// @spotify-internal/encore-web/es/components/HorizontalRule/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { gray30, gray70, spacer24 } from '@spotify-internal/encore-foundation';
import styled, { ThemeProvider } from 'styled-components';
import { rootStyle } from "../../styles";

var isDarkTheme = function isDarkTheme(theme) {
  return theme.horizontalRule === 'dark';
};

var HorizontalRuleElement = styled.hr.withConfig({
  displayName: "HorizontalRule__HorizontalRuleElement",
  componentId: "sc-1v7as93-0"
})(["", ";border:0;border-top:1px solid ", ";margin-bottom:", ";margin-top:", ";"], rootStyle(), function (_ref) {
  var theme = _ref.theme;
  return isDarkTheme(theme) ? gray30 : gray70;
}, spacer24, spacer24);
export function HorizontalRuleComponent(_ref2) {
  var _ref2$theme = _ref2.theme,
      theme = _ref2$theme === void 0 ? {} : _ref2$theme,
      props = _objectWithoutProperties(_ref2, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(HorizontalRuleElement, props));
}
var horizontalRuleHelpers = {
  darkTheme: {
    horizontalRule: 'dark'
  }
};
/**
* **HorizontalRule** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/HorizontalRule) | [Storybook](https://encore-web.spotify.net/?path=/story/components-horizontalrule--default) | [Encore](https://encore.spotify.net/web/components/horizontal-rule) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=3755:6881) |
*
* HorizontalRule is a thin line that is inserted into flow content to denote a break in the content. without introducing a new section as a heading would.
*
* @example
* export const Default = () => <HorizontalRule />;
*
*/

export var HorizontalRule = Object.assign(HorizontalRuleComponent, horizontalRuleHelpers);
export default HorizontalRule;
