// @spotify-internal/ubi-sdk-s4a-web-team-billing/WebTeamBillingEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamBillingEventFactory = exports.createWebTeamBillingEventFactory = void 0;
var SPECIFICATION_ID = "web-team-billing";
var APP = "s4a";
var SPECIFICATION_VERSION = "3.0.2";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebTeamBillingEventFactory(options) {
    return new WebTeamBillingEventFactory(options);
}
exports.createWebTeamBillingEventFactory = createWebTeamBillingEventFactory;
/**
 * edit billing information for your team
 */
var WebTeamBillingEventFactory = /** @class */ (function () {
    function WebTeamBillingEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of BillingInformationEventFactory
    */
    WebTeamBillingEventFactory.prototype.billingInformationFactory = function () {
        return new WebTeamBillingEventFactory.BillingInformationEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of PaymentMethodEventFactory
    */
    WebTeamBillingEventFactory.prototype.paymentMethodFactory = function () {
        return new WebTeamBillingEventFactory.PaymentMethodEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of AddPaymentModalEventFactory
    */
    WebTeamBillingEventFactory.prototype.addPaymentModalFactory = function () {
        return new WebTeamBillingEventFactory.AddPaymentModalEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamBillingEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamBillingEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamBillingEventFactory;
}());
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    /**
     * Create BillingInformationEventFactory class with path
     * @param path to this node
     */
    var BillingInformationEventFactory = /** @class */ (function () {
        function BillingInformationEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "billing_information"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of CountryDropdownEventFactory
        */
        BillingInformationEventFactory.prototype.countryDropdownFactory = function () {
            return new WebTeamBillingEventFactory.BillingInformationEventFactory.CountryDropdownEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ContactEmailDropdownEventFactory
        */
        BillingInformationEventFactory.prototype.contactEmailDropdownFactory = function () {
            return new WebTeamBillingEventFactory.BillingInformationEventFactory.ContactEmailDropdownEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of SaveButtonEventFactory
        */
        BillingInformationEventFactory.prototype.saveButtonFactory = function () {
            return new WebTeamBillingEventFactory.BillingInformationEventFactory.SaveButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        BillingInformationEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        BillingInformationEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return BillingInformationEventFactory;
    }());
    WebTeamBillingEventFactory.BillingInformationEventFactory = BillingInformationEventFactory;
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    /**
     * Create PaymentMethodEventFactory class with path
     * @param path to this node
     */
    var PaymentMethodEventFactory = /** @class */ (function () {
        function PaymentMethodEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "payment_method"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of AddCardButtonEventFactory
        */
        PaymentMethodEventFactory.prototype.addCardButtonFactory = function () {
            return new WebTeamBillingEventFactory.PaymentMethodEventFactory.AddCardButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of UpdateButtonEventFactory
        */
        PaymentMethodEventFactory.prototype.updateButtonFactory = function () {
            return new WebTeamBillingEventFactory.PaymentMethodEventFactory.UpdateButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        PaymentMethodEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        PaymentMethodEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return PaymentMethodEventFactory;
    }());
    WebTeamBillingEventFactory.PaymentMethodEventFactory = PaymentMethodEventFactory;
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    /**
     * Create AddPaymentModalEventFactory class with path
     * @param path to this node
     */
    var AddPaymentModalEventFactory = /** @class */ (function () {
        function AddPaymentModalEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "add_payment_modal"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of CloseButtonEventFactory
        */
        AddPaymentModalEventFactory.prototype.closeButtonFactory = function () {
            return new WebTeamBillingEventFactory.AddPaymentModalEventFactory.CloseButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of UpdateCardButtonEventFactory
        */
        AddPaymentModalEventFactory.prototype.updateCardButtonFactory = function () {
            return new WebTeamBillingEventFactory.AddPaymentModalEventFactory.UpdateCardButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        AddPaymentModalEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AddPaymentModalEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for AddPaymentModalEventFactory.
         */
        AddPaymentModalEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AddPaymentModalEventFactory;
    }());
    WebTeamBillingEventFactory.AddPaymentModalEventFactory = AddPaymentModalEventFactory;
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var BillingInformationEventFactory;
    (function (BillingInformationEventFactory) {
        /**
         * Create CountryDropdownEventFactory class with path
         * @param path to this node
         */
        var CountryDropdownEventFactory = /** @class */ (function () {
            function CountryDropdownEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "country_dropdown"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the CountrySelectionEventFactory
            * @returns A new instance of CountrySelectionEventFactory
            */
            CountryDropdownEventFactory.prototype.countrySelectionFactory = function (data) {
                return new WebTeamBillingEventFactory.BillingInformationEventFactory.CountryDropdownEventFactory.CountrySelectionEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            CountryDropdownEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CountryDropdownEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            CountryDropdownEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CountryDropdownEventFactory;
        }());
        BillingInformationEventFactory.CountryDropdownEventFactory = CountryDropdownEventFactory;
    })(BillingInformationEventFactory = WebTeamBillingEventFactory.BillingInformationEventFactory || (WebTeamBillingEventFactory.BillingInformationEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var BillingInformationEventFactory;
    (function (BillingInformationEventFactory) {
        /**
         * Create ContactEmailDropdownEventFactory class with path
         * @param path to this node
         */
        var ContactEmailDropdownEventFactory = /** @class */ (function () {
            function ContactEmailDropdownEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "contact_email_dropdown"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the UserSelectionEventFactory
            * @returns A new instance of UserSelectionEventFactory
            */
            ContactEmailDropdownEventFactory.prototype.userSelectionFactory = function (data) {
                return new WebTeamBillingEventFactory.BillingInformationEventFactory.ContactEmailDropdownEventFactory.UserSelectionEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            ContactEmailDropdownEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ContactEmailDropdownEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            ContactEmailDropdownEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ContactEmailDropdownEventFactory;
        }());
        BillingInformationEventFactory.ContactEmailDropdownEventFactory = ContactEmailDropdownEventFactory;
    })(BillingInformationEventFactory = WebTeamBillingEventFactory.BillingInformationEventFactory || (WebTeamBillingEventFactory.BillingInformationEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var BillingInformationEventFactory;
    (function (BillingInformationEventFactory) {
        /**
         * Create SaveButtonEventFactory class with path
         * @param path to this node
         */
        var SaveButtonEventFactory = /** @class */ (function () {
            function SaveButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "save_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SaveButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SaveButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit save_billing_info action.
           * @param has_country The value is true when user adds country information or otherwise it is false
           * @param has_email The value is true when user adds email address or otherwise it is false
             */
            SaveButtonEventFactory.prototype.hitSaveBillingInfo = function (_a) {
                var hasCountry = _a.hasCountry, hasEmail = _a.hasEmail;
                var data = {
                    actionParameterNames: ['has_country', 'has_email'],
                    actionParameterValues: [hasCountry.toString(), hasEmail.toString()],
                    actionName: "save_billing_info",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SaveButtonEventFactory;
        }());
        BillingInformationEventFactory.SaveButtonEventFactory = SaveButtonEventFactory;
    })(BillingInformationEventFactory = WebTeamBillingEventFactory.BillingInformationEventFactory || (WebTeamBillingEventFactory.BillingInformationEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var PaymentMethodEventFactory;
    (function (PaymentMethodEventFactory) {
        /**
         * Create AddCardButtonEventFactory class with path
         * @param path to this node
         */
        var AddCardButtonEventFactory = /** @class */ (function () {
            function AddCardButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "add_card_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AddCardButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AddCardButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            AddCardButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create impression for AddCardButtonEventFactory.
             */
            AddCardButtonEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AddCardButtonEventFactory;
        }());
        PaymentMethodEventFactory.AddCardButtonEventFactory = AddCardButtonEventFactory;
    })(PaymentMethodEventFactory = WebTeamBillingEventFactory.PaymentMethodEventFactory || (WebTeamBillingEventFactory.PaymentMethodEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var PaymentMethodEventFactory;
    (function (PaymentMethodEventFactory) {
        /**
         * Create UpdateButtonEventFactory class with path
         * @param path to this node
         */
        var UpdateButtonEventFactory = /** @class */ (function () {
            function UpdateButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "update_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            UpdateButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            UpdateButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            UpdateButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create impression for UpdateButtonEventFactory.
             */
            UpdateButtonEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return UpdateButtonEventFactory;
        }());
        PaymentMethodEventFactory.UpdateButtonEventFactory = UpdateButtonEventFactory;
    })(PaymentMethodEventFactory = WebTeamBillingEventFactory.PaymentMethodEventFactory || (WebTeamBillingEventFactory.PaymentMethodEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var AddPaymentModalEventFactory;
    (function (AddPaymentModalEventFactory) {
        /**
         * Create CloseButtonEventFactory class with path
         * @param path to this node
         */
        var CloseButtonEventFactory = /** @class */ (function () {
            function CloseButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "close_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CloseButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CloseButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_hide action.
             */
            CloseButtonEventFactory.prototype.hitUiHide = function () {
                var data = {
                    actionName: "ui_hide",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CloseButtonEventFactory;
        }());
        AddPaymentModalEventFactory.CloseButtonEventFactory = CloseButtonEventFactory;
    })(AddPaymentModalEventFactory = WebTeamBillingEventFactory.AddPaymentModalEventFactory || (WebTeamBillingEventFactory.AddPaymentModalEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var AddPaymentModalEventFactory;
    (function (AddPaymentModalEventFactory) {
        /**
         * Create UpdateCardButtonEventFactory class with path
         * @param path to this node
         */
        var UpdateCardButtonEventFactory = /** @class */ (function () {
            function UpdateCardButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "update_card_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            UpdateCardButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            UpdateCardButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit edit_credit_card_info action.
           * @param is_new The value is true when adding a new card or false when editing and existing one
           * @param checkout_id The ID of the checkout instance
             */
            UpdateCardButtonEventFactory.prototype.hitEditCreditCardInfo = function (_a) {
                var isNew = _a.isNew, checkoutId = _a.checkoutId;
                var data = {
                    actionParameterNames: ['is_new', 'checkout_id'],
                    actionParameterValues: [isNew.toString(), checkoutId.toString()],
                    actionName: "edit_credit_card_info",
                    actionVersion: 2,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return UpdateCardButtonEventFactory;
        }());
        AddPaymentModalEventFactory.UpdateCardButtonEventFactory = UpdateCardButtonEventFactory;
    })(AddPaymentModalEventFactory = WebTeamBillingEventFactory.AddPaymentModalEventFactory || (WebTeamBillingEventFactory.AddPaymentModalEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var BillingInformationEventFactory;
    (function (BillingInformationEventFactory) {
        var CountryDropdownEventFactory;
        (function (CountryDropdownEventFactory) {
            /**
             * Create CountrySelectionEventFactory class with path
             * @param path to this node
             */
            var CountrySelectionEventFactory = /** @class */ (function () {
                function CountrySelectionEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "country_selection" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                CountrySelectionEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                CountrySelectionEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_select action.
                 */
                CountrySelectionEventFactory.prototype.hitUiSelect = function () {
                    var data = {
                        actionName: "ui_select",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return CountrySelectionEventFactory;
            }());
            CountryDropdownEventFactory.CountrySelectionEventFactory = CountrySelectionEventFactory;
        })(CountryDropdownEventFactory = BillingInformationEventFactory.CountryDropdownEventFactory || (BillingInformationEventFactory.CountryDropdownEventFactory = {}));
    })(BillingInformationEventFactory = WebTeamBillingEventFactory.BillingInformationEventFactory || (WebTeamBillingEventFactory.BillingInformationEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
(function (WebTeamBillingEventFactory) {
    var BillingInformationEventFactory;
    (function (BillingInformationEventFactory) {
        var ContactEmailDropdownEventFactory;
        (function (ContactEmailDropdownEventFactory) {
            /**
             * Create UserSelectionEventFactory class with path
             * @param path to this node
             */
            var UserSelectionEventFactory = /** @class */ (function () {
                function UserSelectionEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "user_selection" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                UserSelectionEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                UserSelectionEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_select action.
                 */
                UserSelectionEventFactory.prototype.hitUiSelect = function () {
                    var data = {
                        actionName: "ui_select",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return UserSelectionEventFactory;
            }());
            ContactEmailDropdownEventFactory.UserSelectionEventFactory = UserSelectionEventFactory;
        })(ContactEmailDropdownEventFactory = BillingInformationEventFactory.ContactEmailDropdownEventFactory || (BillingInformationEventFactory.ContactEmailDropdownEventFactory = {}));
    })(BillingInformationEventFactory = WebTeamBillingEventFactory.BillingInformationEventFactory || (WebTeamBillingEventFactory.BillingInformationEventFactory = {}));
})(WebTeamBillingEventFactory = exports.WebTeamBillingEventFactory || (exports.WebTeamBillingEventFactory = {}));
exports.WebTeamBillingEventFactory = WebTeamBillingEventFactory;
