// @spotify-internal/spoticharts/cjs/components/Legends/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const LegendCircle = styled_components_1.default.div `
  display: inline-block;
  width: 10px;
  height: 10px;
  background-color: ${props => props.color};
  border-radius: 50%;
  margin: 0 8px 0 24px;
  vertical-align: middle;
`;
const LegendLabel = styled_components_1.default.span `
  font-size: 10px;
  font-family: Circular;
  color: ${encore_foundation_1.gray45};
`;
function Legends({ legendsList }) {
    const legendElements = legendsList.map((legends, key) => (react_1.default.createElement("span", { key: key },
        react_1.default.createElement(LegendCircle, { color: legends.color }),
        react_1.default.createElement(LegendLabel, null, legends.label))));
    return react_1.default.createElement("div", null, legendElements);
}
exports.default = Legends;
Legends.propTypes = {
    legendsList: prop_types_1.default.arrayOf(prop_types_1.default.shape({
        label: prop_types_1.default.string,
        color: prop_types_1.default.string,
    })),
};
