// @spotify-internal/adstudio-shared/lib/components/TrackingUrls/TrackingUrlFormField.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _reduxForm = require("redux-form");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _ReduxFormFieldHOC = _interopRequireDefault(require("../ReduxFormFieldHOC"));

var _trackingConstants = _interopRequireDefault(require("./tracking.constants.js"));

var _validators = require("@spotify-internal/adstudio-web-utils/lib/validators");

var _macros = _interopRequireDefault(require("../../config/macros"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var IconX = _adstudioTape.Icons.IconX;

var AddTrackingUrl = _styledComponents["default"].a.withConfig({
  displayName: "TrackingUrlFormField__AddTrackingUrl",
  componentId: "s59bbg-0"
})(["color:", ";&:hover{opacity:0.4;}"], _encoreAdvertisingWeb.azure);

var TrackingList = _styledComponents["default"].ul.withConfig({
  displayName: "TrackingUrlFormField__TrackingList",
  componentId: "s59bbg-1"
})(["list-style:none;"]);

var TrackingRow = _styledComponents["default"].li.withConfig({
  displayName: "TrackingUrlFormField__TrackingRow",
  componentId: "s59bbg-2"
})(["display:flex;flex-direction:row;padding-bottom:", ";"], _encoreAdvertisingWeb.spacer12);

var TrackingCol = _styledComponents["default"].div.withConfig({
  displayName: "TrackingUrlFormField__TrackingCol",
  componentId: "s59bbg-3"
})(["flex:auto 1;"]);

var TrackingRemove = _styledComponents["default"].a.withConfig({
  displayName: "TrackingUrlFormField__TrackingRemove",
  componentId: "s59bbg-4"
})(["color:", ";display:inline-block;margin-left:", ";position:absolute;top:", ";&:hover{opacity:0.4;}"], _encoreAdvertisingWeb.gray60, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.spacer16);

var TrackingRemoveContainer = _styledComponents["default"].div.withConfig({
  displayName: "TrackingUrlFormField__TrackingRemoveContainer",
  componentId: "s59bbg-5"
})(["position:relative;"]);

var checkRequired = function checkRequired() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      key = _ref.key,
      value = _ref.value;

  var requiredType = (0, _validators.required)(key) ? "Please select an event type." : '';
  var requiredURL = (0, _validators.required)(value) ? "Please enter a URL" : '';
  return requiredType || requiredURL;
};

var replacer = function replacer() {
  var str = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '';
  return _macros["default"].reduce(function (accum, item) {
    return accum.replace(item[0], item[1]);
  }, str);
};

var checkUrl = function checkUrl() {
  var obj = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return (0, _validators.url)(obj.value);
};

var TrackingField = (0, _ReduxFormFieldHOC["default"])(function (props) {
  var str = replacer(props.value.value);

  var value = _objectSpread(_objectSpread({}, props.value), {}, {
    value: str
  });

  return /*#__PURE__*/_react["default"].createElement(_adstudioTape.DropdownTextarea, _extends({}, props, {
    options: _trackingConstants["default"],
    selectPlaceholder: 'Please select an event type',
    textareaPlaceholder: 'Enter URL',
    value: value
  }));
});

var TrackingUrlFormField = function TrackingUrlFormField(_ref2) {
  var fields = _ref2.fields;
  return /*#__PURE__*/_react["default"].createElement(TrackingList, null, fields.map(function (member, index) {
    return /*#__PURE__*/_react["default"].createElement(TrackingRow, {
      key: index
    }, /*#__PURE__*/_react["default"].createElement(TrackingCol, null, /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
      component: TrackingField,
      flush: true,
      name: member,
      validate: [checkRequired, checkUrl]
    })), /*#__PURE__*/_react["default"].createElement(TrackingRemoveContainer, null, /*#__PURE__*/_react["default"].createElement(TrackingRemove, {
      onClick: function onClick() {
        return fields.remove(index);
      }
    }, /*#__PURE__*/_react["default"].createElement(IconX, null))));
  }), /*#__PURE__*/_react["default"].createElement("li", null, /*#__PURE__*/_react["default"].createElement(AddTrackingUrl, {
    onClick: function onClick() {
      return fields.push();
    },
    type: "button"
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type, {
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold
  }, "+ ADD"))));
};

TrackingUrlFormField.propTypes = {
  fields: _propTypes["default"].object
};
var _default = TrackingUrlFormField;
exports["default"] = _default;
