// @spotify-internal/spoticharts/cjs/components/ChartWrapper/ErrorBoundaryChart.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const isFunction_1 = __importDefault(require("lodash/isFunction"));
const reactochart_1 = require("reactochart");
class ErrorBoundaryChart extends react_1.Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            error: null,
        };
    }
    componentDidCatch(error) {
        this.setState({
            hasError: true,
            error,
        });
        if (isFunction_1.default(this.props.onError)) {
            this.props.onError(error);
        }
    }
    render() {
        if (this.state.hasError) {
            return (react_1.default.createElement(reactochart_1.XYPlot, Object.assign({ xDomain: [0, 10], yDomain: [0, 5, 100] }, this.props),
                react_1.default.createElement(reactochart_1.YGrid, { tickCount: 2 })));
        }
        return this.props.children;
    }
}
exports.default = ErrorBoundaryChart;
ErrorBoundaryChart.propTypes = {
    children: prop_types_1.default.node,
    onError: prop_types_1.default.func,
};
