// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/AudienceRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AudienceRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.AudienceRequest = (function() {
                    /**
                     * Properties of a AudienceRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IAudienceRequest
                     * @property {number|null} [ageMin] AudienceRequest ageMin
                     * @property {number|null} [ageMax] AudienceRequest ageMax
                     * @property {Array.<string>|null} [genders] AudienceRequest genders
                     * @property {Array.<string>|null} [artistIds] AudienceRequest artistIds
                     * @property {Array.<string>|null} [genres] AudienceRequest genres
                     * @property {Array.<string>|null} [playlists] AudienceRequest playlists
                     * @property {Array.<string>|null} [platforms] AudienceRequest platforms
                     * @property {string|null} [dateBegin] AudienceRequest dateBegin
                     * @property {string|null} [dateEnd] AudienceRequest dateEnd
                     * @property {com.spotify.adstudiobff.proto.PricingRequest|null} [pricing] AudienceRequest pricing
                     * @property {number|null} [budgetMicros] AudienceRequest budgetMicros
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [geos] AudienceRequest geos
                     * @property {string|null} [currencyCode] AudienceRequest currencyCode
                     * @property {Array.<string>|null} [audienceSegmentIds] AudienceRequest audienceSegmentIds
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [format] AudienceRequest format
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [frequencyCaps] AudienceRequest frequencyCaps
                     * @property {Array.<string>|null} [audienceMatchIds] AudienceRequest audienceMatchIds
                     * @property {Array.<com.spotify.adstudiobff.proto.AgeRange>|null} [ageRanges] AudienceRequest ageRanges
                     * @property {Array.<string>|null} [brandSafetyCategories] AudienceRequest brandSafetyCategories
                     * @property {Array.<string>|null} [contextualTargetingCategories] AudienceRequest contextualTargetingCategories
                     * @property {number|null} [setAgeMin] AudienceRequest setAgeMin
                     * @property {number|null} [setAgeMax] AudienceRequest setAgeMax
                     * @property {Array.<string>|null} [setGenders] AudienceRequest setGenders
                     * @property {Array.<string>|null} [setArtistIds] AudienceRequest setArtistIds
                     * @property {Array.<string>|null} [setGenres] AudienceRequest setGenres
                     * @property {Array.<string>|null} [setPlaylists] AudienceRequest setPlaylists
                     * @property {Array.<string>|null} [setPlatforms] AudienceRequest setPlatforms
                     * @property {string|null} [setDateBegin] AudienceRequest setDateBegin
                     * @property {string|null} [setDateEnd] AudienceRequest setDateEnd
                     * @property {com.spotify.adstudiobff.proto.PricingRequest|null} [setPricing] AudienceRequest setPricing
                     * @property {number|null} [setBudgetMicros] AudienceRequest setBudgetMicros
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [setGeos] AudienceRequest setGeos
                     * @property {string|null} [setCurrencyCode] AudienceRequest setCurrencyCode
                     * @property {Array.<string>|null} [setAudienceSegmentIds] AudienceRequest setAudienceSegmentIds
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormat] AudienceRequest setFormat
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [setFrequencyCaps] AudienceRequest setFrequencyCaps
                     * @property {Array.<string>|null} [setAudienceMatchIds] AudienceRequest setAudienceMatchIds
                     * @property {Array.<com.spotify.adstudiobff.proto.AgeRange>|null} [setAgeRanges] AudienceRequest setAgeRanges
                     * @property {Array.<string>|null} [setBrandSafetyCategories] AudienceRequest setBrandSafetyCategories
                     * @property {Array.<string>|null} [setContextualTargetingCategories] AudienceRequest setContextualTargetingCategories
                     */
                    
                    /**
                     * Constructs a new AudienceRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a AudienceRequest.
                     * @implements IAudienceRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IAudienceRequest=} [properties] Properties to set
                     */
                    function AudienceRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * AudienceRequest ageMin.
                     * @member {number} ageMin
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.ageMin = 0;

                    /**
                     * AudienceRequest ageMax.
                     * @member {number} ageMax
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.ageMax = 0;

                    /**
                     * AudienceRequest genders.
                     * @member {Array.<string>} genders
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.genders = [];

                    /**
                     * AudienceRequest artistIds.
                     * @member {Array.<string>} artistIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.artistIds = [];

                    /**
                     * AudienceRequest genres.
                     * @member {Array.<string>} genres
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.genres = [];

                    /**
                     * AudienceRequest playlists.
                     * @member {Array.<string>} playlists
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.playlists = [];

                    /**
                     * AudienceRequest platforms.
                     * @member {Array.<string>} platforms
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.platforms = [];

                    /**
                     * AudienceRequest dateBegin.
                     * @member {string} dateBegin
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.dateBegin = "";

                    /**
                     * AudienceRequest dateEnd.
                     * @member {string} dateEnd
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.dateEnd = "";

                    /**
                     * AudienceRequest pricing.
                     * @member {com.spotify.adstudiobff.proto.PricingRequest} pricing
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.pricing = {};

                    /**
                     * AudienceRequest budgetMicros.
                     * @member {number} budgetMicros
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.budgetMicros = 0;

                    /**
                     * AudienceRequest geos.
                     * @member {Array.<com.spotify.adstudiobff.proto.Geo>} geos
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.geos = [];

                    /**
                     * AudienceRequest currencyCode.
                     * @member {string} currencyCode
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.currencyCode = "";

                    /**
                     * AudienceRequest audienceSegmentIds.
                     * @member {Array.<string>} audienceSegmentIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.audienceSegmentIds = [];

                    /**
                     * AudienceRequest format.
                     * @member {com.spotify.ads.cs.proto.FormatType} format
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.format = {};

                    /**
                     * AudienceRequest frequencyCaps.
                     * @member {Array.<com.spotify.ads.campaignservice.FrequencyCap>} frequencyCaps
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.frequencyCaps = [];

                    /**
                     * AudienceRequest audienceMatchIds.
                     * @member {Array.<string>} audienceMatchIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.audienceMatchIds = [];

                    /**
                     * AudienceRequest ageRanges.
                     * @member {Array.<com.spotify.adstudiobff.proto.AgeRange>} ageRanges
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.ageRanges = [];

                    /**
                     * AudienceRequest brandSafetyCategories.
                     * @member {Array.<string>} brandSafetyCategories
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.brandSafetyCategories = [];

                    /**
                     * AudienceRequest contextualTargetingCategories.
                     * @member {Array.<string>} contextualTargetingCategories
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     */
                    AudienceRequest.prototype.contextualTargetingCategories = [];


                    /**
                     * Sets ageMin
                     * @function setAgeMin
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setAgeMin = function(value) {
                        this['ageMin'] = value;
                        return this;
                    };

                    /**
                     * Sets ageMax
                     * @function setAgeMax
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setAgeMax = function(value) {
                        this['ageMax'] = value;
                        return this;
                    };

                    /**
                     * Sets genders
                     * @function setGenders
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setGenders = function(value) {
                        this['genders'] = value;
                        return this;
                    };

                    /**
                     * Sets artistIds
                     * @function setArtistIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setArtistIds = function(value) {
                        this['artistIds'] = value;
                        return this;
                    };

                    /**
                     * Sets genres
                     * @function setGenres
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setGenres = function(value) {
                        this['genres'] = value;
                        return this;
                    };

                    /**
                     * Sets playlists
                     * @function setPlaylists
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setPlaylists = function(value) {
                        this['playlists'] = value;
                        return this;
                    };

                    /**
                     * Sets platforms
                     * @function setPlatforms
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setPlatforms = function(value) {
                        this['platforms'] = value;
                        return this;
                    };

                    /**
                     * Sets dateBegin
                     * @function setDateBegin
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setDateBegin = function(value) {
                        this['dateBegin'] = value;
                        return this;
                    };

                    /**
                     * Sets dateEnd
                     * @function setDateEnd
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setDateEnd = function(value) {
                        this['dateEnd'] = value;
                        return this;
                    };

                    /**
                     * Sets pricing
                     * @function setPricing
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {com.spotify.adstudiobff.proto.PricingRequest} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setPricing = function(value) {
                        this['pricing'] = value;
                        return this;
                    };

                    /**
                     * Sets budgetMicros
                     * @function setBudgetMicros
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setBudgetMicros = function(value) {
                        this['budgetMicros'] = value;
                        return this;
                    };

                    /**
                     * Sets geos
                     * @function setGeos
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<com.spotify.adstudiobff.proto.Geo>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setGeos = function(value) {
                        this['geos'] = value;
                        return this;
                    };

                    /**
                     * Sets currencyCode
                     * @function setCurrencyCode
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setCurrencyCode = function(value) {
                        this['currencyCode'] = value;
                        return this;
                    };

                    /**
                     * Sets audienceSegmentIds
                     * @function setAudienceSegmentIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setAudienceSegmentIds = function(value) {
                        this['audienceSegmentIds'] = value;
                        return this;
                    };

                    /**
                     * Sets format
                     * @function setFormat
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setFormat = function(value) {
                        this['format'] = value;
                        return this;
                    };

                    /**
                     * Sets frequencyCaps
                     * @function setFrequencyCaps
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<com.spotify.ads.campaignservice.FrequencyCap>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setFrequencyCaps = function(value) {
                        this['frequencyCaps'] = value;
                        return this;
                    };

                    /**
                     * Sets audienceMatchIds
                     * @function setAudienceMatchIds
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setAudienceMatchIds = function(value) {
                        this['audienceMatchIds'] = value;
                        return this;
                    };

                    /**
                     * Sets ageRanges
                     * @function setAgeRanges
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<com.spotify.adstudiobff.proto.AgeRange>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setAgeRanges = function(value) {
                        this['ageRanges'] = value;
                        return this;
                    };

                    /**
                     * Sets brandSafetyCategories
                     * @function setBrandSafetyCategories
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setBrandSafetyCategories = function(value) {
                        this['brandSafetyCategories'] = value;
                        return this;
                    };

                    /**
                     * Sets contextualTargetingCategories
                     * @function setContextualTargetingCategories
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AudienceRequest} AudienceRequest
                     */
                    AudienceRequest.prototype.setContextualTargetingCategories = function(value) {
                        this['contextualTargetingCategories'] = value;
                        return this;
                    };

                    /**
                     * Converts the AudienceRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.AudienceRequest
                     * @instance
                     * @returns {object} object
                     */
                    AudienceRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return AudienceRequest;
                })();
                
                AudienceRequest = proto.AudienceRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["AudienceRequest"] = AudienceRequest;
exports["default"] = $root;
