// @spotify-internal/address-collection-sdk/esm/components/AddressConfirmationModal/AddressConfirmationModal.js
import React from 'react';
import { Backdrop, ButtonPrimary, ButtonSecondary, DialogConfirmation, Type, } from '@spotify-internal/encore-web';
import styled from 'styled-components';
import trackers from '../../lib/tracking/trackers';
const SuggestedAddress = styled.div `
  border: 1px solid #5892da;
  background-color: aliceblue;
  padding: 10px;
  border-radius: 10px;
  margin-top: 12px;
`;
const AddressConfirmationModal = ({ onClose, userAddress, validatedZipCode, country, submitConfirmation, }) => {
    const validatedAddress = {
        street: userAddress.street,
        state: userAddress.state,
        city: userAddress.city,
        postal_code_short: validatedZipCode,
    };
    return (React.createElement(Backdrop, { center: true, onClose: onClose },
        React.createElement(DialogConfirmation, { body: React.createElement(React.Fragment, null,
                React.createElement(Type, { as: "h1", variant: Type.heading3 }, "Confirm your address"),
                React.createElement(Type, { as: "p", variant: Type.body1 }, "The address you entered isn\u2019t valid or may be incomplete. Check your details and try again."),
                React.createElement("div", null,
                    React.createElement(Type, { as: "h2", variant: Type.heading4, weight: "bold", style: { padding: '8px 0' } }, "You entered:"),
                    React.createElement(Type, { as: "p", condensed: true }, userAddress.street),
                    React.createElement(Type, { as: "p", condensed: true },
                        userAddress.city,
                        ", ",
                        userAddress.state,
                        ' ',
                        userAddress.postal_code_short),
                    React.createElement(Type, { as: "p", condensed: true }, country)),
                React.createElement(SuggestedAddress, null,
                    React.createElement(Type, { as: "h2", variant: Type.heading4, weight: "bold", style: { padding: '8px 0' } }, "Suggested address:"),
                    React.createElement(Type, { as: "p", condensed: true }, validatedAddress.street),
                    React.createElement(Type, { as: "p", condensed: true },
                        validatedAddress.city,
                        ", ",
                        validatedAddress.state,
                        ' ',
                        React.createElement(Type, { as: "span", weight: "bold" }, validatedAddress.postal_code_short)),
                    React.createElement(Type, { as: "p", condensed: true }, country))), footer: React.createElement(React.Fragment, null,
                React.createElement(ButtonSecondary, { onClick: onClose }, "Edit Address"),
                React.createElement(ButtonPrimary, { onClick: () => {
                        submitConfirmation(validatedAddress);
                        trackers.trackAddressCollectionConfirmIntent();
                    } }, "Confirm Address")) })));
};
export default AddressConfirmationModal;
