// @spotify-internal/encore-web/es/components/ProgressBar/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { green } from '@spotify-internal/encore-foundation';
import { ThemeProvider } from 'styled-components';
import { ProgressBarBackground } from "./ProgressBarBackground";
import { ProgressBarIndicator } from "./ProgressBarIndicator";

function ProgressBarComponent(_ref) {
  var _ref$value = _ref.value,
      value = _ref$value === void 0 ? 0 : _ref$value,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? green : _ref$color,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["value", "color", "theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(ProgressBarBackground, Object.assign({
    role: "progressbar",
    "aria-valuenow": value,
    "aria-valuemin": "0",
    "aria-valuemax": "100"
  }, props), /*#__PURE__*/React.createElement(ProgressBarIndicator, {
    progressValue: value,
    progressColor: color
  })));
}

var progressBarHelpers = {
  darkTheme: {
    progressBar: 'dark'
  }
};
/**
* **ProgressBar** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/ProgressBar) | [Storybook](https://encore-web.spotify.net/?path=/story/components-progressbar--default) | [Encore](https://encore.spotify.net/web/components/indicators) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1082) |
*
* A modal used to display some contextual information on top of other content when users click on, keypress on, or touch an element.
*
* @example
* export const Default = () => <ProgressBar value={75} />;
*
*export const CustomColor = () => <ProgressBar color={brightRed} value={50} />;
*
*export const ColumnWithLabels = () => {
*  const stages = [
*    {
*      label: 'Research',
*
*/

export var ProgressBar = Object.assign(ProgressBarComponent, progressBarHelpers);
export default ProgressBar;
