// @spotify-internal/encore-web/es/components/TablePagination/index.js
import { gray50, screenSmMin, screenXsMax, spacer4, spacer12 } from '@spotify-internal/encore-foundation';
import styled from 'styled-components';
import { Container as Controls } from "../PaginationControls/Container";
var TablePaginationComponent = styled.div.withConfig({
  displayName: "TablePagination__TablePaginationComponent",
  componentId: "cqwa4n-0"
})(["color:", ";display:flex;align-items:center;@media (max-width:", "){padding:", " ", ";}@media (min-width:", "){padding:", ";}", "{flex:1;display:flex;justify-content:flex-end;}"], gray50, screenXsMax, spacer12, spacer4, screenSmMin, spacer12, Controls);
var tablePaginationHelpers = {
  darkTheme: {
    tablePagination: 'dark'
  }
};
/**
* **TablePagination** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/TablePagination) | [Storybook](https://encore-web.spotify.net/?path=/story/components-table-tablepagination--default) | [Encore](https://encore.spotify.net/web/components/table?format=figma-figma#pagination) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=613:568) |
*
* TablePagination adds Pagination controls to your table.
*
* @example
* export const Default = () => (
*  <TablePagination>
*    <Type>Rows per page:</Type>
*    <PaginationDropdownTrigger>
*      <PaginationDropdown>5</PaginationDropdown>
*    </PaginationDropdownTrigger>
*    <PaginationControls onDecrement={() => {}} onIncrement={() => {}}>
*      5-10 of 100
*    </PaginationControls>
*  </TablePagination>
*);
*
*/

export var TablePagination = Object.assign(TablePaginationComponent, tablePaginationHelpers);
export default TablePagination;
