// @spotify-internal/spoticharts/cjs/components/LineChartMouseOverlay/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const LineChartCollisionBox_1 = __importDefault(require("../LineChartCollisionBox"));
class LineChartMouseOverlay extends react_1.Component {
    render() {
        const _a = this.props, { data, x, y, onMouseEnterContainer, onMouseLeaveContainer } = _a, rest = __rest(_a, ["data", "x", "y", "onMouseEnterContainer", "onMouseLeaveContainer"]);
        return (react_1.default.createElement(react_1.Fragment, null, data.map((d, i) => {
            return (react_1.default.createElement(LineChartCollisionBox_1.default, Object.assign({ key: `colision-box-${i}`, data: d, x: x, y: y }, rest)));
        })));
    }
}
exports.default = LineChartMouseOverlay;
LineChartMouseOverlay.propTypes = {
    data: prop_types_1.default.array.isRequired,
    x: prop_types_1.default.func.isRequired,
    y: prop_types_1.default.func.isRequired,
    xScale: prop_types_1.default.func.isRequired,
    yScale: prop_types_1.default.func.isRequired,
    width: prop_types_1.default.number,
    height: prop_types_1.default.number,
    paddingLeft: prop_types_1.default.number,
    paddingRight: prop_types_1.default.number,
    onMouseEnterContainer: prop_types_1.default.func,
    onMouseLeaveContainer: prop_types_1.default.func,
    onMouseEnter: prop_types_1.default.func.isRequired,
    onMouseLeave: prop_types_1.default.func.isRequired,
};
