// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/CountrySearch/Value/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.CountryValue = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var CountryValue = function CountryValue(props) {
  var _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      item = props.item,
      onClick = props.onClick;
  return /*#__PURE__*/_react["default"].createElement(_Item["default"], {
    country: item,
    disabled: disabled,
    onClick: onClick
  });
};

exports.CountryValue = CountryValue;
CountryValue.propTypes = {
  disabled: _propTypes["default"].bool,
  onClick: _propTypes["default"].func,
  item: _propTypes["default"].object.isRequired,
  keyField: _propTypes["default"].string,
  valueField: _propTypes["default"].string
};
var _default = CountryValue;
exports["default"] = _default;
