// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/CountrySearch/Value/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StyledButtonIcon = (0, _styledComponents["default"])(_encoreAdvertisingWeb.ButtonIcon).withConfig({
  displayName: "Item__StyledButtonIcon",
  componentId: "sc-16z2ylq-0"
})(["display:grid;opacity:0;"]);
var StyledInput = (0, _styledComponents["default"])(_encoreAdvertisingWeb.FormInputIcon).withConfig({
  displayName: "Item__StyledInput",
  componentId: "sc-16z2ylq-1"
})(["input{color:inherit !important;background-color:inherit !important;}:hover{", "{opacity:1;}}"], StyledButtonIcon);

var SelectedCountryValueItem = function SelectedCountryValueItem(props) {
  var country = props.country,
      disabled = props.disabled,
      onClick = props.onClick;
  return /*#__PURE__*/_react["default"].createElement(StyledInput, {
    iconTrailing: !disabled && /*#__PURE__*/_react["default"].createElement(StyledButtonIcon, {
      onClick: onClick
    }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconX, null))
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.FormInput, {
    readOnly: true,
    value: country.name
  }));
};

SelectedCountryValueItem.propTypes = {
  disabled: _propTypes["default"].bool,
  country: _propTypes["default"].object.isRequired,
  onClick: _propTypes["default"].func
};
var _default = SelectedCountryValueItem;
exports["default"] = _default;
