// @spotify-internal/ubi-sdk-s4a-web-team/WebTeamEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamEventFactory = exports.createWebTeamEventFactory = void 0;
var SPECIFICATION_ID = "web-team";
var APP = "s4a";
var SPECIFICATION_VERSION = "2.0.1";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebTeamEventFactory(options) {
    return new WebTeamEventFactory(options);
}
exports.createWebTeamEventFactory = createWebTeamEventFactory;
/**
 * common components across the team pages
 */
var WebTeamEventFactory = /** @class */ (function () {
    function WebTeamEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of BreadcrumbLinkEventFactory
    */
    WebTeamEventFactory.prototype.breadcrumbLinkFactory = function () {
        return new WebTeamEventFactory.BreadcrumbLinkEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of AccessLevelPopoverTriggerEventFactory
    */
    WebTeamEventFactory.prototype.accessLevelPopoverTriggerFactory = function () {
        return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of TeamPageTabsEventFactory
    */
    WebTeamEventFactory.prototype.teamPageTabsFactory = function () {
        return new WebTeamEventFactory.TeamPageTabsEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamEventFactory;
}());
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    /**
     * Create BreadcrumbLinkEventFactory class with path
     * @param path to this node
     */
    var BreadcrumbLinkEventFactory = /** @class */ (function () {
        function BreadcrumbLinkEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "breadcrumb_link"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        BreadcrumbLinkEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        BreadcrumbLinkEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        BreadcrumbLinkEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return BreadcrumbLinkEventFactory;
    }());
    WebTeamEventFactory.BreadcrumbLinkEventFactory = BreadcrumbLinkEventFactory;
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    /**
     * Create AccessLevelPopoverTriggerEventFactory class with path
     * @param path to this node
     */
    var AccessLevelPopoverTriggerEventFactory = /** @class */ (function () {
        function AccessLevelPopoverTriggerEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "access_level_popover_trigger"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of AccessLevelPopoverEventFactory
        */
        AccessLevelPopoverTriggerEventFactory.prototype.accessLevelPopoverFactory = function () {
            return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        AccessLevelPopoverTriggerEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AccessLevelPopoverTriggerEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        AccessLevelPopoverTriggerEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AccessLevelPopoverTriggerEventFactory;
    }());
    WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = AccessLevelPopoverTriggerEventFactory;
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    /**
     * Create TeamPageTabsEventFactory class with path
     * @param path to this node
     */
    var TeamPageTabsEventFactory = /** @class */ (function () {
        function TeamPageTabsEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "team_page_tabs"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of TeamMembersTabEventFactory
        */
        TeamPageTabsEventFactory.prototype.teamMembersTabFactory = function () {
            return new WebTeamEventFactory.TeamPageTabsEventFactory.TeamMembersTabEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ActivityTabEventFactory
        */
        TeamPageTabsEventFactory.prototype.activityTabFactory = function () {
            return new WebTeamEventFactory.TeamPageTabsEventFactory.ActivityTabEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of BillingTabEventFactory
        */
        TeamPageTabsEventFactory.prototype.billingTabFactory = function () {
            return new WebTeamEventFactory.TeamPageTabsEventFactory.BillingTabEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        TeamPageTabsEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TeamPageTabsEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return TeamPageTabsEventFactory;
    }());
    WebTeamEventFactory.TeamPageTabsEventFactory = TeamPageTabsEventFactory;
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        /**
         * Create AccessLevelPopoverEventFactory class with path
         * @param path to this node
         */
        var AccessLevelPopoverEventFactory = /** @class */ (function () {
            function AccessLevelPopoverEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "access_level_popover"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of LeaveTeamLinkEventFactory
            */
            AccessLevelPopoverEventFactory.prototype.leaveTeamLinkFactory = function () {
                return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            AccessLevelPopoverEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AccessLevelPopoverEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for AccessLevelPopoverEventFactory.
             */
            AccessLevelPopoverEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AccessLevelPopoverEventFactory;
        }());
        AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = AccessLevelPopoverEventFactory;
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var TeamPageTabsEventFactory;
    (function (TeamPageTabsEventFactory) {
        /**
         * Create TeamMembersTabEventFactory class with path
         * @param path to this node
         */
        var TeamMembersTabEventFactory = /** @class */ (function () {
            function TeamMembersTabEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "team_members_tab"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TeamMembersTabEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TeamMembersTabEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            TeamMembersTabEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TeamMembersTabEventFactory;
        }());
        TeamPageTabsEventFactory.TeamMembersTabEventFactory = TeamMembersTabEventFactory;
    })(TeamPageTabsEventFactory = WebTeamEventFactory.TeamPageTabsEventFactory || (WebTeamEventFactory.TeamPageTabsEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var TeamPageTabsEventFactory;
    (function (TeamPageTabsEventFactory) {
        /**
         * Create ActivityTabEventFactory class with path
         * @param path to this node
         */
        var ActivityTabEventFactory = /** @class */ (function () {
            function ActivityTabEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "activity_tab"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ActivityTabEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ActivityTabEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            ActivityTabEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ActivityTabEventFactory;
        }());
        TeamPageTabsEventFactory.ActivityTabEventFactory = ActivityTabEventFactory;
    })(TeamPageTabsEventFactory = WebTeamEventFactory.TeamPageTabsEventFactory || (WebTeamEventFactory.TeamPageTabsEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var TeamPageTabsEventFactory;
    (function (TeamPageTabsEventFactory) {
        /**
         * Create BillingTabEventFactory class with path
         * @param path to this node
         */
        var BillingTabEventFactory = /** @class */ (function () {
            function BillingTabEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "billing_tab"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            BillingTabEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            BillingTabEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            BillingTabEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return BillingTabEventFactory;
        }());
        TeamPageTabsEventFactory.BillingTabEventFactory = BillingTabEventFactory;
    })(TeamPageTabsEventFactory = WebTeamEventFactory.TeamPageTabsEventFactory || (WebTeamEventFactory.TeamPageTabsEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            /**
             * Create LeaveTeamLinkEventFactory class with path
             * @param path to this node
             */
            var LeaveTeamLinkEventFactory = /** @class */ (function () {
                function LeaveTeamLinkEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "leave_team_link"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                /**
                * @param id of the LeaveTeamDialogEventFactory
                * @returns A new instance of LeaveTeamDialogEventFactory
                */
                LeaveTeamLinkEventFactory.prototype.leaveTeamDialogFactory = function (data) {
                    return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation,
                        data: data
                    });
                };
                LeaveTeamLinkEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                LeaveTeamLinkEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                LeaveTeamLinkEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return LeaveTeamLinkEventFactory;
            }());
            AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = LeaveTeamLinkEventFactory;
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                /**
                 * Create LeaveTeamDialogEventFactory class with path
                 * @param path to this node
                 */
                var LeaveTeamDialogEventFactory = /** @class */ (function () {
                    function LeaveTeamDialogEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat(__assign(__assign({}, data), { name: "leave_team_dialog" }));
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    /**
                    * @returns A new instance of LeaveTeamConfirmButtonEventFactory
                    */
                    LeaveTeamDialogEventFactory.prototype.leaveTeamConfirmButtonFactory = function () {
                        return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.LeaveTeamConfirmButtonEventFactory({
                            path: this._path,
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        });
                    };
                    /**
                    * @returns A new instance of CancelButtonEventFactory
                    */
                    LeaveTeamDialogEventFactory.prototype.cancelButtonFactory = function () {
                        return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.CancelButtonEventFactory({
                            path: this._path,
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        });
                    };
                    /**
                    * @returns A new instance of GotItButtonEventFactory
                    */
                    LeaveTeamDialogEventFactory.prototype.gotItButtonFactory = function () {
                        return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.GotItButtonEventFactory({
                            path: this._path,
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        });
                    };
                    /**
                    * @returns A new instance of BillingContactDropdownEventFactory
                    */
                    LeaveTeamDialogEventFactory.prototype.billingContactDropdownFactory = function () {
                        return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory({
                            path: this._path,
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        });
                    };
                    LeaveTeamDialogEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    LeaveTeamDialogEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    return LeaveTeamDialogEventFactory;
                }());
                LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = LeaveTeamDialogEventFactory;
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    /**
                     * Create LeaveTeamConfirmButtonEventFactory class with path
                     * @param path to this node
                     */
                    var LeaveTeamConfirmButtonEventFactory = /** @class */ (function () {
                        function LeaveTeamConfirmButtonEventFactory(_a) {
                            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                            this._path = path.concat({
                                name: "leave_team_confirm_button"
                            });
                            if (parentAbsoluteLocation) {
                                this._parentAbsoluteLocation = parentAbsoluteLocation;
                            }
                        }
                        LeaveTeamConfirmButtonEventFactory.prototype._getCurrentLocation = function () {
                            return {
                                pathNodes: this._path,
                                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                            };
                        };
                        LeaveTeamConfirmButtonEventFactory.prototype.getAbsoluteLocation = function () {
                            var _a;
                            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                            return {
                                locations: __spreadArrays((parentLocations || []), [
                                    this._getCurrentLocation()
                                ]),
                            };
                        };
                        /**
                         * Create interaction for hit remove_team_member action.
                         */
                        LeaveTeamConfirmButtonEventFactory.prototype.hitRemoveTeamMember = function () {
                            var data = {
                                actionName: "remove_team_member",
                                actionVersion: 1,
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                interactionType: "hit",
                                specificationMode: SPECIFICATION_MODE,
                                specificationVersion: SPECIFICATION_VERSION,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        return LeaveTeamConfirmButtonEventFactory;
                    }());
                    LeaveTeamDialogEventFactory.LeaveTeamConfirmButtonEventFactory = LeaveTeamConfirmButtonEventFactory;
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    /**
                     * Create CancelButtonEventFactory class with path
                     * @param path to this node
                     */
                    var CancelButtonEventFactory = /** @class */ (function () {
                        function CancelButtonEventFactory(_a) {
                            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                            this._path = path.concat({
                                name: "cancel_button"
                            });
                            if (parentAbsoluteLocation) {
                                this._parentAbsoluteLocation = parentAbsoluteLocation;
                            }
                        }
                        CancelButtonEventFactory.prototype._getCurrentLocation = function () {
                            return {
                                pathNodes: this._path,
                                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                            };
                        };
                        CancelButtonEventFactory.prototype.getAbsoluteLocation = function () {
                            var _a;
                            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                            return {
                                locations: __spreadArrays((parentLocations || []), [
                                    this._getCurrentLocation()
                                ]),
                            };
                        };
                        /**
                         * Create interaction for hit ui_hide action.
                         */
                        CancelButtonEventFactory.prototype.hitUiHide = function () {
                            var data = {
                                actionName: "ui_hide",
                                actionVersion: 1,
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                interactionType: "hit",
                                specificationMode: SPECIFICATION_MODE,
                                specificationVersion: SPECIFICATION_VERSION,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        return CancelButtonEventFactory;
                    }());
                    LeaveTeamDialogEventFactory.CancelButtonEventFactory = CancelButtonEventFactory;
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    /**
                     * Create GotItButtonEventFactory class with path
                     * @param path to this node
                     */
                    var GotItButtonEventFactory = /** @class */ (function () {
                        function GotItButtonEventFactory(_a) {
                            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                            this._path = path.concat({
                                name: "got_it_button"
                            });
                            if (parentAbsoluteLocation) {
                                this._parentAbsoluteLocation = parentAbsoluteLocation;
                            }
                        }
                        GotItButtonEventFactory.prototype._getCurrentLocation = function () {
                            return {
                                pathNodes: this._path,
                                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                            };
                        };
                        GotItButtonEventFactory.prototype.getAbsoluteLocation = function () {
                            var _a;
                            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                            return {
                                locations: __spreadArrays((parentLocations || []), [
                                    this._getCurrentLocation()
                                ]),
                            };
                        };
                        /**
                         * Create interaction for hit ui_hide action.
                         */
                        GotItButtonEventFactory.prototype.hitUiHide = function () {
                            var data = {
                                actionName: "ui_hide",
                                actionVersion: 1,
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                interactionType: "hit",
                                specificationMode: SPECIFICATION_MODE,
                                specificationVersion: SPECIFICATION_VERSION,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        return GotItButtonEventFactory;
                    }());
                    LeaveTeamDialogEventFactory.GotItButtonEventFactory = GotItButtonEventFactory;
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    /**
                     * Create BillingContactDropdownEventFactory class with path
                     * @param path to this node
                     */
                    var BillingContactDropdownEventFactory = /** @class */ (function () {
                        function BillingContactDropdownEventFactory(_a) {
                            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                            this._path = path.concat({
                                name: "billing_contact_dropdown"
                            });
                            if (parentAbsoluteLocation) {
                                this._parentAbsoluteLocation = parentAbsoluteLocation;
                            }
                        }
                        /**
                        * @param id of the BillingContactOptionEventFactory
                        * @returns A new instance of BillingContactOptionEventFactory
                        */
                        BillingContactDropdownEventFactory.prototype.billingContactOptionFactory = function (data) {
                            return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory.BillingContactOptionEventFactory({
                                path: this._path,
                                parentAbsoluteLocation: this._parentAbsoluteLocation,
                                data: data
                            });
                        };
                        /**
                        * @returns A new instance of BillingContactSaveButtonEventFactory
                        */
                        BillingContactDropdownEventFactory.prototype.billingContactSaveButtonFactory = function () {
                            return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory.BillingContactSaveButtonEventFactory({
                                path: this._path,
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            });
                        };
                        /**
                        * @returns A new instance of BillingContactCancelButtonEventFactory
                        */
                        BillingContactDropdownEventFactory.prototype.billingContactCancelButtonFactory = function () {
                            return new WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory.BillingContactCancelButtonEventFactory({
                                path: this._path,
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            });
                        };
                        BillingContactDropdownEventFactory.prototype._getCurrentLocation = function () {
                            return {
                                pathNodes: this._path,
                                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                            };
                        };
                        BillingContactDropdownEventFactory.prototype.getAbsoluteLocation = function () {
                            var _a;
                            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                            return {
                                locations: __spreadArrays((parentLocations || []), [
                                    this._getCurrentLocation()
                                ]),
                            };
                        };
                        /**
                         * Create interaction for hit ui_reveal action.
                         */
                        BillingContactDropdownEventFactory.prototype.hitUiReveal = function () {
                            var data = {
                                actionName: "ui_reveal",
                                actionVersion: 1,
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                interactionType: "hit",
                                specificationMode: SPECIFICATION_MODE,
                                specificationVersion: SPECIFICATION_VERSION,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        return BillingContactDropdownEventFactory;
                    }());
                    LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory = BillingContactDropdownEventFactory;
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    var BillingContactDropdownEventFactory;
                    (function (BillingContactDropdownEventFactory) {
                        /**
                         * Create BillingContactOptionEventFactory class with path
                         * @param path to this node
                         */
                        var BillingContactOptionEventFactory = /** @class */ (function () {
                            function BillingContactOptionEventFactory(_a) {
                                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                                this._path = path.concat(__assign(__assign({}, data), { name: "billing_contact_option" }));
                                if (parentAbsoluteLocation) {
                                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                                }
                            }
                            BillingContactOptionEventFactory.prototype._getCurrentLocation = function () {
                                return {
                                    pathNodes: this._path,
                                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                                };
                            };
                            BillingContactOptionEventFactory.prototype.getAbsoluteLocation = function () {
                                var _a;
                                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                                return {
                                    locations: __spreadArrays((parentLocations || []), [
                                        this._getCurrentLocation()
                                    ]),
                                };
                            };
                            /**
                             * Create interaction for hit ui_select action.
                             */
                            BillingContactOptionEventFactory.prototype.hitUiSelect = function () {
                                var data = {
                                    actionName: "ui_select",
                                    actionVersion: 1,
                                    app: APP,
                                    generatorVersion: GENERATOR_VERSION,
                                    interactionType: "hit",
                                    specificationMode: SPECIFICATION_MODE,
                                    specificationVersion: SPECIFICATION_VERSION,
                                    annotatorConfigurationVersion: '',
                                    annotatorVersion: '',
                                    location: this._getCurrentLocation(),
                                    parentAbsoluteLocation: this._parentAbsoluteLocation
                                };
                                return data;
                            };
                            return BillingContactOptionEventFactory;
                        }());
                        BillingContactDropdownEventFactory.BillingContactOptionEventFactory = BillingContactOptionEventFactory;
                    })(BillingContactDropdownEventFactory = LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory || (LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory = {}));
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    var BillingContactDropdownEventFactory;
                    (function (BillingContactDropdownEventFactory) {
                        /**
                         * Create BillingContactSaveButtonEventFactory class with path
                         * @param path to this node
                         */
                        var BillingContactSaveButtonEventFactory = /** @class */ (function () {
                            function BillingContactSaveButtonEventFactory(_a) {
                                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                                this._path = path.concat({
                                    name: "billing_contact_save_button"
                                });
                                if (parentAbsoluteLocation) {
                                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                                }
                            }
                            BillingContactSaveButtonEventFactory.prototype._getCurrentLocation = function () {
                                return {
                                    pathNodes: this._path,
                                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                                };
                            };
                            BillingContactSaveButtonEventFactory.prototype.getAbsoluteLocation = function () {
                                var _a;
                                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                                return {
                                    locations: __spreadArrays((parentLocations || []), [
                                        this._getCurrentLocation()
                                    ]),
                                };
                            };
                            /**
                             * Create interaction for hit save_billing_info action.
                           * @param has_country The value is true when user adds country information or otherwise it is false
                           * @param has_email The value is true when user adds email address or otherwise it is false
                             */
                            BillingContactSaveButtonEventFactory.prototype.hitSaveBillingInfo = function (_a) {
                                var hasCountry = _a.hasCountry, hasEmail = _a.hasEmail;
                                var data = {
                                    actionParameterNames: ['has_country', 'has_email'],
                                    actionParameterValues: [hasCountry.toString(), hasEmail.toString()],
                                    actionName: "save_billing_info",
                                    actionVersion: 1,
                                    app: APP,
                                    generatorVersion: GENERATOR_VERSION,
                                    interactionType: "hit",
                                    specificationMode: SPECIFICATION_MODE,
                                    specificationVersion: SPECIFICATION_VERSION,
                                    annotatorConfigurationVersion: '',
                                    annotatorVersion: '',
                                    location: this._getCurrentLocation(),
                                    parentAbsoluteLocation: this._parentAbsoluteLocation
                                };
                                return data;
                            };
                            return BillingContactSaveButtonEventFactory;
                        }());
                        BillingContactDropdownEventFactory.BillingContactSaveButtonEventFactory = BillingContactSaveButtonEventFactory;
                    })(BillingContactDropdownEventFactory = LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory || (LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory = {}));
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
(function (WebTeamEventFactory) {
    var AccessLevelPopoverTriggerEventFactory;
    (function (AccessLevelPopoverTriggerEventFactory) {
        var AccessLevelPopoverEventFactory;
        (function (AccessLevelPopoverEventFactory) {
            var LeaveTeamLinkEventFactory;
            (function (LeaveTeamLinkEventFactory) {
                var LeaveTeamDialogEventFactory;
                (function (LeaveTeamDialogEventFactory) {
                    var BillingContactDropdownEventFactory;
                    (function (BillingContactDropdownEventFactory) {
                        /**
                         * Create BillingContactCancelButtonEventFactory class with path
                         * @param path to this node
                         */
                        var BillingContactCancelButtonEventFactory = /** @class */ (function () {
                            function BillingContactCancelButtonEventFactory(_a) {
                                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                                this._path = path.concat({
                                    name: "billing_contact_cancel_button"
                                });
                                if (parentAbsoluteLocation) {
                                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                                }
                            }
                            BillingContactCancelButtonEventFactory.prototype._getCurrentLocation = function () {
                                return {
                                    pathNodes: this._path,
                                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                                };
                            };
                            BillingContactCancelButtonEventFactory.prototype.getAbsoluteLocation = function () {
                                var _a;
                                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                                return {
                                    locations: __spreadArrays((parentLocations || []), [
                                        this._getCurrentLocation()
                                    ]),
                                };
                            };
                            /**
                             * Create interaction for hit ui_hide action.
                             */
                            BillingContactCancelButtonEventFactory.prototype.hitUiHide = function () {
                                var data = {
                                    actionName: "ui_hide",
                                    actionVersion: 1,
                                    app: APP,
                                    generatorVersion: GENERATOR_VERSION,
                                    interactionType: "hit",
                                    specificationMode: SPECIFICATION_MODE,
                                    specificationVersion: SPECIFICATION_VERSION,
                                    annotatorConfigurationVersion: '',
                                    annotatorVersion: '',
                                    location: this._getCurrentLocation(),
                                    parentAbsoluteLocation: this._parentAbsoluteLocation
                                };
                                return data;
                            };
                            return BillingContactCancelButtonEventFactory;
                        }());
                        BillingContactDropdownEventFactory.BillingContactCancelButtonEventFactory = BillingContactCancelButtonEventFactory;
                    })(BillingContactDropdownEventFactory = LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory || (LeaveTeamDialogEventFactory.BillingContactDropdownEventFactory = {}));
                })(LeaveTeamDialogEventFactory = LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory || (LeaveTeamLinkEventFactory.LeaveTeamDialogEventFactory = {}));
            })(LeaveTeamLinkEventFactory = AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory || (AccessLevelPopoverEventFactory.LeaveTeamLinkEventFactory = {}));
        })(AccessLevelPopoverEventFactory = AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory || (AccessLevelPopoverTriggerEventFactory.AccessLevelPopoverEventFactory = {}));
    })(AccessLevelPopoverTriggerEventFactory = WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory || (WebTeamEventFactory.AccessLevelPopoverTriggerEventFactory = {}));
})(WebTeamEventFactory = exports.WebTeamEventFactory || (exports.WebTeamEventFactory = {}));
exports.WebTeamEventFactory = WebTeamEventFactory;
