// @spotify-internal/encore-web/es/components/NavBar/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { gray10, gray95 } from '@spotify-internal/encore-foundation';
import styled, { css, ThemeProvider } from 'styled-components';
var NavBarPage = styled.div.withConfig({
  displayName: "NavBar__NavBarPage",
  componentId: "suvglb-0"
})(["", ";"], function (props) {
  return css(["color:", ";"], props.theme.navBar === 'dark' ? gray95 : gray10);
});

function NavBarComponent(_ref) {
  var list = _ref.list,
      children = _ref.children,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["list", "children", "theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement("section", props, list, /*#__PURE__*/React.createElement(NavBarPage, null, children)));
}

var navBarHelpers = {
  darkTheme: {
    navBar: 'dark'
  }
};
/**
* **NavBar** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavBar) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navbar--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-bar) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=377:171) |
*
* A tab component used to toggle between between pages
*
* @example
* export const Default = () => (
*  <NavBar
*    list={
*      <nav aria-label="Main">
*        <NavBarList>
*          {navBarListItems.map((item, i) => (
*            <NavBarListItem
*              key={i}
*              id={`${item.toLowerCase()}Link`}
*              label={item}
*              active={i === 0}
*              disabled={i === 3}
*              href={`#${item}`}
*            />
*          ))}
*        </NavBarList>
*      </nav>
*    }
*  >
*    <div id="overview-page">
*      <Type as="h1" variant={Type.heading1} condensed>
*        Overview page
*      </Type>
*      <Type>Laborum quis irure sint sunt ipsum culpa culpa consequat labore ut.</Type>
*    </div>
*  </NavBar>
*);
*
*/

export var NavBar = Object.assign(NavBarComponent, navBarHelpers);
export default NavBar;
