// @spotify-internal/encore-web/es/components/FlyOutLink/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import React from 'react';
import styled from 'styled-components';
import { componentDeprecationWarning } from "../../deprecationWarnings";
import { transitionFade } from "../../styles";
import { TextLink } from "../TextLink";
export var FlyOutLink = styled(function FlyOutLinkComponent(_ref) {
  var props = _extends({}, _ref);

  componentDeprecationWarning('FlyoutLink', 'TextLink', 'an upcoming release', 'Pass the "standalone" prop to TextLink to achieve the same visuals.');
  return /*#__PURE__*/React.createElement(TextLink, Object.assign({
    semanticColor: "textSubdued",
    standalone: true
  }, props));
}).withConfig({
  displayName: "FlyOutLink",
  componentId: "uvwq4f-0"
})(["transition:color ", ";"], transitionFade);
/**
* **FlyOutLink** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/FlyOutLink) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-flyout-flyoutlink--default) | [Encore](https://encore.spotify.net/web/components/flyout) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=889:270) |
*
* FlyoutLink is a Textlink in the Flyout header.
*
* @example
* () => (
*  <ColorSetWrapper colorSet="invertedDark">
*    <FlyOutLink component="button">
*      <Type variant="minuetBold">Add Artist</Type>
*    </FlyOutLink>
*  </ColorSetWrapper>
*);
*
*/

export default FlyOutLink;
