// @spotify-internal/adstudio-web-utils/lib/constants/createAd.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CREATIVE_FORMAT = exports.CLIENT_VIDEO_TYPES = exports.CLIENT_IMAGE_TYPES = exports.CLIENT_AUDIO_TYPES = exports.MIN_PORTRAIT_VIDEO_WIDTH = exports.MIN_PORTRAIT_VIDEO_HEIGHT = exports.MIN_LANDSCAPE_VIDEO_WIDTH = exports.MIN_LANDSCAPE_VIDEO_HEIGHT = exports.MAX_VIDEO_FILE_SIZE = exports.MAX_VIDEO_DURATION = exports.MIN_VIDEO_DURATION = exports.PORTRAIT_VIDEO_ASPECT_RATIO = exports.LANDSCAPE_VIDEO_ASPECT_RATIO = exports.MAX_AUDIO_DURATION = exports.MIN_AUDIO_DURATION = exports.MAX_COVER_ART_IMAGE_DIMENTIONS = exports.COVER_ART_IMAGE_DIMENSIONS = exports.FORM_NAMES = void 0;
var IMAGE = 'cover-art-upload';
var AUDIO = 'upload-audio-tab';
var VIDEO = 'video-upload';
var NEW_VOICEOVER = 'request-voiceover-tab';
var EXISTING_VOICEOVER = 'existing-voiceover-tab';
var AUDIO_SUBMISSION = 'audioSubmission';
var FORM_NAMES = {
  HEADLINE: 'name',
  URL: 'clickthroughUrl',
  IMAGE: IMAGE,
  AUDIO: AUDIO,
  VIDEO: VIDEO,
  NEW_VOICEOVER: NEW_VOICEOVER,
  AUDIO_SUBMISSION: AUDIO_SUBMISSION,
  IMAGE_UPLOADER: "".concat(IMAGE, "__uploader"),
  AUDIO_UPLOADER: "".concat(AUDIO, "__uploader"),
  VIDEO_UPLOADER: "".concat(VIDEO, "__uploader"),
  NEW_VOICEOVER_BGMUSIC_UPLOADER: "".concat(NEW_VOICEOVER, "__uploader"),
  EXISTING_VOICEOVER_BGMUSIC_UPLOADER: "".concat(EXISTING_VOICEOVER, "__uploader"),
  NEW_VOICEOVER_BGMUSIC_ID: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".bgmusicId"),
  NEW_VOICEOVER_NAME: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".name"),
  NEW_VOICEOVER_TRANSCRIPT: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".transcript"),
  NEW_VOICEOVER_INSTRUCTIONS: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".instructions"),
  NEW_VOICEOVER_LOCALE: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".locale"),
  NEW_VOICEOVER_VOICE: "".concat(AUDIO_SUBMISSION, ".").concat(NEW_VOICEOVER, ".voice"),
  EXISTING_VOICEOVER_ID: "".concat(AUDIO_SUBMISSION, ".").concat(EXISTING_VOICEOVER, ".id"),
  EXISTING_VOICEOVER_BGMUSIC_ID: "".concat(AUDIO_SUBMISSION, ".").concat(EXISTING_VOICEOVER, ".bgmusicId")
};
exports.FORM_NAMES = FORM_NAMES;
var COVER_ART_IMAGE_DIMENSIONS = {
  width: 640,
  height: 640
};
exports.COVER_ART_IMAGE_DIMENSIONS = COVER_ART_IMAGE_DIMENSIONS;
var MAX_COVER_ART_IMAGE_DIMENTIONS = {
  width: 20000,
  height: 20000
};
exports.MAX_COVER_ART_IMAGE_DIMENTIONS = MAX_COVER_ART_IMAGE_DIMENTIONS;
var MIN_AUDIO_DURATION = null;
exports.MIN_AUDIO_DURATION = MIN_AUDIO_DURATION;
var MAX_AUDIO_DURATION = 31;
exports.MAX_AUDIO_DURATION = MAX_AUDIO_DURATION;
var LANDSCAPE_VIDEO_ASPECT_RATIO = 1.778; // 16:9

exports.LANDSCAPE_VIDEO_ASPECT_RATIO = LANDSCAPE_VIDEO_ASPECT_RATIO;
var PORTRAIT_VIDEO_ASPECT_RATIO = 0.5625; // 9:16

exports.PORTRAIT_VIDEO_ASPECT_RATIO = PORTRAIT_VIDEO_ASPECT_RATIO;
var MIN_VIDEO_DURATION = 1;
exports.MIN_VIDEO_DURATION = MIN_VIDEO_DURATION;
var MAX_VIDEO_DURATION = 30;
exports.MAX_VIDEO_DURATION = MAX_VIDEO_DURATION;
var MAX_VIDEO_FILE_SIZE = 500 * 1024 * 1024;
exports.MAX_VIDEO_FILE_SIZE = MAX_VIDEO_FILE_SIZE;
var MIN_LANDSCAPE_VIDEO_HEIGHT = 720;
exports.MIN_LANDSCAPE_VIDEO_HEIGHT = MIN_LANDSCAPE_VIDEO_HEIGHT;
var MIN_LANDSCAPE_VIDEO_WIDTH = 1280;
exports.MIN_LANDSCAPE_VIDEO_WIDTH = MIN_LANDSCAPE_VIDEO_WIDTH;
var MIN_PORTRAIT_VIDEO_HEIGHT = 1280;
exports.MIN_PORTRAIT_VIDEO_HEIGHT = MIN_PORTRAIT_VIDEO_HEIGHT;
var MIN_PORTRAIT_VIDEO_WIDTH = 720;
exports.MIN_PORTRAIT_VIDEO_WIDTH = MIN_PORTRAIT_VIDEO_WIDTH;
var CLIENT_AUDIO_TYPES = ['audio/ogg', // not supported by safari
'audio/mp3', 'audio/wav', // supported by chrome
'audio/mpeg', 'audio/x-wav' // supported by firefox, safari
];
exports.CLIENT_AUDIO_TYPES = CLIENT_AUDIO_TYPES;
var CLIENT_IMAGE_TYPES = ['image/jpeg', 'image/png'];
exports.CLIENT_IMAGE_TYPES = CLIENT_IMAGE_TYPES;
var CLIENT_VIDEO_TYPES = ['video/mp4', 'video/quicktime'];
exports.CLIENT_VIDEO_TYPES = CLIENT_VIDEO_TYPES;
var AUDIO_FORMAT = 'AUDIO';
var VIDEO_PORTRAIT_FORMAT = 'PORTRAIT';
var VIDEO_LANDSCAPE_FORMAT = 'LANDSCAPE';
var CREATIVE_FORMAT = {
  AUDIO_FORMAT: AUDIO_FORMAT,
  VIDEO_PORTRAIT_FORMAT: VIDEO_PORTRAIT_FORMAT,
  VIDEO_LANDSCAPE_FORMAT: VIDEO_LANDSCAPE_FORMAT
};
exports.CREATIVE_FORMAT = CREATIVE_FORMAT;
