// spotify-harmony/types/errors.js
'use strict';

var TransportErrors = require('spotify-transport/types/errors');
var PlaybackErrors = require('spotify-playback/types/errors');
var debugEnum = require('spotify-debug-tools/debug_enum');

var extend = function(obj) {
  var source;
  for (var i = 1; i < arguments.length; i++) {
    source = arguments[i];
    if (source) {
      for (var prop in source) {
        if (!(prop in source)) {
          continue;
        }
        obj[prop] = source[prop];
      }
    }
  }
  return obj;
};


var Errors = extend(
  {},
  TransportErrors,
  PlaybackErrors,
  {
    // Harmony Client Errors
    USER_INFO_REQUEST_EMPTY_RESPONSE: 'USER_INFO_REQUEST_EMPTY_RESPONSE',
    // User info responded with unexpected status.
    USER_INFO_REQUEST_FAILED_WITH_STATUS:
      'USER_INFO_REQUEST_FAILED_WITH_STATUS',

    // Harmony Errors
    HARMONY_LIFECYCLE_ERROR: 'HARMONY_LIFECYCLE_ERROR',
    HARMONY_NO_TRACKS_LOADED: 'HARMONY_NO_TRACKS_LOADED',
    HARMONY_OPERATION_FORBIDDEN: 'HARMONY_OPERATION_FORBIDDEN',
    HARMONY_LOCAL_PLAYER_DISABLED: 'HARMONY_LOCAL_PLAYER_DISABLED',

    // Player API Client
    CONNECTAPI_CLIENT_INVALID_ARGUMENTS: 'CONNECTAPI_CLIENT_INVALID_ARGUMENTS',
    CONNECTAPI_CLIENT_NO_CONNECTION_ID: 'CONNECT_API_CLIENT_NO_CONNECTION_ID',
    CONNECTAPI_CLIENT_NO_ACTIVE_DEVICE: 'CONNECT_API_CLIENT_NO_ACTIVE_DEVICE',
    CONNECTAPI_CLIENT_NO_DEVICE: 'CONNECTAPI_CLIENT_NO_DEVICE',
    CONNECTAPI_CLIENT_NO_SESSION_ID: 'CONNECTAPI_CLIENT_NO_SESSION_ID',
    CONNECTAPI_CLIENT_NO_STATE: 'CONNECTAPI_CLIENT_NO_STATE',
    CONNECTAPI_CLIENT_INVALID_POSITION: 'CONNECT_API_CLIENT_INVALID_POSITION',
    CONNECTAPI_CLIENT_INVALID_VOLUME: 'CONNECT_API_CLIENT_INVALID_VOLUME',
    CONNECTAPI_MAX_SUBSCRIPTIONS_REACHED:
      'CONNECTAPI_MAX_SUBSCRIPTIONS_REACHED',
    CONNECTAPI_REGISTRATION_FAILED_WITH_STATUS:
      'CONNECTAPI_REGISTRATION_FAILED_WITH_STATUS',

    // TPAPI Client response errors.
    TP_NO_RESPONSE_BODY: 'TP_NO_RESPONSE_BODY',
    TP_REGISTRATION_FAILED_NON_PREMIUM: 'TP_REGISTRATION_FAILED_NON_PREMIUM',
    TP_REGISTRATION_FAILED_WITH_STATUS: 'TP_REGISTRATION_FAILED_WITH_STATUS',
    TP_MAX_SUBSCRIPTIONS_REACHED: 'TP_MAX_SUBSCRIPTIONS_REACHED',
    TP_UPDATE_REQUEST_EMPTY_RESPONSE: 'TP_UPDATE_REQUEST_EMPTY_RESPONSE',
    TP_PARSE_STATE_UPDATE_FAILED_WITH_STATUS:
      'TP_PARSE_STATE_UPDATE_FAILED_WITH_STATUS',

    // TPAPI Client Command errors.
    TP_UNKNOWN_COMMAND: 'TP_UNKNOWN_COMMAND',

    // TPAPI Client State errors
    TP_CANNOT_CREATE_STATE_REF: 'TP_CANNOT_CREATE_STATE_REF',
    TP_MISSING_INITIAL_STATE: 'TP_MISSING_INITIAL_STATE',
    TP_INVALID_STATE_REFERENCE: 'TP_INVALID_STATE_REFERENCE',
    TP_CONFLICT_REQUEST_FAILED_WITH_STATUS:
      'TP_CONFLICT_REQUEST_FAILED_WITH_STATUS',

    // TPAPI Client Stream time error
    TP_STREAM_TIME_VALUE_OUT_OF_RANGE: 'TP_STREAM_TIME_VALUE_OUT_OF_RANGE'
  }
);

module.exports = debugEnum(Errors);
