// spotify-harmony/types/event.js
'use strict';

var debugEnum = require('spotify-debug-tools/debug_enum');

/**
 * The various events that can be emitted from the module.
 *
 * @enum {string}
 */
var Event = {
  AUTHENTICATED: 'authenticated',
  AUTHENTICATION_ERROR: 'authentication_error',
  AUTOPLAY_FAILED: 'autoplay_failed',
  BEFORE_DISCONNECT: 'before_disconnect',
  BEFORE_VOLUME_CHANGE: 'before_volume_change',
  BUFFERING_END: 'buffering_end',
  BUFFERING_START: 'buffering_start',
  CLIENT_DESCRIPTOR_CHANGED: 'descriptor_changed',
  CONNECTED: 'connected',
  CONNECTION_ERROR: 'connection_error',
  DEVICES_CHANGED: 'devices_changed',
  DEVICE_DESCRIPTOR_CHANGED: 'descriptor_changed',
  DISCONNECTED: 'disconnected',
  DURATION_CHANGED: 'duration_changed',
  ERROR: 'error',
  LAST_ACTIVE_DEVICE_INFO_CHANGED: 'last_active_device_info_changed',
  LOCAL_CONTEXT_ENDED: 'local_context_ended',
  LOCAL_PLAYER_DISABLED: 'local_player_disabled',
  LOCAL_PLAYER_ENABLED: 'local_player_enabled',
  LOCAL_PLAYER_LOGGED_OUT: 'local_player_logged_out',
  LOGGED_OUT: 'logged_out',
  MAX_LIST_ERRORS_REACHED: 'max_list_errors_reached',
  MAX_SUBSCRIPTIONS_REACHED: 'max_subscriptions_reached',
  PLAYBACK_CAPPED: 'playback_capped',
  PLAYER_INITIALIZATION_DONE: 'player_initialization_done',
  PLAYER_INITIALIZATION_FAILED: 'player_initialization_failed',
  PRODUCT_STATE_CHANGED: 'product_state_changed',
  PROGRESS: 'progress',
  RECONNECTED: 'reconnected',
  RECONNECTING: 'reconnecting',
  REMOTE_OBSERVER_DISABLED: 'remote_observer_disabled',
  REMOTE_OBSERVER_ENABLED: 'remote_observer_enabled',
  SERVICE_REGISTRATION_ERROR: 'service_registration_error',
  STATE_CHANGED: 'state_changed',
  TRACK_ENDED: 'track_ended',
  UNRECOVERABLE_FAILURE: 'unrecoverable_failure',
  VIDEO_ELEMENT_APPENDED: 'video_element_appended',
  VIDEO_ELEMENT_REMOVED: 'video_element_removed'
};

module.exports = debugEnum(Event);
