// @spotify-internal/encore-web/es/components/FormRadio/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["id", "children", "semanticColor", "isHover", "isFocus", "isActive", "theme"];
import React, { useContext } from 'react';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { renamedPropWarning } from "../../deprecationWarnings";
import { VisuallyHidden } from "../VisuallyHidden";
import { Indicator } from "./Indicator";
import { Label } from "./Label";
import { Radio } from "./Radio";
export var FormRadio = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var id = _ref.id,
      children = _ref.children,
      _ref$semanticColor = _ref.semanticColor,
      semanticColor = _ref$semanticColor === void 0 ? 'essentialBrightAccent' : _ref$semanticColor,
      isHover = _ref.isHover,
      isFocus = _ref.isFocus,
      isActive = _ref.isActive,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    renamedPropWarning('theme', 'semanticColor');
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(Radio, null, /*#__PURE__*/React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "radio",
    ref: ref,
    id: id
  }, props)), /*#__PURE__*/React.createElement("label", {
    htmlFor: id
  }, /*#__PURE__*/React.createElement(Indicator, {
    isUsingKeyboard: isUsingKeyboard,
    semanticColor: semanticColor,
    isHover: isHover,
    isFocus: isFocus,
    isActive: isActive
  }), children && /*#__PURE__*/React.createElement(Label, null, children)));
});
/**
* **FormRadio** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/FormRadio) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-formradio--default) | [Encore](https://encore.spotify.net/web/components/selectors?format=figma-figma#formradio) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:526) |
*
* FormRadio allow users to make one selection within a Form component.
*
* @example
* () => (
*  <FormGroup label="Radio example" withFieldset>
*    <FormRadio id="option-1" name="radio-example" defaultChecked>
*      Option 1
*    </FormRadio>
*    <FormRadio id="option-2" name="radio-example">
*      Option 2
*    </FormRadio>
*  </FormGroup>
*);
*
*/

export default FormRadio;
