// @spotify-internal/dunder-client/models/com/spotify/ads/dunder/ExportHierarchyRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var ExportHierarchyRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.dunder = (function() {
                /**
                 * Namespace dunder.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var dunder = {};

                dunder.ExportHierarchyRequest = (function() {
                    /**
                     * Properties of a ExportHierarchyRequest.
                     * @memberof com.spotify.ads.dunder
                     * @interface IExportHierarchyRequest
                     * @property {string|null} [adAccountId] ExportHierarchyRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.CampaignState.Status|null} [campaignStatus] ExportHierarchyRequest campaignStatus
                     * @property {com.spotify.adstudiobff.proto.FlightState.Status|null} [flightStatus] ExportHierarchyRequest flightStatus
                     * @property {com.spotify.adstudiobff.proto.CreativeState.Status|null} [creativeStatus] ExportHierarchyRequest creativeStatus
                     * @property {Array.<com.spotify.adstudiobff.proto.HierarchyColumn.Column>|null} [fields] ExportHierarchyRequest fields
                     * @property {Array.<com.spotify.ads.dunder.HierarchySortOrder>|null} [sortOrder] ExportHierarchyRequest sortOrder
                     * @property {string|null} [searchQuery] ExportHierarchyRequest searchQuery
                     * @property {string|null} [campaignId] ExportHierarchyRequest campaignId
                     * @property {string|null} [flightId] ExportHierarchyRequest flightId
                     * @property {string|null} [creativeId] ExportHierarchyRequest creativeId
                     * @property {string|null} [timezone] ExportHierarchyRequest timezone
                     * @property {string|null} [locale] ExportHierarchyRequest locale
                     * @property {com.spotify.ads.dunder.DateRange|null} [reportDateFilter] ExportHierarchyRequest reportDateFilter
                     * @property {string|null} [setAdAccountId] ExportHierarchyRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.CampaignState.Status|null} [setCampaignStatus] ExportHierarchyRequest setCampaignStatus
                     * @property {com.spotify.adstudiobff.proto.FlightState.Status|null} [setFlightStatus] ExportHierarchyRequest setFlightStatus
                     * @property {com.spotify.adstudiobff.proto.CreativeState.Status|null} [setCreativeStatus] ExportHierarchyRequest setCreativeStatus
                     * @property {Array.<com.spotify.adstudiobff.proto.HierarchyColumn.Column>|null} [setFields] ExportHierarchyRequest setFields
                     * @property {Array.<com.spotify.ads.dunder.HierarchySortOrder>|null} [setSortOrder] ExportHierarchyRequest setSortOrder
                     * @property {string|null} [setSearchQuery] ExportHierarchyRequest setSearchQuery
                     * @property {string|null} [setCampaignId] ExportHierarchyRequest setCampaignId
                     * @property {string|null} [setFlightId] ExportHierarchyRequest setFlightId
                     * @property {string|null} [setCreativeId] ExportHierarchyRequest setCreativeId
                     * @property {string|null} [setTimezone] ExportHierarchyRequest setTimezone
                     * @property {string|null} [setLocale] ExportHierarchyRequest setLocale
                     * @property {com.spotify.ads.dunder.DateRange|null} [setReportDateFilter] ExportHierarchyRequest setReportDateFilter
                     */
                    
                    /**
                     * Constructs a new ExportHierarchyRequest.
                     * @memberof com.spotify.ads.dunder
                     * @classdesc Represents a ExportHierarchyRequest.
                     * @implements IExportHierarchyRequest
                     * @constructor
                     * @param {com.spotify.ads.dunder.IExportHierarchyRequest=} [properties] Properties to set
                     */
                    function ExportHierarchyRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * ExportHierarchyRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.adAccountId = "";

                    /**
                     * ExportHierarchyRequest campaignStatus.
                     * @member {com.spotify.adstudiobff.proto.CampaignState.Status} campaignStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.campaignStatus = {};

                    /**
                     * ExportHierarchyRequest flightStatus.
                     * @member {com.spotify.adstudiobff.proto.FlightState.Status} flightStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.flightStatus = {};

                    /**
                     * ExportHierarchyRequest creativeStatus.
                     * @member {com.spotify.adstudiobff.proto.CreativeState.Status} creativeStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.creativeStatus = {};

                    /**
                     * ExportHierarchyRequest fields.
                     * @member {Array.<com.spotify.adstudiobff.proto.HierarchyColumn.Column>} fields
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.fields = [];

                    /**
                     * ExportHierarchyRequest sortOrder.
                     * @member {Array.<com.spotify.ads.dunder.HierarchySortOrder>} sortOrder
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.sortOrder = [];

                    /**
                     * ExportHierarchyRequest searchQuery.
                     * @member {string} searchQuery
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.searchQuery = "";

                    /**
                     * ExportHierarchyRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.campaignId = "";

                    /**
                     * ExportHierarchyRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.flightId = "";

                    /**
                     * ExportHierarchyRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.creativeId = "";

                    /**
                     * ExportHierarchyRequest timezone.
                     * @member {string} timezone
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.timezone = "";

                    /**
                     * ExportHierarchyRequest locale.
                     * @member {string} locale
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.locale = "";

                    /**
                     * ExportHierarchyRequest reportDateFilter.
                     * @member {com.spotify.ads.dunder.DateRange} reportDateFilter
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     */
                    ExportHierarchyRequest.prototype.reportDateFilter = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignStatus
                     * @function setCampaignStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {com.spotify.adstudiobff.proto.CampaignState.Status} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setCampaignStatus = function(value) {
                        this['campaignStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets flightStatus
                     * @function setFlightStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {com.spotify.adstudiobff.proto.FlightState.Status} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setFlightStatus = function(value) {
                        this['flightStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeStatus
                     * @function setCreativeStatus
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {com.spotify.adstudiobff.proto.CreativeState.Status} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setCreativeStatus = function(value) {
                        this['creativeStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets fields
                     * @function setFields
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {Array.<com.spotify.adstudiobff.proto.HierarchyColumn.Column>} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setFields = function(value) {
                        this['fields'] = value;
                        return this;
                    };

                    /**
                     * Sets sortOrder
                     * @function setSortOrder
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {Array.<com.spotify.ads.dunder.HierarchySortOrder>} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setSortOrder = function(value) {
                        this['sortOrder'] = value;
                        return this;
                    };

                    /**
                     * Sets searchQuery
                     * @function setSearchQuery
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setSearchQuery = function(value) {
                        this['searchQuery'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets timezone
                     * @function setTimezone
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setTimezone = function(value) {
                        this['timezone'] = value;
                        return this;
                    };

                    /**
                     * Sets locale
                     * @function setLocale
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setLocale = function(value) {
                        this['locale'] = value;
                        return this;
                    };

                    /**
                     * Sets reportDateFilter
                     * @function setReportDateFilter
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @param {com.spotify.ads.dunder.DateRange} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.ExportHierarchyRequest} ExportHierarchyRequest
                     */
                    ExportHierarchyRequest.prototype.setReportDateFilter = function(value) {
                        this['reportDateFilter'] = value;
                        return this;
                    };

                    /**
                     * Converts the ExportHierarchyRequest to an object
                     * @function toObject
                     * @memberof com.spotify.ads.dunder.ExportHierarchyRequest
                     * @instance
                     * @returns {object} object
                     */
                    ExportHierarchyRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return ExportHierarchyRequest;
                })();
                
                ExportHierarchyRequest = dunder.ExportHierarchyRequest;
                return dunder;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["ExportHierarchyRequest"] = ExportHierarchyRequest;
exports["default"] = $root;
