// @spotify-internal/dunder-client/models/com/spotify/ads/dunder/HierarchySortOrder.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var HierarchySortOrder = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.dunder = (function() {
                /**
                 * Namespace dunder.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var dunder = {};

                dunder.HierarchySortOrder = (function() {
                    /**
                     * Properties of a HierarchySortOrder.
                     * @memberof com.spotify.ads.dunder
                     * @interface IHierarchySortOrder
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] HierarchySortOrder sortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [orderBy] HierarchySortOrder orderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] HierarchySortOrder setSortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [setOrderBy] HierarchySortOrder setOrderBy
                     */
                    
                    /**
                     * Constructs a new HierarchySortOrder.
                     * @memberof com.spotify.ads.dunder
                     * @classdesc Represents a HierarchySortOrder.
                     * @implements IHierarchySortOrder
                     * @constructor
                     * @param {com.spotify.ads.dunder.IHierarchySortOrder=} [properties] Properties to set
                     */
                    function HierarchySortOrder(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * HierarchySortOrder sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.ads.dunder.HierarchySortOrder
                     * @instance
                     */
                    HierarchySortOrder.prototype.sortDirection = {};

                    /**
                     * HierarchySortOrder orderBy.
                     * @member {com.spotify.adstudiobff.proto.HierarchyColumn.Column} orderBy
                     * @memberof com.spotify.ads.dunder.HierarchySortOrder
                     * @instance
                     */
                    HierarchySortOrder.prototype.orderBy = {};


                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.ads.dunder.HierarchySortOrder
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.HierarchySortOrder} HierarchySortOrder
                     */
                    HierarchySortOrder.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets orderBy
                     * @function setOrderBy
                     * @memberof com.spotify.ads.dunder.HierarchySortOrder
                     * @param {com.spotify.adstudiobff.proto.HierarchyColumn.Column} value
                     * @instance
                     * @returns {com.spotify.ads.dunder.HierarchySortOrder} HierarchySortOrder
                     */
                    HierarchySortOrder.prototype.setOrderBy = function(value) {
                        this['orderBy'] = value;
                        return this;
                    };

                    /**
                     * Converts the HierarchySortOrder to an object
                     * @function toObject
                     * @memberof com.spotify.ads.dunder.HierarchySortOrder
                     * @instance
                     * @returns {object} object
                     */
                    HierarchySortOrder.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return HierarchySortOrder;
                })();
                
                HierarchySortOrder = dunder.HierarchySortOrder;
                return dunder;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["HierarchySortOrder"] = HierarchySortOrder;
exports["default"] = $root;
