// @spotify-internal/creator-tape/es/components/FormCheckbox/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import VisuallyHidden from '../VisuallyHidden';
import Checkbox from './Checkbox';
import Indicator from './Indicator';
import Label from './Label';
var propTypes = {
  theme: PropTypes.shape({
    formCheckbox: PropTypes.oneOf(['dark'])
  }),
  children: PropTypes.node,
  small: PropTypes.bool,
  indeterminate: PropTypes.bool
};
var defaultProps = {
  theme: {}
};
var FormCheckbox = React.forwardRef(function (_ref, ref) {
  var theme = _ref.theme,
      small = _ref.small,
      indeterminate = _ref.indeterminate,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "small", "indeterminate", "children"]);

  return React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Checkbox, null, React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "checkbox",
    ref: ref
  }, props)), React.createElement(Indicator, {
    indeterminate: indeterminate
  }), children && React.createElement(Label, {
    small: small
  }, children)));
});
FormCheckbox.darkTheme = {
  formCheckbox: 'dark'
};
FormCheckbox.propTypes = propTypes;
FormCheckbox.defaultProps = defaultProps;
FormCheckbox.displayName = 'FormCheckbox';
export default FormCheckbox;
