// @spotify-internal/adstudio-shared/lib/components/AdTargeting/Form/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Targeting = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _global = require("global");

var _reduxForm = require("redux-form");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _api = require("../api");

var strings = _interopRequireWildcard(require("../constants"));

var _constants2 = require("../../TargetingCheckboxList/constants");

var _warnForDeprecatedProp = _interopRequireDefault(require("../../../utils/warnForDeprecatedProp"));

var _AgeRange = _interopRequireDefault(require("../AgeRange"));

var _Gender = _interopRequireDefault(require("../Gender"));

var _Platforms = _interopRequireDefault(require("../Platforms"));

var _LocationTargetingByCountry = _interopRequireDefault(require("../../LocationTargetingByCountry"));

var _ReduxFormArtistTargetingSearch = _interopRequireDefault(require("../../ReduxFormArtistTargetingSearch"));

var _ReduxFormTargetingSearch = _interopRequireDefault(require("../../ReduxFormTargetingSearch"));

var _ReduxFormTargetingCheckboxList = _interopRequireDefault(require("../../ReduxFormTargetingCheckboxList"));

var _service_grpc_web_pb = require("@spotify-internal/targetingservice/proto/com/spotify/targetingservice/proto/service_grpc_web_pb");

var _utilities = require("../../../utilities");

var _i18n = require("../../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var TargetingForm = _styledComponents["default"].div.withConfig({
  displayName: "Form__TargetingForm",
  componentId: "sc-1g5o08r-0"
})(["ul{list-style-type:none;padding:0;}"]);

var FormGroupSection = _styledComponents["default"].div.withConfig({
  displayName: "Form__FormGroupSection",
  componentId: "sc-1g5o08r-1"
})(["margin-bottom:", ";"], _encoreAdvertisingWeb.spacer16);

var Targeting = /*#__PURE__*/function (_Component) {
  _inherits(Targeting, _Component);

  var _super = _createSuper(Targeting);

  function Targeting() {
    var _this;

    _classCallCheck(this, Targeting);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.apiHost = null;
    return _this;
  }

  _createClass(Targeting, [{
    key: "UNSAFE_componentWillMount",
    value: function UNSAFE_componentWillMount() {
      this.setApiHost(this.props.apiHost, this.props.edgeHost);
    }
  }, {
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (nextProps.apiHost !== this.props.apiHost) {
        this.setApiHost(nextProps.apiHost, nextProps.edgeHost);
      }
    } // Eventually: deprecate this whole function, make apiHost a required prop,
    // and directly call the setApiHost method exposed by the api module.

  }, {
    key: "setApiHost",
    value: function setApiHost(newApiHost, newEdgeHost) {
      var hostToUse = newApiHost;
      var edgeHostToUse = newEdgeHost;

      if (!newApiHost || !newEdgeHost) {
        var isOnProdOrigin = _global.window.location.origin.indexOf('adstudio.spotify.com') >= 0 || _global.window.location.origin.indexOf('adstudio-admin2.spotify.net') >= 0;

        if (!newApiHost) {
          hostToUse = "https://".concat(isOnProdOrigin ? 'spclient' : 'exp', ".wg.spotify.com");
        }

        if (!newEdgeHost) {
          edgeHostToUse = "https://".concat(isOnProdOrigin ? 'edge-grpc.spotify.com' : 'edge-proxy-internal-guc.spotify.net');
        }
        /* eslint-disable no-console */


        if (console && console.warn) {
          console.warn("\n          You must provide the API host for calling the AdTargeting API endpoints.\n          We have guessed that you want to use ".concat(hostToUse, " and ").concat(edgeHostToUse, ". Please start providing\n          the apiHost prop to AdTargeting to avoid seeing this message, as this\n          will be deprecated in a future release.\n        "));
        }
        /* eslint-enable no-console */

      }

      var client = new _service_grpc_web_pb.TargetingServiceClient(edgeHostToUse);
      (0, _utilities.enableGrpcDevConsole)([client]);
      (0, _api.setTargetingServiceClient)(client);
      this.apiHost = hostToUse;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          advertiserCountry = _this$props.advertiserCountry,
          formName = _this$props.formName,
          features = _this$props.features,
          parentCountry = _this$props.parentCountry,
          disableLocations = _this$props.disableLocations,
          disableBulkUploadZip = _this$props.disableBulkUploadZip,
          disableTargetWholeCountry = _this$props.disableTargetWholeCountry,
          handleRemoveCountry = _this$props.handleRemoveCountry,
          language = _this$props.language,
          minAge = _this$props.minAge,
          strings = _this$props.strings,
          creativeFormat = _this$props.creativeFormat,
          disableCountry = _this$props.disableCountry;
      var AUDIENCE = strings.AUDIENCE,
          LISTENING_BEHAVIOR = strings.LISTENING_BEHAVIOR,
          headers = strings.HEADERS,
          labels = strings.LABELS,
          placeholders = strings.PLACEHOLDERS,
          releasedSegments = strings.RELEASED_SEGMENTS;
      return /*#__PURE__*/_react["default"].createElement(TargetingForm, null, /*#__PURE__*/_react["default"].createElement(FormGroupSection, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h1, {
        variant: _encoreAdvertisingWeb.Type.heading3,
        weight: _encoreAdvertisingWeb.Type.black
      }, headers[AUDIENCE]), /*#__PURE__*/_react["default"].createElement(_LocationTargetingByCountry["default"], {
        formName: formName,
        excludeCountries: features.excludeCountries,
        parentCountry: parentCountry,
        disableLocations: disableLocations,
        disableBulkUploadZip: disableBulkUploadZip,
        disableTargetWholeCountry: disableTargetWholeCountry,
        handleRemoveCountry: handleRemoveCountry,
        language: language,
        strings: strings,
        disableCountry: disableCountry
      }), /*#__PURE__*/_react["default"].createElement(_AgeRange["default"], {
        formName: formName,
        advertiserCountry: advertiserCountry,
        minAge: minAge,
        strings: strings
      }), /*#__PURE__*/_react["default"].createElement(_Gender["default"], {
        strings: strings
      })), /*#__PURE__*/_react["default"].createElement(FormGroupSection, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h1, {
        variant: _encoreAdvertisingWeb.Type.heading3,
        weight: _encoreAdvertisingWeb.Type.black
      }, headers[LISTENING_BEHAVIOR]), /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
        component: _ReduxFormTargetingSearch["default"],
        debounced: true,
        label: labels[LISTENING_BEHAVIOR].Genres,
        name: "targeting.genres",
        placeholder: placeholders[LISTENING_BEHAVIOR].Genres,
        preloadOptions: true,
        targetingEntity: "genre"
      }), /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
        component: _ReduxFormTargetingSearch["default"],
        debounced: true,
        label: labels[LISTENING_BEHAVIOR].Playlist,
        name: "targeting.moments",
        placeholder: placeholders[LISTENING_BEHAVIOR].Playlist,
        preloadOptions: true,
        targetingEntity: "playlist"
      }), /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
        component: _ReduxFormArtistTargetingSearch["default"],
        debounced: true,
        label: labels[LISTENING_BEHAVIOR].Fans,
        name: "targeting.fanAudiences",
        placeholder: placeholders[LISTENING_BEHAVIOR].Fans
      }), /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
        component: _ReduxFormTargetingCheckboxList["default"],
        label: labels[LISTENING_BEHAVIOR].Interests,
        name: "targeting.segments",
        responseName: "segments",
        podcastTooltipText: strings.PODCAST_TOOLTIP_TEXT,
        segmentNames: releasedSegments,
        fetchUrl: "".concat(this.apiHost, "/adstudio-bff/proto/AdStudioBffAudienceSegmentsService/v1/getAudienceSegments")
      }), /*#__PURE__*/_react["default"].createElement(_Platforms["default"], {
        hideDescription: true,
        creativeFormat: creativeFormat,
        strings: strings
      })));
    }
  }]);

  return Targeting;
}(_react.Component);

exports.Targeting = Targeting;
Targeting.propTypes = {
  advertiserCountry: _propTypes["default"].string,
  apiHost: _propTypes["default"].string,
  edgeHost: _propTypes["default"].string,
  features: _propTypes["default"].objectOf(_propTypes["default"].arrayOf(_propTypes["default"].string)),
  formName: _propTypes["default"].string,
  parentCountry: _propTypes["default"].string,
  disableLocations: _propTypes["default"].bool,
  disableBulkUploadZip: _propTypes["default"].bool,
  disableTargetWholeCountry: _propTypes["default"].bool,
  handleRemoveCountry: _propTypes["default"].func,
  minAge: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  labels: _propTypes["default"].object,
  language: _propTypes["default"].string.isRequired,
  strings: _propTypes["default"].object,
  creativeFormat: _propTypes["default"].string,
  disableCountry: _propTypes["default"].bool
};
Targeting.defaultProps = {
  features: {
    excludeCountries: []
  },
  language: _i18n.DEFAULT_LANGUAGE,
  strings: _objectSpread({
    RELEASED_SEGMENTS: _constants2.RELEASED_SEGMENTS,
    PODCAST_TOOLTIP_TEXT: _constants2.PODCAST_TOOLTIP_TEXT
  }, strings),
  disableCountry: false
};
var _default = Targeting;
exports["default"] = _default;
