// @spotify-internal/encore-web/es/components/Popover/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React from 'react';
import { ThemeProvider } from 'styled-components';
import { overlayDirections } from "../../styles/mixins/overlays";
import { IconX } from "../Icon/IconX";
import { CloseButton } from "./CloseButton";
import { Popover as PopoverElement } from "./Popover";
import { Title } from "./Title";

function PopoverComponent(_ref) {
  var popoverTitle = _ref.popoverTitle,
      children = _ref.children,
      onClose = _ref.onClose,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["popoverTitle", "children", "onClose", "theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(PopoverElement, Object.assign({
    role: "tooltip",
    paddedBody: onClose && !popoverTitle
  }, props), popoverTitle && /*#__PURE__*/React.createElement(Title, null, popoverTitle), children, onClose && /*#__PURE__*/React.createElement(CloseButton, {
    onClick: onClose,
    "aria-label": "Close",
    type: "button"
  }, /*#__PURE__*/React.createElement(IconX, {
    "aria-hidden": "true"
  }))));
}

var popoverHelpers = _objectSpread(_objectSpread({}, overlayDirections), {}, {
  darkTheme: {
    popover: 'dark'
  }
});
/**
* **Popover** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Popover) | [Storybook](https://encore-web.spotify.net/?path=/story/components-popover--default) | [Encore](https://encore.spotify.net/web/components/popover?format=figma-figma#popover) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=522:141) |
*
* A modal used to display some contextual information on top of other content when users click on, keypress on, or touch an element.
*
* @example
* export const Default = () => (
*  <Popover arrow={Popover.topLeft} popoverTitle='Popover "topLeft"' onClose={() => {}}>
*    This is a popover with text and its arrow is aligned to the topLeft
*  </Popover>
*);
*
*/


export var Popover = Object.assign(PopoverComponent, popoverHelpers);
export default Popover;
