// @spotify-internal/encore-web/es/components/ButtonPrimary/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { buttonSizes } from "../../styles";
import { Button } from "./Button";
var ButtonComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var component = _ref.component,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      condensed = _ref.condensed,
      props = _objectWithoutProperties(_ref, ["component", "theme", "condensed"]);

  if (condensed) {
    /* eslint-disable-next-line no-console */
    console.warn("The `condensed` prop had a bug where it was always set to false (https://ghe.spotify.net/encore/web/pull/735/files). Please remove this prop as it's no longer functional.");
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  var rel = props.href && props.target === '_blank' ? 'noopener noreferrer' : null;
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Button, Object.assign({}, props, {
    rel: rel,
    ref: ref,
    component: !component && props.href ? 'a' : component,
    isUsingKeyboard: isUsingKeyboard
  })));
});

var buttonHelpers = _objectSpread(_objectSpread({}, buttonSizes), {}, {
  darkTheme: {
    button: 'dark'
  },
  displayName: 'ButtonPrimary'
});
/**
* **ButtonPrimary** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/ButtonPrimary) | [Storybook](https://encore-web.spotify.net/?path=/story/components-button-buttonprimary--default) | [Encore](https://encore.spotify.net/web/components/button#buttonprimary) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:214) |
*
* Primary buttons are for the action with the highest priority. They use color fills and are green by default.
*
* @example
* export const Default = () => <ButtonPrimary>Default</ButtonPrimary>;
*
*/


export var ButtonPrimary = Object.assign(ButtonComponent, buttonHelpers);
export default ButtonPrimary;
