// @spotify-internal/encore-web/es/components/NavPillPanel/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["active", "role", "tabIndex", "children", "theme"];
import React, { useContext } from 'react';
import styled, { css } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { deprecatedPropWarning } from "../../deprecationWarnings";
import { cssColorValue, semanticColors } from "../../styles";
var NavPillPanelContainer = styled.div.withConfig({
  displayName: "NavPillPanel__NavPillPanelContainer",
  componentId: "sc-3yvzv-0"
})(["", ";"], function (props) {
  return css(["color:", ";&:focus{", "}"], cssColorValue(semanticColors.textBase), !props.isUsingKeyboard ? 'outline: none;' : null);
});
export var NavPillPanel = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var active = _ref.active,
      _ref$role = _ref.role,
      role = _ref$role === void 0 ? 'tabpanel' : _ref$role,
      _ref$tabIndex = _ref.tabIndex,
      tabIndex = _ref$tabIndex === void 0 ? 0 : _ref$tabIndex,
      children = _ref.children,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    deprecatedPropWarning('theme', 'This component is intended to inherit its parent’s color set or color.');
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(NavPillPanelContainer, Object.assign({
    ref: ref,
    role: role,
    hidden: active,
    tabIndex: tabIndex,
    isUsingKeyboard: isUsingKeyboard
  }, props), children);
});
/**
* **NavPillPanel** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/NavPillPanel) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-navpill-navpillpanel--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-pill) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1550) |
*
* null
*
* @example
* () => <NavPillPanel />;
*
*/

export default NavPillPanel;
