// @spotify-internal/encore-foundation/web/index.es6.js
export * from './tokens.es6';
export * from '../themes/themes';
export * from '../themes/sets';

// Temporary type token re-exports to support imports from within Encore Web
// See: https://ghe.spotify.net/encore/web/blob/master/src/core/styles/mixins/type.tsx
// Once these mobile and desktop sizes are no longer in needed in Encore Web
// then these exports can be removed. This is to prevent Encore Web from having
// to import from e.g. '@spotify-internal/encore-foundation/desktop' which was
// causing tree-shaking and bundle size issues.
export { alto as desktopAlto  } from '../desktop';
export { altoBrio as desktopAltoBrio  } from '../desktop';
export { ballad as desktopBallad  } from '../desktop';
export { balladBold as desktopBalladBold  } from '../desktop';
export { bass as desktopBass  } from '../desktop';
export { brio as desktopBrio  } from '../desktop';
export { canon as desktopCanon  } from '../desktop';
export { celloCanon as desktopCelloCanon  } from '../desktop';
export { cello as desktopCello  } from '../desktop';
export { finale as desktopFinale  } from '../desktop';
export { finaleBold as desktopFinaleBold  } from '../desktop';
export { forte as desktopForte  } from '../desktop';
export { mesto as desktopMesto  } from '../desktop';
export { mestoBold as desktopMestoBold  } from '../desktop';
export { metronome as desktopMetronome  } from '../desktop';
export { minuet as desktopMinuet  } from '../desktop';
export { minuetBold as desktopMinuetBold  } from '../desktop';
export { viola as desktopViola  } from '../desktop';
export { violaBold as desktopViolaBold  } from '../desktop';
export { alto as mobileAlto } from '../mobile';
export { altoBrio as mobileAltoBrio } from '../mobile';
export { ballad as mobileBallad } from '../mobile';
export { balladBold as mobileBalladBold } from '../mobile';
export { bass as mobileBass } from '../mobile';
export { brio as mobileBrio } from '../mobile';
export { canon as mobileCanon } from '../mobile';
export { celloCanon as mobileCelloCanon  } from '../mobile';
export { cello as mobileCello } from '../mobile';
export { finale as mobileFinale } from '../mobile';
export { finaleBold as mobileFinaleBold } from '../mobile';
export { forte as mobileForte } from '../mobile';
export { mesto as mobileMesto } from '../mobile';
export { mestoBold as mobileMestoBold } from '../mobile';
export { metronome as mobileMetronome } from '../mobile';
export { minuet as mobileMinuet } from '../mobile';
export { minuetBold as mobileMinuetBold } from '../mobile';
export { viola as mobileViola } from '../mobile';
export { violaBold as mobileViolaBold } from '../mobile'
