// @spotify-internal/spoticharts/cjs/components/OffsetContainer/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const isNull_1 = __importDefault(require("lodash/isNull"));
const isUndefined_1 = __importDefault(require("lodash/isUndefined"));
class OffsetContainer extends react_1.PureComponent {
    render() {
        const _a = this.props, { xPos, yPos, verticalOffset, sideOffset, containerWidth, containerHeight, fullWidth, fullHeight, marginTop, marginRight, marginBottom, marginLeft, children } = _a, rest = __rest(_a, ["xPos", "yPos", "verticalOffset", "sideOffset", "containerWidth", "containerHeight", "fullWidth", "fullHeight", "marginTop", "marginRight", "marginBottom", "marginLeft", "children"]);
        let xOffset;
        let yOffset;
        const chartWidth = fullWidth - marginLeft - marginRight;
        const chartHeight = fullHeight - marginTop - marginBottom;
        yOffset = yPos - containerHeight / 2;
        const collisionHeight = yOffset + containerHeight + verticalOffset;
        if (collisionHeight > chartHeight) {
            yOffset = chartHeight - containerHeight - verticalOffset;
        }
        else if (yOffset < verticalOffset) {
            yOffset = verticalOffset;
        }
        const collisionWidth = xPos + sideOffset + containerWidth + sideOffset;
        if (collisionWidth < chartWidth) {
            xOffset = xPos + sideOffset;
        }
        else {
            xOffset = xPos - sideOffset - containerWidth;
        }
        if (xOffset < 0) {
            xOffset = sideOffset;
        }
        return (react_1.default.createElement("g", { transform: `translate(${xOffset},${yOffset})` }, react_1.default.Children.map(children, child => {
            return isNull_1.default(child) || isUndefined_1.default(child)
                ? null
                : react_1.default.cloneElement(child, Object.assign({}, rest));
        })));
    }
}
exports.default = OffsetContainer;
OffsetContainer.propTypes = {
    xPos: prop_types_1.default.number.isRequired,
    yPos: prop_types_1.default.number,
    verticalOffset: prop_types_1.default.number.isRequired,
    sideOffset: prop_types_1.default.number.isRequired,
    containerWidth: prop_types_1.default.number.isRequired,
    containerHeight: prop_types_1.default.number.isRequired,
    fullWidth: prop_types_1.default.number.isRequired,
    fullHeight: prop_types_1.default.number.isRequired,
    marginTop: prop_types_1.default.number.isRequired,
    marginRight: prop_types_1.default.number.isRequired,
    marginBottom: prop_types_1.default.number.isRequired,
    marginLeft: prop_types_1.default.number.isRequired,
    children: prop_types_1.default.oneOfType([prop_types_1.default.array, prop_types_1.default.object]),
};
OffsetContainer.defaultProps = {
    sideOffset: 22,
    verticalOffset: 12,
    marginTop: 0,
    marginRight: 0,
    marginBottom: 0,
    marginLeft: 0,
};
OffsetContainer.omittedProps = [
    'xPos',
    'yPos',
    'sideMargin',
    'fullWidth',
    'fullHeight',
    'containerWidth',
    'containerHeight',
];
