// @spotify-internal/adstudio-tape/lib/components/PlaybackTimecode/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = PlaybackTimecode;
exports.PlaybackTimecodeContainer = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _secondsToTime = _interopRequireDefault(require("../../utils/helpers/secondsToTime"));

var _encoreWeb = require("@spotify-internal/encore-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var PlaybackTimecodeContainer = _styledComponents["default"].div.withConfig({
  displayName: "PlaybackTimecode__PlaybackTimecodeContainer",
  componentId: "sc-687bni-0"
})(["color:", ";display:flex;align-items:center;justify-content:space-between;"], _encoreFoundation.gray70);

exports.PlaybackTimecodeContainer = PlaybackTimecodeContainer;
var Timecode = (0, _styledComponents["default"])(_encoreWeb.Type).withConfig({
  displayName: "PlaybackTimecode__Timecode",
  componentId: "sc-687bni-1"
})(["padding:2px;"]);

function PlaybackTimecode(_ref) {
  var startTime = _ref.startTime,
      endTime = _ref.endTime,
      _ref$hideDivider = _ref.hideDivider,
      hideDivider = _ref$hideDivider === void 0 ? false : _ref$hideDivider,
      props = _objectWithoutProperties(_ref, ["startTime", "endTime", "hideDivider"]);

  return /*#__PURE__*/_react["default"].createElement(PlaybackTimecodeContainer, props, /*#__PURE__*/_react["default"].createElement(Timecode, {
    variant: _encoreWeb.Type.body3,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, startTime === undefined ? '' : (0, _secondsToTime["default"])(startTime)), !hideDivider && /*#__PURE__*/_react["default"].createElement(Timecode, {
    variant: _encoreWeb.Type.body3,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, "/"), /*#__PURE__*/_react["default"].createElement(Timecode, {
    variant: _encoreWeb.Type.body3,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, endTime === undefined ? '' : (0, _secondsToTime["default"])(endTime)));
}

PlaybackTimecode.propTypes = {
  startTime: _propTypes["default"].number,
  endTime: _propTypes["default"].number,
  hideDivider: _propTypes["default"].bool
};
