// @spotify-internal/encore-web/es/components/NavPill/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { gray10, gray95 } from '@spotify-internal/encore-foundation';
import styled, { css, ThemeProvider } from 'styled-components';
var NavPillPanel = styled.div.withConfig({
  displayName: "NavPill__NavPillPanel",
  componentId: "kouhqe-0"
})(["", ";"], function (props) {
  return css(["color:", ";"], props.theme.navPill === 'dark' ? gray95 : gray10);
});

function NavPillComponent(_ref) {
  var list = _ref.list,
      children = _ref.children,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["list", "children", "theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement("section", props, list, /*#__PURE__*/React.createElement(NavPillPanel, null, children)));
}

var navPillHelpers = {
  darkTheme: {
    navPill: 'dark'
  }
};
/**
* **NavPill** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavPill) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navpill--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-pill) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1550) |
*
* A tab component used to toggle between content within a page
*
* @example
* export const Default = () => (
*  <NavPill
*    list={
*      <NavPillList>
*        <NavPillListItem
*          active
*          label="Monthly Listeners"
*          id="monthly-listeners-tab"
*          aria-controls="monthly-listeners"
*        />
*        <NavPillListItem
*          label="Daily Listeners"
*          id="daily-listeners-tab"
*          aria-controls="daily-listeners"
*        />
*        <NavPillListItem label="Fans" id="fans-tab" aria-controls="fans" />
*        <NavPillListItem label="Disabled" id="disabled-tab" aria-controls="disabled" disabled />
*      </NavPillList>
*    }
*  >
*    <NavPillPanel id="monthly-listeners" aria-labelledby="monthly-listeners-tab">
*      <Type as="p" variant="body2">
*        Monthly Listeners content
*      </Type>
*    </NavPillPanel>
*  </NavPill>
*);
*
*/

export var NavPill = Object.assign(NavPillComponent, navPillHelpers);
export default NavPill;
