// @spotify-internal/adstudio-bff-clients/clients/AdStudioBffAssetService.js

// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";


// Exported root namespace
Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AdStudioBffAssetService = {};
    

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.AdStudioBffAssetService = (function() {
                        
                    /**
                     * Properties of a AdStudioBffAssetService.
                     * @memberof com.spotify.adstudiobff.proto
                     */
                     
                     /**
                      * Fetch function type, set when instantiating this service class.
                      * @callback FetchFunction
                      * @param {string} uri
                      * @param {Object} options
                      * @returns {Promise<any>}
                      */
                     
                    /**
                     * An HttpServiceClient class to send requests to the AdStudioBffAssetService.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc AdStudioBffAssetService to send requests to the AdStudioBffAssetService.
                     * @param {FetchFunction} fetchFn
                     * @param {protorequest.HttpServiceConfigOptions} options
                     * @constructor
                     */
                    AdStudioBffAssetService = function(fetchFn, options) {
                        this.fetchFn = fetchFn;
                        this.host = options.host;
                        this.baseUrl = options.baseUrl;
                        this.version = options.version || "";
                        this.headers = options.headers || new Headers();
                    };
                
                    /**
                     * Sends the fetch request decorated with the configured host.  Only for internal use.
                     */
                    AdStudioBffAssetService.prototype.protoFetch = function(url, opts) {
                      return this.fetchFn(this.host + url, opts);
                    };
                
                    /**
                     * Sets the headers to be sent with all requests to this service.
                     * @function setHeaders
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {Headers} headers
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AdStudioBffAssetService} AdStudioBffAssetService
                     */
                    AdStudioBffAssetService.prototype.setHeaders = function(headers) {
                      this.headers = headers;
                      return this;
                    };
                
                    /**
                     * Get the headers being sent with all requests to this service.
                     * @function getHeaders
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @instance
                     * @returns {Headers}
                     */
                    AdStudioBffAssetService.prototype.getHeaders = function() {
                      return this.headers;
                    };
                
                    /**
                     * Calls AdStudioBffAssetService.createUploadUrl
                     * @function createUploadUrl
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.CreateUploadUrlResponse>} createUploadUrl
                     */
                    AdStudioBffAssetService.prototype.createUploadUrl = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'createUploadUrl';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.approveVoiceover
                     * @function approveVoiceover
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.ApproveVoiceoverRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} approveVoiceover
                     */
                    AdStudioBffAssetService.prototype.approveVoiceover = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'approveVoiceover';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.approveVoiceoverScript
                     * @function approveVoiceoverScript
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.ApproveVoiceoverRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} approveVoiceoverScript
                     */
                    AdStudioBffAssetService.prototype.approveVoiceoverScript = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'approveVoiceoverScript';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.rejectVoiceover
                     * @function rejectVoiceover
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} rejectVoiceover
                     */
                    AdStudioBffAssetService.prototype.rejectVoiceover = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'rejectVoiceover';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.reviseVoiceoverByVoiceoverId
                     * @function reviseVoiceoverByVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.RevisionRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} reviseVoiceoverByVoiceoverId
                     */
                    AdStudioBffAssetService.prototype.reviseVoiceoverByVoiceoverId = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'reviseVoiceoverByVoiceoverId';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.reviseScriptByVoiceoverId
                     * @function reviseScriptByVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.RevisionRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} reviseScriptByVoiceoverId
                     */
                    AdStudioBffAssetService.prototype.reviseScriptByVoiceoverId = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'reviseScriptByVoiceoverId';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.reviseVoiceOverBackground
                     * @function reviseVoiceOverBackground
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} request
                     * @instance
                     * @returns {Promise<google.protobuf.Empty>} reviseVoiceOverBackground
                     */
                    AdStudioBffAssetService.prototype.reviseVoiceOverBackground = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'reviseVoiceOverBackground';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.getMixRequest
                     * @function getMixRequest
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudioasset.MixAssetResponseBatch>} getMixRequest
                     */
                    AdStudioBffAssetService.prototype.getMixRequest = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'getMixRequest';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.createMixPreview
                     * @function createMixPreview
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.AuthCreateMixPreviewResponse>} createMixPreview
                     */
                    AdStudioBffAssetService.prototype.createMixPreview = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'createMixPreview';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAssetService.isMixPreviewComplete
                     * @function isMixPreviewComplete
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAssetService
                     * @param {com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteResponse>} isMixPreviewComplete
                     */
                    AdStudioBffAssetService.prototype.isMixPreviewComplete = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAssetService/' + this.version +  (this.version ? '/' : '') + 'isMixPreviewComplete';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    return AdStudioBffAssetService;
                })();

                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();


exports["default"] = $root;
exports["$root"] = $root;
exports["AdStudioBffAssetService"] = AdStudioBffAssetService;
