// @spotify-internal/adstudio-bff-clients/clients/AdStudioBffAudienceSegmentsService.js

// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";


// Exported root namespace
Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AdStudioBffAudienceSegmentsService = {};
    

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.AdStudioBffAudienceSegmentsService = (function() {
                        
                    /**
                     * Properties of a AdStudioBffAudienceSegmentsService.
                     * @memberof com.spotify.adstudiobff.proto
                     */
                     
                     /**
                      * Fetch function type, set when instantiating this service class.
                      * @callback FetchFunction
                      * @param {string} uri
                      * @param {Object} options
                      * @returns {Promise<any>}
                      */
                     
                    /**
                     * An HttpServiceClient class to send requests to the AdStudioBffAudienceSegmentsService.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc AdStudioBffAudienceSegmentsService to send requests to the AdStudioBffAudienceSegmentsService.
                     * @param {FetchFunction} fetchFn
                     * @param {protorequest.HttpServiceConfigOptions} options
                     * @constructor
                     */
                    AdStudioBffAudienceSegmentsService = function(fetchFn, options) {
                        this.fetchFn = fetchFn;
                        this.host = options.host;
                        this.baseUrl = options.baseUrl;
                        this.version = options.version || "";
                        this.headers = options.headers || new Headers();
                    };
                
                    /**
                     * Sends the fetch request decorated with the configured host.  Only for internal use.
                     */
                    AdStudioBffAudienceSegmentsService.prototype.protoFetch = function(url, opts) {
                      return this.fetchFn(this.host + url, opts);
                    };
                
                    /**
                     * Sets the headers to be sent with all requests to this service.
                     * @function setHeaders
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @param {Headers} headers
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService} AdStudioBffAudienceSegmentsService
                     */
                    AdStudioBffAudienceSegmentsService.prototype.setHeaders = function(headers) {
                      this.headers = headers;
                      return this;
                    };
                
                    /**
                     * Get the headers being sent with all requests to this service.
                     * @function getHeaders
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @instance
                     * @returns {Headers}
                     */
                    AdStudioBffAudienceSegmentsService.prototype.getHeaders = function() {
                      return this.headers;
                    };
                
                    /**
                     * Calls AdStudioBffAudienceSegmentsService.getAudienceSegments
                     * @function getAudienceSegments
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @param {google.protobuf.Empty} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.GetAudienceSegmentsResponse>} getAudienceSegments
                     */
                    AdStudioBffAudienceSegmentsService.prototype.getAudienceSegments = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAudienceSegmentsService/' + this.version +  (this.version ? '/' : '') + 'getAudienceSegments';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAudienceSegmentsService.createAudience
                     * @function createAudience
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @param {com.spotify.adstudiobff.proto.CreateAudienceRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.CreateAudienceResponse>} createAudience
                     */
                    AdStudioBffAudienceSegmentsService.prototype.createAudience = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAudienceSegmentsService/' + this.version +  (this.version ? '/' : '') + 'createAudience';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAudienceSegmentsService.uploadAudience
                     * @function uploadAudience
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @param {com.spotify.adstudiobff.proto.UploadAudienceRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.UploadAudienceResponse>} uploadAudience
                     */
                    AdStudioBffAudienceSegmentsService.prototype.uploadAudience = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAudienceSegmentsService/' + this.version +  (this.version ? '/' : '') + 'uploadAudience';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdStudioBffAudienceSegmentsService.getAudiences
                     * @function getAudiences
                     * @memberof com.spotify.adstudiobff.proto.AdStudioBffAudienceSegmentsService
                     * @param {com.spotify.adstudiobff.proto.GetAudiencesRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.adstudiobff.proto.GetAudiencesBffResponse>} getAudiences
                     */
                    AdStudioBffAudienceSegmentsService.prototype.getAudiences = function(request) {
                        const url = '/' + this.baseUrl + '/AdStudioBffAudienceSegmentsService/' + this.version +  (this.version ? '/' : '') + 'getAudiences';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    return AdStudioBffAudienceSegmentsService;
                })();

                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();


exports["default"] = $root;
exports["$root"] = $root;
exports["AdStudioBffAudienceSegmentsService"] = AdStudioBffAudienceSegmentsService;
