// @spotify-internal/checkout-sdk/esm/types.js
export var InputValidationType;
(function (InputValidationType) {
    InputValidationType["MinLength"] = "minLength";
    InputValidationType["MaxLength"] = "maxLength";
    InputValidationType["Regex"] = "regex";
    InputValidationType["Required"] = "required";
    InputValidationType["Mask"] = "mask";
})(InputValidationType || (InputValidationType = {}));
export var FormLayout;
(function (FormLayout) {
    FormLayout["Pci"] = "pci";
    FormLayout["Row"] = "row";
    FormLayout["Input"] = "input";
    FormLayout["Dropdown"] = "dropdown";
    FormLayout["Text"] = "text";
    FormLayout["StoredPaymentDetails"] = "storedPaymentDetails";
    FormLayout["RadioContent"] = "radioContent";
    FormLayout["KlarnaPrepare"] = "klarnaPrepare";
    FormLayout["KlarnaWidget"] = "klarnaWidget";
    FormLayout["None"] = "none";
})(FormLayout || (FormLayout = {}));
export var TooltipKeys;
(function (TooltipKeys) {
    TooltipKeys["CvcInfo"] = "cvc_info";
    TooltipKeys["SecurityExplained"] = "security_explained";
})(TooltipKeys || (TooltipKeys = {}));
export var RemoteConfigProperties;
(function (RemoteConfigProperties) {
    RemoteConfigProperties["enableUxImprovements"] = "enable_sca_interstitial";
})(RemoteConfigProperties || (RemoteConfigProperties = {}));
export var FlowStep;
(function (FlowStep) {
    FlowStep["Prepare"] = "prepare";
    FlowStep["Proceed"] = "proceed";
})(FlowStep || (FlowStep = {}));
export var FormPosition;
(function (FormPosition) {
    FormPosition["above"] = "above";
    FormPosition["below"] = "below";
})(FormPosition || (FormPosition = {}));
