// @spotify-internal/adstudio-tape/lib/components/Uploader/LoadingDisplay.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = LoadingDisplay;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _encoreWeb = require("@spotify-internal/encore-web");

var _Panel = _interopRequireDefault(require("../Panel"));

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "LoadingDisplay__Container",
  componentId: "sc-1nn93gl-0"
})(["width:100%;"]);

var TrackName = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "LoadingDisplay__TrackName",
  componentId: "sc-1nn93gl-1"
})(["padding-bottom:10px;position:relative;"]);

var ProgressBarBackground = _styledComponents["default"].div.withConfig({
  displayName: "LoadingDisplay__ProgressBarBackground",
  componentId: "sc-1nn93gl-2"
})(["background-color:", ";border-radius:2px;height:4px;width:100%;"], _encoreFoundation.gray90);

var ProgressBar = _styledComponents["default"].div.withConfig({
  displayName: "LoadingDisplay__ProgressBar",
  componentId: "sc-1nn93gl-3"
})(["background-color:", ";border-radius:2px;height:4px;width:", ";"], _encoreFoundation.gray60, function (props) {
  return "".concat(props.progressInt || 0, "%");
});

var StatusText = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "LoadingDisplay__StatusText",
  componentId: "sc-1nn93gl-4"
})(["padding-top:10px;"]);

function LoadingDisplay(_ref) {
  var name = _ref.name,
      progress = _ref.progress,
      uploadingText = _ref.uploadingText,
      processingText = _ref.processingText;
  var progressInt = Math.max(Math.min(parseInt(progress, 10), 100), 0);
  var statusText = progressInt === 100 ? "".concat(processingText, "...") : "".concat(uploadingText, "...");
  return /*#__PURE__*/_react["default"].createElement(_Panel["default"], {
    shadow: true
  }, /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(TrackName, {
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book
  }, name), /*#__PURE__*/_react["default"].createElement(ProgressBarBackground, null, /*#__PURE__*/_react["default"].createElement(ProgressBar, {
    progressInt: progressInt
  })), /*#__PURE__*/_react["default"].createElement(StatusText, {
    color: _encoreFoundation.gray75,
    variant: _encoreWeb.Type.body3,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, statusText)));
}

LoadingDisplay.propTypes = {
  name: _propTypes["default"].string,
  progress: _propTypes["default"].number,
  uploadingText: _propTypes["default"].string,
  processingText: _propTypes["default"].string
};
LoadingDisplay.defaultProps = {
  uploadingText: _constants.UPLOADING,
  processingText: _constants.PROCESSING
};
