// @spotify-internal/adstudio-shared/lib/components/AdTargeting/AgeRange/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _reactRedux = require("react-redux");

var _reduxForm = require("redux-form");

var _AgeRange = require("./AgeRange");

var _age = require("../../../config/age");

var _constants = require("./constants");

var _utils = require("../utils");

function mapStateToProps(state, ownProps) {
  var selector = (0, _reduxForm.formValueSelector)(ownProps.formName); // TODO: getMinTargetingAgeByCountry should be considered deprecated - Voltron will supply minAge
  //  from config service.

  var minAge = ownProps.minAge || (0, _age.getMinTargetingAgeByCountry)(ownProps.advertiserCountry);
  var formAgeRangeStart = selector(state, _constants.REDUX_FORM_START_FIELD_NAME);
  var formAgeRangeEnd = selector(state, _constants.REDUX_FORM_END_FIELD_NAME);
  var selectedAgeRangeStart = Math.max(Number(formAgeRangeStart || minAge), Number(minAge));

  var selectedAgeRangeEnd = formAgeRangeEnd || _age.TARGETING_MAX_AGE.toString();

  return {
    endAge: selectedAgeRangeEnd,
    endRange: (0, _utils.generateAgeRange)('end', selectedAgeRangeStart, _age.TARGETING_MAX_AGE),
    formAgeRangeStart: formAgeRangeStart,
    formMinAge: minAge,
    startAge: selectedAgeRangeStart,
    startRange: (0, _utils.generateAgeRange)('start', minAge, selectedAgeRangeEnd),
    startAgeFieldName: _constants.REDUX_FORM_START_FIELD_NAME,
    endAgeFieldName: _constants.REDUX_FORM_END_FIELD_NAME,
    FieldComponent: _reduxForm.Field
  };
}

function mapDispatchToProps(dispatch, ownProps) {
  return {
    updateEndAge: function updateEndAge(endAge) {
      return dispatch((0, _reduxForm.change)(ownProps.formName, _constants.REDUX_FORM_END_FIELD_NAME, endAge));
    },
    updateStartAge: function updateStartAge(startAge) {
      return dispatch((0, _reduxForm.change)(ownProps.formName, _constants.REDUX_FORM_START_FIELD_NAME, startAge));
    }
  };
}

var _default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(_AgeRange.AgeRange);

exports["default"] = _default;
