// @spotify-internal/adstudio-tape/lib/components/TabsListItem/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _Type = _interopRequireDefault(require("../Type"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var TabsListItemListItem = _styledComponents["default"].li.withConfig({
  displayName: "TabsListItem__TabsListItemListItem",
  componentId: "fioma9-0"
})(["display:inline-block;list-style-type:none;position:relative;"]);

var TabsListItemAnchor = _styledComponents["default"].a.withConfig({
  displayName: "TabsListItem__TabsListItemAnchor",
  componentId: "fioma9-1"
})(["color:", ";", ""], _encoreFoundation.black, function (_ref) {
  var isActive = _ref.isActive;
  return isActive && "\n    ::after {\n      background-color: ".concat(function (props) {
    return props.theme.colors.primaryColor;
  }, ";\n      content: \"\";\n      display: block;\n      height: 2px;\n      margin: 0 auto;\n      position: relative;\n      top: 0;\n      transition: background-color 0.2s;\n      width: 30px;\n    }\n  ");
});

var DisabledType = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "TabsListItem__DisabledType",
  componentId: "fioma9-2"
})(["color:", ";cursor:not-allowed;"], _encoreFoundation.gray45);

var TabsListItem = function TabsListItem(_ref2) {
  var isActive = _ref2.isActive,
      disabled = _ref2.disabled,
      label = _ref2.label,
      href = _ref2.href,
      rest = _objectWithoutProperties(_ref2, ["isActive", "disabled", "label", "href"]);

  return /*#__PURE__*/_react["default"].createElement(TabsListItemListItem, rest, /*#__PURE__*/_react["default"].createElement(TabsListItemAnchor, {
    isActive: isActive,
    href: href
  }, disabled ? /*#__PURE__*/_react["default"].createElement(DisabledType, {
    element: "p",
    variant: "cta-small"
  }, label) : /*#__PURE__*/_react["default"].createElement(_Type["default"], {
    element: "p",
    variant: "cta-small"
  }, label)));
};

TabsListItem.propTypes = {
  isActive: _propTypes["default"].bool,
  disabled: _propTypes["default"].bool,
  label: _propTypes["default"].string,
  href: _propTypes["default"].string
};
var _default = TabsListItem;
exports["default"] = _default;
