// @spotify-internal/ubi-sdk-s4a-web-team-artists/WebTeamArtistsEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamArtistsEventFactory = exports.createWebTeamArtistsEventFactory = void 0;
var SPECIFICATION_ID = "web-team-artists";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.1.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.24";
function createWebTeamArtistsEventFactory(options) {
    return new WebTeamArtistsEventFactory(options);
}
exports.createWebTeamArtistsEventFactory = createWebTeamArtistsEventFactory;
/**
 *
 */
var WebTeamArtistsEventFactory = /** @class */ (function () {
    function WebTeamArtistsEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of ArtistsSearchInputEventFactory
    */
    WebTeamArtistsEventFactory.prototype.artistsSearchInputFactory = function () {
        return new WebTeamArtistsEventFactory.ArtistsSearchInputEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ArtistTableRowEventFactory
    */
    WebTeamArtistsEventFactory.prototype.artistTableRowFactory = function () {
        return new WebTeamArtistsEventFactory.ArtistTableRowEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SearchArtistsLinkEventFactory
    */
    WebTeamArtistsEventFactory.prototype.searchArtistsLinkFactory = function () {
        return new WebTeamArtistsEventFactory.SearchArtistsLinkEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of NoSearchResultsMessageEventFactory
    */
    WebTeamArtistsEventFactory.prototype.noSearchResultsMessageFactory = function () {
        return new WebTeamArtistsEventFactory.NoSearchResultsMessageEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of EmptyStateMessageEventFactory
    */
    WebTeamArtistsEventFactory.prototype.emptyStateMessageFactory = function () {
        return new WebTeamArtistsEventFactory.EmptyStateMessageEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamArtistsEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamArtistsEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamArtistsEventFactory;
}());
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    /**
     * Create ArtistsSearchInputEventFactory class with path
     * @param path to this node
     */
    var ArtistsSearchInputEventFactory = /** @class */ (function () {
        function ArtistsSearchInputEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "artists_search_input"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        ArtistsSearchInputEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ArtistsSearchInputEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for key_stroke search action.
         */
        ArtistsSearchInputEventFactory.prototype.keyStrokeSearch = function () {
            var data = {
                actionName: "search",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "key_stroke",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ArtistsSearchInputEventFactory;
    }());
    WebTeamArtistsEventFactory.ArtistsSearchInputEventFactory = ArtistsSearchInputEventFactory;
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    /**
     * Create ArtistTableRowEventFactory class with path
     * @param path to this node
     */
    var ArtistTableRowEventFactory = /** @class */ (function () {
        function ArtistTableRowEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "artist_table_row"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of ManageArtistButtonEventFactory
        */
        ArtistTableRowEventFactory.prototype.manageArtistButtonFactory = function () {
            return new WebTeamArtistsEventFactory.ArtistTableRowEventFactory.ManageArtistButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        ArtistTableRowEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ArtistTableRowEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return ArtistTableRowEventFactory;
    }());
    WebTeamArtistsEventFactory.ArtistTableRowEventFactory = ArtistTableRowEventFactory;
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    /**
     * Create SearchArtistsLinkEventFactory class with path
     * @param path to this node
     */
    var SearchArtistsLinkEventFactory = /** @class */ (function () {
        function SearchArtistsLinkEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "search_artists_link"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        SearchArtistsLinkEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SearchArtistsLinkEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit quick_scroll action.
         */
        SearchArtistsLinkEventFactory.prototype.hitQuickScroll = function () {
            var data = {
                actionName: "quick_scroll",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SearchArtistsLinkEventFactory;
    }());
    WebTeamArtistsEventFactory.SearchArtistsLinkEventFactory = SearchArtistsLinkEventFactory;
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    /**
     * Create NoSearchResultsMessageEventFactory class with path
     * @param path to this node
     */
    var NoSearchResultsMessageEventFactory = /** @class */ (function () {
        function NoSearchResultsMessageEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "no_search_results_message"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        NoSearchResultsMessageEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        NoSearchResultsMessageEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for NoSearchResultsMessageEventFactory.
         */
        NoSearchResultsMessageEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return NoSearchResultsMessageEventFactory;
    }());
    WebTeamArtistsEventFactory.NoSearchResultsMessageEventFactory = NoSearchResultsMessageEventFactory;
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    /**
     * Create EmptyStateMessageEventFactory class with path
     * @param path to this node
     */
    var EmptyStateMessageEventFactory = /** @class */ (function () {
        function EmptyStateMessageEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "empty_state_message"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of AddArtistLinkEventFactory
        */
        EmptyStateMessageEventFactory.prototype.addArtistLinkFactory = function () {
            return new WebTeamArtistsEventFactory.EmptyStateMessageEventFactory.AddArtistLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        EmptyStateMessageEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        EmptyStateMessageEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for EmptyStateMessageEventFactory.
         */
        EmptyStateMessageEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return EmptyStateMessageEventFactory;
    }());
    WebTeamArtistsEventFactory.EmptyStateMessageEventFactory = EmptyStateMessageEventFactory;
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    var ArtistTableRowEventFactory;
    (function (ArtistTableRowEventFactory) {
        /**
         * Create ManageArtistButtonEventFactory class with path
         * @param path to this node
         */
        var ManageArtistButtonEventFactory = /** @class */ (function () {
            function ManageArtistButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "manage_artist_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of SeeTeamMembersButtonEventFactory
            */
            ManageArtistButtonEventFactory.prototype.seeTeamMembersButtonFactory = function () {
                return new WebTeamArtistsEventFactory.ArtistTableRowEventFactory.ManageArtistButtonEventFactory.SeeTeamMembersButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            ManageArtistButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ManageArtistButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            ManageArtistButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ManageArtistButtonEventFactory;
        }());
        ArtistTableRowEventFactory.ManageArtistButtonEventFactory = ManageArtistButtonEventFactory;
    })(ArtistTableRowEventFactory = WebTeamArtistsEventFactory.ArtistTableRowEventFactory || (WebTeamArtistsEventFactory.ArtistTableRowEventFactory = {}));
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    var EmptyStateMessageEventFactory;
    (function (EmptyStateMessageEventFactory) {
        /**
         * Create AddArtistLinkEventFactory class with path
         * @param path to this node
         */
        var AddArtistLinkEventFactory = /** @class */ (function () {
            function AddArtistLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "add_artist_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AddArtistLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AddArtistLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            AddArtistLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AddArtistLinkEventFactory;
        }());
        EmptyStateMessageEventFactory.AddArtistLinkEventFactory = AddArtistLinkEventFactory;
    })(EmptyStateMessageEventFactory = WebTeamArtistsEventFactory.EmptyStateMessageEventFactory || (WebTeamArtistsEventFactory.EmptyStateMessageEventFactory = {}));
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
(function (WebTeamArtistsEventFactory) {
    var ArtistTableRowEventFactory;
    (function (ArtistTableRowEventFactory) {
        var ManageArtistButtonEventFactory;
        (function (ManageArtistButtonEventFactory) {
            /**
             * Create SeeTeamMembersButtonEventFactory class with path
             * @param path to this node
             */
            var SeeTeamMembersButtonEventFactory = /** @class */ (function () {
                function SeeTeamMembersButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "see_team_members_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                SeeTeamMembersButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                SeeTeamMembersButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                SeeTeamMembersButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return SeeTeamMembersButtonEventFactory;
            }());
            ManageArtistButtonEventFactory.SeeTeamMembersButtonEventFactory = SeeTeamMembersButtonEventFactory;
        })(ManageArtistButtonEventFactory = ArtistTableRowEventFactory.ManageArtistButtonEventFactory || (ArtistTableRowEventFactory.ManageArtistButtonEventFactory = {}));
    })(ArtistTableRowEventFactory = WebTeamArtistsEventFactory.ArtistTableRowEventFactory || (WebTeamArtistsEventFactory.ArtistTableRowEventFactory = {}));
})(WebTeamArtistsEventFactory = exports.WebTeamArtistsEventFactory || (exports.WebTeamArtistsEventFactory = {}));
exports.WebTeamArtistsEventFactory = WebTeamArtistsEventFactory;
