// @spotify-internal/adstudio-tape/lib/components/MobilePreview/styled/CTAButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _ButtonSecondary = _interopRequireDefault(require("../../ButtonSecondary"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var StyledButton = (0, _styledComponents["default"])(_ButtonSecondary["default"]).withConfig({
  displayName: "CTAButton__StyledButton",
  componentId: "sc-3mkb8a-0"
})(["background-color:white;border:none;color:black;margin-top:15px;padding-left:16px;padding-right:16px;pointer-events:none;"]);

var CTAButton = function CTAButton(_ref) {
  var ctaText = _ref.ctaText;
  return /*#__PURE__*/_react["default"].createElement(StyledButton, {
    color: "white",
    buttonSize: "sm"
  }, ctaText);
};

CTAButton.propTypes = {
  ctaText: _propTypes["default"].string.isRequired
};
var _default = CTAButton;
exports["default"] = _default;
