// @spotify-internal/adstudio-tape/lib/components/MobilePreview/styled/Header.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var HeaderWrapper = _styledComponents["default"].div.withConfig({
  displayName: "Header__HeaderWrapper",
  componentId: "zp2sgw-0"
})(["display:flex;flex-direction:column;padding:12px 0 8px;", " & p:last-of-type{letter-spacing:0.1em;}"], function (props) {
  return props.withoutHeadline && (0, _styledComponents.css)(["& p:first-of-type{padding-bottom:", ";}"], _encoreFoundation.spacer16);
});

var InlineType = (0, _styledComponents["default"])(_encoreWeb.Type.span).withConfig({
  displayName: "Header__InlineType",
  componentId: "zp2sgw-1"
})(["display:inline;"]);

var createHeadlineTagLine = function createHeadlineTagLine(headline, tagLine) {
  var output;

  if (tagLine) {
    output = /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
      variant: _encoreWeb.Type.body4,
      weight: _encoreWeb.Type.bold,
      condensed: true
    }, /*#__PURE__*/_react["default"].createElement(InlineType, {
      variant: _encoreWeb.Type.body4,
      weight: _encoreWeb.Type.bold,
      condensed: true
    }, "".concat(headline, ": ")), /*#__PURE__*/_react["default"].createElement(InlineType, {
      variant: _encoreWeb.Type.body4,
      weight: _encoreWeb.Type.book,
      condensed: true
    }, tagLine));
  } else {
    output = /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
      variant: _encoreWeb.Type.body4,
      weight: _encoreWeb.Type.bold,
      condensed: true
    }, headline);
  }

  return output;
};

var Header = function Header(_ref) {
  var title = _ref.title,
      headline = _ref.headline,
      tagLine = _ref.tagLine;
  return /*#__PURE__*/_react["default"].createElement(HeaderWrapper, {
    withoutHeadline: !headline
  }, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    variant: _encoreWeb.Type.body4,
    weight: _encoreWeb.Type.book,
    condensed: !!headline
  }, title), createHeadlineTagLine(headline, tagLine));
};

Header.propTypes = {
  title: _propTypes["default"].string.isRequired,
  headline: _propTypes["default"].string.isRequired,
  tagLine: _propTypes["default"].string
};
var _default = Header;
exports["default"] = _default;
