// @spotify-internal/adstudio-tape/lib/components/MobilePreview/styled/PlaybackControls.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _PlayPauseButton = _interopRequireDefault(require("../../PlayPauseButton"));

var _Icons = _interopRequireDefault(require("../../Icons"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconSeek = _Icons["default"].IconSeek,
    IconSkipBack15 = _Icons["default"].IconSkipBack15,
    IconSkipForward15 = _Icons["default"].IconSkipForward15;
var StyledPlayPause = (0, _styledComponents["default"])(_PlayPauseButton["default"]).withConfig({
  displayName: "PlaybackControls__StyledPlayPause",
  componentId: "sc-1o832pm-0"
})(["color:", ";margin:0 24px;pointer-events:none;"], _encoreFoundation.white);

var ControlsWrapper = _styledComponents["default"].div.withConfig({
  displayName: "PlaybackControls__ControlsWrapper",
  componentId: "sc-1o832pm-1"
})(["display:flex;align-items:center;margin:10px auto 0;justify-content:center;"]);

var PlaybackControls = function PlaybackControls(_ref) {
  var showSkipControls = _ref.showSkipControls,
      showPodcastControls = _ref.showPodcastControls;
  var skipControlColor = showSkipControls ? _encoreFoundation.white : _encoreFoundation.gray20;
  return /*#__PURE__*/_react["default"].createElement(ControlsWrapper, null, showPodcastControls ? /*#__PURE__*/_react["default"].createElement(IconSkipBack15, {
    color: _encoreFoundation.gray20,
    iconSize: _encoreFoundation.spacer20
  }) : /*#__PURE__*/_react["default"].createElement(IconSeek, {
    fill: _encoreFoundation.gray20,
    direction: "prev"
  }), /*#__PURE__*/_react["default"].createElement(StyledPlayPause, {
    circle: true,
    playing: true,
    solid: true,
    size: 64
  }), showPodcastControls ? /*#__PURE__*/_react["default"].createElement(IconSkipForward15, {
    color: skipControlColor,
    iconSize: _encoreFoundation.spacer20
  }) : /*#__PURE__*/_react["default"].createElement(IconSeek, {
    fill: skipControlColor,
    direction: "next"
  }));
};

PlaybackControls.propTypes = {
  showSkipControls: _propTypes["default"].bool.isRequired,
  showPodcastControls: _propTypes["default"].bool
};
var _default = PlaybackControls;
exports["default"] = _default;
