// @spotify-internal/adstudio-tape/lib/components/MobilePreview/styled/ProgressBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ProgressBarWrapper = _styledComponents["default"].div.withConfig({
  displayName: "ProgressBar__ProgressBarWrapper",
  componentId: "r6j43z-0"
})(["align-items:center;display:grid;grid-template-columns:1fr 200px 1fr;margin-top:15px;& span{font-size:", ";}"], _encoreFoundation.spacer8);

var ProgressBarFull = _styledComponents["default"].div.withConfig({
  displayName: "ProgressBar__ProgressBarFull",
  componentId: "r6j43z-1"
})(["background-color:", ";border:0 solid transparent;border-radius:2px;display:flex;flex-grow:11;height:3px;margin-bottom:1px;"], _encoreFoundation.gray30);

var ProgressBarElapsed = _styledComponents["default"].div.withConfig({
  displayName: "ProgressBar__ProgressBarElapsed",
  componentId: "r6j43z-2"
})(["background-color:", ";border:0 solid transparent;border-radius:2px;flex-grow:0.3;height:3px;"], _encoreFoundation.white);

var ProgressBar = function ProgressBar() {
  return /*#__PURE__*/_react["default"].createElement(ProgressBarWrapper, null, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type, {
    variant: _encoreWeb.Type.body4,
    weight: _encoreWeb.Type.book,
    color: _encoreFoundation.gray80,
    condensed: true
  }, "0:06"), /*#__PURE__*/_react["default"].createElement(ProgressBarFull, null, /*#__PURE__*/_react["default"].createElement(ProgressBarElapsed, null)), /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type, {
    variant: _encoreWeb.Type.body4,
    weight: _encoreWeb.Type.book,
    color: _encoreFoundation.gray80,
    condensed: true
  }, "0:25"));
};

var _default = ProgressBar;
exports["default"] = _default;
