// @spotify-internal/spoticharts/cjs/components/AreaChart/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AreaChart = void 0;
const prop_types_1 = __importDefault(require("prop-types"));
const react_1 = __importStar(require("react"));
const d3_shape_1 = require("d3-shape");
const reactochart_1 = require("reactochart");
class AreaChart extends react_1.Component {
    render() {
        const _a = this.props, { curve, pathStyle, fill, opacity } = _a, rest = __rest(_a, ["curve", "pathStyle", "fill", "opacity"]);
        const newPathStyle = Object.assign(Object.assign({}, pathStyle), { fill, opacity });
        return (react_1.default.createElement(reactochart_1.AreaChart, Object.assign({ curve: curve ? curve : d3_shape_1.curveLinear, pathStyle: newPathStyle }, rest)));
    }
}
exports.AreaChart = AreaChart;
AreaChart.propTypes = {
    curve: prop_types_1.default.oneOfType([prop_types_1.default.func, prop_types_1.default.bool]),
    fill: prop_types_1.default.string,
    opacity: prop_types_1.default.number,
    pathStyle: prop_types_1.default.object,
    x: prop_types_1.default.func,
    y: prop_types_1.default.func,
    yEnd: prop_types_1.default.func,
};
AreaChart.defaultProps = {
    curve: d3_shape_1.curveMonotoneX,
    fill: 'black',
    opacity: 0.2,
    x: d => d.x,
    y: d => d.y0,
    yEnd: d => d.y1 || 0,
};
exports.default = AreaChart;
