// @spotify-internal/encore-web/es/components/TableContainer/TableContainer.js
import { black, gray10, gray90, screenXsMax, white } from '@spotify-internal/encore-foundation';
import styled, { css } from 'styled-components';
import { TableHeaderCellElement as TableHeaderCell } from "../TableHeaderCell/TableHeaderCell";

var isDarkTheme = function isDarkTheme(theme) {
  return theme.tableHeaderCell === 'dark';
};

var boxShadow = function boxShadow(theme) {
  return isDarkTheme(theme) ? "box-shadow: -3px 0 0 ".concat(gray10, ", 3px 0 0 ").concat(gray10, ", 0 1px 6px ").concat(black) : "box-shadow: -3px 0 0 ".concat(white, ", 3px 0 0 ").concat(white, ", 0 1px 3px ").concat(gray90);
};

var responsiveWrapper = function responsiveWrapper() {
  return css(["@media screen and (max-width:", "){display:block;width:100%;overflow-x:auto;-webkit-overflow-scrolling:touch;-ms-overflow-style:-ms-autohiding-scrollbar;}"], screenXsMax);
};

var stickyHeaderWrapper = function stickyHeaderWrapper() {
  return css(["overflow:auto;max-height:100%;", "{background-color:", ";position:sticky;top:0;", ";}"], TableHeaderCell, function (props) {
    return isDarkTheme(props.theme) ? gray10 : white;
  }, function (props) {
    return props.isSticky && css(["", ";"], boxShadow(props.theme));
  });
};

export var Container = styled.div.withConfig({
  displayName: "TableContainer__Container",
  componentId: "sc-1lhepyy-0"
})(["", ";", ";"], function (props) {
  return props.responsive && responsiveWrapper;
}, function (props) {
  return props.stickyHeader && stickyHeaderWrapper;
});
