// @spotify-internal/adstudio-tape/lib/components/VideoDesktopPreview/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _VideoPlayer = _interopRequireDefault(require("../VideoPlayer"));

var _TooltipInfo = _interopRequireDefault(require("../TooltipInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var PreviewBody = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__PreviewBody",
  componentId: "sc-375fkr-0"
})(["position:relative;margin-right:1.5em;width:100%;border-radius:4px;box-shadow:0 3px 9px 0 ", ";"], _encoreFoundation.gray90);

var Panel = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__Panel",
  componentId: "sc-375fkr-1"
})(["border-radius:4px;box-shadow:0 3px 9px 0 ", ";padding:20px 15px;width:100%;"], _encoreFoundation.gray90);

var DesktopImagePreview = _styledComponents["default"].img.withConfig({
  displayName: "VideoDesktopPreview__DesktopImagePreview",
  componentId: "sc-375fkr-2"
})(["top:0;left:-1px;width:100%;max-width:initial;"]);

var ImageWrapper = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__ImageWrapper",
  componentId: "sc-375fkr-3"
})(["width:24.3%;position:absolute;z-index:1000;top:52%;left:0;"]);

var ImageTooltip = (0, _styledComponents["default"])(_TooltipInfo["default"]).withConfig({
  displayName: "VideoDesktopPreview__ImageTooltip",
  componentId: "sc-375fkr-4"
})(["position:absolute;margin-top:", ";margin-left:", ";svg{margin-left:0;color:", ";&:hover{color:", ";}}"], _encoreFoundation.spacer4, _encoreFoundation.spacer4, _encoreFoundation.gray50, _encoreFoundation.white);

var VideoWrapper = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__VideoWrapper",
  componentId: "sc-375fkr-5"
})(["position:absolute;top:28%;left:35%;width:50.2%;"]);

var HeaderWrapper = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__HeaderWrapper",
  componentId: "sc-375fkr-6"
})(["width:24.3%;position:absolute;z-index:1000;top:89.8%;padding-left:4px;"]);

var HeaderText = _styledComponents["default"].div.withConfig({
  displayName: "VideoDesktopPreview__HeaderText",
  componentId: "sc-375fkr-7"
})(["padding-bottom:0;color:", ";font-size:", ";min-height:1.4em;line-height:1.4;font-weight:bold;white-space:nowrap;overflow:hidden;"], function (props) {
  return props.color ? props.color : _encoreFoundation.white;
}, function (props) {
  return props.fontSize ? props.fontSize : '6px';
});

var VideoDesktopPreview = function VideoDesktopPreview(_ref) {
  var imageUrl = _ref.imageUrl,
      videoUrl = _ref.videoUrl,
      showPanel = _ref.showPanel,
      title = _ref.title,
      headline = _ref.headline,
      headerFontSize = _ref.headerFontSize,
      imageTooltipMessage = _ref.imageTooltipMessage;

  var body = /*#__PURE__*/_react["default"].createElement(PreviewBody, null, /*#__PURE__*/_react["default"].createElement(DesktopImagePreview, {
    alt: "desktop-preview",
    src: "https://adstudio.scdn.co/assets/video-desktop-preview.png"
  }), /*#__PURE__*/_react["default"].createElement(ImageWrapper, {
    "data-test": "ta-image-preview"
  }, imageTooltipMessage && /*#__PURE__*/_react["default"].createElement(ImageTooltip, {
    tooltipText: imageTooltipMessage,
    triggerOnMouse: true
  }), imageUrl && /*#__PURE__*/_react["default"].createElement("img", {
    alt: "Upload Preview",
    src: imageUrl,
    width: "100%"
  })), /*#__PURE__*/_react["default"].createElement(VideoWrapper, null, videoUrl && /*#__PURE__*/_react["default"].createElement(_VideoPlayer["default"], {
    width: "100%",
    height: "100%",
    videoUrl: videoUrl
  })), /*#__PURE__*/_react["default"].createElement(HeaderWrapper, null, /*#__PURE__*/_react["default"].createElement(HeaderText, {
    fontSize: headerFontSize
  }, title), /*#__PURE__*/_react["default"].createElement(HeaderText, {
    color: _encoreFoundation.gray75,
    fontSize: headerFontSize
  }, headline)));

  if (showPanel) return /*#__PURE__*/_react["default"].createElement(Panel, null, body);
  return body;
};

VideoDesktopPreview.propTypes = {
  imageUrl: _propTypes["default"].string,
  videoUrl: _propTypes["default"].string,
  showPanel: _propTypes["default"].bool,
  title: _propTypes["default"].string,
  headline: _propTypes["default"].string,
  headerFontSize: _propTypes["default"].string,
  imageTooltipMessage: _propTypes["default"].string
};
VideoDesktopPreview.defaultProps = {
  showPanel: true
};
var _default = VideoDesktopPreview;
exports["default"] = _default;
