// @spotify-internal/encore-advertising-web/esm/components/MobilePreview/index.js
import React from 'react';
import styled from 'styled-components';
import { Type, black, gray75, gray90, spacer16, } from '@spotify-internal/encore-web';
import { PlaybackControls } from './styled/PlaybackControls';
import { ProgressBar } from './styled/ProgressBar';
import { Header } from './styled/Header';
import { CTAButton } from './styled/CTAButton';
import { ImageContainer } from './styled/ImageContainer';
const PreviewBody = styled.div `
  background: ${black};
  border-radius: ${spacer16};
  color: white;
  height: 482px;
  margin: ${props => (props.$showPanel ? '0 auto' : '0')};
  text-align: center;
  width: 271px;
`;
const Panel = styled.div `
  border-radius: 4px;
  box-shadow: 0 3px 9px 0 ${gray90};
  padding: 20px 0;
  width: 300px;
`;
const PodcastPanel = styled.div `
  text-align: left;
  margin-top: ${spacer16};
  margin-left: ${spacer16};
`;
export const MobilePreview = ({ title = 'Advertisement', headline, tagLine, imageUrl, podcastSponsorMessage, podcastName, ctaText = 'Learn More', showSkipControls = false, showPanel = true, showCtaText = true, showPodcastControls, imageTooltipMessage, children, }) => {
    const body = (React.createElement(PreviewBody, { "$showPanel": showPanel },
        React.createElement(Header, { title: title, headline: headline, tagLine: tagLine }),
        React.createElement(ImageContainer, { imageTooltipMessage: imageTooltipMessage, imageUrl: imageUrl }, children),
        showPodcastControls && (React.createElement(PodcastPanel, null,
            React.createElement(Type.p, { variant: Type.body2, weight: Type.book, condensed: !!podcastSponsorMessage }, podcastSponsorMessage),
            React.createElement(Type.p, { color: gray75, variant: Type.body4, weight: Type.book, condensed: !!podcastSponsorMessage }, podcastName))),
        showCtaText && React.createElement(CTAButton, { ctaText: ctaText }),
        React.createElement(ProgressBar, null),
        React.createElement(PlaybackControls, { showSkipControls: !!showSkipControls, showPodcastControls: !!showPodcastControls })));
    if (showPanel)
        return React.createElement(Panel, null, body);
    return body;
};
