// @spotify-internal/creator-tape/es/components/FormTextarea/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import { formControlBase, formControlBaseTextarea, formControlTheme, formControlErrorTheme } from '../../styles';

var textareaTheme = function textareaTheme(props) {
  return props.theme.formTextarea === 'dark' ? 'dark' : 'light';
};

var Textarea = styled.textarea.withConfig({
  displayName: "FormTextarea__Textarea",
  componentId: "sc-1jxy1l8-0"
})(["", ";", ";", ";", ";"], formControlBase(), formControlBaseTextarea(), function (props) {
  return formControlTheme(textareaTheme(props), true);
}, function (props) {
  return props.error && formControlErrorTheme(textareaTheme(props), true);
});
var propTypes = {
  /** application theme */
  theme: PropTypes.shape({
    formTextarea: PropTypes.oneOf(['dark'])
  }),
  error: PropTypes.bool
};
var defaultProps = {
  theme: {}
};
var FormTextarea = React.forwardRef(function (_ref, ref) {
  var theme = _ref.theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Textarea, Object.assign({
    ref: ref
  }, props)));
});
FormTextarea.darkTheme = {
  formTextarea: 'dark'
};
FormTextarea.propTypes = propTypes;
FormTextarea.defaultProps = defaultProps;
FormTextarea.displayName = 'FormTextarea';
export default FormTextarea;
