// @spotify-internal/spoticharts/cjs/components/DummyBarChart/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const times_1 = __importDefault(require("lodash/times"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const reactochart_1 = require("reactochart");
const BucketingBarChart_1 = __importDefault(require("../BucketingBarChart"));
const global_1 = __importDefault(require("../../styles/global"));
class DummyBarChart extends react_1.PureComponent {
    render() {
        const { numberOfBars, bucketSpacing, barThickness, color } = this.props;
        const data = times_1.default(numberOfBars, i => ({
            height: i % 2 === 0 ? 4 : 7,
            index: i,
        }));
        return (react_1.default.createElement(react_1.Fragment, null,
            react_1.default.createElement(reactochart_1.XYPlot, Object.assign({ xDomain: [0, numberOfBars - 1], yDomain: [0, 10] }, this.props),
                react_1.default.createElement(reactochart_1.YGrid, { tickCount: 2 }),
                react_1.default.createElement(BucketingBarChart_1.default, Object.assign({}, this.props, { data: data, barThickness: barThickness, valueAccessors: [
                        {
                            start: () => 0,
                            end: d => d.height,
                        },
                    ], bucketAccessor: d => d.index, bucketSpacing: bucketSpacing ||
                        Math.round((this.props.width - barThickness * numberOfBars) /
                            (numberOfBars + 1)), colors: [color] }))),
            react_1.default.createElement(global_1.default, null)));
    }
}
exports.default = DummyBarChart;
DummyBarChart.propTypes = {
    width: prop_types_1.default.number.isRequired,
    height: prop_types_1.default.number.isRequired,
    barThickness: prop_types_1.default.number,
    bucketSpacing: prop_types_1.default.number,
    color: prop_types_1.default.string,
    numberOfBars: prop_types_1.default.number.isRequired,
};
DummyBarChart.defaultProps = {
    barThickness: 40,
    color: encore_foundation_1.gray95,
};
