// @spotify-internal/checkout-sdk/esm/lib/tracking/exportableEventPayloadDefinitions.js
export var ExportableEvents;
(function (ExportableEvents) {
    ExportableEvents["LoadedDefaultPaymentMethod"] = "loadedDefaultPaymentMethod";
    ExportableEvents["TriggeredErrorOnFormValidation"] = "triggeredErrorOnFormValidation";
    ExportableEvents["SubmittedValidPaymentForm"] = "submittedValidPaymentForm";
    ExportableEvents["UserSelectedPaymentMethod"] = "userSelectedPaymentMethod";
    ExportableEvents["UserClickedSubmitButton"] = "userClickedSubmitButton";
})(ExportableEvents || (ExportableEvents = {}));
export var loadedDefaultPaymentMethodEvent = function (data) { return ({
    name: ExportableEvents.LoadedDefaultPaymentMethod,
    data: data,
}); };
export var userSelectedPaymentMethodEvent = function (data) { return ({
    name: ExportableEvents.UserSelectedPaymentMethod,
    data: data,
}); };
export var userClickedSubmitEvent = function (data) { return ({
    name: ExportableEvents.UserClickedSubmitButton,
    data: data,
}); };
export var submittedValidPaymentFormEvent = function (data) { return ({
    name: ExportableEvents.SubmittedValidPaymentForm,
    data: data,
}); };
export var triggeredErrorOnFormValidationEvent = function (data) { return ({
    name: ExportableEvents.TriggeredErrorOnFormValidation,
    data: data,
}); };
