// @spotify-internal/adstudio-tape/lib/styles/mixins/forms.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.formControlTheme = exports.formCheckIndicator = exports.formCheck = exports.formControlBaseSelect = exports.formControlBaseTextarea = exports.formControlBaseInput = exports.formControlBase = void 0;

var _styledComponents = require("styled-components");

var _encoreFoundation = require("@spotify-internal/encore-foundation");

//
// Forms
// --------------------------------------------------
// Shared styles for inputs, textareas, selects
var formControlBase = (0, _styledComponents.css)(["-webkit-appearance:none;-moz-appearance:none;background-image:none;border-style:solid;display:block;font-size:$font-size-base;line-height:1.5;transition:border-color ease-in-out 0.1s,color fade 0.1s;width:100%;@-moz-document url-prefix(\"\"){padding-top:", " + 2px;}&::placeholder{opacity:1;}&:focus,&:hover:focus{outline:0;}&:disabled,&[readonly]{cursor:not-allowed;opacity:1;}&:required{box-shadow:none;}}"], _encoreFoundation.spacer12); // Styles for inputs only

exports.formControlBase = formControlBase;
var formControlBaseInput = (0, _styledComponents.css)(["@media (max-width:", "){border-radius:0;border-width:0 0 1px;padding:0 0 ", ";}@media (min-width:", "){border-radius:", ";border-width:1px;height:", ";padding:", " ", ";}"], _encoreFoundation.screenSmMin, _encoreFoundation.spacer8, _encoreFoundation.screenSmMin, _encoreFoundation.spacer4, _encoreFoundation.spacer48, _encoreFoundation.spacer12, _encoreFoundation.spacer16); // Styles for textareas only

exports.formControlBaseInput = formControlBaseInput;
var formControlBaseTextarea = (0, _styledComponents.css)(["border-radius:", ";border-width:1px;padding:", " ", ";"], _encoreFoundation.spacer4, _encoreFoundation.spacer12, _encoreFoundation.spacer16); // Styles for selects only

exports.formControlBaseTextarea = formControlBaseTextarea;
var formControlBaseSelect = (0, _styledComponents.css)(["border-radius:", ";border-width:1px;height:", ";padding:", " ", " ", " ", ";&::-ms-expand{background-color:transparent;border:0;}"], _encoreFoundation.spacer4, _encoreFoundation.spacer48, _encoreFoundation.spacer12, _encoreFoundation.spacer48, _encoreFoundation.spacer12, _encoreFoundation.spacer16); // Shared styles for radios + checkboxes

exports.formControlBaseSelect = formControlBaseSelect;
var formCheck = (0, _styledComponents.css)(["align-items:center;display:flex;padding-bottom:", ";padding-top:", ";"], _encoreFoundation.spacer4, _encoreFoundation.spacer4);
exports.formCheck = formCheck;
var formCheckIndicator = (0, _styledComponents.css)(["align-self:flex-start;background:transparent;border:1px solid $gray-30;border-radius:$radius;display:inline-block;flex-shrink:0;height:", ";position:relative;top:", ";user-select:none;width:", ";"], _encoreFoundation.spacer16, _encoreFoundation.spacer4, _encoreFoundation.spacer16); // Light Theme

exports.formCheckIndicator = formCheckIndicator;
var themeLightColorDefault = _encoreFoundation.gray20;
var themeLightColorPlaceholder = _encoreFoundation.gray70;
var themeLightColorSelection = _encoreFoundation.gray90;
var themeLightColorDisabled = _encoreFoundation.gray80;
var themeLightBackgroundDefault = _encoreFoundation.white;
var themeLightBackgroundDisabled = _encoreFoundation.gray95;
var themeLightBorderDefault = _encoreFoundation.gray70;
var themeLightBorderFocus = _encoreFoundation.gray20;
var themeLightBorderDisabled = _encoreFoundation.gray80;
var themeLightBorderError = _encoreFoundation.failure;
var themeLightBorderWarning = _encoreFoundation.warning; // Dark Theme

var themeDarkColorDefault = _encoreFoundation.gray95;
var themeDarkColorPlaceholder = _encoreFoundation.gray70;
var themeDarkColorDisabled = _encoreFoundation.gray30;
var themeDarkColorSelection = _encoreFoundation.gray20;
var themeDarkBackgroundDefaultXs = "transparent";
var themeDarkBackgroundDefault = "rgba(0, 0, 0, 0.4)";
var themeDarkBackgroundDisabled = _encoreFoundation.gray20;
var themeDarkBorderDefault = _encoreFoundation.gray70;
var themeDarkBorderFocus = _encoreFoundation.gray95;
var themeDarkBorderDisabled = _encoreFoundation.gray30;
var themeDarkBorderError = _encoreFoundation.failure;
var themeDarkBorderWarning = _encoreFoundation.warning;

var formControlTheme = function formControlTheme(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? 'light' : _ref$theme,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? false : _ref$error,
      _ref$warning = _ref.warning,
      warning = _ref$warning === void 0 ? false : _ref$warning;
  return (0, _styledComponents.css)(["", ""], theme === 'dark' ? (0, _styledComponents.css)(["background-color:", ";border-color:", ";color:", ";&:focus,&:hover:focus{border-color:", ";}&::placeholder{color:", ";}&::selection{background-color:", ";}&:disabled,&[readonly]{border-color:", ";color:", ";@media (min-width:", "){background-color:", ";}}&:invalid{&,&:hover,&:focus,&:hover:focus{border-color:", ";}}}", " ", " @media (min-width:", "){background-color:", ";}"], themeDarkBackgroundDefaultXs, themeDarkBorderDefault, themeDarkColorDefault, themeDarkBorderFocus, themeDarkColorPlaceholder, themeDarkColorSelection, themeDarkBorderDisabled, themeDarkColorDisabled, _encoreFoundation.screenSmMin, themeDarkBackgroundDisabled, themeDarkBorderError, error && (0, _styledComponents.css)(["border-color:", ";&,&:hover,&:focus,&:hover:focus{border-color:", ";}}"], themeDarkBorderError, themeDarkBorderError), warning && (0, _styledComponents.css)(["border-color:", ";&,&:hover,&:focus,&:hover:focus{border-color:", ";}}"], themeDarkBorderWarning, themeDarkBorderWarning), _encoreFoundation.screenSmMin, themeDarkBackgroundDefault) : (0, _styledComponents.css)(["background-color:", ";border-color:", ";color:", ";&:focus,&:hover:focus{border-color:", ";}&::placeholder{color:", ";}&::selection{background-color:", ";}&:disabled,&[readonly]{border-color:", ";color:", ";@media (min-width:", "){background-color:", ";}}&:invalid{&,&:hover,&:focus,&:hover:focus{border-color:", ";}}", " ", ""], themeLightBackgroundDefault, themeLightBorderDefault, themeLightColorDefault, themeLightBorderFocus, themeLightColorPlaceholder, themeLightColorSelection, themeLightBorderDisabled, themeLightColorDisabled, _encoreFoundation.screenSmMin, themeLightBackgroundDisabled, themeLightBorderError, error && (0, _styledComponents.css)(["border-color:", ";&,&:hover,&:focus,&:hover:focus{border-color:", ";}}"], themeLightBorderError, themeLightBorderError), warning && (0, _styledComponents.css)(["border-color:", ";&,&:hover,&:focus,&:hover:focus{border-color:", ";}}"], themeLightBorderWarning, themeLightBorderWarning)));
};

exports.formControlTheme = formControlTheme;
