// @spotify-internal/encore-web/es/styles/mixins/legacyType.js
import { body1, body2, body3, body4, breakpoint, cta1, cta2, cta3, cta4, heading4, type } from '@spotify-internal/encore-foundation';
import { css } from 'styled-components';
export var typeStyleDisplay1 = function typeStyleDisplay1() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], type.display1FontSizeXSmall, type.display1FontWeightXSmall, type.display1LetterSpacingXSmall, type.display1LineHeightXSmall, type.display1PaddingBottomXSmall, breakpoint.screenSmMin, type.display1FontSizeSmall, type.display1LineHeightSmall, type.display1LetterSpacingSmall, breakpoint.screenMdMin, type.display1FontSize, type.display1LineHeight);
};
export var typeStyleDisplay2 = function typeStyleDisplay2() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], type.display2FontSizeXSmall, type.display2FontWeightXSmall, type.display2LetterSpacingXSmall, type.display2LineHeightXSmall, type.display2PaddingBottomXSmall, breakpoint.screenSmMin, type.display2FontSizeSmall, type.display2LineHeightSmall, type.display2LetterSpacingSmall, breakpoint.screenMdMin, type.display2FontSize, type.display2LineHeight);
};
export var typeStyleH1 = function typeStyleH1() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], type.heading1FontSizeXSmall, type.heading1FontWeightXSmall, type.heading1LetterSpacingXSmall, type.heading1LineHeightXSmall, type.heading1PaddingBottomXSmall, breakpoint.screenSmMin, type.heading1FontSizeSmall, type.heading1LineHeightSmall, breakpoint.screenMdMin, type.heading1FontSize, type.heading1LineHeight);
};
export var typeStyleH2 = function typeStyleH2() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}"], type.heading2FontSizeSmall, type.heading2FontWeightSmall, type.heading2LetterSpacingSmall, type.heading2LineHeightSmall, type.heading2PaddingBottomSmall, breakpoint.screenSmMin, type.heading2FontSize, type.heading2LineHeight, type.heading2LetterSpacing);
};
export var typeStyleH3 = function typeStyleH3() {
  return css(["font-size:", ";font-weight:", ";line-height:", ";letter-spacing:", ";@media (min-width:", "){font-size:", ";line-height:", ";}"], type.heading3FontSizeSmall, type.heading3FontWeightSmall, type.heading3LineHeightSmall, type.heading3LetterSpacingSmall, breakpoint.screenSmMin, type.heading3FontSize, type.heading3LineHeight);
};
export var typeStyleH4 = function typeStyleH4() {
  return css(["", ";"], heading4);
};
export var typeStyleBody1 = function typeStyleBody1() {
  return css(["", ";"], body1);
};
export var typeStyleBody2 = function typeStyleBody2() {
  return css(["", ";"], body2);
};
export var typeStyleBody3 = function typeStyleBody3() {
  return css(["", ";"], body3);
};
export var typeStyleBody4 = function typeStyleBody4() {
  return css(["", ";"], body4);
};
export var typeStyleCta1 = function typeStyleCta1() {
  return css(["", ";"], cta1);
};
export var typeStyleCta2 = function typeStyleCta2() {
  return css(["", ";"], cta2);
};
export var typeStyleCta3 = function typeStyleCta3() {
  return css(["", ";"], cta3);
};
export var typeStyleCta4 = function typeStyleCta4() {
  return css(["", ";"], cta4);
};
