// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/LocationTargetingOptions/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LocationTargetingOptions = exports.stringsShape = exports.BulkAddButton = exports.StyledCheckbox = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reduxForm = require("redux-form");

var _recompose = require("recompose");

var _reactRedux = require("react-redux");

var _uniqBy = _interopRequireDefault(require("lodash/uniqBy"));

var _ReduxFormFieldHOC = _interopRequireDefault(require("../../ReduxFormFieldHOC"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _BulkLocationsUploader = _interopRequireDefault(require("../../BulkLocationsUploader"));

var _WithLocationLimitTooltip = _interopRequireDefault(require("../WithLocationLimitTooltip"));

var _constants = require("../constants");

var _i18n = require("../../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var OptionsWrapper = _styledComponents["default"].div.withConfig({
  displayName: "LocationTargetingOptions__OptionsWrapper",
  componentId: "hcm7be-0"
})(["display:grid;grid-template-columns:1fr 265px;grid-gap:16px;align-items:center;justify-items:right;padding-bottom:", ";"], _encoreAdvertisingWeb.spacer16);

var StyledCheckbox = (0, _styledComponents["default"])(_adstudioTape.Checkbox).withConfig({
  displayName: "LocationTargetingOptions__StyledCheckbox",
  componentId: "hcm7be-1"
})(["&&{margin-right:", ";cursor:", ";margin-bottom:0;}& + div > span{font-size:14px;color:", ";line-height:", ";}svg{margin-bottom:0;}"], _encoreAdvertisingWeb.spacer8, function (props) {
  return props.disabled && 'not-allowed';
}, function (props) {
  return props.disabled ? _encoreAdvertisingWeb.gray75 : 'initial';
}, _encoreAdvertisingWeb.spacer16);
exports.StyledCheckbox = StyledCheckbox;
var BulkAddButton = (0, _styledComponents["default"])(_encoreAdvertisingWeb.ButtonSecondary).withConfig({
  displayName: "LocationTargetingOptions__BulkAddButton",
  componentId: "hcm7be-2"
})(["padding:10px;margin-bottom:", ";"], _encoreAdvertisingWeb.spacer32);
exports.BulkAddButton = BulkAddButton;
var StyledField = (0, _styledComponents["default"])(_reduxForm.Field).withConfig({
  displayName: "LocationTargetingOptions__StyledField",
  componentId: "hcm7be-3"
})(["margin-bottom:", ";"], _encoreAdvertisingWeb.spacer16);

var BulkLocationsContainer = _styledComponents["default"].div.withConfig({
  displayName: "LocationTargetingOptions__BulkLocationsContainer",
  componentId: "hcm7be-4"
})(["display:flex;justify-content:flex-end;"]);

var stringsShape = {
  BULK_ADD_MESSAGE: _propTypes["default"].string,
  LABEL_MESSAGE: _propTypes["default"].string
};
/* eslint-disable-next-line */

exports.stringsShape = stringsShape;
var ReduxCheckbox = (0, _ReduxFormFieldHOC["default"])(function (_ref) {
  var value = _ref.value,
      strings = _ref.strings,
      props = _objectWithoutProperties(_ref, ["value", "strings"]);

  return /*#__PURE__*/_react["default"].createElement(StyledCheckbox, _extends({}, props, {
    value: value,
    checked: value,
    label: strings.LABEL_MESSAGE
  }));
});

var LocationTargetingOptions = function LocationTargetingOptions(props) {
  var parentCountry = props.parentCountry,
      disableTargetWholeCountry = props.disableTargetWholeCountry,
      disableBulkUploadZip = props.disableBulkUploadZip,
      formName = props.formName,
      onBlur = props.onBlur,
      setShowModal = props.setShowModal,
      showModal = props.showModal,
      _props$targetedLocati = props.targetedLocations,
      targetedLocations = _props$targetedLocati === void 0 ? [] : _props$targetedLocati,
      edgeHost = props.edgeHost,
      limit = props.limit,
      bulkUploaderStrings = props.bulkUploaderStrings,
      strings = props.strings;
  var overTargetingLimit = targetedLocations.length >= limit;
  return /*#__PURE__*/_react["default"].createElement(OptionsWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledField, {
    flush: true,
    id: "target-country-checkbox",
    name: "targeting.targetWholeCountry",
    component: ReduxCheckbox,
    type: "checkbox",
    disabled: disableTargetWholeCountry,
    strings: strings
  }), /*#__PURE__*/_react["default"].createElement(BulkLocationsContainer, null, /*#__PURE__*/_react["default"].createElement(_WithLocationLimitTooltip["default"], {
    limit: limit,
    overTargetingLimit: overTargetingLimit
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.ButtonSecondary, {
    id: "bulk-location-upload-button",
    buttonSize: _encoreAdvertisingWeb.ButtonSecondary.sm,
    onClick: function onClick() {
      return setShowModal(true);
    },
    disabled: disableBulkUploadZip || overTargetingLimit
  }, strings.BULK_ADD_MESSAGE)), showModal && /*#__PURE__*/_react["default"].createElement(_BulkLocationsUploader["default"], {
    edgeHost: edgeHost,
    targetedLocations: targetedLocations,
    initialTargetCount: targetedLocations.length,
    limit: limit,
    onSubmit: function onSubmit(locations) {
      var mergedLocations = [].concat(_toConsumableArray(targetedLocations), _toConsumableArray(locations));
      onBlur(formName, 'targeting.locations', (0, _uniqBy["default"])(mergedLocations, 'geoId'));
      setShowModal(false);
    },
    parentCountry: parentCountry,
    onModalClose: function onModalClose() {
      return setShowModal(false);
    },
    strings: bulkUploaderStrings
  })));
};

exports.LocationTargetingOptions = LocationTargetingOptions;
LocationTargetingOptions.defaultProps = {
  parentCountry: _i18n.DEFAULT_COUNTRY,
  edgeHost: 'https://edge-grpc.spotify.com',
  strings: {
    BULK_ADD_MESSAGE: _constants.BULK_ADD_MESSAGE,
    LABEL_MESSAGE: _constants.LABEL_MESSAGE
  }
};
LocationTargetingOptions.propTypes = {
  parentCountry: _propTypes["default"].string.isRequired,
  disableTargetWholeCountry: _propTypes["default"].bool,
  disableBulkUploadZip: _propTypes["default"].bool,
  formName: _propTypes["default"].string.isRequired,
  onBlur: _propTypes["default"].func.isRequired,
  setShowModal: _propTypes["default"].func.isRequired,
  showModal: _propTypes["default"].bool.isRequired,
  strings: _propTypes["default"].shape(stringsShape),
  bulkUploaderStrings: _propTypes["default"].object,
  targetedLocations: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    geoId: _propTypes["default"].string,
    name: _propTypes["default"].string,
    parentName: _propTypes["default"].string,
    type: _propTypes["default"].string,
    typeDisplayName: _propTypes["default"].string
  })),
  limit: _propTypes["default"].number.isRequired,
  edgeHost: _propTypes["default"].string
};

function mapStateToProps(state, ownProps) {
  var formSelector = (0, _reduxForm.formValueSelector)(ownProps.formName);
  return {
    targetedLocations: formSelector(state, 'targeting.locations')
  };
}

var _default = (0, _recompose.compose)((0, _reactRedux.connect)(mapStateToProps, {
  onBlur: _reduxForm.blur
}), (0, _recompose.withState)('showModal', 'setShowModal', false))(LocationTargetingOptions);

exports["default"] = _default;
