// @spotify-internal/adstudio-shared/lib/components/DetailViewerListItem/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.DetailViewerListItem = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var GRAY_TITLE = '#919496';
var GRAY_FILLER = '#efefef';

var ItemContainer = _styledComponents["default"].div.withConfig({
  displayName: "DetailViewerListItem__ItemContainer",
  componentId: "dsfldy-0"
})(["flex:1;margin-right:", ";"], _encoreAdvertisingWeb.spacer16);

var FillerRectangle = _styledComponents["default"].div.withConfig({
  displayName: "DetailViewerListItem__FillerRectangle",
  componentId: "dsfldy-1"
})(["background-color:", ";", ""], GRAY_FILLER, function (props) {
  return props.small ? (0, _styledComponents.css)(["display:inline-block;height:12px;margin-top:5px;width:40px;"]) : (0, _styledComponents.css)(["height:36px;margin-right:36px;margin-top:5px;width:128px;"]);
});

var DT = _styledComponents["default"].dt.withConfig({
  displayName: "DetailViewerListItem__DT",
  componentId: "dsfldy-2"
})(["color:", ";font-size:12px;font-weight:300;letter-spacing:1.1px;text-transform:uppercase;"], GRAY_TITLE);

var DTLabel = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.h2).withConfig({
  displayName: "DetailViewerListItem__DTLabel",
  componentId: "dsfldy-3"
})(["", ""], function (props) {
  return props.performanceItem && (0, _styledComponents.css)(["padding-bottom:4px;"]);
});
var LeftSideLabel = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "DetailViewerListItem__LeftSideLabel",
  componentId: "dsfldy-4"
})(["", " margin-right:10px;padding-bottom:0;"], function (props) {
  return props.empty && "color: ".concat(GRAY_TITLE, ";");
});

var DD = _styledComponents["default"].dd.withConfig({
  displayName: "DetailViewerListItem__DD",
  componentId: "dsfldy-5"
})(["margin-left:0;padding-bottom:16px;", " ", ""], function (props) {
  return props.leftSideLabel && (0, _styledComponents.css)(["span[role='button']{padding-right:4px;}button{margin-left:0;}align-items:baseline;display:flex;line-height:24px;padding-bottom:0;"]);
}, function (props) {
  return props.performanceItem && (0, _styledComponents.css)(["white-space:nowrap;padding-bottom:0;h2{padding-bottom:0;}h3{margin-top:0;padding-bottom:0;}small{font-size:14px;}"]);
});

var DetailViewerListItem = function DetailViewerListItem(_ref) {
  var children = _ref.children,
      label = _ref.label,
      leftSideLabel = _ref.leftSideLabel,
      tooltip = _ref.tooltip,
      tooltipPlacement = _ref.tooltipPlacement,
      empty = _ref.empty,
      performanceItem = _ref.performanceItem,
      onShowTooltip = _ref.onShowTooltip;
  return /*#__PURE__*/_react["default"].createElement(ItemContainer, null, children && label && /*#__PURE__*/_react["default"].createElement(DT, null, tooltip && /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
    icon: false,
    onShow: onShowTooltip,
    placement: tooltipPlacement,
    tooltipText: tooltip
  }, /*#__PURE__*/_react["default"].createElement(DTLabel, {
    performanceItem: performanceItem,
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold
  }, label)), !tooltip && /*#__PURE__*/_react["default"].createElement(DTLabel, {
    performanceItem: performanceItem,
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold
  }, label)), children && /*#__PURE__*/_react["default"].createElement(DD, {
    leftSideLabel: !!leftSideLabel,
    performanceItem: performanceItem
  }, tooltip && leftSideLabel && /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
    icon: false,
    onShow: onShowTooltip,
    placement: tooltipPlacement,
    tooltipText: tooltip
  }, /*#__PURE__*/_react["default"].createElement(LeftSideLabel, {
    empty: empty,
    flush: true,
    variant: _encoreAdvertisingWeb.Type.body2,
    weight: _encoreAdvertisingWeb.Type.book
  }, leftSideLabel)), !tooltip && leftSideLabel && /*#__PURE__*/_react["default"].createElement(LeftSideLabel, {
    empty: empty,
    flush: true,
    variant: _encoreAdvertisingWeb.Type.body2,
    weight: _encoreAdvertisingWeb.Type.book
  }, leftSideLabel), empty && leftSideLabel && /*#__PURE__*/_react["default"].createElement(FillerRectangle, {
    small: true
  }), empty && !leftSideLabel && /*#__PURE__*/_react["default"].createElement(FillerRectangle, null), !empty && children));
};

exports.DetailViewerListItem = DetailViewerListItem;
DetailViewerListItem.defaultProps = {
  tooltipPlacement: 'right',
  performanceItem: false
};
DetailViewerListItem.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node, _propTypes["default"].element]),
  className: _propTypes["default"].string,
  empty: _propTypes["default"].bool,
  label: _propTypes["default"].string,
  leftSideLabel: _propTypes["default"].string,
  onShowTooltip: _propTypes["default"].func,
  performanceItem: _propTypes["default"].bool,
  tooltip: _propTypes["default"].string,
  tooltipPlacement: _propTypes["default"].string
};
var _default = DetailViewerListItem;
exports["default"] = _default;
