// @spotify-internal/encore-web/es/components/EmptyState/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { azure } from '@spotify-internal/encore-foundation';
import { ThemeProvider } from 'styled-components';
import { Background } from "./Background";
import { Icon } from "./Icon";
import { IconContainer } from "./IconContainer";
import { MessageContainer } from "./MessageContainer";
export var isDarkTheme = function isDarkTheme(theme) {
  return (theme == null ? void 0 : theme.emptyState) === 'dark';
};
export function EmptyStateComponent(_ref) {
  var background = _ref.background,
      children = _ref.children,
      icon = _ref.icon,
      _ref$iconBackgroundCo = _ref.iconBackgroundColor,
      iconBackgroundColor = _ref$iconBackgroundCo === void 0 ? azure : _ref$iconBackgroundCo,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? 'fullscreen' : _ref$variant,
      props = _objectWithoutProperties(_ref, ["background", "children", "icon", "iconBackgroundColor", "theme", "variant"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Background, Object.assign({
    background: background,
    variant: variant
  }, props), /*#__PURE__*/React.createElement(MessageContainer, {
    background: background,
    variant: variant
  }, icon && /*#__PURE__*/React.createElement(IconContainer, {
    backgroundColor: iconBackgroundColor
  }, /*#__PURE__*/React.createElement(Icon, {
    backgroundColor: iconBackgroundColor,
    icon: icon
  })), children)));
}
var emptyStateHelpers = {
  darkTheme: {
    emptyState: 'dark'
  }
};
export var EmptyState = Object.assign(EmptyStateComponent, emptyStateHelpers);
export default EmptyState;
