// @spotify-internal/encore-web/es/components/Accordion/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { AccordionPanel } from "./Panel";
export function AccordionComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(AccordionPanel, props));
}
var accordionHelpers = {
  darkTheme: {
    accordion: 'dark'
  }
};
/**
* **Accordion** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Accordion) | [Storybook](https://encore-web.spotify.net/?path=/story/components-accordion--repeating-list) | [Encore](https://encore.spotify.net/web/components/accordion) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=4037:6869) |
*
* The Accordion displays items that expand to show more information when clicked or tapped.
*
* @example
* export const RepeatingList = () => {
*  const [activeIndex, updateActive] = useState(-1);
*
*  return (
*    <Accordion aria-label="Accordion group for FAQs">
*      {items.map((item, index) => (
*        <AccordionItem key={`${index}-item`}>
*          <AccordionTitle
*            aria-expanded={index === activeIndex}
*            key={`${index}-title`}
*            onClick={e => {
*              e.preventDefault();
*              updateActive(index === activeIndex ? -1 : index);
*            }}
*            isExpanded={index === activeIndex}
*          >
*            {item.title}
*          </AccordionTitle>
*          <AccordionContent isExpanded={index === activeIndex}>{item.content}</AccordionContent>
*        </AccordionItem>
*      ))}
*    </Accordion>
*  );
*};
*
*/

export var Accordion = Object.assign(AccordionComponent, accordionHelpers);
export default Accordion;
