// @spotify-internal/adstudio-tape/lib/components/Tooltip/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Tooltip;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _encoreWeb = require("@spotify-internal/encore-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Tooltip(_ref) {
  var children = _ref.children,
      className = _ref.className,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? 'light' : _ref$theme,
      _ref$dark = _ref.dark,
      dark = _ref$dark === void 0 ? false : _ref$dark;
  return /*#__PURE__*/_react["default"].createElement(_encoreWeb.Tooltip, {
    className: (0, _classnames["default"])(className),
    theme: dark ? _encoreWeb.Tooltip.darkTheme : _encoreWeb.Tooltip["".concat(theme, "Theme")]
  }, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    variant: _encoreWeb.Type.body2,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, children));
}

Tooltip.propTypes = {
  children: _propTypes["default"].node.isRequired,
  className: _propTypes["default"].string,
  theme: _propTypes["default"].string,
  dark: _propTypes["default"].bool // deprecated - use 'theme="dark"' instead

};
