// @spotify-internal/ubi-sdk-s4a-web-team-activity/WebTeamActivityEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamActivityEventFactory = exports.createWebTeamActivityEventFactory = void 0;
var SPECIFICATION_ID = "web-team-activity";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.1.2";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebTeamActivityEventFactory(options) {
    return new WebTeamActivityEventFactory(options);
}
exports.createWebTeamActivityEventFactory = createWebTeamActivityEventFactory;
/**
 * The team activity page shows activity/audit logs of actions taken on the team, or on behalf of the team.
 */
var WebTeamActivityEventFactory = /** @class */ (function () {
    function WebTeamActivityEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of TeamMemberSearchBarEventFactory
    */
    WebTeamActivityEventFactory.prototype.teamMemberSearchBarFactory = function () {
        return new WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ActivityLogTableEventFactory
    */
    WebTeamActivityEventFactory.prototype.activityLogTableFactory = function () {
        return new WebTeamActivityEventFactory.ActivityLogTableEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of LoadMoreButtonEventFactory
    */
    WebTeamActivityEventFactory.prototype.loadMoreButtonFactory = function () {
        return new WebTeamActivityEventFactory.LoadMoreButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamActivityEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamActivityEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamActivityEventFactory;
}());
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    /**
     * Create TeamMemberSearchBarEventFactory class with path
     * @param path to this node
     */
    var TeamMemberSearchBarEventFactory = /** @class */ (function () {
        function TeamMemberSearchBarEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "team_member_search_bar"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of InputLabelEventFactory
        */
        TeamMemberSearchBarEventFactory.prototype.inputLabelFactory = function () {
            return new WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory.InputLabelEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of SearchInputEventFactory
        */
        TeamMemberSearchBarEventFactory.prototype.searchInputFactory = function () {
            return new WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory.SearchInputEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of SearchResultsEventFactory
        */
        TeamMemberSearchBarEventFactory.prototype.searchResultsFactory = function () {
            return new WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory.SearchResultsEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        TeamMemberSearchBarEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TeamMemberSearchBarEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return TeamMemberSearchBarEventFactory;
    }());
    WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory = TeamMemberSearchBarEventFactory;
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    /**
     * Create ActivityLogTableEventFactory class with path
     * @param path to this node
     */
    var ActivityLogTableEventFactory = /** @class */ (function () {
        function ActivityLogTableEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "activity_log_table"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of ActivityTableHeaderRowEventFactory
        */
        ActivityLogTableEventFactory.prototype.activityTableHeaderRowFactory = function () {
            return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of TableRowActivityRecordEventFactory
        */
        ActivityLogTableEventFactory.prototype.tableRowActivityRecordFactory = function () {
            return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        ActivityLogTableEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ActivityLogTableEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return ActivityLogTableEventFactory;
    }());
    WebTeamActivityEventFactory.ActivityLogTableEventFactory = ActivityLogTableEventFactory;
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    /**
     * Create LoadMoreButtonEventFactory class with path
     * @param path to this node
     */
    var LoadMoreButtonEventFactory = /** @class */ (function () {
        function LoadMoreButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "load_more_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LoadMoreButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LoadMoreButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        LoadMoreButtonEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        /**
         * Create impression for LoadMoreButtonEventFactory.
         */
        LoadMoreButtonEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LoadMoreButtonEventFactory;
    }());
    WebTeamActivityEventFactory.LoadMoreButtonEventFactory = LoadMoreButtonEventFactory;
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var TeamMemberSearchBarEventFactory;
    (function (TeamMemberSearchBarEventFactory) {
        /**
         * Create InputLabelEventFactory class with path
         * @param path to this node
         */
        var InputLabelEventFactory = /** @class */ (function () {
            function InputLabelEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "input_label"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            InputLabelEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            InputLabelEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return InputLabelEventFactory;
        }());
        TeamMemberSearchBarEventFactory.InputLabelEventFactory = InputLabelEventFactory;
    })(TeamMemberSearchBarEventFactory = WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory || (WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var TeamMemberSearchBarEventFactory;
    (function (TeamMemberSearchBarEventFactory) {
        /**
         * Create SearchInputEventFactory class with path
         * @param path to this node
         */
        var SearchInputEventFactory = /** @class */ (function () {
            function SearchInputEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "search_input"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of ClearButtonEventFactory
            */
            SearchInputEventFactory.prototype.clearButtonFactory = function () {
                return new WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory.SearchInputEventFactory.ClearButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            SearchInputEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SearchInputEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for key_stroke search action.
             */
            SearchInputEventFactory.prototype.keyStrokeSearch = function () {
                var data = {
                    actionName: "search",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "key_stroke",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create interaction for key_stroke ui_reveal action.
             */
            SearchInputEventFactory.prototype.keyStrokeUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "key_stroke",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SearchInputEventFactory;
        }());
        TeamMemberSearchBarEventFactory.SearchInputEventFactory = SearchInputEventFactory;
    })(TeamMemberSearchBarEventFactory = WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory || (WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var TeamMemberSearchBarEventFactory;
    (function (TeamMemberSearchBarEventFactory) {
        /**
         * Create SearchResultsEventFactory class with path
         * @param path to this node
         */
        var SearchResultsEventFactory = /** @class */ (function () {
            function SearchResultsEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "search_results"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SearchResultsEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SearchResultsEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit filter action.
             */
            SearchResultsEventFactory.prototype.hitFilter = function () {
                var data = {
                    actionName: "filter",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SearchResultsEventFactory;
        }());
        TeamMemberSearchBarEventFactory.SearchResultsEventFactory = SearchResultsEventFactory;
    })(TeamMemberSearchBarEventFactory = WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory || (WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        /**
         * Create ActivityTableHeaderRowEventFactory class with path
         * @param path to this node
         */
        var ActivityTableHeaderRowEventFactory = /** @class */ (function () {
            function ActivityTableHeaderRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "activity_table_header_row"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of ActivityHeaderEventFactory
            */
            ActivityTableHeaderRowEventFactory.prototype.activityHeaderFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory.ActivityHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of TeamMemberHeaderEventFactory
            */
            ActivityTableHeaderRowEventFactory.prototype.teamMemberHeaderFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory.TeamMemberHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of DateHeaderEventFactory
            */
            ActivityTableHeaderRowEventFactory.prototype.dateHeaderFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory.DateHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            ActivityTableHeaderRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ActivityTableHeaderRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return ActivityTableHeaderRowEventFactory;
        }());
        ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory = ActivityTableHeaderRowEventFactory;
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        /**
         * Create TableRowActivityRecordEventFactory class with path
         * @param path to this node
         */
        var TableRowActivityRecordEventFactory = /** @class */ (function () {
            function TableRowActivityRecordEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "table_row_activity_record"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of ActivityTableCellEventFactory
            */
            TableRowActivityRecordEventFactory.prototype.activityTableCellFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.ActivityTableCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of TeamMemberTableCellEventFactory
            */
            TableRowActivityRecordEventFactory.prototype.teamMemberTableCellFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of DateTableCellEventFactory
            */
            TableRowActivityRecordEventFactory.prototype.dateTableCellFactory = function () {
                return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.DateTableCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            TableRowActivityRecordEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableRowActivityRecordEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return TableRowActivityRecordEventFactory;
        }());
        ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = TableRowActivityRecordEventFactory;
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var TeamMemberSearchBarEventFactory;
    (function (TeamMemberSearchBarEventFactory) {
        var SearchInputEventFactory;
        (function (SearchInputEventFactory) {
            /**
             * Create ClearButtonEventFactory class with path
             * @param path to this node
             */
            var ClearButtonEventFactory = /** @class */ (function () {
                function ClearButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "clear_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ClearButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ClearButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit clear_filter action.
                 */
                ClearButtonEventFactory.prototype.hitClearFilter = function () {
                    var data = {
                        actionName: "clear_filter",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ClearButtonEventFactory;
            }());
            SearchInputEventFactory.ClearButtonEventFactory = ClearButtonEventFactory;
        })(SearchInputEventFactory = TeamMemberSearchBarEventFactory.SearchInputEventFactory || (TeamMemberSearchBarEventFactory.SearchInputEventFactory = {}));
    })(TeamMemberSearchBarEventFactory = WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory || (WebTeamActivityEventFactory.TeamMemberSearchBarEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var ActivityTableHeaderRowEventFactory;
        (function (ActivityTableHeaderRowEventFactory) {
            /**
             * Create ActivityHeaderEventFactory class with path
             * @param path to this node
             */
            var ActivityHeaderEventFactory = /** @class */ (function () {
                function ActivityHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "activity_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ActivityHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ActivityHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return ActivityHeaderEventFactory;
            }());
            ActivityTableHeaderRowEventFactory.ActivityHeaderEventFactory = ActivityHeaderEventFactory;
        })(ActivityTableHeaderRowEventFactory = ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory || (ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var ActivityTableHeaderRowEventFactory;
        (function (ActivityTableHeaderRowEventFactory) {
            /**
             * Create TeamMemberHeaderEventFactory class with path
             * @param path to this node
             */
            var TeamMemberHeaderEventFactory = /** @class */ (function () {
                function TeamMemberHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_member_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TeamMemberHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamMemberHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return TeamMemberHeaderEventFactory;
            }());
            ActivityTableHeaderRowEventFactory.TeamMemberHeaderEventFactory = TeamMemberHeaderEventFactory;
        })(ActivityTableHeaderRowEventFactory = ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory || (ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var ActivityTableHeaderRowEventFactory;
        (function (ActivityTableHeaderRowEventFactory) {
            /**
             * Create DateHeaderEventFactory class with path
             * @param path to this node
             */
            var DateHeaderEventFactory = /** @class */ (function () {
                function DateHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "date_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                DateHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                DateHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return DateHeaderEventFactory;
            }());
            ActivityTableHeaderRowEventFactory.DateHeaderEventFactory = DateHeaderEventFactory;
        })(ActivityTableHeaderRowEventFactory = ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory || (ActivityLogTableEventFactory.ActivityTableHeaderRowEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            /**
             * Create ActivityTableCellEventFactory class with path
             * @param path to this node
             */
            var ActivityTableCellEventFactory = /** @class */ (function () {
                function ActivityTableCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "activity_table_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ActivityTableCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ActivityTableCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return ActivityTableCellEventFactory;
            }());
            TableRowActivityRecordEventFactory.ActivityTableCellEventFactory = ActivityTableCellEventFactory;
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            /**
             * Create TeamMemberTableCellEventFactory class with path
             * @param path to this node
             */
            var TeamMemberTableCellEventFactory = /** @class */ (function () {
                function TeamMemberTableCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_member_table_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                /**
                * @returns A new instance of TeamMemberEventFactory
                */
                TeamMemberTableCellEventFactory.prototype.teamMemberFactory = function () {
                    return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory.TeamMemberEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    });
                };
                /**
                * @returns A new instance of ContextMenuEventFactory
                */
                TeamMemberTableCellEventFactory.prototype.contextMenuFactory = function () {
                    return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory.ContextMenuEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    });
                };
                TeamMemberTableCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamMemberTableCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return TeamMemberTableCellEventFactory;
            }());
            TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory = TeamMemberTableCellEventFactory;
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            /**
             * Create DateTableCellEventFactory class with path
             * @param path to this node
             */
            var DateTableCellEventFactory = /** @class */ (function () {
                function DateTableCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "date_table_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                DateTableCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                DateTableCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return DateTableCellEventFactory;
            }());
            TableRowActivityRecordEventFactory.DateTableCellEventFactory = DateTableCellEventFactory;
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            var TeamMemberTableCellEventFactory;
            (function (TeamMemberTableCellEventFactory) {
                /**
                 * Create TeamMemberEventFactory class with path
                 * @param path to this node
                 */
                var TeamMemberEventFactory = /** @class */ (function () {
                    function TeamMemberEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat({
                            name: "team_member"
                        });
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    TeamMemberEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    TeamMemberEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    return TeamMemberEventFactory;
                }());
                TeamMemberTableCellEventFactory.TeamMemberEventFactory = TeamMemberEventFactory;
            })(TeamMemberTableCellEventFactory = TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory || (TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory = {}));
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            var TeamMemberTableCellEventFactory;
            (function (TeamMemberTableCellEventFactory) {
                /**
                 * Create ContextMenuEventFactory class with path
                 * @param path to this node
                 */
                var ContextMenuEventFactory = /** @class */ (function () {
                    function ContextMenuEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat({
                            name: "context_menu"
                        });
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    /**
                    * @returns A new instance of EditTeamMemberButtonEventFactory
                    */
                    ContextMenuEventFactory.prototype.editTeamMemberButtonFactory = function () {
                        return new WebTeamActivityEventFactory.ActivityLogTableEventFactory.TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory.ContextMenuEventFactory.EditTeamMemberButtonEventFactory({
                            path: this._path,
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        });
                    };
                    ContextMenuEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    ContextMenuEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    /**
                     * Create interaction for hit ui_reveal action.
                     */
                    ContextMenuEventFactory.prototype.hitUiReveal = function () {
                        var data = {
                            actionName: "ui_reveal",
                            actionVersion: 1,
                            app: APP,
                            generatorVersion: GENERATOR_VERSION,
                            interactionType: "hit",
                            specificationMode: SPECIFICATION_MODE,
                            specificationVersion: SPECIFICATION_VERSION,
                            annotatorConfigurationVersion: '',
                            annotatorVersion: '',
                            location: this._getCurrentLocation(),
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        };
                        return data;
                    };
                    return ContextMenuEventFactory;
                }());
                TeamMemberTableCellEventFactory.ContextMenuEventFactory = ContextMenuEventFactory;
            })(TeamMemberTableCellEventFactory = TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory || (TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory = {}));
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
(function (WebTeamActivityEventFactory) {
    var ActivityLogTableEventFactory;
    (function (ActivityLogTableEventFactory) {
        var TableRowActivityRecordEventFactory;
        (function (TableRowActivityRecordEventFactory) {
            var TeamMemberTableCellEventFactory;
            (function (TeamMemberTableCellEventFactory) {
                var ContextMenuEventFactory;
                (function (ContextMenuEventFactory) {
                    /**
                     * Create EditTeamMemberButtonEventFactory class with path
                     * @param path to this node
                     */
                    var EditTeamMemberButtonEventFactory = /** @class */ (function () {
                        function EditTeamMemberButtonEventFactory(_a) {
                            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                            this._path = path.concat({
                                name: "edit_team_member_button"
                            });
                            if (parentAbsoluteLocation) {
                                this._parentAbsoluteLocation = parentAbsoluteLocation;
                            }
                        }
                        EditTeamMemberButtonEventFactory.prototype._getCurrentLocation = function () {
                            return {
                                pathNodes: this._path,
                                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                            };
                        };
                        EditTeamMemberButtonEventFactory.prototype.getAbsoluteLocation = function () {
                            var _a;
                            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                            return {
                                locations: __spreadArrays((parentLocations || []), [
                                    this._getCurrentLocation()
                                ]),
                            };
                        };
                        /**
                         * Create interaction for hit ui_navigate action.
                       * @param destination the destination field
                         */
                        EditTeamMemberButtonEventFactory.prototype.hitUiNavigate = function (_a) {
                            var destination = _a.destination;
                            var data = {
                                actionParameterNames: ['destination'],
                                actionParameterValues: [destination.toString()],
                                actionName: "ui_navigate",
                                actionVersion: 1,
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                interactionType: "hit",
                                specificationMode: SPECIFICATION_MODE,
                                specificationVersion: SPECIFICATION_VERSION,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        /**
                         * Create impression for EditTeamMemberButtonEventFactory.
                         */
                        EditTeamMemberButtonEventFactory.prototype.impression = function () {
                            var data = {
                                app: APP,
                                generatorVersion: GENERATOR_VERSION,
                                specificationVersion: SPECIFICATION_VERSION,
                                specificationMode: SPECIFICATION_MODE,
                                annotatorConfigurationVersion: '',
                                annotatorVersion: '',
                                location: this._getCurrentLocation(),
                                parentAbsoluteLocation: this._parentAbsoluteLocation
                            };
                            return data;
                        };
                        return EditTeamMemberButtonEventFactory;
                    }());
                    ContextMenuEventFactory.EditTeamMemberButtonEventFactory = EditTeamMemberButtonEventFactory;
                })(ContextMenuEventFactory = TeamMemberTableCellEventFactory.ContextMenuEventFactory || (TeamMemberTableCellEventFactory.ContextMenuEventFactory = {}));
            })(TeamMemberTableCellEventFactory = TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory || (TableRowActivityRecordEventFactory.TeamMemberTableCellEventFactory = {}));
        })(TableRowActivityRecordEventFactory = ActivityLogTableEventFactory.TableRowActivityRecordEventFactory || (ActivityLogTableEventFactory.TableRowActivityRecordEventFactory = {}));
    })(ActivityLogTableEventFactory = WebTeamActivityEventFactory.ActivityLogTableEventFactory || (WebTeamActivityEventFactory.ActivityLogTableEventFactory = {}));
})(WebTeamActivityEventFactory = exports.WebTeamActivityEventFactory || (exports.WebTeamActivityEventFactory = {}));
exports.WebTeamActivityEventFactory = WebTeamActivityEventFactory;
