// spotify-harmony/controller.js
'use strict';

var PlayerAPIClient = require('./_internal/lib/connect_service/client');
var ServerTime = require('spotify-transport/server_time');
var HarmonyController = require('./_internal/lib/controller');

module.exports = {
  forClient: function(client, options) {
    var _options = options || {};
    var transport = client.getPublicTransport();
    var version = client.getVersionDescriptor().tagged;

    var playerClient = PlayerAPIClient.create({
      descriptor: client.getClientDescriptor(),
      transport: transport,
      autoregister: 'autoregister' in _options ? _options.autoregister : true
    });

    return HarmonyController.create({
      client: client,
      serverTime: ServerTime.create(transport),
      version: version,
      playerClient: playerClient,
      decorateTrackWindow: 'decorateTrackWindow' in _options ?
        _options.decorateTrackWindow :
        true,
      disableProgress: _options.disableProgress
    });
  },

  of: function(item) {
    return item._controller;
  }
};
