// @spotify-internal/encore-web/es/components/PopoverOnboarding/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { overlayDirections } from "../../styles/mixins/overlays";
import { IconX } from "../Icon/IconX";
import { CloseButton } from "../Popover/CloseButton";
import { Popover } from "../Popover/Popover";
import { Title } from "../Popover/Title";
import { Body } from "./Body";
import { Footer } from "./Footer";

function PopoverOnboardingComponent(_ref) {
  var left = _ref.left,
      top = _ref.top,
      right = _ref.right,
      bottom = _ref.bottom,
      popoverTitle = _ref.popoverTitle,
      onClose = _ref.onClose,
      children = _ref.children,
      navStepper = _ref.navStepper,
      footerButton = _ref.footerButton,
      props = _objectWithoutProperties(_ref, ["left", "top", "right", "bottom", "popoverTitle", "onClose", "children", "navStepper", "footerButton"]);

  return /*#__PURE__*/React.createElement(Popover, Object.assign({
    role: "presentation",
    style: {
      left: left,
      top: top,
      right: right,
      bottom: bottom
    },
    paddedBody: onClose && !popoverTitle
  }, props), popoverTitle && /*#__PURE__*/React.createElement(Title, null, popoverTitle), /*#__PURE__*/React.createElement(Body, null, children), onClose && /*#__PURE__*/React.createElement(CloseButton, {
    onClick: onClose,
    "aria-label": "Close"
  }, /*#__PURE__*/React.createElement(IconX, {
    "aria-hidden": "true"
  })), /*#__PURE__*/React.createElement(Footer, null, /*#__PURE__*/React.createElement("div", null, navStepper), /*#__PURE__*/React.createElement("div", null, footerButton)));
}
/**
* **PopoverOnboarding** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PopoverOnboarding) | [Storybook](https://encore-web.spotify.net/?path=/story/components-popover-popoveronboarding--default) | [Encore](https://encore.spotify.net/web/components/popover?format=figma-figma#popoveronboarding) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=522:377) |
*
* A popover modal used in onboarding, and can be multi-step using NavStepper
*
* @example
* export const Default = () => (
*  <PopoverOnboarding
*    arrow={PopoverOnboarding.leftTop}
*    popoverTitle="Step 1"
*    footerButton={
*      <ButtonTertiary buttonSize={Button.sm} color="green" condensed>
*        Next
*      </ButtonTertiary>
*    }
*    navStepper={
*      <NavStepper>
*        <NavStepperItem visited condensed>
*          Step 1
*        </NavStepperItem>
*        <NavStepperItem active condensed>
*          Step 2
*        </NavStepperItem>
*      </NavStepper>
*    }
*    onClose={() => {}}
*  >
*    Aenean lacinia bibendum nulla sed consectetur. Cras justo odio, dapibus ac facilisis in, egestas
*    eget quam. Duis mollis, est non commodo luctus.
*  </PopoverOnboarding>
*);
*
*/


export var PopoverOnboarding = Object.assign(PopoverOnboardingComponent, overlayDirections);
export default PopoverOnboarding;
