// @spotify-internal/adstudio-shared/lib/config/age.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getMinTargetingAgeByCountry = getMinTargetingAgeByCountry;
exports.REAL_MAX_AGE = exports.TARGETING_MAX_AGE = exports.TARGETING_MIN_AGE_14 = exports.TARGETING_MIN_AGE_15 = exports.TARGETING_MIN_AGE_16 = exports.TARGETING_MIN_AGE = void 0;
var TARGETING_MIN_AGE = 13;
exports.TARGETING_MIN_AGE = TARGETING_MIN_AGE;
var TARGETING_MIN_AGE_16 = 16;
exports.TARGETING_MIN_AGE_16 = TARGETING_MIN_AGE_16;
var TARGETING_MIN_AGE_15 = 15;
exports.TARGETING_MIN_AGE_15 = TARGETING_MIN_AGE_15;
var TARGETING_MIN_AGE_14 = 14;
exports.TARGETING_MIN_AGE_14 = TARGETING_MIN_AGE_14;
var TARGETING_MAX_AGE = 65; // Actually we show '65+' as max

exports.TARGETING_MAX_AGE = TARGETING_MAX_AGE;
var REAL_MAX_AGE = 199; // According to our backend
// TODO: This should be removed once all consuming apps are migrated to the Ad Studio Config
//  Service and pass in the appropriate minAge by property.

exports.REAL_MAX_AGE = REAL_MAX_AGE;

function getMinTargetingAgeByCountry(countryCode) {
  switch (countryCode) {
    case 'FR':
      return TARGETING_MIN_AGE_15;

    case 'IT':
    case 'ES':
      return TARGETING_MIN_AGE_14;

    case 'DE':
    case 'IE':
    case 'NL':
      return TARGETING_MIN_AGE_16;

    case 'DK':
    case 'NO':
    case 'FI':
    case 'US':
    default:
      return TARGETING_MIN_AGE;
  }
}
