// @spotify-internal/adstudio-shared/lib/components/GeoTargetingSearch/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LOCATION_SORT_ORDER = exports.POSTAL_CODE = exports.COUNTRIES = exports.GEO_TYPES = void 0;
// geosearch returns these, in all caps
// https://ghe.spotify.net/ads/geosearch/blob/e85f89af045f1979976c2fda38774486ec0f5a09/src/main/resources/geosearch.conf#L48
var GEO_TYPES = {
  CITY: 'City',
  COUNTRY: 'Country',
  DMA_REGION: 'DMA',
  POSTAL_CODE: 'Postal Code',
  REGION: 'Region',
  COUNTY: 'County'
};
exports.GEO_TYPES = GEO_TYPES;
var COUNTRIES = {
  US: 'United States',
  AD: 'Andorra',
  AR: 'Argentina',
  AU: 'Australia',
  AT: 'Austria',
  BE: 'Belgium',
  BO: 'Bolivia',
  BR: 'Brazil',
  BG: 'Bulgaria',
  CA: 'Canada',
  CL: 'Chile',
  CO: 'Colombia',
  CR: 'Costa Rica',
  CY: 'Cyprus',
  CZ: 'Czech Republic',
  DK: 'Denmark',
  DO: 'Dominican Republic',
  EC: 'Ecuador',
  SV: 'El Salvador',
  EE: 'Estonia',
  FI: 'Finland',
  FR: 'France',
  DE: 'Germany',
  GR: 'Greece',
  GT: 'Guatemala',
  HN: 'Honduras',
  HK: 'Hong Kong',
  HU: 'Hungary',
  IS: 'Iceland',
  ID: 'Indonesia',
  IE: 'Ireland',
  IT: 'Italy',
  JP: 'Japan',
  LV: 'Latvia',
  LI: 'Liechtenstein',
  LT: 'Lithuania',
  LU: 'Luxembourg',
  MY: 'Malaysia',
  MT: 'Malta',
  MX: 'Mexico',
  MC: 'Monaco',
  NL: 'Netherlands',
  NZ: 'New Zealand',
  NI: 'Nicaragua',
  NO: 'Norway',
  PA: 'Panama',
  PY: 'Paraguay',
  PE: 'Peru',
  PH: 'Philippines',
  PL: 'Poland',
  PT: 'Portugal',
  SG: 'Singapore',
  SK: 'Slovakia',
  ES: 'Spain',
  SE: 'Sweden',
  CH: 'Switzerland',
  TW: 'Taiwan',
  TR: 'Turkey',
  GB: 'United Kingdom',
  UY: 'Uruguay',
  Other: 'Other'
};
exports.COUNTRIES = COUNTRIES;
var POSTAL_CODE = 'POSTAL_CODE'; // SORT FROM PIPELINE
// https://ghe.spotify.net/ads/adstudio-targetingservice-pipeline/blob/master/adstudio-targetingservice-pipeline/src/main/scala/com/spotify/data/adstudio/targetingservice/GeoV2Job.scala#L412

exports.POSTAL_CODE = POSTAL_CODE;
var LOCATION_SORT_ORDER = {
  COUNTRY: 1,
  REGION: 2,
  DMA_REGION: 3,
  COUNTY: 4,
  CITY: 5,
  POSTAL_CODE: 6
};
exports.LOCATION_SORT_ORDER = LOCATION_SORT_ORDER;
