// @spotify-internal/adstudio-shared/lib/components/GeoTargetingSearch/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = GeoTargetingSearch;

var _omit2 = _interopRequireDefault(require("lodash/omit"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _constants = require("./constants");

var _TargetingSearch = _interopRequireDefault(require("../TargetingSearch"));

var _Suggestion = _interopRequireDefault(require("./Suggestion"));

var _Value = _interopRequireDefault(require("./Value"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function GeoTargetingSearch(props) {
  return /*#__PURE__*/_react["default"].createElement(_TargetingSearch["default"], _extends({}, props, {
    exclude: function exclude(item) {
      return props.exclude(item) || props.excludedCountries.includes(item.countryCode);
    },
    preloadOptions: false,
    renderSuggestion: function renderSuggestion(suggestionProps) {
      return /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], suggestionProps);
    },
    renderValue: function renderValue(_ref) {
      var disabled = _ref.disabled,
          value = _ref.value,
          deprecatedGeosFaqLink = _ref.deprecatedGeosFaqLink,
          handleDeprecatedGeos = _ref.handleDeprecatedGeos,
          handleRemove = _ref.handleRemove,
          limit = _ref.limit,
          keyField = _ref.keyField,
          valueField = _ref.valueField;
      return /*#__PURE__*/_react["default"].createElement(_Value["default"], {
        disabled: disabled,
        items: value,
        deprecatedGeosFaqLink: deprecatedGeosFaqLink,
        handleDeprecatedGeos: handleDeprecatedGeos,
        keyField: keyField,
        limit: limit,
        onClick: handleRemove,
        valueField: valueField
      });
    }
  }));
}

GeoTargetingSearch.propTypes = _objectSpread(_objectSpread({}, (0, _omit2["default"])(_TargetingSearch["default"].propTypes, ['preloadOptions', 'renderSuggestion', 'renderValue'])), {}, {
  deprecatedGeosFaqLink: _propTypes["default"].string,
  handleDeprecatedGeos: _propTypes["default"].bool,
  limit: _propTypes["default"].number,
  excludedCountries: _propTypes["default"].arrayOf(_propTypes["default"].oneOf(Object.keys(_constants.COUNTRIES)))
});
GeoTargetingSearch.defaultProps = {
  exclude: function exclude() {
    return false;
  },
  excludedCountries: [],
  suggestionsLimit: Infinity
};
