// @spotify-internal/adstudio-tape/lib/styles/mixins.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonStyleOverrides = buttonStyleOverrides;
exports.overlayClose = exports.overlay = exports.overlayBackdropCentered = exports.overlayBackdrop = exports.dialogStyle = exports.overlayBase = void 0;

var _styledComponents = require("styled-components");

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _tokens = require("./tokens");

/* eslint-disable import/prefer-default-export */
function buttonStyleOverrides(_ref) {
  var main = _ref.main,
      hover = _ref.hover,
      focus = _ref.focus,
      active = _ref.active,
      disabled = _ref.disabled;
  return (0, _styledComponents.css)(["", ";&:hover{", ";}&:focus{", ";}&:active,&.active{&,&:hover{", ";}}&.disabled,&[disabled],fieldset[disabled] &{", ";}"], main, hover, focus, active, disabled);
} // Overlays


var overlayBase = (0, _styledComponents.css)(["background:", ";border-radius:", ";box-shadow:", ";color:", ";text-align:left;"], _encoreFoundation.white, _encoreFoundation.spacer4, _tokens.overlayBoxShadow, _encoreFoundation.gray10);
exports.overlayBase = overlayBase;
var dialogStyle = (0, _styledComponents.css)(["", " display:flex;flex-direction:column;max-height:90vh;max-width:calc(100vw - ", ");z-index:", ";@media (min-width:", "){max-height:80vh;}"], overlayBase, _encoreFoundation.spacer48, _tokens.zIndexDialog, _encoreFoundation.screenSmMin);
exports.dialogStyle = dialogStyle;
var overlayBackdrop = (0, _styledComponents.css)(["background-color:rgba(0,0,0,0.4);height:100%;left:0;position:fixed;top:0;width:100%;z-index:", ";"], _tokens.zIndexDialogBackdrop);
exports.overlayBackdrop = overlayBackdrop;
var overlayBackdropCentered = (0, _styledComponents.css)(["align-items:center;display:flex;justify-content:center;"]);
exports.overlayBackdropCentered = overlayBackdropCentered;

var overlay = function overlay() {
  var maxWidth = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '296px';
  return (0, _styledComponents.css)(["", " border:", ";cursor:default;display:inline-block;max-width:", ";position:relative;"], overlayBase, _tokens.overlayBorder, maxWidth);
};

exports.overlay = overlay;
var overlayClose = (0, _styledComponents.css)(["background:none;border:0;color:", ";cursor:pointer;margin:-", ";outline:none;padding:", ";position:absolute;right:", ";top:", ";&:hover{color:", ";}&:active,&:focus{color:", ";}"], _encoreFoundation.gray70, _encoreFoundation.spacer4, _encoreFoundation.spacer4, _encoreFoundation.spacer16, _encoreFoundation.spacer16, _encoreFoundation.gray20, _encoreFoundation.gray30);
exports.overlayClose = overlayClose;
