// @spotify-internal/adstudio-tape/lib/styles/tokens.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.alertHeight = exports.overlayBorder = exports.overlayBorderColor = exports.overlayBoxShadow = exports.transitionEase = exports.transitionHeight = exports.transitionFade = exports.linkHoverDecoration = exports.linkDecoration = exports.linkHoverColor = exports.linkActiveColor = exports.linkColor = exports.zIndexFixed = exports.zIndexDialog = exports.zIndexPopover = exports.zIndexDialogBackdrop = exports.fontWeightBook = exports.fontWeightBold = exports.fontWeightBlack = exports.fontSizeXxs = exports.fontSizeXs = exports.fontSizeSmall = exports.fontSizeBase = exports.sidebarBasePadding = exports.sidebarBaseWidth = exports.plum = exports.azureDisabled = exports.azurePressed = exports.azureFocus = exports.azureHighlight = exports.azureDefault = void 0;

var _polished = require("polished");

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var azureDefault = _encoreFoundation.azure;
exports.azureDefault = azureDefault;
var azureHighlight = '#74b7ff';
exports.azureHighlight = azureHighlight;
var azureFocus = '#337fd1';
exports.azureFocus = azureFocus;
var azurePressed = '#0a67b2';
exports.azurePressed = azurePressed;
var azureDisabled = (0, _polished.lighten)(0.2, _encoreFoundation.azure);
exports.azureDisabled = azureDisabled;
var plum = '#734D85';
exports.plum = plum;
var sidebarBaseWidth = '200px';
exports.sidebarBaseWidth = sidebarBaseWidth;
var sidebarBasePadding = _encoreFoundation.spacer24;
exports.sidebarBasePadding = sidebarBasePadding;
var fontSizeBase = _encoreFoundation.spacer16;
exports.fontSizeBase = fontSizeBase;
var fontSizeSmall = '14px';
exports.fontSizeSmall = fontSizeSmall;
var fontSizeXs = _encoreFoundation.spacer12;
exports.fontSizeXs = fontSizeXs;
var fontSizeXxs = '10px';
exports.fontSizeXxs = fontSizeXxs;
var fontWeightBlack = 900;
exports.fontWeightBlack = fontWeightBlack;
var fontWeightBold = 600;
exports.fontWeightBold = fontWeightBold;
var fontWeightBook = 400; // z-index

exports.fontWeightBook = fontWeightBook;
var zIndexDialogBackdrop = 1031;
exports.zIndexDialogBackdrop = zIndexDialogBackdrop;
var zIndexPopover = zIndexDialogBackdrop + 10;
exports.zIndexPopover = zIndexPopover;
var zIndexDialog = zIndexDialogBackdrop - 1;
exports.zIndexDialog = zIndexDialog;
var zIndexFixed = 1030; // Typography

exports.zIndexFixed = zIndexFixed;
var linkColor = _encoreFoundation.azure;
exports.linkColor = linkColor;
var linkActiveColor = _encoreFoundation.gray10;
exports.linkActiveColor = linkActiveColor;
var linkHoverColor = _encoreFoundation.gray10;
exports.linkHoverColor = linkHoverColor;
var linkDecoration = 'underline';
exports.linkDecoration = linkDecoration;
var linkHoverDecoration = 'underline'; // Transitions + Animations

exports.linkHoverDecoration = linkHoverDecoration;
var transitionFade = '0.1s';
exports.transitionFade = transitionFade;
var transitionHeight = '0.2s';
exports.transitionHeight = transitionHeight;
var transitionEase = '0.15s ease-in-out'; // Overlays

exports.transitionEase = transitionEase;
var overlayBoxShadow = "0 ".concat(_encoreFoundation.spacer4, " ").concat(_encoreFoundation.spacer12, " 0 rgba(0, 0, 0, 0.2)");
exports.overlayBoxShadow = overlayBoxShadow;
var overlayBorderColor = "rgba(0, 0, 0, 0.1)";
exports.overlayBorderColor = overlayBorderColor;
var overlayBorder = "1px solid ".concat(overlayBorderColor); // Misc

exports.overlayBorder = overlayBorder;
var alertHeight = _encoreFoundation.spacer48;
exports.alertHeight = alertHeight;
