// @spotify-internal/encore-web/es/components/CollapseButton/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { Arrow } from "./Arrow";
import { Button } from "./Button";

function CollapseButtonComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      expanded = _ref.expanded,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "expanded", "children"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Button, Object.assign({}, props, {
    "aria-expanded": expanded,
    isUsingKeyboard: isUsingKeyboard
  }), children, /*#__PURE__*/React.createElement(Arrow, {
    expanded: expanded
  })));
}

var collapseButtonHelpers = {
  darkTheme: {
    collapseButton: 'dark'
  }
};
/**
* **CollapseButton** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/CollapseButton) | [Storybook](https://encore-web.spotify.net/?path=/story/components-button-collapsebutton--default) | [Encore](https://encore.spotify.net/web/components/button?format=figma-figma#collapsebutton) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=262:112) |
*
* CollapseButton is used to indicate the ability to expand and collapse content.
*
* @example
* export const Default = () => (
*  <CollapseButton expanded={false} onClick={() => {}}>
*    Reveal
*  </CollapseButton>
*);
*
*/

export var CollapseButton = Object.assign(CollapseButtonComponent, collapseButtonHelpers);
export default CollapseButton;
