// @spotify-internal/adstudio-tape/lib/components/LayoutFooterLink/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = LayoutFooterLink;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _tokens = require("../../styles/tokens");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var propTypes = {
  /** Title for the `href` */
  title: _propTypes["default"].string,

  /** Basic URL link for the `href` */
  href: _propTypes["default"].string,

  /** Pass a custom component instead of the default `href` */
  component: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].string]),
  theme: _propTypes["default"].object
};

var NavItem = _styledComponents["default"].li.withConfig({
  displayName: "LayoutFooterLink__NavItem",
  componentId: "biya5x-0"
})(["&:first-child{@media (min-width:", "){padding-left:0;}}@media (max-width:", "){padding-bottom:", ";padding-top:", ";}@media (min-width:", "){padding-left:", ";padding-right:", ";}"], _encoreFoundation.screenSmMin, _encoreFoundation.screenXsMax, _encoreFoundation.spacer4, _encoreFoundation.spacer4, _encoreFoundation.screenSmMin, _encoreFoundation.spacer24, _encoreFoundation.spacer24);

function LayoutFooterLink(_ref) {
  var title = _ref.title,
      href = _ref.href,
      component = _ref.component,
      elementProps = _objectWithoutProperties(_ref, ["title", "href", "component"]);

  var Element = 'a';

  if (component) {
    Element = component;
  }

  var NavLink = (0, _styledComponents["default"])(Element).withConfig({
    displayName: "LayoutFooterLink__NavLink",
    componentId: "biya5x-1"
  })(["color:", ";display:block;font-weight:", ";text-decoration:none;&:hover,&:focus{color:", ";}"], _encoreFoundation.gray10, _tokens.fontWeightBold, function (props) {
    return props.theme.colors.primaryColor;
  });
  return /*#__PURE__*/_react["default"].createElement(NavItem, null, /*#__PURE__*/_react["default"].createElement(NavLink, _extends({}, elementProps, {
    href: href,
    rel: "noopener noreferrer",
    title: title
  }), title));
}

LayoutFooterLink.propTypes = propTypes;
