// @spotify-internal/ubi-sdk-s4a-web-roster/WebRosterEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebRosterEventFactory = exports.createWebRosterEventFactory = void 0;
var SPECIFICATION_ID = "web-roster";
var APP = "s4a";
var SPECIFICATION_VERSION = "1.0.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.24";
function createWebRosterEventFactory(options) {
    return new WebRosterEventFactory(options);
}
exports.createWebRosterEventFactory = createWebRosterEventFactory;
/**
 *
 */
var WebRosterEventFactory = /** @class */ (function () {
    function WebRosterEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of RosterTableEventFactory
    */
    WebRosterEventFactory.prototype.rosterTableFactory = function () {
        return new WebRosterEventFactory.RosterTableEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebRosterEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebRosterEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebRosterEventFactory;
}());
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    /**
     * Create RosterTableEventFactory class with path
     * @param path to this node
     */
    var RosterTableEventFactory = /** @class */ (function () {
        function RosterTableEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "roster_table"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of RosterTableHeaderRowEventFactory
        */
        RosterTableEventFactory.prototype.rosterTableHeaderRowFactory = function () {
            return new WebRosterEventFactory.RosterTableEventFactory.RosterTableHeaderRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of RosterTableDataRowEventFactory
        */
        RosterTableEventFactory.prototype.rosterTableDataRowFactory = function () {
            return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of PaginationButtonNextEventFactory
        */
        RosterTableEventFactory.prototype.paginationButtonNextFactory = function () {
            return new WebRosterEventFactory.RosterTableEventFactory.PaginationButtonNextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of PaginationButtonPreviousEventFactory
        */
        RosterTableEventFactory.prototype.paginationButtonPreviousFactory = function () {
            return new WebRosterEventFactory.RosterTableEventFactory.PaginationButtonPreviousEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        RosterTableEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        RosterTableEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return RosterTableEventFactory;
    }());
    WebRosterEventFactory.RosterTableEventFactory = RosterTableEventFactory;
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        /**
         * Create RosterTableHeaderRowEventFactory class with path
         * @param path to this node
         */
        var RosterTableHeaderRowEventFactory = /** @class */ (function () {
            function RosterTableHeaderRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "roster_table_header_row"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of ArtistHeaderEventFactory
            */
            RosterTableHeaderRowEventFactory.prototype.artistHeaderFactory = function () {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableHeaderRowEventFactory.ArtistHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of PitchHeaderEventFactory
            */
            RosterTableHeaderRowEventFactory.prototype.pitchHeaderFactory = function () {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableHeaderRowEventFactory.PitchHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of NextReleaseHeaderEventFactory
            */
            RosterTableHeaderRowEventFactory.prototype.nextReleaseHeaderFactory = function () {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableHeaderRowEventFactory.NextReleaseHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            RosterTableHeaderRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            RosterTableHeaderRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return RosterTableHeaderRowEventFactory;
        }());
        RosterTableEventFactory.RosterTableHeaderRowEventFactory = RosterTableHeaderRowEventFactory;
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        /**
         * Create RosterTableDataRowEventFactory class with path
         * @param path to this node
         */
        var RosterTableDataRowEventFactory = /** @class */ (function () {
            function RosterTableDataRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "roster_table_data_row"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param uri of the ArtistNameCellEventFactory
            * @returns A new instance of ArtistNameCellEventFactory
            */
            RosterTableDataRowEventFactory.prototype.artistNameCellFactory = function (data) {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory.ArtistNameCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            /**
            * @returns A new instance of PitchCellEventFactory
            */
            RosterTableDataRowEventFactory.prototype.pitchCellFactory = function () {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory.PitchCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of NextReleaseCellEventFactory
            */
            RosterTableDataRowEventFactory.prototype.nextReleaseCellFactory = function () {
                return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory.NextReleaseCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            RosterTableDataRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            RosterTableDataRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return RosterTableDataRowEventFactory;
        }());
        RosterTableEventFactory.RosterTableDataRowEventFactory = RosterTableDataRowEventFactory;
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        /**
         * Create PaginationButtonNextEventFactory class with path
         * @param path to this node
         */
        var PaginationButtonNextEventFactory = /** @class */ (function () {
            function PaginationButtonNextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "pagination_button_next"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            PaginationButtonNextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            PaginationButtonNextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            PaginationButtonNextEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return PaginationButtonNextEventFactory;
        }());
        RosterTableEventFactory.PaginationButtonNextEventFactory = PaginationButtonNextEventFactory;
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        /**
         * Create PaginationButtonPreviousEventFactory class with path
         * @param path to this node
         */
        var PaginationButtonPreviousEventFactory = /** @class */ (function () {
            function PaginationButtonPreviousEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "pagination_button_previous"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            PaginationButtonPreviousEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            PaginationButtonPreviousEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            PaginationButtonPreviousEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return PaginationButtonPreviousEventFactory;
        }());
        RosterTableEventFactory.PaginationButtonPreviousEventFactory = PaginationButtonPreviousEventFactory;
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableHeaderRowEventFactory;
        (function (RosterTableHeaderRowEventFactory) {
            /**
             * Create ArtistHeaderEventFactory class with path
             * @param path to this node
             */
            var ArtistHeaderEventFactory = /** @class */ (function () {
                function ArtistHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "artist_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ArtistHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ArtistHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return ArtistHeaderEventFactory;
            }());
            RosterTableHeaderRowEventFactory.ArtistHeaderEventFactory = ArtistHeaderEventFactory;
        })(RosterTableHeaderRowEventFactory = RosterTableEventFactory.RosterTableHeaderRowEventFactory || (RosterTableEventFactory.RosterTableHeaderRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableHeaderRowEventFactory;
        (function (RosterTableHeaderRowEventFactory) {
            /**
             * Create PitchHeaderEventFactory class with path
             * @param path to this node
             */
            var PitchHeaderEventFactory = /** @class */ (function () {
                function PitchHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "pitch_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                /**
                * @returns A new instance of PitchColumnTooltipIconEventFactory
                */
                PitchHeaderEventFactory.prototype.pitchColumnTooltipIconFactory = function () {
                    return new WebRosterEventFactory.RosterTableEventFactory.RosterTableHeaderRowEventFactory.PitchHeaderEventFactory.PitchColumnTooltipIconEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    });
                };
                PitchHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                PitchHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return PitchHeaderEventFactory;
            }());
            RosterTableHeaderRowEventFactory.PitchHeaderEventFactory = PitchHeaderEventFactory;
        })(RosterTableHeaderRowEventFactory = RosterTableEventFactory.RosterTableHeaderRowEventFactory || (RosterTableEventFactory.RosterTableHeaderRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableHeaderRowEventFactory;
        (function (RosterTableHeaderRowEventFactory) {
            /**
             * Create NextReleaseHeaderEventFactory class with path
             * @param path to this node
             */
            var NextReleaseHeaderEventFactory = /** @class */ (function () {
                function NextReleaseHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "next_release_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                NextReleaseHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                NextReleaseHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return NextReleaseHeaderEventFactory;
            }());
            RosterTableHeaderRowEventFactory.NextReleaseHeaderEventFactory = NextReleaseHeaderEventFactory;
        })(RosterTableHeaderRowEventFactory = RosterTableEventFactory.RosterTableHeaderRowEventFactory || (RosterTableEventFactory.RosterTableHeaderRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableDataRowEventFactory;
        (function (RosterTableDataRowEventFactory) {
            /**
             * Create ArtistNameCellEventFactory class with path
             * @param path to this node
             */
            var ArtistNameCellEventFactory = /** @class */ (function () {
                function ArtistNameCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "artist_name_cell" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ArtistNameCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ArtistNameCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                ArtistNameCellEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ArtistNameCellEventFactory;
            }());
            RosterTableDataRowEventFactory.ArtistNameCellEventFactory = ArtistNameCellEventFactory;
        })(RosterTableDataRowEventFactory = RosterTableEventFactory.RosterTableDataRowEventFactory || (RosterTableEventFactory.RosterTableDataRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableDataRowEventFactory;
        (function (RosterTableDataRowEventFactory) {
            /**
             * Create PitchCellEventFactory class with path
             * @param path to this node
             */
            var PitchCellEventFactory = /** @class */ (function () {
                function PitchCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "pitch_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                /**
                * @param uri of the PitchASongButtonEventFactory
                * @returns A new instance of PitchASongButtonEventFactory
                */
                PitchCellEventFactory.prototype.pitchASongButtonFactory = function (data) {
                    return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory.PitchCellEventFactory.PitchASongButtonEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation,
                        data: data
                    });
                };
                /**
                * @param uri of the SeeAPitchButtonEventFactory
                * @returns A new instance of SeeAPitchButtonEventFactory
                */
                PitchCellEventFactory.prototype.seeAPitchButtonFactory = function (data) {
                    return new WebRosterEventFactory.RosterTableEventFactory.RosterTableDataRowEventFactory.PitchCellEventFactory.SeeAPitchButtonEventFactory({
                        path: this._path,
                        parentAbsoluteLocation: this._parentAbsoluteLocation,
                        data: data
                    });
                };
                PitchCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                PitchCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return PitchCellEventFactory;
            }());
            RosterTableDataRowEventFactory.PitchCellEventFactory = PitchCellEventFactory;
        })(RosterTableDataRowEventFactory = RosterTableEventFactory.RosterTableDataRowEventFactory || (RosterTableEventFactory.RosterTableDataRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableDataRowEventFactory;
        (function (RosterTableDataRowEventFactory) {
            /**
             * Create NextReleaseCellEventFactory class with path
             * @param path to this node
             */
            var NextReleaseCellEventFactory = /** @class */ (function () {
                function NextReleaseCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "next_release_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                NextReleaseCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                NextReleaseCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return NextReleaseCellEventFactory;
            }());
            RosterTableDataRowEventFactory.NextReleaseCellEventFactory = NextReleaseCellEventFactory;
        })(RosterTableDataRowEventFactory = RosterTableEventFactory.RosterTableDataRowEventFactory || (RosterTableEventFactory.RosterTableDataRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableHeaderRowEventFactory;
        (function (RosterTableHeaderRowEventFactory) {
            var PitchHeaderEventFactory;
            (function (PitchHeaderEventFactory) {
                /**
                 * Create PitchColumnTooltipIconEventFactory class with path
                 * @param path to this node
                 */
                var PitchColumnTooltipIconEventFactory = /** @class */ (function () {
                    function PitchColumnTooltipIconEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat({
                            name: "pitch_column_tooltip_icon"
                        });
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    PitchColumnTooltipIconEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    PitchColumnTooltipIconEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    /**
                     * Create interaction for hit ui_reveal action.
                     */
                    PitchColumnTooltipIconEventFactory.prototype.hitUiReveal = function () {
                        var data = {
                            actionName: "ui_reveal",
                            actionVersion: 1,
                            app: APP,
                            generatorVersion: GENERATOR_VERSION,
                            interactionType: "hit",
                            specificationMode: SPECIFICATION_MODE,
                            specificationVersion: SPECIFICATION_VERSION,
                            annotatorConfigurationVersion: '',
                            annotatorVersion: '',
                            location: this._getCurrentLocation(),
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        };
                        return data;
                    };
                    return PitchColumnTooltipIconEventFactory;
                }());
                PitchHeaderEventFactory.PitchColumnTooltipIconEventFactory = PitchColumnTooltipIconEventFactory;
            })(PitchHeaderEventFactory = RosterTableHeaderRowEventFactory.PitchHeaderEventFactory || (RosterTableHeaderRowEventFactory.PitchHeaderEventFactory = {}));
        })(RosterTableHeaderRowEventFactory = RosterTableEventFactory.RosterTableHeaderRowEventFactory || (RosterTableEventFactory.RosterTableHeaderRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableDataRowEventFactory;
        (function (RosterTableDataRowEventFactory) {
            var PitchCellEventFactory;
            (function (PitchCellEventFactory) {
                /**
                 * Create PitchASongButtonEventFactory class with path
                 * @param path to this node
                 */
                var PitchASongButtonEventFactory = /** @class */ (function () {
                    function PitchASongButtonEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat(__assign(__assign({}, data), { name: "pitch_a_song_button" }));
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    PitchASongButtonEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    PitchASongButtonEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    /**
                     * Create interaction for hit ui_navigate action.
                   * @param destination the destination field
                     */
                    PitchASongButtonEventFactory.prototype.hitUiNavigate = function (_a) {
                        var destination = _a.destination;
                        var data = {
                            actionParameterNames: ['destination'],
                            actionParameterValues: [destination.toString()],
                            actionName: "ui_navigate",
                            actionVersion: 1,
                            app: APP,
                            generatorVersion: GENERATOR_VERSION,
                            interactionType: "hit",
                            specificationMode: SPECIFICATION_MODE,
                            specificationVersion: SPECIFICATION_VERSION,
                            annotatorConfigurationVersion: '',
                            annotatorVersion: '',
                            location: this._getCurrentLocation(),
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        };
                        return data;
                    };
                    return PitchASongButtonEventFactory;
                }());
                PitchCellEventFactory.PitchASongButtonEventFactory = PitchASongButtonEventFactory;
            })(PitchCellEventFactory = RosterTableDataRowEventFactory.PitchCellEventFactory || (RosterTableDataRowEventFactory.PitchCellEventFactory = {}));
        })(RosterTableDataRowEventFactory = RosterTableEventFactory.RosterTableDataRowEventFactory || (RosterTableEventFactory.RosterTableDataRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
(function (WebRosterEventFactory) {
    var RosterTableEventFactory;
    (function (RosterTableEventFactory) {
        var RosterTableDataRowEventFactory;
        (function (RosterTableDataRowEventFactory) {
            var PitchCellEventFactory;
            (function (PitchCellEventFactory) {
                /**
                 * Create SeeAPitchButtonEventFactory class with path
                 * @param path to this node
                 */
                var SeeAPitchButtonEventFactory = /** @class */ (function () {
                    function SeeAPitchButtonEventFactory(_a) {
                        var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                        this._path = path.concat(__assign(__assign({}, data), { name: "see_a_pitch_button" }));
                        if (parentAbsoluteLocation) {
                            this._parentAbsoluteLocation = parentAbsoluteLocation;
                        }
                    }
                    SeeAPitchButtonEventFactory.prototype._getCurrentLocation = function () {
                        return {
                            pathNodes: this._path,
                            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                        };
                    };
                    SeeAPitchButtonEventFactory.prototype.getAbsoluteLocation = function () {
                        var _a;
                        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                        return {
                            locations: __spreadArrays((parentLocations || []), [
                                this._getCurrentLocation()
                            ]),
                        };
                    };
                    /**
                     * Create interaction for hit ui_navigate action.
                   * @param destination the destination field
                     */
                    SeeAPitchButtonEventFactory.prototype.hitUiNavigate = function (_a) {
                        var destination = _a.destination;
                        var data = {
                            actionParameterNames: ['destination'],
                            actionParameterValues: [destination.toString()],
                            actionName: "ui_navigate",
                            actionVersion: 1,
                            app: APP,
                            generatorVersion: GENERATOR_VERSION,
                            interactionType: "hit",
                            specificationMode: SPECIFICATION_MODE,
                            specificationVersion: SPECIFICATION_VERSION,
                            annotatorConfigurationVersion: '',
                            annotatorVersion: '',
                            location: this._getCurrentLocation(),
                            parentAbsoluteLocation: this._parentAbsoluteLocation
                        };
                        return data;
                    };
                    return SeeAPitchButtonEventFactory;
                }());
                PitchCellEventFactory.SeeAPitchButtonEventFactory = SeeAPitchButtonEventFactory;
            })(PitchCellEventFactory = RosterTableDataRowEventFactory.PitchCellEventFactory || (RosterTableDataRowEventFactory.PitchCellEventFactory = {}));
        })(RosterTableDataRowEventFactory = RosterTableEventFactory.RosterTableDataRowEventFactory || (RosterTableEventFactory.RosterTableDataRowEventFactory = {}));
    })(RosterTableEventFactory = WebRosterEventFactory.RosterTableEventFactory || (WebRosterEventFactory.RosterTableEventFactory = {}));
})(WebRosterEventFactory = exports.WebRosterEventFactory || (exports.WebRosterEventFactory = {}));
exports.WebRosterEventFactory = WebRosterEventFactory;
