// @spotify-internal/spoticharts/cjs/components/RoundedContainer/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const FLAT = 'flat';
const TALL = 'tall';
class RoundedContainer extends react_1.PureComponent {
    render() {
        const { xPos, yPos, width, height, cornerRadius, fill, filter } = this.props;
        const diameter = cornerRadius * 2;
        const Corner = (_a) => {
            var cPos = __rest(_a, []);
            return (react_1.default.createElement("circle", Object.assign({ r: cornerRadius, fill: fill }, cPos)));
        };
        const Side = (_a) => {
            var { orientation = TALL } = _a, pos = __rest(_a, ["orientation"]);
            return (react_1.default.createElement("rect", Object.assign({ width: orientation === FLAT ? width - diameter : cornerRadius, height: orientation === FLAT ? cornerRadius : height - diameter, fill: fill }, pos)));
        };
        return (react_1.default.createElement("g", { filter: filter, transform: `translate(${xPos}, ${yPos})` },
            react_1.default.createElement(Corner, { cx: cornerRadius, cy: cornerRadius }),
            react_1.default.createElement(Corner, { cx: width - cornerRadius, cy: cornerRadius }),
            react_1.default.createElement(Corner, { cx: width - cornerRadius, cy: height - cornerRadius }),
            react_1.default.createElement(Corner, { cx: cornerRadius, cy: height - cornerRadius }),
            react_1.default.createElement(Side, { x: cornerRadius, y: 0, orientation: FLAT }),
            react_1.default.createElement(Side, { x: width - cornerRadius, y: cornerRadius, orientation: TALL }),
            react_1.default.createElement(Side, { x: cornerRadius, y: height - cornerRadius, orientation: FLAT }),
            react_1.default.createElement(Side, { x: 0, y: cornerRadius, orientation: TALL }),
            react_1.default.createElement("rect", { width: width - diameter, height: height - diameter, x: cornerRadius, y: cornerRadius, fill: fill })));
    }
}
exports.default = RoundedContainer;
RoundedContainer.propTypes = {
    xPos: prop_types_1.default.number,
    yPos: prop_types_1.default.number,
    width: prop_types_1.default.number,
    height: prop_types_1.default.number,
    cornerRadius: prop_types_1.default.number,
    fill: prop_types_1.default.string,
    filter: prop_types_1.default.string,
};
RoundedContainer.defaultProps = {
    xPos: 0,
    yPos: 0,
    cornerRadius: 5,
    fill: encore_foundation_1.white,
    filter: '',
};
