// @spotify-internal/address-collection-sdk/esm/components/ErrorContainer/ErrorContainer.js
import { red, white } from '@spotify-internal/encore-web';
import React, { useEffect, useRef } from 'react';
import styled from 'styled-components';
import SvgIcon from './Icon';
export const red5 = '#fdf2f4';
const Wrapper = styled.div `
  background: ${red5};
  text-align: left;
  border-radius: 3px;
  margin: 12px 0;
  padding: 16px;
  display: flex;
  flex-direction: row;

  svg {
    margin-right: 9px;
    margin-left: 0;

    width: 19px;
    height: 19px;
    min-width: 19px;

    circle:first-child {
      fill: ${red};
      stroke: ${red};
    }

    line {
      stroke: ${white};
    }

    circle:last-child {
      fill: ${white};
      stroke: ${white};
    }
  }

  p {
    padding: 0;
    margin: 0;
    font-family: Circular, Helvetica, Arial, sans-serif;
    font-size: 14px;
    font-weight: 400;
    color: #cd1a2b;
    line-height: 1.5;

    a {
      color: ${red} !important;
    }
  }
`;
export default function ErrorContainer({ errors }) {
    const ref = useRef(null);
    useEffect(() => {
        const element = ref.current;
        if (errors && errors.length > 0 && element !== null) {
            element.scrollIntoView({
                behavior: 'smooth',
            });
        }
    }, [errors]);
    return (React.createElement(React.Fragment, null, errors && errors.length > 0 && (React.createElement(Wrapper, { ref: ref, "data-testid": "error-container" },
        React.createElement(SvgIcon, null),
        React.createElement("div", null, errors &&
            errors.map((error, id) => {
                return React.createElement("p", { key: id }, error);
            }))))));
}
