// @spotify-internal/spoticharts/cjs/components/PathPointLine/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleLine = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const Utils_1 = require("../Utils");
const PathPointFind_1 = __importDefault(require("../PathPointFind"));
class SimpleLine extends react_1.PureComponent {
    render() {
        const _a = this.props, { xPos, yPos, yScale, stroke, strokeWidth } = _a, rest = __rest(_a, ["xPos", "yPos", "yScale", "stroke", "strokeWidth"]);
        const cleanedProps = Utils_1.cleanProps(rest);
        if (!yScale || (yScale && !yScale.range)) {
            return react_1.default.createElement("g", null);
        }
        return (react_1.default.createElement("line", Object.assign({ x1: xPos, y1: yScale.range()[0], x2: xPos, y2: yPos, stroke: stroke, strokeWidth: strokeWidth }, cleanedProps)));
    }
}
exports.SimpleLine = SimpleLine;
SimpleLine.propTypes = {
    xPos: prop_types_1.default.number.isRequired,
    yPos: prop_types_1.default.number.isRequired,
    yScale: prop_types_1.default.func.isRequired,
    stroke: prop_types_1.default.string,
    strokeWidth: prop_types_1.default.number,
};
SimpleLine.defaultProps = {
    stroke: encore_foundation_1.black,
    strokeWidth: 1,
};
exports.default = PathPointFind_1.default(SimpleLine);
