// @spotify-internal/adstudio-tape/lib/components/MobilePreview/styled/ImageContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _TooltipInfo = _interopRequireDefault(require("../../TooltipInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IMAGE_SIZE = '245px';

var Image = _styledComponents["default"].img.withConfig({
  displayName: "ImageContainer__Image",
  componentId: "ol2txh-0"
})(["height:inherit;width:inherit;"]);

var ImageWrapper = _styledComponents["default"].div.withConfig({
  displayName: "ImageContainer__ImageWrapper",
  componentId: "ol2txh-1"
})(["background:", ";margin:0 auto;height:", ";width:", ";"], _encoreFoundation.gray90, IMAGE_SIZE, IMAGE_SIZE);

var ImageTooltip = (0, _styledComponents["default"])(_TooltipInfo["default"]).withConfig({
  displayName: "ImageContainer__ImageTooltip",
  componentId: "ol2txh-2"
})(["position:absolute;margin-left:", ";margin-top:", ";svg{color:", ";&:hover{color:", ";}}"], _encoreFoundation.spacer8, _encoreFoundation.spacer16, _encoreFoundation.gray50, _encoreFoundation.white);

var ImageContainer = function ImageContainer(_ref) {
  var imageUrl = _ref.imageUrl,
      imageTooltipMessage = _ref.imageTooltipMessage,
      children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement(ImageWrapper, null, imageTooltipMessage && /*#__PURE__*/_react["default"].createElement(ImageTooltip, {
    tooltipText: imageTooltipMessage,
    triggerOnMouse: true
  }), imageUrl ? /*#__PURE__*/_react["default"].createElement(Image, {
    src: imageUrl
  }) : children);
};

ImageContainer.propTypes = {
  imageUrl: _propTypes["default"].string,
  children: _propTypes["default"].node,
  imageTooltipMessage: _propTypes["default"].string
};
var _default = ImageContainer;
exports["default"] = _default;
