// @spotify-internal/spoticharts/cjs/components/BucketedBar/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const isUndefined_1 = __importDefault(require("lodash/isUndefined"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const HoverableBar = styled_components_1.default('rect') `
  fill: ${props => props.fillColor};

  &:hover {
    ${props => {
    if (props.darkenOnHover) {
        return 'filter: url(#brightness);';
    }
    return '';
}}
  }
`;
class BucketedBar extends react_1.PureComponent {
    render() {
        const { xScale, yScale, x, xEnd, y, yEnd, bucketOffset, thickness, style, onMouseEnter, onMouseMove, onMouseLeave, onClick, showLabel, fill, labelFormat, labelDistance, labelClassName, labelStyle, darkenOnHover, } = this.props;
        const orientation = isUndefined_1.default(xEnd) ? 'vertical' : 'horizontal';
        const className = `rct-chart-bar-${orientation} ${this.props.className || ''}`;
        const labelClass = `rct-chart-bar-label ${labelClassName || ''}`;
        let rectX;
        let rectY;
        let width;
        let height;
        let xText;
        let yText;
        let textAnchor;
        let textValue;
        if (orientation === 'horizontal') {
            rectY = bucketOffset;
            const x0 = xScale(x);
            const x1 = xScale(xEnd);
            rectX = Math.min(x0, x1);
            width = Math.abs(x1 - x0);
            height = thickness;
            xText = Math.max(x0, x1) + labelDistance;
            yText = rectY + thickness / 2 + 5;
            textAnchor = '';
            textValue = xEnd;
        }
        else {
            rectX = bucketOffset;
            const y0 = yScale(y);
            const y1 = yScale(yEnd);
            rectY = Math.min(y0, y1);
            height = Math.abs(y1 - y0);
            width = thickness;
            xText = rectX + thickness / 2;
            yText = rectY - labelDistance;
            textAnchor = 'middle';
            textValue = yEnd;
        }
        const rect = (react_1.default.createElement(HoverableBar, Object.assign({}, {
            x: rectX,
            y: rectY,
            width,
            height,
            className,
            style,
            onMouseEnter,
            onMouseMove,
            onMouseLeave,
            onClick,
            fillColor: fill,
            darkenOnHover,
        })));
        const text = (react_1.default.createElement("text", Object.assign({}, {
            textAnchor,
            x: xText,
            y: yText,
            className: labelClass,
            style: labelStyle,
        }), labelFormat ? labelFormat(textValue) : textValue));
        if (showLabel) {
            return (react_1.default.createElement("g", null,
                rect,
                text));
        }
        return rect;
    }
}
exports.default = BucketedBar;
BucketedBar.propTypes = {
    x: prop_types_1.default.oneOfType([
        prop_types_1.default.number,
        prop_types_1.default.string,
        prop_types_1.default.instanceOf(Date),
    ]),
    y: prop_types_1.default.oneOfType([
        prop_types_1.default.number,
        prop_types_1.default.string,
        prop_types_1.default.instanceOf(Date),
    ]),
    xEnd: prop_types_1.default.oneOfType([
        prop_types_1.default.number,
        prop_types_1.default.string,
        prop_types_1.default.instanceOf(Date),
    ]),
    yEnd: prop_types_1.default.oneOfType([
        prop_types_1.default.number,
        prop_types_1.default.string,
        prop_types_1.default.instanceOf(Date),
    ]),
    thickness: prop_types_1.default.number,
    className: prop_types_1.default.string,
    style: prop_types_1.default.object,
    onMouseMove: prop_types_1.default.func,
    onMouseEnter: prop_types_1.default.func,
    onMouseLeave: prop_types_1.default.func,
    onClick: prop_types_1.default.func,
    xScale: prop_types_1.default.func,
    yScale: prop_types_1.default.func,
    showLabel: prop_types_1.default.bool,
    labelFormat: prop_types_1.default.func,
    labelDistance: prop_types_1.default.number,
    labelClassName: prop_types_1.default.string,
    labelStyle: prop_types_1.default.object,
    bucketOffset: prop_types_1.default.number,
    fill: prop_types_1.default.string,
    darkenOnHover: prop_types_1.default.bool,
};
BucketedBar.defaultProps = {
    x: 0,
    y: 0,
    thickness: 8,
    className: '',
    style: {},
    labelDistance: 24,
    bucketOffset: 0,
    fill: encore_foundation_1.royalBlue,
};
