// @spotify-internal/ubi-sdk-s4a-web-teams/WebTeamsEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamsEventFactory = exports.createWebTeamsEventFactory = void 0;
var SPECIFICATION_ID = "web-teams";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.3.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.24";
function createWebTeamsEventFactory(options) {
    return new WebTeamsEventFactory(options);
}
exports.createWebTeamsEventFactory = createWebTeamsEventFactory;
/**
 *
 */
var WebTeamsEventFactory = /** @class */ (function () {
    function WebTeamsEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of AddTeamButtonEventFactory
    */
    WebTeamsEventFactory.prototype.addTeamButtonFactory = function () {
        return new WebTeamsEventFactory.AddTeamButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @param id of the TableOrGridViewToggleEventFactory
    * @returns A new instance of TableOrGridViewToggleEventFactory
    */
    WebTeamsEventFactory.prototype.tableOrGridViewToggleFactory = function (data) {
        return new WebTeamsEventFactory.TableOrGridViewToggleEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation,
            data: data
        });
    };
    /**
    * @returns A new instance of SearchBoxEventFactory
    */
    WebTeamsEventFactory.prototype.searchBoxFactory = function () {
        return new WebTeamsEventFactory.SearchBoxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SearchBoxLabelEventFactory
    */
    WebTeamsEventFactory.prototype.searchBoxLabelFactory = function () {
        return new WebTeamsEventFactory.SearchBoxLabelEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of YourTeamsTableEventFactory
    */
    WebTeamsEventFactory.prototype.yourTeamsTableFactory = function () {
        return new WebTeamsEventFactory.YourTeamsTableEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of PaginationButtonNextEventFactory
    */
    WebTeamsEventFactory.prototype.paginationButtonNextFactory = function () {
        return new WebTeamsEventFactory.PaginationButtonNextEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of PaginationButtonPreviousEventFactory
    */
    WebTeamsEventFactory.prototype.paginationButtonPreviousFactory = function () {
        return new WebTeamsEventFactory.PaginationButtonPreviousEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamsEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamsEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamsEventFactory;
}());
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create AddTeamButtonEventFactory class with path
     * @param path to this node
     */
    var AddTeamButtonEventFactory = /** @class */ (function () {
        function AddTeamButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "add_team_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        AddTeamButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AddTeamButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        AddTeamButtonEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AddTeamButtonEventFactory;
    }());
    WebTeamsEventFactory.AddTeamButtonEventFactory = AddTeamButtonEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create TableOrGridViewToggleEventFactory class with path
     * @param path to this node
     */
    var TableOrGridViewToggleEventFactory = /** @class */ (function () {
        function TableOrGridViewToggleEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat(__assign(__assign({}, data), { name: "table_or_grid_view_toggle" }));
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        TableOrGridViewToggleEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TableOrGridViewToggleEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_element_toggle action.
         */
        TableOrGridViewToggleEventFactory.prototype.hitUiElementToggle = function () {
            var data = {
                actionName: "ui_element_toggle",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return TableOrGridViewToggleEventFactory;
    }());
    WebTeamsEventFactory.TableOrGridViewToggleEventFactory = TableOrGridViewToggleEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create SearchBoxEventFactory class with path
     * @param path to this node
     */
    var SearchBoxEventFactory = /** @class */ (function () {
        function SearchBoxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "search_box"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of ClearTextBoxButtonEventFactory
        */
        SearchBoxEventFactory.prototype.clearTextBoxButtonFactory = function () {
            return new WebTeamsEventFactory.SearchBoxEventFactory.ClearTextBoxButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        SearchBoxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SearchBoxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for key_stroke search action.
         */
        SearchBoxEventFactory.prototype.keyStrokeSearch = function () {
            var data = {
                actionName: "search",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "key_stroke",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SearchBoxEventFactory;
    }());
    WebTeamsEventFactory.SearchBoxEventFactory = SearchBoxEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create SearchBoxLabelEventFactory class with path
     * @param path to this node
     */
    var SearchBoxLabelEventFactory = /** @class */ (function () {
        function SearchBoxLabelEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "search_box_label"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        SearchBoxLabelEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SearchBoxLabelEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return SearchBoxLabelEventFactory;
    }());
    WebTeamsEventFactory.SearchBoxLabelEventFactory = SearchBoxLabelEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create YourTeamsTableEventFactory class with path
     * @param path to this node
     */
    var YourTeamsTableEventFactory = /** @class */ (function () {
        function YourTeamsTableEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "your_teams_table"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of TableHeaderRowEventFactory
        */
        YourTeamsTableEventFactory.prototype.tableHeaderRowFactory = function () {
            return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableHeaderRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param uri of the TableDataRowEventFactory
        * @returns A new instance of TableDataRowEventFactory
        */
        YourTeamsTableEventFactory.prototype.tableDataRowFactory = function (data) {
            return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableDataRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        YourTeamsTableEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        YourTeamsTableEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return YourTeamsTableEventFactory;
    }());
    WebTeamsEventFactory.YourTeamsTableEventFactory = YourTeamsTableEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create PaginationButtonNextEventFactory class with path
     * @param path to this node
     */
    var PaginationButtonNextEventFactory = /** @class */ (function () {
        function PaginationButtonNextEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "pagination_button_next"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        PaginationButtonNextEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        PaginationButtonNextEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        PaginationButtonNextEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return PaginationButtonNextEventFactory;
    }());
    WebTeamsEventFactory.PaginationButtonNextEventFactory = PaginationButtonNextEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    /**
     * Create PaginationButtonPreviousEventFactory class with path
     * @param path to this node
     */
    var PaginationButtonPreviousEventFactory = /** @class */ (function () {
        function PaginationButtonPreviousEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "pagination_button_previous"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        PaginationButtonPreviousEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        PaginationButtonPreviousEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        PaginationButtonPreviousEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return PaginationButtonPreviousEventFactory;
    }());
    WebTeamsEventFactory.PaginationButtonPreviousEventFactory = PaginationButtonPreviousEventFactory;
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var SearchBoxEventFactory;
    (function (SearchBoxEventFactory) {
        /**
         * Create ClearTextBoxButtonEventFactory class with path
         * @param path to this node
         */
        var ClearTextBoxButtonEventFactory = /** @class */ (function () {
            function ClearTextBoxButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "clear_text_box_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ClearTextBoxButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ClearTextBoxButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit text_clear action.
             */
            ClearTextBoxButtonEventFactory.prototype.hitTextClear = function () {
                var data = {
                    actionName: "text_clear",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ClearTextBoxButtonEventFactory;
        }());
        SearchBoxEventFactory.ClearTextBoxButtonEventFactory = ClearTextBoxButtonEventFactory;
    })(SearchBoxEventFactory = WebTeamsEventFactory.SearchBoxEventFactory || (WebTeamsEventFactory.SearchBoxEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        /**
         * Create TableHeaderRowEventFactory class with path
         * @param path to this node
         */
        var TableHeaderRowEventFactory = /** @class */ (function () {
            function TableHeaderRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "table_header_row"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of TeamNameHeaderEventFactory
            */
            TableHeaderRowEventFactory.prototype.teamNameHeaderFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableHeaderRowEventFactory.TeamNameHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of AdminHeaderEventFactory
            */
            TableHeaderRowEventFactory.prototype.adminHeaderFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableHeaderRowEventFactory.AdminHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of TeamTypeHeaderEventFactory
            */
            TableHeaderRowEventFactory.prototype.teamTypeHeaderFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableHeaderRowEventFactory.TeamTypeHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of AccessLevelHeaderEventFactory
            */
            TableHeaderRowEventFactory.prototype.accessLevelHeaderFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableHeaderRowEventFactory.AccessLevelHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            TableHeaderRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableHeaderRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return TableHeaderRowEventFactory;
        }());
        YourTeamsTableEventFactory.TableHeaderRowEventFactory = TableHeaderRowEventFactory;
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        /**
         * Create TableDataRowEventFactory class with path
         * @param path to this node
         */
        var TableDataRowEventFactory = /** @class */ (function () {
            function TableDataRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "table_data_row" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of TeamNameCellEventFactory
            */
            TableDataRowEventFactory.prototype.teamNameCellFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableDataRowEventFactory.TeamNameCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of AdminCellEventFactory
            */
            TableDataRowEventFactory.prototype.adminCellFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableDataRowEventFactory.AdminCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of TeamTypeCellEventFactory
            */
            TableDataRowEventFactory.prototype.teamTypeCellFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableDataRowEventFactory.TeamTypeCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of AccessLevelCellEventFactory
            */
            TableDataRowEventFactory.prototype.accessLevelCellFactory = function () {
                return new WebTeamsEventFactory.YourTeamsTableEventFactory.TableDataRowEventFactory.AccessLevelCellEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            TableDataRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableDataRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            TableDataRowEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TableDataRowEventFactory;
        }());
        YourTeamsTableEventFactory.TableDataRowEventFactory = TableDataRowEventFactory;
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableHeaderRowEventFactory;
        (function (TableHeaderRowEventFactory) {
            /**
             * Create TeamNameHeaderEventFactory class with path
             * @param path to this node
             */
            var TeamNameHeaderEventFactory = /** @class */ (function () {
                function TeamNameHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_name_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TeamNameHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamNameHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit sort action.
                 */
                TeamNameHeaderEventFactory.prototype.hitSort = function () {
                    var data = {
                        actionName: "sort",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TeamNameHeaderEventFactory;
            }());
            TableHeaderRowEventFactory.TeamNameHeaderEventFactory = TeamNameHeaderEventFactory;
        })(TableHeaderRowEventFactory = YourTeamsTableEventFactory.TableHeaderRowEventFactory || (YourTeamsTableEventFactory.TableHeaderRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableHeaderRowEventFactory;
        (function (TableHeaderRowEventFactory) {
            /**
             * Create AdminHeaderEventFactory class with path
             * @param path to this node
             */
            var AdminHeaderEventFactory = /** @class */ (function () {
                function AdminHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "admin_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AdminHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AdminHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return AdminHeaderEventFactory;
            }());
            TableHeaderRowEventFactory.AdminHeaderEventFactory = AdminHeaderEventFactory;
        })(TableHeaderRowEventFactory = YourTeamsTableEventFactory.TableHeaderRowEventFactory || (YourTeamsTableEventFactory.TableHeaderRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableHeaderRowEventFactory;
        (function (TableHeaderRowEventFactory) {
            /**
             * Create TeamTypeHeaderEventFactory class with path
             * @param path to this node
             */
            var TeamTypeHeaderEventFactory = /** @class */ (function () {
                function TeamTypeHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_type_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TeamTypeHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamTypeHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit sort action.
                 */
                TeamTypeHeaderEventFactory.prototype.hitSort = function () {
                    var data = {
                        actionName: "sort",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TeamTypeHeaderEventFactory;
            }());
            TableHeaderRowEventFactory.TeamTypeHeaderEventFactory = TeamTypeHeaderEventFactory;
        })(TableHeaderRowEventFactory = YourTeamsTableEventFactory.TableHeaderRowEventFactory || (YourTeamsTableEventFactory.TableHeaderRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableHeaderRowEventFactory;
        (function (TableHeaderRowEventFactory) {
            /**
             * Create AccessLevelHeaderEventFactory class with path
             * @param path to this node
             */
            var AccessLevelHeaderEventFactory = /** @class */ (function () {
                function AccessLevelHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "access_level_header"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AccessLevelHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AccessLevelHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return AccessLevelHeaderEventFactory;
            }());
            TableHeaderRowEventFactory.AccessLevelHeaderEventFactory = AccessLevelHeaderEventFactory;
        })(TableHeaderRowEventFactory = YourTeamsTableEventFactory.TableHeaderRowEventFactory || (YourTeamsTableEventFactory.TableHeaderRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableDataRowEventFactory;
        (function (TableDataRowEventFactory) {
            /**
             * Create TeamNameCellEventFactory class with path
             * @param path to this node
             */
            var TeamNameCellEventFactory = /** @class */ (function () {
                function TeamNameCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_name_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TeamNameCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamNameCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return TeamNameCellEventFactory;
            }());
            TableDataRowEventFactory.TeamNameCellEventFactory = TeamNameCellEventFactory;
        })(TableDataRowEventFactory = YourTeamsTableEventFactory.TableDataRowEventFactory || (YourTeamsTableEventFactory.TableDataRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableDataRowEventFactory;
        (function (TableDataRowEventFactory) {
            /**
             * Create AdminCellEventFactory class with path
             * @param path to this node
             */
            var AdminCellEventFactory = /** @class */ (function () {
                function AdminCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "admin_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AdminCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AdminCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                AdminCellEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return AdminCellEventFactory;
            }());
            TableDataRowEventFactory.AdminCellEventFactory = AdminCellEventFactory;
        })(TableDataRowEventFactory = YourTeamsTableEventFactory.TableDataRowEventFactory || (YourTeamsTableEventFactory.TableDataRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableDataRowEventFactory;
        (function (TableDataRowEventFactory) {
            /**
             * Create TeamTypeCellEventFactory class with path
             * @param path to this node
             */
            var TeamTypeCellEventFactory = /** @class */ (function () {
                function TeamTypeCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "team_type_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TeamTypeCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TeamTypeCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return TeamTypeCellEventFactory;
            }());
            TableDataRowEventFactory.TeamTypeCellEventFactory = TeamTypeCellEventFactory;
        })(TableDataRowEventFactory = YourTeamsTableEventFactory.TableDataRowEventFactory || (YourTeamsTableEventFactory.TableDataRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
(function (WebTeamsEventFactory) {
    var YourTeamsTableEventFactory;
    (function (YourTeamsTableEventFactory) {
        var TableDataRowEventFactory;
        (function (TableDataRowEventFactory) {
            /**
             * Create AccessLevelCellEventFactory class with path
             * @param path to this node
             */
            var AccessLevelCellEventFactory = /** @class */ (function () {
                function AccessLevelCellEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "access_level_cell"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AccessLevelCellEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AccessLevelCellEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                return AccessLevelCellEventFactory;
            }());
            TableDataRowEventFactory.AccessLevelCellEventFactory = AccessLevelCellEventFactory;
        })(TableDataRowEventFactory = YourTeamsTableEventFactory.TableDataRowEventFactory || (YourTeamsTableEventFactory.TableDataRowEventFactory = {}));
    })(YourTeamsTableEventFactory = WebTeamsEventFactory.YourTeamsTableEventFactory || (WebTeamsEventFactory.YourTeamsTableEventFactory = {}));
})(WebTeamsEventFactory = exports.WebTeamsEventFactory || (exports.WebTeamsEventFactory = {}));
exports.WebTeamsEventFactory = WebTeamsEventFactory;
