// @spotify-internal/encore-web/es/components/AccordionTitle/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["isExpanded", "heading", "children", "theme"];
import React, { useContext } from 'react';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { deprecatedPropWarning } from "../../deprecationWarnings";
import { Type } from "../Type";
import { AccordionIcon } from "./Icon";
import { Title } from "./Title";
export var AccordionTitle = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$isExpanded = _ref.isExpanded,
      isExpanded = _ref$isExpanded === void 0 ? false : _ref$isExpanded,
      _ref$heading = _ref.heading,
      heading = _ref$heading === void 0 ? 'h2' : _ref$heading,
      children = _ref.children,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    deprecatedPropWarning('theme', 'This component is intended to inherit its parent’s color set or color.');
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(Type, {
    as: heading,
    condensed: true,
    ref: ref
  }, /*#__PURE__*/React.createElement(Title, Object.assign({
    isUsingKeyboard: isUsingKeyboard
  }, props), /*#__PURE__*/React.createElement(Type, {
    condensed: true,
    variant: "celloCanon"
  }, children), /*#__PURE__*/React.createElement(AccordionIcon, {
    isExpanded: isExpanded,
    semanticColor: "textSubdued",
    role: "presentation"
  })));
});
/**
* **AccordionTitle** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/AccordionTitle) |  [Encore](https://encore.spotify.net/web/components/accordion) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=4037:6872) |
*
* Title of the accordion component, seen on open and closed views.
*
* @example
* No example for this component
*
*/

export default AccordionTitle;
