// @spotify-internal/encore-web/es/components/NavPillPanel/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { gray10, gray95 } from '@spotify-internal/encore-foundation';
import styled, { css, ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
var NavPillPanelContainer = styled.div.withConfig({
  displayName: "NavPillPanel__NavPillPanelContainer",
  componentId: "sc-1mvj4lh-0"
})(["", ";"], function (props) {
  return css(["color:", ";&:focus{", "}"], props.theme.navPillPanel === 'dark' ? gray95 : gray10, !props.isUsingKeyboard ? 'outline: none;' : null);
});
var NavPillPanelComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      active = _ref.active,
      _ref$role = _ref.role,
      role = _ref$role === void 0 ? 'tabpanel' : _ref$role,
      _ref$tabIndex = _ref.tabIndex,
      tabIndex = _ref$tabIndex === void 0 ? 0 : _ref$tabIndex,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "active", "role", "tabIndex", "children"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(NavPillPanelContainer, Object.assign({
    ref: ref,
    role: role,
    hidden: active,
    tabIndex: tabIndex,
    isUsingKeyboard: isUsingKeyboard
  }, props), children));
});
var navPillPanelHelpers = {
  darkTheme: {
    navPillPanel: 'dark'
  }
};
/**
* **NavPillPanel** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavPillPanel) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navpill-navpillpanel--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-pill) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1550) |
*
* null
*
* @example
* export const Default = () => <NavPillPanel />;
*
*/

export var NavPillPanel = Object.assign(NavPillPanelComponent, navPillPanelHelpers);
export default NavPillPanel;
