// @spotify-internal/adstudio-tape/lib/components/LayoutFooter/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = LayoutFooter;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _tokens = require("../../styles/tokens");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var propTypes = {
  links: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node)]),
  showCopyright: _propTypes["default"].bool,
  darkTheme: _propTypes["default"].bool,
  className: _propTypes["default"].string
};
var defaultProps = {
  showCopyright: true,
  darkTheme: false
};

var Footer = _styledComponents["default"].footer.withConfig({
  displayName: "LayoutFooter__Footer",
  componentId: "sc-904be3-0"
})(["border-top:1px solid ", ";@media (max-width:", "){font-size:", ";padding-bottom:", ";padding-top:", ";}@media (min-width:", "){font-size:", ";padding:", " ", ";}", ";"], _encoreFoundation.gray60, _encoreFoundation.screenXsMax, _tokens.fontSizeSmall, _encoreFoundation.spacer24, _encoreFoundation.spacer24, _encoreFoundation.screenSmMin, _tokens.fontSizeXs, _encoreFoundation.spacer24, _encoreFoundation.spacer8, function (props) {
  return props.darkTheme && (0, _styledComponents.css)(["background-color:", ";"], _encoreFoundation.black);
});

var NavList = _styledComponents["default"].ul.withConfig({
  displayName: "LayoutFooter__NavList",
  componentId: "sc-904be3-1"
})(["list-style:none;padding-left:0;@media (min-width:", "){align-items:center;display:flex;justify-content:flex-start;margin-bottom:0;margin-top:0;padding-bottom:0;}"], _encoreFoundation.screenSmMin);

var Copyright = _styledComponents["default"].li.withConfig({
  displayName: "LayoutFooter__Copyright",
  componentId: "sc-904be3-2"
})(["color:", ";&:first-child{@media (min-width:", "){padding-left:0;}}@media (max-width:", "){padding-bottom:", ";padding-top:", ";float:right;margin-top:-26px;}@media (min-width:", "){flex:1;padding-left:", ";padding-right:0;text-align:right;}@media (min-width:", "){padding-right:0;}"], _encoreFoundation.gray60, _encoreFoundation.screenSmMin, _encoreFoundation.screenXsMax, _encoreFoundation.spacer4, _encoreFoundation.spacer4, _encoreFoundation.screenSmMin, _encoreFoundation.spacer24, _encoreFoundation.screenMdMin);

function LayoutFooter(props) {
  var links = props.links,
      showCopyright = props.showCopyright,
      darkTheme = props.darkTheme,
      className = props.className;
  return /*#__PURE__*/_react["default"].createElement(Footer, {
    darkTheme: darkTheme,
    className: className
  }, /*#__PURE__*/_react["default"].createElement("nav", null, /*#__PURE__*/_react["default"].createElement(NavList, null, links, showCopyright && /*#__PURE__*/_react["default"].createElement(Copyright, null, "\xA9 ", new Date().getFullYear(), " Spotify AB"))));
}

LayoutFooter.propTypes = propTypes;
LayoutFooter.defaultProps = defaultProps;
