// @spotify-internal/adstudio-shared/lib/components/Autosuggest/SelectedPillsList/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var PillLink = _styledComponents["default"].span.withConfig({
  displayName: "SelectedPillsList__PillLink",
  componentId: "sc-1t0zb12-0"
})(["text-decoration:", ";"], function (props) {
  return props.href ? 'underline' : 'none';
});

var SelectedPillsList = function SelectedPillsList(_ref) {
  var disabled = _ref.disabled,
      clickthroughUrl = _ref.clickthroughUrl,
      items = _ref.items,
      keyField = _ref.keyField,
      _onClick = _ref.onClick,
      _onKeyEnter = _ref.onKeyEnter,
      valueField = _ref.valueField;
  return /*#__PURE__*/_react["default"].createElement("ul", null, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(_Item["default"], {
      disabled: disabled,
      item: item,
      key: item[keyField],
      onClick: function onClick() {
        return _onClick(item);
      },
      onKeyEnter: function onKeyEnter() {
        return _onKeyEnter(item);
      }
    }, /*#__PURE__*/_react["default"].createElement(PillLink, {
      as: !!clickthroughUrl && 'a',
      href: !!clickthroughUrl && "".concat(clickthroughUrl, "/").concat(item[keyField]),
      target: "_blank"
    }, item[valueField]));
  }));
};

SelectedPillsList.propTypes = {
  disabled: _propTypes["default"].bool,
  clickthroughUrl: _propTypes["default"].string,
  items: _propTypes["default"].array.isRequired,
  keyField: _propTypes["default"].string.isRequired,
  onClick: _propTypes["default"].func.isRequired,
  onKeyEnter: _propTypes["default"].func.isRequired,
  valueField: _propTypes["default"].string.isRequired
};
var _default = SelectedPillsList;
exports["default"] = _default;
