// @spotify-internal/adstudio-shared/lib/components/Autosuggest/SelectedPillsList/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SelectedPillsItem = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var IconX = _adstudioTape.Icons.IconX;

var Pill = _styledComponents["default"].li.withConfig({
  displayName: "Item__Pill",
  componentId: "vck5u0-0"
})(["display:inline-block;margin:0 9px 20px 0;height:36px;p{margin-bottom:0;margin-top:0;padding-bottom:11px;padding-top:11px;}div{background-color:", ";border:solid 1px ", ";border-radius:2px;color:", ";cursor:pointer;display:inline-block;font-weight:700;padding:10px;a,span{color:", ";&:hover,&:visited{color:", ";background-color:", " !important;}}span{&:focus{outline:none;}}}"], _encoreAdvertisingWeb.white, function (props) {
  return props.theme.colors.primaryColor;
}, function (props) {
  return props.theme.colors.primaryColor;
}, function (props) {
  return props.theme.colors.primaryColor;
}, function (props) {
  return props.theme.colors.primaryColor;
}, _encoreAdvertisingWeb.white);

var RemoveIcon = (0, _styledComponents["default"])(IconX).withConfig({
  displayName: "Item__RemoveIcon",
  componentId: "vck5u0-1"
})(["left:3px;margin-left:", ";position:relative;top:3px;"], _encoreAdvertisingWeb.spacer8);
/* eslint-disable no-unused-vars */

var SelectedPillsItem = function SelectedPillsItem(_ref) {
  var disabled = _ref.disabled,
      item = _ref.item,
      children = _ref.children,
      onClick = _ref.onClick,
      onKeyEnter = _ref.onKeyEnter,
      props = _objectWithoutProperties(_ref, ["disabled", "item", "children", "onClick", "onKeyEnter"]);

  return /*#__PURE__*/_react["default"].createElement(Pill, null, /*#__PURE__*/_react["default"].createElement("div", props, children, !disabled && /*#__PURE__*/_react["default"].createElement("span", {
    onClick: disabled ? null : onClick,
    onKeyDown: disabled ? null : onKeyEnter,
    role: "button",
    tabIndex: "0"
  }, /*#__PURE__*/_react["default"].createElement(RemoveIcon, {
    iconSize: 16
  }))));
};

exports.SelectedPillsItem = SelectedPillsItem;
SelectedPillsItem.propTypes = {
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node, _propTypes["default"].element]),
  disabled: _propTypes["default"].bool,
  item: _propTypes["default"].object,
  onClick: _propTypes["default"].func,
  onKeyEnter: _propTypes["default"].func
};
var _default = SelectedPillsItem;
exports["default"] = _default;
