// @spotify-internal/encore-advertising-web/esm/components/VideoMobilePreview/styles.js
import React from 'react';
import styled, { css } from 'styled-components';
import { VideoPlayer } from '../VideoPlayer';
import { black, gray90, spacer16, ButtonSecondary, gray20, } from '@spotify-internal/encore-web';
const VideoWrapper = styled.div `
  position: absolute;
  right: 0;
  top: ${props => (props.$isLandscape ? '98px' : '0')};
  width: ${props => props.$width};
  height: ${props => props.$height};
  background-color: ${gray20};
`;
export function VideoContainer({ videoUrl, isLandscape }) {
    const [width, height] = isLandscape ? ['271px', '153px'] : ['271px', '482px'];
    return (React.createElement(VideoWrapper, { "$isLandscape": isLandscape, "$width": width, "$height": height }, videoUrl && (React.createElement(VideoPlayer, { width: width, height: height, videoProps: { src: videoUrl } }))));
}
export const PreviewBody = styled.div `
  background: ${black};
  border-radius: ${spacer16};
  overflow: hidden;
  color: white;
  height: 482px;
  margin: ${props => (props.$showPanel ? '0 auto' : '0')};
  text-align: center;
  width: 271px;
  position: relative;
`;
export const Panel = styled.div `
  border-radius: 4px;
  box-shadow: 0 3px 9px 0 ${gray90};
  padding: 20px 0;
  width: 300px;
`;
export const HeaderWrapper = styled.div `
  display: flex;
  flex-direction: column;
  padding: 12px 0 8px;
  z-index: 5;
  position: relative;
  ${props => props.$withoutHeadline &&
    css `
      & p:first-of-type {
        padding-bottom: ${spacer16};
      }
    `}
`;
export const CTAButton = styled(ButtonSecondary) `
  background-color: white;
  border: none;
  color: black;
  padding-left: 16px;
  padding-right: 16px;
  pointer-events: none;
  z-index: 5;
  position: relative;
  margin-top: ${props => (props.$isLandscape ? '260px' : '320px')};
`;
export const SkipControls = styled.div `
  position: relative;
  margin-top: ${props => (props.$isLandscape ? '76px' : '16px')};
  z-index: 5;
`;
