// @spotify-internal/encore-web/es/components/FlyOut/Wrapper.js
import { screenSmMin, screenXsMax, spacer8, spacer16, white } from '@spotify-internal/encore-foundation';
import styled, { css } from 'styled-components';
import { rootStyle, triangle } from "../../styles";
import { flyoutControlsBgDesktop, flyoutMainBgDesktop, flyoutWidth } from "./variables";
export var Wrapper = styled.nav.withConfig({
  displayName: "Wrapper",
  componentId: "ohnfrw-0"
})(["", ";color:", ";", ";", ";@media (min-width:", "){width:", ";}"], rootStyle(), white, function (props) {
  return props.isLightArrow ? css(["@media (min-width:", "){", ";}"], screenSmMin, triangle({
    color: flyoutControlsBgDesktop,
    position: 'top',
    offset: spacer16,
    width: spacer8
  })) : css(["@media (min-width:", "){", ";}"], screenSmMin, triangle({
    color: flyoutMainBgDesktop,
    position: 'top',
    offset: spacer16,
    width: spacer8
  }));
}, function (props) {
  return props.isMobileFlex && css(["@media (max-width:", "){display:flex;flex-direction:column;height:100vh;}"], screenXsMax);
}, screenSmMin, flyoutWidth);
