// @spotify-internal/encore-web/es/components/PopoverNavigationLink/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

/* stylelint-disable no-descending-specificity */
import React from 'react';
import { gray30, gray50, gray70, gray90, gray95, spacer4, spacer8, spacer12, spacer24, white } from '@spotify-internal/encore-foundation';
import styled, { css } from 'styled-components';
import { buttonLinkReset, rootStyle, typeStyleH4 } from "../../styles";
import { PopoverDesktop } from "../PopoverNavigation/PopoverDesktop";
import { PopoverMobile } from "../PopoverNavigation/PopoverMobile";
import { PopoverNavigationItem } from "../PopoverNavigationItem";

var isDarkTheme = function isDarkTheme(props) {
  var _props$theme;

  return ((_props$theme = props.theme) == null ? void 0 : _props$theme.popover) === 'dark';
};

export var PopoverNavigationLink = styled(function (_ref) {
  var _ref$component = _ref.component,
      Component = _ref$component === void 0 ? 'a' : _ref$component,
      props = _objectWithoutProperties(_ref, ["component"]);

  return /*#__PURE__*/React.createElement(Component, props);
}).withConfig({
  displayName: "PopoverNavigationLink",
  componentId: "lu84ag-0"
})(["", ";", ";color:", ";text-decoration:none;display:inline-block;", " &{border-radius:", ";display:block;padding:10px ", ";margin-left:-", ";margin-right:-", ";&:hover{background-color:", ";}&:active,&:focus{&,&:hover{background-color:", ";}}", ";}", " ", ":first-child &{margin-top:-", ";}", " ", ":last-child &{margin-bottom:-", ";}", " &{", ";padding-bottom:", ";padding-top:", ";}&:hover{color:", ";}&:active,&:focus{&,&:hover{outline:none;color:", ";}}"], function (props) {
  return props.component === 'button' && css(["", ";font-size:inherit;"], buttonLinkReset({
    lineHeight: 'inherit'
  }));
}, rootStyle(), function (props) {
  return isDarkTheme(props) ? gray70 : gray30;
}, PopoverDesktop, spacer4, spacer8, function (props) {
  return isDarkTheme(props) ? '13px' : spacer12;
}, function (props) {
  return isDarkTheme(props) ? '13px' : spacer12;
}, function (props) {
  return isDarkTheme(props) ? gray50 : gray95;
}, function (props) {
  return isDarkTheme(props) ? gray30 : gray90;
}, function (props) {
  return props.component === 'button' && css(["width:calc(100% + ", ");text-align:left;"], isDarkTheme(props) ? '26px' : spacer24);
}, PopoverDesktop, PopoverNavigationItem, spacer8, PopoverDesktop, PopoverNavigationItem, spacer8, PopoverMobile, typeStyleH4(), spacer12, spacer12, function (props) {
  return isDarkTheme(props) ? white : gray30;
}, function (props) {
  return isDarkTheme(props) ? white : gray30;
});
/**
* **PopoverNavigationLink** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PopoverNavigationLink) | [Storybook](https://encore-web.spotify.net/?path=/story/components-popover-popovernavigationlink--default) | [Encore](https://encore.spotify.net/web/components/popover?format=figma-figma#popovernavigation) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=7907:6134) |
*
* PopoverNavigationLinks are links in the PopoverNaviagationList
*
* @example
* export const Default = () => <PopoverNavigationLink href="#">Anchor link</PopoverNavigationLink>;
*
*/

export default PopoverNavigationLink;
