// @spotify-internal/spoticharts/cjs/components/LineChart/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LineChart = void 0;
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const isFunction_1 = __importDefault(require("lodash/isFunction"));
const d3_shape_1 = require("d3-shape");
const reactochart_1 = require("reactochart");
const LineChartMouseOverlay_1 = __importDefault(require("../LineChartMouseOverlay"));
const global_1 = __importDefault(require("../../styles/global"));
class LineChart extends react_1.Component {
    render() {
        const _a = this.props, { curve, fill, lineStyle, opacity, stroke, strokeWidth, onMouseEnter, onMouseLeave } = _a, rest = __rest(_a, ["curve", "fill", "lineStyle", "opacity", "stroke", "strokeWidth", "onMouseEnter", "onMouseLeave"]);
        const newLineStyle = Object.assign(Object.assign({}, lineStyle), { stroke, opacity, fill, strokeWidth });
        return (react_1.default.createElement(react_1.Fragment, null,
            react_1.default.createElement(reactochart_1.LineChart, Object.assign({ curve: !!curve ? curve : undefined, lineStyle: newLineStyle }, rest)),
            isFunction_1.default(onMouseEnter) && (react_1.default.createElement(LineChartMouseOverlay_1.default, Object.assign({ onMouseEnter: onMouseEnter, onMouseLeave: onMouseLeave }, rest))),
            react_1.default.createElement(global_1.default, null)));
    }
}
exports.LineChart = LineChart;
LineChart.propTypes = {
    curve: prop_types_1.default.oneOfType([prop_types_1.default.bool, prop_types_1.default.func]),
    fill: prop_types_1.default.string,
    lineStyle: prop_types_1.default.object,
    opacity: prop_types_1.default.number,
    stroke: prop_types_1.default.string,
    strokeWidth: prop_types_1.default.number,
    onMouseEnter: prop_types_1.default.func,
    onMouseLeave: prop_types_1.default.func,
    x: prop_types_1.default.func,
    y: prop_types_1.default.func,
};
LineChart.defaultProps = {
    curve: d3_shape_1.curveMonotoneX,
    fill: 'none',
    opacity: 1,
    stroke: '#000',
    strokeWidth: 2,
    x: d => d.x,
    y: d => d.y,
    onMouseLeave() { },
};
exports.default = LineChart;
