// @spotify-internal/dunder-client/clients/DunderService.js

// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";


// Exported root namespace
Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var DunderService = {};
    

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.dunder = (function() {
                /**
                 * Namespace dunder.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var dunder = {};

                dunder.DunderService = (function() {
                        
                    /**
                     * Properties of a DunderService.
                     * @memberof com.spotify.ads.dunder
                     */
                     
                     /**
                      * Fetch function type, set when instantiating this service class.
                      * @callback FetchFunction
                      * @param {string} uri
                      * @param {Object} options
                      * @returns {Promise<any>}
                      */
                     
                    /**
                     * An HttpServiceClient class to send requests to the DunderService.
                     * @memberof com.spotify.ads.dunder
                     * @classdesc DunderService to send requests to the DunderService.
                     * @param {FetchFunction} fetchFn
                     * @param {protorequest.HttpServiceConfigOptions} options
                     * @constructor
                     */
                    DunderService = function(fetchFn, options) {
                        this.fetchFn = fetchFn;
                        this.host = options.host;
                        this.baseUrl = options.baseUrl;
                        this.version = options.version || "";
                        this.headers = options.headers || new Headers();
                    };
                
                    /**
                     * Sends the fetch request decorated with the configured host.  Only for internal use.
                     */
                    DunderService.prototype.protoFetch = function(url, opts) {
                      return this.fetchFn(this.host + url, opts);
                    };
                
                    /**
                     * Sets the headers to be sent with all requests to this service.
                     * @function setHeaders
                     * @memberof com.spotify.ads.dunder.DunderService
                     * @param {Headers} headers
                     * @instance
                     * @returns {com.spotify.ads.dunder.DunderService} DunderService
                     */
                    DunderService.prototype.setHeaders = function(headers) {
                      this.headers = headers;
                      return this;
                    };
                
                    /**
                     * Get the headers being sent with all requests to this service.
                     * @function getHeaders
                     * @memberof com.spotify.ads.dunder.DunderService
                     * @instance
                     * @returns {Headers}
                     */
                    DunderService.prototype.getHeaders = function() {
                      return this.headers;
                    };
                
                    /**
                     * Calls DunderService.exportHierarchyCsv
                     * @function exportHierarchyCsv
                     * @memberof com.spotify.ads.dunder.DunderService
                     * @param {com.spotify.ads.dunder.ExportHierarchyRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.dunder.ExportResponse>} exportHierarchyCsv
                     */
                    DunderService.prototype.exportHierarchyCsv = function(request) {
                        const url = '/' + this.baseUrl + '/DunderService/' + this.version +  (this.version ? '/' : '') + 'exportHierarchyCsv';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    return DunderService;
                })();

                return dunder;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();


exports["default"] = $root;
exports["$root"] = $root;
exports["DunderService"] = DunderService;
