// @spotify-internal/adstudio-shared/lib/components/AudienceSizer/BudgetPanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BudgetPanel = BudgetPanel;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _FormatType = require("../../config/FormatType");

var _RadialMeter = _interopRequireDefault(require("../RadialMeter"));

var _MarkdownList = require("./MarkdownList");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var MeterContainerWidth = '230px';
var ICON_SIZE = 16;
var LOW_VALUE_THRESHOLD = 50;
var MED_VALUE_THRESHOLD = 79;

var LoadingSpinnerDiv = _styledComponents["default"].div.withConfig({
  displayName: "BudgetPanel__LoadingSpinnerDiv",
  componentId: "sc-1uq768k-0"
})(["margin:-6px -3px 0 -3px;visibility:", ""], function (p) {
  return p.show ? 'visible' : 'hidden';
});

var BudgetText = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.h4).withConfig({
  displayName: "BudgetPanel__BudgetText",
  componentId: "sc-1uq768k-1"
})(["margin-bottom:", ";"], _encoreAdvertisingWeb.spacer4);

function BudgetPanel(_ref) {
  var audienceDeliveryLikelihood = _ref.audienceDeliveryLikelihood,
      className = _ref.className,
      costRaw = _ref.cost,
      costModel = _ref.costModel,
      currencyFormatter = _ref.currencyFormatter,
      error = _ref.error,
      hasMetRequiredMinReach = _ref.hasMetRequiredMinReach,
      isEditing = _ref.isEditing,
      format = _ref.format,
      strings = _ref.strings,
      serveOnMegaphone = _ref.serveOnMegaphone,
      showLoadSpinner = _ref.showLoadSpinner;
  var cost = costRaw && costModel === strings.IMPRESSIONS_MODEL ? costRaw * 1000 : costRaw;
  var deliveryStatus = inferDeliveryStatus();
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: className
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.BoxGroup, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Box, {
    p: "".concat(_encoreAdvertisingWeb.spacer8, " ").concat(_encoreAdvertisingWeb.spacer8, " ").concat(_encoreAdvertisingWeb.spacer16, " ").concat(_encoreAdvertisingWeb.spacer16),
    flexDirection: "row",
    alignItems: "flex-start"
  }, /*#__PURE__*/_react["default"].createElement(Cost, {
    cost: cost,
    costModel: costModel,
    currencyFormatter: currencyFormatter,
    format: format,
    serveOnMegaphone: serveOnMegaphone,
    strings: strings,
    isEditing: isEditing
  }), /*#__PURE__*/_react["default"].createElement(LoadingSpinnerDiv, {
    show: showLoadSpinner
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.LoadingSpinner, {
    diameter: 32
  }))), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Box, {
    p: _encoreAdvertisingWeb.spacer16
  }, /*#__PURE__*/_react["default"].createElement(BudgetText, {
    variant: _encoreAdvertisingWeb.Type.heading4,
    weight: _encoreAdvertisingWeb.Type.bold
  }, strings.BUDGET_DELIVERY_LABEL), /*#__PURE__*/_react["default"].createElement(Meter, {
    audienceDeliveryLikelihood: audienceDeliveryLikelihood,
    deliveryStatus: deliveryStatus,
    hasMetRequiredMinReach: hasMetRequiredMinReach,
    strings: strings
  }), /*#__PURE__*/_react["default"].createElement(Recommendation, {
    cost: cost,
    deliveryStatus: deliveryStatus,
    serveOnMegaphone: serveOnMegaphone,
    strings: strings
  }))));

  function inferDeliveryStatus() {
    if (error) return strings.DELIVERY_STATUS.ERROR;
    if (typeof audienceDeliveryLikelihood !== 'number' || isNaN(audienceDeliveryLikelihood)) return strings.DELIVERY_STATUS.INVALID;
    if (!hasMetRequiredMinReach) return strings.DELIVERY_STATUS.REACH_TOO_LOW;
    if (audienceDeliveryLikelihood <= LOW_VALUE_THRESHOLD) return strings.DELIVERY_STATUS.LOW_VALUE_THRESHOLD;
    if (audienceDeliveryLikelihood <= MED_VALUE_THRESHOLD) return strings.DELIVERY_STATUS.MED_VALUE_THRESHOLD;
    return null;
  }

  ;
}

;

function Cost(_ref2) {
  var cost = _ref2.cost,
      costModel = _ref2.costModel,
      currencyFormatter = _ref2.currencyFormatter,
      format = _ref2.format,
      serveOnMegaphone = _ref2.serveOnMegaphone,
      isEditing = _ref2.isEditing,
      strings = _ref2.strings;
  var currencyOverrides = getCurrencyOverrides(cost);
  var isVideoFormat = format === _FormatType.Format.VIDEO;

  var labelContent = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null);

  var costJsx = /*#__PURE__*/_react["default"].createElement("strong", null, currencyFormatter(cost, currencyOverrides));

  if (costModel === strings.PAID_LISTENS_MODEL) labelContent = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, costJsx, " ", isVideoFormat ? strings.PER_VIEW : strings.PER_LISTEN);else if (serveOnMegaphone) labelContent = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, costJsx, " ", strings.MAXIMUM_CPM_LABEL, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconWithTooltip, {
    withPortal: true,
    placement: "top",
    content: strings.MAXIMUM_CPM_TOOLTIP
  }));else labelContent = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, costJsx, " ", strings.CPM_LABEL, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconWithTooltip, {
    withPortal: true,
    placement: "top",
    content: strings.CPM_TOOLTIP
  }));
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold,
    style: {
      lineHeight: '1.7em'
    }
  }, strings.REACH_COST_LABEL_TEXT), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, _extends({
    condensed: !isEditing || !serveOnMegaphone,
    variant: _encoreAdvertisingWeb.Type.body2
  }, !cost && {
    color: _encoreAdvertisingWeb.gray60
  }), labelContent), isEditing && serveOnMegaphone && /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
    variant: _encoreAdvertisingWeb.Type.body3,
    color: _encoreAdvertisingWeb.gray50,
    condensed: true
  }, strings.MAXIMUM_CPM_EDIT_DISCLAIMER));
}

function getCurrencyOverrides(cost) {
  var overrides = {}; // calculate the best decimal number based on the cost value. should always show
  // two places no matter how small. 0.00001566 should be 0.000016, etc.

  if (cost) {
    var exponentialPower = Number(Number(cost).toExponential().split('e')[1]);

    if (exponentialPower < -1) {
      overrides.decimals = Math.abs(exponentialPower) + 1;
    }
  }

  return overrides;
}

var MeterContainer = _styledComponents["default"].div.withConfig({
  displayName: "BudgetPanel__MeterContainer",
  componentId: "sc-1uq768k-2"
})(["color:", ";font-size:", ";margin-bottom:", ";width:", ";"], _encoreAdvertisingWeb.gray60, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.spacer16, MeterContainerWidth);

function Meter(_ref3) {
  var audienceDeliveryLikelihood = _ref3.audienceDeliveryLikelihood,
      deliveryStatus = _ref3.deliveryStatus,
      hasMetRequiredMinReach = _ref3.hasMetRequiredMinReach,
      strings = _ref3.strings;
  var radialMeterValue = deliveryStatus === strings.DELIVERY_STATUS.INVALID ? null : audienceDeliveryLikelihood;
  return /*#__PURE__*/_react["default"].createElement(MeterContainer, null, /*#__PURE__*/_react["default"].createElement(_RadialMeter["default"], {
    fill: deliveryStatus === strings.DELIVERY_STATUS.INVALID ? _encoreAdvertisingWeb.gray60 : 'url(#stoplight-gradient)',
    highLabel: strings.HIGH_LABEL,
    lowLabel: strings.LOW_LABEL,
    value: hasMetRequiredMinReach ? radialMeterValue : 0
  }, /*#__PURE__*/_react["default"].createElement("linearGradient", {
    id: "stoplight-gradient",
    x1: "0%",
    y1: "0%",
    y2: "0%"
  }, /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "0%",
    stopColor: "#FF4632"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "50%",
    stopColor: "#F8E95A"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "80%",
    stopColor: "#F8E95A"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "85%",
    stopColor: "#CDF564"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "100%",
    stopColor: "#1ED760"
  }))));
}

function Recommendation(_ref4) {
  var cost = _ref4.cost,
      deliveryStatus = _ref4.deliveryStatus,
      serveOnMegaphone = _ref4.serveOnMegaphone,
      strings = _ref4.strings;

  var _icon;

  switch (deliveryStatus) {
    case _constants.stringConstants.DELIVERY_STATUS.ERROR:
    case _constants.stringConstants.DELIVERY_STATUS.REACH_TOO_LOW:
    case _constants.stringConstants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD:
      _icon = /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconExclamationAlt, {
        color: "failure",
        iconSize: ICON_SIZE
      });
      break;

    case _constants.stringConstants.DELIVERY_STATUS.INVALID:
    case _constants.stringConstants.DELIVERY_STATUS.MED_VALUE_THRESHOLD:
      _icon = null;
      break;

    default:
      _icon = cost ? /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconCheckAlt, {
        color: "success",
        iconSize: ICON_SIZE
      }) : /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconExclamationAlt, {
        color: _encoreAdvertisingWeb.gray60,
        iconSize: ICON_SIZE
      });
  }

  var labelAndBody;

  switch (deliveryStatus) {
    case _constants.stringConstants.DELIVERY_STATUS.ERROR:
      labelAndBody = strings.RECOMMENDATIONTEXT.Error;
      break;

    case _constants.stringConstants.DELIVERY_STATUS.INVALID:
      labelAndBody = strings.RECOMMENDATIONTEXT.NoValue;
      break;

    case _constants.stringConstants.DELIVERY_STATUS.REACH_TOO_LOW:
      labelAndBody = strings.RECOMMENDATIONTEXT.ReachTooLow;
      break;

    case _constants.stringConstants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD:
    case _constants.stringConstants.DELIVERY_STATUS.MED_VALUE_THRESHOLD:
      {
        var recText = strings.RECOMMENDATIONTEXT;
        labelAndBody = {
          Label: deliveryStatus === _constants.stringConstants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD ? recText.LowValue.Label : recText.MedValue.Label,
          Body: serveOnMegaphone ? strings.LOW_OR_MED_VALUE_PODCAST_MESSAGE : strings.LOW_OR_MED_VALUE_MESSAGE
        };
        break;
      }

    default:
      labelAndBody = cost ? strings.RECOMMENDATIONTEXT.HighValue : _objectSpread(_objectSpread({}, strings.RECOMMENDATIONTEXT.NoValue), {}, {
        Body: strings.PLACEHOLDER_MESSAGE
      });
  }

  var _labelAndBody = labelAndBody,
      label = _labelAndBody.Label,
      body = _labelAndBody.Body;
  return body && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconWithText, {
    rightAlign: true,
    icon: function icon() {
      return _icon;
    },
    spacer: _encoreAdvertisingWeb.spacer8,
    style: {
      marginBottom: _encoreAdvertisingWeb.spacer8
    }
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type, {
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold,
    color: _encoreAdvertisingWeb.gray30
  }, /*#__PURE__*/_react["default"].createElement("span", null, label))), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type, {
    variant: _encoreAdvertisingWeb.Type.body3,
    color: _encoreAdvertisingWeb.gray30
  }, /*#__PURE__*/_react["default"].createElement(_MarkdownList.MarkdownList, {
    source: body
  })));
}
