// @spotify-internal/adstudio-shared/lib/components/AudienceSizer/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stringConstants = void 0;

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

var stringConstants = /*#__PURE__*/function () {
  function stringConstants() {
    _classCallCheck(this, stringConstants);
  }

  _createClass(stringConstants, null, [{
    key: "withOverides",
    value: // Everything below here can be overriden by string injection

    /**
     * Will create a new strings object with overrides
     */
    function withOverides() {
      var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      return Object.assign({}, stringConstants, props);
    }
  }]);

  return stringConstants;
}();

exports.stringConstants = stringConstants;
stringConstants.IMPRESSIONS_MODEL = 'CPM';
stringConstants.PAID_LISTENS_MODEL = 'CPCL';
stringConstants.DELIVERY_STATUS = {
  ERROR: 'ERROR',
  INVALID: 'INVALID',
  REACH_TOO_LOW: 'REACH_TOO_LOW',
  LOW_VALUE_THRESHOLD: 'LOW_VALUE_THRESHOLD',
  MED_VALUE_THRESHOLD: 'MED_VALUE_THRESHOLD'
};
stringConstants.PRICER_FORMAT = {
  AUDIO: 'AUDIO',
  VIDEO: 'VIDEO'
};
stringConstants.IMPRESSIONS_MODEL_ADS_SERVED_LABEL = 'Estimated ads served';
stringConstants.PAID_LISTENS_MODEL_ADS_SERVED_LABEL = 'Estimated ad listens';
stringConstants.ESTIMATED_DAILY_REACH_LABEL = 'Estimated reach';
stringConstants.ESTIMATED_FREQUENCY_LABEL = 'Estimated frequency';
stringConstants.BUDGET_DELIVERY_LABEL = 'Budget delivery likelihood';
stringConstants.PAID_LISTENS_MODEL_ADS_VIEWED_LABEL = "Estimated ad views";
stringConstants.PAID_LISTENS_MODEL_ADS_SERVED_TOOLTIP_TEXT = 'Your cost per listen increases as you add more targeting selections.';
stringConstants.IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT = 'Your cost per ad served increases as you add more targeting selections.';
stringConstants.ESTIMATED_DAILY_REACH_TOOLTIP_TEXT = 'The estimated number of unique users who will be served your ad at least once.';
stringConstants.ESTIMATED_FREQUENCY_TOOLTIP_TEXT = 'The estimated average number of times each user will be served your ad.';
stringConstants.PAID_LISTENS_MODEL_ADS_VIEWED_TOOLTIP_TEXT = 'Your cost per view increases as you add more targeting selections.';
stringConstants.DEFAULT_PRICER_ERROR = 'We’re unable to get pricing information. Check your connection or try reloading the page.';
stringConstants.DEFAULT_AUDIENCE_ESTIMATE_ERROR = 'We’re unable to estimate your audience. Check your connection or try reloading the page.';
stringConstants.PLACEHOLDER_MESSAGE = 'Complete required fields';
stringConstants.PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE = "\n  - Choose a format\n  - Select platforms\n  - Set your budget\n  - Select your targeting location\n  ";
stringConstants.INSUFFICIENT_RANGE_TEXT = 'Insufficient';
stringConstants.ERROR_LABEL = 'Unable to estimate audience';
stringConstants.ERROR_MESSAGE = 'Please check your connection or try reloading the page.';
stringConstants.LOW_RECOMMENDATION_LABEL = 'Not likely to be spent';
stringConstants.MED_RECOMMENDATION_LABEL = 'May not be spent';
stringConstants.LOW_OR_MED_VALUE_MESSAGE = "To increase the likelihood of spending your full budget, do one or more of the following:\n  - Extend your schedule\n  - Expand your audience\n  - Add locations (cities, states, provinces, DMAs, etc.)";
stringConstants.LOW_OR_MED_VALUE_PODCAST_MESSAGE = "\n  - Include more episode content topics\n  - Extend your schedule\n  - Expand your audience\n  - Add locations (cities, states, provinces, DMAs, etc.)\n  ";
stringConstants.GOOD_RECOMMENDATION_LABEL = 'Likely to be spent';
stringConstants.GOOD_VALUE_MESSAGE = 'Based on your audience selections, your budget will likely spend in full.';
stringConstants.ACCEPTABLE_AUDIENCE_MESSAGE = 'Your selected audience is suited to your budget.';
stringConstants.REACH_TOO_LOW_LABEL = 'Cannot be spent';
stringConstants.REACH_TOO_LOW_MESSAGE = 'Broaden your audience to proceed.';
stringConstants.GREATER_THAN_PREFIX = 'Greater than ';
stringConstants.LESS_THAN_PREFIX = 'Less than ';
stringConstants.RECOMMENDATIONTEXT = {
  Error: {
    Label: stringConstants.ERROR_LABEL,
    Body: stringConstants.ERROR_MESSAGE
  },
  NoValue: {
    Label: 'Recommendation',
    Body: stringConstants.PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE
  },
  NoValueLegacy: {
    Label: 'Recommendation',
    Body: null
  },
  LowValue: {
    Label: stringConstants.LOW_RECOMMENDATION_LABEL,
    Body: stringConstants.LOW_OR_MED_VALUE_MESSAGE
  },
  MedValue: {
    Label: stringConstants.MED_RECOMMENDATION_LABEL,
    Body: stringConstants.LOW_OR_MED_VALUE_MESSAGE
  },
  HighValue: {
    Label: stringConstants.GOOD_RECOMMENDATION_LABEL,
    Body: stringConstants.ACCEPTABLE_AUDIENCE_MESSAGE
  },
  ReachTooLow: {
    Label: stringConstants.REACH_TOO_LOW_LABEL,
    Body: stringConstants.REACH_TOO_LOW_MESSAGE
  }
};
stringConstants.REACH_COST_LABEL_TEXT = 'Based on your targeting, you can reach this audience for';
stringConstants.PER_LISTEN = 'per listen';
stringConstants.PER_VIEW = 'per view';
stringConstants.HIGH_LABEL = 'High';
stringConstants.LOW_LABEL = 'Low';
stringConstants.CPM_LABEL = stringConstants.IMPRESSIONS_MODEL;
stringConstants.CPM_TOOLTIP = 'Your CPM increases as you add more targeting selections.';
stringConstants.MAXIMUM_CPM_LABEL = 'maximum CPM';
stringConstants.MAXIMUM_CPM_TOOLTIP = 'Maximum CPM is the most you\'ll pay per thousand impressions. Ads that we deliver off Spotify have a lower CPM than ads on Spotify, so your final CPM will vary based on delivery.';
stringConstants.MAXIMUM_CPM_EDIT_DISCLAIMER = 'For ads delivered off Spotify, all impressions today are billed at the new CPM. For ads on Spotify, changes apply going forward.';
