// @spotify-internal/adstudio-shared/lib/components/AudienceSizer/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = AudienceSizer;

var _isEqual2 = _interopRequireDefault(require("lodash/isEqual"));

var _cloneDeep2 = _interopRequireDefault(require("lodash/cloneDeep"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _audienceSizerHelpers = require("../../utils/audienceSizerHelpers");

var _BudgetPanel = require("./BudgetPanel");

var _AudienceEstimatesPanel = require("./AudienceEstimatesPanel");

var _constants = require("./constants");

var _i18n = require("../../config/i18n");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "AudienceSizer__Container",
  componentId: "sc-1rehflq-0"
})(["position:relative;color:", ";"], _encoreAdvertisingWeb.gray10);

function AudienceSizer(_ref) {
  var _state$audience, _state$pricing$cost, _state$pricing, _state$pricing2, _state$pricing3, _state$audience2, _state$audience3, _state$audience4, _state$audience5, _state$audience6, _state$audience7;

  var currencyFormatter = _ref.currencyFormatter,
      _ref$countryCode = _ref.countryCode,
      countryCode = _ref$countryCode === void 0 ? _i18n.DEFAULT_COUNTRY : _ref$countryCode,
      format = _ref.format,
      _ref$hasMetRequiredMi = _ref.hasMetRequiredMinReach,
      hasMetRequiredMinReach = _ref$hasMetRequiredMi === void 0 ? true : _ref$hasMetRequiredMi,
      stringOverrides = _ref.strings,
      _ref$serveOnMegaphone = _ref.serveOnMegaphone,
      serveOnMegaphone = _ref$serveOnMegaphone === void 0 ? false : _ref$serveOnMegaphone,
      _ref$isEditing = _ref.isEditing,
      isEditing = _ref$isEditing === void 0 ? false : _ref$isEditing,
      adAccountId = _ref.adAccountId,
      adAccountCountry = _ref.adAccountCountry,
      artistIds = _ref.artistIds,
      apiHost = _ref.apiHost,
      audienceSegments = _ref.audienceSegments,
      contextualTargetingIds = _ref.contextualTargetingIds,
      contentSafetyIds = _ref.contentSafetyIds,
      customAudiences = _ref.customAudiences,
      flightId = _ref.flightId,
      genres = _ref.genres,
      geos = _ref.geos,
      isScmCampaign = _ref.isScmCampaign,
      onPricingError = _ref.onPricingError,
      onPricingRequest = _ref.onPricingRequest,
      onPricingSuccess = _ref.onPricingSuccess,
      platforms = _ref.platforms,
      playlists = _ref.playlists,
      pricerEndpoint = _ref.pricerEndpoint,
      ageRanges = _ref.ageRanges,
      budget = _ref.budget,
      dateBegin = _ref.dateBegin,
      dateEnd = _ref.dateEnd,
      edgeProxyHost = _ref.edgeProxyHost,
      frequencyCaps = _ref.frequencyCaps,
      genders = _ref.genders,
      onAudienceEstimateError = _ref.onAudienceEstimateError,
      onAudienceEstimateRequest = _ref.onAudienceEstimateRequest,
      onAudienceEstimateSuccess = _ref.onAudienceEstimateSuccess;

  var strings = _constants.stringConstants.withOverides(stringOverrides);

  var _useState = (0, _react.useState)({
    isLoading: true,
    error: null,
    pricing: null,
    audience: null
  }),
      _useState2 = _slicedToArray(_useState, 2),
      state = _useState2[0],
      setState = _useState2[1];

  var refreshProps = [// Pricing props
  adAccountId, adAccountCountry, artistIds, apiHost, audienceSegments, contextualTargetingIds, contentSafetyIds, customAudiences, flightId, format, genres, geos, isScmCampaign, platforms, playlists, pricerEndpoint, // Audience props that aren't already included in pricing
  audienceSegments, ageRanges, budget, customAudiences, dateBegin, dateEnd, edgeProxyHost, frequencyCaps, genders];
  var prevProps = (0, _react.useRef)();
  (0, _react.useEffect)(function () {
    refresh();
  }, refreshProps);
  return /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.VStack, {
    gap: _encoreAdvertisingWeb.spacer16
  }, /*#__PURE__*/_react["default"].createElement(_BudgetPanel.BudgetPanel, {
    audienceDeliveryLikelihood: (_state$audience = state.audience) === null || _state$audience === void 0 ? void 0 : _state$audience.audienceDeliveryLikelihood,
    cost: (_state$pricing$cost = (_state$pricing = state.pricing) === null || _state$pricing === void 0 ? void 0 : _state$pricing.cost) !== null && _state$pricing$cost !== void 0 ? _state$pricing$cost : 0,
    costModel: (_state$pricing2 = state.pricing) === null || _state$pricing2 === void 0 ? void 0 : _state$pricing2.model,
    currencyFormatter: currencyFormatter,
    error: state.error,
    hasMetRequiredMinReach: hasMetRequiredMinReach,
    isEditing: isEditing,
    format: format,
    strings: strings,
    serveOnMegaphone: serveOnMegaphone,
    showLoadSpinner: state.isLoading
  }), /*#__PURE__*/_react["default"].createElement(_AudienceEstimatesPanel.AudienceEstimatesPanel, {
    costModel: (_state$pricing3 = state.pricing) === null || _state$pricing3 === void 0 ? void 0 : _state$pricing3.model,
    countryCode: countryCode,
    format: format,
    hasAudienceEstimateFetched: !!state.audience,
    maxAdsServed: (_state$audience2 = state.audience) === null || _state$audience2 === void 0 ? void 0 : _state$audience2.maxAdsServed,
    maxReach: (_state$audience3 = state.audience) === null || _state$audience3 === void 0 ? void 0 : _state$audience3.maxReach,
    minAdsServed: (_state$audience4 = state.audience) === null || _state$audience4 === void 0 ? void 0 : _state$audience4.minAdsServed,
    minReach: (_state$audience5 = state.audience) === null || _state$audience5 === void 0 ? void 0 : _state$audience5.minReach,
    minFrequency: (_state$audience6 = state.audience) === null || _state$audience6 === void 0 ? void 0 : _state$audience6.minFrequency,
    maxFrequency: (_state$audience7 = state.audience) === null || _state$audience7 === void 0 ? void 0 : _state$audience7.maxFrequency,
    hasMetRequiredMinReach: hasMetRequiredMinReach,
    strings: strings
  })));

  function refresh() {
    return _refresh.apply(this, arguments);
  }

  function _refresh() {
    _refresh = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
      var _pricingRes$costMicro, _pricingRes$model;

      var propsAtStart, pricingReq, pricingRes, audReq, audRes, _setState;

      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _setState = function _setState2(callback) {
                if (!(0, _isEqual2["default"])(propsAtStart, prevProps.current)) return;
                setState(callback);
              };

              if (!(0, _isEqual2["default"])(prevProps.current, refreshProps)) {
                _context.next = 3;
                break;
              }

              return _context.abrupt("return");

            case 3:
              prevProps.current = (0, _cloneDeep2["default"])(refreshProps);
              propsAtStart = (0, _cloneDeep2["default"])(refreshProps);

              _setState(function (s) {
                return _objectSpread(_objectSpread({}, s), {}, {
                  isLoading: true,
                  error: null
                });
              });

              pricingReq = (0, _audienceSizerHelpers.buildPricingRequest)({
                adAccountId: adAccountId,
                adAccountCountry: adAccountCountry,
                artistIds: artistIds,
                apiHost: apiHost,
                audienceSegments: audienceSegments,
                contextualTargetingIds: contextualTargetingIds,
                contentSafetyIds: contentSafetyIds,
                customAudiences: customAudiences,
                flightId: flightId,
                format: format,
                genres: genres,
                geos: geos,
                isScmCampaign: isScmCampaign,
                onPricingError: onPricingError,
                onPricingRequest: onPricingRequest,
                onPricingSuccess: onPricingSuccess,
                platforms: platforms,
                playlists: playlists,
                pricerEndpoint: pricerEndpoint
              });
              _context.next = 9;
              return (0, _audienceSizerHelpers.fetchPricing)(pricingReq)["catch"](function (e) {
                return e;
              });

            case 9:
              pricingRes = _context.sent;

              if (!(pricingRes instanceof Error)) {
                _context.next = 13;
                break;
              }

              _setState({
                isLoading: false,
                error: pricingRes,
                pricing: null,
                audience: null
              });

              return _context.abrupt("return");

            case 13:
              if (pricingRes) {
                _context.next = 16;
                break;
              }

              _setState({
                isLoading: false,
                error: null,
                pricing: null,
                audience: null
              });

              return _context.abrupt("return");

            case 16:
              audReq = (0, _audienceSizerHelpers.buildAudienceEstimateRequest)({
                audienceSegments: audienceSegments,
                ageRanges: ageRanges,
                artistIds: artistIds,
                budget: budget,
                contentSafetyIds: contentSafetyIds,
                contextualTargetingIds: contextualTargetingIds,
                currencyCode: pricingRes.currency,
                customAudiences: customAudiences,
                dateBegin: dateBegin,
                dateEnd: dateEnd,
                edgeProxyHost: edgeProxyHost,
                format: format,
                frequencyCaps: frequencyCaps,
                genders: genders,
                genres: genres,
                geos: geos,
                onAudienceEstimateError: onAudienceEstimateError,
                onAudienceEstimateRequest: onAudienceEstimateRequest,
                onAudienceEstimateSuccess: onAudienceEstimateSuccess,
                platforms: platforms,
                playlists: playlists,
                pricing: {
                  costMicros: (_pricingRes$costMicro = pricingRes.costMicro) !== null && _pricingRes$costMicro !== void 0 ? _pricingRes$costMicro : 0,
                  costModel: (_pricingRes$model = pricingRes.model) !== null && _pricingRes$model !== void 0 ? _pricingRes$model : _constants.stringConstants.IMPRESSIONS_MODEL
                }
              });
              _context.next = 19;
              return (0, _audienceSizerHelpers.fetchAudienceEstimate)(audReq)["catch"](function (e) {
                return e;
              });

            case 19:
              audRes = _context.sent;

              if (!(audRes instanceof Error)) {
                _context.next = 23;
                break;
              }

              _setState({
                isLoading: false,
                error: audRes,
                pricing: null,
                audience: null
              });

              return _context.abrupt("return");

            case 23:
              _setState({
                isLoading: false,
                error: null,
                pricing: pricingRes,
                audience: audRes
              }); // setState that cancels if props change


            case 24:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
    return _refresh.apply(this, arguments);
  }
}
