// @spotify-internal/spoticharts/cjs/components/ShadowedRoundedContainer/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const RoundedContainer_1 = __importDefault(require("../RoundedContainer"));
class ShadowedRoundedContainer extends react_1.PureComponent {
    render() {
        const { xPos, yPos, width, height, cornerRadius, fill } = this.props;
        const filterName = `rounded-container-shadow-filter`;
        return (react_1.default.createElement("g", null,
            react_1.default.createElement("defs", null,
                react_1.default.createElement("filter", { id: filterName, y: "-50%", x: "-50%", width: "250%", height: "250%" },
                    react_1.default.createElement("feColorMatrix", { in: "SourceAlpha", result: "lightened", values: "0 0 1 0 0  0 0 1 0 0  0 0 1 0 0  0 0 0 .5 0" }),
                    react_1.default.createElement("feOffset", { in: "lightened", result: "shadowOffset", dy: "1" }),
                    react_1.default.createElement("feGaussianBlur", { in: "shadowOffset", result: "shadowBlur", stdDeviation: "2" }),
                    react_1.default.createElement("feBlend", { in: "SourceGraphic", in2: "shadowBlur", mode: "normal" }))),
            react_1.default.createElement(RoundedContainer_1.default, { filter: `url(#${filterName})`, xPos: xPos, yPos: yPos, width: width, height: height, cornerRadius: cornerRadius, fill: fill })));
    }
}
exports.default = ShadowedRoundedContainer;
ShadowedRoundedContainer.propTypes = {
    xPos: prop_types_1.default.number.isRequired,
    yPos: prop_types_1.default.number.isRequired,
    width: prop_types_1.default.number,
    height: prop_types_1.default.number,
    cornerRadius: prop_types_1.default.number,
    fill: prop_types_1.default.string,
};
ShadowedRoundedContainer.defaultProps = {
    xPos: 0,
    yPos: 0,
    cornerRadius: 5,
};
