// @spotify-internal/adstudio-webgate-fetch/dist/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoggedOutError = LoggedOutError;
exports.AuthSystemError = AuthSystemError;
exports.THROW_ABOVE_STATUS_CODE = exports.OFFLINE_ERROR_MESSAGE = exports.NETWORK_ERROR_MESSAGE = exports.AUTH_URL_STORAGE_KEY = exports.WEBGATE_TOKEN_STORAGE_KEY = void 0;
var WEBGATE_TOKEN_STORAGE_KEY = 'webgateToken';
exports.WEBGATE_TOKEN_STORAGE_KEY = WEBGATE_TOKEN_STORAGE_KEY;
var AUTH_URL_STORAGE_KEY = 'webgateAuthUrl';
exports.AUTH_URL_STORAGE_KEY = AUTH_URL_STORAGE_KEY;
var NETWORK_ERROR_MESSAGE = 'There was a network error. Please try again.';
exports.NETWORK_ERROR_MESSAGE = NETWORK_ERROR_MESSAGE;
var OFFLINE_ERROR_MESSAGE = 'You are currently offline.';
exports.OFFLINE_ERROR_MESSAGE = OFFLINE_ERROR_MESSAGE;
var THROW_ABOVE_STATUS_CODE = 500;
exports.THROW_ABOVE_STATUS_CODE = THROW_ABOVE_STATUS_CODE;

function LoggedOutError() {
  Error.call(this);
}

LoggedOutError.prototype = Object.create(Error.prototype);

function AuthSystemError(message) {
  Error.call(this);
  this.message = message;
}

AuthSystemError.prototype = Object.create(Error.prototype);
