// @spotify-internal/adstudio-webgate-fetch/dist/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "getAuthUrl", {
  enumerable: true,
  get: function get() {
    return _authUrl.getAuthUrl;
  }
});
Object.defineProperty(exports, "setAuthUrl", {
  enumerable: true,
  get: function get() {
    return _authUrl.setAuthUrl;
  }
});
Object.defineProperty(exports, "LoggedOutError", {
  enumerable: true,
  get: function get() {
    return _constants.LoggedOutError;
  }
});
Object.defineProperty(exports, "AuthSystemError", {
  enumerable: true,
  get: function get() {
    return _constants.AuthSystemError;
  }
});
Object.defineProperty(exports, "webgateFetch", {
  enumerable: true,
  get: function get() {
    return _fetch["default"];
  }
});
Object.defineProperty(exports, "getWebgateToken", {
  enumerable: true,
  get: function get() {
    return _getToken["default"];
  }
});
Object.defineProperty(exports, "fetchWebgateToken", {
  enumerable: true,
  get: function get() {
    return _fetchToken["default"];
  }
});
Object.defineProperty(exports, "protoFetch", {
  enumerable: true,
  get: function get() {
    return _protoFetch["default"];
  }
});
Object.defineProperty(exports, "getCachedWebgateToken", {
  enumerable: true,
  get: function get() {
    return _tokenCache.getCachedWebgateToken;
  }
});
Object.defineProperty(exports, "setCachedWebgateToken", {
  enumerable: true,
  get: function get() {
    return _tokenCache.setCachedWebgateToken;
  }
});
Object.defineProperty(exports, "removeCachedWebgateToken", {
  enumerable: true,
  get: function get() {
    return _tokenCache.removeCachedWebgateToken;
  }
});

var _authUrl = require("./authUrl");

var _constants = require("./constants");

var _fetch = _interopRequireDefault(require("./fetch"));

var _getToken = _interopRequireDefault(require("./getToken"));

var _fetchToken = _interopRequireDefault(require("./fetchToken"));

var _protoFetch = _interopRequireDefault(require("./protoFetch"));

var _tokenCache = require("./tokenCache");
