// @spotify-internal/encore-web/advertising/components/LoadingSpinner/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoadingSpinner = void 0;

var _react = _interopRequireDefault(require("react"));

var _styles = require("../../../cjs/styles");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var offset = 187;
var duration = '2.4s';

var StyledSpinner = _styledComponents["default"].svg.withConfig({
  displayName: "LoadingSpinner__StyledSpinner",
  componentId: "sc-155nxrz-0"
})(["animation:rotator ", " linear infinite;&& .pathIndeterminate{animation:dash ", " ease-in-out infinite;stroke-dasharray:", ";stroke-dashoffset:0;transform-origin:50% 50%;}@keyframes rotator{0%{transform:rotate(0deg);}100%{transform:rotate(270deg);}}@keyframes dash{0%{stroke-dashoffset:", ";}50%{stroke-dashoffset:", ";transform:rotate(135deg);}100%{stroke-dashoffset:", ";transform:rotate(450deg);}}"], duration, duration, offset, offset, offset / 4, offset);

var BrightAccentCircle = _styledComponents["default"].circle.withConfig({
  displayName: "LoadingSpinner__BrightAccentCircle",
  componentId: "sc-155nxrz-1"
})(["stroke:", ";"], (0, _styles.cssColorValue)(_styles.semanticEssentialColors.essentialBrightAccent));

var LoadingSpinner = function LoadingSpinner(_ref) {
  var _ref$diameter = _ref.diameter,
      diameter = _ref$diameter === void 0 ? '210px' : _ref$diameter;
  return /*#__PURE__*/_react["default"].createElement(StyledSpinner, {
    viewBox: "0 0 100 100",
    width: diameter,
    xmlns: "http://www.w3.org/2000/svg"
  }, /*#__PURE__*/_react["default"].createElement(BrightAccentCircle, {
    className: "pathIndeterminate",
    cx: "50",
    cy: "50",
    fill: "none",
    r: "25",
    strokeLinecap: "round",
    strokeWidth: "2.5"
  }));
};

exports.LoadingSpinner = LoadingSpinner;
