// @spotify-internal/adstudio-tape/lib/components/TableRow/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = TableRow;
exports.StyledTableRow = exports.FlexRow = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _flex = require("../../styles/common/flex");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var filledStyle = (0, _styledComponents.css)(["1px solid ", ""], _encoreFoundation.gray90);

var FlexRow = _styledComponents["default"].div.withConfig({
  displayName: "TableRow__FlexRow",
  componentId: "hid58o-0"
})(["display:flex;align-items:center;", " ", ""], _flex.row, _flex.rowAlignCenter);

exports.FlexRow = FlexRow;

var StyledTableRow = _styledComponents["default"].div.withConfig({
  displayName: "TableRow__StyledTableRow",
  componentId: "hid58o-1"
})(["border-top:", ";margin-left:0;margin-right:0;flex-direction:column;&:hover{background:", ";outline:none;}&:focus{background:", ";outline:none;}&:active{background:white;}&:first-child{border-top:1px solid transparent;}&:last-child{border-bottom:", ";}", " ", ""], filledStyle, _encoreFoundation.gray95, _encoreFoundation.gray90, filledStyle, function (props) {
  return props.onClick ? 'cursor: pointer;' : '';
}, function (props) {
  return props.active ? "background-color: ".concat(_encoreFoundation.gray90, ";") : '';
});

exports.StyledTableRow = StyledTableRow;

var StyledAnchor = _styledComponents["default"].a.withConfig({
  displayName: "TableRow__StyledAnchor",
  componentId: "hid58o-2"
})(["color:", ";text-decoration:none;width:100%;&:focus{background:", ";outline:", ";}", " ", ""], _encoreFoundation.black, _encoreFoundation.gray95, filledStyle, _flex.row, _flex.rowAlignCenter);

var propTypes = {
  /** Can accept TableData or any <td> */
  active: _propTypes["default"].bool,
  children: _propTypes["default"].node.isRequired,
  href: _propTypes["default"].string,
  onClick: _propTypes["default"].func,
  onHrefClick: _propTypes["default"].func,
  extraContent: _propTypes["default"].node
};

function TableRow(_ref) {
  var children = _ref.children,
      href = _ref.href,
      onClick = _ref.onClick,
      active = _ref.active,
      extraContent = _ref.extraContent;

  var content = _react["default"].Children.map(children, function (child) {
    return child;
  });

  var handleHrefClick = function handleHrefClick(e) {
    var aNewTab = e.metaKey || e.ctrlKey;
    var anExternalLink = href.startsWith('http');

    if (!aNewTab && !anExternalLink) {
      e.preventDefault();
      onClick(href);
    }
  };

  var anchorProps = href && onClick ? {
    onClick: handleHrefClick
  } : {};
  var innerContent = !href ? content : /*#__PURE__*/_react["default"].createElement(StyledAnchor, _extends({
    href: href
  }, anchorProps), content);
  var props = onClick && !href ? {
    role: 'button',
    tabIndex: '0',
    onClick: onClick
  } : {};
  return /*#__PURE__*/_react["default"].createElement(StyledTableRow, _extends({
    active: active,
    "data-test": "table-row"
  }, props), /*#__PURE__*/_react["default"].createElement(FlexRow, null, innerContent), extraContent && /*#__PURE__*/_react["default"].createElement("div", null, extraContent));
}

TableRow.propTypes = propTypes;
