// @spotify-internal/adstudio-tape/lib/components/Radio/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _encoreWeb = require("@spotify-internal/encore-web");

var _animation = require("../../styles/mixins/animation");

var _Icons = _interopRequireDefault(require("../Icons"));

var _styles = require("./styles");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Wrapper = _styledComponents["default"].div.withConfig({
  displayName: "Radio__Wrapper",
  componentId: "sc-17j5cjr-0"
})(["width:100%;"]);

var IconColumn = _styledComponents["default"].div.withConfig({
  displayName: "Radio__IconColumn",
  componentId: "sc-17j5cjr-1"
})(["", " flex-basis:", ";"], _styles.flexColumnStyles, _encoreFoundation.spacer24);

var RadioColumn = _styledComponents["default"].div.withConfig({
  displayName: "Radio__RadioColumn",
  componentId: "sc-17j5cjr-2"
})(["", " &::after{", "}"], _styles.flexColumnStyles, _styles.radioInputAfterStyles);

var ContentColumn = _styledComponents["default"].div.withConfig({
  displayName: "Radio__ContentColumn",
  componentId: "sc-17j5cjr-3"
})(["", ";min-width:0;"], _styles.contentColumnStyles);

var RadioHitbox = _styledComponents["default"].label.withConfig({
  displayName: "Radio__RadioHitbox",
  componentId: "sc-17j5cjr-4"
})(["display:flex;flex-grow:1;cursor:", ";min-width:0;"], function (props) {
  return props.disabled && 'not-allowed';
});

var RadioLabel = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "Radio__RadioLabel",
  componentId: "sc-17j5cjr-5"
})(["padding-bottom:", ";", ";"], function (props) {
  return props.checked && props.hasContent && !props.group && _encoreFoundation.spacer4;
}, function (props) {
  return props.theme === 'dark' && (0, _styledComponents.css)(["color:", ";"], _encoreFoundation.gray95);
});
var Description = (0, _styledComponents["default"])(RadioLabel).withConfig({
  displayName: "Radio__Description",
  componentId: "sc-17j5cjr-6"
})(["", ""], _styles.descriptionStyles);
var Label = (0, _styledComponents["default"])(RadioLabel).withConfig({
  displayName: "Radio__Label",
  componentId: "sc-17j5cjr-7"
})(["", ";"], function (props) {
  return props.group && (0, _styledComponents.css)(["", " font-weight:bold;color:", ";"], _styles.descriptionStyles, _encoreFoundation.black);
});
var Heading = (0, _styledComponents["default"])(RadioLabel).withConfig({
  displayName: "Radio__Heading",
  componentId: "sc-17j5cjr-8"
})(["", " font-weight:bold;padding-bottom:", ";", ";"], _styles.descriptionStyles, _encoreFoundation.spacer16, function (props) {
  return props.group && (0, _styledComponents.css)(["color:", ";"], _encoreFoundation.black);
});

var RadioInput = _styledComponents["default"].input.withConfig({
  displayName: "Radio__RadioInput",
  componentId: "sc-17j5cjr-9"
})(["position:relative;visibility:hidden;cursor:", ";"], function (props) {
  return props.disabled && 'not-allowed';
});

var Children = _styledComponents["default"].div.withConfig({
  displayName: "Radio__Children",
  componentId: "sc-17j5cjr-10"
})(["", ";", ""], function (props) {
  return props.checked ? (0, _styledComponents.css)(["visibility:visible;height:100%;"]) : (0, _styledComponents.css)(["visibility:collapse;height:0;"]);
}, function (props) {
  return props.animation && _animation.basicAnimation;
});

var Radio = function Radio(_ref) {
  var animation = _ref.animation,
      checked = _ref.checked,
      children = _ref.children,
      description = _ref.description,
      disabled = _ref.disabled,
      group = _ref.group,
      heading = _ref.heading,
      hideListItemIcons = _ref.hideListItemIcons,
      highContrast = _ref.highContrast,
      id = _ref.id,
      invertOnSelect = _ref.invertOnSelect,
      label = _ref.label,
      radioValue = _ref.radioValue,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, ["animation", "checked", "children", "description", "disabled", "group", "heading", "hideListItemIcons", "highContrast", "id", "invertOnSelect", "label", "radioValue", "theme"]);

  var ContentIcon = checked ? _Icons["default"].IconChevronUp : _Icons["default"].IconChevronDown;
  var iconColor = invertOnSelect && checked || highContrast ? 'white' : 'black';
  return /*#__PURE__*/_react["default"].createElement(Wrapper, null, heading && /*#__PURE__*/_react["default"].createElement(Heading, {
    condensed: true,
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book,
    group: group,
    invertOnSelect: invertOnSelect,
    highContrast: highContrast
  }, heading), /*#__PURE__*/_react["default"].createElement(RadioHitbox, {
    htmlFor: id,
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement(RadioColumn, {
    checked: checked,
    disabled: disabled,
    highContrast: highContrast,
    invertOnSelect: invertOnSelect
  }, /*#__PURE__*/_react["default"].createElement(RadioInput, _extends({}, props, {
    checked: checked,
    disabled: disabled,
    group: group,
    highContrast: highContrast,
    id: id,
    invertOnSelect: invertOnSelect,
    type: "radio",
    value: radioValue
  }))), /*#__PURE__*/_react["default"].createElement(ContentColumn, {
    checked: checked,
    highContrast: highContrast,
    invertOnSelect: invertOnSelect
  }, /*#__PURE__*/_react["default"].createElement(Label, {
    condensed: true,
    group: group,
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book,
    checked: checked,
    hasContent: !!children,
    theme: theme
  }, label), description && /*#__PURE__*/_react["default"].createElement(Description, {
    condensed: true,
    group: group,
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book,
    checked: checked,
    hasContent: !!children,
    invertOnSelect: invertOnSelect,
    highContrast: highContrast
  }, description), children && /*#__PURE__*/_react["default"].createElement(Children, {
    animation: animation,
    group: group,
    checked: checked,
    animate: checked
  }, /*#__PURE__*/_react["default"].createElement("div", null, children))), children && !hideListItemIcons && /*#__PURE__*/_react["default"].createElement(IconColumn, null, /*#__PURE__*/_react["default"].createElement(ContentIcon, {
    iconSize: 24,
    color: iconColor
  }))));
};

Radio.propTypes = {
  /* eslint-disable */
  animation: _propTypes["default"].object,

  /* eslint-enable */
  checked: _propTypes["default"].bool,
  children: _propTypes["default"].node,
  description: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  group: _propTypes["default"].bool,
  heading: _propTypes["default"].string,
  hideListItemIcons: _propTypes["default"].bool,
  highContrast: _propTypes["default"].bool,
  id: _propTypes["default"].string,
  invertOnSelect: _propTypes["default"].bool,
  label: _propTypes["default"].string.isRequired,
  radioValue: _propTypes["default"].string.isRequired,
  theme: _propTypes["default"].string
};

var _default = (0, _styledComponents.withTheme)(Radio);

exports["default"] = _default;
