// @spotify-internal/encore-web/es/components/UnsupportedBrowser/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { black, spacer24, spacer64, white } from '@spotify-internal/encore-foundation';
import styled from 'styled-components';
import { rootStyle, zIndexFixed } from "../../styles";
var Wrapper = styled.div.withConfig({
  displayName: "UnsupportedBrowser__Wrapper",
  componentId: "kx5l7g-0"
})(["", ";color:", ";background-color:", ";position:fixed;top:0;left:0;width:100%;height:100%;z-index:", ";display:flex;justify-content:center;align-items:center;flex-direction:column;text-align:center;padding:", ";"], rootStyle(), white, black, zIndexFixed, spacer24);
var BodySvg = styled.svg.withConfig({
  displayName: "UnsupportedBrowser__BodySvg",
  componentId: "kx5l7g-1"
})(["stroke:", ";margin-bottom:", ";"], white, spacer64);
export function UnsupportedBrowser(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/React.createElement(Wrapper, props, /*#__PURE__*/React.createElement(BodySvg, {
    role: "img",
    width: "214",
    height: "155"
  }, /*#__PURE__*/React.createElement("rect", {
    stroke: "#FFF",
    strokeWidth: "3",
    x: "1.5",
    y: "1.5",
    width: "211",
    height: "152",
    rx: "9"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "16",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "32",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "48",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M55 83c3.28 2.667 7.28 4 12 4s8.72-1.333 12-4M135 83c3.28 2.667 7.28 4 12 4s8.72-1.333 12-4",
    stroke: "#FFF",
    strokeWidth: "3",
    strokeLinecap: "square"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M68 122.374c13-4.25 26-6.374 39-6.374s26 2.125 39 6.374M2.5 34.5h210",
    stroke: "#FFF",
    strokeWidth: "3",
    strokeLinecap: "square"
  })), children);
}
/**
* **UnsupportedBrowser** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/UnsupportedBrowser) | [Storybook](https://encore-web.spotify.net/?path=/story/components-unsupportedbrowser--default) | [Encore](https://encore.spotify.net/web/components/unsupported-browser) | 
*
* UnsupportedBrowser shows a simple message and image when a browser is unsupported.
*
* @example
* export const Default = () => (
*  <UnsupportedBrowser>
*    <Type as="p" variant={Type.heading1} condensed>
*      Your browser doesn’t work with this application.
*    </Type>
*    <Type as="p" variant={Type.heading1}>
*      <TextLink href="https://browser-update.org/" theme={TextLink.darkTheme} standalone>
*        Switch or update browsers.
*      </TextLink>
*    </Type>
*  </UnsupportedBrowser>
*);
*
*/

export default UnsupportedBrowser;
