// @spotify-internal/encore-web/cjs/styles/mixins/legacyType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.typeStyleH4 = exports.typeStyleH3 = exports.typeStyleH2 = exports.typeStyleH1 = exports.typeStyleDisplay2 = exports.typeStyleDisplay1 = exports.typeStyleCta4 = exports.typeStyleCta3 = exports.typeStyleCta2 = exports.typeStyleCta1 = exports.typeStyleBody4 = exports.typeStyleBody3 = exports.typeStyleBody2 = exports.typeStyleBody1 = void 0;

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _styledComponents = require("styled-components");

var typeStyleDisplay1 = function typeStyleDisplay1() {
  return (0, _styledComponents.css)(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], _encoreFoundation.type.display1FontSizeXSmall, _encoreFoundation.type.display1FontWeightXSmall, _encoreFoundation.type.display1LetterSpacingXSmall, _encoreFoundation.type.display1LineHeightXSmall, _encoreFoundation.type.display1PaddingBottomXSmall, _encoreFoundation.breakpoint.screenSmMin, _encoreFoundation.type.display1FontSizeSmall, _encoreFoundation.type.display1LineHeightSmall, _encoreFoundation.type.display1LetterSpacingSmall, _encoreFoundation.breakpoint.screenMdMin, _encoreFoundation.type.display1FontSize, _encoreFoundation.type.display1LineHeight);
};

exports.typeStyleDisplay1 = typeStyleDisplay1;

var typeStyleDisplay2 = function typeStyleDisplay2() {
  return (0, _styledComponents.css)(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], _encoreFoundation.type.display2FontSizeXSmall, _encoreFoundation.type.display2FontWeightXSmall, _encoreFoundation.type.display2LetterSpacingXSmall, _encoreFoundation.type.display2LineHeightXSmall, _encoreFoundation.type.display2PaddingBottomXSmall, _encoreFoundation.breakpoint.screenSmMin, _encoreFoundation.type.display2FontSizeSmall, _encoreFoundation.type.display2LineHeightSmall, _encoreFoundation.type.display2LetterSpacingSmall, _encoreFoundation.breakpoint.screenMdMin, _encoreFoundation.type.display2FontSize, _encoreFoundation.type.display2LineHeight);
};

exports.typeStyleDisplay2 = typeStyleDisplay2;

var typeStyleH1 = function typeStyleH1() {
  return (0, _styledComponents.css)(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";}@media (min-width:", "){font-size:", ";line-height:", ";}"], _encoreFoundation.type.heading1FontSizeXSmall, _encoreFoundation.type.heading1FontWeightXSmall, _encoreFoundation.type.heading1LetterSpacingXSmall, _encoreFoundation.type.heading1LineHeightXSmall, _encoreFoundation.type.heading1PaddingBottomXSmall, _encoreFoundation.breakpoint.screenSmMin, _encoreFoundation.type.heading1FontSizeSmall, _encoreFoundation.type.heading1LineHeightSmall, _encoreFoundation.breakpoint.screenMdMin, _encoreFoundation.type.heading1FontSize, _encoreFoundation.type.heading1LineHeight);
};

exports.typeStyleH1 = typeStyleH1;

var typeStyleH2 = function typeStyleH2() {
  return (0, _styledComponents.css)(["font-size:", ";font-weight:", ";letter-spacing:", ";line-height:", ";padding-bottom:", ";@media (min-width:", "){font-size:", ";line-height:", ";letter-spacing:", ";}"], _encoreFoundation.type.heading2FontSizeSmall, _encoreFoundation.type.heading2FontWeightSmall, _encoreFoundation.type.heading2LetterSpacingSmall, _encoreFoundation.type.heading2LineHeightSmall, _encoreFoundation.type.heading2PaddingBottomSmall, _encoreFoundation.breakpoint.screenSmMin, _encoreFoundation.type.heading2FontSize, _encoreFoundation.type.heading2LineHeight, _encoreFoundation.type.heading2LetterSpacing);
};

exports.typeStyleH2 = typeStyleH2;

var typeStyleH3 = function typeStyleH3() {
  return (0, _styledComponents.css)(["font-size:", ";font-weight:", ";line-height:", ";letter-spacing:", ";@media (min-width:", "){font-size:", ";line-height:", ";}"], _encoreFoundation.type.heading3FontSizeSmall, _encoreFoundation.type.heading3FontWeightSmall, _encoreFoundation.type.heading3LineHeightSmall, _encoreFoundation.type.heading3LetterSpacingSmall, _encoreFoundation.breakpoint.screenSmMin, _encoreFoundation.type.heading3FontSize, _encoreFoundation.type.heading3LineHeight);
};

exports.typeStyleH3 = typeStyleH3;

var typeStyleH4 = function typeStyleH4() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.heading4);
};

exports.typeStyleH4 = typeStyleH4;

var typeStyleBody1 = function typeStyleBody1() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.body1);
};

exports.typeStyleBody1 = typeStyleBody1;

var typeStyleBody2 = function typeStyleBody2() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.body2);
};

exports.typeStyleBody2 = typeStyleBody2;

var typeStyleBody3 = function typeStyleBody3() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.body3);
};

exports.typeStyleBody3 = typeStyleBody3;

var typeStyleBody4 = function typeStyleBody4() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.body4);
};

exports.typeStyleBody4 = typeStyleBody4;

var typeStyleCta1 = function typeStyleCta1() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.cta1);
};

exports.typeStyleCta1 = typeStyleCta1;

var typeStyleCta2 = function typeStyleCta2() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.cta2);
};

exports.typeStyleCta2 = typeStyleCta2;

var typeStyleCta3 = function typeStyleCta3() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.cta3);
};

exports.typeStyleCta3 = typeStyleCta3;

var typeStyleCta4 = function typeStyleCta4() {
  return (0, _styledComponents.css)(["", ";"], _encoreFoundation.cta4);
};

exports.typeStyleCta4 = typeStyleCta4;
