// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/ApproveVoiceoverRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var ApproveVoiceoverRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.ApproveVoiceoverRequest = (function() {
                    /**
                     * Properties of a ApproveVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IApproveVoiceoverRequest
                     * @property {string|null} [voiceoverId] ApproveVoiceoverRequest voiceoverId
                     * @property {string|null} [adAccountId] ApproveVoiceoverRequest adAccountId
                     * @property {string|null} [mixId] ApproveVoiceoverRequest mixId
                     * @property {string|null} [setVoiceoverId] ApproveVoiceoverRequest setVoiceoverId
                     * @property {string|null} [setAdAccountId] ApproveVoiceoverRequest setAdAccountId
                     * @property {string|null} [setMixId] ApproveVoiceoverRequest setMixId
                     */
                    
                    /**
                     * Constructs a new ApproveVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a ApproveVoiceoverRequest.
                     * @implements IApproveVoiceoverRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IApproveVoiceoverRequest=} [properties] Properties to set
                     */
                    function ApproveVoiceoverRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * ApproveVoiceoverRequest voiceoverId.
                     * @member {string} voiceoverId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @instance
                     */
                    ApproveVoiceoverRequest.prototype.voiceoverId = "";

                    /**
                     * ApproveVoiceoverRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @instance
                     */
                    ApproveVoiceoverRequest.prototype.adAccountId = "";

                    /**
                     * ApproveVoiceoverRequest mixId.
                     * @member {string} mixId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @instance
                     */
                    ApproveVoiceoverRequest.prototype.mixId = "";


                    /**
                     * Sets voiceoverId
                     * @function setVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ApproveVoiceoverRequest} ApproveVoiceoverRequest
                     */
                    ApproveVoiceoverRequest.prototype.setVoiceoverId = function(value) {
                        this['voiceoverId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ApproveVoiceoverRequest} ApproveVoiceoverRequest
                     */
                    ApproveVoiceoverRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets mixId
                     * @function setMixId
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ApproveVoiceoverRequest} ApproveVoiceoverRequest
                     */
                    ApproveVoiceoverRequest.prototype.setMixId = function(value) {
                        this['mixId'] = value;
                        return this;
                    };

                    /**
                     * Converts the ApproveVoiceoverRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.ApproveVoiceoverRequest
                     * @instance
                     * @returns {object} object
                     */
                    ApproveVoiceoverRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return ApproveVoiceoverRequest;
                })();
                
                ApproveVoiceoverRequest = proto.ApproveVoiceoverRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["ApproveVoiceoverRequest"] = ApproveVoiceoverRequest;
exports["default"] = $root;
