// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/ArchiveHierarchyDraftsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var ArchiveHierarchyDraftsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.ArchiveHierarchyDraftsRequest = (function() {
                    /**
                     * Properties of a ArchiveHierarchyDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IArchiveHierarchyDraftsRequest
                     * @property {string|null} [adAccountId] ArchiveHierarchyDraftsRequest adAccountId
                     * @property {Array.<string>|null} [hierarchyDraftIds] ArchiveHierarchyDraftsRequest hierarchyDraftIds
                     * @property {string|null} [setAdAccountId] ArchiveHierarchyDraftsRequest setAdAccountId
                     * @property {Array.<string>|null} [setHierarchyDraftIds] ArchiveHierarchyDraftsRequest setHierarchyDraftIds
                     */
                    
                    /**
                     * Constructs a new ArchiveHierarchyDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a ArchiveHierarchyDraftsRequest.
                     * @implements IArchiveHierarchyDraftsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IArchiveHierarchyDraftsRequest=} [properties] Properties to set
                     */
                    function ArchiveHierarchyDraftsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * ArchiveHierarchyDraftsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest
                     * @instance
                     */
                    ArchiveHierarchyDraftsRequest.prototype.adAccountId = "";

                    /**
                     * ArchiveHierarchyDraftsRequest hierarchyDraftIds.
                     * @member {Array.<string>} hierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest
                     * @instance
                     */
                    ArchiveHierarchyDraftsRequest.prototype.hierarchyDraftIds = [];


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest} ArchiveHierarchyDraftsRequest
                     */
                    ArchiveHierarchyDraftsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftIds
                     * @function setHierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest} ArchiveHierarchyDraftsRequest
                     */
                    ArchiveHierarchyDraftsRequest.prototype.setHierarchyDraftIds = function(value) {
                        this['hierarchyDraftIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the ArchiveHierarchyDraftsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.ArchiveHierarchyDraftsRequest
                     * @instance
                     * @returns {object} object
                     */
                    ArchiveHierarchyDraftsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return ArchiveHierarchyDraftsRequest;
                })();
                
                ArchiveHierarchyDraftsRequest = proto.ArchiveHierarchyDraftsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["ArchiveHierarchyDraftsRequest"] = ArchiveHierarchyDraftsRequest;
exports["default"] = $root;
