// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/AuthCreateMixPreviewRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AuthCreateMixPreviewRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.AuthCreateMixPreviewRequest = (function() {
                    /**
                     * Properties of a AuthCreateMixPreviewRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IAuthCreateMixPreviewRequest
                     * @property {string|null} [adAccountId] AuthCreateMixPreviewRequest adAccountId
                     * @property {string|null} [voiceFile] AuthCreateMixPreviewRequest voiceFile
                     * @property {string|null} [musicFile] AuthCreateMixPreviewRequest musicFile
                     * @property {boolean|null} [playFullMusic] AuthCreateMixPreviewRequest playFullMusic
                     * @property {com.spotify.adstudioasset.MixMode|null} [mixMode] AuthCreateMixPreviewRequest mixMode
                     * @property {number|null} [voiceStart] AuthCreateMixPreviewRequest voiceStart
                     * @property {number|null} [musicStart] AuthCreateMixPreviewRequest musicStart
                     * @property {number|null} [musicVolume] AuthCreateMixPreviewRequest musicVolume
                     * @property {string|null} [voiceFormat] AuthCreateMixPreviewRequest voiceFormat
                     * @property {string|null} [musicFormat] AuthCreateMixPreviewRequest musicFormat
                     * @property {string|null} [outputFormat] AuthCreateMixPreviewRequest outputFormat
                     * @property {number|null} [duration] AuthCreateMixPreviewRequest duration
                     * @property {string|null} [setAdAccountId] AuthCreateMixPreviewRequest setAdAccountId
                     * @property {string|null} [setVoiceFile] AuthCreateMixPreviewRequest setVoiceFile
                     * @property {string|null} [setMusicFile] AuthCreateMixPreviewRequest setMusicFile
                     * @property {boolean|null} [setPlayFullMusic] AuthCreateMixPreviewRequest setPlayFullMusic
                     * @property {com.spotify.adstudioasset.MixMode|null} [setMixMode] AuthCreateMixPreviewRequest setMixMode
                     * @property {number|null} [setVoiceStart] AuthCreateMixPreviewRequest setVoiceStart
                     * @property {number|null} [setMusicStart] AuthCreateMixPreviewRequest setMusicStart
                     * @property {number|null} [setMusicVolume] AuthCreateMixPreviewRequest setMusicVolume
                     * @property {string|null} [setVoiceFormat] AuthCreateMixPreviewRequest setVoiceFormat
                     * @property {string|null} [setMusicFormat] AuthCreateMixPreviewRequest setMusicFormat
                     * @property {string|null} [setOutputFormat] AuthCreateMixPreviewRequest setOutputFormat
                     * @property {number|null} [setDuration] AuthCreateMixPreviewRequest setDuration
                     */
                    
                    /**
                     * Constructs a new AuthCreateMixPreviewRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a AuthCreateMixPreviewRequest.
                     * @implements IAuthCreateMixPreviewRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IAuthCreateMixPreviewRequest=} [properties] Properties to set
                     */
                    function AuthCreateMixPreviewRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * AuthCreateMixPreviewRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.adAccountId = "";

                    /**
                     * AuthCreateMixPreviewRequest voiceFile.
                     * @member {string} voiceFile
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.voiceFile = "";

                    /**
                     * AuthCreateMixPreviewRequest musicFile.
                     * @member {string} musicFile
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.musicFile = "";

                    /**
                     * AuthCreateMixPreviewRequest playFullMusic.
                     * @member {boolean} playFullMusic
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.playFullMusic = false;

                    /**
                     * AuthCreateMixPreviewRequest mixMode.
                     * @member {com.spotify.adstudioasset.MixMode} mixMode
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.mixMode = {};

                    /**
                     * AuthCreateMixPreviewRequest voiceStart.
                     * @member {number} voiceStart
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.voiceStart = 0;

                    /**
                     * AuthCreateMixPreviewRequest musicStart.
                     * @member {number} musicStart
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.musicStart = 0;

                    /**
                     * AuthCreateMixPreviewRequest musicVolume.
                     * @member {number} musicVolume
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.musicVolume = 0;

                    /**
                     * AuthCreateMixPreviewRequest voiceFormat.
                     * @member {string} voiceFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.voiceFormat = "";

                    /**
                     * AuthCreateMixPreviewRequest musicFormat.
                     * @member {string} musicFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.musicFormat = "";

                    /**
                     * AuthCreateMixPreviewRequest outputFormat.
                     * @member {string} outputFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.outputFormat = "";

                    /**
                     * AuthCreateMixPreviewRequest duration.
                     * @member {number} duration
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     */
                    AuthCreateMixPreviewRequest.prototype.duration = 0;


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceFile
                     * @function setVoiceFile
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setVoiceFile = function(value) {
                        this['voiceFile'] = value;
                        return this;
                    };

                    /**
                     * Sets musicFile
                     * @function setMusicFile
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setMusicFile = function(value) {
                        this['musicFile'] = value;
                        return this;
                    };

                    /**
                     * Sets playFullMusic
                     * @function setPlayFullMusic
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setPlayFullMusic = function(value) {
                        this['playFullMusic'] = value;
                        return this;
                    };

                    /**
                     * Sets mixMode
                     * @function setMixMode
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {com.spotify.adstudioasset.MixMode} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setMixMode = function(value) {
                        this['mixMode'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceStart
                     * @function setVoiceStart
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setVoiceStart = function(value) {
                        this['voiceStart'] = value;
                        return this;
                    };

                    /**
                     * Sets musicStart
                     * @function setMusicStart
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setMusicStart = function(value) {
                        this['musicStart'] = value;
                        return this;
                    };

                    /**
                     * Sets musicVolume
                     * @function setMusicVolume
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setMusicVolume = function(value) {
                        this['musicVolume'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceFormat
                     * @function setVoiceFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setVoiceFormat = function(value) {
                        this['voiceFormat'] = value;
                        return this;
                    };

                    /**
                     * Sets musicFormat
                     * @function setMusicFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setMusicFormat = function(value) {
                        this['musicFormat'] = value;
                        return this;
                    };

                    /**
                     * Sets outputFormat
                     * @function setOutputFormat
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setOutputFormat = function(value) {
                        this['outputFormat'] = value;
                        return this;
                    };

                    /**
                     * Sets duration
                     * @function setDuration
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest} AuthCreateMixPreviewRequest
                     */
                    AuthCreateMixPreviewRequest.prototype.setDuration = function(value) {
                        this['duration'] = value;
                        return this;
                    };

                    /**
                     * Converts the AuthCreateMixPreviewRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.AuthCreateMixPreviewRequest
                     * @instance
                     * @returns {object} object
                     */
                    AuthCreateMixPreviewRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return AuthCreateMixPreviewRequest;
                })();
                
                AuthCreateMixPreviewRequest = proto.AuthCreateMixPreviewRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["AuthCreateMixPreviewRequest"] = AuthCreateMixPreviewRequest;
exports["default"] = $root;
