// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/AuthIsMixPreviewCompleteRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AuthIsMixPreviewCompleteRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.AuthIsMixPreviewCompleteRequest = (function() {
                    /**
                     * Properties of a AuthIsMixPreviewCompleteRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IAuthIsMixPreviewCompleteRequest
                     * @property {string|null} [adAccountId] AuthIsMixPreviewCompleteRequest adAccountId
                     * @property {string|null} [outputFile] AuthIsMixPreviewCompleteRequest outputFile
                     * @property {string|null} [setAdAccountId] AuthIsMixPreviewCompleteRequest setAdAccountId
                     * @property {string|null} [setOutputFile] AuthIsMixPreviewCompleteRequest setOutputFile
                     */
                    
                    /**
                     * Constructs a new AuthIsMixPreviewCompleteRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a AuthIsMixPreviewCompleteRequest.
                     * @implements IAuthIsMixPreviewCompleteRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IAuthIsMixPreviewCompleteRequest=} [properties] Properties to set
                     */
                    function AuthIsMixPreviewCompleteRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * AuthIsMixPreviewCompleteRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest
                     * @instance
                     */
                    AuthIsMixPreviewCompleteRequest.prototype.adAccountId = "";

                    /**
                     * AuthIsMixPreviewCompleteRequest outputFile.
                     * @member {string} outputFile
                     * @memberof com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest
                     * @instance
                     */
                    AuthIsMixPreviewCompleteRequest.prototype.outputFile = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest} AuthIsMixPreviewCompleteRequest
                     */
                    AuthIsMixPreviewCompleteRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets outputFile
                     * @function setOutputFile
                     * @memberof com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest} AuthIsMixPreviewCompleteRequest
                     */
                    AuthIsMixPreviewCompleteRequest.prototype.setOutputFile = function(value) {
                        this['outputFile'] = value;
                        return this;
                    };

                    /**
                     * Converts the AuthIsMixPreviewCompleteRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.AuthIsMixPreviewCompleteRequest
                     * @instance
                     * @returns {object} object
                     */
                    AuthIsMixPreviewCompleteRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return AuthIsMixPreviewCompleteRequest;
                })();
                
                AuthIsMixPreviewCompleteRequest = proto.AuthIsMixPreviewCompleteRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["AuthIsMixPreviewCompleteRequest"] = AuthIsMixPreviewCompleteRequest;
exports["default"] = $root;
