// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/BackgroundRevisionRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var BackgroundRevisionRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.BackgroundRevisionRequest = (function() {
                    /**
                     * Properties of a BackgroundRevisionRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IBackgroundRevisionRequest
                     * @property {string|null} [creativeId] BackgroundRevisionRequest creativeId
                     * @property {string|null} [voiceoverMixId] BackgroundRevisionRequest voiceoverMixId
                     * @property {string|null} [adAccountId] BackgroundRevisionRequest adAccountId
                     * @property {string|null} [bgAssetId] BackgroundRevisionRequest bgAssetId
                     * @property {boolean|null} [playFullMusic] BackgroundRevisionRequest playFullMusic
                     * @property {com.spotify.adstudioasset.MixMode|null} [mixMode] BackgroundRevisionRequest mixMode
                     * @property {number|null} [voiceStart] BackgroundRevisionRequest voiceStart
                     * @property {number|null} [musicStart] BackgroundRevisionRequest musicStart
                     * @property {number|null} [backgroundVolume] BackgroundRevisionRequest backgroundVolume
                     * @property {number|null} [requestedDuration] BackgroundRevisionRequest requestedDuration
                     * @property {string|null} [setCreativeId] BackgroundRevisionRequest setCreativeId
                     * @property {string|null} [setVoiceoverMixId] BackgroundRevisionRequest setVoiceoverMixId
                     * @property {string|null} [setAdAccountId] BackgroundRevisionRequest setAdAccountId
                     * @property {string|null} [setBgAssetId] BackgroundRevisionRequest setBgAssetId
                     * @property {boolean|null} [setPlayFullMusic] BackgroundRevisionRequest setPlayFullMusic
                     * @property {com.spotify.adstudioasset.MixMode|null} [setMixMode] BackgroundRevisionRequest setMixMode
                     * @property {number|null} [setVoiceStart] BackgroundRevisionRequest setVoiceStart
                     * @property {number|null} [setMusicStart] BackgroundRevisionRequest setMusicStart
                     * @property {number|null} [setBackgroundVolume] BackgroundRevisionRequest setBackgroundVolume
                     * @property {number|null} [setRequestedDuration] BackgroundRevisionRequest setRequestedDuration
                     */
                    
                    /**
                     * Constructs a new BackgroundRevisionRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a BackgroundRevisionRequest.
                     * @implements IBackgroundRevisionRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IBackgroundRevisionRequest=} [properties] Properties to set
                     */
                    function BackgroundRevisionRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * BackgroundRevisionRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.creativeId = "";

                    /**
                     * BackgroundRevisionRequest voiceoverMixId.
                     * @member {string} voiceoverMixId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.voiceoverMixId = "";

                    /**
                     * BackgroundRevisionRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.adAccountId = "";

                    /**
                     * BackgroundRevisionRequest bgAssetId.
                     * @member {string} bgAssetId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.bgAssetId = "";

                    /**
                     * BackgroundRevisionRequest playFullMusic.
                     * @member {boolean} playFullMusic
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.playFullMusic = false;

                    /**
                     * BackgroundRevisionRequest mixMode.
                     * @member {com.spotify.adstudioasset.MixMode} mixMode
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.mixMode = {};

                    /**
                     * BackgroundRevisionRequest voiceStart.
                     * @member {number} voiceStart
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.voiceStart = 0;

                    /**
                     * BackgroundRevisionRequest musicStart.
                     * @member {number} musicStart
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.musicStart = 0;

                    /**
                     * BackgroundRevisionRequest backgroundVolume.
                     * @member {number} backgroundVolume
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.backgroundVolume = 0;

                    /**
                     * BackgroundRevisionRequest requestedDuration.
                     * @member {number} requestedDuration
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     */
                    BackgroundRevisionRequest.prototype.requestedDuration = 0;


                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceoverMixId
                     * @function setVoiceoverMixId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setVoiceoverMixId = function(value) {
                        this['voiceoverMixId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets bgAssetId
                     * @function setBgAssetId
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setBgAssetId = function(value) {
                        this['bgAssetId'] = value;
                        return this;
                    };

                    /**
                     * Sets playFullMusic
                     * @function setPlayFullMusic
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setPlayFullMusic = function(value) {
                        this['playFullMusic'] = value;
                        return this;
                    };

                    /**
                     * Sets mixMode
                     * @function setMixMode
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {com.spotify.adstudioasset.MixMode} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setMixMode = function(value) {
                        this['mixMode'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceStart
                     * @function setVoiceStart
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setVoiceStart = function(value) {
                        this['voiceStart'] = value;
                        return this;
                    };

                    /**
                     * Sets musicStart
                     * @function setMusicStart
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setMusicStart = function(value) {
                        this['musicStart'] = value;
                        return this;
                    };

                    /**
                     * Sets backgroundVolume
                     * @function setBackgroundVolume
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setBackgroundVolume = function(value) {
                        this['backgroundVolume'] = value;
                        return this;
                    };

                    /**
                     * Sets requestedDuration
                     * @function setRequestedDuration
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BackgroundRevisionRequest} BackgroundRevisionRequest
                     */
                    BackgroundRevisionRequest.prototype.setRequestedDuration = function(value) {
                        this['requestedDuration'] = value;
                        return this;
                    };

                    /**
                     * Converts the BackgroundRevisionRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.BackgroundRevisionRequest
                     * @instance
                     * @returns {object} object
                     */
                    BackgroundRevisionRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return BackgroundRevisionRequest;
                })();
                
                BackgroundRevisionRequest = proto.BackgroundRevisionRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["BackgroundRevisionRequest"] = BackgroundRevisionRequest;
exports["default"] = $root;
