// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/BffTargeting.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var BffTargeting = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.BffTargeting = (function() {
                    /**
                     * Properties of a BffTargeting.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IBffTargeting
                     * @property {number|null} [ageMin] BffTargeting ageMin
                     * @property {number|null} [ageMax] BffTargeting ageMax
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [geo] BffTargeting geo
                     * @property {Array.<string>|null} [platforms] BffTargeting platforms
                     * @property {Array.<string>|null} [genders] BffTargeting genders
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [genres] BffTargeting genres
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [moments] BffTargeting moments
                     * @property {Array.<string>|null} [audiences] BffTargeting audiences
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [audienceData] BffTargeting audienceData
                     * @property {Array.<string>|null} [audienceSegmentIds] BffTargeting audienceSegmentIds
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [targetedCountry] BffTargeting targetedCountry
                     * @property {Array.<string>|null} [customAudienceIds] BffTargeting customAudienceIds
                     * @property {Array.<com.spotify.adstudiobff.proto.AgeRange>|null} [ageRanges] BffTargeting ageRanges
                     * @property {Array.<string>|null} [brandSafetyCategories] BffTargeting brandSafetyCategories
                     * @property {Array.<string>|null} [contextualTargetingCategories] BffTargeting contextualTargetingCategories
                     * @property {string|null} [competitiveSeparationCategory] BffTargeting competitiveSeparationCategory
                     * @property {string|null} [competitiveSeparationSubcategory] BffTargeting competitiveSeparationSubcategory
                     * @property {number|null} [setAgeMin] BffTargeting setAgeMin
                     * @property {number|null} [setAgeMax] BffTargeting setAgeMax
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [setGeo] BffTargeting setGeo
                     * @property {Array.<string>|null} [setPlatforms] BffTargeting setPlatforms
                     * @property {Array.<string>|null} [setGenders] BffTargeting setGenders
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [setGenres] BffTargeting setGenres
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [setMoments] BffTargeting setMoments
                     * @property {Array.<string>|null} [setAudiences] BffTargeting setAudiences
                     * @property {Array.<com.spotify.adstudiobff.proto.Target>|null} [setAudienceData] BffTargeting setAudienceData
                     * @property {Array.<string>|null} [setAudienceSegmentIds] BffTargeting setAudienceSegmentIds
                     * @property {Array.<com.spotify.adstudiobff.proto.Geo>|null} [setTargetedCountry] BffTargeting setTargetedCountry
                     * @property {Array.<string>|null} [setCustomAudienceIds] BffTargeting setCustomAudienceIds
                     * @property {Array.<com.spotify.adstudiobff.proto.AgeRange>|null} [setAgeRanges] BffTargeting setAgeRanges
                     * @property {Array.<string>|null} [setBrandSafetyCategories] BffTargeting setBrandSafetyCategories
                     * @property {Array.<string>|null} [setContextualTargetingCategories] BffTargeting setContextualTargetingCategories
                     * @property {string|null} [setCompetitiveSeparationCategory] BffTargeting setCompetitiveSeparationCategory
                     * @property {string|null} [setCompetitiveSeparationSubcategory] BffTargeting setCompetitiveSeparationSubcategory
                     */
                    
                    /**
                     * Constructs a new BffTargeting.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a BffTargeting.
                     * @implements IBffTargeting
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IBffTargeting=} [properties] Properties to set
                     */
                    function BffTargeting(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * BffTargeting ageMin.
                     * @member {number} ageMin
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.ageMin = 0;

                    /**
                     * BffTargeting ageMax.
                     * @member {number} ageMax
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.ageMax = 0;

                    /**
                     * BffTargeting geo.
                     * @member {Array.<com.spotify.adstudiobff.proto.Geo>} geo
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.geo = [];

                    /**
                     * BffTargeting platforms.
                     * @member {Array.<string>} platforms
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.platforms = [];

                    /**
                     * BffTargeting genders.
                     * @member {Array.<string>} genders
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.genders = [];

                    /**
                     * BffTargeting genres.
                     * @member {Array.<com.spotify.adstudiobff.proto.Target>} genres
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.genres = [];

                    /**
                     * BffTargeting moments.
                     * @member {Array.<com.spotify.adstudiobff.proto.Target>} moments
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.moments = [];

                    /**
                     * BffTargeting audiences.
                     * @member {Array.<string>} audiences
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.audiences = [];

                    /**
                     * BffTargeting audienceData.
                     * @member {Array.<com.spotify.adstudiobff.proto.Target>} audienceData
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.audienceData = [];

                    /**
                     * BffTargeting audienceSegmentIds.
                     * @member {Array.<string>} audienceSegmentIds
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.audienceSegmentIds = [];

                    /**
                     * BffTargeting targetedCountry.
                     * @member {Array.<com.spotify.adstudiobff.proto.Geo>} targetedCountry
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.targetedCountry = [];

                    /**
                     * BffTargeting customAudienceIds.
                     * @member {Array.<string>} customAudienceIds
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.customAudienceIds = [];

                    /**
                     * BffTargeting ageRanges.
                     * @member {Array.<com.spotify.adstudiobff.proto.AgeRange>} ageRanges
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.ageRanges = [];

                    /**
                     * BffTargeting brandSafetyCategories.
                     * @member {Array.<string>} brandSafetyCategories
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.brandSafetyCategories = [];

                    /**
                     * BffTargeting contextualTargetingCategories.
                     * @member {Array.<string>} contextualTargetingCategories
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.contextualTargetingCategories = [];

                    /**
                     * BffTargeting competitiveSeparationCategory.
                     * @member {string} competitiveSeparationCategory
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.competitiveSeparationCategory = "";

                    /**
                     * BffTargeting competitiveSeparationSubcategory.
                     * @member {string} competitiveSeparationSubcategory
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     */
                    BffTargeting.prototype.competitiveSeparationSubcategory = "";


                    /**
                     * Sets ageMin
                     * @function setAgeMin
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAgeMin = function(value) {
                        this['ageMin'] = value;
                        return this;
                    };

                    /**
                     * Sets ageMax
                     * @function setAgeMax
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAgeMax = function(value) {
                        this['ageMax'] = value;
                        return this;
                    };

                    /**
                     * Sets geo
                     * @function setGeo
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.Geo>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setGeo = function(value) {
                        this['geo'] = value;
                        return this;
                    };

                    /**
                     * Sets platforms
                     * @function setPlatforms
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setPlatforms = function(value) {
                        this['platforms'] = value;
                        return this;
                    };

                    /**
                     * Sets genders
                     * @function setGenders
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setGenders = function(value) {
                        this['genders'] = value;
                        return this;
                    };

                    /**
                     * Sets genres
                     * @function setGenres
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.Target>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setGenres = function(value) {
                        this['genres'] = value;
                        return this;
                    };

                    /**
                     * Sets moments
                     * @function setMoments
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.Target>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setMoments = function(value) {
                        this['moments'] = value;
                        return this;
                    };

                    /**
                     * Sets audiences
                     * @function setAudiences
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAudiences = function(value) {
                        this['audiences'] = value;
                        return this;
                    };

                    /**
                     * Sets audienceData
                     * @function setAudienceData
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.Target>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAudienceData = function(value) {
                        this['audienceData'] = value;
                        return this;
                    };

                    /**
                     * Sets audienceSegmentIds
                     * @function setAudienceSegmentIds
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAudienceSegmentIds = function(value) {
                        this['audienceSegmentIds'] = value;
                        return this;
                    };

                    /**
                     * Sets targetedCountry
                     * @function setTargetedCountry
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.Geo>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setTargetedCountry = function(value) {
                        this['targetedCountry'] = value;
                        return this;
                    };

                    /**
                     * Sets customAudienceIds
                     * @function setCustomAudienceIds
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setCustomAudienceIds = function(value) {
                        this['customAudienceIds'] = value;
                        return this;
                    };

                    /**
                     * Sets ageRanges
                     * @function setAgeRanges
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<com.spotify.adstudiobff.proto.AgeRange>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setAgeRanges = function(value) {
                        this['ageRanges'] = value;
                        return this;
                    };

                    /**
                     * Sets brandSafetyCategories
                     * @function setBrandSafetyCategories
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setBrandSafetyCategories = function(value) {
                        this['brandSafetyCategories'] = value;
                        return this;
                    };

                    /**
                     * Sets contextualTargetingCategories
                     * @function setContextualTargetingCategories
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setContextualTargetingCategories = function(value) {
                        this['contextualTargetingCategories'] = value;
                        return this;
                    };

                    /**
                     * Sets competitiveSeparationCategory
                     * @function setCompetitiveSeparationCategory
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setCompetitiveSeparationCategory = function(value) {
                        this['competitiveSeparationCategory'] = value;
                        return this;
                    };

                    /**
                     * Sets competitiveSeparationSubcategory
                     * @function setCompetitiveSeparationSubcategory
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffTargeting} BffTargeting
                     */
                    BffTargeting.prototype.setCompetitiveSeparationSubcategory = function(value) {
                        this['competitiveSeparationSubcategory'] = value;
                        return this;
                    };

                    /**
                     * Converts the BffTargeting to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.BffTargeting
                     * @instance
                     * @returns {object} object
                     */
                    BffTargeting.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return BffTargeting;
                })();
                
                BffTargeting = proto.BffTargeting;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["BffTargeting"] = BffTargeting;
exports["default"] = $root;
