// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/BffUpdateAdAccountRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var BffUpdateAdAccountRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.BffUpdateAdAccountRequest = (function() {
                    /**
                     * Properties of a BffUpdateAdAccountRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IBffUpdateAdAccountRequest
                     * @property {string|null} [adAccountId] BffUpdateAdAccountRequest adAccountId
                     * @property {com.spotify.adaccountservice.service.model.AccountAddress|null} [businessAddress] BffUpdateAdAccountRequest businessAddress
                     * @property {string|null} [taxId] BffUpdateAdAccountRequest taxId
                     * @property {string|null} [industry] BffUpdateAdAccountRequest industry
                     * @property {string|null} [businessType] BffUpdateAdAccountRequest businessType
                     * @property {string|null} [setAdAccountId] BffUpdateAdAccountRequest setAdAccountId
                     * @property {com.spotify.adaccountservice.service.model.AccountAddress|null} [setBusinessAddress] BffUpdateAdAccountRequest setBusinessAddress
                     * @property {string|null} [setTaxId] BffUpdateAdAccountRequest setTaxId
                     * @property {string|null} [setIndustry] BffUpdateAdAccountRequest setIndustry
                     * @property {string|null} [setBusinessType] BffUpdateAdAccountRequest setBusinessType
                     */
                    
                    /**
                     * Constructs a new BffUpdateAdAccountRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a BffUpdateAdAccountRequest.
                     * @implements IBffUpdateAdAccountRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IBffUpdateAdAccountRequest=} [properties] Properties to set
                     */
                    function BffUpdateAdAccountRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * BffUpdateAdAccountRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     */
                    BffUpdateAdAccountRequest.prototype.adAccountId = "";

                    /**
                     * BffUpdateAdAccountRequest businessAddress.
                     * @member {com.spotify.adaccountservice.service.model.AccountAddress} businessAddress
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     */
                    BffUpdateAdAccountRequest.prototype.businessAddress = {};

                    /**
                     * BffUpdateAdAccountRequest taxId.
                     * @member {string} taxId
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     */
                    BffUpdateAdAccountRequest.prototype.taxId = "";

                    /**
                     * BffUpdateAdAccountRequest industry.
                     * @member {string} industry
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     */
                    BffUpdateAdAccountRequest.prototype.industry = "";

                    /**
                     * BffUpdateAdAccountRequest businessType.
                     * @member {string} businessType
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     */
                    BffUpdateAdAccountRequest.prototype.businessType = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest} BffUpdateAdAccountRequest
                     */
                    BffUpdateAdAccountRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets businessAddress
                     * @function setBusinessAddress
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @param {com.spotify.adaccountservice.service.model.AccountAddress} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest} BffUpdateAdAccountRequest
                     */
                    BffUpdateAdAccountRequest.prototype.setBusinessAddress = function(value) {
                        this['businessAddress'] = value;
                        return this;
                    };

                    /**
                     * Sets taxId
                     * @function setTaxId
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest} BffUpdateAdAccountRequest
                     */
                    BffUpdateAdAccountRequest.prototype.setTaxId = function(value) {
                        this['taxId'] = value;
                        return this;
                    };

                    /**
                     * Sets industry
                     * @function setIndustry
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest} BffUpdateAdAccountRequest
                     */
                    BffUpdateAdAccountRequest.prototype.setIndustry = function(value) {
                        this['industry'] = value;
                        return this;
                    };

                    /**
                     * Sets businessType
                     * @function setBusinessType
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest} BffUpdateAdAccountRequest
                     */
                    BffUpdateAdAccountRequest.prototype.setBusinessType = function(value) {
                        this['businessType'] = value;
                        return this;
                    };

                    /**
                     * Converts the BffUpdateAdAccountRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.BffUpdateAdAccountRequest
                     * @instance
                     * @returns {object} object
                     */
                    BffUpdateAdAccountRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return BffUpdateAdAccountRequest;
                })();
                
                BffUpdateAdAccountRequest = proto.BffUpdateAdAccountRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["BffUpdateAdAccountRequest"] = BffUpdateAdAccountRequest;
exports["default"] = $root;
