// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/BulkGetCreativesRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var BulkGetCreativesRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.BulkGetCreativesRequest = (function() {
                    /**
                     * Properties of a BulkGetCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IBulkGetCreativesRequest
                     * @property {string|null} [adAccountId] BulkGetCreativesRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [creativeIds] BulkGetCreativesRequest creativeIds
                     * @property {string|null} [flightId] BulkGetCreativesRequest flightId
                     * @property {string|null} [setAdAccountId] BulkGetCreativesRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCreativeIds] BulkGetCreativesRequest setCreativeIds
                     * @property {string|null} [setFlightId] BulkGetCreativesRequest setFlightId
                     */
                    
                    /**
                     * Constructs a new BulkGetCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a BulkGetCreativesRequest.
                     * @implements IBulkGetCreativesRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IBulkGetCreativesRequest=} [properties] Properties to set
                     */
                    function BulkGetCreativesRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * BulkGetCreativesRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @instance
                     */
                    BulkGetCreativesRequest.prototype.adAccountId = "";

                    /**
                     * BulkGetCreativesRequest creativeIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} creativeIds
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @instance
                     */
                    BulkGetCreativesRequest.prototype.creativeIds = {};

                    /**
                     * BulkGetCreativesRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @instance
                     */
                    BulkGetCreativesRequest.prototype.flightId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BulkGetCreativesRequest} BulkGetCreativesRequest
                     */
                    BulkGetCreativesRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeIds
                     * @function setCreativeIds
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BulkGetCreativesRequest} BulkGetCreativesRequest
                     */
                    BulkGetCreativesRequest.prototype.setCreativeIds = function(value) {
                        this['creativeIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.BulkGetCreativesRequest} BulkGetCreativesRequest
                     */
                    BulkGetCreativesRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Converts the BulkGetCreativesRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.BulkGetCreativesRequest
                     * @instance
                     * @returns {object} object
                     */
                    BulkGetCreativesRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return BulkGetCreativesRequest;
                })();
                
                BulkGetCreativesRequest = proto.BulkGetCreativesRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["BulkGetCreativesRequest"] = BulkGetCreativesRequest;
exports["default"] = $root;
