// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CompleteHierarchyDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CompleteHierarchyDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CompleteHierarchyDraftRequest = (function() {
                    /**
                     * Properties of a CompleteHierarchyDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICompleteHierarchyDraftRequest
                     * @property {string|null} [adAccountId] CompleteHierarchyDraftRequest adAccountId
                     * @property {string|null} [hierarchyDraftId] CompleteHierarchyDraftRequest hierarchyDraftId
                     * @property {string|null} [setAdAccountId] CompleteHierarchyDraftRequest setAdAccountId
                     * @property {string|null} [setHierarchyDraftId] CompleteHierarchyDraftRequest setHierarchyDraftId
                     */
                    
                    /**
                     * Constructs a new CompleteHierarchyDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CompleteHierarchyDraftRequest.
                     * @implements ICompleteHierarchyDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICompleteHierarchyDraftRequest=} [properties] Properties to set
                     */
                    function CompleteHierarchyDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CompleteHierarchyDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest
                     * @instance
                     */
                    CompleteHierarchyDraftRequest.prototype.adAccountId = "";

                    /**
                     * CompleteHierarchyDraftRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest
                     * @instance
                     */
                    CompleteHierarchyDraftRequest.prototype.hierarchyDraftId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest} CompleteHierarchyDraftRequest
                     */
                    CompleteHierarchyDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest} CompleteHierarchyDraftRequest
                     */
                    CompleteHierarchyDraftRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Converts the CompleteHierarchyDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CompleteHierarchyDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    CompleteHierarchyDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CompleteHierarchyDraftRequest;
                })();
                
                CompleteHierarchyDraftRequest = proto.CompleteHierarchyDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CompleteHierarchyDraftRequest"] = CompleteHierarchyDraftRequest;
exports["default"] = $root;
