// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateColdStartDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateColdStartDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateColdStartDraftRequest = (function() {
                    /**
                     * Properties of a CreateColdStartDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateColdStartDraftRequest
                     * @property {string|null} [adAccountId] CreateColdStartDraftRequest adAccountId
                     * @property {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues|null} [campaignDraftFormValues] CreateColdStartDraftRequest campaignDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [flightDraftFormValues] CreateColdStartDraftRequest flightDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [creativeDraftFormValues] CreateColdStartDraftRequest creativeDraftFormValues
                     * @property {string|null} [setAdAccountId] CreateColdStartDraftRequest setAdAccountId
                     * @property {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues|null} [setCampaignDraftFormValues] CreateColdStartDraftRequest setCampaignDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [setFlightDraftFormValues] CreateColdStartDraftRequest setFlightDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [setCreativeDraftFormValues] CreateColdStartDraftRequest setCreativeDraftFormValues
                     */
                    
                    /**
                     * Constructs a new CreateColdStartDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateColdStartDraftRequest.
                     * @implements ICreateColdStartDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateColdStartDraftRequest=} [properties] Properties to set
                     */
                    function CreateColdStartDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateColdStartDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @instance
                     */
                    CreateColdStartDraftRequest.prototype.adAccountId = "";

                    /**
                     * CreateColdStartDraftRequest campaignDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues} campaignDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @instance
                     */
                    CreateColdStartDraftRequest.prototype.campaignDraftFormValues = {};

                    /**
                     * CreateColdStartDraftRequest flightDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} flightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @instance
                     */
                    CreateColdStartDraftRequest.prototype.flightDraftFormValues = {};

                    /**
                     * CreateColdStartDraftRequest creativeDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} creativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @instance
                     */
                    CreateColdStartDraftRequest.prototype.creativeDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateColdStartDraftRequest} CreateColdStartDraftRequest
                     */
                    CreateColdStartDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignDraftFormValues
                     * @function setCampaignDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateColdStartDraftRequest} CreateColdStartDraftRequest
                     */
                    CreateColdStartDraftRequest.prototype.setCampaignDraftFormValues = function(value) {
                        this['campaignDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Sets flightDraftFormValues
                     * @function setFlightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateColdStartDraftRequest} CreateColdStartDraftRequest
                     */
                    CreateColdStartDraftRequest.prototype.setFlightDraftFormValues = function(value) {
                        this['flightDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeDraftFormValues
                     * @function setCreativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateColdStartDraftRequest} CreateColdStartDraftRequest
                     */
                    CreateColdStartDraftRequest.prototype.setCreativeDraftFormValues = function(value) {
                        this['creativeDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateColdStartDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateColdStartDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateColdStartDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateColdStartDraftRequest;
                })();
                
                CreateColdStartDraftRequest = proto.CreateColdStartDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateColdStartDraftRequest"] = CreateColdStartDraftRequest;
exports["default"] = $root;
