// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateUploadUrlRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateUploadUrlRequest = {};
var Type = {};
var Orientation = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateUploadUrlRequest = (function() {
                    /**
                     * Properties of a CreateUploadUrlRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateUploadUrlRequest
                     * @property {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Type|null} [type] CreateUploadUrlRequest type
                     * @property {string|null} [mediaType] CreateUploadUrlRequest mediaType
                     * @property {string|null} [adAccountId] CreateUploadUrlRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Orientation|null} [orientation] CreateUploadUrlRequest orientation
                     * @property {number|null} [aspectRatio] CreateUploadUrlRequest aspectRatio
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [formatType] CreateUploadUrlRequest formatType
                     * @property {com.spotify.adstudioasset.AudioMediaType|null} [audioMediaType] CreateUploadUrlRequest audioMediaType
                     * @property {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Type|null} [setType] CreateUploadUrlRequest setType
                     * @property {string|null} [setMediaType] CreateUploadUrlRequest setMediaType
                     * @property {string|null} [setAdAccountId] CreateUploadUrlRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Orientation|null} [setOrientation] CreateUploadUrlRequest setOrientation
                     * @property {number|null} [setAspectRatio] CreateUploadUrlRequest setAspectRatio
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormatType] CreateUploadUrlRequest setFormatType
                     * @property {com.spotify.adstudioasset.AudioMediaType|null} [setAudioMediaType] CreateUploadUrlRequest setAudioMediaType
                     */
                    
                    /**
                     * Constructs a new CreateUploadUrlRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateUploadUrlRequest.
                     * @implements ICreateUploadUrlRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateUploadUrlRequest=} [properties] Properties to set
                     */
                    function CreateUploadUrlRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateUploadUrlRequest type.
                     * @member {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Type} type
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.type = {};

                    /**
                     * CreateUploadUrlRequest mediaType.
                     * @member {string} mediaType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.mediaType = "";

                    /**
                     * CreateUploadUrlRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.adAccountId = "";

                    /**
                     * CreateUploadUrlRequest orientation.
                     * @member {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Orientation} orientation
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.orientation = {};

                    /**
                     * CreateUploadUrlRequest aspectRatio.
                     * @member {number} aspectRatio
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.aspectRatio = 0;

                    /**
                     * CreateUploadUrlRequest formatType.
                     * @member {com.spotify.ads.cs.proto.FormatType} formatType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.formatType = {};

                    /**
                     * CreateUploadUrlRequest audioMediaType.
                     * @member {com.spotify.adstudioasset.AudioMediaType} audioMediaType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     */
                    CreateUploadUrlRequest.prototype.audioMediaType = {};


                    /**
                     * Sets type
                     * @function setType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Type} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setType = function(value) {
                        this['type'] = value;
                        return this;
                    };

                    /**
                     * Sets mediaType
                     * @function setMediaType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setMediaType = function(value) {
                        this['mediaType'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets orientation
                     * @function setOrientation
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Orientation} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setOrientation = function(value) {
                        this['orientation'] = value;
                        return this;
                    };

                    /**
                     * Sets aspectRatio
                     * @function setAspectRatio
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setAspectRatio = function(value) {
                        this['aspectRatio'] = value;
                        return this;
                    };

                    /**
                     * Sets formatType
                     * @function setFormatType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setFormatType = function(value) {
                        this['formatType'] = value;
                        return this;
                    };

                    /**
                     * Sets audioMediaType
                     * @function setAudioMediaType
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @param {com.spotify.adstudioasset.AudioMediaType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateUploadUrlRequest} CreateUploadUrlRequest
                     */
                    CreateUploadUrlRequest.prototype.setAudioMediaType = function(value) {
                        this['audioMediaType'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateUploadUrlRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateUploadUrlRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateUploadUrlRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    /**
                     * Type enum.
                     * @name com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Type
                     * @enum {string}

                     * @property {string} bgmusic=bgmusic bgmusic value
                     * @property {string} fullmix=fullmix fullmix value
                     * @property {string} image=image image value
                     * @property {string} video=video video value
                     * @property {string} voiceover=voiceover voiceover value
                     */
                    CreateUploadUrlRequest.Type = (function() {
                        var values = {};

                        values["bgmusic"] = "bgmusic";
                        values["fullmix"] = "fullmix";
                        values["image"] = "image";
                        values["video"] = "video";
                        values["voiceover"] = "voiceover";
                        return values;
                    })();
                    
                    Type = CreateUploadUrlRequest.Type;

                    /**
                     * Orientation enum.
                     * @name com.spotify.adstudiobff.proto.CreateUploadUrlRequest.Orientation
                     * @enum {string}

                     * @property {string} LANDSCAPE=LANDSCAPE LANDSCAPE value
                     * @property {string} PORTRAIT=PORTRAIT PORTRAIT value
                     */
                    CreateUploadUrlRequest.Orientation = (function() {
                        var values = {};

                        values["LANDSCAPE"] = "LANDSCAPE";
                        values["PORTRAIT"] = "PORTRAIT";
                        return values;
                    })();
                    
                    Orientation = CreateUploadUrlRequest.Orientation;
                    return CreateUploadUrlRequest;
                })();
                
                CreateUploadUrlRequest = proto.CreateUploadUrlRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateUploadUrlRequest"] = CreateUploadUrlRequest;
exports["Type"] = Type;
exports["Orientation"] = Orientation;
exports["default"] = $root;
