// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateFlightWithCreativesRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateFlightWithCreativesRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateFlightWithCreativesRequest = (function() {
                    /**
                     * Properties of a CreateFlightWithCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateFlightWithCreativesRequest
                     * @property {string|null} [adAccountId] CreateFlightWithCreativesRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.CreateFlightRequest|null} [flight] CreateFlightWithCreativesRequest flight
                     * @property {Array.<com.spotify.adstudiobff.proto.CreateCreativeRequest>|null} [creatives] CreateFlightWithCreativesRequest creatives
                     * @property {string|null} [setAdAccountId] CreateFlightWithCreativesRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.CreateFlightRequest|null} [setFlight] CreateFlightWithCreativesRequest setFlight
                     * @property {Array.<com.spotify.adstudiobff.proto.CreateCreativeRequest>|null} [setCreatives] CreateFlightWithCreativesRequest setCreatives
                     */
                    
                    /**
                     * Constructs a new CreateFlightWithCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateFlightWithCreativesRequest.
                     * @implements ICreateFlightWithCreativesRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateFlightWithCreativesRequest=} [properties] Properties to set
                     */
                    function CreateFlightWithCreativesRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateFlightWithCreativesRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @instance
                     */
                    CreateFlightWithCreativesRequest.prototype.adAccountId = "";

                    /**
                     * CreateFlightWithCreativesRequest flight.
                     * @member {com.spotify.adstudiobff.proto.CreateFlightRequest} flight
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @instance
                     */
                    CreateFlightWithCreativesRequest.prototype.flight = {};

                    /**
                     * CreateFlightWithCreativesRequest creatives.
                     * @member {Array.<com.spotify.adstudiobff.proto.CreateCreativeRequest>} creatives
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @instance
                     */
                    CreateFlightWithCreativesRequest.prototype.creatives = [];


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest} CreateFlightWithCreativesRequest
                     */
                    CreateFlightWithCreativesRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets flight
                     * @function setFlight
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.CreateFlightRequest} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest} CreateFlightWithCreativesRequest
                     */
                    CreateFlightWithCreativesRequest.prototype.setFlight = function(value) {
                        this['flight'] = value;
                        return this;
                    };

                    /**
                     * Sets creatives
                     * @function setCreatives
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @param {Array.<com.spotify.adstudiobff.proto.CreateCreativeRequest>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest} CreateFlightWithCreativesRequest
                     */
                    CreateFlightWithCreativesRequest.prototype.setCreatives = function(value) {
                        this['creatives'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateFlightWithCreativesRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightWithCreativesRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateFlightWithCreativesRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateFlightWithCreativesRequest;
                })();
                
                CreateFlightWithCreativesRequest = proto.CreateFlightWithCreativesRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateFlightWithCreativesRequest"] = CreateFlightWithCreativesRequest;
exports["default"] = $root;
