// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateFlightRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateFlightRequest = (function() {
                    /**
                     * Properties of a CreateFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateFlightRequest
                     * @property {string|null} [adAccountId] CreateFlightRequest adAccountId
                     * @property {string|null} [campaignId] CreateFlightRequest campaignId
                     * @property {string|null} [name] CreateFlightRequest name
                     * @property {google.protobuf.Timestamp|null} [dateBegin] CreateFlightRequest dateBegin
                     * @property {google.protobuf.Timestamp|null} [dateEnd] CreateFlightRequest dateEnd
                     * @property {com.spotify.adstudiobff.proto.BffTargeting|null} [targeting] CreateFlightRequest targeting
                     * @property {com.spotify.adstudiobff.proto.Money|null} [totalBudget] CreateFlightRequest totalBudget
                     * @property {string|null} [objective] CreateFlightRequest objective
                     * @property {string|null} [artistId] CreateFlightRequest artistId
                     * @property {string|null} [purchaseOrderNumber] CreateFlightRequest purchaseOrderNumber
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [frequencyCap] CreateFlightRequest frequencyCap
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [format] CreateFlightRequest format
                     * @property {number|null} [aspectRatio] CreateFlightRequest aspectRatio
                     * @property {string|null} [draftId] CreateFlightRequest draftId
                     * @property {string|null} [duplicatedFlightId] CreateFlightRequest duplicatedFlightId
                     * @property {string|null} [copiedFromExistingFlightId] CreateFlightRequest copiedFromExistingFlightId
                     * @property {string|null} [hierarchyDraftId] CreateFlightRequest hierarchyDraftId
                     * @property {com.spotify.adstudiobff.proto.DuplicationType|null} [duplicationType] CreateFlightRequest duplicationType
                     * @property {boolean|null} [serveOnMegaphone] CreateFlightRequest serveOnMegaphone
                     * @property {string|null} [setAdAccountId] CreateFlightRequest setAdAccountId
                     * @property {string|null} [setCampaignId] CreateFlightRequest setCampaignId
                     * @property {string|null} [setName] CreateFlightRequest setName
                     * @property {google.protobuf.Timestamp|null} [setDateBegin] CreateFlightRequest setDateBegin
                     * @property {google.protobuf.Timestamp|null} [setDateEnd] CreateFlightRequest setDateEnd
                     * @property {com.spotify.adstudiobff.proto.BffTargeting|null} [setTargeting] CreateFlightRequest setTargeting
                     * @property {com.spotify.adstudiobff.proto.Money|null} [setTotalBudget] CreateFlightRequest setTotalBudget
                     * @property {string|null} [setObjective] CreateFlightRequest setObjective
                     * @property {string|null} [setArtistId] CreateFlightRequest setArtistId
                     * @property {string|null} [setPurchaseOrderNumber] CreateFlightRequest setPurchaseOrderNumber
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [setFrequencyCap] CreateFlightRequest setFrequencyCap
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormat] CreateFlightRequest setFormat
                     * @property {number|null} [setAspectRatio] CreateFlightRequest setAspectRatio
                     * @property {string|null} [setDraftId] CreateFlightRequest setDraftId
                     * @property {string|null} [setDuplicatedFlightId] CreateFlightRequest setDuplicatedFlightId
                     * @property {string|null} [setCopiedFromExistingFlightId] CreateFlightRequest setCopiedFromExistingFlightId
                     * @property {string|null} [setHierarchyDraftId] CreateFlightRequest setHierarchyDraftId
                     * @property {com.spotify.adstudiobff.proto.DuplicationType|null} [setDuplicationType] CreateFlightRequest setDuplicationType
                     * @property {boolean|null} [setServeOnMegaphone] CreateFlightRequest setServeOnMegaphone
                     */
                    
                    /**
                     * Constructs a new CreateFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateFlightRequest.
                     * @implements ICreateFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateFlightRequest=} [properties] Properties to set
                     */
                    function CreateFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.adAccountId = "";

                    /**
                     * CreateFlightRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.campaignId = "";

                    /**
                     * CreateFlightRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.name = "";

                    /**
                     * CreateFlightRequest dateBegin.
                     * @member {google.protobuf.Timestamp} dateBegin
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.dateBegin = {};

                    /**
                     * CreateFlightRequest dateEnd.
                     * @member {google.protobuf.Timestamp} dateEnd
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.dateEnd = {};

                    /**
                     * CreateFlightRequest targeting.
                     * @member {com.spotify.adstudiobff.proto.BffTargeting} targeting
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.targeting = {};

                    /**
                     * CreateFlightRequest totalBudget.
                     * @member {com.spotify.adstudiobff.proto.Money} totalBudget
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.totalBudget = {};

                    /**
                     * CreateFlightRequest objective.
                     * @member {string} objective
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.objective = "";

                    /**
                     * CreateFlightRequest artistId.
                     * @member {string} artistId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.artistId = "";

                    /**
                     * CreateFlightRequest purchaseOrderNumber.
                     * @member {string} purchaseOrderNumber
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.purchaseOrderNumber = "";

                    /**
                     * CreateFlightRequest frequencyCap.
                     * @member {Array.<com.spotify.ads.campaignservice.FrequencyCap>} frequencyCap
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.frequencyCap = [];

                    /**
                     * CreateFlightRequest format.
                     * @member {com.spotify.ads.cs.proto.FormatType} format
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.format = {};

                    /**
                     * CreateFlightRequest aspectRatio.
                     * @member {number} aspectRatio
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.aspectRatio = 0;

                    /**
                     * CreateFlightRequest draftId.
                     * @member {string} draftId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.draftId = "";

                    /**
                     * CreateFlightRequest duplicatedFlightId.
                     * @member {string} duplicatedFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.duplicatedFlightId = "";

                    /**
                     * CreateFlightRequest copiedFromExistingFlightId.
                     * @member {string} copiedFromExistingFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.copiedFromExistingFlightId = "";

                    /**
                     * CreateFlightRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.hierarchyDraftId = "";

                    /**
                     * CreateFlightRequest duplicationType.
                     * @member {com.spotify.adstudiobff.proto.DuplicationType} duplicationType
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.duplicationType = {};

                    /**
                     * CreateFlightRequest serveOnMegaphone.
                     * @member {boolean} serveOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     */
                    CreateFlightRequest.prototype.serveOnMegaphone = false;


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets dateBegin
                     * @function setDateBegin
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setDateBegin = function(value) {
                        this['dateBegin'] = value;
                        return this;
                    };

                    /**
                     * Sets dateEnd
                     * @function setDateEnd
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setDateEnd = function(value) {
                        this['dateEnd'] = value;
                        return this;
                    };

                    /**
                     * Sets targeting
                     * @function setTargeting
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {com.spotify.adstudiobff.proto.BffTargeting} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setTargeting = function(value) {
                        this['targeting'] = value;
                        return this;
                    };

                    /**
                     * Sets totalBudget
                     * @function setTotalBudget
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {com.spotify.adstudiobff.proto.Money} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setTotalBudget = function(value) {
                        this['totalBudget'] = value;
                        return this;
                    };

                    /**
                     * Sets objective
                     * @function setObjective
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setObjective = function(value) {
                        this['objective'] = value;
                        return this;
                    };

                    /**
                     * Sets artistId
                     * @function setArtistId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setArtistId = function(value) {
                        this['artistId'] = value;
                        return this;
                    };

                    /**
                     * Sets purchaseOrderNumber
                     * @function setPurchaseOrderNumber
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setPurchaseOrderNumber = function(value) {
                        this['purchaseOrderNumber'] = value;
                        return this;
                    };

                    /**
                     * Sets frequencyCap
                     * @function setFrequencyCap
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {Array.<com.spotify.ads.campaignservice.FrequencyCap>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setFrequencyCap = function(value) {
                        this['frequencyCap'] = value;
                        return this;
                    };

                    /**
                     * Sets format
                     * @function setFormat
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setFormat = function(value) {
                        this['format'] = value;
                        return this;
                    };

                    /**
                     * Sets aspectRatio
                     * @function setAspectRatio
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setAspectRatio = function(value) {
                        this['aspectRatio'] = value;
                        return this;
                    };

                    /**
                     * Sets draftId
                     * @function setDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setDraftId = function(value) {
                        this['draftId'] = value;
                        return this;
                    };

                    /**
                     * Sets duplicatedFlightId
                     * @function setDuplicatedFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setDuplicatedFlightId = function(value) {
                        this['duplicatedFlightId'] = value;
                        return this;
                    };

                    /**
                     * Sets copiedFromExistingFlightId
                     * @function setCopiedFromExistingFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setCopiedFromExistingFlightId = function(value) {
                        this['copiedFromExistingFlightId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets duplicationType
                     * @function setDuplicationType
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {com.spotify.adstudiobff.proto.DuplicationType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setDuplicationType = function(value) {
                        this['duplicationType'] = value;
                        return this;
                    };

                    /**
                     * Sets serveOnMegaphone
                     * @function setServeOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateFlightRequest} CreateFlightRequest
                     */
                    CreateFlightRequest.prototype.setServeOnMegaphone = function(value) {
                        this['serveOnMegaphone'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateFlightRequest;
                })();
                
                CreateFlightRequest = proto.CreateFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateFlightRequest"] = CreateFlightRequest;
exports["default"] = $root;
