// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateCreativeRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateCreativeRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateCreativeRequest = (function() {
                    /**
                     * Properties of a CreateCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateCreativeRequest
                     * @property {string|null} [name] CreateCreativeRequest name
                     * @property {string|null} [banner] CreateCreativeRequest banner
                     * @property {string|null} [fullmixId] CreateCreativeRequest fullmixId
                     * @property {string|null} [clickthroughUrl] CreateCreativeRequest clickthroughUrl
                     * @property {string|null} [adAccountId] CreateCreativeRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.BffVoiceover|null} [voiceover] CreateCreativeRequest voiceover
                     * @property {string|null} [bgmusicId] CreateCreativeRequest bgmusicId
                     * @property {boolean|null} [playFullMusic] CreateCreativeRequest playFullMusic
                     * @property {string|null} [ctaText] CreateCreativeRequest ctaText
                     * @property {string|null} [objective] CreateCreativeRequest objective
                     * @property {string|null} [artistId] CreateCreativeRequest artistId
                     * @property {string|null} [flightId] CreateCreativeRequest flightId
                     * @property {string|null} [videoId] CreateCreativeRequest videoId
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [creativeRotationParameters] CreateCreativeRequest creativeRotationParameters
                     * @property {string|null} [brandName] CreateCreativeRequest brandName
                     * @property {string|null} [tagLine] CreateCreativeRequest tagLine
                     * @property {string|null} [duplicatedCreativeId] CreateCreativeRequest duplicatedCreativeId
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [format] CreateCreativeRequest format
                     * @property {string|null} [copiedFromExistingCreativeId] CreateCreativeRequest copiedFromExistingCreativeId
                     * @property {boolean|null} [moatEnabled] CreateCreativeRequest moatEnabled
                     * @property {string|null} [iasPixel] CreateCreativeRequest iasPixel
                     * @property {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>|null} [trackingPixel] CreateCreativeRequest trackingPixel
                     * @property {string|null} [hierarchyDraftId] CreateCreativeRequest hierarchyDraftId
                     * @property {com.spotify.adstudiobff.proto.DuplicationType|null} [duplicationType] CreateCreativeRequest duplicationType
                     * @property {boolean|null} [serveOnMegaphone] CreateCreativeRequest serveOnMegaphone
                     * @property {string|null} [setName] CreateCreativeRequest setName
                     * @property {string|null} [setBanner] CreateCreativeRequest setBanner
                     * @property {string|null} [setFullmixId] CreateCreativeRequest setFullmixId
                     * @property {string|null} [setClickthroughUrl] CreateCreativeRequest setClickthroughUrl
                     * @property {string|null} [setAdAccountId] CreateCreativeRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.BffVoiceover|null} [setVoiceover] CreateCreativeRequest setVoiceover
                     * @property {string|null} [setBgmusicId] CreateCreativeRequest setBgmusicId
                     * @property {boolean|null} [setPlayFullMusic] CreateCreativeRequest setPlayFullMusic
                     * @property {string|null} [setCtaText] CreateCreativeRequest setCtaText
                     * @property {string|null} [setObjective] CreateCreativeRequest setObjective
                     * @property {string|null} [setArtistId] CreateCreativeRequest setArtistId
                     * @property {string|null} [setFlightId] CreateCreativeRequest setFlightId
                     * @property {string|null} [setVideoId] CreateCreativeRequest setVideoId
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [setCreativeRotationParameters] CreateCreativeRequest setCreativeRotationParameters
                     * @property {string|null} [setBrandName] CreateCreativeRequest setBrandName
                     * @property {string|null} [setTagLine] CreateCreativeRequest setTagLine
                     * @property {string|null} [setDuplicatedCreativeId] CreateCreativeRequest setDuplicatedCreativeId
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormat] CreateCreativeRequest setFormat
                     * @property {string|null} [setCopiedFromExistingCreativeId] CreateCreativeRequest setCopiedFromExistingCreativeId
                     * @property {boolean|null} [setMoatEnabled] CreateCreativeRequest setMoatEnabled
                     * @property {string|null} [setIasPixel] CreateCreativeRequest setIasPixel
                     * @property {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>|null} [setTrackingPixel] CreateCreativeRequest setTrackingPixel
                     * @property {string|null} [setHierarchyDraftId] CreateCreativeRequest setHierarchyDraftId
                     * @property {com.spotify.adstudiobff.proto.DuplicationType|null} [setDuplicationType] CreateCreativeRequest setDuplicationType
                     * @property {boolean|null} [setServeOnMegaphone] CreateCreativeRequest setServeOnMegaphone
                     */
                    
                    /**
                     * Constructs a new CreateCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateCreativeRequest.
                     * @implements ICreateCreativeRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateCreativeRequest=} [properties] Properties to set
                     */
                    function CreateCreativeRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateCreativeRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.name = "";

                    /**
                     * CreateCreativeRequest banner.
                     * @member {string} banner
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.banner = "";

                    /**
                     * CreateCreativeRequest fullmixId.
                     * @member {string} fullmixId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.fullmixId = "";

                    /**
                     * CreateCreativeRequest clickthroughUrl.
                     * @member {string} clickthroughUrl
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.clickthroughUrl = "";

                    /**
                     * CreateCreativeRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.adAccountId = "";

                    /**
                     * CreateCreativeRequest voiceover.
                     * @member {com.spotify.adstudiobff.proto.BffVoiceover} voiceover
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.voiceover = {};

                    /**
                     * CreateCreativeRequest bgmusicId.
                     * @member {string} bgmusicId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.bgmusicId = "";

                    /**
                     * CreateCreativeRequest playFullMusic.
                     * @member {boolean} playFullMusic
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.playFullMusic = false;

                    /**
                     * CreateCreativeRequest ctaText.
                     * @member {string} ctaText
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.ctaText = "";

                    /**
                     * CreateCreativeRequest objective.
                     * @member {string} objective
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.objective = "";

                    /**
                     * CreateCreativeRequest artistId.
                     * @member {string} artistId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.artistId = "";

                    /**
                     * CreateCreativeRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.flightId = "";

                    /**
                     * CreateCreativeRequest videoId.
                     * @member {string} videoId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.videoId = "";

                    /**
                     * CreateCreativeRequest creativeRotationParameters.
                     * @member {com.spotify.adstudiobff.proto.CreativeRotationParameters} creativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.creativeRotationParameters = {};

                    /**
                     * CreateCreativeRequest brandName.
                     * @member {string} brandName
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.brandName = "";

                    /**
                     * CreateCreativeRequest tagLine.
                     * @member {string} tagLine
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.tagLine = "";

                    /**
                     * CreateCreativeRequest duplicatedCreativeId.
                     * @member {string} duplicatedCreativeId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.duplicatedCreativeId = "";

                    /**
                     * CreateCreativeRequest format.
                     * @member {com.spotify.ads.cs.proto.FormatType} format
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.format = {};

                    /**
                     * CreateCreativeRequest copiedFromExistingCreativeId.
                     * @member {string} copiedFromExistingCreativeId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.copiedFromExistingCreativeId = "";

                    /**
                     * CreateCreativeRequest moatEnabled.
                     * @member {boolean} moatEnabled
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.moatEnabled = false;

                    /**
                     * CreateCreativeRequest iasPixel.
                     * @member {string} iasPixel
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.iasPixel = "";

                    /**
                     * CreateCreativeRequest trackingPixel.
                     * @member {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>} trackingPixel
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.trackingPixel = [];

                    /**
                     * CreateCreativeRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.hierarchyDraftId = "";

                    /**
                     * CreateCreativeRequest duplicationType.
                     * @member {com.spotify.adstudiobff.proto.DuplicationType} duplicationType
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.duplicationType = {};

                    /**
                     * CreateCreativeRequest serveOnMegaphone.
                     * @member {boolean} serveOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     */
                    CreateCreativeRequest.prototype.serveOnMegaphone = false;


                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets banner
                     * @function setBanner
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setBanner = function(value) {
                        this['banner'] = value;
                        return this;
                    };

                    /**
                     * Sets fullmixId
                     * @function setFullmixId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setFullmixId = function(value) {
                        this['fullmixId'] = value;
                        return this;
                    };

                    /**
                     * Sets clickthroughUrl
                     * @function setClickthroughUrl
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setClickthroughUrl = function(value) {
                        this['clickthroughUrl'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets voiceover
                     * @function setVoiceover
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {com.spotify.adstudiobff.proto.BffVoiceover} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setVoiceover = function(value) {
                        this['voiceover'] = value;
                        return this;
                    };

                    /**
                     * Sets bgmusicId
                     * @function setBgmusicId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setBgmusicId = function(value) {
                        this['bgmusicId'] = value;
                        return this;
                    };

                    /**
                     * Sets playFullMusic
                     * @function setPlayFullMusic
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setPlayFullMusic = function(value) {
                        this['playFullMusic'] = value;
                        return this;
                    };

                    /**
                     * Sets ctaText
                     * @function setCtaText
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setCtaText = function(value) {
                        this['ctaText'] = value;
                        return this;
                    };

                    /**
                     * Sets objective
                     * @function setObjective
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setObjective = function(value) {
                        this['objective'] = value;
                        return this;
                    };

                    /**
                     * Sets artistId
                     * @function setArtistId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setArtistId = function(value) {
                        this['artistId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets videoId
                     * @function setVideoId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setVideoId = function(value) {
                        this['videoId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeRotationParameters
                     * @function setCreativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {com.spotify.adstudiobff.proto.CreativeRotationParameters} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setCreativeRotationParameters = function(value) {
                        this['creativeRotationParameters'] = value;
                        return this;
                    };

                    /**
                     * Sets brandName
                     * @function setBrandName
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setBrandName = function(value) {
                        this['brandName'] = value;
                        return this;
                    };

                    /**
                     * Sets tagLine
                     * @function setTagLine
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setTagLine = function(value) {
                        this['tagLine'] = value;
                        return this;
                    };

                    /**
                     * Sets duplicatedCreativeId
                     * @function setDuplicatedCreativeId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setDuplicatedCreativeId = function(value) {
                        this['duplicatedCreativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets format
                     * @function setFormat
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setFormat = function(value) {
                        this['format'] = value;
                        return this;
                    };

                    /**
                     * Sets copiedFromExistingCreativeId
                     * @function setCopiedFromExistingCreativeId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setCopiedFromExistingCreativeId = function(value) {
                        this['copiedFromExistingCreativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets moatEnabled
                     * @function setMoatEnabled
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setMoatEnabled = function(value) {
                        this['moatEnabled'] = value;
                        return this;
                    };

                    /**
                     * Sets iasPixel
                     * @function setIasPixel
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setIasPixel = function(value) {
                        this['iasPixel'] = value;
                        return this;
                    };

                    /**
                     * Sets trackingPixel
                     * @function setTrackingPixel
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setTrackingPixel = function(value) {
                        this['trackingPixel'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets duplicationType
                     * @function setDuplicationType
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {com.spotify.adstudiobff.proto.DuplicationType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setDuplicationType = function(value) {
                        this['duplicationType'] = value;
                        return this;
                    };

                    /**
                     * Sets serveOnMegaphone
                     * @function setServeOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCreativeRequest} CreateCreativeRequest
                     */
                    CreateCreativeRequest.prototype.setServeOnMegaphone = function(value) {
                        this['serveOnMegaphone'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateCreativeRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateCreativeRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateCreativeRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateCreativeRequest;
                })();
                
                CreateCreativeRequest = proto.CreateCreativeRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateCreativeRequest"] = CreateCreativeRequest;
exports["default"] = $root;
