// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateCampaignRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateCampaignRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateCampaignRequest = (function() {
                    /**
                     * Properties of a CreateCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateCampaignRequest
                     * @property {string|null} [adAccountId] CreateCampaignRequest adAccountId
                     * @property {string|null} [name] CreateCampaignRequest name
                     * @property {string|null} [objective] CreateCampaignRequest objective
                     * @property {string|null} [purchaseOrderNumber] CreateCampaignRequest purchaseOrderNumber
                     * @property {string|null} [artistId] CreateCampaignRequest artistId
                     * @property {string|null} [hierarchyDraftId] CreateCampaignRequest hierarchyDraftId
                     * @property {string|null} [setAdAccountId] CreateCampaignRequest setAdAccountId
                     * @property {string|null} [setName] CreateCampaignRequest setName
                     * @property {string|null} [setObjective] CreateCampaignRequest setObjective
                     * @property {string|null} [setPurchaseOrderNumber] CreateCampaignRequest setPurchaseOrderNumber
                     * @property {string|null} [setArtistId] CreateCampaignRequest setArtistId
                     * @property {string|null} [setHierarchyDraftId] CreateCampaignRequest setHierarchyDraftId
                     */
                    
                    /**
                     * Constructs a new CreateCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateCampaignRequest.
                     * @implements ICreateCampaignRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateCampaignRequest=} [properties] Properties to set
                     */
                    function CreateCampaignRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateCampaignRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.adAccountId = "";

                    /**
                     * CreateCampaignRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.name = "";

                    /**
                     * CreateCampaignRequest objective.
                     * @member {string} objective
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.objective = "";

                    /**
                     * CreateCampaignRequest purchaseOrderNumber.
                     * @member {string} purchaseOrderNumber
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.purchaseOrderNumber = "";

                    /**
                     * CreateCampaignRequest artistId.
                     * @member {string} artistId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.artistId = "";

                    /**
                     * CreateCampaignRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     */
                    CreateCampaignRequest.prototype.hierarchyDraftId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets objective
                     * @function setObjective
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setObjective = function(value) {
                        this['objective'] = value;
                        return this;
                    };

                    /**
                     * Sets purchaseOrderNumber
                     * @function setPurchaseOrderNumber
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setPurchaseOrderNumber = function(value) {
                        this['purchaseOrderNumber'] = value;
                        return this;
                    };

                    /**
                     * Sets artistId
                     * @function setArtistId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setArtistId = function(value) {
                        this['artistId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateCampaignRequest} CreateCampaignRequest
                     */
                    CreateCampaignRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateCampaignRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateCampaignRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateCampaignRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateCampaignRequest;
                })();
                
                CreateCampaignRequest = proto.CreateCampaignRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateCampaignRequest"] = CreateCampaignRequest;
exports["default"] = $root;
