// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateDraftForExistingCampaignRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateDraftForExistingCampaignRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateDraftForExistingCampaignRequest = (function() {
                    /**
                     * Properties of a CreateDraftForExistingCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateDraftForExistingCampaignRequest
                     * @property {string|null} [adAccountId] CreateDraftForExistingCampaignRequest adAccountId
                     * @property {string|null} [campaignId] CreateDraftForExistingCampaignRequest campaignId
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [flightDraftFormValues] CreateDraftForExistingCampaignRequest flightDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [creativeDraftFormValues] CreateDraftForExistingCampaignRequest creativeDraftFormValues
                     * @property {string|null} [setAdAccountId] CreateDraftForExistingCampaignRequest setAdAccountId
                     * @property {string|null} [setCampaignId] CreateDraftForExistingCampaignRequest setCampaignId
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [setFlightDraftFormValues] CreateDraftForExistingCampaignRequest setFlightDraftFormValues
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [setCreativeDraftFormValues] CreateDraftForExistingCampaignRequest setCreativeDraftFormValues
                     */
                    
                    /**
                     * Constructs a new CreateDraftForExistingCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateDraftForExistingCampaignRequest.
                     * @implements ICreateDraftForExistingCampaignRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateDraftForExistingCampaignRequest=} [properties] Properties to set
                     */
                    function CreateDraftForExistingCampaignRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateDraftForExistingCampaignRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @instance
                     */
                    CreateDraftForExistingCampaignRequest.prototype.adAccountId = "";

                    /**
                     * CreateDraftForExistingCampaignRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @instance
                     */
                    CreateDraftForExistingCampaignRequest.prototype.campaignId = "";

                    /**
                     * CreateDraftForExistingCampaignRequest flightDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} flightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @instance
                     */
                    CreateDraftForExistingCampaignRequest.prototype.flightDraftFormValues = {};

                    /**
                     * CreateDraftForExistingCampaignRequest creativeDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} creativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @instance
                     */
                    CreateDraftForExistingCampaignRequest.prototype.creativeDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest} CreateDraftForExistingCampaignRequest
                     */
                    CreateDraftForExistingCampaignRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest} CreateDraftForExistingCampaignRequest
                     */
                    CreateDraftForExistingCampaignRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightDraftFormValues
                     * @function setFlightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @param {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest} CreateDraftForExistingCampaignRequest
                     */
                    CreateDraftForExistingCampaignRequest.prototype.setFlightDraftFormValues = function(value) {
                        this['flightDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeDraftFormValues
                     * @function setCreativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @param {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest} CreateDraftForExistingCampaignRequest
                     */
                    CreateDraftForExistingCampaignRequest.prototype.setCreativeDraftFormValues = function(value) {
                        this['creativeDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateDraftForExistingCampaignRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingCampaignRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateDraftForExistingCampaignRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateDraftForExistingCampaignRequest;
                })();
                
                CreateDraftForExistingCampaignRequest = proto.CreateDraftForExistingCampaignRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateDraftForExistingCampaignRequest"] = CreateDraftForExistingCampaignRequest;
exports["default"] = $root;
