// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateAudienceRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateAudienceRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateAudienceRequest = (function() {
                    /**
                     * Properties of a CreateAudienceRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateAudienceRequest
                     * @property {string|null} [adAccountId] CreateAudienceRequest adAccountId
                     * @property {string|null} [name] CreateAudienceRequest name
                     * @property {string|null} [description] CreateAudienceRequest description
                     * @property {string|null} [setAdAccountId] CreateAudienceRequest setAdAccountId
                     * @property {string|null} [setName] CreateAudienceRequest setName
                     * @property {string|null} [setDescription] CreateAudienceRequest setDescription
                     */
                    
                    /**
                     * Constructs a new CreateAudienceRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateAudienceRequest.
                     * @implements ICreateAudienceRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateAudienceRequest=} [properties] Properties to set
                     */
                    function CreateAudienceRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateAudienceRequest ad_account_id.
                     * @member {string} ad_account_id
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @instance
                     */
                    CreateAudienceRequest.prototype.adAccountId = "";

                    /**
                     * CreateAudienceRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @instance
                     */
                    CreateAudienceRequest.prototype.name = "";

                    /**
                     * CreateAudienceRequest description.
                     * @member {string} description
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @instance
                     */
                    CreateAudienceRequest.prototype.description = "";


                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateAudienceRequest} CreateAudienceRequest
                     */
                    CreateAudienceRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateAudienceRequest} CreateAudienceRequest
                     */
                    CreateAudienceRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets description
                     * @function setDescription
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateAudienceRequest} CreateAudienceRequest
                     */
                    CreateAudienceRequest.prototype.setDescription = function(value) {
                        this['description'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateAudienceRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateAudienceRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateAudienceRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateAudienceRequest;
                })();
                
                CreateAudienceRequest = proto.CreateAudienceRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateAudienceRequest"] = CreateAudienceRequest;
exports["default"] = $root;
