// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/CreateDraftForExistingFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var CreateDraftForExistingFlightRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.CreateDraftForExistingFlightRequest = (function() {
                    /**
                     * Properties of a CreateDraftForExistingFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ICreateDraftForExistingFlightRequest
                     * @property {string|null} [adAccountId] CreateDraftForExistingFlightRequest adAccountId
                     * @property {string|null} [campaignId] CreateDraftForExistingFlightRequest campaignId
                     * @property {string|null} [flightId] CreateDraftForExistingFlightRequest flightId
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [creativeDraftFormValues] CreateDraftForExistingFlightRequest creativeDraftFormValues
                     * @property {string|null} [setAdAccountId] CreateDraftForExistingFlightRequest setAdAccountId
                     * @property {string|null} [setCampaignId] CreateDraftForExistingFlightRequest setCampaignId
                     * @property {string|null} [setFlightId] CreateDraftForExistingFlightRequest setFlightId
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [setCreativeDraftFormValues] CreateDraftForExistingFlightRequest setCreativeDraftFormValues
                     */
                    
                    /**
                     * Constructs a new CreateDraftForExistingFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a CreateDraftForExistingFlightRequest.
                     * @implements ICreateDraftForExistingFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ICreateDraftForExistingFlightRequest=} [properties] Properties to set
                     */
                    function CreateDraftForExistingFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * CreateDraftForExistingFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @instance
                     */
                    CreateDraftForExistingFlightRequest.prototype.adAccountId = "";

                    /**
                     * CreateDraftForExistingFlightRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @instance
                     */
                    CreateDraftForExistingFlightRequest.prototype.campaignId = "";

                    /**
                     * CreateDraftForExistingFlightRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @instance
                     */
                    CreateDraftForExistingFlightRequest.prototype.flightId = "";

                    /**
                     * CreateDraftForExistingFlightRequest creativeDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} creativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @instance
                     */
                    CreateDraftForExistingFlightRequest.prototype.creativeDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest} CreateDraftForExistingFlightRequest
                     */
                    CreateDraftForExistingFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest} CreateDraftForExistingFlightRequest
                     */
                    CreateDraftForExistingFlightRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest} CreateDraftForExistingFlightRequest
                     */
                    CreateDraftForExistingFlightRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeDraftFormValues
                     * @function setCreativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @param {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest} CreateDraftForExistingFlightRequest
                     */
                    CreateDraftForExistingFlightRequest.prototype.setCreativeDraftFormValues = function(value) {
                        this['creativeDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the CreateDraftForExistingFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.CreateDraftForExistingFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    CreateDraftForExistingFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return CreateDraftForExistingFlightRequest;
                })();
                
                CreateDraftForExistingFlightRequest = proto.CreateDraftForExistingFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["CreateDraftForExistingFlightRequest"] = CreateDraftForExistingFlightRequest;
exports["default"] = $root;
