// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/DateRange.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var DateRange = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.DateRange = (function() {
                    /**
                     * Properties of a DateRange.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IDateRange
                     * @property {google.protobuf.Timestamp|null} [dateBegin] DateRange dateBegin
                     * @property {google.protobuf.Timestamp|null} [dateEnd] DateRange dateEnd
                     * @property {google.protobuf.Timestamp|null} [setDateBegin] DateRange setDateBegin
                     * @property {google.protobuf.Timestamp|null} [setDateEnd] DateRange setDateEnd
                     */
                    
                    /**
                     * Constructs a new DateRange.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a DateRange.
                     * @implements IDateRange
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IDateRange=} [properties] Properties to set
                     */
                    function DateRange(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * DateRange dateBegin.
                     * @member {google.protobuf.Timestamp} dateBegin
                     * @memberof com.spotify.adstudiobff.proto.DateRange
                     * @instance
                     */
                    DateRange.prototype.dateBegin = {};

                    /**
                     * DateRange dateEnd.
                     * @member {google.protobuf.Timestamp} dateEnd
                     * @memberof com.spotify.adstudiobff.proto.DateRange
                     * @instance
                     */
                    DateRange.prototype.dateEnd = {};


                    /**
                     * Sets dateBegin
                     * @function setDateBegin
                     * @memberof com.spotify.adstudiobff.proto.DateRange
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.DateRange} DateRange
                     */
                    DateRange.prototype.setDateBegin = function(value) {
                        this['dateBegin'] = value;
                        return this;
                    };

                    /**
                     * Sets dateEnd
                     * @function setDateEnd
                     * @memberof com.spotify.adstudiobff.proto.DateRange
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.DateRange} DateRange
                     */
                    DateRange.prototype.setDateEnd = function(value) {
                        this['dateEnd'] = value;
                        return this;
                    };

                    /**
                     * Converts the DateRange to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.DateRange
                     * @instance
                     * @returns {object} object
                     */
                    DateRange.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return DateRange;
                })();
                
                DateRange = proto.DateRange;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["DateRange"] = DateRange;
exports["default"] = $root;
