// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/DismissHierarchyDraftReviewRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var DismissHierarchyDraftReviewRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.DismissHierarchyDraftReviewRequest = (function() {
                    /**
                     * Properties of a DismissHierarchyDraftReviewRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IDismissHierarchyDraftReviewRequest
                     * @property {string|null} [adAccountId] DismissHierarchyDraftReviewRequest adAccountId
                     * @property {string|null} [hierarchyDraftId] DismissHierarchyDraftReviewRequest hierarchyDraftId
                     * @property {string|null} [setAdAccountId] DismissHierarchyDraftReviewRequest setAdAccountId
                     * @property {string|null} [setHierarchyDraftId] DismissHierarchyDraftReviewRequest setHierarchyDraftId
                     */
                    
                    /**
                     * Constructs a new DismissHierarchyDraftReviewRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a DismissHierarchyDraftReviewRequest.
                     * @implements IDismissHierarchyDraftReviewRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IDismissHierarchyDraftReviewRequest=} [properties] Properties to set
                     */
                    function DismissHierarchyDraftReviewRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * DismissHierarchyDraftReviewRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest
                     * @instance
                     */
                    DismissHierarchyDraftReviewRequest.prototype.adAccountId = "";

                    /**
                     * DismissHierarchyDraftReviewRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest
                     * @instance
                     */
                    DismissHierarchyDraftReviewRequest.prototype.hierarchyDraftId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest} DismissHierarchyDraftReviewRequest
                     */
                    DismissHierarchyDraftReviewRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest} DismissHierarchyDraftReviewRequest
                     */
                    DismissHierarchyDraftReviewRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Converts the DismissHierarchyDraftReviewRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.DismissHierarchyDraftReviewRequest
                     * @instance
                     * @returns {object} object
                     */
                    DismissHierarchyDraftReviewRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return DismissHierarchyDraftReviewRequest;
                })();
                
                DismissHierarchyDraftReviewRequest = proto.DismissHierarchyDraftReviewRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["DismissHierarchyDraftReviewRequest"] = DismissHierarchyDraftReviewRequest;
exports["default"] = $root;
