// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditFlightRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditFlightRequest = (function() {
                    /**
                     * Properties of a EditFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditFlightRequest
                     * @property {string|null} [flightId] EditFlightRequest flightId
                     * @property {string|null} [name] EditFlightRequest name
                     * @property {string|null} [adAccountId] EditFlightRequest adAccountId
                     * @property {google.protobuf.Timestamp|null} [dateBegin] EditFlightRequest dateBegin
                     * @property {google.protobuf.Timestamp|null} [dateEnd] EditFlightRequest dateEnd
                     * @property {com.spotify.adstudiobff.proto.BffTargeting|null} [targeting] EditFlightRequest targeting
                     * @property {com.spotify.adstudiobff.proto.Money|null} [totalBudget] EditFlightRequest totalBudget
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [frequencyCap] EditFlightRequest frequencyCap
                     * @property {string|null} [setFlightId] EditFlightRequest setFlightId
                     * @property {string|null} [setName] EditFlightRequest setName
                     * @property {string|null} [setAdAccountId] EditFlightRequest setAdAccountId
                     * @property {google.protobuf.Timestamp|null} [setDateBegin] EditFlightRequest setDateBegin
                     * @property {google.protobuf.Timestamp|null} [setDateEnd] EditFlightRequest setDateEnd
                     * @property {com.spotify.adstudiobff.proto.BffTargeting|null} [setTargeting] EditFlightRequest setTargeting
                     * @property {com.spotify.adstudiobff.proto.Money|null} [setTotalBudget] EditFlightRequest setTotalBudget
                     * @property {Array.<com.spotify.ads.campaignservice.FrequencyCap>|null} [setFrequencyCap] EditFlightRequest setFrequencyCap
                     */
                    
                    /**
                     * Constructs a new EditFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditFlightRequest.
                     * @implements IEditFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditFlightRequest=} [properties] Properties to set
                     */
                    function EditFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditFlightRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.flightId = "";

                    /**
                     * EditFlightRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.name = "";

                    /**
                     * EditFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.adAccountId = "";

                    /**
                     * EditFlightRequest dateBegin.
                     * @member {google.protobuf.Timestamp} dateBegin
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.dateBegin = {};

                    /**
                     * EditFlightRequest dateEnd.
                     * @member {google.protobuf.Timestamp} dateEnd
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.dateEnd = {};

                    /**
                     * EditFlightRequest targeting.
                     * @member {com.spotify.adstudiobff.proto.BffTargeting} targeting
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.targeting = {};

                    /**
                     * EditFlightRequest totalBudget.
                     * @member {com.spotify.adstudiobff.proto.Money} totalBudget
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.totalBudget = {};

                    /**
                     * EditFlightRequest frequencyCap.
                     * @member {Array.<com.spotify.ads.campaignservice.FrequencyCap>} frequencyCap
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     */
                    EditFlightRequest.prototype.frequencyCap = [];


                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets dateBegin
                     * @function setDateBegin
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setDateBegin = function(value) {
                        this['dateBegin'] = value;
                        return this;
                    };

                    /**
                     * Sets dateEnd
                     * @function setDateEnd
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {google.protobuf.Timestamp} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setDateEnd = function(value) {
                        this['dateEnd'] = value;
                        return this;
                    };

                    /**
                     * Sets targeting
                     * @function setTargeting
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {com.spotify.adstudiobff.proto.BffTargeting} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setTargeting = function(value) {
                        this['targeting'] = value;
                        return this;
                    };

                    /**
                     * Sets totalBudget
                     * @function setTotalBudget
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {com.spotify.adstudiobff.proto.Money} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setTotalBudget = function(value) {
                        this['totalBudget'] = value;
                        return this;
                    };

                    /**
                     * Sets frequencyCap
                     * @function setFrequencyCap
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @param {Array.<com.spotify.ads.campaignservice.FrequencyCap>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightRequest} EditFlightRequest
                     */
                    EditFlightRequest.prototype.setFrequencyCap = function(value) {
                        this['frequencyCap'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    EditFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditFlightRequest;
                })();
                
                EditFlightRequest = proto.EditFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditFlightRequest"] = EditFlightRequest;
exports["default"] = $root;
