// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditCampaignRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditCampaignRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditCampaignRequest = (function() {
                    /**
                     * Properties of a EditCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditCampaignRequest
                     * @property {string|null} [adAccountId] EditCampaignRequest adAccountId
                     * @property {string|null} [campaignId] EditCampaignRequest campaignId
                     * @property {string|null} [name] EditCampaignRequest name
                     * @property {string|null} [setAdAccountId] EditCampaignRequest setAdAccountId
                     * @property {string|null} [setCampaignId] EditCampaignRequest setCampaignId
                     * @property {string|null} [setName] EditCampaignRequest setName
                     */
                    
                    /**
                     * Constructs a new EditCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditCampaignRequest.
                     * @implements IEditCampaignRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditCampaignRequest=} [properties] Properties to set
                     */
                    function EditCampaignRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditCampaignRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @instance
                     */
                    EditCampaignRequest.prototype.adAccountId = "";

                    /**
                     * EditCampaignRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @instance
                     */
                    EditCampaignRequest.prototype.campaignId = "";

                    /**
                     * EditCampaignRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @instance
                     */
                    EditCampaignRequest.prototype.name = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCampaignRequest} EditCampaignRequest
                     */
                    EditCampaignRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCampaignRequest} EditCampaignRequest
                     */
                    EditCampaignRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCampaignRequest} EditCampaignRequest
                     */
                    EditCampaignRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditCampaignRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditCampaignRequest
                     * @instance
                     * @returns {object} object
                     */
                    EditCampaignRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditCampaignRequest;
                })();
                
                EditCampaignRequest = proto.EditCampaignRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditCampaignRequest"] = EditCampaignRequest;
exports["default"] = $root;
