// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditUserRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditUserRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditUserRequest = (function() {
                    /**
                     * Properties of a EditUserRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditUserRequest
                     * @property {string|null} [firstName] EditUserRequest firstName
                     * @property {string|null} [lastName] EditUserRequest lastName
                     * @property {string|null} [businessEmail] EditUserRequest businessEmail
                     * @property {boolean|null} [emailSubscribe] EditUserRequest emailSubscribe
                     * @property {string|null} [setFirstName] EditUserRequest setFirstName
                     * @property {string|null} [setLastName] EditUserRequest setLastName
                     * @property {string|null} [setBusinessEmail] EditUserRequest setBusinessEmail
                     * @property {boolean|null} [setEmailSubscribe] EditUserRequest setEmailSubscribe
                     */
                    
                    /**
                     * Constructs a new EditUserRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditUserRequest.
                     * @implements IEditUserRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditUserRequest=} [properties] Properties to set
                     */
                    function EditUserRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditUserRequest firstName.
                     * @member {string} firstName
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @instance
                     */
                    EditUserRequest.prototype.firstName = "";

                    /**
                     * EditUserRequest lastName.
                     * @member {string} lastName
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @instance
                     */
                    EditUserRequest.prototype.lastName = "";

                    /**
                     * EditUserRequest businessEmail.
                     * @member {string} businessEmail
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @instance
                     */
                    EditUserRequest.prototype.businessEmail = "";

                    /**
                     * EditUserRequest emailSubscribe.
                     * @member {boolean} emailSubscribe
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @instance
                     */
                    EditUserRequest.prototype.emailSubscribe = false;


                    /**
                     * Sets firstName
                     * @function setFirstName
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditUserRequest} EditUserRequest
                     */
                    EditUserRequest.prototype.setFirstName = function(value) {
                        this['firstName'] = value;
                        return this;
                    };

                    /**
                     * Sets lastName
                     * @function setLastName
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditUserRequest} EditUserRequest
                     */
                    EditUserRequest.prototype.setLastName = function(value) {
                        this['lastName'] = value;
                        return this;
                    };

                    /**
                     * Sets businessEmail
                     * @function setBusinessEmail
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditUserRequest} EditUserRequest
                     */
                    EditUserRequest.prototype.setBusinessEmail = function(value) {
                        this['businessEmail'] = value;
                        return this;
                    };

                    /**
                     * Sets emailSubscribe
                     * @function setEmailSubscribe
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditUserRequest} EditUserRequest
                     */
                    EditUserRequest.prototype.setEmailSubscribe = function(value) {
                        this['emailSubscribe'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditUserRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditUserRequest
                     * @instance
                     * @returns {object} object
                     */
                    EditUserRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditUserRequest;
                })();
                
                EditUserRequest = proto.EditUserRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditUserRequest"] = EditUserRequest;
exports["default"] = $root;
