// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditFlightLinksRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditFlightLinksRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditFlightLinksRequest = (function() {
                    /**
                     * Properties of a EditFlightLinksRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditFlightLinksRequest
                     * @property {string|null} [adAccountId] EditFlightLinksRequest adAccountId
                     * @property {Array.<com.spotify.adstudiobff.proto.EditFlightLinkParameters>|null} [editFlightLinkParameters] EditFlightLinksRequest editFlightLinkParameters
                     * @property {string|null} [setAdAccountId] EditFlightLinksRequest setAdAccountId
                     * @property {Array.<com.spotify.adstudiobff.proto.EditFlightLinkParameters>|null} [setEditFlightLinkParameters] EditFlightLinksRequest setEditFlightLinkParameters
                     */
                    
                    /**
                     * Constructs a new EditFlightLinksRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditFlightLinksRequest.
                     * @implements IEditFlightLinksRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditFlightLinksRequest=} [properties] Properties to set
                     */
                    function EditFlightLinksRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditFlightLinksRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinksRequest
                     * @instance
                     */
                    EditFlightLinksRequest.prototype.adAccountId = "";

                    /**
                     * EditFlightLinksRequest editFlightLinkParameters.
                     * @member {Array.<com.spotify.adstudiobff.proto.EditFlightLinkParameters>} editFlightLinkParameters
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinksRequest
                     * @instance
                     */
                    EditFlightLinksRequest.prototype.editFlightLinkParameters = [];


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinksRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightLinksRequest} EditFlightLinksRequest
                     */
                    EditFlightLinksRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets editFlightLinkParameters
                     * @function setEditFlightLinkParameters
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinksRequest
                     * @param {Array.<com.spotify.adstudiobff.proto.EditFlightLinkParameters>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightLinksRequest} EditFlightLinksRequest
                     */
                    EditFlightLinksRequest.prototype.setEditFlightLinkParameters = function(value) {
                        this['editFlightLinkParameters'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditFlightLinksRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinksRequest
                     * @instance
                     * @returns {object} object
                     */
                    EditFlightLinksRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditFlightLinksRequest;
                })();
                
                EditFlightLinksRequest = proto.EditFlightLinksRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditFlightLinksRequest"] = EditFlightLinksRequest;
exports["default"] = $root;
