// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditCreativeRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditCreativeRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditCreativeRequest = (function() {
                    /**
                     * Properties of a EditCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditCreativeRequest
                     * @property {string|null} [creativeId] EditCreativeRequest creativeId
                     * @property {string|null} [name] EditCreativeRequest name
                     * @property {string|null} [adAccountId] EditCreativeRequest adAccountId
                     * @property {string|null} [banner] EditCreativeRequest banner
                     * @property {string|null} [fullmixId] EditCreativeRequest fullmixId
                     * @property {string|null} [videoId] EditCreativeRequest videoId
                     * @property {string|null} [clickthroughUrl] EditCreativeRequest clickthroughUrl
                     * @property {string|null} [ctaText] EditCreativeRequest ctaText
                     * @property {string|null} [brandName] EditCreativeRequest brandName
                     * @property {string|null} [tagLine] EditCreativeRequest tagLine
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [format] EditCreativeRequest format
                     * @property {boolean|null} [moatEnabled] EditCreativeRequest moatEnabled
                     * @property {string|null} [iasPixel] EditCreativeRequest iasPixel
                     * @property {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>|null} [trackingPixel] EditCreativeRequest trackingPixel
                     * @property {string|null} [setCreativeId] EditCreativeRequest setCreativeId
                     * @property {string|null} [setName] EditCreativeRequest setName
                     * @property {string|null} [setAdAccountId] EditCreativeRequest setAdAccountId
                     * @property {string|null} [setBanner] EditCreativeRequest setBanner
                     * @property {string|null} [setFullmixId] EditCreativeRequest setFullmixId
                     * @property {string|null} [setVideoId] EditCreativeRequest setVideoId
                     * @property {string|null} [setClickthroughUrl] EditCreativeRequest setClickthroughUrl
                     * @property {string|null} [setCtaText] EditCreativeRequest setCtaText
                     * @property {string|null} [setBrandName] EditCreativeRequest setBrandName
                     * @property {string|null} [setTagLine] EditCreativeRequest setTagLine
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormat] EditCreativeRequest setFormat
                     * @property {boolean|null} [setMoatEnabled] EditCreativeRequest setMoatEnabled
                     * @property {string|null} [setIasPixel] EditCreativeRequest setIasPixel
                     * @property {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>|null} [setTrackingPixel] EditCreativeRequest setTrackingPixel
                     */
                    
                    /**
                     * Constructs a new EditCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditCreativeRequest.
                     * @implements IEditCreativeRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditCreativeRequest=} [properties] Properties to set
                     */
                    function EditCreativeRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditCreativeRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.creativeId = "";

                    /**
                     * EditCreativeRequest name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.name = "";

                    /**
                     * EditCreativeRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.adAccountId = "";

                    /**
                     * EditCreativeRequest banner.
                     * @member {string} banner
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.banner = "";

                    /**
                     * EditCreativeRequest fullmixId.
                     * @member {string} fullmixId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.fullmixId = "";

                    /**
                     * EditCreativeRequest videoId.
                     * @member {string} videoId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.videoId = "";

                    /**
                     * EditCreativeRequest clickthroughUrl.
                     * @member {string} clickthroughUrl
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.clickthroughUrl = "";

                    /**
                     * EditCreativeRequest ctaText.
                     * @member {string} ctaText
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.ctaText = "";

                    /**
                     * EditCreativeRequest brandName.
                     * @member {string} brandName
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.brandName = "";

                    /**
                     * EditCreativeRequest tagLine.
                     * @member {string} tagLine
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.tagLine = "";

                    /**
                     * EditCreativeRequest format.
                     * @member {com.spotify.ads.cs.proto.FormatType} format
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.format = {};

                    /**
                     * EditCreativeRequest moatEnabled.
                     * @member {boolean} moatEnabled
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.moatEnabled = false;

                    /**
                     * EditCreativeRequest iasPixel.
                     * @member {string} iasPixel
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.iasPixel = "";

                    /**
                     * EditCreativeRequest trackingPixel.
                     * @member {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>} trackingPixel
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     */
                    EditCreativeRequest.prototype.trackingPixel = [];


                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets banner
                     * @function setBanner
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setBanner = function(value) {
                        this['banner'] = value;
                        return this;
                    };

                    /**
                     * Sets fullmixId
                     * @function setFullmixId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setFullmixId = function(value) {
                        this['fullmixId'] = value;
                        return this;
                    };

                    /**
                     * Sets videoId
                     * @function setVideoId
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setVideoId = function(value) {
                        this['videoId'] = value;
                        return this;
                    };

                    /**
                     * Sets clickthroughUrl
                     * @function setClickthroughUrl
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setClickthroughUrl = function(value) {
                        this['clickthroughUrl'] = value;
                        return this;
                    };

                    /**
                     * Sets ctaText
                     * @function setCtaText
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setCtaText = function(value) {
                        this['ctaText'] = value;
                        return this;
                    };

                    /**
                     * Sets brandName
                     * @function setBrandName
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setBrandName = function(value) {
                        this['brandName'] = value;
                        return this;
                    };

                    /**
                     * Sets tagLine
                     * @function setTagLine
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setTagLine = function(value) {
                        this['tagLine'] = value;
                        return this;
                    };

                    /**
                     * Sets format
                     * @function setFormat
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setFormat = function(value) {
                        this['format'] = value;
                        return this;
                    };

                    /**
                     * Sets moatEnabled
                     * @function setMoatEnabled
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setMoatEnabled = function(value) {
                        this['moatEnabled'] = value;
                        return this;
                    };

                    /**
                     * Sets iasPixel
                     * @function setIasPixel
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setIasPixel = function(value) {
                        this['iasPixel'] = value;
                        return this;
                    };

                    /**
                     * Sets trackingPixel
                     * @function setTrackingPixel
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @param {Array.<com.spotify.ads.campaignservice.ThirdPartyPixel>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditCreativeRequest} EditCreativeRequest
                     */
                    EditCreativeRequest.prototype.setTrackingPixel = function(value) {
                        this['trackingPixel'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditCreativeRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditCreativeRequest
                     * @instance
                     * @returns {object} object
                     */
                    EditCreativeRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditCreativeRequest;
                })();
                
                EditCreativeRequest = proto.EditCreativeRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditCreativeRequest"] = EditCreativeRequest;
exports["default"] = $root;
