// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EditFlightLinkParameters.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EditFlightLinkParameters = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.EditFlightLinkParameters = (function() {
                    /**
                     * Properties of a EditFlightLinkParameters.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IEditFlightLinkParameters
                     * @property {string|null} [flightLinkId] EditFlightLinkParameters flightLinkId
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [creativeRotationParameters] EditFlightLinkParameters creativeRotationParameters
                     * @property {string|null} [setFlightLinkId] EditFlightLinkParameters setFlightLinkId
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [setCreativeRotationParameters] EditFlightLinkParameters setCreativeRotationParameters
                     */
                    
                    /**
                     * Constructs a new EditFlightLinkParameters.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a EditFlightLinkParameters.
                     * @implements IEditFlightLinkParameters
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IEditFlightLinkParameters=} [properties] Properties to set
                     */
                    function EditFlightLinkParameters(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * EditFlightLinkParameters flightLinkId.
                     * @member {string} flightLinkId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinkParameters
                     * @instance
                     */
                    EditFlightLinkParameters.prototype.flightLinkId = "";

                    /**
                     * EditFlightLinkParameters creativeRotationParameters.
                     * @member {com.spotify.adstudiobff.proto.CreativeRotationParameters} creativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinkParameters
                     * @instance
                     */
                    EditFlightLinkParameters.prototype.creativeRotationParameters = {};


                    /**
                     * Sets flightLinkId
                     * @function setFlightLinkId
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinkParameters
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightLinkParameters} EditFlightLinkParameters
                     */
                    EditFlightLinkParameters.prototype.setFlightLinkId = function(value) {
                        this['flightLinkId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeRotationParameters
                     * @function setCreativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinkParameters
                     * @param {com.spotify.adstudiobff.proto.CreativeRotationParameters} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.EditFlightLinkParameters} EditFlightLinkParameters
                     */
                    EditFlightLinkParameters.prototype.setCreativeRotationParameters = function(value) {
                        this['creativeRotationParameters'] = value;
                        return this;
                    };

                    /**
                     * Converts the EditFlightLinkParameters to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.EditFlightLinkParameters
                     * @instance
                     * @returns {object} object
                     */
                    EditFlightLinkParameters.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return EditFlightLinkParameters;
                })();
                
                EditFlightLinkParameters = proto.EditFlightLinkParameters;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EditFlightLinkParameters"] = EditFlightLinkParameters;
exports["default"] = $root;
