// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/EffectiveVoState.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var EffectiveVoState = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                /**
                 * EffectiveVoState enum.
                 * @name com.spotify.adstudiobff.proto.EffectiveVoState
                 * @enum {string}

                 * @property {string} VO_STATE_UNKNOWN=VO_STATE_UNKNOWN VO_STATE_UNKNOWN value
                 * @property {string} VO_READY=VO_READY VO_READY value
                 * @property {string} VO_APPROVED=VO_APPROVED VO_APPROVED value
                 * @property {string} VO_REJECTED=VO_REJECTED VO_REJECTED value
                 * @property {string} VO_PROCESSING=VO_PROCESSING VO_PROCESSING value
                 * @property {string} VO_FAILED=VO_FAILED VO_FAILED value
                 * @property {string} VO_SCRIPT_READY=VO_SCRIPT_READY VO_SCRIPT_READY value
                 * @property {string} VO_WAITING_SCRIPT_APPROVAL=VO_WAITING_SCRIPT_APPROVAL VO_WAITING_SCRIPT_APPROVAL value
                 * @property {string} VO_WAITING_VOICEOVER=VO_WAITING_VOICEOVER VO_WAITING_VOICEOVER value
                 * @property {string} VO_WAITING_APPROVAL=VO_WAITING_APPROVAL VO_WAITING_APPROVAL value
                 * @property {string} VO_WAITING_SCRIPT=VO_WAITING_SCRIPT VO_WAITING_SCRIPT value
                 */
                proto.EffectiveVoState = (function() {
                    var values = {};

                    values["VO_STATE_UNKNOWN"] = "VO_STATE_UNKNOWN";
                    values["VO_READY"] = "VO_READY";
                    values["VO_APPROVED"] = "VO_APPROVED";
                    values["VO_REJECTED"] = "VO_REJECTED";
                    values["VO_PROCESSING"] = "VO_PROCESSING";
                    values["VO_FAILED"] = "VO_FAILED";
                    values["VO_SCRIPT_READY"] = "VO_SCRIPT_READY";
                    values["VO_WAITING_SCRIPT_APPROVAL"] = "VO_WAITING_SCRIPT_APPROVAL";
                    values["VO_WAITING_VOICEOVER"] = "VO_WAITING_VOICEOVER";
                    values["VO_WAITING_APPROVAL"] = "VO_WAITING_APPROVAL";
                    values["VO_WAITING_SCRIPT"] = "VO_WAITING_SCRIPT";
                    return values;
                })();
                
                EffectiveVoState = proto.EffectiveVoState;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["EffectiveVoState"] = EffectiveVoState;
exports["default"] = $root;
