// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/FlightLinkRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var FlightLinkRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.FlightLinkRequest = (function() {
                    /**
                     * Properties of a FlightLinkRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IFlightLinkRequest
                     * @property {string|null} [adAccountId] FlightLinkRequest adAccountId
                     * @property {Array.<string>|null} [creativeIds] FlightLinkRequest creativeIds
                     * @property {string|null} [flightId] FlightLinkRequest flightId
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [format] FlightLinkRequest format
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [creativeRotationParameters] FlightLinkRequest creativeRotationParameters
                     * @property {boolean|null} [serveOnMegaphone] FlightLinkRequest serveOnMegaphone
                     * @property {string|null} [setAdAccountId] FlightLinkRequest setAdAccountId
                     * @property {Array.<string>|null} [setCreativeIds] FlightLinkRequest setCreativeIds
                     * @property {string|null} [setFlightId] FlightLinkRequest setFlightId
                     * @property {com.spotify.ads.cs.proto.FormatType|null} [setFormat] FlightLinkRequest setFormat
                     * @property {com.spotify.adstudiobff.proto.CreativeRotationParameters|null} [setCreativeRotationParameters] FlightLinkRequest setCreativeRotationParameters
                     * @property {boolean|null} [setServeOnMegaphone] FlightLinkRequest setServeOnMegaphone
                     */
                    
                    /**
                     * Constructs a new FlightLinkRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a FlightLinkRequest.
                     * @implements IFlightLinkRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IFlightLinkRequest=} [properties] Properties to set
                     */
                    function FlightLinkRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * FlightLinkRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.adAccountId = "";

                    /**
                     * FlightLinkRequest creativeIds.
                     * @member {Array.<string>} creativeIds
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.creativeIds = [];

                    /**
                     * FlightLinkRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.flightId = "";

                    /**
                     * FlightLinkRequest format.
                     * @member {com.spotify.ads.cs.proto.FormatType} format
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.format = {};

                    /**
                     * FlightLinkRequest creativeRotationParameters.
                     * @member {com.spotify.adstudiobff.proto.CreativeRotationParameters} creativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.creativeRotationParameters = {};

                    /**
                     * FlightLinkRequest serveOnMegaphone.
                     * @member {boolean} serveOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     */
                    FlightLinkRequest.prototype.serveOnMegaphone = false;


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeIds
                     * @function setCreativeIds
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setCreativeIds = function(value) {
                        this['creativeIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets format
                     * @function setFormat
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {com.spotify.ads.cs.proto.FormatType} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setFormat = function(value) {
                        this['format'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeRotationParameters
                     * @function setCreativeRotationParameters
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {com.spotify.adstudiobff.proto.CreativeRotationParameters} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setCreativeRotationParameters = function(value) {
                        this['creativeRotationParameters'] = value;
                        return this;
                    };

                    /**
                     * Sets serveOnMegaphone
                     * @function setServeOnMegaphone
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.FlightLinkRequest} FlightLinkRequest
                     */
                    FlightLinkRequest.prototype.setServeOnMegaphone = function(value) {
                        this['serveOnMegaphone'] = value;
                        return this;
                    };

                    /**
                     * Converts the FlightLinkRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.FlightLinkRequest
                     * @instance
                     * @returns {object} object
                     */
                    FlightLinkRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return FlightLinkRequest;
                })();
                
                FlightLinkRequest = proto.FlightLinkRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["FlightLinkRequest"] = FlightLinkRequest;
exports["default"] = $root;
