// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/Geo.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var Geo = {};
var Type = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.Geo = (function() {
                    /**
                     * Properties of a Geo.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGeo
                     * @property {string|null} [id] Geo id
                     * @property {string|null} [countryCode] Geo countryCode
                     * @property {string|null} [name] Geo name
                     * @property {string|null} [typeDisplayName] Geo typeDisplayName
                     * @property {com.spotify.targetingservice.proto.GeoLocation.Type|null} [geoType] Geo geoType
                     * @property {string|null} [parentName] Geo parentName
                     * @property {boolean|null} [searchable] Geo searchable
                     * @property {boolean|null} [targetableByAdserving] Geo targetableByAdserving
                     * @property {string|null} [setId] Geo setId
                     * @property {string|null} [setCountryCode] Geo setCountryCode
                     * @property {string|null} [setName] Geo setName
                     * @property {string|null} [setTypeDisplayName] Geo setTypeDisplayName
                     * @property {com.spotify.targetingservice.proto.GeoLocation.Type|null} [setGeoType] Geo setGeoType
                     * @property {string|null} [setParentName] Geo setParentName
                     * @property {boolean|null} [setSearchable] Geo setSearchable
                     * @property {boolean|null} [setTargetableByAdserving] Geo setTargetableByAdserving
                     */
                    
                    /**
                     * Constructs a new Geo.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a Geo.
                     * @implements IGeo
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGeo=} [properties] Properties to set
                     */
                    function Geo(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * Geo id.
                     * @member {string} id
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.id = "";

                    /**
                     * Geo countryCode.
                     * @member {string} countryCode
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.countryCode = "";

                    /**
                     * Geo name.
                     * @member {string} name
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.name = "";

                    /**
                     * Geo typeDisplayName.
                     * @member {string} typeDisplayName
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.typeDisplayName = "";

                    /**
                     * Geo geoType.
                     * @member {com.spotify.targetingservice.proto.GeoLocation.Type} geoType
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.geoType = {};

                    /**
                     * Geo parentName.
                     * @member {string} parentName
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.parentName = "";

                    /**
                     * Geo searchable.
                     * @member {boolean} searchable
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.searchable = false;

                    /**
                     * Geo targetableByAdserving.
                     * @member {boolean} targetableByAdserving
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     */
                    Geo.prototype.targetableByAdserving = false;


                    /**
                     * Sets id
                     * @function setId
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setId = function(value) {
                        this['id'] = value;
                        return this;
                    };

                    /**
                     * Sets countryCode
                     * @function setCountryCode
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setCountryCode = function(value) {
                        this['countryCode'] = value;
                        return this;
                    };

                    /**
                     * Sets name
                     * @function setName
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setName = function(value) {
                        this['name'] = value;
                        return this;
                    };

                    /**
                     * Sets typeDisplayName
                     * @function setTypeDisplayName
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setTypeDisplayName = function(value) {
                        this['typeDisplayName'] = value;
                        return this;
                    };

                    /**
                     * Sets geoType
                     * @function setGeoType
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {com.spotify.targetingservice.proto.GeoLocation.Type} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setGeoType = function(value) {
                        this['geoType'] = value;
                        return this;
                    };

                    /**
                     * Sets parentName
                     * @function setParentName
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setParentName = function(value) {
                        this['parentName'] = value;
                        return this;
                    };

                    /**
                     * Sets searchable
                     * @function setSearchable
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setSearchable = function(value) {
                        this['searchable'] = value;
                        return this;
                    };

                    /**
                     * Sets targetableByAdserving
                     * @function setTargetableByAdserving
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @param {boolean} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Geo} Geo
                     */
                    Geo.prototype.setTargetableByAdserving = function(value) {
                        this['targetableByAdserving'] = value;
                        return this;
                    };

                    /**
                     * Converts the Geo to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.Geo
                     * @instance
                     * @returns {object} object
                     */
                    Geo.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    /**
                     * Type enum.
                     * @name com.spotify.adstudiobff.proto.Geo.Type
                     * @enum {string}

                     * @property {string} CITY=CITY CITY value
                     * @property {string} DMA_REGION=DMA_REGION DMA_REGION value
                     * @property {string} STATE=STATE STATE value
                     * @property {string} COUNTRY=COUNTRY COUNTRY value
                     * @property {string} PROVINCE=PROVINCE PROVINCE value
                     * @property {string} POSTAL_CODE=POSTAL_CODE POSTAL_CODE value
                     * @property {string} COUNTY=COUNTY COUNTY value
                     */
                    Geo.Type = (function() {
                        var values = {};

                        values["CITY"] = "CITY";
                        values["DMA_REGION"] = "DMA_REGION";
                        values["STATE"] = "STATE";
                        values["COUNTRY"] = "COUNTRY";
                        values["PROVINCE"] = "PROVINCE";
                        values["POSTAL_CODE"] = "POSTAL_CODE";
                        values["COUNTY"] = "COUNTY";
                        return values;
                    })();
                    
                    Type = Geo.Type;
                    return Geo;
                })();
                
                Geo = proto.Geo;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["Geo"] = Geo;
exports["Type"] = Type;
exports["default"] = $root;
