// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetAvailableOffersRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetAvailableOffersRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetAvailableOffersRequest = (function() {
                    /**
                     * Properties of a GetAvailableOffersRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetAvailableOffersRequest
                     * @property {string|null} [adAccountId] GetAvailableOffersRequest adAccountId
                     * @property {string|null} [flightId] GetAvailableOffersRequest flightId
                     * @property {string|null} [setAdAccountId] GetAvailableOffersRequest setAdAccountId
                     * @property {string|null} [setFlightId] GetAvailableOffersRequest setFlightId
                     */
                    
                    /**
                     * Constructs a new GetAvailableOffersRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetAvailableOffersRequest.
                     * @implements IGetAvailableOffersRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetAvailableOffersRequest=} [properties] Properties to set
                     */
                    function GetAvailableOffersRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetAvailableOffersRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetAvailableOffersRequest
                     * @instance
                     */
                    GetAvailableOffersRequest.prototype.adAccountId = "";

                    /**
                     * GetAvailableOffersRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.GetAvailableOffersRequest
                     * @instance
                     */
                    GetAvailableOffersRequest.prototype.flightId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetAvailableOffersRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetAvailableOffersRequest} GetAvailableOffersRequest
                     */
                    GetAvailableOffersRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.GetAvailableOffersRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetAvailableOffersRequest} GetAvailableOffersRequest
                     */
                    GetAvailableOffersRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetAvailableOffersRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetAvailableOffersRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetAvailableOffersRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetAvailableOffersRequest;
                })();
                
                GetAvailableOffersRequest = proto.GetAvailableOffersRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetAvailableOffersRequest"] = GetAvailableOffersRequest;
exports["default"] = $root;
