// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetBillDetailsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetBillDetailsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetBillDetailsRequest = (function() {
                    /**
                     * Properties of a GetBillDetailsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetBillDetailsRequest
                     * @property {string|null} [adAccountId] GetBillDetailsRequest adAccountId
                     * @property {Array.<string>|null} [billId] GetBillDetailsRequest billId
                     * @property {string|null} [setAdAccountId] GetBillDetailsRequest setAdAccountId
                     * @property {Array.<string>|null} [setBillId] GetBillDetailsRequest setBillId
                     */
                    
                    /**
                     * Constructs a new GetBillDetailsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetBillDetailsRequest.
                     * @implements IGetBillDetailsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetBillDetailsRequest=} [properties] Properties to set
                     */
                    function GetBillDetailsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetBillDetailsRequest ad_account_id.
                     * @member {string} ad_account_id
                     * @memberof com.spotify.adstudiobff.proto.GetBillDetailsRequest
                     * @instance
                     */
                    GetBillDetailsRequest.prototype.adAccountId = "";

                    /**
                     * GetBillDetailsRequest bill_id.
                     * @member {Array.<string>} bill_id
                     * @memberof com.spotify.adstudiobff.proto.GetBillDetailsRequest
                     * @instance
                     */
                    GetBillDetailsRequest.prototype.billId = [];


                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetBillDetailsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetBillDetailsRequest} GetBillDetailsRequest
                     */
                    GetBillDetailsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets bill_id
                     * @function setBillId
                     * @memberof com.spotify.adstudiobff.proto.GetBillDetailsRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetBillDetailsRequest} GetBillDetailsRequest
                     */
                    GetBillDetailsRequest.prototype.setBillId = function(value) {
                        this['billId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetBillDetailsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetBillDetailsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetBillDetailsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetBillDetailsRequest;
                })();
                
                GetBillDetailsRequest = proto.GetBillDetailsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetBillDetailsRequest"] = GetBillDetailsRequest;
exports["default"] = $root;
