// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetBillsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetBillsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetBillsRequest = (function() {
                    /**
                     * Properties of a GetBillsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetBillsRequest
                     * @property {Array.<string>|null} [adAccountId] GetBillsRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.Page|null} [paging] GetBillsRequest paging
                     * @property {Array.<string>|null} [setAdAccountId] GetBillsRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.Page|null} [setPaging] GetBillsRequest setPaging
                     */
                    
                    /**
                     * Constructs a new GetBillsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetBillsRequest.
                     * @implements IGetBillsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetBillsRequest=} [properties] Properties to set
                     */
                    function GetBillsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetBillsRequest ad_account_id.
                     * @member {Array.<string>} ad_account_id
                     * @memberof com.spotify.adstudiobff.proto.GetBillsRequest
                     * @instance
                     */
                    GetBillsRequest.prototype.adAccountId = [];

                    /**
                     * GetBillsRequest paging.
                     * @member {com.spotify.adstudiobff.proto.Page} paging
                     * @memberof com.spotify.adstudiobff.proto.GetBillsRequest
                     * @instance
                     */
                    GetBillsRequest.prototype.paging = {};


                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetBillsRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetBillsRequest} GetBillsRequest
                     */
                    GetBillsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets paging
                     * @function setPaging
                     * @memberof com.spotify.adstudiobff.proto.GetBillsRequest
                     * @param {com.spotify.adstudiobff.proto.Page} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetBillsRequest} GetBillsRequest
                     */
                    GetBillsRequest.prototype.setPaging = function(value) {
                        this['paging'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetBillsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetBillsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetBillsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetBillsRequest;
                })();
                
                GetBillsRequest = proto.GetBillsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetBillsRequest"] = GetBillsRequest;
exports["default"] = $root;
