// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCampaignRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCampaignRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCampaignRequest = (function() {
                    /**
                     * Properties of a GetCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCampaignRequest
                     * @property {string|null} [campaignId] GetCampaignRequest campaignId
                     * @property {string|null} [adAccountId] GetCampaignRequest adAccountId
                     * @property {string|null} [setCampaignId] GetCampaignRequest setCampaignId
                     * @property {string|null} [setAdAccountId] GetCampaignRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new GetCampaignRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCampaignRequest.
                     * @implements IGetCampaignRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCampaignRequest=} [properties] Properties to set
                     */
                    function GetCampaignRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCampaignRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignRequest
                     * @instance
                     */
                    GetCampaignRequest.prototype.campaignId = "";

                    /**
                     * GetCampaignRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignRequest
                     * @instance
                     */
                    GetCampaignRequest.prototype.adAccountId = "";


                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignRequest} GetCampaignRequest
                     */
                    GetCampaignRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignRequest} GetCampaignRequest
                     */
                    GetCampaignRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCampaignRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCampaignRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCampaignRequest;
                })();
                
                GetCampaignRequest = proto.GetCampaignRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCampaignRequest"] = GetCampaignRequest;
exports["default"] = $root;
