// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCreativesRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCreativesRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCreativesRequest = (function() {
                    /**
                     * Properties of a GetCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCreativesRequest
                     * @property {number|null} [limit] GetCreativesRequest limit
                     * @property {number|null} [offset] GetCreativesRequest offset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetCreativesRequest sortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [sortCriteria] GetCreativesRequest sortCriteria
                     * @property {string|null} [adAccountId] GetCreativesRequest adAccountId
                     * @property {string|null} [searchWord] GetCreativesRequest searchWord
                     * @property {string|null} [flightId] GetCreativesRequest flightId
                     * @property {string|null} [campaignId] GetCreativesRequest campaignId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [campaignIds] GetCreativesRequest campaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [flightIds] GetCreativesRequest flightIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [creativeIds] GetCreativesRequest creativeIds
                     * @property {com.spotify.adstudiobff.proto.CreativeState.Status|null} [creativeState] GetCreativesRequest creativeState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [reportDateFilter] GetCreativesRequest reportDateFilter
                     * @property {number|null} [setLimit] GetCreativesRequest setLimit
                     * @property {number|null} [setOffset] GetCreativesRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetCreativesRequest setSortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [setSortCriteria] GetCreativesRequest setSortCriteria
                     * @property {string|null} [setAdAccountId] GetCreativesRequest setAdAccountId
                     * @property {string|null} [setSearchWord] GetCreativesRequest setSearchWord
                     * @property {string|null} [setFlightId] GetCreativesRequest setFlightId
                     * @property {string|null} [setCampaignId] GetCreativesRequest setCampaignId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCampaignIds] GetCreativesRequest setCampaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setFlightIds] GetCreativesRequest setFlightIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCreativeIds] GetCreativesRequest setCreativeIds
                     * @property {com.spotify.adstudiobff.proto.CreativeState.Status|null} [setCreativeState] GetCreativesRequest setCreativeState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [setReportDateFilter] GetCreativesRequest setReportDateFilter
                     */
                    
                    /**
                     * Constructs a new GetCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCreativesRequest.
                     * @implements IGetCreativesRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCreativesRequest=} [properties] Properties to set
                     */
                    function GetCreativesRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCreativesRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.limit = 0;

                    /**
                     * GetCreativesRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.offset = 0;

                    /**
                     * GetCreativesRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.sortDirection = {};

                    /**
                     * GetCreativesRequest sortCriteria.
                     * @member {com.spotify.adstudiobff.proto.HierarchyColumn.Column} sortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.sortCriteria = {};

                    /**
                     * GetCreativesRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.adAccountId = "";

                    /**
                     * GetCreativesRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.searchWord = "";

                    /**
                     * GetCreativesRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.flightId = "";

                    /**
                     * GetCreativesRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.campaignId = "";

                    /**
                     * GetCreativesRequest campaignIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} campaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.campaignIds = {};

                    /**
                     * GetCreativesRequest flightIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} flightIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.flightIds = {};

                    /**
                     * GetCreativesRequest creativeIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} creativeIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.creativeIds = {};

                    /**
                     * GetCreativesRequest creativeState.
                     * @member {com.spotify.adstudiobff.proto.CreativeState.Status} creativeState
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.creativeState = {};

                    /**
                     * GetCreativesRequest reportDateFilter.
                     * @member {com.spotify.adstudiobff.proto.DateRange} reportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     */
                    GetCreativesRequest.prototype.reportDateFilter = {};


                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets sortCriteria
                     * @function setSortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.HierarchyColumn.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setSortCriteria = function(value) {
                        this['sortCriteria'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignIds
                     * @function setCampaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightIds
                     * @function setFlightIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setFlightIds = function(value) {
                        this['flightIds'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeIds
                     * @function setCreativeIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setCreativeIds = function(value) {
                        this['creativeIds'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeState
                     * @function setCreativeState
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.CreativeState.Status} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setCreativeState = function(value) {
                        this['creativeState'] = value;
                        return this;
                    };

                    /**
                     * Sets reportDateFilter
                     * @function setReportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @param {com.spotify.adstudiobff.proto.DateRange} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativesRequest} GetCreativesRequest
                     */
                    GetCreativesRequest.prototype.setReportDateFilter = function(value) {
                        this['reportDateFilter'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCreativesRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCreativesRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCreativesRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCreativesRequest;
                })();
                
                GetCreativesRequest = proto.GetCreativesRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCreativesRequest"] = GetCreativesRequest;
exports["default"] = $root;
