// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCampaignsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCampaignsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCampaignsRequest = (function() {
                    /**
                     * Properties of a GetCampaignsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCampaignsRequest
                     * @property {number|null} [limit] GetCampaignsRequest limit
                     * @property {number|null} [offset] GetCampaignsRequest offset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetCampaignsRequest sortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [sortCriteria] GetCampaignsRequest sortCriteria
                     * @property {string|null} [adAccountId] GetCampaignsRequest adAccountId
                     * @property {string|null} [searchWord] GetCampaignsRequest searchWord
                     * @property {com.spotify.adstudiobff.proto.CampaignState.Status|null} [campaignState] GetCampaignsRequest campaignState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [reportDateFilter] GetCampaignsRequest reportDateFilter
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [campaignIds] GetCampaignsRequest campaignIds
                     * @property {number|null} [setLimit] GetCampaignsRequest setLimit
                     * @property {number|null} [setOffset] GetCampaignsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetCampaignsRequest setSortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [setSortCriteria] GetCampaignsRequest setSortCriteria
                     * @property {string|null} [setAdAccountId] GetCampaignsRequest setAdAccountId
                     * @property {string|null} [setSearchWord] GetCampaignsRequest setSearchWord
                     * @property {com.spotify.adstudiobff.proto.CampaignState.Status|null} [setCampaignState] GetCampaignsRequest setCampaignState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [setReportDateFilter] GetCampaignsRequest setReportDateFilter
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCampaignIds] GetCampaignsRequest setCampaignIds
                     */
                    
                    /**
                     * Constructs a new GetCampaignsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCampaignsRequest.
                     * @implements IGetCampaignsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCampaignsRequest=} [properties] Properties to set
                     */
                    function GetCampaignsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCampaignsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.limit = 0;

                    /**
                     * GetCampaignsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.offset = 0;

                    /**
                     * GetCampaignsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.sortDirection = {};

                    /**
                     * GetCampaignsRequest sortCriteria.
                     * @member {com.spotify.adstudiobff.proto.HierarchyColumn.Column} sortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.sortCriteria = {};

                    /**
                     * GetCampaignsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.adAccountId = "";

                    /**
                     * GetCampaignsRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.searchWord = "";

                    /**
                     * GetCampaignsRequest campaignState.
                     * @member {com.spotify.adstudiobff.proto.CampaignState.Status} campaignState
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.campaignState = {};

                    /**
                     * GetCampaignsRequest reportDateFilter.
                     * @member {com.spotify.adstudiobff.proto.DateRange} reportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.reportDateFilter = {};

                    /**
                     * GetCampaignsRequest campaignIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} campaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     */
                    GetCampaignsRequest.prototype.campaignIds = {};


                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets sortCriteria
                     * @function setSortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {com.spotify.adstudiobff.proto.HierarchyColumn.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setSortCriteria = function(value) {
                        this['sortCriteria'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignState
                     * @function setCampaignState
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {com.spotify.adstudiobff.proto.CampaignState.Status} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setCampaignState = function(value) {
                        this['campaignState'] = value;
                        return this;
                    };

                    /**
                     * Sets reportDateFilter
                     * @function setReportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {com.spotify.adstudiobff.proto.DateRange} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setReportDateFilter = function(value) {
                        this['reportDateFilter'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignIds
                     * @function setCampaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignsRequest} GetCampaignsRequest
                     */
                    GetCampaignsRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCampaignsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCampaignsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCampaignsRequest;
                })();
                
                GetCampaignsRequest = proto.GetCampaignsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCampaignsRequest"] = GetCampaignsRequest;
exports["default"] = $root;
