// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCreativeDraftsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCreativeDraftsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCreativeDraftsRequest = (function() {
                    /**
                     * Properties of a GetCreativeDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCreativeDraftsRequest
                     * @property {string|null} [adAccountId] GetCreativeDraftsRequest adAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [hierarchyDraftStatus] GetCreativeDraftsRequest hierarchyDraftStatus
                     * @property {string|null} [searchWord] GetCreativeDraftsRequest searchWord
                     * @property {number|null} [limit] GetCreativeDraftsRequest limit
                     * @property {number|null} [offset] GetCreativeDraftsRequest offset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [orderBy] GetCreativeDraftsRequest orderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetCreativeDraftsRequest sortDirection
                     * @property {string|null} [campaignId] GetCreativeDraftsRequest campaignId
                     * @property {string|null} [flightId] GetCreativeDraftsRequest flightId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [campaignIds] GetCreativeDraftsRequest campaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [flightIds] GetCreativeDraftsRequest flightIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [hierarchyDraftIds] GetCreativeDraftsRequest hierarchyDraftIds
                     * @property {string|null} [setAdAccountId] GetCreativeDraftsRequest setAdAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [setHierarchyDraftStatus] GetCreativeDraftsRequest setHierarchyDraftStatus
                     * @property {string|null} [setSearchWord] GetCreativeDraftsRequest setSearchWord
                     * @property {number|null} [setLimit] GetCreativeDraftsRequest setLimit
                     * @property {number|null} [setOffset] GetCreativeDraftsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [setOrderBy] GetCreativeDraftsRequest setOrderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetCreativeDraftsRequest setSortDirection
                     * @property {string|null} [setCampaignId] GetCreativeDraftsRequest setCampaignId
                     * @property {string|null} [setFlightId] GetCreativeDraftsRequest setFlightId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCampaignIds] GetCreativeDraftsRequest setCampaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setFlightIds] GetCreativeDraftsRequest setFlightIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setHierarchyDraftIds] GetCreativeDraftsRequest setHierarchyDraftIds
                     */
                    
                    /**
                     * Constructs a new GetCreativeDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCreativeDraftsRequest.
                     * @implements IGetCreativeDraftsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCreativeDraftsRequest=} [properties] Properties to set
                     */
                    function GetCreativeDraftsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCreativeDraftsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.adAccountId = "";

                    /**
                     * GetCreativeDraftsRequest hierarchyDraftStatus.
                     * @member {com.spotify.ads.adstudiodrafts.DraftStatus} hierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.hierarchyDraftStatus = {};

                    /**
                     * GetCreativeDraftsRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.searchWord = "";

                    /**
                     * GetCreativeDraftsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.limit = 0;

                    /**
                     * GetCreativeDraftsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.offset = 0;

                    /**
                     * GetCreativeDraftsRequest orderBy.
                     * @member {com.spotify.adstudiobff.proto.Column} orderBy
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.orderBy = {};

                    /**
                     * GetCreativeDraftsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.sortDirection = {};

                    /**
                     * GetCreativeDraftsRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.campaignId = "";

                    /**
                     * GetCreativeDraftsRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.flightId = "";

                    /**
                     * GetCreativeDraftsRequest campaignIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} campaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.campaignIds = {};

                    /**
                     * GetCreativeDraftsRequest flightIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} flightIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.flightIds = {};

                    /**
                     * GetCreativeDraftsRequest hierarchyDraftIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} hierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     */
                    GetCreativeDraftsRequest.prototype.hierarchyDraftIds = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftStatus
                     * @function setHierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.ads.adstudiodrafts.DraftStatus} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setHierarchyDraftStatus = function(value) {
                        this['hierarchyDraftStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets orderBy
                     * @function setOrderBy
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setOrderBy = function(value) {
                        this['orderBy'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignIds
                     * @function setCampaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightIds
                     * @function setFlightIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setFlightIds = function(value) {
                        this['flightIds'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftIds
                     * @function setHierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeDraftsRequest} GetCreativeDraftsRequest
                     */
                    GetCreativeDraftsRequest.prototype.setHierarchyDraftIds = function(value) {
                        this['hierarchyDraftIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCreativeDraftsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeDraftsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCreativeDraftsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCreativeDraftsRequest;
                })();
                
                GetCreativeDraftsRequest = proto.GetCreativeDraftsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCreativeDraftsRequest"] = GetCreativeDraftsRequest;
exports["default"] = $root;
