// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCreativeRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCreativeRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCreativeRequest = (function() {
                    /**
                     * Properties of a GetCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCreativeRequest
                     * @property {string|null} [creativeId] GetCreativeRequest creativeId
                     * @property {string|null} [adAccountId] GetCreativeRequest adAccountId
                     * @property {string|null} [setCreativeId] GetCreativeRequest setCreativeId
                     * @property {string|null} [setAdAccountId] GetCreativeRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new GetCreativeRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCreativeRequest.
                     * @implements IGetCreativeRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCreativeRequest=} [properties] Properties to set
                     */
                    function GetCreativeRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCreativeRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeRequest
                     * @instance
                     */
                    GetCreativeRequest.prototype.creativeId = "";

                    /**
                     * GetCreativeRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeRequest
                     * @instance
                     */
                    GetCreativeRequest.prototype.adAccountId = "";


                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeRequest} GetCreativeRequest
                     */
                    GetCreativeRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCreativeRequest} GetCreativeRequest
                     */
                    GetCreativeRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCreativeRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCreativeRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCreativeRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCreativeRequest;
                })();
                
                GetCreativeRequest = proto.GetCreativeRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCreativeRequest"] = GetCreativeRequest;
exports["default"] = $root;
