// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetCampaignDraftsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetCampaignDraftsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetCampaignDraftsRequest = (function() {
                    /**
                     * Properties of a GetCampaignDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetCampaignDraftsRequest
                     * @property {string|null} [adAccountId] GetCampaignDraftsRequest adAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [hierarchyDraftStatus] GetCampaignDraftsRequest hierarchyDraftStatus
                     * @property {string|null} [searchWord] GetCampaignDraftsRequest searchWord
                     * @property {number|null} [limit] GetCampaignDraftsRequest limit
                     * @property {number|null} [offset] GetCampaignDraftsRequest offset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [orderBy] GetCampaignDraftsRequest orderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetCampaignDraftsRequest sortDirection
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [hierarchyDraftIds] GetCampaignDraftsRequest hierarchyDraftIds
                     * @property {string|null} [setAdAccountId] GetCampaignDraftsRequest setAdAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [setHierarchyDraftStatus] GetCampaignDraftsRequest setHierarchyDraftStatus
                     * @property {string|null} [setSearchWord] GetCampaignDraftsRequest setSearchWord
                     * @property {number|null} [setLimit] GetCampaignDraftsRequest setLimit
                     * @property {number|null} [setOffset] GetCampaignDraftsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [setOrderBy] GetCampaignDraftsRequest setOrderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetCampaignDraftsRequest setSortDirection
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setHierarchyDraftIds] GetCampaignDraftsRequest setHierarchyDraftIds
                     */
                    
                    /**
                     * Constructs a new GetCampaignDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetCampaignDraftsRequest.
                     * @implements IGetCampaignDraftsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetCampaignDraftsRequest=} [properties] Properties to set
                     */
                    function GetCampaignDraftsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetCampaignDraftsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.adAccountId = "";

                    /**
                     * GetCampaignDraftsRequest hierarchyDraftStatus.
                     * @member {com.spotify.ads.adstudiodrafts.DraftStatus} hierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.hierarchyDraftStatus = {};

                    /**
                     * GetCampaignDraftsRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.searchWord = "";

                    /**
                     * GetCampaignDraftsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.limit = 0;

                    /**
                     * GetCampaignDraftsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.offset = 0;

                    /**
                     * GetCampaignDraftsRequest orderBy.
                     * @member {com.spotify.adstudiobff.proto.Column} orderBy
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.orderBy = {};

                    /**
                     * GetCampaignDraftsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.sortDirection = {};

                    /**
                     * GetCampaignDraftsRequest hierarchyDraftIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} hierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     */
                    GetCampaignDraftsRequest.prototype.hierarchyDraftIds = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftStatus
                     * @function setHierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {com.spotify.ads.adstudiodrafts.DraftStatus} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setHierarchyDraftStatus = function(value) {
                        this['hierarchyDraftStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets orderBy
                     * @function setOrderBy
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setOrderBy = function(value) {
                        this['orderBy'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftIds
                     * @function setHierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetCampaignDraftsRequest} GetCampaignDraftsRequest
                     */
                    GetCampaignDraftsRequest.prototype.setHierarchyDraftIds = function(value) {
                        this['hierarchyDraftIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetCampaignDraftsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetCampaignDraftsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetCampaignDraftsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetCampaignDraftsRequest;
                })();
                
                GetCampaignDraftsRequest = proto.GetCampaignDraftsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetCampaignDraftsRequest"] = GetCampaignDraftsRequest;
exports["default"] = $root;
