// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetEnrichedHierarchyDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetEnrichedHierarchyDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetEnrichedHierarchyDraftRequest = (function() {
                    /**
                     * Properties of a GetEnrichedHierarchyDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetEnrichedHierarchyDraftRequest
                     * @property {string|null} [id] GetEnrichedHierarchyDraftRequest id
                     * @property {string|null} [adAccountId] GetEnrichedHierarchyDraftRequest adAccountId
                     * @property {string|null} [setId] GetEnrichedHierarchyDraftRequest setId
                     * @property {string|null} [setAdAccountId] GetEnrichedHierarchyDraftRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new GetEnrichedHierarchyDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetEnrichedHierarchyDraftRequest.
                     * @implements IGetEnrichedHierarchyDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetEnrichedHierarchyDraftRequest=} [properties] Properties to set
                     */
                    function GetEnrichedHierarchyDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetEnrichedHierarchyDraftRequest id.
                     * @member {string} id
                     * @memberof com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest
                     * @instance
                     */
                    GetEnrichedHierarchyDraftRequest.prototype.id = "";

                    /**
                     * GetEnrichedHierarchyDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest
                     * @instance
                     */
                    GetEnrichedHierarchyDraftRequest.prototype.adAccountId = "";


                    /**
                     * Sets id
                     * @function setId
                     * @memberof com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest} GetEnrichedHierarchyDraftRequest
                     */
                    GetEnrichedHierarchyDraftRequest.prototype.setId = function(value) {
                        this['id'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest} GetEnrichedHierarchyDraftRequest
                     */
                    GetEnrichedHierarchyDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetEnrichedHierarchyDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetEnrichedHierarchyDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetEnrichedHierarchyDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetEnrichedHierarchyDraftRequest;
                })();
                
                GetEnrichedHierarchyDraftRequest = proto.GetEnrichedHierarchyDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetEnrichedHierarchyDraftRequest"] = GetEnrichedHierarchyDraftRequest;
exports["default"] = $root;
