// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetFlightDraftsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetFlightDraftsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetFlightDraftsRequest = (function() {
                    /**
                     * Properties of a GetFlightDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetFlightDraftsRequest
                     * @property {string|null} [adAccountId] GetFlightDraftsRequest adAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [hierarchyDraftStatus] GetFlightDraftsRequest hierarchyDraftStatus
                     * @property {string|null} [searchWord] GetFlightDraftsRequest searchWord
                     * @property {number|null} [limit] GetFlightDraftsRequest limit
                     * @property {number|null} [offset] GetFlightDraftsRequest offset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [orderBy] GetFlightDraftsRequest orderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetFlightDraftsRequest sortDirection
                     * @property {string|null} [campaignId] GetFlightDraftsRequest campaignId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [campaignIds] GetFlightDraftsRequest campaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [hierarchyDraftIds] GetFlightDraftsRequest hierarchyDraftIds
                     * @property {string|null} [setAdAccountId] GetFlightDraftsRequest setAdAccountId
                     * @property {com.spotify.ads.adstudiodrafts.DraftStatus|null} [setHierarchyDraftStatus] GetFlightDraftsRequest setHierarchyDraftStatus
                     * @property {string|null} [setSearchWord] GetFlightDraftsRequest setSearchWord
                     * @property {number|null} [setLimit] GetFlightDraftsRequest setLimit
                     * @property {number|null} [setOffset] GetFlightDraftsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [setOrderBy] GetFlightDraftsRequest setOrderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetFlightDraftsRequest setSortDirection
                     * @property {string|null} [setCampaignId] GetFlightDraftsRequest setCampaignId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCampaignIds] GetFlightDraftsRequest setCampaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setHierarchyDraftIds] GetFlightDraftsRequest setHierarchyDraftIds
                     */
                    
                    /**
                     * Constructs a new GetFlightDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetFlightDraftsRequest.
                     * @implements IGetFlightDraftsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetFlightDraftsRequest=} [properties] Properties to set
                     */
                    function GetFlightDraftsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetFlightDraftsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.adAccountId = "";

                    /**
                     * GetFlightDraftsRequest hierarchyDraftStatus.
                     * @member {com.spotify.ads.adstudiodrafts.DraftStatus} hierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.hierarchyDraftStatus = {};

                    /**
                     * GetFlightDraftsRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.searchWord = "";

                    /**
                     * GetFlightDraftsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.limit = 0;

                    /**
                     * GetFlightDraftsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.offset = 0;

                    /**
                     * GetFlightDraftsRequest orderBy.
                     * @member {com.spotify.adstudiobff.proto.Column} orderBy
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.orderBy = {};

                    /**
                     * GetFlightDraftsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.sortDirection = {};

                    /**
                     * GetFlightDraftsRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.campaignId = "";

                    /**
                     * GetFlightDraftsRequest campaignIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} campaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.campaignIds = {};

                    /**
                     * GetFlightDraftsRequest hierarchyDraftIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} hierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     */
                    GetFlightDraftsRequest.prototype.hierarchyDraftIds = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftStatus
                     * @function setHierarchyDraftStatus
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {com.spotify.ads.adstudiodrafts.DraftStatus} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setHierarchyDraftStatus = function(value) {
                        this['hierarchyDraftStatus'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets orderBy
                     * @function setOrderBy
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setOrderBy = function(value) {
                        this['orderBy'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignIds
                     * @function setCampaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftIds
                     * @function setHierarchyDraftIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightDraftsRequest} GetFlightDraftsRequest
                     */
                    GetFlightDraftsRequest.prototype.setHierarchyDraftIds = function(value) {
                        this['hierarchyDraftIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetFlightDraftsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetFlightDraftsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetFlightDraftsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetFlightDraftsRequest;
                })();
                
                GetFlightDraftsRequest = proto.GetFlightDraftsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetFlightDraftsRequest"] = GetFlightDraftsRequest;
exports["default"] = $root;
