// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetFlightRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetFlightRequest = (function() {
                    /**
                     * Properties of a GetFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetFlightRequest
                     * @property {string|null} [flightId] GetFlightRequest flightId
                     * @property {string|null} [adAccountId] GetFlightRequest adAccountId
                     * @property {string|null} [setFlightId] GetFlightRequest setFlightId
                     * @property {string|null} [setAdAccountId] GetFlightRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new GetFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetFlightRequest.
                     * @implements IGetFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetFlightRequest=} [properties] Properties to set
                     */
                    function GetFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetFlightRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightRequest
                     * @instance
                     */
                    GetFlightRequest.prototype.flightId = "";

                    /**
                     * GetFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightRequest
                     * @instance
                     */
                    GetFlightRequest.prototype.adAccountId = "";


                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightRequest} GetFlightRequest
                     */
                    GetFlightRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightRequest} GetFlightRequest
                     */
                    GetFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetFlightRequest;
                })();
                
                GetFlightRequest = proto.GetFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetFlightRequest"] = GetFlightRequest;
exports["default"] = $root;
