// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetFlightsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetFlightsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetFlightsRequest = (function() {
                    /**
                     * Properties of a GetFlightsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetFlightsRequest
                     * @property {number|null} [limit] GetFlightsRequest limit
                     * @property {number|null} [offset] GetFlightsRequest offset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetFlightsRequest sortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [sortCriteria] GetFlightsRequest sortCriteria
                     * @property {string|null} [adAccountId] GetFlightsRequest adAccountId
                     * @property {string|null} [searchWord] GetFlightsRequest searchWord
                     * @property {string|null} [campaignId] GetFlightsRequest campaignId
                     * @property {string|null} [creativeId] GetFlightsRequest creativeId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [campaignIds] GetFlightsRequest campaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [flightIds] GetFlightsRequest flightIds
                     * @property {com.spotify.adstudiobff.proto.FlightState.Status|null} [flightState] GetFlightsRequest flightState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [reportDateFilter] GetFlightsRequest reportDateFilter
                     * @property {number|null} [setLimit] GetFlightsRequest setLimit
                     * @property {number|null} [setOffset] GetFlightsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetFlightsRequest setSortDirection
                     * @property {com.spotify.adstudiobff.proto.HierarchyColumn.Column|null} [setSortCriteria] GetFlightsRequest setSortCriteria
                     * @property {string|null} [setAdAccountId] GetFlightsRequest setAdAccountId
                     * @property {string|null} [setSearchWord] GetFlightsRequest setSearchWord
                     * @property {string|null} [setCampaignId] GetFlightsRequest setCampaignId
                     * @property {string|null} [setCreativeId] GetFlightsRequest setCreativeId
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setCampaignIds] GetFlightsRequest setCampaignIds
                     * @property {com.spotify.adstudiobff.proto.IdFilter|null} [setFlightIds] GetFlightsRequest setFlightIds
                     * @property {com.spotify.adstudiobff.proto.FlightState.Status|null} [setFlightState] GetFlightsRequest setFlightState
                     * @property {com.spotify.adstudiobff.proto.DateRange|null} [setReportDateFilter] GetFlightsRequest setReportDateFilter
                     */
                    
                    /**
                     * Constructs a new GetFlightsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetFlightsRequest.
                     * @implements IGetFlightsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetFlightsRequest=} [properties] Properties to set
                     */
                    function GetFlightsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetFlightsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.limit = 0;

                    /**
                     * GetFlightsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.offset = 0;

                    /**
                     * GetFlightsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.sortDirection = {};

                    /**
                     * GetFlightsRequest sortCriteria.
                     * @member {com.spotify.adstudiobff.proto.HierarchyColumn.Column} sortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.sortCriteria = {};

                    /**
                     * GetFlightsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.adAccountId = "";

                    /**
                     * GetFlightsRequest searchWord.
                     * @member {string} searchWord
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.searchWord = "";

                    /**
                     * GetFlightsRequest campaignId.
                     * @member {string} campaignId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.campaignId = "";

                    /**
                     * GetFlightsRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.creativeId = "";

                    /**
                     * GetFlightsRequest campaignIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} campaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.campaignIds = {};

                    /**
                     * GetFlightsRequest flightIds.
                     * @member {com.spotify.adstudiobff.proto.IdFilter} flightIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.flightIds = {};

                    /**
                     * GetFlightsRequest flightState.
                     * @member {com.spotify.adstudiobff.proto.FlightState.Status} flightState
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.flightState = {};

                    /**
                     * GetFlightsRequest reportDateFilter.
                     * @member {com.spotify.adstudiobff.proto.DateRange} reportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     */
                    GetFlightsRequest.prototype.reportDateFilter = {};


                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Sets sortCriteria
                     * @function setSortCriteria
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.HierarchyColumn.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setSortCriteria = function(value) {
                        this['sortCriteria'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets searchWord
                     * @function setSearchWord
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setSearchWord = function(value) {
                        this['searchWord'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignId
                     * @function setCampaignId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setCampaignId = function(value) {
                        this['campaignId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignIds
                     * @function setCampaignIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightIds
                     * @function setFlightIds
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.IdFilter} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setFlightIds = function(value) {
                        this['flightIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flightState
                     * @function setFlightState
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.FlightState.Status} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setFlightState = function(value) {
                        this['flightState'] = value;
                        return this;
                    };

                    /**
                     * Sets reportDateFilter
                     * @function setReportDateFilter
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @param {com.spotify.adstudiobff.proto.DateRange} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetFlightsRequest} GetFlightsRequest
                     */
                    GetFlightsRequest.prototype.setReportDateFilter = function(value) {
                        this['reportDateFilter'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetFlightsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetFlightsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetFlightsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetFlightsRequest;
                })();
                
                GetFlightsRequest = proto.GetFlightsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetFlightsRequest"] = GetFlightsRequest;
exports["default"] = $root;
