// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetMixFromVoiceoverRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetMixFromVoiceoverRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetMixFromVoiceoverRequest = (function() {
                    /**
                     * Properties of a GetMixFromVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetMixFromVoiceoverRequest
                     * @property {string|null} [voiceoverId] GetMixFromVoiceoverRequest voiceoverId
                     * @property {string|null} [adAccountId] GetMixFromVoiceoverRequest adAccountId
                     * @property {string|null} [setVoiceoverId] GetMixFromVoiceoverRequest setVoiceoverId
                     * @property {string|null} [setAdAccountId] GetMixFromVoiceoverRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new GetMixFromVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetMixFromVoiceoverRequest.
                     * @implements IGetMixFromVoiceoverRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetMixFromVoiceoverRequest=} [properties] Properties to set
                     */
                    function GetMixFromVoiceoverRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetMixFromVoiceoverRequest voiceoverId.
                     * @member {string} voiceoverId
                     * @memberof com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest
                     * @instance
                     */
                    GetMixFromVoiceoverRequest.prototype.voiceoverId = "";

                    /**
                     * GetMixFromVoiceoverRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest
                     * @instance
                     */
                    GetMixFromVoiceoverRequest.prototype.adAccountId = "";


                    /**
                     * Sets voiceoverId
                     * @function setVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest} GetMixFromVoiceoverRequest
                     */
                    GetMixFromVoiceoverRequest.prototype.setVoiceoverId = function(value) {
                        this['voiceoverId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest} GetMixFromVoiceoverRequest
                     */
                    GetMixFromVoiceoverRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetMixFromVoiceoverRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetMixFromVoiceoverRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetMixFromVoiceoverRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetMixFromVoiceoverRequest;
                })();
                
                GetMixFromVoiceoverRequest = proto.GetMixFromVoiceoverRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetMixFromVoiceoverRequest"] = GetMixFromVoiceoverRequest;
exports["default"] = $root;
