// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/GetReviewableHierarchyDraftsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var GetReviewableHierarchyDraftsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.GetReviewableHierarchyDraftsRequest = (function() {
                    /**
                     * Properties of a GetReviewableHierarchyDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IGetReviewableHierarchyDraftsRequest
                     * @property {string|null} [adAccountId] GetReviewableHierarchyDraftsRequest adAccountId
                     * @property {number|null} [limit] GetReviewableHierarchyDraftsRequest limit
                     * @property {number|null} [offset] GetReviewableHierarchyDraftsRequest offset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [orderBy] GetReviewableHierarchyDraftsRequest orderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [sortDirection] GetReviewableHierarchyDraftsRequest sortDirection
                     * @property {string|null} [setAdAccountId] GetReviewableHierarchyDraftsRequest setAdAccountId
                     * @property {number|null} [setLimit] GetReviewableHierarchyDraftsRequest setLimit
                     * @property {number|null} [setOffset] GetReviewableHierarchyDraftsRequest setOffset
                     * @property {com.spotify.adstudiobff.proto.Column|null} [setOrderBy] GetReviewableHierarchyDraftsRequest setOrderBy
                     * @property {com.spotify.adstudiobff.proto.SortDirection|null} [setSortDirection] GetReviewableHierarchyDraftsRequest setSortDirection
                     */
                    
                    /**
                     * Constructs a new GetReviewableHierarchyDraftsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a GetReviewableHierarchyDraftsRequest.
                     * @implements IGetReviewableHierarchyDraftsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IGetReviewableHierarchyDraftsRequest=} [properties] Properties to set
                     */
                    function GetReviewableHierarchyDraftsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * GetReviewableHierarchyDraftsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.adAccountId = "";

                    /**
                     * GetReviewableHierarchyDraftsRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.limit = 0;

                    /**
                     * GetReviewableHierarchyDraftsRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.offset = 0;

                    /**
                     * GetReviewableHierarchyDraftsRequest orderBy.
                     * @member {com.spotify.adstudiobff.proto.Column} orderBy
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.orderBy = {};

                    /**
                     * GetReviewableHierarchyDraftsRequest sortDirection.
                     * @member {com.spotify.adstudiobff.proto.SortDirection} sortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.sortDirection = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest} GetReviewableHierarchyDraftsRequest
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest} GetReviewableHierarchyDraftsRequest
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest} GetReviewableHierarchyDraftsRequest
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets orderBy
                     * @function setOrderBy
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.Column} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest} GetReviewableHierarchyDraftsRequest
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.setOrderBy = function(value) {
                        this['orderBy'] = value;
                        return this;
                    };

                    /**
                     * Sets sortDirection
                     * @function setSortDirection
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @param {com.spotify.adstudiobff.proto.SortDirection} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest} GetReviewableHierarchyDraftsRequest
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.setSortDirection = function(value) {
                        this['sortDirection'] = value;
                        return this;
                    };

                    /**
                     * Converts the GetReviewableHierarchyDraftsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.GetReviewableHierarchyDraftsRequest
                     * @instance
                     * @returns {object} object
                     */
                    GetReviewableHierarchyDraftsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return GetReviewableHierarchyDraftsRequest;
                })();
                
                GetReviewableHierarchyDraftsRequest = proto.GetReviewableHierarchyDraftsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["GetReviewableHierarchyDraftsRequest"] = GetReviewableHierarchyDraftsRequest;
exports["default"] = $root;
