// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/ManageAdAccountRolesRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var ManageAdAccountRolesRequest = {};
var Action = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.ManageAdAccountRolesRequest = (function() {
                    /**
                     * Properties of a ManageAdAccountRolesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IManageAdAccountRolesRequest
                     * @property {string|null} [adAccountId] ManageAdAccountRolesRequest adAccountId
                     * @property {string|null} [spotifyUserExternalId] ManageAdAccountRolesRequest spotifyUserExternalId
                     * @property {string|null} [roleName] ManageAdAccountRolesRequest roleName
                     * @property {string|null} [prevRoleName] ManageAdAccountRolesRequest prevRoleName
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [action] ManageAdAccountRolesRequest action
                     * @property {string|null} [setAdAccountId] ManageAdAccountRolesRequest setAdAccountId
                     * @property {string|null} [setSpotifyUserExternalId] ManageAdAccountRolesRequest setSpotifyUserExternalId
                     * @property {string|null} [setRoleName] ManageAdAccountRolesRequest setRoleName
                     * @property {string|null} [setPrevRoleName] ManageAdAccountRolesRequest setPrevRoleName
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [setAction] ManageAdAccountRolesRequest setAction
                     */
                    
                    /**
                     * Constructs a new ManageAdAccountRolesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a ManageAdAccountRolesRequest.
                     * @implements IManageAdAccountRolesRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IManageAdAccountRolesRequest=} [properties] Properties to set
                     */
                    function ManageAdAccountRolesRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * ManageAdAccountRolesRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     */
                    ManageAdAccountRolesRequest.prototype.adAccountId = "";

                    /**
                     * ManageAdAccountRolesRequest spotifyUserExternalId.
                     * @member {string} spotifyUserExternalId
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     */
                    ManageAdAccountRolesRequest.prototype.spotifyUserExternalId = "";

                    /**
                     * ManageAdAccountRolesRequest roleName.
                     * @member {string} roleName
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     */
                    ManageAdAccountRolesRequest.prototype.roleName = "";

                    /**
                     * ManageAdAccountRolesRequest prevRoleName.
                     * @member {string} prevRoleName
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     */
                    ManageAdAccountRolesRequest.prototype.prevRoleName = "";

                    /**
                     * ManageAdAccountRolesRequest action.
                     * @member {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} action
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     */
                    ManageAdAccountRolesRequest.prototype.action = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest} ManageAdAccountRolesRequest
                     */
                    ManageAdAccountRolesRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets spotifyUserExternalId
                     * @function setSpotifyUserExternalId
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest} ManageAdAccountRolesRequest
                     */
                    ManageAdAccountRolesRequest.prototype.setSpotifyUserExternalId = function(value) {
                        this['spotifyUserExternalId'] = value;
                        return this;
                    };

                    /**
                     * Sets roleName
                     * @function setRoleName
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest} ManageAdAccountRolesRequest
                     */
                    ManageAdAccountRolesRequest.prototype.setRoleName = function(value) {
                        this['roleName'] = value;
                        return this;
                    };

                    /**
                     * Sets prevRoleName
                     * @function setPrevRoleName
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest} ManageAdAccountRolesRequest
                     */
                    ManageAdAccountRolesRequest.prototype.setPrevRoleName = function(value) {
                        this['prevRoleName'] = value;
                        return this;
                    };

                    /**
                     * Sets action
                     * @function setAction
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @param {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest} ManageAdAccountRolesRequest
                     */
                    ManageAdAccountRolesRequest.prototype.setAction = function(value) {
                        this['action'] = value;
                        return this;
                    };

                    /**
                     * Converts the ManageAdAccountRolesRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest
                     * @instance
                     * @returns {object} object
                     */
                    ManageAdAccountRolesRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    /**
                     * Action enum.
                     * @name com.spotify.adstudiobff.proto.ManageAdAccountRolesRequest.Action
                     * @enum {string}

                     * @property {string} UNSET=UNSET UNSET value
                     * @property {string} ADD=ADD ADD value
                     * @property {string} REMOVE=REMOVE REMOVE value
                     * @property {string} UPDATE=UPDATE UPDATE value
                     */
                    ManageAdAccountRolesRequest.Action = (function() {
                        var values = {};

                        values["UNSET"] = "UNSET";
                        values["ADD"] = "ADD";
                        values["REMOVE"] = "REMOVE";
                        values["UPDATE"] = "UPDATE";
                        return values;
                    })();
                    
                    Action = ManageAdAccountRolesRequest.Action;
                    return ManageAdAccountRolesRequest;
                })();
                
                ManageAdAccountRolesRequest = proto.ManageAdAccountRolesRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["ManageAdAccountRolesRequest"] = ManageAdAccountRolesRequest;
exports["Action"] = Action;
exports["default"] = $root;
