// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/Page.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var Page = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.Page = (function() {
                    /**
                     * Properties of a Page.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IPage
                     * @property {number|null} [limit] Page limit
                     * @property {number|null} [offset] Page offset
                     * @property {number|null} [total] Page total
                     * @property {number|null} [setLimit] Page setLimit
                     * @property {number|null} [setOffset] Page setOffset
                     * @property {number|null} [setTotal] Page setTotal
                     */
                    
                    /**
                     * Constructs a new Page.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a Page.
                     * @implements IPage
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IPage=} [properties] Properties to set
                     */
                    function Page(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * Page limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @instance
                     */
                    Page.prototype.limit = 0;

                    /**
                     * Page offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @instance
                     */
                    Page.prototype.offset = 0;

                    /**
                     * Page total.
                     * @member {number} total
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @instance
                     */
                    Page.prototype.total = 0;


                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Page} Page
                     */
                    Page.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Page} Page
                     */
                    Page.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets total
                     * @function setTotal
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.Page} Page
                     */
                    Page.prototype.setTotal = function(value) {
                        this['total'] = value;
                        return this;
                    };

                    /**
                     * Converts the Page to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.Page
                     * @instance
                     * @returns {object} object
                     */
                    Page.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return Page;
                })();
                
                Page = proto.Page;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["Page"] = Page;
exports["default"] = $root;
