// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/PauseResumeFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var PauseResumeFlightRequest = {};
var Action = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.PauseResumeFlightRequest = (function() {
                    /**
                     * Properties of a PauseResumeFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IPauseResumeFlightRequest
                     * @property {string|null} [adAccountId] PauseResumeFlightRequest adAccountId
                     * @property {string|null} [flightId] PauseResumeFlightRequest flightId
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [action] PauseResumeFlightRequest action
                     * @property {string|null} [setAdAccountId] PauseResumeFlightRequest setAdAccountId
                     * @property {string|null} [setFlightId] PauseResumeFlightRequest setFlightId
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [setAction] PauseResumeFlightRequest setAction
                     */
                    
                    /**
                     * Constructs a new PauseResumeFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a PauseResumeFlightRequest.
                     * @implements IPauseResumeFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IPauseResumeFlightRequest=} [properties] Properties to set
                     */
                    function PauseResumeFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * PauseResumeFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @instance
                     */
                    PauseResumeFlightRequest.prototype.adAccountId = "";

                    /**
                     * PauseResumeFlightRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @instance
                     */
                    PauseResumeFlightRequest.prototype.flightId = "";

                    /**
                     * PauseResumeFlightRequest action.
                     * @member {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} action
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @instance
                     */
                    PauseResumeFlightRequest.prototype.action = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightRequest} PauseResumeFlightRequest
                     */
                    PauseResumeFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightRequest} PauseResumeFlightRequest
                     */
                    PauseResumeFlightRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets action
                     * @function setAction
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @param {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightRequest} PauseResumeFlightRequest
                     */
                    PauseResumeFlightRequest.prototype.setAction = function(value) {
                        this['action'] = value;
                        return this;
                    };

                    /**
                     * Converts the PauseResumeFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    PauseResumeFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    /**
                     * Action enum.
                     * @name com.spotify.adstudiobff.proto.PauseResumeFlightRequest.Action
                     * @enum {string}

                     * @property {string} UNSET=UNSET UNSET value
                     * @property {string} PAUSE=PAUSE PAUSE value
                     * @property {string} RESUME=RESUME RESUME value
                     */
                    PauseResumeFlightRequest.Action = (function() {
                        var values = {};

                        values["UNSET"] = "UNSET";
                        values["PAUSE"] = "PAUSE";
                        values["RESUME"] = "RESUME";
                        return values;
                    })();
                    
                    Action = PauseResumeFlightRequest.Action;
                    return PauseResumeFlightRequest;
                })();
                
                PauseResumeFlightRequest = proto.PauseResumeFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["PauseResumeFlightRequest"] = PauseResumeFlightRequest;
exports["Action"] = Action;
exports["default"] = $root;
