// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/PauseResumeFlightLinkRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var PauseResumeFlightLinkRequest = {};
var Action = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.PauseResumeFlightLinkRequest = (function() {
                    /**
                     * Properties of a PauseResumeFlightLinkRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IPauseResumeFlightLinkRequest
                     * @property {string|null} [adAccountId] PauseResumeFlightLinkRequest adAccountId
                     * @property {string|null} [creativeId] PauseResumeFlightLinkRequest creativeId
                     * @property {string|null} [flightId] PauseResumeFlightLinkRequest flightId
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [action] PauseResumeFlightLinkRequest action
                     * @property {string|null} [setAdAccountId] PauseResumeFlightLinkRequest setAdAccountId
                     * @property {string|null} [setCreativeId] PauseResumeFlightLinkRequest setCreativeId
                     * @property {string|null} [setFlightId] PauseResumeFlightLinkRequest setFlightId
                     * @property {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action|null} [setAction] PauseResumeFlightLinkRequest setAction
                     */
                    
                    /**
                     * Constructs a new PauseResumeFlightLinkRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a PauseResumeFlightLinkRequest.
                     * @implements IPauseResumeFlightLinkRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IPauseResumeFlightLinkRequest=} [properties] Properties to set
                     */
                    function PauseResumeFlightLinkRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * PauseResumeFlightLinkRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @instance
                     */
                    PauseResumeFlightLinkRequest.prototype.adAccountId = "";

                    /**
                     * PauseResumeFlightLinkRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @instance
                     */
                    PauseResumeFlightLinkRequest.prototype.creativeId = "";

                    /**
                     * PauseResumeFlightLinkRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @instance
                     */
                    PauseResumeFlightLinkRequest.prototype.flightId = "";

                    /**
                     * PauseResumeFlightLinkRequest action.
                     * @member {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} action
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @instance
                     */
                    PauseResumeFlightLinkRequest.prototype.action = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest} PauseResumeFlightLinkRequest
                     */
                    PauseResumeFlightLinkRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest} PauseResumeFlightLinkRequest
                     */
                    PauseResumeFlightLinkRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest} PauseResumeFlightLinkRequest
                     */
                    PauseResumeFlightLinkRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets action
                     * @function setAction
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @param {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest} PauseResumeFlightLinkRequest
                     */
                    PauseResumeFlightLinkRequest.prototype.setAction = function(value) {
                        this['action'] = value;
                        return this;
                    };

                    /**
                     * Converts the PauseResumeFlightLinkRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest
                     * @instance
                     * @returns {object} object
                     */
                    PauseResumeFlightLinkRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    /**
                     * Action enum.
                     * @name com.spotify.adstudiobff.proto.PauseResumeFlightLinkRequest.Action
                     * @enum {string}

                     * @property {string} UNKNOWN=UNKNOWN UNKNOWN value
                     * @property {string} PAUSE=PAUSE PAUSE value
                     * @property {string} RESUME=RESUME RESUME value
                     */
                    PauseResumeFlightLinkRequest.Action = (function() {
                        var values = {};

                        values["UNKNOWN"] = "UNKNOWN";
                        values["PAUSE"] = "PAUSE";
                        values["RESUME"] = "RESUME";
                        return values;
                    })();
                    
                    Action = PauseResumeFlightLinkRequest.Action;
                    return PauseResumeFlightLinkRequest;
                })();
                
                PauseResumeFlightLinkRequest = proto.PauseResumeFlightLinkRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["PauseResumeFlightLinkRequest"] = PauseResumeFlightLinkRequest;
exports["Action"] = Action;
exports["default"] = $root;
