// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/PendingUserApprovalCreativesRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var PendingUserApprovalCreativesRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.PendingUserApprovalCreativesRequest = (function() {
                    /**
                     * Properties of a PendingUserApprovalCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IPendingUserApprovalCreativesRequest
                     * @property {string|null} [adAccountId] PendingUserApprovalCreativesRequest adAccountId
                     * @property {number|null} [limit] PendingUserApprovalCreativesRequest limit
                     * @property {number|null} [offset] PendingUserApprovalCreativesRequest offset
                     * @property {string|null} [creativeId] PendingUserApprovalCreativesRequest creativeId
                     * @property {string|null} [setAdAccountId] PendingUserApprovalCreativesRequest setAdAccountId
                     * @property {number|null} [setLimit] PendingUserApprovalCreativesRequest setLimit
                     * @property {number|null} [setOffset] PendingUserApprovalCreativesRequest setOffset
                     * @property {string|null} [setCreativeId] PendingUserApprovalCreativesRequest setCreativeId
                     */
                    
                    /**
                     * Constructs a new PendingUserApprovalCreativesRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a PendingUserApprovalCreativesRequest.
                     * @implements IPendingUserApprovalCreativesRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IPendingUserApprovalCreativesRequest=} [properties] Properties to set
                     */
                    function PendingUserApprovalCreativesRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * PendingUserApprovalCreativesRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @instance
                     */
                    PendingUserApprovalCreativesRequest.prototype.adAccountId = "";

                    /**
                     * PendingUserApprovalCreativesRequest limit.
                     * @member {number} limit
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @instance
                     */
                    PendingUserApprovalCreativesRequest.prototype.limit = 0;

                    /**
                     * PendingUserApprovalCreativesRequest offset.
                     * @member {number} offset
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @instance
                     */
                    PendingUserApprovalCreativesRequest.prototype.offset = 0;

                    /**
                     * PendingUserApprovalCreativesRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @instance
                     */
                    PendingUserApprovalCreativesRequest.prototype.creativeId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest} PendingUserApprovalCreativesRequest
                     */
                    PendingUserApprovalCreativesRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets limit
                     * @function setLimit
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest} PendingUserApprovalCreativesRequest
                     */
                    PendingUserApprovalCreativesRequest.prototype.setLimit = function(value) {
                        this['limit'] = value;
                        return this;
                    };

                    /**
                     * Sets offset
                     * @function setOffset
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @param {number} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest} PendingUserApprovalCreativesRequest
                     */
                    PendingUserApprovalCreativesRequest.prototype.setOffset = function(value) {
                        this['offset'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest} PendingUserApprovalCreativesRequest
                     */
                    PendingUserApprovalCreativesRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Converts the PendingUserApprovalCreativesRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.PendingUserApprovalCreativesRequest
                     * @instance
                     * @returns {object} object
                     */
                    PendingUserApprovalCreativesRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return PendingUserApprovalCreativesRequest;
                })();
                
                PendingUserApprovalCreativesRequest = proto.PendingUserApprovalCreativesRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["PendingUserApprovalCreativesRequest"] = PendingUserApprovalCreativesRequest;
exports["default"] = $root;
