// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/RejectVoiceoverRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var RejectVoiceoverRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.RejectVoiceoverRequest = (function() {
                    /**
                     * Properties of a RejectVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IRejectVoiceoverRequest
                     * @property {string|null} [voiceoverId] RejectVoiceoverRequest voiceoverId
                     * @property {string|null} [adAccountId] RejectVoiceoverRequest adAccountId
                     * @property {string|null} [flightId] RejectVoiceoverRequest flightId
                     * @property {string|null} [creativeId] RejectVoiceoverRequest creativeId
                     * @property {string|null} [rejectionReason] RejectVoiceoverRequest rejectionReason
                     * @property {com.spotify.adstudioasset.RejectionCode|null} [rejectionCode] RejectVoiceoverRequest rejectionCode
                     * @property {string|null} [setVoiceoverId] RejectVoiceoverRequest setVoiceoverId
                     * @property {string|null} [setAdAccountId] RejectVoiceoverRequest setAdAccountId
                     * @property {string|null} [setFlightId] RejectVoiceoverRequest setFlightId
                     * @property {string|null} [setCreativeId] RejectVoiceoverRequest setCreativeId
                     * @property {string|null} [setRejectionReason] RejectVoiceoverRequest setRejectionReason
                     * @property {com.spotify.adstudioasset.RejectionCode|null} [setRejectionCode] RejectVoiceoverRequest setRejectionCode
                     */
                    
                    /**
                     * Constructs a new RejectVoiceoverRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a RejectVoiceoverRequest.
                     * @implements IRejectVoiceoverRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IRejectVoiceoverRequest=} [properties] Properties to set
                     */
                    function RejectVoiceoverRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * RejectVoiceoverRequest voiceoverId.
                     * @member {string} voiceoverId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.voiceoverId = "";

                    /**
                     * RejectVoiceoverRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.adAccountId = "";

                    /**
                     * RejectVoiceoverRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.flightId = "";

                    /**
                     * RejectVoiceoverRequest creativeId.
                     * @member {string} creativeId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.creativeId = "";

                    /**
                     * RejectVoiceoverRequest rejectionReason.
                     * @member {string} rejectionReason
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.rejectionReason = "";

                    /**
                     * RejectVoiceoverRequest rejectionCode.
                     * @member {com.spotify.adstudioasset.RejectionCode} rejectionCode
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     */
                    RejectVoiceoverRequest.prototype.rejectionCode = {};


                    /**
                     * Sets voiceoverId
                     * @function setVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setVoiceoverId = function(value) {
                        this['voiceoverId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeId
                     * @function setCreativeId
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setCreativeId = function(value) {
                        this['creativeId'] = value;
                        return this;
                    };

                    /**
                     * Sets rejectionReason
                     * @function setRejectionReason
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setRejectionReason = function(value) {
                        this['rejectionReason'] = value;
                        return this;
                    };

                    /**
                     * Sets rejectionCode
                     * @function setRejectionCode
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @param {com.spotify.adstudioasset.RejectionCode} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RejectVoiceoverRequest} RejectVoiceoverRequest
                     */
                    RejectVoiceoverRequest.prototype.setRejectionCode = function(value) {
                        this['rejectionCode'] = value;
                        return this;
                    };

                    /**
                     * Converts the RejectVoiceoverRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.RejectVoiceoverRequest
                     * @instance
                     * @returns {object} object
                     */
                    RejectVoiceoverRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return RejectVoiceoverRequest;
                })();
                
                RejectVoiceoverRequest = proto.RejectVoiceoverRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["RejectVoiceoverRequest"] = RejectVoiceoverRequest;
exports["default"] = $root;
