// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/RevisionRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var RevisionRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.RevisionRequest = (function() {
                    /**
                     * Properties of a RevisionRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IRevisionRequest
                     * @property {string|null} [voiceoverId] RevisionRequest voiceoverId
                     * @property {string|null} [instructions] RevisionRequest instructions
                     * @property {string|null} [adAccountId] RevisionRequest adAccountId
                     * @property {string|null} [setVoiceoverId] RevisionRequest setVoiceoverId
                     * @property {string|null} [setInstructions] RevisionRequest setInstructions
                     * @property {string|null} [setAdAccountId] RevisionRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new RevisionRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a RevisionRequest.
                     * @implements IRevisionRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IRevisionRequest=} [properties] Properties to set
                     */
                    function RevisionRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * RevisionRequest voiceoverId.
                     * @member {string} voiceoverId
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @instance
                     */
                    RevisionRequest.prototype.voiceoverId = "";

                    /**
                     * RevisionRequest instructions.
                     * @member {string} instructions
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @instance
                     */
                    RevisionRequest.prototype.instructions = "";

                    /**
                     * RevisionRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @instance
                     */
                    RevisionRequest.prototype.adAccountId = "";


                    /**
                     * Sets voiceoverId
                     * @function setVoiceoverId
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RevisionRequest} RevisionRequest
                     */
                    RevisionRequest.prototype.setVoiceoverId = function(value) {
                        this['voiceoverId'] = value;
                        return this;
                    };

                    /**
                     * Sets instructions
                     * @function setInstructions
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RevisionRequest} RevisionRequest
                     */
                    RevisionRequest.prototype.setInstructions = function(value) {
                        this['instructions'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.RevisionRequest} RevisionRequest
                     */
                    RevisionRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the RevisionRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.RevisionRequest
                     * @instance
                     * @returns {object} object
                     */
                    RevisionRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return RevisionRequest;
                })();
                
                RevisionRequest = proto.RevisionRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["RevisionRequest"] = RevisionRequest;
exports["default"] = $root;
