// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/SetUserAsOwnerRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var SetUserAsOwnerRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.SetUserAsOwnerRequest = (function() {
                    /**
                     * Properties of a SetUserAsOwnerRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ISetUserAsOwnerRequest
                     * @property {string|null} [adAccountId] SetUserAsOwnerRequest adAccountId
                     * @property {string|null} [spotifyUserExternalId] SetUserAsOwnerRequest spotifyUserExternalId
                     * @property {string|null} [setAdAccountId] SetUserAsOwnerRequest setAdAccountId
                     * @property {string|null} [setSpotifyUserExternalId] SetUserAsOwnerRequest setSpotifyUserExternalId
                     */
                    
                    /**
                     * Constructs a new SetUserAsOwnerRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a SetUserAsOwnerRequest.
                     * @implements ISetUserAsOwnerRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ISetUserAsOwnerRequest=} [properties] Properties to set
                     */
                    function SetUserAsOwnerRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * SetUserAsOwnerRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.SetUserAsOwnerRequest
                     * @instance
                     */
                    SetUserAsOwnerRequest.prototype.adAccountId = "";

                    /**
                     * SetUserAsOwnerRequest spotifyUserExternalId.
                     * @member {string} spotifyUserExternalId
                     * @memberof com.spotify.adstudiobff.proto.SetUserAsOwnerRequest
                     * @instance
                     */
                    SetUserAsOwnerRequest.prototype.spotifyUserExternalId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.SetUserAsOwnerRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.SetUserAsOwnerRequest} SetUserAsOwnerRequest
                     */
                    SetUserAsOwnerRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets spotifyUserExternalId
                     * @function setSpotifyUserExternalId
                     * @memberof com.spotify.adstudiobff.proto.SetUserAsOwnerRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.SetUserAsOwnerRequest} SetUserAsOwnerRequest
                     */
                    SetUserAsOwnerRequest.prototype.setSpotifyUserExternalId = function(value) {
                        this['spotifyUserExternalId'] = value;
                        return this;
                    };

                    /**
                     * Converts the SetUserAsOwnerRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.SetUserAsOwnerRequest
                     * @instance
                     * @returns {object} object
                     */
                    SetUserAsOwnerRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return SetUserAsOwnerRequest;
                })();
                
                SetUserAsOwnerRequest = proto.SetUserAsOwnerRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["SetUserAsOwnerRequest"] = SetUserAsOwnerRequest;
exports["default"] = $root;
