// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/StatsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var StatsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.StatsRequest = (function() {
                    /**
                     * Properties of a StatsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IStatsRequest
                     * @property {string|null} [flightId] StatsRequest flightId
                     * @property {string|null} [adAccountId] StatsRequest adAccountId
                     * @property {com.spotify.adstudiobff.proto.Targeting|null} [targetingType] StatsRequest targetingType
                     * @property {string|null} [setFlightId] StatsRequest setFlightId
                     * @property {string|null} [setAdAccountId] StatsRequest setAdAccountId
                     * @property {com.spotify.adstudiobff.proto.Targeting|null} [setTargetingType] StatsRequest setTargetingType
                     */
                    
                    /**
                     * Constructs a new StatsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a StatsRequest.
                     * @implements IStatsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IStatsRequest=} [properties] Properties to set
                     */
                    function StatsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * StatsRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @instance
                     */
                    StatsRequest.prototype.flightId = "";

                    /**
                     * StatsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @instance
                     */
                    StatsRequest.prototype.adAccountId = "";

                    /**
                     * StatsRequest targetingType.
                     * @member {com.spotify.adstudiobff.proto.Targeting} targetingType
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @instance
                     */
                    StatsRequest.prototype.targetingType = {};


                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.StatsRequest} StatsRequest
                     */
                    StatsRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.StatsRequest} StatsRequest
                     */
                    StatsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets targetingType
                     * @function setTargetingType
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @param {com.spotify.adstudiobff.proto.Targeting} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.StatsRequest} StatsRequest
                     */
                    StatsRequest.prototype.setTargetingType = function(value) {
                        this['targetingType'] = value;
                        return this;
                    };

                    /**
                     * Converts the StatsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.StatsRequest
                     * @instance
                     * @returns {object} object
                     */
                    StatsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return StatsRequest;
                })();
                
                StatsRequest = proto.StatsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["StatsRequest"] = StatsRequest;
exports["default"] = $root;
