// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/StopFlightRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var StopFlightRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.StopFlightRequest = (function() {
                    /**
                     * Properties of a StopFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IStopFlightRequest
                     * @property {string|null} [adAccountId] StopFlightRequest adAccountId
                     * @property {string|null} [flightId] StopFlightRequest flightId
                     * @property {string|null} [setAdAccountId] StopFlightRequest setAdAccountId
                     * @property {string|null} [setFlightId] StopFlightRequest setFlightId
                     */
                    
                    /**
                     * Constructs a new StopFlightRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a StopFlightRequest.
                     * @implements IStopFlightRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IStopFlightRequest=} [properties] Properties to set
                     */
                    function StopFlightRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * StopFlightRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.StopFlightRequest
                     * @instance
                     */
                    StopFlightRequest.prototype.adAccountId = "";

                    /**
                     * StopFlightRequest flightId.
                     * @member {string} flightId
                     * @memberof com.spotify.adstudiobff.proto.StopFlightRequest
                     * @instance
                     */
                    StopFlightRequest.prototype.flightId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.StopFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.StopFlightRequest} StopFlightRequest
                     */
                    StopFlightRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightId
                     * @function setFlightId
                     * @memberof com.spotify.adstudiobff.proto.StopFlightRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.StopFlightRequest} StopFlightRequest
                     */
                    StopFlightRequest.prototype.setFlightId = function(value) {
                        this['flightId'] = value;
                        return this;
                    };

                    /**
                     * Converts the StopFlightRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.StopFlightRequest
                     * @instance
                     * @returns {object} object
                     */
                    StopFlightRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return StopFlightRequest;
                })();
                
                StopFlightRequest = proto.StopFlightRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["StopFlightRequest"] = StopFlightRequest;
exports["default"] = $root;
