// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/TaxIdValidationRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var TaxIdValidationRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.TaxIdValidationRequest = (function() {
                    /**
                     * Properties of a TaxIdValidationRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ITaxIdValidationRequest
                     * @property {string|null} [taxId] TaxIdValidationRequest taxId
                     * @property {string|null} [countryCode] TaxIdValidationRequest countryCode
                     * @property {string|null} [setTaxId] TaxIdValidationRequest setTaxId
                     * @property {string|null} [setCountryCode] TaxIdValidationRequest setCountryCode
                     */
                    
                    /**
                     * Constructs a new TaxIdValidationRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a TaxIdValidationRequest.
                     * @implements ITaxIdValidationRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ITaxIdValidationRequest=} [properties] Properties to set
                     */
                    function TaxIdValidationRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * TaxIdValidationRequest taxId.
                     * @member {string} taxId
                     * @memberof com.spotify.adstudiobff.proto.TaxIdValidationRequest
                     * @instance
                     */
                    TaxIdValidationRequest.prototype.taxId = "";

                    /**
                     * TaxIdValidationRequest countryCode.
                     * @member {string} countryCode
                     * @memberof com.spotify.adstudiobff.proto.TaxIdValidationRequest
                     * @instance
                     */
                    TaxIdValidationRequest.prototype.countryCode = "";


                    /**
                     * Sets taxId
                     * @function setTaxId
                     * @memberof com.spotify.adstudiobff.proto.TaxIdValidationRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.TaxIdValidationRequest} TaxIdValidationRequest
                     */
                    TaxIdValidationRequest.prototype.setTaxId = function(value) {
                        this['taxId'] = value;
                        return this;
                    };

                    /**
                     * Sets countryCode
                     * @function setCountryCode
                     * @memberof com.spotify.adstudiobff.proto.TaxIdValidationRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.TaxIdValidationRequest} TaxIdValidationRequest
                     */
                    TaxIdValidationRequest.prototype.setCountryCode = function(value) {
                        this['countryCode'] = value;
                        return this;
                    };

                    /**
                     * Converts the TaxIdValidationRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.TaxIdValidationRequest
                     * @instance
                     * @returns {object} object
                     */
                    TaxIdValidationRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return TaxIdValidationRequest;
                })();
                
                TaxIdValidationRequest = proto.TaxIdValidationRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["TaxIdValidationRequest"] = TaxIdValidationRequest;
exports["default"] = $root;
