// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/TermsAndConditionsRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var TermsAndConditionsRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.TermsAndConditionsRequest = (function() {
                    /**
                     * Properties of a TermsAndConditionsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface ITermsAndConditionsRequest
                     * @property {string|null} [adAccountId] TermsAndConditionsRequest adAccountId
                     * @property {string|null} [clientId] TermsAndConditionsRequest clientId
                     * @property {string|null} [setAdAccountId] TermsAndConditionsRequest setAdAccountId
                     * @property {string|null} [setClientId] TermsAndConditionsRequest setClientId
                     */
                    
                    /**
                     * Constructs a new TermsAndConditionsRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a TermsAndConditionsRequest.
                     * @implements ITermsAndConditionsRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.ITermsAndConditionsRequest=} [properties] Properties to set
                     */
                    function TermsAndConditionsRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * TermsAndConditionsRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.TermsAndConditionsRequest
                     * @instance
                     */
                    TermsAndConditionsRequest.prototype.adAccountId = "";

                    /**
                     * TermsAndConditionsRequest clientId.
                     * @member {string} clientId
                     * @memberof com.spotify.adstudiobff.proto.TermsAndConditionsRequest
                     * @instance
                     */
                    TermsAndConditionsRequest.prototype.clientId = "";


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.TermsAndConditionsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.TermsAndConditionsRequest} TermsAndConditionsRequest
                     */
                    TermsAndConditionsRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets clientId
                     * @function setClientId
                     * @memberof com.spotify.adstudiobff.proto.TermsAndConditionsRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.TermsAndConditionsRequest} TermsAndConditionsRequest
                     */
                    TermsAndConditionsRequest.prototype.setClientId = function(value) {
                        this['clientId'] = value;
                        return this;
                    };

                    /**
                     * Converts the TermsAndConditionsRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.TermsAndConditionsRequest
                     * @instance
                     * @returns {object} object
                     */
                    TermsAndConditionsRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return TermsAndConditionsRequest;
                })();
                
                TermsAndConditionsRequest = proto.TermsAndConditionsRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["TermsAndConditionsRequest"] = TermsAndConditionsRequest;
exports["default"] = $root;
