// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/UpdateCreativeDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var UpdateCreativeDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.UpdateCreativeDraftRequest = (function() {
                    /**
                     * Properties of a UpdateCreativeDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IUpdateCreativeDraftRequest
                     * @property {string|null} [adAccountId] UpdateCreativeDraftRequest adAccountId
                     * @property {string|null} [hierarchyDraftId] UpdateCreativeDraftRequest hierarchyDraftId
                     * @property {string|null} [creativeDraftId] UpdateCreativeDraftRequest creativeDraftId
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [creativeDraftFormValues] UpdateCreativeDraftRequest creativeDraftFormValues
                     * @property {string|null} [setAdAccountId] UpdateCreativeDraftRequest setAdAccountId
                     * @property {string|null} [setHierarchyDraftId] UpdateCreativeDraftRequest setHierarchyDraftId
                     * @property {string|null} [setCreativeDraftId] UpdateCreativeDraftRequest setCreativeDraftId
                     * @property {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues|null} [setCreativeDraftFormValues] UpdateCreativeDraftRequest setCreativeDraftFormValues
                     */
                    
                    /**
                     * Constructs a new UpdateCreativeDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a UpdateCreativeDraftRequest.
                     * @implements IUpdateCreativeDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IUpdateCreativeDraftRequest=} [properties] Properties to set
                     */
                    function UpdateCreativeDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * UpdateCreativeDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @instance
                     */
                    UpdateCreativeDraftRequest.prototype.adAccountId = "";

                    /**
                     * UpdateCreativeDraftRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @instance
                     */
                    UpdateCreativeDraftRequest.prototype.hierarchyDraftId = "";

                    /**
                     * UpdateCreativeDraftRequest creativeDraftId.
                     * @member {string} creativeDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @instance
                     */
                    UpdateCreativeDraftRequest.prototype.creativeDraftId = "";

                    /**
                     * UpdateCreativeDraftRequest creativeDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} creativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @instance
                     */
                    UpdateCreativeDraftRequest.prototype.creativeDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest} UpdateCreativeDraftRequest
                     */
                    UpdateCreativeDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest} UpdateCreativeDraftRequest
                     */
                    UpdateCreativeDraftRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeDraftId
                     * @function setCreativeDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest} UpdateCreativeDraftRequest
                     */
                    UpdateCreativeDraftRequest.prototype.setCreativeDraftId = function(value) {
                        this['creativeDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets creativeDraftFormValues
                     * @function setCreativeDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.CreativeDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest} UpdateCreativeDraftRequest
                     */
                    UpdateCreativeDraftRequest.prototype.setCreativeDraftFormValues = function(value) {
                        this['creativeDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the UpdateCreativeDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.UpdateCreativeDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    UpdateCreativeDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return UpdateCreativeDraftRequest;
                })();
                
                UpdateCreativeDraftRequest = proto.UpdateCreativeDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["UpdateCreativeDraftRequest"] = UpdateCreativeDraftRequest;
exports["default"] = $root;
