// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/UpdateCampaignDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var UpdateCampaignDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.UpdateCampaignDraftRequest = (function() {
                    /**
                     * Properties of a UpdateCampaignDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IUpdateCampaignDraftRequest
                     * @property {string|null} [adAccountId] UpdateCampaignDraftRequest adAccountId
                     * @property {string|null} [hierarchyDraftId] UpdateCampaignDraftRequest hierarchyDraftId
                     * @property {string|null} [campaignDraftId] UpdateCampaignDraftRequest campaignDraftId
                     * @property {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues|null} [campaignDraftFormValues] UpdateCampaignDraftRequest campaignDraftFormValues
                     * @property {string|null} [setAdAccountId] UpdateCampaignDraftRequest setAdAccountId
                     * @property {string|null} [setHierarchyDraftId] UpdateCampaignDraftRequest setHierarchyDraftId
                     * @property {string|null} [setCampaignDraftId] UpdateCampaignDraftRequest setCampaignDraftId
                     * @property {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues|null} [setCampaignDraftFormValues] UpdateCampaignDraftRequest setCampaignDraftFormValues
                     */
                    
                    /**
                     * Constructs a new UpdateCampaignDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a UpdateCampaignDraftRequest.
                     * @implements IUpdateCampaignDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IUpdateCampaignDraftRequest=} [properties] Properties to set
                     */
                    function UpdateCampaignDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * UpdateCampaignDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @instance
                     */
                    UpdateCampaignDraftRequest.prototype.adAccountId = "";

                    /**
                     * UpdateCampaignDraftRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @instance
                     */
                    UpdateCampaignDraftRequest.prototype.hierarchyDraftId = "";

                    /**
                     * UpdateCampaignDraftRequest campaignDraftId.
                     * @member {string} campaignDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @instance
                     */
                    UpdateCampaignDraftRequest.prototype.campaignDraftId = "";

                    /**
                     * UpdateCampaignDraftRequest campaignDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues} campaignDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @instance
                     */
                    UpdateCampaignDraftRequest.prototype.campaignDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest} UpdateCampaignDraftRequest
                     */
                    UpdateCampaignDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest} UpdateCampaignDraftRequest
                     */
                    UpdateCampaignDraftRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignDraftId
                     * @function setCampaignDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest} UpdateCampaignDraftRequest
                     */
                    UpdateCampaignDraftRequest.prototype.setCampaignDraftId = function(value) {
                        this['campaignDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaignDraftFormValues
                     * @function setCampaignDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.CampaignDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest} UpdateCampaignDraftRequest
                     */
                    UpdateCampaignDraftRequest.prototype.setCampaignDraftFormValues = function(value) {
                        this['campaignDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the UpdateCampaignDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.UpdateCampaignDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    UpdateCampaignDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return UpdateCampaignDraftRequest;
                })();
                
                UpdateCampaignDraftRequest = proto.UpdateCampaignDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["UpdateCampaignDraftRequest"] = UpdateCampaignDraftRequest;
exports["default"] = $root;
