// @spotify-internal/adstudio-bff-clients/models/com/spotify/adstudiobff/proto/UpdateFlightDraftRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var UpdateFlightDraftRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adstudiobff = (function() {
            /**
             * Namespace adstudiobff.
             * @memberof com.spotify
             * @namespace
             */
            var adstudiobff = {};

            adstudiobff.proto = (function() {
                /**
                 * Namespace proto.
                 * @memberof com.spotify.adstudiobff
                 * @namespace
                 */
                var proto = {};

                proto.UpdateFlightDraftRequest = (function() {
                    /**
                     * Properties of a UpdateFlightDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @interface IUpdateFlightDraftRequest
                     * @property {string|null} [adAccountId] UpdateFlightDraftRequest adAccountId
                     * @property {string|null} [hierarchyDraftId] UpdateFlightDraftRequest hierarchyDraftId
                     * @property {string|null} [flightDraftId] UpdateFlightDraftRequest flightDraftId
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [flightDraftFormValues] UpdateFlightDraftRequest flightDraftFormValues
                     * @property {string|null} [setAdAccountId] UpdateFlightDraftRequest setAdAccountId
                     * @property {string|null} [setHierarchyDraftId] UpdateFlightDraftRequest setHierarchyDraftId
                     * @property {string|null} [setFlightDraftId] UpdateFlightDraftRequest setFlightDraftId
                     * @property {com.spotify.ads.adstudiodrafts.FlightDraftFormValues|null} [setFlightDraftFormValues] UpdateFlightDraftRequest setFlightDraftFormValues
                     */
                    
                    /**
                     * Constructs a new UpdateFlightDraftRequest.
                     * @memberof com.spotify.adstudiobff.proto
                     * @classdesc Represents a UpdateFlightDraftRequest.
                     * @implements IUpdateFlightDraftRequest
                     * @constructor
                     * @param {com.spotify.adstudiobff.proto.IUpdateFlightDraftRequest=} [properties] Properties to set
                     */
                    function UpdateFlightDraftRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * UpdateFlightDraftRequest adAccountId.
                     * @member {string} adAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @instance
                     */
                    UpdateFlightDraftRequest.prototype.adAccountId = "";

                    /**
                     * UpdateFlightDraftRequest hierarchyDraftId.
                     * @member {string} hierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @instance
                     */
                    UpdateFlightDraftRequest.prototype.hierarchyDraftId = "";

                    /**
                     * UpdateFlightDraftRequest flightDraftId.
                     * @member {string} flightDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @instance
                     */
                    UpdateFlightDraftRequest.prototype.flightDraftId = "";

                    /**
                     * UpdateFlightDraftRequest flightDraftFormValues.
                     * @member {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} flightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @instance
                     */
                    UpdateFlightDraftRequest.prototype.flightDraftFormValues = {};


                    /**
                     * Sets adAccountId
                     * @function setAdAccountId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateFlightDraftRequest} UpdateFlightDraftRequest
                     */
                    UpdateFlightDraftRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets hierarchyDraftId
                     * @function setHierarchyDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateFlightDraftRequest} UpdateFlightDraftRequest
                     */
                    UpdateFlightDraftRequest.prototype.setHierarchyDraftId = function(value) {
                        this['hierarchyDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightDraftId
                     * @function setFlightDraftId
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateFlightDraftRequest} UpdateFlightDraftRequest
                     */
                    UpdateFlightDraftRequest.prototype.setFlightDraftId = function(value) {
                        this['flightDraftId'] = value;
                        return this;
                    };

                    /**
                     * Sets flightDraftFormValues
                     * @function setFlightDraftFormValues
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @param {com.spotify.ads.adstudiodrafts.FlightDraftFormValues} value
                     * @instance
                     * @returns {com.spotify.adstudiobff.proto.UpdateFlightDraftRequest} UpdateFlightDraftRequest
                     */
                    UpdateFlightDraftRequest.prototype.setFlightDraftFormValues = function(value) {
                        this['flightDraftFormValues'] = value;
                        return this;
                    };

                    /**
                     * Converts the UpdateFlightDraftRequest to an object
                     * @function toObject
                     * @memberof com.spotify.adstudiobff.proto.UpdateFlightDraftRequest
                     * @instance
                     * @returns {object} object
                     */
                    UpdateFlightDraftRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return UpdateFlightDraftRequest;
                })();
                
                UpdateFlightDraftRequest = proto.UpdateFlightDraftRequest;
                return proto;
            })();

            return adstudiobff;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["UpdateFlightDraftRequest"] = UpdateFlightDraftRequest;
exports["default"] = $root;
