// @spotify-internal/encore-web/es/components/FormCheckbox/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { VisuallyHidden } from "../VisuallyHidden";
import { Checkbox } from "./Checkbox";
import { Indicator } from "./Indicator";
import { Label } from "./Label";
var FormCheckboxComponent = React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      small = _ref.small,
      indeterminate = _ref.indeterminate,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "small", "indeterminate", "children"]);

  return React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Checkbox, null, React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "checkbox",
    ref: ref
  }, props)), React.createElement(Indicator, {
    indeterminate: indeterminate
  }), children && React.createElement(Label, {
    small: small
  }, children)));
});
var formCheckboxHelpers = {
  darkTheme: {
    formCheckbox: 'dark'
  },
  displayName: 'FormCheckbox'
};
export var FormCheckbox = Object.assign(FormCheckboxComponent, formCheckboxHelpers);
export default FormCheckbox;
