// @spotify-internal/encore-web-v3/es/components/FormInput/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { Input } from "./Input";
var FormInputComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      error = _ref.error,
      props = _objectWithoutProperties(_ref, ["theme", "error"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Input, Object.assign({
    ref: ref,
    isUsingKeyboard: isUsingKeyboard,
    error: error,
    "aria-invalid": error
  }, props)));
});
var formInputHelpers = {
  darkTheme: {
    formInput: 'dark'
  },
  displayName: 'FormInput'
};
/**
* **FormInput** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/FormInput) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-forminput--default) | [Encore](https://encore.spotify.net/web/components/form?format=figma-figma#forminput) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:858) |
*
* A FormInput collects user input and edit text or other information, then submits it to a server.
*
* @example
* export const Default = () => <FormInput defaultValue="Default" />;
*
*/

export var FormInput = Object.assign(FormInputComponent, formInputHelpers);
export default FormInput;
