// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LOCATION_UNAVAILABLE_PLURAL = exports.LOCATION_UNAVAILABLE_SINGULAR = exports.CTA_REMOVE_ALL_SINGULAR = exports.CTA_REMOVE_ALL_PLURAL = exports.CTA_LEARN_MORE = exports.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION = exports.ARIA_REMOVE_THIS_LOCATION = exports.LABEL_MESSAGE = exports.BULK_ADD_MESSAGE = exports.DEPRECATED_GEOS_ERROR = exports.COUNTRY_ERROR_MESSAGE = exports.COUNTRY_PLACEHOLDER = exports.COUNTRY_TOOLTIP_DESCRIPTION = exports.COUNTRY_LABEL = exports.LOCATION_TOOLTIP_DESCRIPTION = exports.LOCATION_PLACEHOLDER = exports.LOCATIONS_DESCRIPTION = exports.LOCATION_LABEL = exports.KEY_FIELD = void 0;
var KEY_FIELD = 'geoId';
exports.KEY_FIELD = KEY_FIELD;
var LOCATION_LABEL = 'Location(s)';
exports.LOCATION_LABEL = LOCATION_LABEL;
var LOCATIONS_DESCRIPTION = 'Reach listeners in a specific country, state, city, or DMA.'; // TODO: Update copy

exports.LOCATIONS_DESCRIPTION = LOCATIONS_DESCRIPTION;
var LOCATION_PLACEHOLDER = 'Search for a location. E.g., 90210, Santa Monica, or California';
exports.LOCATION_PLACEHOLDER = LOCATION_PLACEHOLDER;
var LOCATION_TOOLTIP_DESCRIPTION = 'Search by city, state, DMA, or zip/postal code. For more info, visit our FAQ.';
exports.LOCATION_TOOLTIP_DESCRIPTION = LOCATION_TOOLTIP_DESCRIPTION;
var COUNTRY_LABEL = 'Country';
exports.COUNTRY_LABEL = COUNTRY_LABEL;
var COUNTRY_TOOLTIP_DESCRIPTION = 'Choose a country before targeting locations within it. You can only target one country.';
exports.COUNTRY_TOOLTIP_DESCRIPTION = COUNTRY_TOOLTIP_DESCRIPTION;
var COUNTRY_PLACEHOLDER = 'Search for a country. E.g., United States';
exports.COUNTRY_PLACEHOLDER = COUNTRY_PLACEHOLDER;
var COUNTRY_ERROR_MESSAGE = 'Selecting a country is required.';
exports.COUNTRY_ERROR_MESSAGE = COUNTRY_ERROR_MESSAGE;
var DEPRECATED_GEOS_ERROR = 'Please remove any unavailable locations.';
exports.DEPRECATED_GEOS_ERROR = DEPRECATED_GEOS_ERROR;
var BULK_ADD_MESSAGE = 'Bulk add postal codes';
exports.BULK_ADD_MESSAGE = BULK_ADD_MESSAGE;
var LABEL_MESSAGE = 'Target the whole country';
exports.LABEL_MESSAGE = LABEL_MESSAGE;
var ARIA_REMOVE_THIS_LOCATION = 'Remove this location';
exports.ARIA_REMOVE_THIS_LOCATION = ARIA_REMOVE_THIS_LOCATION;
var ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION = 'This location is unavailable. Please remove.';
exports.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION = ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION;
var CTA_LEARN_MORE = 'Learn more';
exports.CTA_LEARN_MORE = CTA_LEARN_MORE;
var CTA_REMOVE_ALL_PLURAL = 'Remove all';
exports.CTA_REMOVE_ALL_PLURAL = CTA_REMOVE_ALL_PLURAL;
var CTA_REMOVE_ALL_SINGULAR = 'Remove';
exports.CTA_REMOVE_ALL_SINGULAR = CTA_REMOVE_ALL_SINGULAR;
var LOCATION_UNAVAILABLE_SINGULAR = '[deprecatedCount] location unavailable.  Remove it to continue.';
exports.LOCATION_UNAVAILABLE_SINGULAR = LOCATION_UNAVAILABLE_SINGULAR;
var LOCATION_UNAVAILABLE_PLURAL = '[deprecatedCount] locations unavailable.  Remove them to continue.';
exports.LOCATION_UNAVAILABLE_PLURAL = LOCATION_UNAVAILABLE_PLURAL;
