// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _api = require("./api");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _CountrySearch = _interopRequireWildcard(require("./CountrySearch"));

var _LocationSearch = _interopRequireWildcard(require("./LocationSearch"));

var _LocationTargetingOptions = _interopRequireWildcard(require("./LocationTargetingOptions"));

var _i18n = require("../../config/i18n");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "LocationTargetingByCountry__Container",
  componentId: "sc-1iuc0t1-0"
})([""]);

var sharedSearchStyles = (0, _styledComponents.css)(["& > div{padding-bottom:", ";}"], _encoreAdvertisingWeb.spacer16);
var StyledCountrySearch = (0, _styledComponents["default"])(_CountrySearch["default"]).withConfig({
  displayName: "LocationTargetingByCountry__StyledCountrySearch",
  componentId: "sc-1iuc0t1-1"
})(["", ";"], sharedSearchStyles);
var StyledLocationSearch = (0, _styledComponents["default"])(_LocationSearch["default"]).withConfig({
  displayName: "LocationTargetingByCountry__StyledLocationSearch",
  componentId: "sc-1iuc0t1-2"
})(["", ";"], sharedSearchStyles);

var LocationTargetingByCountry = /*#__PURE__*/function (_React$Component) {
  _inherits(LocationTargetingByCountry, _React$Component);

  var _super = _createSuper(LocationTargetingByCountry);

  function LocationTargetingByCountry() {
    _classCallCheck(this, LocationTargetingByCountry);

    return _super.apply(this, arguments);
  }

  _createClass(LocationTargetingByCountry, [{
    key: "UNSAFE_componentWillMount",
    value: function UNSAFE_componentWillMount() {
      this.setApiHost(this.props.apiHost);
    }
  }, {
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (nextProps.apiHost !== this.props.apiHost) {
        this.setApiHost(nextProps.apiHost);
      }
    }
  }, {
    key: "setApiHost",
    value: function setApiHost(newApiHost) {
      var hostToUse = newApiHost;

      if (!newApiHost) {
        var isOnProdOrigin = window.location.origin.indexOf('adstudio.spotify.com') >= 0 || window.location.origin.indexOf('adstudio-admin2.spotify.net') >= 0;
        hostToUse = "https://edge-grpc.spotify.com";
        /* eslint-disable no-console */

        if (console && console.warn) {
          console.warn("\n          You must provide the API host for calling the AdTargeting API endpoints.\n          We have guessed that you want to use ".concat(hostToUse, ". Please start providing\n          the apiHost prop to AdTargeting to avoid seeing this message, as this\n          will be deprecated in a future release.\n        "));
        }
        /* eslint-enable no-console */

      }

      (0, _api.setEdgeHost)(hostToUse);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          disableCountry = _this$props.disableCountry,
          disableLocations = _this$props.disableLocations,
          excludeCountries = _this$props.excludeCountries,
          parentCountry = _this$props.parentCountry,
          handleRemoveCountry = _this$props.handleRemoveCountry,
          disableTargetWholeCountry = _this$props.disableTargetWholeCountry,
          disableBulkUploadZip = _this$props.disableBulkUploadZip,
          deprecatedGeosFaqLink = _this$props.deprecatedGeosFaqLink,
          handleDeprecatedGeos = _this$props.handleDeprecatedGeos,
          formName = _this$props.formName,
          language = _this$props.language,
          limit = _this$props.limit,
          strings = _this$props.strings;
      return /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(StyledCountrySearch, {
        disableCountry: disableCountry,
        excludeCountries: excludeCountries,
        handleRemoveCountry: handleRemoveCountry,
        language: language,
        strings: strings.countrySearch
      }), /*#__PURE__*/_react["default"].createElement(StyledLocationSearch, {
        disableLocations: disableLocations,
        parentCountry: parentCountry,
        language: language,
        limit: limit,
        bulkUploaderStrings: strings.bulkUploader,
        strings: strings.locationSearch,
        subdivisionCategories: strings.subdivisionCategories,
        deprecatedGeosFaqLink: deprecatedGeosFaqLink,
        handleDeprecatedGeos: handleDeprecatedGeos
      }), /*#__PURE__*/_react["default"].createElement(_LocationTargetingOptions["default"], {
        disableTargetWholeCountry: disableTargetWholeCountry,
        disableBulkUploadZip: disableBulkUploadZip,
        parentCountry: parentCountry,
        formName: formName,
        limit: limit,
        strings: strings.targetingOptions,
        bulkUploaderStrings: strings.bulkUploader
      }));
    }
  }]);

  return LocationTargetingByCountry;
}(_react["default"].Component);

exports["default"] = LocationTargetingByCountry;
LocationTargetingByCountry.propTypes = {
  apiHost: _propTypes["default"].string,
  disableCountry: _propTypes["default"].bool,
  disableLocations: _propTypes["default"].bool,
  excludeCountries: _propTypes["default"].arrayOf(_propTypes["default"].string),
  disableTargetWholeCountry: _propTypes["default"].bool,
  disableBulkUploadZip: _propTypes["default"].bool,
  formName: _propTypes["default"].string.isRequired,
  deprecatedGeosFaqLink: _propTypes["default"].string,
  handleDeprecatedGeos: _propTypes["default"].bool,
  handleRemoveCountry: _propTypes["default"].func,
  parentCountry: _propTypes["default"].string,
  language: _propTypes["default"].string.isRequired,
  limit: _propTypes["default"].number.isRequired,
  strings: _propTypes["default"].shape({
    bulkUploader: _propTypes["default"].object,
    countrySearch: _propTypes["default"].shape(_CountrySearch.stringsShape),
    locationSearch: _propTypes["default"].shape(_LocationSearch.stringsShape),
    subdivisionCategories: _propTypes["default"].object,
    targetingOptions: _propTypes["default"].shape(_LocationTargetingOptions.stringsShape)
  })
};
LocationTargetingByCountry.defaultProps = {
  excludeCountries: [],
  language: _i18n.DEFAULT_LANGUAGE,
  limit: 1000,
  strings: {
    countrySearch: _CountrySearch.strings,
    locationSearch: _LocationSearch.strings,
    targetingOptions: _LocationTargetingOptions.strings
  }
};
