// @spotify-internal/encore-web/es/components/Snackbar/index.js
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useEffect, useState } from 'react';
import { CSSTransition } from 'react-transition-group';
import { ThemeProvider } from 'styled-components';
import { Container } from "./Container";
import { Text } from "./Text";
export var isDarkTheme = function isDarkTheme(theme) {
  return theme.snackbar === 'dark';
};
export var snackbarTransitionDuration = 300;
export var snackbarRestDuration = 5000;
export var SnackbarComponent = function SnackbarComponent(_ref) {
  var onExited = _ref.onExited,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["onExited", "theme", "children"]);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      showSnackbar = _useState2[0],
      setShowSnackbar = _useState2[1];

  useEffect(function () {
    setShowSnackbar(true);
    var timer = setTimeout(function () {
      return setShowSnackbar(false);
    }, snackbarRestDuration);
    return function () {
      return clearTimeout(timer);
    };
  }, []);
  return /*#__PURE__*/React.createElement(CSSTransition, {
    "in": showSnackbar,
    onExited: onExited,
    timeout: {
      enter: 0,
      exit: snackbarTransitionDuration
    },
    unmountOnExit: true,
    mountOnEnter: true
  }, function (state) {
    return /*#__PURE__*/React.createElement(ThemeProvider, {
      theme: theme
    }, /*#__PURE__*/React.createElement(Container, Object.assign({
      role: "status",
      state: state
    }, props), /*#__PURE__*/React.createElement(Text, null, children)));
  });
};
var snackbarHelpers = {
  darkTheme: {
    snackbar: 'dark'
  }
};
export var Snackbar = Object.assign(SnackbarComponent, snackbarHelpers);
export default Snackbar;
