// @spotify-internal/adstudio-bff-clients/models/com/spotify/ads/campaignservice/ThirdPartyPixel.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var ThirdPartyPixel = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.campaignservice = (function() {
                /**
                 * Namespace campaignservice.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var campaignservice = {};

                campaignservice.ThirdPartyPixel = (function() {
                    /**
                     * Properties of a ThirdPartyPixel.
                     * @memberof com.spotify.ads.campaignservice
                     * @interface IThirdPartyPixel
                     * @property {com.spotify.ads.common.TrackingEventType|null} [trackingEventType] ThirdPartyPixel trackingEventType
                     * @property {string|null} [url] ThirdPartyPixel url
                     * @property {com.spotify.ads.common.TrackingEventType|null} [setTrackingEventType] ThirdPartyPixel setTrackingEventType
                     * @property {string|null} [setUrl] ThirdPartyPixel setUrl
                     */
                    
                    /**
                     * Constructs a new ThirdPartyPixel.
                     * @memberof com.spotify.ads.campaignservice
                     * @classdesc Represents a ThirdPartyPixel.
                     * @implements IThirdPartyPixel
                     * @constructor
                     * @param {com.spotify.ads.campaignservice.IThirdPartyPixel=} [properties] Properties to set
                     */
                    function ThirdPartyPixel(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * ThirdPartyPixel trackingEventType.
                     * @member {com.spotify.ads.common.TrackingEventType} trackingEventType
                     * @memberof com.spotify.ads.campaignservice.ThirdPartyPixel
                     * @instance
                     */
                    ThirdPartyPixel.prototype.trackingEventType = {};

                    /**
                     * ThirdPartyPixel url.
                     * @member {string} url
                     * @memberof com.spotify.ads.campaignservice.ThirdPartyPixel
                     * @instance
                     */
                    ThirdPartyPixel.prototype.url = "";


                    /**
                     * Sets trackingEventType
                     * @function setTrackingEventType
                     * @memberof com.spotify.ads.campaignservice.ThirdPartyPixel
                     * @param {com.spotify.ads.common.TrackingEventType} value
                     * @instance
                     * @returns {com.spotify.ads.campaignservice.ThirdPartyPixel} ThirdPartyPixel
                     */
                    ThirdPartyPixel.prototype.setTrackingEventType = function(value) {
                        this['trackingEventType'] = value;
                        return this;
                    };

                    /**
                     * Sets url
                     * @function setUrl
                     * @memberof com.spotify.ads.campaignservice.ThirdPartyPixel
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.campaignservice.ThirdPartyPixel} ThirdPartyPixel
                     */
                    ThirdPartyPixel.prototype.setUrl = function(value) {
                        this['url'] = value;
                        return this;
                    };

                    /**
                     * Converts the ThirdPartyPixel to an object
                     * @function toObject
                     * @memberof com.spotify.ads.campaignservice.ThirdPartyPixel
                     * @instance
                     * @returns {object} object
                     */
                    ThirdPartyPixel.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return ThirdPartyPixel;
                })();
                
                ThirdPartyPixel = campaignservice.ThirdPartyPixel;
                return campaignservice;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["ThirdPartyPixel"] = ThirdPartyPixel;
exports["default"] = $root;
