// @spotify-internal/encore-web/es/styles/mixins/buttons.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

var _buttonHeightMinimum, _buttonPaddingTopBott, _buttonPaddingLeftRig, _buttonWithIconSidePa, _buttonWithIconMargin, _buttonIconSize;

//
// Buttons
// --------------------------------------------------
import { spacer8, spacer12, spacer16, spacer20, spacer24, spacer32, spacer48, spacer56 } from '@spotify-internal/encore-foundation';
import { css } from 'styled-components';
import { encoreContextStatus } from "../../contexts/EncoreContext";
import { balladBold, browserFocusReset, cello, cursorDisabled, finaleBold, mestoBold, opacityActive, opacityDisabled, rootStyle, violaBold } from "../../styles";
import { pxToInt } from "../../typeHelpers";
export var buttonSizes = {
  sm: 'sm',
  md: 'md',
  lg: 'lg'
};

/** Height of single-line buttons */
export var buttonHeightMinimum = (_buttonHeightMinimum = {}, _defineProperty(_buttonHeightMinimum, buttonSizes.sm, spacer32), _defineProperty(_buttonHeightMinimum, buttonSizes.md, spacer48), _defineProperty(_buttonHeightMinimum, buttonSizes.lg, spacer56), _buttonHeightMinimum);
export var buttonBorderRadius = '500px';
export var buttonTransitionFade = '33ms';
export var buttonTextTransform = 'none';
export var buttonHoverScale = '1.04';
export var buttonPaddingTopBottom = (_buttonPaddingTopBott = {}, _defineProperty(_buttonPaddingTopBott, buttonSizes.sm, spacer8), _defineProperty(_buttonPaddingTopBott, buttonSizes.md, '14px'), _defineProperty(_buttonPaddingTopBott, buttonSizes.lg, spacer16), _buttonPaddingTopBott);
/**
 * Left and right padding for buttons without icons.
 * For buttons with `iconLeading` or `iconTrailing`, see `getButtonWithIconLeftRightPadding()`.
 */

export var buttonPaddingLeftRight = (_buttonPaddingLeftRig = {}, _defineProperty(_buttonPaddingLeftRig, buttonSizes.sm, spacer16), _defineProperty(_buttonPaddingLeftRig, buttonSizes.md, spacer32), _defineProperty(_buttonPaddingLeftRig, buttonSizes.lg, spacer48), _buttonPaddingLeftRig);
/**
 * Observed left &/or right padding for buttons with `iconLeading` or `iconTrailing`.
 * Note: "Observed" because the true padding is calculated, see `getButtonWithIconLeftRightPadding()`.
 */

export var buttonWithIconSidePadding = (_buttonWithIconSidePa = {}, _defineProperty(_buttonWithIconSidePa, buttonSizes.sm, spacer12), _defineProperty(_buttonWithIconSidePa, buttonSizes.md, spacer20), _defineProperty(_buttonWithIconSidePa, buttonSizes.lg, spacer24), _buttonWithIconSidePa);
/** Observed margin between icons and text (absolute positioned, not real margin) */

export var buttonWithIconMargin = (_buttonWithIconMargin = {}, _defineProperty(_buttonWithIconMargin, buttonSizes.sm, '5px'), _defineProperty(_buttonWithIconMargin, buttonSizes.md, spacer12), _defineProperty(_buttonWithIconMargin, buttonSizes.lg, '10px'), _buttonWithIconMargin);
/**
 * Standard icon size to use with each button size.
 * Note: These standard sizes aren't always used. See `getButtonIconSize()`.
 * */

export var buttonIconSize = (_buttonIconSize = {}, _defineProperty(_buttonIconSize, buttonSizes.sm, spacer16), _defineProperty(_buttonIconSize, buttonSizes.md, spacer24), _defineProperty(_buttonIconSize, buttonSizes.lg, spacer24), _buttonIconSize);
export var defaultTypeVariants = {
  sm: 'finaleBold',
  md: 'violaBold',
  lg: 'balladBold'
};
export var nextTypeVariants = {
  sm: 'mestoBold',
  md: 'balladBold',
  lg: 'cello'
};

var getNextButtonFont = function getNextButtonFont(buttonSize) {
  switch (buttonSize) {
    case buttonSizes.sm:
      return mestoBold();

    case buttonSizes.md:
      return balladBold();

    case buttonSizes.lg:
      return cello();

    default:
      return balladBold();
  }
};

export var getButtonFont = function getButtonFont(buttonSize, status) {
  if (status === encoreContextStatus.next) {
    return getNextButtonFont(buttonSize);
  }

  switch (buttonSize) {
    case buttonSizes.sm:
      return finaleBold(true);

    case buttonSizes.md:
      return violaBold(true);

    case buttonSizes.lg:
      return balladBold(true);

    default:
      return violaBold(true);
  }
};
/**
 * Calculates side padding for buttons with `iconLeading` &/or `iconTrailing`.
 * Accounts for the icon's size, padding on the button edge, and margin between icons and text.
 * Used in ButtonPrimary and ButtonSecondary.
 */

var getButtonWithIconLeftRightPadding = function getButtonWithIconLeftRightPadding(buttonSize) {
  var padding = buttonWithIconSidePadding[buttonSize];
  var margin = buttonWithIconMargin[buttonSize];
  var iconSize = buttonIconSize[buttonSize];
  return pxToInt(padding) + pxToInt(iconSize) + pxToInt(margin);
};

export var getButtonPaddingValues = function getButtonPaddingValues(buttonSize) {
  var borderWidth = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 0;
  var paddingTopBottom = "".concat(pxToInt(buttonPaddingTopBottom[buttonSize]) - borderWidth, "px");
  var paddingLeftRight = "".concat(pxToInt(buttonPaddingLeftRight[buttonSize]) - borderWidth, "px");
  var withIconPaddingLeftRight = "".concat(getButtonWithIconLeftRightPadding(buttonSize) - borderWidth, "px");
  return {
    paddingTopBottom: paddingTopBottom,
    paddingLeftRight: paddingLeftRight,
    withIconPaddingLeftRight: withIconPaddingLeftRight
  };
};
export var getButtonPadding = function getButtonPadding(buttonSize, iconLeading, iconTrailing, iconOnly, borderWidth) {
  var _getButtonPaddingValu = getButtonPaddingValues(buttonSize, borderWidth),
      paddingTopBottom = _getButtonPaddingValu.paddingTopBottom,
      paddingLeftRight = _getButtonPaddingValu.paddingLeftRight,
      withIconPaddingLeftRight = _getButtonPaddingValu.withIconPaddingLeftRight;
  /** iconOnly buttons used fixed width & height instead of padding */


  var iconOnlyDimensions = css(["width:", ";height:", ";"], buttonHeightMinimum[buttonSize], buttonHeightMinimum[buttonSize]);
  var noIconPadding = css(["padding:", " ", ";"], paddingTopBottom, paddingLeftRight);
  var symmetricalIconPadding = css(["padding:", " ", ";"], paddingTopBottom, withIconPaddingLeftRight);
  var asymmetricalIconPadding = css(["padding-top:", ";padding-bottom:", ";padding-left:", ";padding-right:", ";[dir='rtl'] &{padding-right:", ";padding-left:", ";}"], paddingTopBottom, paddingTopBottom, iconLeading ? withIconPaddingLeftRight : paddingLeftRight, iconTrailing ? withIconPaddingLeftRight : paddingLeftRight, iconLeading ? withIconPaddingLeftRight : paddingLeftRight, iconTrailing ? withIconPaddingLeftRight : paddingLeftRight);

  if (iconOnly) {
    return iconOnlyDimensions;
  }

  if (!iconLeading && !iconTrailing) {
    return noIconPadding;
  }

  if (iconLeading && iconTrailing) {
    return symmetricalIconPadding;
  }

  return asymmetricalIconPadding;
};
/** Base styles shared by most buttons, including root style, focus reset and font sizes */

export var buttonStylesBase = function buttonStylesBase(buttonSize, useBrowserDefaultFocusStyle, status) {
  return css(["", ";", " ", ";background-color:transparent;border:0;border-radius:", ";display:inline-block;position:relative;text-align:center;text-decoration:none;text-transform:", ";touch-action:manipulation;transition-duration:", ";transition-property:background-color,border-color,color,box-shadow,filter,transform;user-select:none;vertical-align:middle;transform:translate3d(0,0,0);"], rootStyle(), !useBrowserDefaultFocusStyle && browserFocusReset, getButtonFont(buttonSize, status), buttonBorderRadius, buttonTextTransform, buttonTransitionFade);
};
/** Disabled styles shared by most buttons */

export var buttonStylesDisabled = css(["cursor:", ";opacity:", ";transform:scale(1);"], cursorDisabled, opacityDisabled);
/** Active styles shares by most buttons */

export var buttonStylesActive = css(["opacity:", ";outline:none;transform:scale(1);"], opacityActive);
/** Hover styles shares by most buttons */

export var buttonStylesHover = css(["transform:scale(", ");"], buttonHoverScale); // Reset a button to a link looking thing

export var buttonLinkReset = function buttonLinkReset() {
  var useBrowserDefaultFocusStyle = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : false;
  return css(["background-color:transparent;border:0;color:inherit;line-height:1;letter-spacing:inherit;padding:0;", ""], !useBrowserDefaultFocusStyle && browserFocusReset);
};
