// @spotify-internal/encore-web/es/styles/mixins/forms.js
/* stylelint-disable no-descending-specificity */
//
// Forms
// --------------------------------------------------
import { spacer4, spacer16, spacer32 } from '@spotify-internal/encore-foundation';
import { css } from 'styled-components';
import { useEncoreContextType } from "../../hooks/useEncoreContextType";
import { cssColorValue, semanticColors } from "../semantic-theme";
import { opacityDisabled } from "../variables";
import { rootStyle } from "./baseline";
import { ballad, typeStyleBody1 } from "./type"; // Shared styles for inputs, textareas, selects

export var formControlBase = function formControlBase() {
  return css(["-webkit-appearance:none;background-image:none;border:0;display:block;", ";transition:box-shadow ease-in-out 0.1s,color fade 0.1s;width:100%;&::placeholder{opacity:1;}&:focus,&:hover:focus{outline:0;}&:disabled,&[readonly]{cursor:not-allowed;opacity:1;}"], function () {
    return useEncoreContextType(typeStyleBody1(), ballad());
  });
}; // Styles for selects only

export var formControlBaseSelect = function formControlBaseSelect() {
  return css(["", ";margin-top:0;margin-bottom:0;appearance:none;box-shadow:none;text-indent:0.01px;text-overflow:'';&::-ms-expand{display:none;}border-radius:", ";padding:14px 44px 14px 14px;"], rootStyle(), spacer4);
}; // Styles the down arrow icon on FormSelect and Dropdown

export var formChevronDown = function formChevronDown() {
  return css(["color:", ";pointer-events:none;position:absolute;right:14px;"], cssColorValue(semanticColors.essentialSubdued));
}; // Shared styles for radios + checkboxes

export var formCheck = function formCheck() {
  return css(["", ";align-items:center;display:flex;padding-bottom:", ";padding-top:", ";position:relative;min-height:", ";"], rootStyle(), spacer4, spacer4, spacer32);
};
export var formCheckIndicator = function formCheckIndicator(radius) {
  return css(["", ";background:transparent;border-radius:", ";display:inline-block;height:", ";position:relative;user-select:none;width:", ";flex-shrink:0;align-self:flex-start;top:", ";"], rootStyle(), radius, spacer16, spacer16, spacer4);
};
export var formControlTheme = function formControlTheme(_ref) {
  var _ref$isUsingKeyboard = _ref.isUsingKeyboard,
      isUsingKeyboard = _ref$isUsingKeyboard === void 0 ? true : _ref$isUsingKeyboard,
      isHover = _ref.isHover,
      isFocus = _ref.isFocus,
      error = _ref.error;
  var hoverStyles = css(["box-shadow:inset 0 0 0 1px ", ";"], cssColorValue(semanticColors.essentialBase));
  var focusStyles = css(["box-shadow:inset 0 0 0 ", "px ", ";"], isUsingKeyboard ? 3 : 1.5, cssColorValue(semanticColors.essentialBase));
  var errorStyles = css(["&,&:hover{box-shadow:inset 0 0 0 1px ", ";}&:focus{box-shadow:inset 0 0 0 ", "px ", ";}"], cssColorValue(semanticColors.essentialNegative), isUsingKeyboard ? 3 : 1.5, cssColorValue(semanticColors.essentialNegative));
  return css(["background-color:", ";box-shadow:inset 0 0 0 1px ", ";color:", ";&:hover{", "}", " &:focus{", "}&&{", ";}&::placeholder{color:", ";}&:disabled,&[readonly]{box-shadow:inset 0 0 0 1px ", ";opacity:", ";}&:invalid{", ";}", ""], cssColorValue(semanticColors.backgroundBase), cssColorValue(semanticColors.essentialSubdued), cssColorValue(semanticColors.textBase), hoverStyles, isHover && hoverStyles, focusStyles, isFocus && focusStyles, cssColorValue(semanticColors.textSubdued), cssColorValue(semanticColors.essentialSubdued), opacityDisabled, error === undefined && errorStyles, error && errorStyles);
};
