// @spotify-internal/adstudio-tape/lib/components/FormGroup/StyledType.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SubDescription = exports.Span = exports.Description = exports.OptionalLabel = exports.Label = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _Type = _interopRequireDefault(require("../Type"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Label = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "StyledType__Label",
  componentId: "rxwws7-0"
})(["display:block;padding-bottom:", ";color:", ";"], _encoreFoundation.spacer4, function (props) {
  if (props.disabled) return _encoreFoundation.gray30;else if (props.theme === 'dark') return _encoreFoundation.gray95;
  return 'initial';
});
exports.Label = Label;
var OptionalLabel = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "StyledType__OptionalLabel",
  componentId: "rxwws7-1"
})(["display:inline-block;margin-left:", ";padding-bottom:0;color:", ";"], _encoreFoundation.spacer8, function (props) {
  return props.theme === 'dark' ? _encoreFoundation.gray95 : _encoreFoundation.gray30;
});
exports.OptionalLabel = OptionalLabel;
var Description = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "StyledType__Description",
  componentId: "rxwws7-2"
})(["padding-bottom:", ";color:", ";"], _encoreFoundation.spacer4, _encoreFoundation.gray30);
exports.Description = Description;
var Span = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "StyledType__Span",
  componentId: "rxwws7-3"
})(["", ""], function (props) {
  return props.inline && (0, _styledComponents.css)(["display:flex;flex-wrap:wrap;"]);
});
exports.Span = Span;
var SubDescription = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "StyledType__SubDescription",
  componentId: "rxwws7-4"
})(["padding-top:", ";color:", ";"], _encoreFoundation.spacer4, function (props) {
  if (props.error) return _encoreFoundation.failure;else if (props.warning) return _encoreFoundation.warning;else if (props.subDescription) return _encoreFoundation.gray30;
  return 'initial';
});
exports.SubDescription = SubDescription;
