// @spotify-internal/adstudio-tape/lib/components/UploaderProgress/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = UploaderProgress;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _Panel = _interopRequireDefault(require("../Panel"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "UploaderProgress__Container",
  componentId: "x9g0j9-0"
})(["width:100%;"]);

var TrackName = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "UploaderProgress__TrackName",
  componentId: "x9g0j9-1"
})(["padding-bottom:10px;position:relative;"]);

var ProgressBarBackground = _styledComponents["default"].div.withConfig({
  displayName: "UploaderProgress__ProgressBarBackground",
  componentId: "x9g0j9-2"
})(["background-color:", ";border-radius:2px;height:4px;width:100%;"], _encoreFoundation.gray90);

var ProgressBar = _styledComponents["default"].div.withConfig({
  displayName: "UploaderProgress__ProgressBar",
  componentId: "x9g0j9-3"
})(["background-color:", ";border-radius:2px;height:4px;width:0;"], _encoreFoundation.gray60);

var StatusText = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "UploaderProgress__StatusText",
  componentId: "x9g0j9-4"
})(["padding-top:10px;"]);

function UploaderProgress(_ref) {
  var name = _ref.name,
      progress = _ref.progress;
  var progressInt = Math.max(Math.min(parseInt(progress, 10), 100), 0);
  var statusText = progressInt === 100 ? 'Processing...' : 'Uploading...';
  return /*#__PURE__*/_react["default"].createElement(_Panel["default"], {
    shadow: true
  }, /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(TrackName, {
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book
  }, name)), /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(ProgressBarBackground, null, /*#__PURE__*/_react["default"].createElement(ProgressBar, {
    style: {
      width: "".concat(progressInt, "%")
    }
  }))), /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(StatusText, {
    color: _encoreFoundation.gray75,
    variant: _encoreWeb.Type.body3,
    weight: _encoreWeb.Type.book,
    condensed: true
  }, statusText))));
}

UploaderProgress.propTypes = {
  name: _propTypes["default"].string,
  progress: _propTypes["default"].number
};
