// @spotify-internal/adstudio-tape/lib/config/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ASPECT_RATIOS = exports.CREATIVE_FORMAT = exports.CLIENT_AUDIO_TYPES = exports.ACCEPTED_VIDEO_FILE_TYPES = exports.ACCEPTED_IMAGE_FILE_TYPES = exports.ACCEPTED_AUDIO_FILE_TYPES = void 0;
var ACCEPTED_AUDIO_FILE_TYPES = {
  'audio/mp3': 'MP3',
  'audio/ogg': 'OGG',
  'audio/wav': 'WAV'
};
exports.ACCEPTED_AUDIO_FILE_TYPES = ACCEPTED_AUDIO_FILE_TYPES;
var ACCEPTED_IMAGE_FILE_TYPES = {
  'image/jpeg': 'JPEG',
  'image/png': 'PNG'
};
exports.ACCEPTED_IMAGE_FILE_TYPES = ACCEPTED_IMAGE_FILE_TYPES;
var ACCEPTED_VIDEO_FILE_TYPES = {
  'video/mp4': 'MP4',
  'video/quicktime': 'MOV'
};
exports.ACCEPTED_VIDEO_FILE_TYPES = ACCEPTED_VIDEO_FILE_TYPES;
var CLIENT_AUDIO_TYPES = {
  'audio/ogg': 'OGG',
  // not supported by safari or firefox
  'audio/mp3': 'MP3',
  'audio/mpeg': 'MP3',
  // supported by safari
  'audio/wav': 'WAV',
  // supported by chrome
  'audio/x-wav': 'WAV' // supported by firefox, safari

};
exports.CLIENT_AUDIO_TYPES = CLIENT_AUDIO_TYPES;
var AUDIO_FORMAT = 'AUDIO';
var VIDEO_PORTRAIT_FORMAT = 'PORTRAIT';
var VIDEO_LANDSCAPE_FORMAT = 'LANDSCAPE';
var CREATIVE_FORMAT = {
  AUDIO_FORMAT: AUDIO_FORMAT,
  VIDEO_PORTRAIT_FORMAT: VIDEO_PORTRAIT_FORMAT,
  VIDEO_LANDSCAPE_FORMAT: VIDEO_LANDSCAPE_FORMAT
};
exports.CREATIVE_FORMAT = CREATIVE_FORMAT;
var ASPECT_RATIOS = {
  LANDSCAPE: 1.778,
  PORTRAIT: 0.5625,
  UNKNOWN: 1
};
exports.ASPECT_RATIOS = ASPECT_RATIOS;
