// @spotify-internal/encore-web/cjs/components/ButtonTertiary/ButtonLegacy.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonLegacyComponent = exports.ButtonLegacy = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _BrowserDefaultFocusStyleContext = require("../../contexts/BrowserDefaultFocusStyleContext");

var _styles = require("../../styles");

var _excluded = ["component", "semanticColor", "condensed", "buttonSize", "hover", "active", "focus", "isUsingKeyboard", "useBrowserDefaultFocusStyle", "iconOnly", "iconLeading", "iconTrailing"],
    _excluded2 = ["component"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/**
 * Button Legacy
 * TODO: Archive in v6
 * */
var ButtonLegacy = (0, _styledComponents["default"])( /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var _ref$component = _ref.component,
      Component = _ref$component === void 0 ? 'button' : _ref$component,
      semanticColor = _ref.semanticColor,
      condensed = _ref.condensed,
      buttonSize = _ref.buttonSize,
      hover = _ref.hover,
      active = _ref.active,
      focus = _ref.focus,
      isUsingKeyboard = _ref.isUsingKeyboard,
      useBrowserDefaultFocusStyle = _ref.useBrowserDefaultFocusStyle,
      iconOnly = _ref.iconOnly,
      iconLeading = _ref.iconLeading,
      iconTrailing = _ref.iconTrailing,
      props = (0, _objectWithoutProperties2["default"])(_ref, _excluded);

  if (iconOnly || iconLeading || iconTrailing) {
    // eslint-disable-next-line no-console
    console.warn("ButtonTertiary's `buttonLegacy` feature does not support `iconOnly`, `iconLeading`, or `iconTrailing`");
  }

  return /*#__PURE__*/_react["default"].createElement(Component, Object.assign({}, props, {
    ref: ref
  }));
})).withConfig({
  displayName: "ButtonLegacy",
  componentId: "sc-1oh201n-0"
})(["", ""], function (_ref2) {
  var _ref2$buttonSize = _ref2.buttonSize,
      buttonSize = _ref2$buttonSize === void 0 ? 'md' : _ref2$buttonSize,
      _ref2$semanticColor = _ref2.semanticColor,
      semanticColor = _ref2$semanticColor === void 0 ? 'textSubdued' : _ref2$semanticColor,
      hover = _ref2.hover,
      active = _ref2.active,
      focus = _ref2.focus,
      useBrowserDefaultFocusStyle = _ref2.useBrowserDefaultFocusStyle,
      isUsingKeyboard = _ref2.isUsingKeyboard,
      condensed = _ref2.condensed;
  var hoverStyles = (0, _styledComponents.css)(["", " ", ""], _styles.buttonStylesHover, semanticColor === 'textSubdued' && (0, _styledComponents.css)(["color:", ";"], (0, _styles.cssColorValue)(_styles.semanticColors.textBase)));
  var activeStyles = (0, _styledComponents.css)(["", " color:", ";"], _styles.buttonStylesActive, (0, _styles.cssColorValue)(_styles.semanticColors.textSubdued));
  var disabledStyles = (0, _styledComponents.css)(["", " color:", ";&::after{border-color:transparent;}"], _styles.buttonStylesDisabled, (0, _styles.cssColorValue)(_styles.semanticColors.textSubdued));

  var _getButtonPaddingValu = (0, _styles.getButtonPaddingValuesLegacy)(buttonSize),
      paddingTopBottom = _getButtonPaddingValu.paddingTopBottom,
      paddingLeftRight = _getButtonPaddingValu.paddingLeftRight;

  var focusBorderBase = paddingTopBottom + _styles.focusBorderWidth;
  return (0, _styledComponents.css)(["", " color:", ";padding:", "px ", "px;", " &::after{width:", ";}&:hover{", "}", " &:active{", "}", " &[disabled],&[aria-disabled='true'],fieldset[disabled] &{", "}"], (0, _styles.buttonStylesBaseLegacy)(buttonSize, useBrowserDefaultFocusStyle), (0, _styles.cssColorValue)(semanticColor), paddingTopBottom, condensed ? 0 : paddingLeftRight, !useBrowserDefaultFocusStyle && (0, _styles.focusBottomBorder)(focusBorderBase, isUsingKeyboard, focus), "calc(100% - ".concat(condensed ? 0 : paddingLeftRight * 2, "px)"), hoverStyles, hover && hoverStyles, activeStyles, active && activeStyles, disabledStyles);
});
exports.ButtonLegacy = ButtonLegacy;

var ButtonLegacyComponent = /*#__PURE__*/_react["default"].forwardRef(function (_ref3, ref) {
  var component = _ref3.component,
      props = (0, _objectWithoutProperties2["default"])(_ref3, _excluded2);
  var rel = props.href && props.target === '_blank' ? 'noopener noreferrer' : null;

  var _useContext = (0, _react.useContext)(_BrowserDefaultFocusStyleContext.BrowserDefaultFocusStyleContext),
      useBrowserDefaultFocusStyle = _useContext.useBrowserDefaultFocusStyle;

  return /*#__PURE__*/_react["default"].createElement(ButtonLegacy, Object.assign({}, props, {
    rel: rel,
    ref: ref,
    component: !component && props.href ? 'a' : component,
    useBrowserDefaultFocusStyle: useBrowserDefaultFocusStyle
  }));
});

exports.ButtonLegacyComponent = ButtonLegacyComponent;
