// @spotify-internal/spoticharts/cjs/components/DotMarker/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const strokeWidth = 2;
function DotMarker({ xPos, yPos, diameter, fill, stroke }) {
    const radius = Math.round((diameter - strokeWidth) / 2);
    const filterName = `dot-marker-filter-${diameter}-${fill.substr(1)}`;
    return (react_1.default.createElement("g", null,
        react_1.default.createElement("defs", null,
            react_1.default.createElement("filter", { id: filterName, y: "-50%", x: "-50%", width: "250%", height: "250%" },
                react_1.default.createElement("feColorMatrix", { in: "SourceAlpha", result: "lightened", values: "0 0 1 0 0  0 0 1 0 0  0 0 1 0 0  0 0 0 .5 0" }),
                react_1.default.createElement("feOffset", { in: "lightened", result: "shadowOffset", dy: "2" }),
                react_1.default.createElement("feGaussianBlur", { in: "shadowOffset", result: "shadowBlur", stdDeviation: "2" }),
                react_1.default.createElement("feBlend", { in: "SourceGraphic", in2: "shadowBlur", mode: "normal" }))),
        react_1.default.createElement("circle", { filter: `url(#${filterName})`, r: radius, cx: xPos, cy: yPos, stroke: stroke, strokeWidth: strokeWidth, fill: fill })));
}
exports.default = DotMarker;
DotMarker.propTypes = {
    xPos: prop_types_1.default.number,
    yPos: prop_types_1.default.number,
    diameter: prop_types_1.default.number,
    fill: prop_types_1.default.string.isRequired,
    stroke: prop_types_1.default.string,
};
DotMarker.defaultProps = {
    xPos: 0,
    yPos: 0,
    diameter: 12,
    stroke: encore_foundation_1.white,
};
