// @spotify-internal/adstudio-tape/lib/components/NavigationListItem/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = NavigationListItem;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _tokens = require("../../styles/tokens");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var propTypes = {
  /** Sets an `active` CSS class */
  active: _propTypes["default"].bool,
  className: _propTypes["default"].string,

  /** Pass a custom component instead of the default `href` */
  component: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].string]),

  /** Disabled the link entirely */
  disabled: _propTypes["default"].bool,
  flush: _propTypes["default"].bool,
  href: _propTypes["default"].string,
  linkActiveClass: _propTypes["default"].string,
  linkClass: _propTypes["default"].string,
  linkDisabledClass: _propTypes["default"].string,

  /** A title list item */
  title: _propTypes["default"].bool,
  to: _propTypes["default"].string
};
var navigationItemTitle = (0, _styledComponents.css)(["color:", ";font-size:", ";font-weight:", ";letter-spacing:0.2em;margin-bottom:", ";margin-left:", ";text-transform:uppercase;&:not(:first-child){margin-top:", ";}"], _encoreFoundation.gray60, _tokens.fontSizeSmall, _tokens.fontWeightBook, _encoreFoundation.spacer4, _tokens.sidebarBasePadding, _tokens.sidebarBasePadding);
var flushStyles = (0, _styledComponents.css)(["margin:0;padding:0;"]);

var NavigationItem = _styledComponents["default"].dd.withConfig({
  displayName: "NavigationListItem__NavigationItem",
  componentId: "sc-1sj2a02-0"
})(["margin-left:0;", ";", ""], function (props) {
  return props.title ? navigationItemTitle : null;
}, function (props) {
  return props.flush ? flushStyles : null;
});

var NavigationLink = _styledComponents["default"].a.withConfig({
  displayName: "NavigationListItem__NavigationLink",
  componentId: "sc-1sj2a02-1"
})(["color:", ";display:block;font-size:", ";font-weight:", ";letter-spacing:0.014em;line-height:1;overflow:hidden;padding:", " ", ";position:relative;text-overflow:ellipsis;transition:color 0.1s;white-space:nowrap;&:focus,&:hover{color:", ";outline:none;}&:active{color:", ";}", ";", ";", ";"], _encoreFoundation.gray60, _tokens.fontSizeSmall, _tokens.fontWeightBold, _encoreFoundation.spacer8, _tokens.sidebarBasePadding, _encoreFoundation.white, _encoreFoundation.white, function (props) {
  return props.flush ? flushStyles : null;
}, function (props) {
  return props.disabled ? (0, _styledComponents.css)(["&,&:focus,&:active,&:hover{opacity:0.4;pointer-events:none;}&::after{border-color:transparent;}"]) : null;
}, function (props) {
  return props.active ? (0, _styledComponents.css)(["&,&:hover{color:", ";}&::after{border-left:4px solid ", ";content:'';height:18px;left:0;position:absolute;top:6px;}"], _encoreFoundation.white, props.theme.colors.primaryColor) : null;
});

function NavigationListItem(_ref) {
  var active = _ref.active,
      className = _ref.className,
      disabled = _ref.disabled,
      title = _ref.title,
      flush = _ref.flush,
      href = _ref.href,
      component = _ref.component,
      to = _ref.to,
      props = _objectWithoutProperties(_ref, ["active", "className", "disabled", "title", "flush", "href", "component", "to"]);

  var LinkElement = NavigationLink;
  var navigationElement = 'dd';

  if (title) {
    navigationElement = 'dt';
    LinkElement = 'span';
  }

  if (to) {
    if (component) {
      props.to = to;
    } else {
      props.href = to;
    }
  } else if (href) {
    props.href = href;
  }

  return /*#__PURE__*/_react["default"].createElement(NavigationItem, {
    className: className,
    flush: flush,
    title: title,
    as: navigationElement
  }, /*#__PURE__*/_react["default"].createElement(LinkElement, _extends({}, props, {
    flush: flush,
    disabled: disabled,
    active: active,
    as: component,
    "aria-current": active && 'page',
    rel: !title ? 'noopener noreferrer' : undefined
  })));
}

NavigationListItem.propTypes = propTypes;
NavigationListItem.NavigationItem = NavigationItem;
NavigationListItem.NavigationLink = NavigationLink;
