// @spotify-internal/encore-advertising-web/esm/utils/helpers/styleOverrides/border.js
import { createParser } from './createParser';
export const borderConfig = {
    border: ['border'],
    borderWidth: ['borderWidth'],
    borderStyle: ['borderStyle'],
    borderColor: ['borderColor'],
    borderRadius: ['borderRadius'],
    borderTop: ['borderTop'],
    borderTopLeftRadius: ['borderTopLeftRadius'],
    borderTopRightRadius: ['borderTopRightRadius'],
    borderRight: ['borderRight'],
    borderBottom: ['borderBottom'],
    borderBottomLeftRadius: ['borderBottomLeftRadius'],
    borderBottomRightRadius: ['borderBottomRightRadius'],
    borderLeft: ['borderLeft'],
    borderX: ['borderLeft', 'borderRight'],
    borderY: ['borderTop', 'borderBottom'],
    borderTopWidth: ['borderTopWidth'],
    borderTopColor: ['borderTopColor'],
    borderTopStyle: ['borderTopStyle'],
    borderBottomWidth: ['borderBottomWidth'],
    borderBottomColor: ['borderBottomColor'],
    borderBottomStyle: ['borderBottomStyle'],
    borderLeftWidth: ['borderLeftWidth'],
    borderLeftColor: ['borderLeftColor'],
    borderLeftStyle: ['borderLeftStyle'],
    borderRightWidth: ['borderRightWidth'],
    borderRightColor: ['borderRightColor'],
    borderRightStyle: ['borderRightStyle'],
};
export const borderOverrides = createParser(borderConfig);
