// @spotify-internal/encore-web/es/components/Popover/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["colorSet", "popoverTitle", "children", "onClose", "className", "theme"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React from 'react';
import { renamedPropWarning } from "../../deprecationWarnings";
import { addColorSet, overlayDirections } from "../../styles";
import { IconX } from "../Icon/IconX";
import { CloseButton } from "./CloseButton";
import { Popover as PopoverElement } from "./Popover";
import { Title } from "./Title";

function PopoverComponent(_ref) {
  var _ref$colorSet = _ref.colorSet,
      colorSet = _ref$colorSet === void 0 ? 'base' : _ref$colorSet,
      popoverTitle = _ref.popoverTitle,
      children = _ref.children,
      onClose = _ref.onClose,
      className = _ref.className,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    renamedPropWarning('theme', 'colorSet');
  }

  return /*#__PURE__*/React.createElement(PopoverElement, Object.assign({
    role: "tooltip",
    paddedBody: onClose && !popoverTitle,
    className: addColorSet(colorSet, className),
    colorSet: colorSet
  }, props), popoverTitle && /*#__PURE__*/React.createElement(Title, null, popoverTitle), children, onClose && /*#__PURE__*/React.createElement(CloseButton, {
    onClick: onClose,
    "aria-label": "Close",
    type: "button",
    iconOnly: IconX
  }));
}

var popoverHelpers = _objectSpread({}, overlayDirections);
/**
* **Popover** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/Popover) | [Storybook](https://encore-web.spotify.net/?path=/story/components-popover--default) | [Encore](https://encore.spotify.net/web/components/popover?format=figma-figma#popover) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=522:141) |
*
* A modal used to display some contextual information on top of other content when users click on, keypress on, or touch an element.
*
* @example
* () => (
*  <Popover arrow={Popover.topLeft} popoverTitle='Popover "topLeft"' onClose={() => {}}>
*    This is a popover with text and its arrow is aligned to the topLeft
*  </Popover>
*);
*
*/


export var Popover = Object.assign(PopoverComponent, popoverHelpers);
export default Popover;
