// @spotify-internal/encore-web/es/components/Popover/Popover.js
import { spacer16, spacer20, spacer64 } from '@spotify-internal/encore-foundation';
import styled from 'styled-components';
import { useEncoreContextType } from "../../hooks/useEncoreContextType";
import { overlayArrow, overlayColorTheme, rootStyle, typeStyleBody2, viola } from "../../styles";
import Overlay from "../OverlayTrigger/Overlay";

var overlayWidth = function overlayWidth(props) {
  return props.large ? '600px' : '296px';
};
/* If there is no title and a close button,
add extra padding to body text to avoid overlap */


var padding = function padding(props) {
  return props.paddedBody ? "".concat(spacer16, " ").concat(spacer64, " ").concat(spacer16, " ").concat(spacer20) : "".concat(spacer16, " ").concat(spacer20);
};

export var Popover = styled.div.withConfig({
  displayName: "Popover",
  componentId: "sc-1iog353-0"
})(["", ";", ";", ";", ";padding:", ";text-transform:initial;", " &{pointer-events:all;}"], function (props) {
  return overlayColorTheme({
    maxWidth: overlayWidth(props),
    colorSet: props.colorSet
  });
}, function (props) {
  return props.arrow && overlayArrow(props.arrow);
}, rootStyle(), function () {
  return useEncoreContextType(typeStyleBody2(), viola());
}, padding, Overlay);
