// @spotify-internal/encore-web/es/components/PaginationDropdown/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Arrow } from "./Arrow";
import { Button } from "./Button";
import { Container } from "./Container";

function PaginationDropdownComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Container, null, /*#__PURE__*/React.createElement(Button, props), /*#__PURE__*/React.createElement(Arrow, null)));
}

var paginationDropdownHelpers = {
  darkTheme: {
    paginationDropdown: 'dark'
  }
};
/**
* **PaginationDropdown** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PaginationDropdown) | [Storybook](https://encore-web.spotify.net/?path=/story/components-pagination-paginationdropdown--default) | [Encore](https://encore.spotify.net/web/components/pagination?format=figma-figma#paginationdropdown) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=2311:140) |
*
* Use the PaginationDropdown component when you need a user to navigate between pages of data, by selecting a pagination number from a list.
*
* @example
* export const Default = () => <PaginationDropdown>10</PaginationDropdown>;
*
*/

export var PaginationDropdown = Object.assign(PaginationDropdownComponent, paginationDropdownHelpers);
export default PaginationDropdown;
