// @spotify-internal/creator-tape/es/styles/variables.js
import { spacer24, spacer64, spotifyGreenHighlight, spotifyGreen, black, gray7 } from '@spotify-internal/tokens/creator/web/tokens.common';
export * from '@spotify-internal/tokens/creator/web/tokens.common'; // Typography

export var fontSizeBody1 = '16px';
export var fontSizeBody2 = '14px';
export var fontSizeBody3 = '12px';
export var fontSizeBody4 = '10px';
export var fontWeightBook = 400;
export var fontWeightBold = 700;
export var fontWeightBlack = 900; // z-index master list

export var zIndexDropdown = 1000;
export var zIndexSticky = 1020;
export var zIndexFixed = 1030;
export var zIndexDialogBackdrop = 1040;
export var zIndexDialog = 1050;
export var zIndexPopover = 1060;
export var zIndexTooltip = 1070; // Links

export var linkColor = black;
export var linkDecoration = 'underline';
export var linkHoverColor = spotifyGreenHighlight;
export var linkHoverDecoration = 'underline';
export var linkActiveColor = spotifyGreen; // Shared Nav Styles

export var navBg = gray7;
export var navFontSize = fontSizeBody1; // Sidebar

export var sidebarBaseWidth = '200px';
export var sidebarSlimBaseWidth = spacer64;
export var sidebarPadding = spacer24; // Transitions + Animations

export var transitionFade = '0.1s';
export var transitionHeight = '0.2s'; // Misc

export var cursorDisabled = 'not-allowed'; // Disabled cursor for form controls and buttons.
