// @spotify-internal/encore-web/cjs/contexts/EncoreContext.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.hasStatus = exports.getStatus = exports.encoreContextStatus = exports.encoreContextKeyword = exports.EncoreContextDefault = exports.EncoreContext = void 0;

var _react = _interopRequireDefault(require("react"));

var encoreContextStatus = {
  experimental: 'experimental',
  next: 'next',
  deprecated: 'deprecated'
};
exports.encoreContextStatus = encoreContextStatus;
var encoreContextKeyword = {
  type: 'type',
  button: 'button',
  icon: 'icon'
};
exports.encoreContextKeyword = encoreContextKeyword;
var EncoreContextDefault = {
  experimental: [],
  next: [],
  deprecated: []
};
exports.EncoreContextDefault = EncoreContextDefault;

var hasStatus = function hasStatus(keyword, status) {
  return status.indexOf(keyword) > -1;
};

exports.hasStatus = hasStatus;

var getStatus = function getStatus(keyword, config) {
  var lifecycle = undefined;
  var statusKeys = Object.keys(encoreContextStatus);
  statusKeys.forEach(function (status) {
    if (hasStatus(keyword, config[status])) lifecycle = status;
  });
  return lifecycle;
};

exports.getStatus = getStatus;

var EncoreContext = /*#__PURE__*/_react["default"].createContext(EncoreContextDefault);

exports.EncoreContext = EncoreContext;
EncoreContext.displayName = 'Encore';
