// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DEFAULT_PO_NUMBER_TEXT = exports.DEFAULT_NONE_TEXT = exports.DEFAULT_CANNOT_ADD_CARD_SUBTEXT = exports.DEFAULT_CARD_SUCCESS_TEXT = exports.RETRY_TEXT = exports.DEFAULT_ERROR_HELP_TEXT = exports.DEFAULT_REDIRECTING_CTA_TEXT = exports.DEFAULT_INVOICE_DESCRIPTION = exports.DEFAULT_INVOICE_TEXT = exports.DEFAULT_CARD_DESCRIPTION_SUB = exports.DEFAULT_CARD_DESCRIPTION = exports.DEFAULT_CARD_CTA_TEXT = exports.DEFAULT_ADD_CARD_DESCRIPTION = exports.DEFAULT_ADD_CARD_CTA_TEXT = exports.DEFAULT_ADD_CARD_SUBTEXT = exports.DEFAULT_ADD_CARD_TEXT = exports.DEFAULT_CARD_CONFIG = exports.SUPPORTED_CARDS = exports.CREDIT_CARD_CONFIG = exports.DETAILS_TYPES = exports.DETAILS_INVOICE = exports.DETAILS_CARD = void 0;

var _IconVisa = _interopRequireDefault(require("./Icon/IconVisa"));

var _IconMastercard = _interopRequireDefault(require("./Icon/IconMastercard"));

var _IconAmex = _interopRequireDefault(require("./Icon/IconAmex"));

var _IconDiscover = _interopRequireDefault(require("./Icon/IconDiscover"));

var _IconDefault = _interopRequireDefault(require("./Icon/IconDefault"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var DETAILS_CARD = 'CARD';
exports.DETAILS_CARD = DETAILS_CARD;
var DETAILS_INVOICE = 'INVOICE';
exports.DETAILS_INVOICE = DETAILS_INVOICE;
var DETAILS_TYPES = [DETAILS_CARD, DETAILS_INVOICE];
exports.DETAILS_TYPES = DETAILS_TYPES;
var CREDIT_CARD_CONFIG = {
  visa: {
    format: [4, 4, 4, 4],
    name: 'Visa',
    icon: _IconVisa["default"]
  },
  mc: {
    format: [4, 4, 4, 4],
    name: 'Mastercard',
    icon: _IconMastercard["default"]
  },
  amex: {
    format: [4, 6, 5],
    name: 'Amex',
    icon: _IconAmex["default"]
  },
  discover: {
    format: [4, 4, 4, 4],
    name: 'Discover',
    icon: _IconDiscover["default"]
  }
};
exports.CREDIT_CARD_CONFIG = CREDIT_CARD_CONFIG;
var SUPPORTED_CARDS = [CREDIT_CARD_CONFIG.visa, CREDIT_CARD_CONFIG.mc, CREDIT_CARD_CONFIG.amex];
exports.SUPPORTED_CARDS = SUPPORTED_CARDS;
var DEFAULT_CARD_CONFIG = {
  format: CREDIT_CARD_CONFIG.mc.format,
  icon: _IconDefault["default"]
};
exports.DEFAULT_CARD_CONFIG = DEFAULT_CARD_CONFIG;
var DEFAULT_ADD_CARD_TEXT = '**Add your card to continue.**';
exports.DEFAULT_ADD_CARD_TEXT = DEFAULT_ADD_CARD_TEXT;
var DEFAULT_ADD_CARD_SUBTEXT = "You'll be redirected to a new page. Your work will be saved.";
exports.DEFAULT_ADD_CARD_SUBTEXT = DEFAULT_ADD_CARD_SUBTEXT;
var DEFAULT_ADD_CARD_CTA_TEXT = 'Add card';
exports.DEFAULT_ADD_CARD_CTA_TEXT = DEFAULT_ADD_CARD_CTA_TEXT;
var DEFAULT_ADD_CARD_DESCRIPTION = 'Payment method must be from the same country as your account.';
exports.DEFAULT_ADD_CARD_DESCRIPTION = DEFAULT_ADD_CARD_DESCRIPTION;
var DEFAULT_CARD_CTA_TEXT = 'Update';
exports.DEFAULT_CARD_CTA_TEXT = DEFAULT_CARD_CTA_TEXT;
var DEFAULT_CARD_DESCRIPTION = 'We’ll charge your credit card at the end of each month or when your ad spend reaches a given [billing threshold](https://adstudio.spotify.com/help/billing-cycle) across all your campaigns.';
exports.DEFAULT_CARD_DESCRIPTION = DEFAULT_CARD_DESCRIPTION;
var DEFAULT_CARD_DESCRIPTION_SUB = '**Note:** only one credit card can be kept on file for billing.';
exports.DEFAULT_CARD_DESCRIPTION_SUB = DEFAULT_CARD_DESCRIPTION_SUB;
var DEFAULT_INVOICE_TEXT = 'You have invoicing set up.';
exports.DEFAULT_INVOICE_TEXT = DEFAULT_INVOICE_TEXT;
var DEFAULT_INVOICE_DESCRIPTION = "\n  If you have questions about billing, visit our FAQ or please email us at [adstudio@spotify.com](mailto:adstudio@spotify.com)\n";
exports.DEFAULT_INVOICE_DESCRIPTION = DEFAULT_INVOICE_DESCRIPTION;
var DEFAULT_REDIRECTING_CTA_TEXT = 'Redirecting';
exports.DEFAULT_REDIRECTING_CTA_TEXT = DEFAULT_REDIRECTING_CTA_TEXT;
var DEFAULT_ERROR_HELP_TEXT = "\n  Contact [adstudio@spotify.com](mailto:adstudio@spotify.com)\n  if you're having trouble.\n";
exports.DEFAULT_ERROR_HELP_TEXT = DEFAULT_ERROR_HELP_TEXT;
var RETRY_TEXT = 'Retry';
exports.RETRY_TEXT = RETRY_TEXT;
var DEFAULT_CARD_SUCCESS_TEXT = 'Card saved';
exports.DEFAULT_CARD_SUCCESS_TEXT = DEFAULT_CARD_SUCCESS_TEXT;
var DEFAULT_CANNOT_ADD_CARD_SUBTEXT = "Viewers aren't permitted to add a payment method. Please contact your Admin.";
exports.DEFAULT_CANNOT_ADD_CARD_SUBTEXT = DEFAULT_CANNOT_ADD_CARD_SUBTEXT;
var DEFAULT_NONE_TEXT = 'none';
exports.DEFAULT_NONE_TEXT = DEFAULT_NONE_TEXT;
var DEFAULT_PO_NUMBER_TEXT = 'PO number';
exports.DEFAULT_PO_NUMBER_TEXT = DEFAULT_PO_NUMBER_TEXT;
