// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/Container.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Container;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _Type = _interopRequireDefault(require("../Type"));

var _Panel = _interopRequireDefault(require("../Panel"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ContentIcon = _styledComponents["default"].div.withConfig({
  displayName: "Container__ContentIcon",
  componentId: "z6mwnj-0"
})(["display:flex;flex-direction:column;height:", ";justify-content:center;padding-right:", ";"], _encoreFoundation.spacer32, _encoreFoundation.spacer16);

var PaymentDetailsContent = _styledComponents["default"].div.withConfig({
  displayName: "Container__PaymentDetailsContent",
  componentId: "z6mwnj-1"
})(["", ""], function (props) {
  return props.loading && (0, _styledComponents.css)(["align-items:center;display:flex;justify-content:center;width:100%;"]);
});

var Description = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "Container__Description",
  componentId: "z6mwnj-2"
})(["margin-bottom:0;padding-bottom:0;padding-top:", ";&&& a{color:", ";}"], _encoreFoundation.spacer8, _encoreFoundation.black);

function Container(_ref) {
  var className = _ref.className,
      children = _ref.children,
      cta = _ref.cta,
      description = _ref.description,
      hideShadow = _ref.hideShadow,
      Icon = _ref.icon,
      error = _ref.error,
      warning = _ref.warning,
      loading = _ref.loading;
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: className
  }, /*#__PURE__*/_react["default"].createElement(_Panel["default"], {
    ctaList: cta ? [cta] : undefined,
    shadow: !hideShadow,
    error: error,
    warning: warning
  }, Icon && /*#__PURE__*/_react["default"].createElement(ContentIcon, null, Icon && /*#__PURE__*/_react["default"].createElement(Icon, null)), /*#__PURE__*/_react["default"].createElement(PaymentDetailsContent, {
    loading: loading
  }, children)), description && /*#__PURE__*/_react["default"].createElement(Description, {
    element: "p",
    variant: "body-2-book"
  }, description));
}

Container.propTypes = {
  className: _propTypes["default"].string,
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node, _propTypes["default"].element]),
  cta: _propTypes["default"].shape({
    label: _propTypes["default"].string,
    disabled: _propTypes["default"].bool,
    onClick: _propTypes["default"].func
  }),
  description: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node, _propTypes["default"].element]),
  error: _propTypes["default"].bool,
  hideShadow: _propTypes["default"].bool,
  icon: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].object]),
  warning: _propTypes["default"].bool,
  loading: _propTypes["default"].bool
};
