// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/Error.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = PaymentMethodError;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

var _Type = _interopRequireDefault(require("../Type"));

var _constants = require("./constants");

var _Container = _interopRequireDefault(require("./Container"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function PaymentMethodError(_ref) {
  var error = _ref.error,
      errorHelpText = _ref.errorHelpText,
      retryText = _ref.retryText,
      onClickRetry = _ref.onClickRetry,
      hideShadow = _ref.hideShadow,
      warning = _ref.warning;
  return /*#__PURE__*/_react["default"].createElement(_Container["default"], {
    cta: {
      label: retryText,
      onClick: onClickRetry
    },
    description: /*#__PURE__*/_react["default"].createElement(_Type["default"], {
      element: "span",
      variant: "body-2-book"
    }, /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
      source: errorHelpText,
      renderers: {
        /* eslint-disable react/prop-types */
        paragraph: function paragraph(_ref2) {
          var children = _ref2.children;
          return /*#__PURE__*/_react["default"].createElement("span", null, children);
        }
        /* eslint-enable react/prop-types */

      }
    })),
    error: !!error,
    hideShadow: hideShadow,
    warning: !!warning
  }, /*#__PURE__*/_react["default"].createElement(_Type["default"], {
    element: "span",
    variant: "body-1-book"
  }, error || warning));
}

PaymentMethodError.propTypes = {
  error: _propTypes["default"].string,
  errorHelpText: _propTypes["default"].string,
  hideShadow: _propTypes["default"].bool,
  onClickRetry: _propTypes["default"].func,
  retryText: _propTypes["default"].string,
  warning: _propTypes["default"].string
};
PaymentMethodError.defaultProps = {
  errorHelpText: _constants.DEFAULT_ERROR_HELP_TEXT,
  retryText: _constants.RETRY_TEXT
};
