// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = PaymentDetails;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _constants = require("./constants");

var _AddCard = _interopRequireDefault(require("./AddCard"));

var _Card = _interopRequireDefault(require("./Card"));

var _Invoice = _interopRequireDefault(require("./Invoice"));

var _Loading = _interopRequireDefault(require("./Loading"));

var _Error = _interopRequireDefault(require("./Error"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function PaymentDetails(props) {
  var error = props.error,
      loading = props.loading,
      type = props.type,
      warning = props.warning; // when loading, show the loading component

  if (loading) return /*#__PURE__*/_react["default"].createElement(_Loading["default"], props); // when there is an error, and we're not loading, show the error

  if (error || warning) return /*#__PURE__*/_react["default"].createElement(_Error["default"], props); // render the Card component for the CARD type

  if (type === _constants.DETAILS_CARD) return /*#__PURE__*/_react["default"].createElement(_Card["default"], props); // render the Invoice component for the INVOICE type

  if (type === _constants.DETAILS_INVOICE) return /*#__PURE__*/_react["default"].createElement(_Invoice["default"], props); // render AddCard for all other empty cases

  return /*#__PURE__*/_react["default"].createElement(_AddCard["default"], props);
}

PaymentDetails.propTypes = {
  addCardCTAtext: _propTypes["default"].string,
  addCardDescription: _propTypes["default"].string,
  addCardSubtext: _propTypes["default"].string,
  addCardText: _propTypes["default"].string,
  cardCTAtext: _propTypes["default"].string,
  cardDescription: _propTypes["default"].string,
  cardDescriptionSubtext: _propTypes["default"].string,
  cardSuccessText: _propTypes["default"].string,
  detail: _propTypes["default"].shape({
    cardNumber: _propTypes["default"].string,
    paymentMethod: _propTypes["default"].string,
    expiryMonth: _propTypes["default"].string,
    expiryYear: _propTypes["default"].string
  }),
  error: _propTypes["default"].string,
  errorHelpText: _propTypes["default"].string,
  invoiceDescription: _propTypes["default"].string,
  invoiceText: _propTypes["default"].string,
  loading: _propTypes["default"].bool,
  onClickCheckout: _propTypes["default"].func,
  onClickRetry: _propTypes["default"].func,
  noPONumberText: _propTypes["default"].string,
  poNumberText: _propTypes["default"].string,
  redirecting: _propTypes["default"].bool,
  redirectingCTAtext: _propTypes["default"].string,
  retryText: _propTypes["default"].string,
  type: _propTypes["default"].oneOf(_constants.DETAILS_TYPES),
  viewOnly: _propTypes["default"].bool,
  viewOnlySubText: _propTypes["default"].string,
  warning: _propTypes["default"].string
};
