// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/Invoice.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Invoice;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Type = _interopRequireDefault(require("../Type"));

var _constants = require("./constants");

var _IconInvoice = _interopRequireDefault(require("./Icon/IconInvoice"));

var _Container = _interopRequireDefault(require("./Container"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var InvoiceType = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "Invoice__InvoiceType",
  componentId: "sc-1xhyba4-0"
})(["padding-top:1em;padding-bottom:4px;"]);

function Invoice(_ref) {
  var hideShadow = _ref.hideShadow,
      invoiceDescription = _ref.invoiceDescription,
      invoiceText = _ref.invoiceText,
      noPONumberText = _ref.noPONumberText,
      poNumber = _ref.poNumber,
      poNumberText = _ref.poNumberText;
  return /*#__PURE__*/_react["default"].createElement(_Container["default"], {
    hideShadow: hideShadow,
    icon: _IconInvoice["default"],
    description: /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
      source: invoiceDescription,
      renderers: {
        /* eslint-disable react/prop-types */
        paragraph: function paragraph(_ref2) {
          var children = _ref2.children;
          return /*#__PURE__*/_react["default"].createElement("span", null, children);
        }
        /* eslint-enable react/prop-types */

      }
    })
  }, /*#__PURE__*/_react["default"].createElement(InvoiceType, {
    element: "p",
    variant: "body-2-bold"
  }, invoiceText), /*#__PURE__*/_react["default"].createElement(_Type["default"], {
    element: "p",
    variant: "body-2-book"
  }, poNumberText, ": ", poNumber ? poNumber : noPONumberText));
}

Invoice.propTypes = {
  hideShadow: _propTypes["default"].bool,
  invoiceDescription: _propTypes["default"].string,
  invoiceText: _propTypes["default"].string,
  noPONumberText: _propTypes["default"].string,
  poNumber: _propTypes["default"].string,
  poNumberText: _propTypes["default"].string
};
Invoice.defaultProps = {
  invoiceDescription: _constants.DEFAULT_INVOICE_DESCRIPTION,
  invoiceText: _constants.DEFAULT_INVOICE_TEXT,
  noPONumberText: _constants.DEFAULT_NONE_TEXT,
  poNumberText: _constants.DEFAULT_PO_NUMBER_TEXT
};
