// @spotify-internal/encore-web-v3/es/styles/mixins/buttons.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

var _buttonFontSize, _buttonPadding;

//
// Buttons
// --------------------------------------------------

/* eslint-disable import/prefer-default-export */
import { css } from 'styled-components';
import { fontWeightBold, rootStyle } from "../../styles";
import { browserFocusReset, focusBorderUsingProp, focusBottomBorder, focusGapBorder } from "./focusBorders";
export var buttonSizes = {
  sm: 'sm',
  md: 'md',
  lg: 'lg'
};
export var buttonFontWeight = fontWeightBold;
export var buttonBorderRadius = '500px';
export var buttonLineHeight = '1';
export var buttonFontSize = (_buttonFontSize = {}, _defineProperty(_buttonFontSize, buttonSizes.sm, '12px'), _defineProperty(_buttonFontSize, buttonSizes.md, '14px'), _defineProperty(_buttonFontSize, buttonSizes.lg, '16px'), _buttonFontSize);
export var buttonPadding = (_buttonPadding = {}, _defineProperty(_buttonPadding, buttonSizes.sm, '10px'), _defineProperty(_buttonPadding, buttonSizes.md, '17px'), _defineProperty(_buttonPadding, buttonSizes.lg, '20px'), _buttonPadding);
export var buttonTransitionFade = '33ms';
export var baseButtonStyles = function baseButtonStyles(buttonSize) {
  return css(["", ";background-color:transparent;border:0;border-radius:", ";display:inline-block;font-size:", ";font-weight:", ";letter-spacing:2px;line-height:", ";text-align:center;text-decoration:none;text-transform:uppercase;touch-action:manipulation;transition-duration:", ";transition-property:background-color,border-color,color,box-shadow,filter,transform;user-select:none;vertical-align:middle;transform:translate3d(0,0,0);"], rootStyle(), buttonBorderRadius, buttonFontSize[buttonSize], buttonFontWeight, buttonLineHeight, buttonTransitionFade);
};
export function getButtonPadding(buttonSize) {
  var size = parseInt(buttonFontSize[buttonSize], 10);
  var paddingTopBottom = parseInt(buttonPadding[buttonSize], 10);
  var paddingLeftRight = size + paddingTopBottom * 2;
  return {
    paddingTopBottom: paddingTopBottom,
    paddingLeftRight: paddingLeftRight
  };
}
export var buttonStyle = function buttonStyle(_ref) {
  var buttonSize = _ref.buttonSize,
      condensed = _ref.condensed,
      variantRules = _ref.variantRules,
      theme = _ref.theme,
      focus = _ref.focus,
      _ref$isUsingKeyboard = _ref.isUsingKeyboard,
      isUsingKeyboard = _ref$isUsingKeyboard === void 0 ? true : _ref$isUsingKeyboard,
      _ref$tertiary = _ref.tertiary,
      tertiary = _ref$tertiary === void 0 ? false : _ref$tertiary;
  return css(["", ";", ";", ";", ";"], baseButtonStyles(buttonSize), function () {
    var _getButtonPadding = getButtonPadding(buttonSize),
        paddingTopBottom = _getButtonPadding.paddingTopBottom,
        paddingLeftRight = _getButtonPadding.paddingLeftRight;

    var isDarkTheme = (theme == null ? void 0 : theme.button) === 'dark';
    var focusBorderStyle = tertiary ? focusBottomBorder(isDarkTheme, paddingTopBottom, isUsingKeyboard) : focusGapBorder(isDarkTheme, buttonBorderRadius, isUsingKeyboard); // custom css to override the default 100% width since we want a focus border
    // to expand only under the button text (excluding the padding in a tertiary button)

    var tertiaryFocusStyles = tertiary ? css(["&::after{width:", ";}"], "calc(100% - ".concat(condensed ? 0 : paddingLeftRight * 2, "px)")) : null;
    return css(["padding:", "px ", "px;", ";", ";"], paddingTopBottom, condensed ? 0 : paddingLeftRight, focusBorderStyle, tertiaryFocusStyles);
  }, function () {
    var isDarkTheme = (theme == null ? void 0 : theme.button) === 'dark';
    return focus && focusBorderUsingProp(isDarkTheme, isUsingKeyboard);
  }, variantRules);
}; // Reset a button to a link looking thing

export var buttonLinkReset = function buttonLinkReset() {
  var _ref2 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref2$lineHeight = _ref2.lineHeight,
      lineHeight = _ref2$lineHeight === void 0 ? buttonLineHeight : _ref2$lineHeight;

  return css(["background-color:transparent;border:0;color:inherit;line-height:", ";letter-spacing:inherit;padding:0;", ""], lineHeight, browserFocusReset);
}; //  Hover primary and secondary buttons

export var buttonHover = function buttonHover() {
  return css(["transform:scale(1.04);"]);
}; // Make hoverable area of a button or link span the full height and width of its container

export var stretchedLink = function stretchedLink() {
  return css(["&::after{position:absolute;top:0;right:0;bottom:0;left:0;z-index:1;pointer-events:auto;content:'';}"]);
};
