// @spotify-internal/encore-web/es/components/ButtonPrimary/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["component", "colorSet", "buttonSize", "fullWidth", "iconLeading", "iconTrailing", "iconOnly", "children", "className", "UNSAFE_colorSet", "aria-label", "aria-labelledby", "buttonLegacy", "theme"],
    _excluded2 = ["buttonLegacy"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { useContext } from 'react';
import { BrowserDefaultFocusStyleContext } from "../../contexts/BrowserDefaultFocusStyleContext";
import { EncoreContext, encoreContextKeyword, getStatus } from "../../contexts/EncoreContext";
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { deprecatedPropWarning, renamedPropWarning } from "../../deprecationWarnings";
import { addColorSet, buttonSizes, defaultTheme } from "../../styles";
import { Button } from "./Button";
import { ButtonChildren } from "./ButtonChildren";
import { ButtonFocus } from "./ButtonFocus";
import { ButtonInner } from "./ButtonInner";
import { ButtonLegacyComponent } from "./ButtonLegacy";
var ButtonComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var component = _ref.component,
      _ref$colorSet = _ref.colorSet,
      colorSet = _ref$colorSet === void 0 ? 'brightAccent' : _ref$colorSet,
      _ref$buttonSize = _ref.buttonSize,
      buttonSize = _ref$buttonSize === void 0 ? 'md' : _ref$buttonSize,
      fullWidth = _ref.fullWidth,
      iconLeading = _ref.iconLeading,
      iconTrailing = _ref.iconTrailing,
      iconOnly = _ref.iconOnly,
      children = _ref.children,
      className = _ref.className,
      UNSAFE_colorSet = _ref.UNSAFE_colorSet,
      ariaLabel = _ref['aria-label'],
      ariaLabelledby = _ref['aria-labelledby'],
      buttonLegacy = _ref.buttonLegacy,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    renamedPropWarning('theme', 'colorSet');
  }

  if (iconOnly && !ariaLabel && !ariaLabelledby) {
    // eslint-disable-next-line no-console
    console.warn('ButtonPrimary must have an aria-label or aria-labelledby attribute defined when using the iconOnly prop in order to be accessible.');
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  var _useContext2 = useContext(BrowserDefaultFocusStyleContext),
      useBrowserDefaultFocusStyle = _useContext2.useBrowserDefaultFocusStyle;

  var config = useContext(EncoreContext);
  var status = getStatus(encoreContextKeyword.button, config);
  var rel = props.href && props.target === '_blank' ? 'noopener noreferrer' : null;
  var fallbackSet = defaultTheme[colorSet];
  return /*#__PURE__*/React.createElement(Button, Object.assign({}, props, {
    rel: rel,
    ref: ref,
    component: !component && props.href ? 'a' : component,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledby,
    buttonSize: buttonSize,
    fullWidth: fullWidth,
    fallbackSet: fallbackSet,
    useBrowserDefaultFocusStyle: useBrowserDefaultFocusStyle,
    status: status
  }), /*#__PURE__*/React.createElement(ButtonInner, {
    className: UNSAFE_colorSet ? className : addColorSet(colorSet, className),
    fullWidth: fullWidth,
    iconLeading: iconLeading,
    iconTrailing: iconTrailing,
    iconOnly: iconOnly,
    UNSAFE_colorSet: UNSAFE_colorSet,
    buttonSize: buttonSize,
    fallbackSet: fallbackSet
  }, /*#__PURE__*/React.createElement(ButtonChildren, {
    iconOnly: iconOnly,
    iconLeading: iconLeading,
    iconTrailing: iconTrailing,
    buttonSize: buttonSize,
    children: children
  })), !useBrowserDefaultFocusStyle && /*#__PURE__*/React.createElement(ButtonFocus, {
    isUsingKeyboard: isUsingKeyboard
  }));
});
/** Component that figures out if we are rendering the legacy or current button */

var ButtonToRender = /*#__PURE__*/React.forwardRef(function (_ref2, ref) {
  var buttonLegacy = _ref2.buttonLegacy,
      props = _objectWithoutProperties(_ref2, _excluded2);

  if (buttonLegacy) {
    deprecatedPropWarning('buttonLegacy', "This prop will be removed soon. Please migrate to new button styles by removing this prop.\nRead more: https://encore.spotify.net/web/whats-new/green-buttons/?format=web-js");
    return /*#__PURE__*/React.createElement(ButtonLegacyComponent, Object.assign({
      ref: ref
    }, props));
  }

  return /*#__PURE__*/React.createElement(ButtonComponent, Object.assign({
    ref: ref
  }, props));
});
/**
 * TODO: Remove when we deprecate `buttonSize` helpers like `ButtonPrimary.sm`.
 */

var buttonHelpers = _objectSpread(_objectSpread({}, buttonSizes), {}, {
  displayName: 'ButtonPrimary'
});
/**
* **ButtonPrimary** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/ButtonPrimary) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-button-buttonprimary--default) | [Encore](https://encore.spotify.net/web/components/button#buttonprimary) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:214) |
*
* Primary buttons are for the action with the highest priority. They use color fills and are green by default.
*
* @example
* () => <ButtonPrimary>Default</ButtonPrimary>;
*
*/


export var ButtonPrimary = Object.assign(ButtonToRender, buttonHelpers);
export default ButtonPrimary;
