// @spotify-internal/encore-web/es/components/FormToggle/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { VisuallyHidden } from "../VisuallyHidden";
import { Checkbox } from "./Checkbox";
import { Indicator } from "./Indicator";
import { Label } from "./Label";
import { Wrapper } from "./Wrapper";
var FormToggleComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      small = _ref.small,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "small", "children"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Checkbox, null, /*#__PURE__*/React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "checkbox",
    ref: ref
  }, props)), /*#__PURE__*/React.createElement(Wrapper, {
    isUsingKeyboard: isUsingKeyboard
  }, /*#__PURE__*/React.createElement(Indicator, null)), children && /*#__PURE__*/React.createElement(Label, {
    small: small
  }, children)));
});
var formToggleHelpers = {
  darkTheme: {
    formToggle: 'dark'
  },
  displayName: 'FormToggle'
};
/**
* **FormToggle** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/FormToggle) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-formtoggle--default) | [Encore](https://encore.spotify.net/web/components/selectors?format=figma-figma#formtoggle) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:620) |
*
* FormToggle allow users to switch the state of a single item on or off, within a Form component.
*
* @example
* export const Default = () => (
*  <FormGroup label="FormToggle example" withFieldset>
*    <FormToggle defaultChecked>Option 1</FormToggle>
*    <FormToggle>Option 2</FormToggle>
*  </FormGroup>
*);
*
*/

export var FormToggle = Object.assign(FormToggleComponent, formToggleHelpers);
export default FormToggle;
