// @spotify-internal/adstudio-tape/lib/components/Input/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _forms = require("../../styles/mixins/forms");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var StyledInput = _styledComponents["default"].input.withConfig({
  displayName: "Input__StyledInput",
  componentId: "zuvgql-0"
})(["", " ", " ", ""], _forms.formControlBase, _forms.formControlBaseInput, function (_ref) {
  var theme = _ref.theme,
      error = _ref.error,
      warning = _ref.warning;
  return (0, _forms.formControlTheme)({
    theme: theme,
    error: error,
    warning: warning
  });
});

var propTypes = {
  className: _propTypes["default"].string,

  /** application theme */
  theme: _propTypes["default"].oneOf(['dark', 'light']),
  error: _propTypes["default"].oneOfType([_propTypes["default"].bool, _propTypes["default"].node]),
  warning: _propTypes["default"].oneOfType([_propTypes["default"].bool, _propTypes["default"].node])
};
var defaultProps = {
  theme: 'light',
  error: false,
  warning: false
};

var Input = /*#__PURE__*/_react["default"].forwardRef(function (_ref2, ref) {
  var theme = _ref2.theme,
      error = _ref2.error,
      warning = _ref2.warning,
      className = _ref2.className,
      props = _objectWithoutProperties(_ref2, ["theme", "error", "warning", "className"]);

  return /*#__PURE__*/_react["default"].createElement(StyledInput, _extends({
    theme: theme,
    error: error,
    warning: warning,
    className: className,
    ref: ref
  }, props));
});

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;
var _default = Input;
exports["default"] = _default;
