// @spotify-internal/encore-web/es/components/Image/Image.js
import styled, { css } from 'styled-components';
import { imageStyle } from "../../styles";

var imageStyles = function imageStyles() {
  return css(["", ";", ";", ";", ";", ";"], function (props) {
    return props.imageWidth && css(["width:", ";"], props.imageWidth);
  }, function (props) {
    return props.imageHeight && css(["height:", ";"], props.imageHeight);
  }, function (props) {
    return props.circle && css(["border-radius:50%;"]);
  }, function (props) {
    return props.crop && css(["object-fit:cover;"]);
  }, function (props) {
    return props.fluid && css(["max-width:100%;height:auto;"]);
  });
};

export var PlaceholderImageWrapper = styled.div.withConfig({
  displayName: "Image__PlaceholderImageWrapper",
  componentId: "sc-1u215sg-0"
})(["overflow:hidden;img{", ";filter:blur(20px);transform:scale(1.03);}", ";"], imageStyle(), imageStyles());
export var PlaceholderImage = styled.img.withConfig({
  displayName: "Image__PlaceholderImage",
  componentId: "sc-1u215sg-1"
})(["", ";width:100%;height:100%;"], imageStyle());
export var StyledLazyImage = styled.img.withConfig({
  displayName: "Image__StyledLazyImage",
  componentId: "sc-1u215sg-2"
})(["", ";display:", ";", ";"], imageStyle(), function (props) {
  return props.isLoaded ? 'initial' : 'none';
}, imageStyles());
export default styled.img.withConfig({
  displayName: "Image",
  componentId: "sc-1u215sg-3"
})(["", ";"], imageStyles());
