// @spotify-internal/encore-web/es/components/Tag/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { TagElement } from "./Tag";

function TagComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(TagElement, Object.assign({
    isUsingKeyboard: isUsingKeyboard
  }, props)));
}

var tagHelpers = {
  darkTheme: {
    tag: 'dark'
  }
};
/**
* **Tag** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Tag) | [Storybook](https://encore-web.spotify.net/?path=/story/components-tag--default) | [Encore](https://encore.spotify.net/web/components/tags) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=522:6) |
*
* Use Tags to present a collection of attributes (metadata, filters, multiple values in a text field, or selectable short choices in a form) to describe a piece of content.
*
* @example
* export const Default = () => <Tag>Hip Hop</Tag>;
*
*/

export var Tag = Object.assign(TagComponent, tagHelpers);
export default Tag;
