// @spotify-internal/adstudio-shared/lib/components/AudienceSizerLegacy/BudgetPanel.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getCurrencyOverrides = getCurrencyOverrides;
exports["default"] = exports.BudgetPanel = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _Panel = require("@spotify-internal/adstudio-tape/lib/components/Panel");

var _FormatType = require("../../config/FormatType");

var _constants = require("./constants");

var _LabelledItem = _interopRequireWildcard(require("./LabelledItem"));

var _RadialMeter = _interopRequireDefault(require("../RadialMeter"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var MeterContainerWidth = '230px';

var invalidAudienceDeliveryLikelihood = function invalidAudienceDeliveryLikelihood(audienceDeliveryLikelihood) {
  return typeof audienceDeliveryLikelihood !== 'number' || isNaN(audienceDeliveryLikelihood);
};

function getCurrencyOverrides(cost) {
  var overrides = {}; // calculate the best decimal number based on the cost value. should always show
  // two places no matter how small. 0.00001566 should be 0.000016, etc.

  if (cost) {
    var exponentialPower = Number(Number(cost).toExponential().split('e')[1]);

    if (exponentialPower < -1) {
      overrides.decimals = Math.abs(exponentialPower) + 1;
    }
  }

  return overrides;
}

var ICON_SIZE = 16;
var LOW_VALUE_THRESHOLD = 50;
var MED_VALUE_THRESHOLD = 79;
var StyledPanel = (0, _styledComponents["default"])(_adstudioTape.Panel).withConfig({
  displayName: "BudgetPanel__StyledPanel",
  componentId: "sc-1c90sfh-0"
})(["&&{border-color:", ";flex-direction:column;background:", ";flex-grow:1;flex-shrink:1;align-items:unset;}&:first-child{border-bottom-left-radius:0;border-bottom-right-radius:0;}&:last-child{border-top:unset;border-top-left-radius:0;border-top-right-radius:0;}", "{align-items:unset;}"], _encoreAdvertisingWeb.gray90, _encoreAdvertisingWeb.white, _Panel.PanelContent);

var MeterContainer = _styledComponents["default"].div.withConfig({
  displayName: "BudgetPanel__MeterContainer",
  componentId: "sc-1c90sfh-1"
})(["color:", ";font-size:", ";margin-bottom:", ";width:", ";"], _encoreAdvertisingWeb.gray60, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.spacer16, MeterContainerWidth);

var BudgetText = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.h4).withConfig({
  displayName: "BudgetPanel__BudgetText",
  componentId: "sc-1c90sfh-2"
})(["&&{padding-bottom:1em;}"]);

var BudgetPanel = function BudgetPanel(_ref) {
  var audienceDeliveryLikelihood = _ref.audienceDeliveryLikelihood,
      currencyFormatter = _ref.currencyFormatter,
      costRaw = _ref.cost,
      costModel = _ref.costModel,
      error = _ref.error,
      _ref$hasMetRequiredMi = _ref.hasMetRequiredMinReach,
      hasMetRequiredMinReach = _ref$hasMetRequiredMi === void 0 ? true : _ref$hasMetRequiredMi,
      format = _ref.format,
      _ref$serveOnMegaphone = _ref.serveOnMegaphone,
      serveOnMegaphone = _ref$serveOnMegaphone === void 0 ? false : _ref$serveOnMegaphone,
      strings = _ref.strings,
      props = _objectWithoutProperties(_ref, ["audienceDeliveryLikelihood", "currencyFormatter", "cost", "costModel", "error", "hasMetRequiredMinReach", "format", "serveOnMegaphone", "strings"]);

  var isVideoFormat = format === _FormatType.Format.VIDEO;
  var cost = costRaw && costModel === _constants.IMPRESSIONS_MODEL ? costRaw * 1000 : costRaw;
  var currencyOverrides = getCurrencyOverrides(cost);

  function getRecommendationIcon(status) {
    var IconExclamationAlt = _adstudioTape.Icons.IconExclamationAlt,
        IconCheckAlt = _adstudioTape.Icons.IconCheckAlt;

    switch (status) {
      case _constants.DELIVERY_STATUS.ERROR:
      case _constants.DELIVERY_STATUS.REACH_TOO_LOW:
      case _constants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD:
        return /*#__PURE__*/_react["default"].createElement(IconExclamationAlt, {
          color: "failure",
          iconSize: ICON_SIZE
        });

      case _constants.DELIVERY_STATUS.INVALID:
      case _constants.DELIVERY_STATUS.MED_VALUE_THRESHOLD:
        return null;

      default:
        return /*#__PURE__*/_react["default"].createElement(IconCheckAlt, {
          color: "success",
          iconSize: ICON_SIZE
        });
    }
  }

  function getRecommendationText(status) {
    switch (status) {
      case _constants.DELIVERY_STATUS.ERROR:
        return strings.RECOMMENDATIONTEXT.Error;

      case _constants.DELIVERY_STATUS.INVALID:
        return strings.RECOMMENDATIONTEXT.NoValue;

      case _constants.DELIVERY_STATUS.REACH_TOO_LOW:
        return strings.RECOMMENDATIONTEXT.ReachTooLow;

      case _constants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD:
      case _constants.DELIVERY_STATUS.MED_VALUE_THRESHOLD:
        {
          var recText = strings.RECOMMENDATIONTEXT;
          return {
            Label: status === _constants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD ? recText.LowValue.Label : recText.MedValue.Label,
            Body: /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
              escapeHtml: false,
              renderers: {
                paragraph: function paragraph(mProps) {
                  return /*#__PURE__*/_react["default"].createElement(_LabelledItem.RecommendationText, _extends({
                    variant: _encoreAdvertisingWeb.Type.body3,
                    weight: _encoreAdvertisingWeb.Type.book
                  }, mProps));
                },
                list: function list(lProps) {
                  return /*#__PURE__*/_react["default"].createElement(_LabelledItem.ListContainer, null, ' ', /*#__PURE__*/_react["default"].createElement(_LabelledItem.StyledList, lProps), ' ');
                },
                listItem: _LabelledItem.StyledListItem
              },
              source: serveOnMegaphone ? strings.LOW_OR_MED_VALUE_PODCAST_MESSAGE : strings.LOW_OR_MED_VALUE_MESSAGE
            })
          };
        }

      default:
        return strings.RECOMMENDATIONTEXT.HighValue;
    }
  }

  function getLabelledItem() {
    return /*#__PURE__*/_react["default"].createElement(_LabelledItem["default"], {
      labelText: strings.REACH_COST_LABEL_TEXT,
      placeholder: "".concat(currencyFormatter(0, currencyOverrides), " ").concat(strings.CPM_LABEL)
    }, cost && costModel === _constants.PAID_LISTENS_MODEL && !isVideoFormat && "".concat(currencyFormatter(cost, currencyOverrides), " ").concat(strings.PER_LISTEN), cost && costModel === _constants.PAID_LISTENS_MODEL && isVideoFormat && "".concat(currencyFormatter(cost, currencyOverrides), " ").concat(strings.PER_VIEW), cost && costModel === _constants.IMPRESSIONS_MODEL && !isVideoFormat && /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
      placement: "left",
      tooltipText: strings.CPM_TOOLTIP
    }, "".concat(currencyFormatter(cost, currencyOverrides), " ").concat(strings.CPM_LABEL)), cost && costModel === _constants.IMPRESSIONS_MODEL && isVideoFormat && "".concat(currencyFormatter(cost, currencyOverrides), " ").concat(strings.CPM_LABEL));
  }

  var getDeliveryStatus = function getDeliveryStatus(audienceDeliveryLikelihood, error, hasMetRequiredMinReach) {
    if (error) {
      return _constants.DELIVERY_STATUS.ERROR;
    }

    if (invalidAudienceDeliveryLikelihood(audienceDeliveryLikelihood)) {
      return _constants.DELIVERY_STATUS.INVALID;
    }

    if (!hasMetRequiredMinReach) {
      return _constants.DELIVERY_STATUS.REACH_TOO_LOW;
    }

    if (audienceDeliveryLikelihood <= LOW_VALUE_THRESHOLD) {
      return _constants.DELIVERY_STATUS.LOW_VALUE_THRESHOLD;
    }

    if (audienceDeliveryLikelihood <= MED_VALUE_THRESHOLD) {
      return _constants.DELIVERY_STATUS.MED_VALUE_THRESHOLD;
    }

    return null;
  };

  var deliveryStatus = getDeliveryStatus(audienceDeliveryLikelihood, error, hasMetRequiredMinReach);
  var recommendationIcon = getRecommendationIcon(deliveryStatus);
  var recommendationText = getRecommendationText(deliveryStatus);
  var radialMeterValue = deliveryStatus === _constants.DELIVERY_STATUS.INVALID ? null : audienceDeliveryLikelihood;
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(StyledPanel, props, getLabelledItem()), /*#__PURE__*/_react["default"].createElement(StyledPanel, props, /*#__PURE__*/_react["default"].createElement(BudgetText, _extends({}, props, {
    variant: _encoreAdvertisingWeb.Type.heading4,
    weight: _encoreAdvertisingWeb.Type.bold
  }), strings.BUDGET_DELIVERY_LABEL), /*#__PURE__*/_react["default"].createElement(MeterContainer, null, /*#__PURE__*/_react["default"].createElement(_RadialMeter["default"], {
    fill: deliveryStatus === _constants.DELIVERY_STATUS.INVALID ? _encoreAdvertisingWeb.gray60 : 'url(#stoplight-gradient)',
    highLabel: strings.HIGH_LABEL,
    lowLabel: strings.LOW_LABEL,
    value: hasMetRequiredMinReach ? radialMeterValue : 0
  }, /*#__PURE__*/_react["default"].createElement("linearGradient", {
    id: "stoplight-gradient",
    x1: "0%",
    y1: "0%",
    y2: "0%"
  }, /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "0%",
    stopColor: "#FF4632"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "50%",
    stopColor: "#F8E95A"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "80%",
    stopColor: "#F8E95A"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "85%",
    stopColor: "#CDF564"
  }), /*#__PURE__*/_react["default"].createElement("stop", {
    offset: "100%",
    stopColor: "#1ED760"
  })))), /*#__PURE__*/_react["default"].createElement(_LabelledItem.RecommendationText, _extends({}, props, {
    labelIcon: recommendationIcon,
    labelText: recommendationText.Label,
    styled: !!recommendationText,
    variant: recommendationText ? _encoreAdvertisingWeb.Type.body3 : _encoreAdvertisingWeb.Type.body2,
    weight: _encoreAdvertisingWeb.Type.book
  }), recommendationText.Body)));
};

exports.BudgetPanel = BudgetPanel;
BudgetPanel.propTypes = {
  audienceDeliveryLikelihood: _propTypes["default"].number,
  cost: _propTypes["default"].number,
  costModel: _propTypes["default"].oneOf([_constants.IMPRESSIONS_MODEL, _constants.PAID_LISTENS_MODEL]),
  currencyFormatter: _propTypes["default"].func.isRequired,
  error: _propTypes["default"].string,
  hasMetRequiredMinReach: _propTypes["default"].bool,
  isVideoFormat: _propTypes["default"].bool,
  isPodcastFormat: _propTypes["default"].bool,
  serveOnMegaphone: _propTypes["default"].bool,
  strings: _propTypes["default"].shape({
    BUDGET_DELIVERY_LABEL: _constants.BUDGET_DELIVERY_LABEL,
    LOW_OR_MED_VALUE_MESSAGE: _constants.LOW_OR_MED_VALUE_MESSAGE,
    RECOMMENDATIONTEXT: _constants.RECOMMENDATIONTEXT,
    IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT: _constants.IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT,
    REACH_COST_LABEL_TEXT: _constants.REACH_COST_LABEL_TEXT,
    PER_LISTEN: _constants.PER_LISTEN,
    HIGH_LABEL: _constants.HIGH_LABEL,
    LOW_LABEL: _constants.LOW_LABEL,
    PER_VIEW: _constants.PER_VIEW,
    CPM_LABEL: _constants.CPM_LABEL,
    CPM_TOOLTIP: _constants.CPM_TOOLTIP,
    MAXIMUM_CPM_LABEL: _constants.MAXIMUM_CPM_LABEL,
    MAXIMUM_CPM_TOOLTIP: _constants.MAXIMUM_CPM_TOOLTIP,
    LOW_OR_MED_VALUE_PODCAST_MESSAGE: _constants.LOW_OR_MED_VALUE_PODCAST_MESSAGE
  })
};
BudgetPanel.defaultProps = {
  strings: {
    BUDGET_DELIVERY_LABEL: _constants.BUDGET_DELIVERY_LABEL,
    LOW_OR_MED_VALUE_MESSAGE: _constants.LOW_OR_MED_VALUE_MESSAGE,
    RECOMMENDATIONTEXT: _constants.RECOMMENDATIONTEXT,
    IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT: _constants.IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT,
    REACH_COST_LABEL_TEXT: _constants.REACH_COST_LABEL_TEXT,
    PER_VIEW: _constants.PER_VIEW,
    PER_LISTEN: _constants.PER_LISTEN,
    HIGH_LABEL: _constants.HIGH_LABEL,
    LOW_LABEL: _constants.LOW_LABEL,
    CPM_LABEL: _constants.CPM_LABEL,
    CPM_TOOLTIP: _constants.CPM_TOOLTIP,
    MAXIMUM_CPM_LABEL: _constants.MAXIMUM_CPM_LABEL,
    MAXIMUM_CPM_TOOLTIP: _constants.MAXIMUM_CPM_TOOLTIP,
    LOW_OR_MED_VALUE_PODCAST_MESSAGE: _constants.LOW_OR_MED_VALUE_PODCAST_MESSAGE
  },
  format: _FormatType.Format.AUDIO,
  serveOnMegaphone: false
};
var _default = BudgetPanel;
exports["default"] = _default;
