// @spotify-internal/adstudio-shared/lib/components/AudienceSizerLegacy/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MAXIMUM_CPM_TOOLTIP = exports.MAXIMUM_CPM_LABEL = exports.CPM_TOOLTIP = exports.CPM_LABEL = exports.LOW_LABEL = exports.HIGH_LABEL = exports.PER_VIEW = exports.PER_LISTEN = exports.REACH_COST_LABEL_TEXT = exports.RECOMMENDATIONTEXT = exports.LESS_THAN_PREFIX = exports.GREATER_THAN_PREFIX = exports.LOW_OR_MED_VALUE_PODCAST_MESSAGE = exports.LOW_OR_MED_VALUE_MESSAGE = exports.INSUFFICIENT_RANGE_TEXT = exports.PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE = exports.PLACEHOLDER_MESSAGE = exports.DEFAULT_AUDIENCE_ESTIMATE_ERROR = exports.DEFAULT_PRICER_ERROR = exports.PAID_LISTENS_MODEL_ADS_VIEWED_TOOLTIP_TEXT = exports.ESTIMATED_FREQUENCY_TOOLTIP_TEXT = exports.ESTIMATED_DAILY_REACH_TOOLTIP_TEXT = exports.IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT = exports.PAID_LISTENS_MODEL_ADS_SERVED_TOOLTIP_TEXT = exports.PAID_LISTENS_MODEL_ADS_VIEWED_LABEL = exports.BUDGET_DELIVERY_LABEL = exports.ESTIMATED_FREQUENCY_LABEL = exports.ESTIMATED_DAILY_REACH_LABEL = exports.PAID_LISTENS_MODEL_ADS_SERVED_LABEL = exports.IMPRESSIONS_MODEL_ADS_SERVED_LABEL = exports.PRICER_FORMAT = exports.DELIVERY_STATUS = exports.PAID_LISTENS_MODEL = exports.IMPRESSIONS_MODEL = void 0;
var IMPRESSIONS_MODEL = 'CPM';
exports.IMPRESSIONS_MODEL = IMPRESSIONS_MODEL;
var PAID_LISTENS_MODEL = 'CPCL';
exports.PAID_LISTENS_MODEL = PAID_LISTENS_MODEL;
var DELIVERY_STATUS = {
  ERROR: 'ERROR',
  INVALID: 'INVALID',
  REACH_TOO_LOW: 'REACH_TOO_LOW',
  LOW_VALUE_THRESHOLD: 'LOW_VALUE_THRESHOLD',
  MED_VALUE_THRESHOLD: 'MED_VALUE_THRESHOLD'
};
exports.DELIVERY_STATUS = DELIVERY_STATUS;
var PRICER_FORMAT = {
  AUDIO: 'AUDIO',
  VIDEO: 'VIDEO'
}; // Everything below here can be overriden by string injection

exports.PRICER_FORMAT = PRICER_FORMAT;
var IMPRESSIONS_MODEL_ADS_SERVED_LABEL = 'Estimated ads served';
exports.IMPRESSIONS_MODEL_ADS_SERVED_LABEL = IMPRESSIONS_MODEL_ADS_SERVED_LABEL;
var PAID_LISTENS_MODEL_ADS_SERVED_LABEL = 'Estimated ad listens';
exports.PAID_LISTENS_MODEL_ADS_SERVED_LABEL = PAID_LISTENS_MODEL_ADS_SERVED_LABEL;
var ESTIMATED_DAILY_REACH_LABEL = 'Estimated reach';
exports.ESTIMATED_DAILY_REACH_LABEL = ESTIMATED_DAILY_REACH_LABEL;
var ESTIMATED_FREQUENCY_LABEL = 'Estimated frequency';
exports.ESTIMATED_FREQUENCY_LABEL = ESTIMATED_FREQUENCY_LABEL;
var BUDGET_DELIVERY_LABEL = 'Budget delivery likelihood';
exports.BUDGET_DELIVERY_LABEL = BUDGET_DELIVERY_LABEL;
var PAID_LISTENS_MODEL_ADS_VIEWED_LABEL = "Estimated ad views";
exports.PAID_LISTENS_MODEL_ADS_VIEWED_LABEL = PAID_LISTENS_MODEL_ADS_VIEWED_LABEL;
var PAID_LISTENS_MODEL_ADS_SERVED_TOOLTIP_TEXT = 'Your cost per listen increases as you add more targeting selections.';
exports.PAID_LISTENS_MODEL_ADS_SERVED_TOOLTIP_TEXT = PAID_LISTENS_MODEL_ADS_SERVED_TOOLTIP_TEXT;
var IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT = 'Your cost per ad served increases as you add more targeting selections.';
exports.IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT = IMPRESSIONS_MODEL_ADS_SERVED_TOOLTIP_TEXT;
var ESTIMATED_DAILY_REACH_TOOLTIP_TEXT = 'The estimated number of unique users who will be served your ad at least once.';
exports.ESTIMATED_DAILY_REACH_TOOLTIP_TEXT = ESTIMATED_DAILY_REACH_TOOLTIP_TEXT;
var ESTIMATED_FREQUENCY_TOOLTIP_TEXT = 'The estimated average number of times each user will be served your ad.';
exports.ESTIMATED_FREQUENCY_TOOLTIP_TEXT = ESTIMATED_FREQUENCY_TOOLTIP_TEXT;
var PAID_LISTENS_MODEL_ADS_VIEWED_TOOLTIP_TEXT = 'Your cost per view increases as you add more targeting selections.';
exports.PAID_LISTENS_MODEL_ADS_VIEWED_TOOLTIP_TEXT = PAID_LISTENS_MODEL_ADS_VIEWED_TOOLTIP_TEXT;
var DEFAULT_PRICER_ERROR = 'We’re unable to get pricing information. Check your connection or try reloading the page.';
exports.DEFAULT_PRICER_ERROR = DEFAULT_PRICER_ERROR;
var DEFAULT_AUDIENCE_ESTIMATE_ERROR = 'We’re unable to estimate your audience. Check your connection or try reloading the page.';
exports.DEFAULT_AUDIENCE_ESTIMATE_ERROR = DEFAULT_AUDIENCE_ESTIMATE_ERROR;
var PLACEHOLDER_MESSAGE = 'Complete required fields';
exports.PLACEHOLDER_MESSAGE = PLACEHOLDER_MESSAGE;
var PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE = "\n- Choose a format\n- Select platforms\n- Set your budget\n- Select your targeting location\n";
exports.PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE = PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE;
var INSUFFICIENT_RANGE_TEXT = 'Insufficient';
exports.INSUFFICIENT_RANGE_TEXT = INSUFFICIENT_RANGE_TEXT;
var ERROR_LABEL = 'Unable to estimate audience';
var ERROR_MESSAGE = 'Please check your connection or try reloading the page.';
var LOW_RECOMMENDATION_LABEL = 'Not likely to be spent';
var MED_RECOMMENDATION_LABEL = 'May not be spent';
var LOW_OR_MED_VALUE_MESSAGE = "To increase the likelihood of spending your full budget, do one or more of the following:\n- Extend your schedule\n- Expand your audience\n- Add locations (cities, states, provinces, DMAs, etc.)";
exports.LOW_OR_MED_VALUE_MESSAGE = LOW_OR_MED_VALUE_MESSAGE;
var LOW_OR_MED_VALUE_PODCAST_MESSAGE = "\n- Include more episode content topics\n- Extend your schedule\n- Expand your audience\n- Add locations (cities, states, provinces, DMAs, etc.)\n";
exports.LOW_OR_MED_VALUE_PODCAST_MESSAGE = LOW_OR_MED_VALUE_PODCAST_MESSAGE;
var GOOD_RECOMMENDATION_LABEL = 'Likely to be spent';
var GOOD_VALUE_MESSAGE = 'Based on your audience selections, your budget will likely spend in full.';
var ACCEPTABLE_AUDIENCE_MESSAGE = 'Your selected audience is suited to your budget.';
var REACH_TOO_LOW_LABEL = 'Cannot be spent';
var REACH_TOO_LOW_MESSAGE = 'Broaden your audience to proceed.';
var GREATER_THAN_PREFIX = 'Greater than ';
exports.GREATER_THAN_PREFIX = GREATER_THAN_PREFIX;
var LESS_THAN_PREFIX = 'Less than ';
exports.LESS_THAN_PREFIX = LESS_THAN_PREFIX;
var RECOMMENDATIONTEXT = {
  Error: {
    Label: ERROR_LABEL,
    Body: ERROR_MESSAGE
  },
  NoValue: {
    Label: 'Recommendation',
    Body: null
  },
  LowValue: {
    Label: LOW_RECOMMENDATION_LABEL,
    Body: LOW_OR_MED_VALUE_MESSAGE
  },
  MedValue: {
    Label: MED_RECOMMENDATION_LABEL,
    Body: LOW_OR_MED_VALUE_MESSAGE
  },
  HighValue: {
    Label: GOOD_RECOMMENDATION_LABEL,
    Body: ACCEPTABLE_AUDIENCE_MESSAGE
  },
  ReachTooLow: {
    Label: REACH_TOO_LOW_LABEL,
    Body: REACH_TOO_LOW_MESSAGE
  }
};
exports.RECOMMENDATIONTEXT = RECOMMENDATIONTEXT;
var REACH_COST_LABEL_TEXT = 'Based on your targeting, you can reach this audience for';
exports.REACH_COST_LABEL_TEXT = REACH_COST_LABEL_TEXT;
var PER_LISTEN = 'per listen';
exports.PER_LISTEN = PER_LISTEN;
var PER_VIEW = 'per view';
exports.PER_VIEW = PER_VIEW;
var HIGH_LABEL = 'High';
exports.HIGH_LABEL = HIGH_LABEL;
var LOW_LABEL = 'Low';
exports.LOW_LABEL = LOW_LABEL;
var CPM_LABEL = IMPRESSIONS_MODEL;
exports.CPM_LABEL = CPM_LABEL;
var CPM_TOOLTIP = 'Your CPM increases as you add more targeting selections.';
exports.CPM_TOOLTIP = CPM_TOOLTIP;
var MAXIMUM_CPM_LABEL = 'maximum CPM';
exports.MAXIMUM_CPM_LABEL = MAXIMUM_CPM_LABEL;
var MAXIMUM_CPM_TOOLTIP = 'Maximum CPM is the most you\'ll pay per thousand impressions. Ads that we deliver off Spotify have a lower CPM than ads on Spotify, so your final CPM will vary based on delivery.';
exports.MAXIMUM_CPM_TOOLTIP = MAXIMUM_CPM_TOOLTIP;
