// @spotify-internal/adstudio-shared/lib/components/AudienceSizerLegacy/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.V4_PRICER_URL = void 0;

var _omit2 = _interopRequireDefault(require("lodash/omit"));

var _isEqual2 = _interopRequireDefault(require("lodash/isEqual"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _audienceSizerHelpers = require("../../utils/audienceSizerHelpers");

var _selectors = require("./selectors");

var _BudgetPanel = _interopRequireDefault(require("./BudgetPanel"));

var _AudienceEstimatesPanel = _interopRequireDefault(require("./AudienceEstimatesPanel"));

var _componentHelpers = require("../../utils/componentHelpers");

var audiencePickerStrings = _interopRequireWildcard(require("./constants"));

var _i18n = require("../../config/i18n");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var V4_PRICER_URL = 'adstudiopricer/v4/price';
exports.V4_PRICER_URL = V4_PRICER_URL;

var Container = _styledComponents["default"].div.withConfig({
  displayName: "AudienceSizerLegacy__Container",
  componentId: "okbcx9-0"
})(["position:relative;"]);

var LoadingSpinnerContainer = _styledComponents["default"].div.withConfig({
  displayName: "AudienceSizerLegacy__LoadingSpinnerContainer",
  componentId: "okbcx9-1"
})(["position:absolute;right:", ";top:", ";"], _encoreAdvertisingWeb.spacer4, _encoreAdvertisingWeb.spacer4);

var AudienceSizer = /*#__PURE__*/function (_PureComponent) {
  _inherits(AudienceSizer, _PureComponent);

  var _super = _createSuper(AudienceSizer);

  function AudienceSizer() {
    var _this;

    _classCallCheck(this, AudienceSizer);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.state = _objectSpread({}, _selectors.DEFAULT_STATE);
    _this.setStateAsync = _componentHelpers.setStateAsync;
    return _this;
  }

  _createClass(AudienceSizer, [{
    key: "componentDidMount",
    value: function () {
      var _componentDidMount = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                _context.next = 2;
                return this.fetchAudienceSizerDetails({});

              case 2:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this);
      }));

      function componentDidMount() {
        return _componentDidMount.apply(this, arguments);
      }

      return componentDidMount;
    }()
  }, {
    key: "componentDidUpdate",
    value: function () {
      var _componentDidUpdate = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2(prevProps) {
        return regeneratorRuntime.wrap(function _callee2$(_context2) {
          while (1) {
            switch (_context2.prev = _context2.next) {
              case 0:
                _context2.next = 2;
                return this.fetchAudienceSizerDetails(prevProps);

              case 2:
              case "end":
                return _context2.stop();
            }
          }
        }, _callee2, this);
      }));

      function componentDidUpdate(_x) {
        return _componentDidUpdate.apply(this, arguments);
      }

      return componentDidUpdate;
    }()
  }, {
    key: "fetchAudienceSizerDetails",
    value: function () {
      var _fetchAudienceSizerDetails = _asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee3(prevProps) {
        var _this$props, strings, webgateHost, edgeProxyHost, audienceEstimate, pricing, error, props, payload, _pricing$costMicro, _pricing$model, _props, _payload;

        return regeneratorRuntime.wrap(function _callee3$(_context3) {
          while (1) {
            switch (_context3.prev = _context3.next) {
              case 0:
                _this$props = this.props, strings = _this$props.strings, webgateHost = _this$props.webgateHost, edgeProxyHost = _this$props.edgeProxyHost;
                audienceEstimate = (0, _selectors.getAudienceEstimate)(this.state);
                pricing = (0, _selectors.getPricing)(this.state);

                if (!(JSON.stringify(prevProps) === JSON.stringify(this.props) || (0, _isEqual2["default"])((0, _audienceSizerHelpers.buildPricingRequest)(this.props), (0, _audienceSizerHelpers.buildPricingRequest)(prevProps)) && (0, _isEqual2["default"])((0, _audienceSizerHelpers.buildAudienceEstimateRequest)(this.props, pricing), (0, _audienceSizerHelpers.buildAudienceEstimateRequest)(prevProps, pricing)))) {
                  _context3.next = 5;
                  break;
                }

                return _context3.abrupt("return");

              case 5:
                _context3.next = 7;
                return this.setStateAsync(function () {
                  return {
                    loading: true
                  };
                });

              case 7:
                _context3.prev = 7;
                props = (0, _omit2["default"])(this.props, 'countryCode');
                payload = (0, _audienceSizerHelpers.buildPricingRequest)(_objectSpread(_objectSpread({}, props), {}, {
                  apiHost: webgateHost,
                  pricerEndpoint: V4_PRICER_URL
                }));
                _context3.next = 12;
                return (0, _audienceSizerHelpers.fetchPricing)(payload);

              case 12:
                pricing = _context3.sent;
                _context3.next = 19;
                break;

              case 15:
                _context3.prev = 15;
                _context3.t0 = _context3["catch"](7);
                pricing = null;
                error = strings.DEFAULT_PRICER_ERROR;

              case 19:
                if (!(error || !pricing || !this.props.budget)) {
                  _context3.next = 23;
                  break;
                }

                audienceEstimate = null;
                _context3.next = 35;
                break;

              case 23:
                _context3.prev = 23;
                _props = (0, _omit2["default"])(this.props, 'countryCode');
                _payload = (0, _audienceSizerHelpers.buildAudienceEstimateRequest)(_objectSpread(_objectSpread({}, _props), {}, {
                  edgeProxyHost: edgeProxyHost,
                  currencyCode: pricing.currency,
                  pricing: {
                    costMicros: (_pricing$costMicro = pricing.costMicro) !== null && _pricing$costMicro !== void 0 ? _pricing$costMicro : 0,
                    costModel: (_pricing$model = pricing.model) !== null && _pricing$model !== void 0 ? _pricing$model : strings.IMPRESSIONS_MODEL
                  }
                }));
                _context3.next = 28;
                return (0, _audienceSizerHelpers.fetchAudienceEstimate)(_payload);

              case 28:
                audienceEstimate = _context3.sent;
                _context3.next = 35;
                break;

              case 31:
                _context3.prev = 31;
                _context3.t1 = _context3["catch"](23);
                audienceEstimate = null;
                error = strings.DEFAULT_AUDIENCE_ESTIMATE_ERROR;

              case 35:
                _context3.next = 37;
                return this.setStateAsync(function () {
                  return {
                    loading: false,
                    error: error,
                    audienceEstimate: audienceEstimate,
                    pricing: pricing
                  };
                });

              case 37:
              case "end":
                return _context3.stop();
            }
          }
        }, _callee3, this, [[7, 15], [23, 31]]);
      }));

      function fetchAudienceSizerDetails(_x2) {
        return _fetchAudienceSizerDetails.apply(this, arguments);
      }

      return fetchAudienceSizerDetails;
    }()
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          currencyFormatter = _this$props2.currencyFormatter,
          countryCode = _this$props2.countryCode,
          hasMetRequiredMinReach = _this$props2.hasMetRequiredMinReach,
          format = _this$props2.format,
          strings = _this$props2.strings,
          serveOnMegaphone = _this$props2.serveOnMegaphone;
      return /*#__PURE__*/_react["default"].createElement(Container, null, (0, _selectors.getLoading)(this.state) && /*#__PURE__*/_react["default"].createElement(LoadingSpinnerContainer, null, /*#__PURE__*/_react["default"].createElement(_adstudioTape.LoadingSpinner, {
        diameter: 32
      })), /*#__PURE__*/_react["default"].createElement(_BudgetPanel["default"], {
        audienceDeliveryLikelihood: (0, _selectors.getAudienceDeliveryLikelihood)(this.state),
        cost: (0, _selectors.getCost)(this.state),
        costModel: (0, _selectors.getCostModel)(this.state),
        currencyFormatter: currencyFormatter,
        error: (0, _selectors.getError)(this.state),
        hasMetRequiredMinReach: hasMetRequiredMinReach,
        format: format,
        placeholder: strings.PLACEHOLDER_MESSAGE,
        strings: strings,
        serveOnMegaphone: serveOnMegaphone
      }), /*#__PURE__*/_react["default"].createElement(_AudienceEstimatesPanel["default"], {
        costModel: (0, _selectors.getCostModel)(this.state),
        countryCode: countryCode,
        hasAudienceEstimateFetched: !!(0, _selectors.getAudienceEstimate)(this.state),
        maxAdsServed: (0, _selectors.getMaxAdsServed)(this.state),
        maxReach: (0, _selectors.getMaxReach)(this.state),
        minAdsServed: (0, _selectors.getMinAdsServed)(this.state),
        minReach: (0, _selectors.getMinReach)(this.state),
        minFrequency: (0, _selectors.getMinFrequency)(this.state),
        maxFrequency: (0, _selectors.getMaxFrequency)(this.state),
        hasMetRequiredMinReach: hasMetRequiredMinReach,
        isVideoFormat: format === audiencePickerStrings.PRICER_FORMAT.VIDEO,
        strings: strings
      }));
    }
  }]);

  return AudienceSizer;
}(_react.PureComponent);

exports["default"] = AudienceSizer;
AudienceSizer.propTypes = {
  flightId: _propTypes["default"].string,
  adAccountId: _propTypes["default"].string,
  ageRanges: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    ageMin: _propTypes["default"].number,
    ageMax: _propTypes["default"].number
  })),
  webgateHost: _propTypes["default"].string,
  edgeProxyHost: _propTypes["default"].string,
  artistIds: _propTypes["default"].arrayOf(_propTypes["default"].string),
  budget: _propTypes["default"].number,
  countryCode: _propTypes["default"].string,
  currencyFormatter: _propTypes["default"].func.isRequired,
  dateBegin: _propTypes["default"].string,
  dateEnd: _propTypes["default"].string,
  genders: _propTypes["default"].arrayOf(_propTypes["default"].string),
  genres: _propTypes["default"].arrayOf(_propTypes["default"].string),
  geos: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    id: _propTypes["default"].string,
    type: _propTypes["default"].string,
    geoType: _propTypes["default"].string,
    countryCode: _propTypes["default"].string
  })),
  onAudienceEstimateError: _propTypes["default"].func,
  onAudienceEstimateRequest: _propTypes["default"].func,
  onAudienceEstimateSuccess: _propTypes["default"].func,
  onPricingError: _propTypes["default"].func,
  onPricingRequest: _propTypes["default"].func,
  onPricingSuccess: _propTypes["default"].func,
  platforms: _propTypes["default"].arrayOf(_propTypes["default"].string),
  playlists: _propTypes["default"].arrayOf(_propTypes["default"].string),
  hasMetRequiredMinReach: _propTypes["default"].bool,
  segments: _propTypes["default"].arrayOf(_propTypes["default"].string),
  strings: _propTypes["default"].shape({
    DEFAULT_AUDIENCE_ESTIMATE_ERROR: _propTypes["default"].string,
    DEFAULT_PRICER_ERROR: _propTypes["default"].string
  }),
  isScmCampaign: _propTypes["default"].bool,
  frequencyCaps: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    maxImpressions: _propTypes["default"].number
  })),
  format: _propTypes["default"].string,
  serveOnMegaphone: _propTypes["default"].bool,
  contextualTargetingIds: _propTypes["default"].arrayOf(_propTypes["default"].string),
  contentSafetyIds: _propTypes["default"].arrayOf(_propTypes["default"].string)
};
AudienceSizer.defaultProps = {
  webgateHost: 'https://exp.wg.spotify.com',
  edgeProxyHost: 'https://adstudio-app-api-sandbox.spotify.net',
  countryCode: _i18n.DEFAULT_COUNTRY,
  strings: audiencePickerStrings,
  serveOnMegaphone: false
};
