// @spotify-internal/adstudio-shared/lib/components/AudienceSizerLegacy/LabelledItem.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = LabelledItem;
exports.RecommendationText = exports.StyledListItem = exports.StyledList = exports.ListContainer = void 0;

var _identity2 = _interopRequireDefault(require("lodash/identity"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _constants = require("./constants");

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var StyledLabel = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "LabelledItem__StyledLabel",
  componentId: "zdm51v-0"
})(["display:flex;margin-bottom:", ";margin-top:", ";max-width:90%;&:first-of-type{margin-top:0;}"], _encoreAdvertisingWeb.spacer4, _encoreAdvertisingWeb.spacer12);
var LabelStyles = (0, _styledComponents.css)(["display:inline;height:", ";margin-left:", ";position:relative;top:-2px;"], _encoreAdvertisingWeb.spacer16, _encoreAdvertisingWeb.spacer8);

var LabelIcon = _styledComponents["default"].span.withConfig({
  displayName: "LabelledItem__LabelIcon",
  componentId: "zdm51v-1"
})(["", ";"], LabelStyles);

var LabelTooltip = (0, _styledComponents["default"])(_adstudioTape.TooltipInfo).withConfig({
  displayName: "LabelledItem__LabelTooltip",
  componentId: "zdm51v-2"
})(["", ";"], LabelStyles);
var PlaceHolderText = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "LabelledItem__PlaceHolderText",
  componentId: "zdm51v-3"
})(["color:", ";"], function (props) {
  return props.emptyChildren && _encoreAdvertisingWeb.gray60;
});

var ListContainer = _styledComponents["default"].div.withConfig({
  displayName: "LabelledItem__ListContainer",
  componentId: "zdm51v-4"
})(["padding-left:", ";"], _encoreAdvertisingWeb.spacer16);

exports.ListContainer = ListContainer;

var StyledList = _styledComponents["default"].ul.withConfig({
  displayName: "LabelledItem__StyledList",
  componentId: "zdm51v-5"
})(["padding-bottom:0;"]);

exports.StyledList = StyledList;

var StyledListItem = _styledComponents["default"].li.withConfig({
  displayName: "LabelledItem__StyledListItem",
  componentId: "zdm51v-6"
})(["display:list-item;font-size:inherit;"]);

exports.StyledListItem = StyledListItem;
var RecommendationText = (0, _styledComponents["default"])(LabelledItem).withConfig({
  displayName: "LabelledItem__RecommendationText",
  componentId: "zdm51v-7"
})(["&&{", ";}"], function (props) {
  return props.styled && (0, _styledComponents.css)(["padding-right:", ";display:inline-block;line-height:20px;"], _encoreAdvertisingWeb.spacer24);
});
exports.RecommendationText = RecommendationText;

function LabelledItem(_ref) {
  var children = _ref.children,
      className = _ref.className,
      labelText = _ref.labelText,
      labelIcon = _ref.labelIcon,
      labelTooltip = _ref.labelTooltip,
      placeholder = _ref.placeholder,
      variant = _ref.variant,
      weight = _ref.weight;

  var emptyChildren = _react.Children.toArray(children).every(function (child) {
    return !(0, _identity2["default"])(child);
  });

  return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, /*#__PURE__*/_react["default"].createElement(StyledLabel, {
    condensed: true,
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold
  }, /*#__PURE__*/_react["default"].createElement("span", null, labelText), labelIcon && /*#__PURE__*/_react["default"].createElement(LabelIcon, null, labelIcon), labelTooltip && /*#__PURE__*/_react["default"].createElement(LabelTooltip, {
    className: className,
    placement: "left",
    tooltipText: labelTooltip
  })), /*#__PURE__*/_react["default"].createElement(PlaceHolderText, {
    className: className,
    emptyChildren: emptyChildren,
    condensed: true,
    variant: variant,
    weight: weight
  }, !emptyChildren ? children : /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
    escapeHtml: false,
    renderers: {
      paragraph: function paragraph(props) {
        return /*#__PURE__*/_react["default"].createElement(RecommendationText, _extends({
          variant: _encoreAdvertisingWeb.Type.body3,
          weight: _encoreAdvertisingWeb.Type.book
        }, props));
      },
      list: function list(props) {
        return /*#__PURE__*/_react["default"].createElement(ListContainer, null, ' ', /*#__PURE__*/_react["default"].createElement(StyledList, props), ' ');
      },
      listItem: StyledListItem
    },
    source: placeholder
  })));
}

LabelledItem.propTypes = {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  labelIcon: _propTypes["default"].node,
  labelText: _propTypes["default"].node,
  labelTooltip: _propTypes["default"].node,
  placeholder: _propTypes["default"].node,
  variant: _propTypes["default"].string,
  weight: _propTypes["default"].string
};
LabelledItem.defaultProps = {
  placeholder: _constants.PLACEHOLDER_WITH_SUGGESTIONS_MESSAGE,
  variant: _encoreAdvertisingWeb.Type.body2,
  weight: _encoreAdvertisingWeb.Type.book
};
