// @spotify-internal/adstudio-shared/lib/components/AudienceSizerLegacy/selectors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getMinFrequency = exports.getMaxFrequency = exports.getMinReach = exports.getMaxReach = exports.getAudienceDeliveryLikelihood = exports.getMaxAdsServed = exports.getMinAdsServed = exports.getAudienceEstimate = exports.getCostModel = exports.getCost = exports.getPricing = exports.getLoading = exports.getError = exports.DEFAULT_STATE = void 0;

var _get2 = _interopRequireDefault(require("lodash/get"));

var _numbers = require("../../utils/numbers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var DEFAULT_STATE = {
  error: null,
  loading: false,
  audienceEstimate: null,
  pricing: null
};
exports.DEFAULT_STATE = DEFAULT_STATE;

var getError = function getError(state) {
  return (0, _get2["default"])(state, 'error');
};

exports.getError = getError;

var getLoading = function getLoading(state) {
  return (0, _get2["default"])(state, 'loading');
};

exports.getLoading = getLoading;

var getPricing = function getPricing(state) {
  return (0, _get2["default"])(state, 'pricing');
};

exports.getPricing = getPricing;

var getCost = function getCost(state) {
  return (0, _get2["default"])(getPricing(state), 'cost');
};

exports.getCost = getCost;

var getCostModel = function getCostModel(state) {
  return (0, _get2["default"])(getPricing(state), 'model');
};

exports.getCostModel = getCostModel;

var getAudienceEstimate = function getAudienceEstimate(state) {
  return (0, _get2["default"])(state, 'audienceEstimate');
};

exports.getAudienceEstimate = getAudienceEstimate;

var getMinAdsServed = function getMinAdsServed(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'minAdsServed'));
};

exports.getMinAdsServed = getMinAdsServed;

var getMaxAdsServed = function getMaxAdsServed(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'maxAdsServed'));
};

exports.getMaxAdsServed = getMaxAdsServed;

var getAudienceDeliveryLikelihood = function getAudienceDeliveryLikelihood(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'audienceDeliveryLikelihood'));
};

exports.getAudienceDeliveryLikelihood = getAudienceDeliveryLikelihood;

var getMaxReach = function getMaxReach(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'maxReach'));
};

exports.getMaxReach = getMaxReach;

var getMinReach = function getMinReach(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'minReach'));
};

exports.getMinReach = getMinReach;

var getMaxFrequency = function getMaxFrequency(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'maxFrequency'));
};

exports.getMaxFrequency = getMaxFrequency;

var getMinFrequency = function getMinFrequency(state) {
  return (0, _numbers.toNumber)((0, _get2["default"])(getAudienceEstimate(state), 'minFrequency'));
};

exports.getMinFrequency = getMinFrequency;
