// @spotify-internal/encore-web/es/components/DialogFullScreen/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["banner", "body", "bodyProps", "bodyStyle", "bodyTitle", "dialogTitle", "footer", "navStepper", "onClose", "onEscape", "className", "colorSet", "theme"];
import React, { useEffect } from 'react';
import { renamedPropWarning } from "../../deprecationWarnings";
import { addColorSet } from "../../styles";
import { IconX } from "../Icon/IconX";
import { Body } from "./Body";
import { BodyTitle } from "./BodyTitle";
import { CloseButton } from "./CloseButton";
import { Container } from "./Container";
import { Footer } from "./Footer";
import { FooterNav } from "./FooterNav";
import { Header } from "./Header";
import { HeaderNav } from "./HeaderNav";
import { HeaderTitle } from "./HeaderTitle";
export var DialogFullScreen = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var banner = _ref.banner,
      body = _ref.body,
      bodyProps = _ref.bodyProps,
      bodyStyle = _ref.bodyStyle,
      bodyTitle = _ref.bodyTitle,
      dialogTitle = _ref.dialogTitle,
      footer = _ref.footer,
      navStepper = _ref.navStepper,
      onClose = _ref.onClose,
      onEscape = _ref.onEscape,
      className = _ref.className,
      _ref$colorSet = _ref.colorSet,
      colorSet = _ref$colorSet === void 0 ? 'base' : _ref$colorSet,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    renamedPropWarning('theme', 'colorSet');
  }

  useEffect(function () {
    var onKeydown = function onKeydown(event) {
      if (event.keyCode === 27 && onEscape) {
        onEscape(event);
      }
    };

    window.addEventListener('keydown', onKeydown, true);
    return function cleanup() {
      window.removeEventListener('keydown', onKeydown, true);
    };
  });
  return /*#__PURE__*/React.createElement(Container, Object.assign({
    ref: ref,
    colorSet: colorSet,
    className: addColorSet(colorSet, className)
  }, props), banner, dialogTitle && /*#__PURE__*/React.createElement(Header, {
    colorSet: colorSet
  }, /*#__PURE__*/React.createElement(HeaderTitle, null, dialogTitle), navStepper && /*#__PURE__*/React.createElement(HeaderNav, null, navStepper), onClose && /*#__PURE__*/React.createElement(CloseButton, {
    onClick: onClose,
    "aria-label": "Close",
    iconOnly: IconX
  })), /*#__PURE__*/React.createElement(Body, Object.assign({
    style: bodyStyle,
    colorSet: colorSet
  }, bodyProps), bodyTitle && /*#__PURE__*/React.createElement(BodyTitle, null, bodyTitle), body), /*#__PURE__*/React.createElement(Footer, null, navStepper && /*#__PURE__*/React.createElement(FooterNav, null, navStepper), footer));
});
/**
* **DialogFullScreen** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/DialogFullScreen) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-dialog-dialogfullscreen--default) | [Encore](https://encore.spotify.net/web/components/dialog?format=figma-figma#dialogfullscreen) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:1223) |
*
* DialogFullScreen is a fullscreen view and is used when there's a lot of information or several steps within one Dialog.
*
* @example
* () => (
*  <DialogFullScreen
*    onClose={() => {}}
*    dialogTitle="Dialog Title"
*    bodyTitle="Body Title"
*    body={
*      <Type as="p" style={{ textAlign: 'center' }}>
*        Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.
*      </Type>
*    }
*    footer={
*      <div>
*        <ButtonTertiary buttonSize={ButtonTertiary.sm} condensed>
*          Button
*        </ButtonTertiary>
*        <ButtonPrimary buttonSize={ButtonTertiary.sm}>Button</ButtonPrimary>
*      </div>
*    }
*  />
*);
*
*/

export default DialogFullScreen;
