// @spotify-internal/adstudio-tape/lib/components/PaginationV2/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PaginationV2 = PaginationV2;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _encoreWeb = require("@spotify-internal/encore-web");

var _tokens = require("../../styles/tokens");

var _GridWrapper = _interopRequireDefault(require("../GridWrapper"));

var _GridColumns = _interopRequireDefault(require("../GridColumns"));

var _Icons = _interopRequireDefault(require("../Icons"));

var _onEnter = _interopRequireDefault(require("../../utils/helpers/onEnter"));

var _Item = _interopRequireDefault(require("../Pagination/Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconChevronLeft = _Icons["default"].IconChevronLeft,
    IconChevronRight = _Icons["default"].IconChevronRight,
    IconChevronDown = _Icons["default"].IconChevronDown;
var DEFAULT_PAGE_SIZE_OPTIONS = [{
  key: 5,
  value: 5
}, {
  key: 10,
  value: 10
}, {
  key: 20,
  value: 20
}, {
  key: 50,
  value: 50
}, {
  key: 100,
  value: 100
}];
var StyledItem = (0, _styledComponents["default"])(_Item["default"]).withConfig({
  displayName: "PaginationV2__StyledItem",
  componentId: "ippju-0"
})(["padding:0;"]);

var Spacer = _styledComponents["default"].span.withConfig({
  displayName: "PaginationV2__Spacer",
  componentId: "ippju-1"
})(["display:inline-block;width:10px;"]);

var PaginationList = _styledComponents["default"].ul.withConfig({
  displayName: "PaginationV2__PaginationList",
  componentId: "ippju-2"
})(["font-size:14px;list-style:none;margin:0;padding:0;text-align:right;"]);

var GrayType = (0, _styledComponents["default"])(_encoreWeb.Type).withConfig({
  displayName: "PaginationV2__GrayType",
  componentId: "ippju-3"
})(["color:", ";vertical-align:top;"], _encoreFoundation.gray45);

var PageSizeSection = _styledComponents["default"].div.withConfig({
  displayName: "PaginationV2__PageSizeSection",
  componentId: "ippju-4"
})(["text-align:left;"]);

var StyledSelect = _styledComponents["default"].select.withConfig({
  displayName: "PaginationV2__StyledSelect",
  componentId: "ippju-5"
})(["-webkit-appearance:none;text-indent:0.01px;text-overflow:'';&::-ms-expand{display:none;}border:none;padding:0 27px 7px 10px;background-color:", ";outline:none;font-size:", ";font-weight:", ";letter-spacing:0.025em;line-height:20px;vertical-align:top;"], _encoreFoundation.white, _tokens.fontSizeSmall, _tokens.fontWeightBold);

var StyledIconChevronDown = (0, _styledComponents["default"])(IconChevronDown).withConfig({
  displayName: "PaginationV2__StyledIconChevronDown",
  componentId: "ippju-6"
})(["vertical-align:top;position:relative;left:-24px;color:", ";pointer-events:none;"], _encoreFoundation.gray45);

function PaginationV2(_ref) {
  var offset = _ref.offset,
      limit = _ref.limit,
      handleSelect = _ref.handleSelect,
      handlePageSizeChange = _ref.handlePageSizeChange,
      total = _ref.total,
      _ref$pageSizeOptions = _ref.pageSizeOptions,
      pageSizeOptions = _ref$pageSizeOptions === void 0 ? DEFAULT_PAGE_SIZE_OPTIONS : _ref$pageSizeOptions;

  var handlePrev = function handlePrev() {
    return handleSelect(Math.max(offset - limit, 0));
  };

  var handleNext = function handleNext() {
    return handleSelect(Math.min(offset + limit, total));
  };

  var firstElementNumber = Math.max(offset + 1, 1);
  var lastElementNumber = Math.min(offset + limit, total);
  var selectedPageSizeOption = pageSizeOptions.find(function (opt) {
    return opt.key === limit;
  }) || pageSizeOptions[0];
  return /*#__PURE__*/_react["default"].createElement(_GridWrapper["default"], null, /*#__PURE__*/_react["default"].createElement(_GridColumns["default"], {
    defaultColumnSpanCount: 6
  }, /*#__PURE__*/_react["default"].createElement(PageSizeSection, null, /*#__PURE__*/_react["default"].createElement(GrayType, {
    weight: _encoreWeb.Type.bold,
    variant: _encoreWeb.Type.body2
  }, 'Rows per page: '), /*#__PURE__*/_react["default"].createElement(StyledSelect, {
    onChange: function onChange(e) {
      return handlePageSizeChange(e.target.value);
    }
  }, pageSizeOptions.map(function (opt) {
    return /*#__PURE__*/_react["default"].createElement("option", {
      value: opt.key,
      selected: opt.value === selectedPageSizeOption.value
    }, opt.value);
  })), /*#__PURE__*/_react["default"].createElement(StyledIconChevronDown, {
    iconSize: 16
  }))), /*#__PURE__*/_react["default"].createElement(_GridColumns["default"], {
    defaultColumnSpanCount: 6
  }, /*#__PURE__*/_react["default"].createElement(PaginationList, null, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type, {
    weight: _encoreWeb.Type.bold,
    variant: _encoreWeb.Type.body2
  }, firstElementNumber, " - ", lastElementNumber), /*#__PURE__*/_react["default"].createElement(GrayType, {
    weight: _encoreWeb.Type.bold,
    variant: _encoreWeb.Type.body2
  }, " of ".concat(total)), /*#__PURE__*/_react["default"].createElement(Spacer, null), /*#__PURE__*/_react["default"].createElement(StyledItem, {
    isArrow: true,
    key: "prev",
    "data-disabled": offset === 0
  }, /*#__PURE__*/_react["default"].createElement("a", {
    onClick: handlePrev,
    onKeyPress: (0, _onEnter["default"])(handlePrev),
    role: "button",
    "aria-label": "Previous page",
    tabIndex: "0"
  }, /*#__PURE__*/_react["default"].createElement(IconChevronLeft, {
    iconSize: 16
  }))), /*#__PURE__*/_react["default"].createElement(StyledItem, {
    isArrow: true,
    "data-disabled": offset + limit >= total,
    key: "next"
  }, /*#__PURE__*/_react["default"].createElement("a", {
    onClick: handleNext,
    onKeyPress: (0, _onEnter["default"])(handleNext),
    role: "button",
    "aria-label": "Next page",
    tabIndex: "0"
  }, /*#__PURE__*/_react["default"].createElement(IconChevronRight, {
    iconSize: 16
  }))))));
}

PaginationV2.propTypes = {
  offset: _propTypes["default"].number,
  limit: _propTypes["default"].number,
  handleSelect: _propTypes["default"].func,
  // the call back to be called when a user clicks on a button
  handlePageSizeChange: _propTypes["default"].func,
  // the call back to be called when a user changes the page size
  total: _propTypes["default"].number,
  // number of items
  pageSizeOptions: _propTypes["default"].arrayOf(_propTypes["default"].object)
};
var _default = PaginationV2;
exports["default"] = _default;
