// @spotify-internal/encore-web-v3/es/components/TableHeaderCell/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { TableHeaderCellElement } from "./TableHeaderCell";

function TableHeaderCellComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$align = _ref.align,
      align = _ref$align === void 0 ? 'left' : _ref$align,
      _ref$scope = _ref.scope,
      scope = _ref$scope === void 0 ? 'col' : _ref$scope,
      props = _objectWithoutProperties(_ref, ["theme", "align", "scope"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(TableHeaderCellElement, Object.assign({
    align: align,
    scope: scope
  }, props)));
}

var tableHeaderCellHelpers = {
  darkTheme: {
    tableHeaderCell: 'dark'
  }
};
/**
* **TableHeaderCell** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/TableHeaderCell) | [Storybook](https://encore-web.spotify.net/?path=/story/components-table-tableheadercell--default) | [Encore](https://encore.spotify.net/web/components/table?format=figma-figma#header-cell) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=848:2648) |
*
* Use a TableHeaderCell in table's header row to label columns or enable sorting.
*
* @example
* export const Default = () => (
*  <Table>
*    <thead>
*      <TableRow>
*        <TableHeaderCell>Default</TableHeaderCell>
*        <TableHeaderCell selectable>Selectable</TableHeaderCell>
*        <TableHeaderCell active>Active</TableHeaderCell>
*        <TableHeaderCell align="center">Center-aligned</TableHeaderCell>
*        <TableHeaderCell align="right">Right-aligned</TableHeaderCell>
*      </TableRow>
*    </thead>
*  </Table>
*);
*
*/

export var TableHeaderCell = Object.assign(TableHeaderCellComponent, tableHeaderCellHelpers);
export default TableHeaderCell;
