// @spotify-internal/encore-web/es/components/FormRadio/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { VisuallyHidden } from "../VisuallyHidden";
import { Indicator } from "./Indicator";
import { Label } from "./Label";
import { Radio } from "./Radio";
var FormRadioComponent = React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, ["theme", "children"]);

  return React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Radio, null, React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "radio",
    ref: ref
  }, props)), React.createElement(Indicator, null), children && React.createElement(Label, null, children)));
});
var formRadioHelpers = {
  darkTheme: {
    formRadio: 'dark'
  },
  displayName: 'FormRadio'
};
export var FormRadio = Object.assign(FormRadioComponent, formRadioHelpers);
export default FormRadio;
