// @spotify-internal/adstudio-config/proto/com/spotify/ads/protobuf/annotationValues_pb.js
// source: com/spotify/ads/protobuf/annotationValues.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var google_protobuf_descriptor_pb = require('google-protobuf/google/protobuf/descriptor_pb.js');
goog.object.extend(proto, google_protobuf_descriptor_pb);
goog.exportSymbol('proto.com.spotify.ads.protobuf.EndpointDefinition', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.EndpointDefinition.PatternCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.MessageDefaults', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.MethodDefaults', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.ServiceDefaults', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.MessageDefaults = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.MessageDefaults, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.MessageDefaults.displayName = 'proto.com.spotify.ads.protobuf.MessageDefaults';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.MethodDefaults = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.MethodDefaults, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.MethodDefaults.displayName = 'proto.com.spotify.ads.protobuf.MethodDefaults';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.EndpointDefinition = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_);
};
goog.inherits(proto.com.spotify.ads.protobuf.EndpointDefinition, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.EndpointDefinition.displayName = 'proto.com.spotify.ads.protobuf.EndpointDefinition';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.ServiceDefaults = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.ads.protobuf.ServiceDefaults.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.ServiceDefaults, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.ServiceDefaults.displayName = 'proto.com.spotify.ads.protobuf.ServiceDefaults';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.MessageDefaults.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.MessageDefaults.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.MessageDefaults} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MessageDefaults.toObject = function(includeInstance, msg) {
  var f, obj = {
    wrapper: jspb.Message.getBooleanFieldWithDefault(msg, 1, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.MessageDefaults}
 */
proto.com.spotify.ads.protobuf.MessageDefaults.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.MessageDefaults;
  return proto.com.spotify.ads.protobuf.MessageDefaults.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.MessageDefaults} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.MessageDefaults}
 */
proto.com.spotify.ads.protobuf.MessageDefaults.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setWrapper(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.MessageDefaults.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.MessageDefaults.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.MessageDefaults} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MessageDefaults.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getWrapper();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
};


/**
 * optional bool wrapper = 1;
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.MessageDefaults.prototype.getWrapper = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.ads.protobuf.MessageDefaults} returns this
 */
proto.com.spotify.ads.protobuf.MessageDefaults.prototype.setWrapper = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.MethodDefaults.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.MethodDefaults.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.MethodDefaults} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MethodDefaults.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.MethodDefaults}
 */
proto.com.spotify.ads.protobuf.MethodDefaults.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.MethodDefaults;
  return proto.com.spotify.ads.protobuf.MethodDefaults.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.MethodDefaults} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.MethodDefaults}
 */
proto.com.spotify.ads.protobuf.MethodDefaults.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.MethodDefaults.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.MethodDefaults.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.MethodDefaults} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MethodDefaults.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_ = [[1,2,3,4,5]];

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.PatternCase = {
  PATTERN_NOT_SET: 0,
  GET: 1,
  PUT: 2,
  POST: 3,
  PATCH: 4,
  DELETE: 5
};

/**
 * @return {proto.com.spotify.ads.protobuf.EndpointDefinition.PatternCase}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getPatternCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.EndpointDefinition.PatternCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.EndpointDefinition.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.EndpointDefinition} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.toObject = function(includeInstance, msg) {
  var f, obj = {
    get: jspb.Message.getFieldWithDefault(msg, 1, ""),
    put: jspb.Message.getFieldWithDefault(msg, 2, ""),
    post: jspb.Message.getFieldWithDefault(msg, 3, ""),
    patch: jspb.Message.getFieldWithDefault(msg, 4, ""),
    pb_delete: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.EndpointDefinition;
  return proto.com.spotify.ads.protobuf.EndpointDefinition.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.EndpointDefinition} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setGet(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setPut(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setPost(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setPatch(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setDelete(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.EndpointDefinition.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.EndpointDefinition} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {string} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeString(
      1,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeString(
      2,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeString(
      3,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeString(
      4,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 5));
  if (f != null) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional string get = 1;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getGet = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.setGet = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.clearGet = function() {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.hasGet = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string put = 2;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getPut = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.setPut = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.clearPut = function() {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.hasPut = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string post = 3;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getPost = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.setPost = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.clearPost = function() {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.hasPost = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional string patch = 4;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getPatch = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.setPatch = function(value) {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.clearPatch = function() {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.hasPatch = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional string delete = 5;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.getDelete = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.setDelete = function(value) {
  return jspb.Message.setOneofField(this, 5, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.EndpointDefinition} returns this
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.clearDelete = function() {
  return jspb.Message.setOneofField(this, 5, proto.com.spotify.ads.protobuf.EndpointDefinition.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.EndpointDefinition.prototype.hasDelete = function() {
  return jspb.Message.getField(this, 5) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.ServiceDefaults.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.ServiceDefaults} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.toObject = function(includeInstance, msg) {
  var f, obj = {
    version: jspb.Message.getFieldWithDefault(msg, 1, 0),
    servicenamesList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f,
    enablegrpc: jspb.Message.getBooleanFieldWithDefault(msg, 3, false),
    disableprotorpc: jspb.Message.getBooleanFieldWithDefault(msg, 4, false),
    servicename: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.ServiceDefaults;
  return proto.com.spotify.ads.protobuf.ServiceDefaults.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.ServiceDefaults} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setVersion(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addServicenames(value);
      break;
    case 3:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setEnablegrpc(value);
      break;
    case 4:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setDisableprotorpc(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setServicename(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.ServiceDefaults.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.ServiceDefaults} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getVersion();
  if (f !== 0) {
    writer.writeInt32(
      1,
      f
    );
  }
  f = message.getServicenamesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
  f = message.getEnablegrpc();
  if (f) {
    writer.writeBool(
      3,
      f
    );
  }
  f = message.getDisableprotorpc();
  if (f) {
    writer.writeBool(
      4,
      f
    );
  }
  f = message.getServicename();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional int32 version = 1;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.getVersion = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.setVersion = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * repeated string serviceNames = 2;
 * @return {!Array<string>}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.getServicenamesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.setServicenamesList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.addServicenames = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.clearServicenamesList = function() {
  return this.setServicenamesList([]);
};


/**
 * optional bool enableGrpc = 3;
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.getEnablegrpc = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 3, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.setEnablegrpc = function(value) {
  return jspb.Message.setProto3BooleanField(this, 3, value);
};


/**
 * optional bool disableProtoRpc = 4;
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.getDisableprotorpc = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 4, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.setDisableprotorpc = function(value) {
  return jspb.Message.setProto3BooleanField(this, 4, value);
};


/**
 * optional string serviceName = 5;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.getServicename = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.ServiceDefaults} returns this
 */
proto.com.spotify.ads.protobuf.ServiceDefaults.prototype.setServicename = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


goog.object.extend(exports, proto.com.spotify.ads.protobuf);
