// @spotify-internal/encore-advertising-web/esm/components/Box/index.js
import { spacer16, spacer32, white, gray85, failure, warning, opacity70, } from '@spotify-internal/encore-web';
import styled from 'styled-components';
import { defaultBoxShadow } from '../../styles/shadows';
import { cssOverrides } from '../../utils/helpers/styleOverrides';
function getBorderColor(props) {
    var _a;
    if (props.error)
        return failure;
    if (props.warning)
        return warning;
    if (props.elevated)
        return 'transparent';
    return (_a = props.defaultBorderColor) !== null && _a !== void 0 ? _a : gray85;
}
export const Box = styled.div `
  /* ----------------------------- Default Styles ----------------------------- */
  display: flex;
  position: relative;
  flex-direction: column;
  width: 100%;
  overflow: hidden;
  padding: ${spacer32};
  border-radius: ${spacer16};
  background-color: ${white}; // TODO replace with cssColorValue(semanticColors.backgroundBase);
  border: 1px solid ${props => getBorderColor(props)};
  opacity: ${props => (props.loading ? opacity70 : 1)};
  box-shadow: ${props => (props.elevated ? defaultBoxShadow : 'none')};
  text-align: left;
  overflow: hidden;

  hr {
    margin-left: 0;
    margin-right: 0;
  }

  /* -------- Styled Overrides -------- */
  && {
    ${cssOverrides}
  }
`;
export const BoxGroup = styled(Box) `
  padding: 0;
  overflow: hidden;
  > ${Box} {
    border: none;
    box-shadow: none;
    border-radius: 0;
  }
  > ${Box} + ${Box} {
    border-top: 1px solid ${gray85};
  }
`;
