// @spotify-internal/adstudio-bff-clients/models/com/spotify/adaccountservice/service/model/AdAccountCreationInfo.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AdAccountCreationInfo = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adaccountservice = (function() {
            /**
             * Namespace adaccountservice.
             * @memberof com.spotify
             * @namespace
             */
            var adaccountservice = {};

            adaccountservice.service = (function() {
                /**
                 * Namespace service.
                 * @memberof com.spotify.adaccountservice
                 * @namespace
                 */
                var service = {};

                service.model = (function() {
                    /**
                     * Namespace model.
                     * @memberof com.spotify.adaccountservice.service
                     * @namespace
                     */
                    var model = {};

                    model.AdAccountCreationInfo = (function() {
                        /**
                         * Properties of a AdAccountCreationInfo.
                         * @memberof com.spotify.adaccountservice.service.model
                         * @interface IAdAccountCreationInfo
                         * @property {string|null} [businessName] AdAccountCreationInfo businessName
                         * @property {string|null} [country] AdAccountCreationInfo country
                         * @property {string|null} [businessType] AdAccountCreationInfo businessType
                         * @property {string|null} [industry] AdAccountCreationInfo industry
                         * @property {string|null} [utm] AdAccountCreationInfo utm
                         * @property {string|null} [setBusinessName] AdAccountCreationInfo setBusinessName
                         * @property {string|null} [setCountry] AdAccountCreationInfo setCountry
                         * @property {string|null} [setBusinessType] AdAccountCreationInfo setBusinessType
                         * @property {string|null} [setIndustry] AdAccountCreationInfo setIndustry
                         * @property {string|null} [setUtm] AdAccountCreationInfo setUtm
                         */
                        
                        /**
                         * Constructs a new AdAccountCreationInfo.
                         * @memberof com.spotify.adaccountservice.service.model
                         * @classdesc Represents a AdAccountCreationInfo.
                         * @implements IAdAccountCreationInfo
                         * @constructor
                         * @param {com.spotify.adaccountservice.service.model.IAdAccountCreationInfo=} [properties] Properties to set
                         */
                        function AdAccountCreationInfo(properties) {
                            if (properties)
                                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                    if (properties[keys[i]] != null)
                                        this[keys[i]] = properties[keys[i]];
                        }


                        /**
                         * AdAccountCreationInfo business_name.
                         * @member {string} business_name
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         */
                        AdAccountCreationInfo.prototype.businessName = "";

                        /**
                         * AdAccountCreationInfo country.
                         * @member {string} country
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         */
                        AdAccountCreationInfo.prototype.country = "";

                        /**
                         * AdAccountCreationInfo business_type.
                         * @member {string} business_type
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         */
                        AdAccountCreationInfo.prototype.businessType = "";

                        /**
                         * AdAccountCreationInfo industry.
                         * @member {string} industry
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         */
                        AdAccountCreationInfo.prototype.industry = "";

                        /**
                         * AdAccountCreationInfo utm.
                         * @member {string} utm
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         */
                        AdAccountCreationInfo.prototype.utm = "";


                        /**
                         * Sets business_name
                         * @function setBusinessName
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.AdAccountCreationInfo} AdAccountCreationInfo
                         */
                        AdAccountCreationInfo.prototype.setBusinessName = function(value) {
                            this['businessName'] = value;
                            return this;
                        };

                        /**
                         * Sets country
                         * @function setCountry
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.AdAccountCreationInfo} AdAccountCreationInfo
                         */
                        AdAccountCreationInfo.prototype.setCountry = function(value) {
                            this['country'] = value;
                            return this;
                        };

                        /**
                         * Sets business_type
                         * @function setBusinessType
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.AdAccountCreationInfo} AdAccountCreationInfo
                         */
                        AdAccountCreationInfo.prototype.setBusinessType = function(value) {
                            this['businessType'] = value;
                            return this;
                        };

                        /**
                         * Sets industry
                         * @function setIndustry
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.AdAccountCreationInfo} AdAccountCreationInfo
                         */
                        AdAccountCreationInfo.prototype.setIndustry = function(value) {
                            this['industry'] = value;
                            return this;
                        };

                        /**
                         * Sets utm
                         * @function setUtm
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.AdAccountCreationInfo} AdAccountCreationInfo
                         */
                        AdAccountCreationInfo.prototype.setUtm = function(value) {
                            this['utm'] = value;
                            return this;
                        };

                        /**
                         * Converts the AdAccountCreationInfo to an object
                         * @function toObject
                         * @memberof com.spotify.adaccountservice.service.model.AdAccountCreationInfo
                         * @instance
                         * @returns {object} object
                         */
                        AdAccountCreationInfo.prototype.toObject = function() {
                            return JSON.parse(JSON.stringify(this));
                        };


                        return AdAccountCreationInfo;
                    })();
                    
                    AdAccountCreationInfo = model.AdAccountCreationInfo;
                    return model;
                })();

                return service;
            })();

            return adaccountservice;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["AdAccountCreationInfo"] = AdAccountCreationInfo;
exports["default"] = $root;
