// @spotify-internal/adstudio-bff-clients/models/com/spotify/adaccountservice/service/model/UserAccountSignUp.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var UserAccountSignUp = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.adaccountservice = (function() {
            /**
             * Namespace adaccountservice.
             * @memberof com.spotify
             * @namespace
             */
            var adaccountservice = {};

            adaccountservice.service = (function() {
                /**
                 * Namespace service.
                 * @memberof com.spotify.adaccountservice
                 * @namespace
                 */
                var service = {};

                service.model = (function() {
                    /**
                     * Namespace model.
                     * @memberof com.spotify.adaccountservice.service
                     * @namespace
                     */
                    var model = {};

                    model.UserAccountSignUp = (function() {
                        /**
                         * Properties of a UserAccountSignUp.
                         * @memberof com.spotify.adaccountservice.service.model
                         * @interface IUserAccountSignUp
                         * @property {string|null} [firstName] UserAccountSignUp firstName
                         * @property {string|null} [lastName] UserAccountSignUp lastName
                         * @property {string|null} [businessName] UserAccountSignUp businessName
                         * @property {string|null} [businessEmail] UserAccountSignUp businessEmail
                         * @property {string|null} [country] UserAccountSignUp country
                         * @property {string|null} [businessType] UserAccountSignUp businessType
                         * @property {string|null} [industry] UserAccountSignUp industry
                         * @property {number|null} [timezoneOffsetMillis] UserAccountSignUp timezoneOffsetMillis
                         * @property {boolean|null} [emailSubscription] UserAccountSignUp emailSubscription
                         * @property {string|null} [taxId] UserAccountSignUp taxId
                         * @property {string|null} [adStudioLead] UserAccountSignUp adStudioLead
                         * @property {string|null} [utm] UserAccountSignUp utm
                         * @property {string|null} [businessWebsite] UserAccountSignUp businessWebsite
                         * @property {string|null} [setFirstName] UserAccountSignUp setFirstName
                         * @property {string|null} [setLastName] UserAccountSignUp setLastName
                         * @property {string|null} [setBusinessName] UserAccountSignUp setBusinessName
                         * @property {string|null} [setBusinessEmail] UserAccountSignUp setBusinessEmail
                         * @property {string|null} [setCountry] UserAccountSignUp setCountry
                         * @property {string|null} [setBusinessType] UserAccountSignUp setBusinessType
                         * @property {string|null} [setIndustry] UserAccountSignUp setIndustry
                         * @property {number|null} [setTimezoneOffsetMillis] UserAccountSignUp setTimezoneOffsetMillis
                         * @property {boolean|null} [setEmailSubscription] UserAccountSignUp setEmailSubscription
                         * @property {string|null} [setTaxId] UserAccountSignUp setTaxId
                         * @property {string|null} [setAdStudioLead] UserAccountSignUp setAdStudioLead
                         * @property {string|null} [setUtm] UserAccountSignUp setUtm
                         * @property {string|null} [setBusinessWebsite] UserAccountSignUp setBusinessWebsite
                         */
                        
                        /**
                         * Constructs a new UserAccountSignUp.
                         * @memberof com.spotify.adaccountservice.service.model
                         * @classdesc Represents a UserAccountSignUp.
                         * @implements IUserAccountSignUp
                         * @constructor
                         * @param {com.spotify.adaccountservice.service.model.IUserAccountSignUp=} [properties] Properties to set
                         */
                        function UserAccountSignUp(properties) {
                            if (properties)
                                for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                    if (properties[keys[i]] != null)
                                        this[keys[i]] = properties[keys[i]];
                        }


                        /**
                         * UserAccountSignUp first_name.
                         * @member {string} first_name
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.firstName = "";

                        /**
                         * UserAccountSignUp last_name.
                         * @member {string} last_name
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.lastName = "";

                        /**
                         * UserAccountSignUp business_name.
                         * @member {string} business_name
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.businessName = "";

                        /**
                         * UserAccountSignUp business_email.
                         * @member {string} business_email
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.businessEmail = "";

                        /**
                         * UserAccountSignUp country.
                         * @member {string} country
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.country = "";

                        /**
                         * UserAccountSignUp business_type.
                         * @member {string} business_type
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.businessType = "";

                        /**
                         * UserAccountSignUp industry.
                         * @member {string} industry
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.industry = "";

                        /**
                         * UserAccountSignUp timezone_offset_millis.
                         * @member {number} timezone_offset_millis
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.timezoneOffsetMillis = 0;

                        /**
                         * UserAccountSignUp email_subscription.
                         * @member {boolean} email_subscription
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.emailSubscription = false;

                        /**
                         * UserAccountSignUp tax_id.
                         * @member {string} tax_id
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.taxId = "";

                        /**
                         * UserAccountSignUp ad_studio_lead.
                         * @member {string} ad_studio_lead
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.adStudioLead = "";

                        /**
                         * UserAccountSignUp utm.
                         * @member {string} utm
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.utm = "";

                        /**
                         * UserAccountSignUp business_website.
                         * @member {string} business_website
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         */
                        UserAccountSignUp.prototype.businessWebsite = "";


                        /**
                         * Sets first_name
                         * @function setFirstName
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setFirstName = function(value) {
                            this['firstName'] = value;
                            return this;
                        };

                        /**
                         * Sets last_name
                         * @function setLastName
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setLastName = function(value) {
                            this['lastName'] = value;
                            return this;
                        };

                        /**
                         * Sets business_name
                         * @function setBusinessName
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setBusinessName = function(value) {
                            this['businessName'] = value;
                            return this;
                        };

                        /**
                         * Sets business_email
                         * @function setBusinessEmail
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setBusinessEmail = function(value) {
                            this['businessEmail'] = value;
                            return this;
                        };

                        /**
                         * Sets country
                         * @function setCountry
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setCountry = function(value) {
                            this['country'] = value;
                            return this;
                        };

                        /**
                         * Sets business_type
                         * @function setBusinessType
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setBusinessType = function(value) {
                            this['businessType'] = value;
                            return this;
                        };

                        /**
                         * Sets industry
                         * @function setIndustry
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setIndustry = function(value) {
                            this['industry'] = value;
                            return this;
                        };

                        /**
                         * Sets timezone_offset_millis
                         * @function setTimezoneOffsetMillis
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {number} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setTimezoneOffsetMillis = function(value) {
                            this['timezoneOffsetMillis'] = value;
                            return this;
                        };

                        /**
                         * Sets email_subscription
                         * @function setEmailSubscription
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {boolean} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setEmailSubscription = function(value) {
                            this['emailSubscription'] = value;
                            return this;
                        };

                        /**
                         * Sets tax_id
                         * @function setTaxId
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setTaxId = function(value) {
                            this['taxId'] = value;
                            return this;
                        };

                        /**
                         * Sets ad_studio_lead
                         * @function setAdStudioLead
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setAdStudioLead = function(value) {
                            this['adStudioLead'] = value;
                            return this;
                        };

                        /**
                         * Sets utm
                         * @function setUtm
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setUtm = function(value) {
                            this['utm'] = value;
                            return this;
                        };

                        /**
                         * Sets business_website
                         * @function setBusinessWebsite
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @param {string} value
                         * @instance
                         * @returns {com.spotify.adaccountservice.service.model.UserAccountSignUp} UserAccountSignUp
                         */
                        UserAccountSignUp.prototype.setBusinessWebsite = function(value) {
                            this['businessWebsite'] = value;
                            return this;
                        };

                        /**
                         * Converts the UserAccountSignUp to an object
                         * @function toObject
                         * @memberof com.spotify.adaccountservice.service.model.UserAccountSignUp
                         * @instance
                         * @returns {object} object
                         */
                        UserAccountSignUp.prototype.toObject = function() {
                            return JSON.parse(JSON.stringify(this));
                        };


                        return UserAccountSignUp;
                    })();
                    
                    UserAccountSignUp = model.UserAccountSignUp;
                    return model;
                })();

                return service;
            })();

            return adaccountservice;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["UserAccountSignUp"] = UserAccountSignUp;
exports["default"] = $root;
