// @spotify-internal/ubi-sdk-s4a-web-access-email-validation-timeout-dialog/WebAccessEmailValidationTimeoutDialogEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAccessEmailValidationTimeoutDialogEventFactory = exports.createWebAccessEmailValidationTimeoutDialogEventFactory = void 0;
var SPECIFICATION_ID = "web-access-email-validation-timeout-dialog";
var APP = "s4a";
var SPECIFICATION_VERSION = "1.0.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.2.0";
function createWebAccessEmailValidationTimeoutDialogEventFactory(options) {
    return new WebAccessEmailValidationTimeoutDialogEventFactory(options);
}
exports.createWebAccessEmailValidationTimeoutDialogEventFactory = createWebAccessEmailValidationTimeoutDialogEventFactory;
/**
 * the user sees a dialog letting them know their email validation token has expired
 */
var WebAccessEmailValidationTimeoutDialogEventFactory = /** @class */ (function () {
    function WebAccessEmailValidationTimeoutDialogEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of OkButtonEventFactory
    */
    WebAccessEmailValidationTimeoutDialogEventFactory.prototype.okButtonFactory = function () {
        return new WebAccessEmailValidationTimeoutDialogEventFactory.OkButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebAccessEmailValidationTimeoutDialogEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebAccessEmailValidationTimeoutDialogEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    /**
     * Create impression for WebAccessEmailValidationTimeoutDialogEventFactory.
     */
    WebAccessEmailValidationTimeoutDialogEventFactory.prototype.impression = function () {
        var data = {
            app: APP,
            generatorVersion: GENERATOR_VERSION,
            specificationVersion: SPECIFICATION_VERSION,
            specificationMode: SPECIFICATION_MODE,
            annotatorConfigurationVersion: '',
            annotatorVersion: '',
            location: this._getCurrentLocation(),
            parentAbsoluteLocation: this._parentAbsoluteLocation
        };
        return data;
    };
    return WebAccessEmailValidationTimeoutDialogEventFactory;
}());
exports.WebAccessEmailValidationTimeoutDialogEventFactory = WebAccessEmailValidationTimeoutDialogEventFactory;
(function (WebAccessEmailValidationTimeoutDialogEventFactory) {
    /**
     * Create OkButtonEventFactory class with path
     * @param path to this node
     */
    var OkButtonEventFactory = /** @class */ (function () {
        function OkButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "ok_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        OkButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        OkButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit logout action.
         */
        OkButtonEventFactory.prototype.hitLogout = function () {
            var data = {
                actionName: "logout",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return OkButtonEventFactory;
    }());
    WebAccessEmailValidationTimeoutDialogEventFactory.OkButtonEventFactory = OkButtonEventFactory;
})(WebAccessEmailValidationTimeoutDialogEventFactory = exports.WebAccessEmailValidationTimeoutDialogEventFactory || (exports.WebAccessEmailValidationTimeoutDialogEventFactory = {}));
exports.WebAccessEmailValidationTimeoutDialogEventFactory = WebAccessEmailValidationTimeoutDialogEventFactory;
