// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/CountrySearch/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CountryTargetingSearch = CountryTargetingSearch;
exports["default"] = exports.strings = exports.stringsShape = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reduxForm = require("redux-form");

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _validators = require("@spotify-internal/adstudio-web-utils/lib/validators");

var _validators2 = require("../../../utils/validators");

var _ReduxFormAutosuggest = _interopRequireDefault(require("../../ReduxFormAutosuggest"));

var _Suggestion = _interopRequireDefault(require("./Suggestion"));

var _Value = _interopRequireDefault(require("./Value"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _constants = require("../constants");

var _api = require("../api");

var _i18n = require("../../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

/* eslint-disable */
var CountrySearchWrapper = _styledComponents["default"].div.withConfig({
  displayName: "CountrySearch__CountrySearchWrapper",
  componentId: "sc-11pbc8b-0"
})([""]);

var stringsShape = {
  COUNTRY_LABEL: _propTypes["default"].string,
  COUNTRY_TOOLTIP_DESCRIPTION: _propTypes["default"].string,
  COUNTRY_PLACEHOLDER: _propTypes["default"].string,
  COUNTRY_ERROR_MESSAGE: _propTypes["default"].string
};
exports.stringsShape = stringsShape;
var strings = {
  COUNTRY_LABEL: _constants.COUNTRY_LABEL,
  COUNTRY_TOOLTIP_DESCRIPTION: _constants.COUNTRY_TOOLTIP_DESCRIPTION,
  COUNTRY_PLACEHOLDER: _constants.COUNTRY_PLACEHOLDER,
  COUNTRY_ERROR_MESSAGE: _constants.COUNTRY_ERROR_MESSAGE
};
exports.strings = strings;

function CountryTargetingSearch(_ref) {
  var disableCountry = _ref.disableCountry,
      handleRemoveCountry = _ref.handleRemoveCountry,
      className = _ref.className,
      excludeCountries = _ref.excludeCountries,
      strings = _ref.strings,
      keyField = _ref.keyField,
      language = _ref.language;
  var validator = (0, _validators2.withCustomMessage)(_validators.required, strings.COUNTRY_ERROR_MESSAGE);
  return /*#__PURE__*/_react["default"].createElement(CountrySearchWrapper, {
    className: className
  }, /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
    id: "country-autosuggest",
    name: "targeting.country",
    keyField: keyField || strings.KEY_FIELD,
    debounced: true,
    inlineValue: true,
    component: _ReduxFormAutosuggest["default"],
    exclude: function exclude(item) {
      return excludeCountries.includes(item.countryCode);
    },
    onInputChange: function onInputChange(query) {
      return (0, _api.countrySearch)(query, language);
    },
    placeholder: strings.COUNTRY_PLACEHOLDER,
    label: /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
      placement: "leftBottom",
      tooltipText: strings.COUNTRY_TOOLTIP_DESCRIPTION
    }, strings.COUNTRY_LABEL),
    renderSuggestion: function renderSuggestion(suggestionProps) {
      return /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], suggestionProps);
    },
    renderValue: function renderValue(props) {
      var disabled = props.disabled,
          _props$value = props.value,
          value = _props$value === void 0 ? [] : _props$value,
          handleRemove = props.handleRemove,
          keyField = props.keyField,
          valueField = props.valueField;
      var item = value[0] || {};
      return /*#__PURE__*/_react["default"].createElement(_Value["default"], {
        disabled: disabled || disableCountry,
        item: item,
        onClick: function onClick() {
          if (!disableCountry) {
            handleRemove(item);
            handleRemoveCountry();
          }
        },
        keyField: keyField,
        key: item[keyField],
        valueField: valueField
      });
    },
    suggestionsLimit: Infinity,
    validate: [validator]
  }));
}

CountryTargetingSearch.propTypes = {
  excludeCountries: _propTypes["default"].arrayOf(_propTypes["default"].string),
  keyField: _propTypes["default"].string,
  language: _propTypes["default"].string.isRequired,
  string: _propTypes["default"].shape(stringsShape)
};
CountryTargetingSearch.defaultProps = {
  excludeCountries: [],
  keyField: _constants.KEY_FIELD,
  language: _i18n.DEFAULT_LANGUAGE,
  strings: strings
};
var _default = CountryTargetingSearch;
exports["default"] = _default;
