// @spotify-internal/adstudio-tape/lib/components/Pagination/Items.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icons = _interopRequireDefault(require("../Icons"));

var _onEnter = _interopRequireDefault(require("../../utils/helpers/onEnter"));

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var IconMore = _Icons["default"].IconMore;

var Items = /*#__PURE__*/function (_Component) {
  _inherits(Items, _Component);

  var _super = _createSuper(Items);

  function Items() {
    var _this;

    _classCallCheck(this, Items);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "handleSelectPage", function (page) {
      return function () {
        return _this.props.handleSelect(page);
      };
    });

    _defineProperty(_assertThisInitialized(_this), "items", function (startPage, endPage) {
      var currentPage = _this.props.currentPage;
      var itemsList = [];

      for (var page = startPage; page <= endPage; page += 1) {
        itemsList.push( /*#__PURE__*/_react["default"].createElement(_Item["default"], {
          "data-current-page": currentPage === page,
          key: page
        }, /*#__PURE__*/_react["default"].createElement("a", {
          onClick: _this.handleSelectPage(page),
          onKeyPress: (0, _onEnter["default"])(_this.handleSelectPage(page)),
          role: "button",
          tabIndex: "0"
        }, page)));
      }

      return itemsList;
    });

    _defineProperty(_assertThisInitialized(_this), "firstItem", function () {
      return /*#__PURE__*/_react["default"].createElement(_Item["default"], {
        key: 1
      }, /*#__PURE__*/_react["default"].createElement("a", {
        onClick: _this.handleSelectPage(1),
        onKeyPress: (0, _onEnter["default"])(_this.handleSelectPage(1)),
        role: "button",
        tabIndex: "0"
      }, "1"));
    });

    return _this;
  }

  _createClass(Items, [{
    key: "numberOfPagesToDisplay",
    get:
    /**
     * Calculate the number of pages to display
     * @param {Number} currentPage
     * @param {Number} numPages
     * @param {Number} numPagesBeforeEllipsis
     */
    function get() {
      var _this$props = this.props,
          numPagesBeforeEllipsis = _this$props.numPagesBeforeEllipsis,
          numPages = _this$props.numPages,
          currentPage = _this$props.currentPage;
      var startPage = 2;
      var endPage = 0;

      if (numPagesBeforeEllipsis && numPagesBeforeEllipsis < numPages) {
        startPage = Math.max(Math.min(currentPage - Math.floor(numPagesBeforeEllipsis / 2, 10), numPages - numPagesBeforeEllipsis + 1), 1);
        endPage = startPage + numPagesBeforeEllipsis - 1;
      } else {
        startPage = 1;
        endPage = numPages;
      }

      return {
        startPage: startPage,
        endPage: endPage
      };
    }
  }, {
    key: "render",
    value: function render() {
      var numPages = this.props.numPages;
      var _this$numberOfPagesTo = this.numberOfPagesToDisplay,
          startPage = _this$numberOfPagesTo.startPage,
          endPage = _this$numberOfPagesTo.endPage;
      return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, startPage > 1 && this.firstItem(), startPage > 2 && /*#__PURE__*/_react["default"].createElement(_Item["default"], {
        key: "ellipsis-1",
        isEllipsis: true
      }, /*#__PURE__*/_react["default"].createElement(IconMore, {
        size: 16
      })), this.items(startPage, endPage), endPage < numPages - 1 && /*#__PURE__*/_react["default"].createElement(_Item["default"], {
        key: "ellipsis-2",
        isEllipsis: true
      }, /*#__PURE__*/_react["default"].createElement(IconMore, {
        size: 16
      })), endPage < numPages && /*#__PURE__*/_react["default"].createElement(_Item["default"], {
        key: numPages
      }, /*#__PURE__*/_react["default"].createElement("a", {
        onClick: this.handleSelectPage(numPages),
        onKeyPress: (0, _onEnter["default"])(this.handleSelectPage(numPages)),
        role: "button",
        tabIndex: "0"
      }, numPages)));
    }
  }]);

  return Items;
}(_react.Component);

exports["default"] = Items;

_defineProperty(Items, "propTypes", {
  currentPage: _propTypes["default"].number.isRequired,
  numPages: _propTypes["default"].number.isRequired,
  handleSelect: _propTypes["default"].func.isRequired,
  numPagesBeforeEllipsis: _propTypes["default"].number
});

_defineProperty(Items, "defaultProps", {
  numPagesBeforeEllipsis: 5
});

;
