// @spotify-internal/encore-web/es/components/TableThumbnail/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["img", "imgAlt", "thumbnailTitle", "subtitle", "small", "truncate", "condensed", "theme"];
import React from 'react';
import { deprecatedPropWarning } from "../../deprecationWarnings";
import { Container } from "./Container";
import { Image } from "./Image";
import { Subtitle } from "./Subtitle";
import { Title } from "./Title";
import { Wrapper } from "./Wrapper";
export var TableThumbnail = function TableThumbnail(_ref) {
  var img = _ref.img,
      imgAlt = _ref.imgAlt,
      thumbnailTitle = _ref.thumbnailTitle,
      subtitle = _ref.subtitle,
      small = _ref.small,
      truncate = _ref.truncate,
      condensed = _ref.condensed,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    deprecatedPropWarning('theme', 'This component is intended to inherit its parent’s color set or color.');
  }

  return /*#__PURE__*/React.createElement(Container, props, img && /*#__PURE__*/React.createElement(Image, {
    src: img,
    alt: imgAlt,
    small: small
  }), /*#__PURE__*/React.createElement(Wrapper, {
    condensed: condensed
  }, thumbnailTitle && /*#__PURE__*/React.createElement(Title, {
    truncate: truncate
  }, thumbnailTitle), subtitle && /*#__PURE__*/React.createElement(Subtitle, {
    truncate: truncate
  }, subtitle)));
};
/**
* **TableThumbnail** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/TableThumbnail) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-table-tablethumbnail--default) | [Encore](https://encore.spotify.net/web/components/table?format=figma-figma#thumbnail-cell) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=561:135) |
*
* The TableThumbnail cell displays an image at the beginning of a row.
*
* @example
* () => (
*  <Table>
*    <thead>
*      <TableRow>
*        <TableHeaderCell>Header</TableHeaderCell>
*      </TableRow>
*    </thead>
*    <tbody>
*      <TableRow>
*        <TableCell>
*          <TableThumbnail
*            img="https://i.scdn.co/image/a8f3fe2c87edf4f67acb629d832e65846c451afc"
*            imgAlt="Image Title"
*            thumbnailTitle="Title"
*            subtitle={
*              <div>
*                <Type as="p">Optional text</Type>
*                <Type as="p">Additional text</Type>
*              </div>
*            }
*          />
*        </TableCell>
*      </TableRow>
*      <TableRow>
*        <TableCell>
*          <TableThumbnail
*            img="https://i.scdn.co/image/a8f3fe2c87edf4f67acb629d832e65846c451afc"
*            imgAlt="Image Title"
*            thumbnailTitle="Title"
*            subtitle={
*              <div>
*                <Type as="p">Optional text</Type>
*                <Type as="p">Additional text</Type>
*              </div>
*            }
*          />
*        </TableCell>
*      </TableRow>
*    </tbody>
*  </Table>
*);
*
*/

export default TableThumbnail;
