// @spotify-internal/ubi-sdk-s4a-web-team-member-details-form/WebTeamMemberDetailsFormEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTeamMemberDetailsFormEventFactory = exports.createWebTeamMemberDetailsFormEventFactory = void 0;
var SPECIFICATION_ID = "web-team-member-details-form";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.1.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.24";
function createWebTeamMemberDetailsFormEventFactory(options) {
    return new WebTeamMemberDetailsFormEventFactory(options);
}
exports.createWebTeamMemberDetailsFormEventFactory = createWebTeamMemberDetailsFormEventFactory;
/**
 *
 */
var WebTeamMemberDetailsFormEventFactory = /** @class */ (function () {
    function WebTeamMemberDetailsFormEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of FirstNameTextboxEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.firstNameTextboxFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.FirstNameTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of LastNameTextboxEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.lastNameTextboxFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.LastNameTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of BusinessEmailTextboxEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.businessEmailTextboxFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.BusinessEmailTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of RoleDropdownEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.roleDropdownFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.RoleDropdownEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CompanyTextboxEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.companyTextboxFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.CompanyTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of AdminRadioButtonEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.adminRadioButtonFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.AdminRadioButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of EditorRadioButtonEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.editorRadioButtonFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.EditorRadioButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ReaderRadioButtonEventFactory
    */
    WebTeamMemberDetailsFormEventFactory.prototype.readerRadioButtonFactory = function () {
        return new WebTeamMemberDetailsFormEventFactory.ReaderRadioButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebTeamMemberDetailsFormEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebTeamMemberDetailsFormEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebTeamMemberDetailsFormEventFactory;
}());
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create FirstNameTextboxEventFactory class with path
     * @param path to this node
     */
    var FirstNameTextboxEventFactory = /** @class */ (function () {
        function FirstNameTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "first_name_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        FirstNameTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        FirstNameTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        FirstNameTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return FirstNameTextboxEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.FirstNameTextboxEventFactory = FirstNameTextboxEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create LastNameTextboxEventFactory class with path
     * @param path to this node
     */
    var LastNameTextboxEventFactory = /** @class */ (function () {
        function LastNameTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "last_name_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LastNameTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LastNameTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        LastNameTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LastNameTextboxEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.LastNameTextboxEventFactory = LastNameTextboxEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create BusinessEmailTextboxEventFactory class with path
     * @param path to this node
     */
    var BusinessEmailTextboxEventFactory = /** @class */ (function () {
        function BusinessEmailTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "business_email_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        BusinessEmailTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        BusinessEmailTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        BusinessEmailTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return BusinessEmailTextboxEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.BusinessEmailTextboxEventFactory = BusinessEmailTextboxEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create RoleDropdownEventFactory class with path
     * @param path to this node
     */
    var RoleDropdownEventFactory = /** @class */ (function () {
        function RoleDropdownEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "role_dropdown"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the RoleOptionsEventFactory
        * @returns A new instance of RoleOptionsEventFactory
        */
        RoleDropdownEventFactory.prototype.roleOptionsFactory = function (data) {
            return new WebTeamMemberDetailsFormEventFactory.RoleDropdownEventFactory.RoleOptionsEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        RoleDropdownEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        RoleDropdownEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        RoleDropdownEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return RoleDropdownEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.RoleDropdownEventFactory = RoleDropdownEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create CompanyTextboxEventFactory class with path
     * @param path to this node
     */
    var CompanyTextboxEventFactory = /** @class */ (function () {
        function CompanyTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "company_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        CompanyTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CompanyTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        CompanyTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CompanyTextboxEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.CompanyTextboxEventFactory = CompanyTextboxEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create AdminRadioButtonEventFactory class with path
     * @param path to this node
     */
    var AdminRadioButtonEventFactory = /** @class */ (function () {
        function AdminRadioButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "admin_radio_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        AdminRadioButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AdminRadioButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_select action.
         */
        AdminRadioButtonEventFactory.prototype.hitUiSelect = function () {
            var data = {
                actionName: "ui_select",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AdminRadioButtonEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.AdminRadioButtonEventFactory = AdminRadioButtonEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create EditorRadioButtonEventFactory class with path
     * @param path to this node
     */
    var EditorRadioButtonEventFactory = /** @class */ (function () {
        function EditorRadioButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "editor_radio_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        EditorRadioButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        EditorRadioButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_select action.
         */
        EditorRadioButtonEventFactory.prototype.hitUiSelect = function () {
            var data = {
                actionName: "ui_select",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return EditorRadioButtonEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.EditorRadioButtonEventFactory = EditorRadioButtonEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    /**
     * Create ReaderRadioButtonEventFactory class with path
     * @param path to this node
     */
    var ReaderRadioButtonEventFactory = /** @class */ (function () {
        function ReaderRadioButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "reader_radio_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        ReaderRadioButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ReaderRadioButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_select action.
         */
        ReaderRadioButtonEventFactory.prototype.hitUiSelect = function () {
            var data = {
                actionName: "ui_select",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ReaderRadioButtonEventFactory;
    }());
    WebTeamMemberDetailsFormEventFactory.ReaderRadioButtonEventFactory = ReaderRadioButtonEventFactory;
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
(function (WebTeamMemberDetailsFormEventFactory) {
    var RoleDropdownEventFactory;
    (function (RoleDropdownEventFactory) {
        /**
         * Create RoleOptionsEventFactory class with path
         * @param path to this node
         */
        var RoleOptionsEventFactory = /** @class */ (function () {
            function RoleOptionsEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "role_options" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            RoleOptionsEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            RoleOptionsEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_select action.
             */
            RoleOptionsEventFactory.prototype.hitUiSelect = function () {
                var data = {
                    actionName: "ui_select",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return RoleOptionsEventFactory;
        }());
        RoleDropdownEventFactory.RoleOptionsEventFactory = RoleOptionsEventFactory;
    })(RoleDropdownEventFactory = WebTeamMemberDetailsFormEventFactory.RoleDropdownEventFactory || (WebTeamMemberDetailsFormEventFactory.RoleDropdownEventFactory = {}));
})(WebTeamMemberDetailsFormEventFactory = exports.WebTeamMemberDetailsFormEventFactory || (exports.WebTeamMemberDetailsFormEventFactory = {}));
exports.WebTeamMemberDetailsFormEventFactory = WebTeamMemberDetailsFormEventFactory;
