// @spotify-internal/encore-web/es/components/FormHelpText/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { failure, gray50, spacer4, spacer8 } from '@spotify-internal/encore-foundation';
import styled, { css, ThemeProvider } from 'styled-components';
import { rootStyle, typeStyleBody2 } from "../../styles";
import { IconX } from "../Icon/IconX";
var IconXForText = styled(IconX).withConfig({
  displayName: "FormHelpText__IconXForText",
  componentId: "e48exm-0"
})(["height:1em;width:1em;vertical-align:-0.175em;margin-right:", ";"], spacer4);
var Help = styled.div.withConfig({
  displayName: "FormHelpText__Help",
  componentId: "e48exm-1"
})(["", ";", ";margin-top:", ";", ";", ";", ";"], rootStyle(), typeStyleBody2(), spacer8, function (props) {
  return props.error && css(["color:", ";"], failure);
}, function (props) {
  return !props.error && css(["color:", ";"], gray50);
}, function (props) {
  return props.maxChars && css(["font-feature-settings:'tnum';text-align:right;"]);
});

function FormHelpTextComponent(_ref) {
  var error = _ref.error,
      children = _ref.children,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["error", "children", "theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Help, Object.assign({
    error: error
  }, props), error && /*#__PURE__*/React.createElement(IconXForText, {
    "aria-label": "Error:"
  }), children));
}

var formHelpTextHelpers = {
  darkTheme: {
    formHelpText: 'dark'
  }
};
/**
* **FormHelpText** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/FormHelpText) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-formhelptext--default) | [Encore](https://encore.spotify.net/web/components/form?format=figma-figma#formhelptext) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:866) |
*
* FormHelpText is placed after a form field and can display contextual information, character count or error notifications.
*
* @example
* export const Default = () => <FormHelpText>Helpful information</FormHelpText>;
*
*/

export var FormHelpText = Object.assign(FormHelpTextComponent, formHelpTextHelpers);
export default FormHelpText;
