// @spotify-internal/encore-web/es/components/NavPillListItem/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["label", "active", "isHover", "isFocus", "component", "theme"];
import React, { useContext } from 'react';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { deprecatedPropWarning } from "../../deprecationWarnings";
import { Link } from "./Link";
import { ListItem } from "./ListItem";
export var NavPillListItem = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var label = _ref.label,
      active = _ref.active,
      isHover = _ref.isHover,
      isFocus = _ref.isFocus,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    deprecatedPropWarning('theme', 'This component is intended to inherit its parent’s color set or color.');
  }

  var tabIndexAttribute = active ? null : {
    tabIndex: '-1'
  };

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ListItem, {
    role: "presentation"
  }, /*#__PURE__*/React.createElement(Link, Object.assign({
    ref: ref,
    role: "tab"
  }, tabIndexAttribute, {
    isActive: active,
    isHover: isHover,
    isFocus: isFocus,
    component: component,
    "aria-selected": active,
    isUsingKeyboard: isUsingKeyboard
  }, props), label));
});
/**
* **NavPillListItem** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/NavPillListItem) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-navpill-navpilllistitem--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-pill) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1550) |
*
* An individual tab within the NavPill component
*
* @example
* () => (
*  <NavPillList>
*    <NavPillListItem label="Overview" href="#NavPill-overview" />
*  </NavPillList>
*);
*
*/

export default NavPillListItem;
