// reactochart/MarkerLineChart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _isUndefined = _interopRequireDefault(require("lodash/isUndefined"));

var _first = _interopRequireDefault(require("lodash/first"));

var _last = _interopRequireDefault(require("lodash/last"));

var _clamp = _interopRequireDefault(require("lodash/clamp"));

var _isFunction = _interopRequireDefault(require("lodash/isFunction"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _util = require("./util.js");

var CustomPropTypes = _interopRequireWildcard(require("./utils/CustomPropTypes"));

var _Data = require("./utils/Data");

var _Scale = require("./utils/Scale");

var _xyPropsEqual = _interopRequireDefault(require("./utils/xyPropsEqual"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function getTickType(props) {
  const {
    xEnd,
    yEnd,
    horizontal
  } = props; // warn if a range is passed for the dependent variable, which is expected to be a value

  if (!horizontal && !(0, _isUndefined.default)(yEnd) || horizontal && !(0, _isUndefined.default)(xEnd)) {
    // eslint-disable-next-line no-console
    console.warn('Warning: MarkerLineChart can only show the independent variable as a range, not the dependent variable.');
  }

  if (!horizontal && !(0, _isUndefined.default)(xEnd) || horizontal && !(0, _isUndefined.default)(yEnd)) return 'RangeValue';
  return 'ValueValue';
}
/**
 * `MarkerLineChart` is similar to a bar chart,
 * except that it just draws a line at the data value, rather than a full bar.
 * If the independent variable is a range, the length of the line will represent that range,
 * otherwise all lines will be the same length.
 * The dependent variable must be a single value, not a range.
 */


class MarkerLineChart extends _react.default.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "onMouseEnterLine", (e, d) => {
      this.props.onMouseEnterLine(e, d);
    });

    _defineProperty(this, "onMouseMoveLine", (e, d) => {
      this.props.onMouseMoveLine(e, d);
    });

    _defineProperty(this, "onMouseLeaveLine", (e, d) => {
      this.props.onMouseLeaveLine(e, d);
    });

    _defineProperty(this, "renderRangeValueLine", (d, i) => {
      const [onMouseEnter, onMouseMove, onMouseLeave] = ['onMouseEnterLine', 'onMouseMoveLine', 'onMouseLeaveLine'].map(eventName => {
        // partially apply this line's data point as 2nd callback argument
        const callback = (0, _util.methodIfFuncProp)(eventName, this.props, this);
        return (0, _isFunction.default)(callback) ? (0, _util.bindTrailingArgs)(callback, d) : null;
      });
      const {
        x,
        xEnd,
        y,
        yEnd,
        horizontal,
        xScale,
        yScale,
        lineClassName,
        lineStyle
      } = this.props;
      const xVal = xScale((0, _Data.makeAccessor2)(x)(d));
      const yVal = yScale((0, _Data.makeAccessor2)(y)(d));
      const xEndVal = (0, _isUndefined.default)(xEnd) ? 0 : xScale((0, _Data.makeAccessor2)(xEnd)(d));
      const yEndVal = (0, _isUndefined.default)(yEnd) ? 0 : yScale((0, _Data.makeAccessor2)(yEnd)(d));
      const [x1, y1] = [xVal, yVal];
      const x2 = horizontal ? xVal : xEndVal;
      const y2 = horizontal ? yEndVal : yVal;
      const key = `marker-line-${i}`;
      if (![x1, x2, y1, y2].every(val => val !== null && isFinite(val))) return null;
      return /*#__PURE__*/_react.default.createElement("line", _extends({
        className: `${(0, _Data.getValue)(lineClassName, d, i)}`,
        style: (0, _Data.getValue)(lineStyle, d, i)
      }, {
        x1,
        x2,
        y1,
        y2,
        key,
        onMouseEnter,
        onMouseMove,
        onMouseLeave
      }));
    });

    _defineProperty(this, "renderValueValueLine", (d, i) => {
      const [onMouseEnter, onMouseMove, onMouseLeave] = ['onMouseEnterLine', 'onMouseMoveLine', 'onMouseLeaveLine'].map(eventName => {
        // partially apply this line's data point as 2nd callback argument
        const callback = (0, _util.methodIfFuncProp)(eventName, this.props, this);
        return (0, _isFunction.default)(callback) ? (0, _util.bindTrailingArgs)(callback, d) : null;
      });
      const {
        x,
        y,
        horizontal,
        lineLength,
        xScale,
        yScale,
        lineClassName,
        lineStyle
      } = this.props;
      const xVal = xScale((0, _Data.makeAccessor2)(x)(d));
      const yVal = yScale((0, _Data.makeAccessor2)(y)(d));
      const x1 = !horizontal ? xVal - lineLength / 2 : xVal;
      const x2 = !horizontal ? xVal + lineLength / 2 : xVal;
      const y1 = !horizontal ? yVal : yVal - lineLength / 2;
      const y2 = !horizontal ? yVal : yVal + lineLength / 2;
      const key = `marker-line-${i}`;
      if (![x1, x2, y1, y2].every(val => val !== null && isFinite(val))) return null;
      return /*#__PURE__*/_react.default.createElement("line", _extends({
        className: `${(0, _Data.getValue)(lineClassName, d, i)}`,
        style: (0, _Data.getValue)(lineStyle, d, i)
      }, {
        x1,
        x2,
        y1,
        y2,
        key,
        onMouseEnter,
        onMouseMove,
        onMouseLeave
      }));
    });
  }

  static getSpacing(props) {
    const tickType = getTickType(props); // no spacing for rangeValue marker charts since line start and end are set explicitly

    if (tickType === 'RangeValue') return {
      spacingTop: 0,
      spacingRight: 0,
      spacingBottom: 0,
      spacingLeft: 0
    };
    const {
      lineLength,
      horizontal,
      data,
      xDomain,
      yDomain,
      xScale,
      yScale,
      x,
      y
    } = props;
    const P = lineLength / 2; // padding

    const markDomain = horizontal ? yDomain : xDomain;
    const markScale = horizontal ? yScale : xScale;
    const markAccessor = horizontal ? (0, _Data.makeAccessor2)(y) : (0, _Data.makeAccessor2)(x);
    const markDataDomain = (0, _Data.domainFromData)(data, markAccessor); // find the edges of the tick domain, and map them through the scale function

    const [domainHead, domainTail] = [(0, _first.default)(markDomain), (0, _last.default)(markDomain)].map(markScale).sort(); // sort the pixel values return by the domain extents
    // find the edges of the data domain, and map them through the scale function

    const [dataDomainHead, dataDomainTail] = [(0, _first.default)(markDataDomain), (0, _last.default)(markDataDomain)].map(markScale).sort(); // sort the pixel values return by the domain extents
    // find the necessary spacing (based on bar width) to push the bars completely inside the tick domain

    const [spacingTail, spacingHead] = [(0, _clamp.default)(P - (domainTail - dataDomainTail), 0, P), (0, _clamp.default)(P - (dataDomainHead - domainHead), 0, P)];

    if (horizontal) {
      return {
        spacingTop: spacingHead,
        spacingBottom: spacingTail,
        spacingLeft: 0,
        spacingRight: 0
      };
    }

    return {
      spacingTop: 0,
      spacingBottom: 0,
      spacingLeft: spacingHead,
      spacingRight: spacingTail
    };
  }

  static getDomain(props) {
    if (getTickType(props) === 'RangeValue') {
      // set range domain for range type
      const {
        data,
        x,
        xEnd,
        y,
        yEnd,
        xScaleType,
        yScaleType,
        horizontal
      } = props; // only have to specify range axis domain, other axis uses default domainFromData
      // in this chart type, the range axis, if there is one, is always the *independent* variable

      const rangeAxis = horizontal ? 'y' : 'x';
      const rangeStartAccessor = horizontal ? (0, _Data.makeAccessor2)(y) : (0, _Data.makeAccessor2)(x);
      const rangeEndAccessor = horizontal ? (0, _Data.makeAccessor2)(yEnd) : (0, _Data.makeAccessor2)(xEnd);
      const rangeDataType = (0, _Scale.dataTypeFromScaleType)(horizontal ? yScaleType : xScaleType);
      return {
        [`${rangeAxis}Domain`]: (0, _Data.domainFromRangeData)(data, rangeStartAccessor, rangeEndAccessor, rangeDataType)
      };
    }

    return {};
  }

  shouldComponentUpdate(nextProps) {
    const shouldUpdate = !(0, _xyPropsEqual.default)(this.props, nextProps, []);
    return shouldUpdate;
  }

  render() {
    const tickType = getTickType(this.props);
    return /*#__PURE__*/_react.default.createElement("g", {
      className: "rct-marker-line-chart",
      "aria-hidden": "true"
    }, tickType === 'RangeValue' ? this.props.data.map(this.renderRangeValueLine) : this.props.data.map(this.renderValueValueLine));
  }

}

exports.default = MarkerLineChart;

_defineProperty(MarkerLineChart, "propTypes", {
  /**
   * Array of data objects. One marker line will be rendered per datum in the array.
   */
  data: _propTypes.default.array.isRequired,

  /**
   * Accessor function for marker line's X values, called once per line (datum), or a single value to be used for all marker lines.
   * If `horizontal` is `false`, this gets the *independent* variable value on which the line is centered.
   * If `horizontal` is `true`, this gets the start (minimum value) of the *dependent* variable.
   */
  x: CustomPropTypes.valueOrAccessor,

  /**
   * Accessor function for marker line's Y values, called once per line (datum), or a single value to be used for all marker lines.
   * If `horizontal` is `false`, this gets the start (minimum value) of the *dependent* variable.
   * If `horizontal` is `true`, this gets the *independent* variable value on which the line is centered.
   */
  y: CustomPropTypes.valueOrAccessor,

  /**
   * Accessor function for the end (maximum X-values) of the *dependent* variable, which is where the marker line is rendered,
   * or a single value to be used for all marker lines.
   * Should only be passed when `horizontal` is `true` (ignored otherwise).
   */
  xEnd: CustomPropTypes.valueOrAccessor,

  /**
   * Accessor function for the end (maximum Y-values) of the *dependent* variable range which is where the marker line is rendered,
   * or a single value to be used for all marker lines.
   * Should only be passed when `horizontal` is `false` (ignored otherwise).
   */
  yEnd: CustomPropTypes.valueOrAccessor,

  /**
   * Boolean which determines whether the chart will be horizontal.
   * When `true` the X-axis will be treated as the dependent axis.
   */
  horizontal: _propTypes.default.bool,

  /**
   * Length of the marker line.
   */
  lineLength: _propTypes.default.number,

  /**
   * D3 scale type for X axis - provided by XYPlot.
   */
  xScaleType: _propTypes.default.string,

  /**
   * D3 scale type for Y axis - provided by XYPlot.
   */
  yScaleType: _propTypes.default.string,

  /**
   * D3 scale for X axis - provided by XYPlot.
   */
  xScale: _propTypes.default.func,

  /**
   * D3 scale for Y axis - provided by XYPlot.
   */
  yScale: _propTypes.default.func,

  /**
   * Class attribute to be applied to the line path,
   * or an accessor function which returns a class.
   */
  lineClassName: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func]),

  /**
   * Inline style object to be applied to each marker line,
   * or an accessor function which returns a style object.
   */
  lineStyle: _propTypes.default.oneOfType([_propTypes.default.object, _propTypes.default.func]),

  /**
   * `mouseenter` event handler callback, called when user's mouse enters a marker line.
   */
  onMouseEnterLine: _propTypes.default.func,

  /**
   * `mousemove` event handler callback, called when user's mouse moves within a marker line.
   */
  onMouseMoveLine: _propTypes.default.func,

  /**
   * `mouseleave` event handler callback, called when user's mouse leaves a marker line.
   */
  onMouseLeaveLine: _propTypes.default.func
});

_defineProperty(MarkerLineChart, "defaultProps", {
  horizontal: false,
  lineLength: 10,
  lineClassName: '',
  lineStyle: {}
});
