// reactochart/MeasuredValueLabel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _defaults = _interopRequireDefault(require("lodash/defaults"));

var _omit = _interopRequireDefault(require("lodash/omit"));

var _identity = _interopRequireDefault(require("lodash/identity"));

var _measureText = _interopRequireDefault(require("./utils/measureText"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class MeasuredValueLabel extends _react.default.Component {
  static getLabel(props) {
    const {
      value,
      format
    } = props;
    const style = (0, _defaults.default)(props.style, MeasuredValueLabel.defaultProps.style);
    const labelStr = format(value);
    const labelWithStyle = Object.assign({
      text: labelStr
    }, style);
    const measured = (0, _measureText.default)(labelWithStyle);
    return {
      value: props.value,
      text: measured.text,
      height: measured.height.value,
      width: measured.width.value
    };
  }

  render() {
    const {
      value,
      format
    } = this.props;
    const passedProps = (0, _omit.default)(this.props, ['value', 'format']);
    return /*#__PURE__*/_react.default.createElement("text", passedProps, _react.default.Children.count(this.props.children) ? this.props.children : format(value));
  }

}

exports.default = MeasuredValueLabel;

_defineProperty(MeasuredValueLabel, "propTypes", {
  value: _propTypes.default.any,
  format: _propTypes.default.func,
  children: _propTypes.default.any
});

_defineProperty(MeasuredValueLabel, "defaultProps", {
  format: _identity.default,
  style: {
    fontFamily: 'Helvetica, sans-serif',
    fontSize: '20px',
    lineHeight: 1,
    textAnchor: 'middle'
  }
});
