// reactochart/YLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/**
 * `YLine` is a horizontal line rendered on the y axis
 */
class YLine extends _react.default.Component {
  render() {
    const {
      width,
      yScale,
      value,
      xScale,
      xLimit,
      spacingLeft,
      spacingRight,
      style
    } = this.props;
    const className = `rct-chart-line-y ${this.props.className || ''}`;
    const lineY = yScale(value);
    const lineX = typeof xLimit === 'undefined' ? width + spacingRight : xScale(xLimit);
    return /*#__PURE__*/_react.default.createElement("line", _extends({
      "aria-hidden": "true"
    }, {
      x1: -spacingLeft,
      x2: lineX,
      y1: lineY,
      y2: lineY,
      className,
      style
    }));
  }

}

exports.default = YLine;

_defineProperty(YLine, "propTypes", {
  /**
   * D3 scale for Y axis - provided by XYPlot
   */
  yScale: _propTypes.default.func,

  /**
   * Width of chart - provided by XYPlot.
   */
  width: _propTypes.default.number,
  value: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string, _propTypes.default.instanceOf(Date)]).isRequired,

  /**
   * D3 scale for X axis - provided by XYPlot
   */
  xScale: _propTypes.default.func,
  xLimit: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string, _propTypes.default.instanceOf(Date)]),

  /**
   * Spacing left - provided by XYPlot
   */
  spacingLeft: _propTypes.default.number,

  /**
   * Spacing right - provided by XYPlot
   */
  spacingRight: _propTypes.default.number,

  /**
   * Inline style object to be applied to the line
   */
  style: _propTypes.default.object,

  /**
   * Class attribute to be applied to the line
   */
  className: _propTypes.default.string
});

_defineProperty(YLine, "defaultProps", {
  style: {},
  className: '',
  spacingLeft: 0,
  spacingRight: 0
});
