// @spotify-internal/ubi-sdk-s4a-web-bulk-invite/WebBulkInviteEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebBulkInviteEventFactory = exports.createWebBulkInviteEventFactory = void 0;
var SPECIFICATION_ID = "web-bulk-invite";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.1.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.15";
function createWebBulkInviteEventFactory(options) {
    return new WebBulkInviteEventFactory(options);
}
exports.createWebBulkInviteEventFactory = createWebBulkInviteEventFactory;
/**
 *
 */
var WebBulkInviteEventFactory = /** @class */ (function () {
    function WebBulkInviteEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of DownloadTemplateLinkEventFactory
    */
    WebBulkInviteEventFactory.prototype.downloadTemplateLinkFactory = function () {
        return new WebBulkInviteEventFactory.DownloadTemplateLinkEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of UploadCsvFileEventFactory
    */
    WebBulkInviteEventFactory.prototype.uploadCsvFileFactory = function () {
        return new WebBulkInviteEventFactory.UploadCsvFileEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CsvFileParsingErrorEventFactory
    */
    WebBulkInviteEventFactory.prototype.csvFileParsingErrorFactory = function () {
        return new WebBulkInviteEventFactory.CsvFileParsingErrorEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CsvMissingHeadersErrorEventFactory
    */
    WebBulkInviteEventFactory.prototype.csvMissingHeadersErrorFactory = function () {
        return new WebBulkInviteEventFactory.CsvMissingHeadersErrorEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of InvalidFileTypeErrorEventFactory
    */
    WebBulkInviteEventFactory.prototype.invalidFileTypeErrorFactory = function () {
        return new WebBulkInviteEventFactory.InvalidFileTypeErrorEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of InvitationRowEventFactory
    */
    WebBulkInviteEventFactory.prototype.invitationRowFactory = function () {
        return new WebBulkInviteEventFactory.InvitationRowEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of EditInvitationFormEventFactory
    */
    WebBulkInviteEventFactory.prototype.editInvitationFormFactory = function () {
        return new WebBulkInviteEventFactory.EditInvitationFormEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SendButtonEventFactory
    */
    WebBulkInviteEventFactory.prototype.sendButtonFactory = function () {
        return new WebBulkInviteEventFactory.SendButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebBulkInviteEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebBulkInviteEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebBulkInviteEventFactory;
}());
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create DownloadTemplateLinkEventFactory class with path
     * @param path to this node
     */
    var DownloadTemplateLinkEventFactory = /** @class */ (function () {
        function DownloadTemplateLinkEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "download_template_link"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        DownloadTemplateLinkEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        DownloadTemplateLinkEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit download_file action.
         */
        DownloadTemplateLinkEventFactory.prototype.hitDownloadFile = function () {
            var data = {
                actionName: "download_file",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return DownloadTemplateLinkEventFactory;
    }());
    WebBulkInviteEventFactory.DownloadTemplateLinkEventFactory = DownloadTemplateLinkEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create UploadCsvFileEventFactory class with path
     * @param path to this node
     */
    var UploadCsvFileEventFactory = /** @class */ (function () {
        function UploadCsvFileEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "upload_csv_file"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        UploadCsvFileEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        UploadCsvFileEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for drag download_file action.
         */
        UploadCsvFileEventFactory.prototype.dragDownloadFile = function () {
            var data = {
                actionName: "download_file",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "drag",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        /**
         * Create interaction for hit download_file action.
         */
        UploadCsvFileEventFactory.prototype.hitDownloadFile = function () {
            var data = {
                actionName: "download_file",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return UploadCsvFileEventFactory;
    }());
    WebBulkInviteEventFactory.UploadCsvFileEventFactory = UploadCsvFileEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create CsvFileParsingErrorEventFactory class with path
     * @param path to this node
     */
    var CsvFileParsingErrorEventFactory = /** @class */ (function () {
        function CsvFileParsingErrorEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "csv_file_parsing_error"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        CsvFileParsingErrorEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CsvFileParsingErrorEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for CsvFileParsingErrorEventFactory.
         */
        CsvFileParsingErrorEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CsvFileParsingErrorEventFactory;
    }());
    WebBulkInviteEventFactory.CsvFileParsingErrorEventFactory = CsvFileParsingErrorEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create CsvMissingHeadersErrorEventFactory class with path
     * @param path to this node
     */
    var CsvMissingHeadersErrorEventFactory = /** @class */ (function () {
        function CsvMissingHeadersErrorEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "csv_missing_headers_error"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        CsvMissingHeadersErrorEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CsvMissingHeadersErrorEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for CsvMissingHeadersErrorEventFactory.
         */
        CsvMissingHeadersErrorEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CsvMissingHeadersErrorEventFactory;
    }());
    WebBulkInviteEventFactory.CsvMissingHeadersErrorEventFactory = CsvMissingHeadersErrorEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create InvalidFileTypeErrorEventFactory class with path
     * @param path to this node
     */
    var InvalidFileTypeErrorEventFactory = /** @class */ (function () {
        function InvalidFileTypeErrorEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "invalid_file_type_error"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        InvalidFileTypeErrorEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        InvalidFileTypeErrorEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for InvalidFileTypeErrorEventFactory.
         */
        InvalidFileTypeErrorEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return InvalidFileTypeErrorEventFactory;
    }());
    WebBulkInviteEventFactory.InvalidFileTypeErrorEventFactory = InvalidFileTypeErrorEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create InvitationRowEventFactory class with path
     * @param path to this node
     */
    var InvitationRowEventFactory = /** @class */ (function () {
        function InvitationRowEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "invitation_row"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of EditButtonEventFactory
        */
        InvitationRowEventFactory.prototype.editButtonFactory = function () {
            return new WebBulkInviteEventFactory.InvitationRowEventFactory.EditButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        InvitationRowEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        InvitationRowEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return InvitationRowEventFactory;
    }());
    WebBulkInviteEventFactory.InvitationRowEventFactory = InvitationRowEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create EditInvitationFormEventFactory class with path
     * @param path to this node
     */
    var EditInvitationFormEventFactory = /** @class */ (function () {
        function EditInvitationFormEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "edit_invitation_form"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        EditInvitationFormEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        EditInvitationFormEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return EditInvitationFormEventFactory;
    }());
    WebBulkInviteEventFactory.EditInvitationFormEventFactory = EditInvitationFormEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    /**
     * Create SendButtonEventFactory class with path
     * @param path to this node
     */
    var SendButtonEventFactory = /** @class */ (function () {
        function SendButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "send_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        SendButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SendButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit invite_team_member action.
       * @param invite_type the invite_type field
         */
        SendButtonEventFactory.prototype.hitInviteTeamMember = function (_a) {
            var inviteType = _a.inviteType;
            var data = {
                actionParameterNames: ['invite_type'],
                actionParameterValues: [inviteType.toString()],
                actionName: "invite_team_member",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SendButtonEventFactory;
    }());
    WebBulkInviteEventFactory.SendButtonEventFactory = SendButtonEventFactory;
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
(function (WebBulkInviteEventFactory) {
    var InvitationRowEventFactory;
    (function (InvitationRowEventFactory) {
        /**
         * Create EditButtonEventFactory class with path
         * @param path to this node
         */
        var EditButtonEventFactory = /** @class */ (function () {
            function EditButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "edit_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            EditButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            EditButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            EditButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return EditButtonEventFactory;
        }());
        InvitationRowEventFactory.EditButtonEventFactory = EditButtonEventFactory;
    })(InvitationRowEventFactory = WebBulkInviteEventFactory.InvitationRowEventFactory || (WebBulkInviteEventFactory.InvitationRowEventFactory = {}));
})(WebBulkInviteEventFactory = exports.WebBulkInviteEventFactory || (exports.WebBulkInviteEventFactory = {}));
exports.WebBulkInviteEventFactory = WebBulkInviteEventFactory;
