// @spotify-internal/adstudio-shared/lib/components/CreativePreviewContainer/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _FormatType = require("../../config/FormatType");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "CreativePreviewContainer__Container",
  componentId: "sqea9v-0"
})(["width:", ";max-width:", ";"], function (props) {
  return props.width || '271px';
}, function (props) {
  return props.width || '271px';
});

var IconContainer = _styledComponents["default"].div.withConfig({
  displayName: "CreativePreviewContainer__IconContainer",
  componentId: "sqea9v-1"
})(["text-align:center;"]);

var StyledNavBarList = (0, _styledComponents["default"])(_encoreAdvertisingWeb.NavBarList).withConfig({
  displayName: "CreativePreviewContainer__StyledNavBarList",
  componentId: "sqea9v-2"
})(["justify-content:space-evenly;"]);
var StyledNavBar = (0, _styledComponents["default"])(_encoreAdvertisingWeb.NavBar).withConfig({
  displayName: "CreativePreviewContainer__StyledNavBar",
  componentId: "sqea9v-3"
})(["width:100%;", " + div > div{margin-right:0;}"], StyledNavBarList);
var StyledNavBarListItem = (0, _styledComponents["default"])(_encoreAdvertisingWeb.NavBarListItem).withConfig({
  displayName: "CreativePreviewContainer__StyledNavBarListItem",
  componentId: "sqea9v-4"
})(["&:after{height:6px;width:6px;border-radius:100%;}"]);

var StyledNavBarListItemDisabledWrapper = _styledComponents["default"].div.withConfig({
  displayName: "CreativePreviewContainer__StyledNavBarListItemDisabledWrapper",
  componentId: "sqea9v-5"
})(["pointer-events:all;cursor:not-allowed;"]);

var TABS = {
  MOBILE: 'MOBILE',
  DESKTOP: 'DESKTOP'
};
var PODCAST_PREVIEW_IMAGE = 'https://storage.googleapis.com/adstudio-inbox/fdac2c5f-9a36-4a89-a84f-1636331d9d96';

var NavBarListItemWrapper = function NavBarListItemWrapper(_ref) {
  var children = _ref.children,
      disabled = _ref.disabled,
      disabledText = _ref.disabledText;

  if (disabled && disabledText) {
    return /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
      placement: "right",
      icon: false,
      triggerOnMouse: true,
      iconMarginTop: "0",
      tooltipText: disabledText
    }, /*#__PURE__*/_react["default"].createElement(StyledNavBarListItemDisabledWrapper, null, children));
  } else if (disabled) {
    return /*#__PURE__*/_react["default"].createElement(StyledNavBarListItemDisabledWrapper, null, children);
  }

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, children);
};

var CreativePreviewContainer = /*#__PURE__*/function (_React$Component) {
  _inherits(CreativePreviewContainer, _React$Component);

  var _super = _createSuper(CreativePreviewContainer);

  function CreativePreviewContainer() {
    var _this;

    _classCallCheck(this, CreativePreviewContainer);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.state = {
      activeTab: TABS.MOBILE
    };

    _this.toggleActiveTab = function (activeTab) {
      return function () {
        _this.setState({
          activeTab: activeTab
        });
      };
    };

    _this.getPreviewComponent = function (activeTab) {
      var _this$props = _this.props,
          ctaText = _this$props.ctaText,
          videoUrl = _this$props.videoUrl,
          imageUrl = _this$props.imageUrl,
          previewTitle = _this$props.previewTitle,
          showSkipControls = _this$props.showSkipControls,
          brandName = _this$props.brandName,
          tagLine = _this$props.tagLine,
          videoSkipText = _this$props.videoSkipText,
          podcastSponsorMessage = _this$props.podcastSponsorMessage,
          podcastName = _this$props.podcastName,
          aspectRatio = _this$props.aspectRatio,
          format = _this$props.format,
          podcastImageTooltipMessage = _this$props.podcastImageTooltipMessage;
      var isVideoFormat = format === _FormatType.Format.VIDEO;
      var isPodcast = format === _FormatType.Format.AUDIO_PODCAST;
      var imgSrc = isPodcast ? PODCAST_PREVIEW_IMAGE : imageUrl;
      var imgAlt = isPodcast ? 'Podcast Preview' : 'Upload Preview';
      var imageTooltipMessage = isPodcast ? podcastImageTooltipMessage : undefined;

      if (activeTab === TABS.MOBILE) {
        if (isVideoFormat) {
          return /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.VideoMobilePreview, {
            aspectRatio: aspectRatio,
            ctaText: ctaText,
            videoUrl: videoUrl,
            skipText: showSkipControls ? videoSkipText : undefined,
            showPanel: false
          });
        }

        return /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.MobilePreview, {
          title: previewTitle,
          ctaText: ctaText,
          headline: brandName,
          tagLine: tagLine,
          showSkipControls: showSkipControls,
          showPodcastControls: isPodcast,
          showCtaText: !isPodcast,
          showPanel: false,
          podcastSponsorMessage: podcastSponsorMessage,
          podcastName: podcastName,
          imageTooltipMessage: imageTooltipMessage
        }, imgSrc && /*#__PURE__*/_react["default"].createElement("img", {
          alt: imgAlt,
          src: imgSrc,
          width: "245",
          height: "245",
          "data-test": "ta-image-preview"
        }));
      }

      if (activeTab === TABS.DESKTOP) {
        return /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.VideoDesktopPreview, {
          videoUrl: videoUrl,
          imageUrl: imgSrc,
          showPanel: false,
          imageTooltipMessage: imageTooltipMessage,
          title: isPodcast ? podcastSponsorMessage : undefined,
          headline: isPodcast ? podcastName : previewTitle
        });
      }

      return null;
    };

    return _this;
  }

  _createClass(CreativePreviewContainer, [{
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          strings = _this$props2.strings,
          desktopDisabledText = _this$props2.desktopDisabledText,
          desktopDisabled = _this$props2.desktopDisabled,
          mobileDisabled = _this$props2.mobileDisabled,
          mobileDisabledText = _this$props2.mobileDisabledText,
          theme = _this$props2.theme;
      var activeTab = this.state.activeTab;
      var activeIconColor = theme ? theme.colors.primaryColor : _encoreAdvertisingWeb.azure;
      var desktopIconColor = activeTab === TABS.DESKTOP ? activeIconColor : _encoreAdvertisingWeb.black;
      var mobileIconColor = activeTab === TABS.MOBILE ? activeIconColor : _encoreAdvertisingWeb.black;
      if (desktopDisabled) desktopIconColor = _encoreAdvertisingWeb.gray70;
      if (mobileDisabled) mobileIconColor = _encoreAdvertisingWeb.gray70;
      return /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement("div", null, this.getPreviewComponent(activeTab)), /*#__PURE__*/_react["default"].createElement(StyledNavBar, {
        list: /*#__PURE__*/_react["default"].createElement(StyledNavBarList, null, /*#__PURE__*/_react["default"].createElement(NavBarListItemWrapper, {
          disabled: mobileDisabled,
          disabledText: mobileDisabledText
        }, /*#__PURE__*/_react["default"].createElement(StyledNavBarListItem, {
          active: activeTab === TABS.MOBILE,
          disabled: mobileDisabled,
          label: /*#__PURE__*/_react["default"].createElement(IconContainer, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconMobile, {
            iconSize: 32,
            color: mobileIconColor
          }), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
            variant: _encoreAdvertisingWeb.Type.cta4,
            condensed: true,
            color: mobileIconColor
          }, strings.TABS.MOBILE)),
          onClick: this.toggleActiveTab(TABS.MOBILE)
        })), /*#__PURE__*/_react["default"].createElement(NavBarListItemWrapper, {
          disabled: desktopDisabled,
          disabledText: desktopDisabledText
        }, /*#__PURE__*/_react["default"].createElement(StyledNavBarListItem, {
          active: activeTab === TABS.DESKTOP,
          disabled: desktopDisabled,
          label: /*#__PURE__*/_react["default"].createElement(IconContainer, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.IconDevicesComputer, {
            iconSize: 32,
            color: desktopIconColor
          }), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
            variant: _encoreAdvertisingWeb.Type.cta4,
            condensed: true,
            color: desktopIconColor
          }, strings.TABS.DESKTOP)),
          onClick: this.toggleActiveTab(TABS.DESKTOP)
        })))
      }));
    }
  }]);

  return CreativePreviewContainer;
}(_react["default"].Component);

CreativePreviewContainer.defaultProps = {
  strings: {
    TABS: {
      DESKTOP: 'Desktop',
      MOBILE: 'Mobile'
    }
  }
};

var _default = (0, _styledComponents.withTheme)(CreativePreviewContainer);

exports["default"] = _default;
