// @spotify-internal/encore-web/es/components/PaginationDropdownLink/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Link } from "./Link";

function PaginationDropdownLinkComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Link, props));
}

var paginationDropdownLinkHelpers = {
  darkTheme: {
    paginationDropdownLink: 'dark'
  }
};
/**
* **PaginationDropdownLink** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PaginationDropdownLink) | [Storybook](https://encore-web.spotify.net/?path=/story/components-pagination-paginationdropdownlink--default) | [Encore](https://encore.spotify.net/web/components/pagination?format=figma-figma#paginationdropdown) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=2311:140) |
*
* PaginationDropdownLink is a line item link in the PaginationDropdownList
*
* @example
* export const Default = () => <PaginationDropdownLink>100</PaginationDropdownLink>;
*
*/

export var PaginationDropdownLink = Object.assign(PaginationDropdownLinkComponent, paginationDropdownLinkHelpers);
export default PaginationDropdownLink;
