// @spotify-internal/adstudio-shared/lib/components/ArtistTargetingSearch/Suggestion/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _selectors = require("../selectors");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ArtistSuggestion = function ArtistSuggestion(_ref) {
  var parts = _ref.parts,
      _ref$small = _ref.small,
      small = _ref$small === void 0 ? false : _ref$small,
      suggestion = _ref.suggestion;
  var name = suggestion.name,
      id = suggestion.id;
  var imageUrl = (0, _selectors.getSuggestionImageUrl)(suggestion);
  var artistPageUrl = "https://open.spotify.com/artist/".concat(id);

  var openArtistPage = function openArtistPage(e) {
    e.stopPropagation();
    window.open(artistPageUrl, '_blank');
  };

  var ArtistSuggestionContainer = _styledComponents["default"].div.withConfig({
    displayName: "Suggestion__ArtistSuggestionContainer",
    componentId: "tmfeb1-0"
  })(["display:-webkit-box;display:-ms-flexbox;display:flex;padding:", ";margin:0;span:first-child{text-transform:capitalize;}"], small ? '0.5em' : '0.5em 1em 0');

  var MoImageWrapper = _styledComponents["default"].div.withConfig({
    displayName: "Suggestion__MoImageWrapper",
    componentId: "tmfeb1-1"
  })(["height:", ";width:", ";"], small ? '30px' : '50px', small ? '30px' : '50px');

  var MoImage = _styledComponents["default"].div.withConfig({
    displayName: "Suggestion__MoImage",
    componentId: "tmfeb1-2"
  })(["background-position:center;background-size:cover;border-radius:", ";height:100%;width:100%;", ""], small ? '300px' : '500px', function () {
    return typeof imageUrl === 'string' && imageUrl.length > 0 ? "\n        background-image: url(".concat(imageUrl, ");\n      ") : "\n        box-shadow: none;\n        img-circle: true;\n        margin: 0;\n        user-icon-container: true;\n      ";
  });

  var NameWrapper = _styledComponents["default"].div.withConfig({
    displayName: "Suggestion__NameWrapper",
    componentId: "tmfeb1-3"
  })(["", ""], small ? "\n        padding: 0.25em 0 0 0.5em;\n      " : "");

  var Highlight = _styledComponents["default"].span.withConfig({
    displayName: "Suggestion__Highlight",
    componentId: "tmfeb1-4"
  })(["", ""], function (_ref2) {
    var highlight = _ref2.highlight;
    return highlight && "\n        color: ".concat(_encoreAdvertisingWeb.gray75, ";\n        position: relative;\n      ");
  });

  var StyledP = _styledComponents["default"].p.withConfig({
    displayName: "Suggestion__StyledP",
    componentId: "tmfeb1-5"
  })(["padding-bottom:0;"]);

  return /*#__PURE__*/_react["default"].createElement(ArtistSuggestionContainer, {
    "data-test": "artist-autosuggest-option"
  }, /*#__PURE__*/_react["default"].createElement(MoImageWrapper, null, /*#__PURE__*/_react["default"].createElement(MoImage, {
    role: "button",
    onClick: openArtistPage
  })), /*#__PURE__*/_react["default"].createElement(NameWrapper, null, /*#__PURE__*/_react["default"].createElement(StyledP, null, parts && parts.map(function (_ref3, i) {
    var highlight = _ref3.highlight,
        text = _ref3.text;
    return /*#__PURE__*/_react["default"].createElement(Highlight, {
      highlight: highlight,
      key: i
    }, text);
  }), !parts && /*#__PURE__*/_react["default"].createElement("span", null, name))));
};

ArtistSuggestion.propTypes = {
  parts: _propTypes["default"].array,
  small: _propTypes["default"].bool,
  suggestion: _propTypes["default"].shape({
    name: _propTypes["default"].string,
    id: _propTypes["default"].string,
    images: _propTypes["default"].arrayOf(_propTypes["default"].shape({
      url: _propTypes["default"].string
    }))
  })
};
var _default = ArtistSuggestion;
exports["default"] = _default;
