// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/LocationSearch/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LocationSearchByCountry = exports.stringsShape = exports.strings = exports.TargetingLocation = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reduxForm = require("redux-form");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _validators = require("../../../utils/validators");

var _api = require("../api");

var _constants = require("../constants");

var _ReduxFormAutosuggest = _interopRequireDefault(require("../../ReduxFormAutosuggest"));

var _Suggestion = _interopRequireDefault(require("../../GeoTargetingSearch/Suggestion"));

var _Value = _interopRequireDefault(require("../../GeoTargetingSearch/Value"));

var _selectors = require("../../GeoTargetingSearch/selectors");

var _i18n = require("../../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var TargetingLocation = _styledComponents["default"].div.withConfig({
  displayName: "LocationSearch__TargetingLocation",
  componentId: "ak0mbo-0"
})([".location-type{background-clip:content-box;background-color:", ";color:", ";font-weight:400;text-transform:capitalize;p{padding-left:10px;}}.selected-locations-list-item p{margin-bottom:0;margin-top:0;padding-bottom:", ";padding-top:", ";}.selected-locations-list.populated{margin-bottom:10px;overflow:hidden;}.location-name{color:", ";}.dma-geo-type{text-transform:uppercase;}"], _encoreAdvertisingWeb.gray95, _encoreAdvertisingWeb.gray40, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.gray60);

exports.TargetingLocation = TargetingLocation;
var strings = {
  LOCATION_LABEL: _constants.LOCATION_LABEL,
  LOCATION_PLACEHOLDER: _constants.LOCATION_PLACEHOLDER,
  LOCATION_TOOLTIP_DESCRIPTION: _constants.LOCATION_TOOLTIP_DESCRIPTION,
  ARIA_REMOVE_THIS_LOCATION: _constants.ARIA_REMOVE_THIS_LOCATION,
  ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION: _constants.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION,
  CTA_LEARN_MORE: _constants.CTA_LEARN_MORE,
  CTA_REMOVE_ALL_PLURAL: _constants.CTA_REMOVE_ALL_PLURAL,
  CTA_REMOVE_ALL_SINGULAR: _constants.CTA_REMOVE_ALL_SINGULAR,
  DEPRECATED_GEOS_ERROR: _constants.DEPRECATED_GEOS_ERROR,
  LOCATION_UNAVAILABLE_SINGULAR: _constants.LOCATION_UNAVAILABLE_SINGULAR,
  LOCATION_UNAVAILABLE_PLURAL: _constants.LOCATION_UNAVAILABLE_PLURAL
};
exports.strings = strings;
var stringsShape = {
  LOCATION_LABEL: _propTypes["default"].string,
  LOCATION_PLACEHOLDER: _propTypes["default"].string,
  LOCATION_TOOLTIP_DESCRIPTION: _propTypes["default"].string,
  ARIA_REMOVE_THIS_LOCATION: _propTypes["default"].string,
  ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION: _propTypes["default"].string,
  CTA_LEARN_MORE: _propTypes["default"].string,
  CTA_REMOVE_ALL_PLURAL: _propTypes["default"].string,
  CTA_REMOVE_ALL_SINGULAR: _propTypes["default"].string,
  DEPRECATED_GEOS_ERROR: _propTypes["default"].string,
  LOCATION_UNAVAILABLE_SINGULAR: _propTypes["default"].string,
  LOCATION_UNAVAILABLE_PLURAL: _propTypes["default"].string
};
exports.stringsShape = stringsShape;

var LocationSearchByCountry = /*#__PURE__*/function (_React$PureComponent) {
  _inherits(LocationSearchByCountry, _React$PureComponent);

  var _super = _createSuper(LocationSearchByCountry);

  function LocationSearchByCountry() {
    _classCallCheck(this, LocationSearchByCountry);

    return _super.apply(this, arguments);
  }

  _createClass(LocationSearchByCountry, [{
    key: "render",
    value: function render() {
      var _this = this;

      var _this$props = this.props,
          bulkUploaderStrings = _this$props.bulkUploaderStrings,
          className = _this$props.className,
          excludeCountries = _this$props.excludeCountries,
          hideDescription = _this$props.hideDescription,
          disableLocations = _this$props.disableLocations,
          deprecatedGeosFaqLink = _this$props.deprecatedGeosFaqLink,
          handleDeprecatedGeos = _this$props.handleDeprecatedGeos,
          language = _this$props.language,
          limit = _this$props.limit,
          strings = _this$props.strings,
          subdivisionCategories = _this$props.subdivisionCategories,
          keyField = _this$props.keyField,
          geoLocationSearchableTypes = _this$props.geoLocationSearchableTypes;
      var deprecatedGeos = (0, _validators.withCustomMessage)(_validators.deprecatedGeosValidator, strings.DEPRECATED_GEOS_ERROR);
      return /*#__PURE__*/_react["default"].createElement(TargetingLocation, {
        className: className
      }, /*#__PURE__*/_react["default"].createElement(_reduxForm.Field, {
        component: _ReduxFormAutosuggest["default"],
        debounced: true,
        limit: limit,
        exclude: function exclude(item) {
          return excludeCountries.includes(item.countryCode);
        },
        id: "locations-autosuggest",
        label: hideDescription ? undefined : /*#__PURE__*/_react["default"].createElement(_adstudioTape.TooltipInfo, {
          placement: "leftBottom",
          tooltipText: strings.LOCATION_TOOLTIP_DESCRIPTION
        }, strings.LOCATION_LABEL),
        disabled: disableLocations,
        name: "targeting.locations",
        keyField: keyField,
        onInputChange: function onInputChange(query) {
          // Optimization to not bog down the UI by showing every result for "N", for instance
          if (query.length < 2) return;
          var parentCountry = _this.props.parentCountry;

          if (_this.props.tracking && typeof _this.props.tracking.locations === 'function') {
            _this.props.tracking.locations(query);
          }

          return (0, _api.locationByCountrySearch)(query, parentCountry, language, 0, geoLocationSearchableTypes);
        },
        hasLazyLoading: true,
        onFetchMore: function onFetchMore(query, page) {
          var parentCountry = _this.props.parentCountry;
          return (0, _api.locationByCountrySearch)(query, parentCountry, language, page, geoLocationSearchableTypes);
        },
        placeholder: strings.LOCATION_PLACEHOLDER,
        formatSuggestion: _selectors.getGeoLabelFromSuggestion,
        renderSuggestion: function renderSuggestion(props) {
          return /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], _extends({}, props, {
            strings: subdivisionCategories
          }));
        },
        renderValue: function renderValue(_ref) {
          var disabled = _ref.disabled,
              value = _ref.value,
              handleRemove = _ref.handleRemove,
              keyField = _ref.keyField,
              valueField = _ref.valueField;
          return /*#__PURE__*/_react["default"].createElement(_Value["default"], {
            bulkUploaderStrings: bulkUploaderStrings,
            deprecatedGeoStrings: strings,
            disabled: disabled,
            deprecatedGeosFaqLink: deprecatedGeosFaqLink,
            handleDeprecatedGeos: handleDeprecatedGeos,
            items: value,
            keyField: keyField || strings.KEY_FIELD,
            onClick: handleRemove,
            valueField: valueField,
            limit: limit,
            strings: subdivisionCategories
          });
        },
        suggestionsLimit: Infinity,
        validate: [_validators.locationByCountryValidator, deprecatedGeos]
      }));
    }
  }]);

  return LocationSearchByCountry;
}(_react["default"].PureComponent);

exports.LocationSearchByCountry = LocationSearchByCountry;
LocationSearchByCountry.propTypes = {
  bulkUploaderStrings: _propTypes["default"].object,
  excludeCountries: _propTypes["default"].arrayOf(_propTypes["default"].string),
  hideDescription: _propTypes["default"].bool,
  onInputChange: _propTypes["default"].func,
  disableLocations: _propTypes["default"].bool,
  deprecatedGeosFaqLink: _propTypes["default"].string,
  handleDeprecatedGeos: _propTypes["default"].bool,
  parentCountry: _propTypes["default"].string.isRequired,
  tracking: _propTypes["default"].object,
  language: _propTypes["default"].string.isRequired,
  limit: _propTypes["default"].number,
  strings: _propTypes["default"].shape(stringsShape),
  subdivisionCategories: _propTypes["default"].object,
  geoLocationSearchableTypes: _propTypes["default"].arrayOf(_propTypes["default"].string)
};
LocationSearchByCountry.defaultProps = {
  bulkUploaderStrings: {},
  excludeCountries: [],
  keyField: _constants.KEY_FIELD,
  language: _i18n.DEFAULT_LANGUAGE,
  strings: {
    LOCATION_LABEL: _constants.LOCATION_LABEL,
    LOCATION_PLACEHOLDER: _constants.LOCATION_PLACEHOLDER,
    LOCATION_TOOLTIP_DESCRIPTION: _constants.LOCATION_TOOLTIP_DESCRIPTION,
    ARIA_REMOVE_THIS_LOCATION: _constants.ARIA_REMOVE_THIS_LOCATION,
    ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION: _constants.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION,
    CTA_LEARN_MORE: _constants.CTA_LEARN_MORE,
    CTA_REMOVE_ALL_PLURAL: _constants.CTA_REMOVE_ALL_PLURAL,
    CTA_REMOVE_ALL_SINGULAR: _constants.CTA_REMOVE_ALL_SINGULAR,
    DEPRECATED_GEOS_ERROR: _constants.DEPRECATED_GEOS_ERROR,
    LOCATION_UNAVAILABLE_SINGULAR: _constants.LOCATION_UNAVAILABLE_SINGULAR,
    LOCATION_UNAVAILABLE_PLURAL: _constants.LOCATION_UNAVAILABLE_PLURAL
  }
};
var _default = LocationSearchByCountry;
exports["default"] = _default;
