// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/LocationSearch/LocationSearch.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LocationSearchByCountry = exports.DEFAULT_STRINGS = exports.TargetingLocation = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _api = require("../api");

var _constants = require("../constants");

var _Autosuggest = _interopRequireDefault(require("../../Autosuggest"));

var _Suggestion = _interopRequireDefault(require("../../GeoTargetingSearch/Suggestion"));

var _Value = _interopRequireDefault(require("../../GeoTargetingSearch/Value"));

var _selectors = require("../../GeoTargetingSearch/selectors");

var _i18n = require("../../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var GeoValue = _Value["default"];

var TargetingLocation = _styledComponents["default"].div.withConfig({
  displayName: "LocationSearch__TargetingLocation",
  componentId: "sc-1ol0pl-0"
})([".location-type{background-clip:content-box;background-color:", ";color:", ";font-weight:400;text-transform:capitalize;p{padding-left:10px;}}.selected-locations-list-item p{margin-bottom:0;margin-top:0;padding-bottom:", ";padding-top:", ";}.selected-locations-list.populated{margin-bottom:10px;overflow:hidden;}.location-name{color:", ";}.dma-geo-type{text-transform:uppercase;}"], _encoreAdvertisingWeb.gray95, _encoreAdvertisingWeb.gray40, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.spacer12, _encoreAdvertisingWeb.gray60);

exports.TargetingLocation = TargetingLocation;
var DEFAULT_STRINGS = {
  LOCATION_LABEL: _constants.LOCATION_LABEL,
  LOCATION_PLACEHOLDER: _constants.LOCATION_PLACEHOLDER,
  LOCATION_TOOLTIP_DESCRIPTION: _constants.LOCATION_TOOLTIP_DESCRIPTION,
  ARIA_REMOVE_THIS_LOCATION: _constants.ARIA_REMOVE_THIS_LOCATION,
  ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION: _constants.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION,
  CTA_LEARN_MORE: _constants.CTA_LEARN_MORE,
  CTA_REMOVE_ALL_PLURAL: _constants.CTA_REMOVE_ALL_PLURAL,
  CTA_REMOVE_ALL_SINGULAR: _constants.CTA_REMOVE_ALL_SINGULAR,
  DEPRECATED_GEOS_ERROR: _constants.DEPRECATED_GEOS_ERROR,
  LOCATION_UNAVAILABLE_SINGULAR: _constants.LOCATION_UNAVAILABLE_SINGULAR,
  LOCATION_UNAVAILABLE_PLURAL: _constants.LOCATION_UNAVAILABLE_PLURAL
};
exports.DEFAULT_STRINGS = DEFAULT_STRINGS;

var LocationSearchByCountry = /*#__PURE__*/function (_React$PureComponent) {
  _inherits(LocationSearchByCountry, _React$PureComponent);

  var _super = _createSuper(LocationSearchByCountry);

  function LocationSearchByCountry() {
    _classCallCheck(this, LocationSearchByCountry);

    return _super.apply(this, arguments);
  }

  _createClass(LocationSearchByCountry, [{
    key: "render",
    value: function render() {
      var _this = this;

      var _this$props = this.props,
          bulkUploaderStrings = _this$props.bulkUploaderStrings,
          className = _this$props.className,
          excludeCountries = _this$props.excludeCountries,
          disableLocations = _this$props.disableLocations,
          deprecatedGeosFaqLink = _this$props.deprecatedGeosFaqLink,
          handleDeprecatedGeos = _this$props.handleDeprecatedGeos,
          _this$props$language = _this$props.language,
          language = _this$props$language === void 0 ? _i18n.DEFAULT_LANGUAGE : _this$props$language,
          limit = _this$props.limit,
          fieldRenderProps = _this$props.fieldRenderProps,
          onChange = _this$props.onChange,
          subdivisionCategories = _this$props.subdivisionCategories,
          validate = _this$props.validate,
          geoLocationSearchableTypes = _this$props.geoLocationSearchableTypes,
          rest = _objectWithoutProperties(_this$props, ["bulkUploaderStrings", "className", "excludeCountries", "disableLocations", "deprecatedGeosFaqLink", "handleDeprecatedGeos", "language", "limit", "fieldRenderProps", "onChange", "subdivisionCategories", "validate", "geoLocationSearchableTypes"]);

      var strings = rest.strings || DEFAULT_STRINGS;
      return /*#__PURE__*/_react["default"].createElement(TargetingLocation, {
        className: className
      }, /*#__PURE__*/_react["default"].createElement(_Autosuggest["default"], _extends({}, fieldRenderProps.input, {
        debounced: true,
        limit: limit,
        exclude: function exclude(item) {
          return excludeCountries.includes(item.countryCode);
        },
        id: "locations-autosuggest",
        disabled: disableLocations,
        keyField: rest.keyField || _constants.KEY_FIELD,
        onInputChange: function onInputChange(query) {
          // Optimization to not bog down the UI by showing every result for "N", for instance
          if (query.length < 2) return;
          var parentCountry = _this.props.parentCountry;

          if (_this.props.tracking && typeof _this.props.tracking.locations === 'function') {
            _this.props.tracking.locations(query);
          }

          return (0, _api.locationByCountrySearch)(query, parentCountry, language, 0, geoLocationSearchableTypes);
        },
        hasLazyLoading: true,
        onFetchMore: function onFetchMore(query, page) {
          var parentCountry = _this.props.parentCountry;
          return (0, _api.locationByCountrySearch)(query, parentCountry, language, page, geoLocationSearchableTypes);
        },
        placeholder: strings.LOCATION_PLACEHOLDER,
        formatSuggestion: _selectors.getGeoLabelFromSuggestion,
        renderSuggestion: function renderSuggestion(props) {
          return /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], _extends({}, props, {
            strings: subdivisionCategories
          }));
        },
        renderValue: function renderValue(_ref) {
          var disabled = _ref.disabled,
              value = _ref.value,
              handleRemove = _ref.handleRemove,
              keyField = _ref.keyField,
              valueField = _ref.valueField;
          return /*#__PURE__*/_react["default"].createElement(GeoValue, {
            deprecatedGeosFaqLink: deprecatedGeosFaqLink,
            handleDeprecatedGeos: handleDeprecatedGeos,
            bulkUploaderStrings: bulkUploaderStrings,
            deprecatedGeoStrings: strings,
            disabled: disabled,
            items: value,
            keyField: keyField,
            onClick: handleRemove,
            valueField: valueField,
            limit: limit,
            strings: subdivisionCategories
          });
        },
        suggestionsLimit: Infinity,
        validate: validate,
        onBlur: function onBlur(val) {
          onChange(val);
          fieldRenderProps.input.onBlur();
        }
      })));
    }
  }]);

  return LocationSearchByCountry;
}(_react["default"].PureComponent);

exports.LocationSearchByCountry = LocationSearchByCountry;
LocationSearchByCountry.defaultProps = {
  excludeCountries: []
};
var _default = LocationSearchByCountry;
exports["default"] = _default;
