// @spotify-internal/encore-web/es/components/PaginationDropdownList/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { List } from "./List";

function PaginationDropdownListComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(List, props));
}

var paginationDropdownListHelpers = {
  darkTheme: {
    paginationDropdownList: 'dark'
  }
};
/**
* **PaginationDropdownList** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PaginationDropdownList) | [Storybook](https://encore-web.spotify.net/?path=/story/components-pagination-paginationdropdownlist--default) | [Encore](https://encore.spotify.net/web/components/pagination?format=figma-figma#paginationdropdown) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=2311:140) |
*
* PaginationDropdownList is a popover containing PaginationDropdownLinks
*
* @example
* export const Default = () => (
*  <PaginationDropdownList>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>10</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>20</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>30</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>40</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>50</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>60</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>70</PaginationDropdownLink>
*    </PaginationDropdownItem>
*    <PaginationDropdownItem>
*      <PaginationDropdownLink>80</PaginationDropdownLink>
*    </PaginationDropdownItem>
*  </PaginationDropdownList>
*);
*
*/

export var PaginationDropdownList = Object.assign(PaginationDropdownListComponent, paginationDropdownListHelpers);
export default PaginationDropdownList;
