// @spotify-internal/encore-web/es/components/FormRadio/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { VisuallyHidden } from "../VisuallyHidden";
import { Indicator } from "./Indicator";
import { Label } from "./Label";
import { Radio } from "./Radio";
var FormRadioComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      children = _ref.children,
      id = _ref.id,
      props = _objectWithoutProperties(_ref, ["theme", "children", "id"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Radio, null, /*#__PURE__*/React.createElement(VisuallyHidden, Object.assign({
    component: "input",
    type: "radio",
    ref: ref,
    id: id
  }, props)), /*#__PURE__*/React.createElement("label", {
    htmlFor: id
  }, /*#__PURE__*/React.createElement(Indicator, {
    isUsingKeyboard: isUsingKeyboard
  }), children && /*#__PURE__*/React.createElement(Label, null, children))));
});
var formRadioHelpers = {
  darkTheme: {
    formRadio: 'dark'
  },
  displayName: 'FormRadio'
};
/**
* **FormRadio** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/FormRadio) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-formradio--default) | [Encore](https://encore.spotify.net/web/components/selectors?format=figma-figma#formradio) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:526) |
*
* FormRadio allow users to make one selection within a Form component.
*
* @example
* export const Default = () => (
*  <FormGroup label="Radio example" withFieldset>
*    <FormRadio id="option-1" name="radio-example" defaultChecked>
*      Option 1
*    </FormRadio>
*    <FormRadio id="option-2" name="radio-example">
*      Option 2
*    </FormRadio>
*  </FormGroup>
*);
*
*/

export var FormRadio = Object.assign(FormRadioComponent, formRadioHelpers);
export default FormRadio;
