// @spotify-internal/adstudio-tape/lib/components/ImageDisplay/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ImageDisplay = ImageDisplay;
exports["default"] = exports.Thumbnail = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _Thumb = _interopRequireDefault(require("../Thumb"));

var _Panel = _interopRequireDefault(require("../Panel"));

var _onEnter = _interopRequireDefault(require("../../utils/helpers/onEnter"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _constants = require("./constants");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Display = _styledComponents["default"].div.withConfig({
  displayName: "ImageDisplay__Display",
  componentId: "sc-1fqrnvj-0"
})(["align-items:center;display:flex;justify-content:flex-start;min-width:0;"]);

var Thumbnail = (0, _styledComponents["default"])(_Thumb["default"]).withConfig({
  displayName: "ImageDisplay__Thumbnail",
  componentId: "sc-1fqrnvj-1"
})(["flex-shrink:0;margin-right:", ";"], _encoreFoundation.spacer16);
exports.Thumbnail = Thumbnail;
var Name = (0, _styledComponents["default"])(_encoreWeb.Type.p).withConfig({
  displayName: "ImageDisplay__Name",
  componentId: "sc-1fqrnvj-2"
})(["flex:1;flex-direction:column;overflow:hidden;padding:0;text-overflow:ellipsis;white-space:nowrap;"]);

function ImageDisplay(_ref) {
  var className = _ref.className,
      deleteLabel = _ref.deleteLabel,
      name = _ref.name,
      onClickDelete = _ref.onClickDelete,
      onClickImage = _ref.onClickImage,
      panelBorder = _ref.panelBorder,
      src = _ref.src;

  var content = /*#__PURE__*/_react["default"].createElement(Display, {
    className: className
  }, /*#__PURE__*/_react["default"].createElement(Thumbnail, {
    onClick: onClickImage,
    size: 80,
    src: src
  }), /*#__PURE__*/_react["default"].createElement(Name, {
    variant: _encoreWeb.Type.body1,
    weight: _encoreWeb.Type.book
  }, name));

  if (onClickImage) {
    content = /*#__PURE__*/_react["default"].createElement("a", {
      onClick: onClickImage,
      onKeyPress: (0, _onEnter["default"])(onClickImage),
      role: "button",
      tabIndex: "0"
    }, content);
  }

  var ctaList = [];

  if (onClickDelete) {
    ctaList.push({
      label: deleteLabel,
      onClick: onClickDelete
    });
  }

  return /*#__PURE__*/_react["default"].createElement(_Panel["default"], {
    border: false,
    ctaList: ctaList,
    ctaListStacked: true,
    shadow: panelBorder
  }, content);
}

ImageDisplay.propTypes = {
  className: _propTypes["default"].string,
  deleteLabel: _propTypes["default"].string,
  name: _propTypes["default"].string,
  onClickDelete: _propTypes["default"].func,
  onClickImage: _propTypes["default"].func,
  panelBorder: _propTypes["default"].bool,
  src: _propTypes["default"].string.isRequired
};
ImageDisplay.defaultProps = {
  deleteLabel: _constants.DELETE_LABEL,
  panelBorder: false
};
var _default = ImageDisplay;
exports["default"] = _default;
