// @spotify-internal/adstudio-tape/lib/components/Panel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Panel;
exports.PanelContent = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _ButtonTertiary = _interopRequireDefault(require("../ButtonTertiary"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Container = _styledComponents["default"].div.withConfig({
  displayName: "Panel__Container",
  componentId: "f8ma1i-0"
})(["align-items:center;border-radius:", ";border:", ";border-color:", ";box-shadow:", ";display:flex;justify-content:space-between;padding:", ";"], _encoreFoundation.spacer4, function (props) {
  if (props.shadow) return 'none';
  if (props.border) return "1px solid ".concat(_encoreFoundation.gray70);
  return 'none';
}, function (props) {
  if (props.error) return _encoreFoundation.failure;
  if (props.warning) return _encoreFoundation.warning;
  return null;
}, function (props) {
  return props.shadow && "0 1px 3px 0 ".concat(_encoreFoundation.gray85);
}, _encoreFoundation.spacer16);

var PanelContent = _styledComponents["default"].div.withConfig({
  displayName: "Panel__PanelContent",
  componentId: "f8ma1i-1"
})(["align-items:center;display:flex;flex-grow:1;justify-content:flex-start;min-width:0;"]);

exports.PanelContent = PanelContent;

var CTA = _styledComponents["default"].div.withConfig({
  displayName: "Panel__CTA",
  componentId: "f8ma1i-2"
})(["flex-shrink:0;"]);

var CTAButton = (0, _styledComponents["default"])(_ButtonTertiary["default"]).withConfig({
  displayName: "Panel__CTAButton",
  componentId: "f8ma1i-3"
})(["padding-right:0;display:", ";flex-direction:", ";flex-shrink:0;color:", ";"], function (props) {
  return props.ctaListStacked && 'flex';
}, function (props) {
  return props.ctaListStacked && 'column';
}, function (props) {
  return props.theme && props.theme.colors ? props.theme.colors.primaryColor : _encoreFoundation.azure;
});

function Panel(_ref) {
  var border = _ref.border,
      children = _ref.children,
      className = _ref.className,
      ctaList = _ref.ctaList,
      ctaListStacked = _ref.ctaListStacked,
      error = _ref.error,
      shadow = _ref.shadow,
      warning = _ref.warning;

  var result = /*#__PURE__*/_react["default"].createElement(Container, {
    className: className,
    error: error,
    warning: warning,
    border: border,
    shadow: shadow
  }, /*#__PURE__*/_react["default"].createElement(PanelContent, {
    className: className
  }, children), ctaList && ctaList.length > 0 && /*#__PURE__*/_react["default"].createElement(CTA, null, ctaList.map(function (_ref2) {
    var label = _ref2.label,
        ctaClassName = _ref2.className,
        ctaProps = _objectWithoutProperties(_ref2, ["label", "className"]);

    return /*#__PURE__*/_react["default"].createElement(CTAButton, _extends({
      key: "panel-cta-".concat(label),
      buttonSize: "sm",
      className: ctaClassName,
      ctaListStacked: ctaListStacked
    }, ctaProps), label);
  })));

  return result;
}

Panel.defaultProps = {
  border: true
};
Panel.propTypes = {
  border: _propTypes["default"].bool,
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].node, _propTypes["default"].element]),
  className: _propTypes["default"].string,
  ctaList: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    label: _propTypes["default"].string
  })),
  ctaListStacked: _propTypes["default"].bool,
  shadow: _propTypes["default"].bool
};
