// @spotify-internal/encore-foundation/web/index.js
module.exports = {
  ...require('./tokens.common'),
  ...require('../themes/themes'),
  ...require('../themes/sets'),
  // Temporary type token re-exports to support imports from within Encore Web
  // See: https://ghe.spotify.net/encore/web/blob/master/src/core/styles/mixins/type.tsx
  // Once these mobile and desktop sizes are no longer in needed in Encore Web
  // then these exports can be removed. This is to prevent Encore Web from having
  // to import from e.g. '@spotify-internal/encore-foundation/desktop' which was
  // causing tree-shaking and bundle size issues.
  desktopAlto: require('../desktop').alto,
  desktopAltoBrio: require('../desktop').altoBrio,
  desktopBallad: require('../desktop').ballad,
  desktopBalladBold: require('../desktop').balladBold,
  desktopBass: require('../desktop').bass,
  desktopBrio: require('../desktop').brio,
  desktopCanon: require('../desktop').canon,
  desktopCelloCanon: require('../desktop').celloCanon,
  desktopCello: require('../desktop').cello,
  desktopFinale: require('../desktop').finale,
  desktopFinaleBold: require('../desktop').finaleBold,
  desktopForte: require('../desktop').forte,
  desktopMesto: require('../desktop').mesto,
  desktopMestoBold: require('../desktop').mestoBold,
  desktopMetronome: require('../desktop').metronome,
  desktopMinuet: require('../desktop').minuet,
  desktopMinuetBold: require('../desktop').minuetBold,
  desktopViola: require('../desktop').viola,
  desktopViolaBold: require('../desktop').violaBold,
  mobileAlto: require('../mobile').alto,
  mobileAltoBrio: require('../mobile').altoBrio,
  mobileBallad: require('../mobile').ballad,
  mobileBalladBold: require('../mobile').balladBold,
  mobileBass: require('../mobile').bass,
  mobileBrio: require('../mobile').brio,
  mobileCanon: require('../mobile').canon,
  mobileCelloCanon: require('../mobile').celloCanon,
  mobileCello: require('../mobile').cello,
  mobileFinale: require('../mobile').finale,
  mobileFinaleBold: require('../mobile').finaleBold,
  mobileForte: require('../mobile').forte,
  mobileMesto: require('../mobile').mesto,
  mobileMestoBold: require('../mobile').mestoBold,
  mobileMetronome: require('../mobile').metronome,
  mobileMinuet: require('../mobile').minuet,
  mobileMinuetBold: require('../mobile').minuetBold,
  mobileViola: require('../mobile').viola,
  mobileViolaBold: require('../mobile').violaBold,
};
