// @spotify-internal/encore-advertising-web/esm/components/MobilePreview/styled/ImageContainer.js
import React from 'react';
import styled from 'styled-components';
import { ButtonIcon, gray50, gray90, IconHelpAlt, spacer8, Tooltip, TooltipTrigger, white, } from '@spotify-internal/encore-web';
const IMAGE_SIZE = '245px';
const Image = styled.img `
  height: inherit;
  width: inherit;
`;
const ImageWrapper = styled.div `
  position: relative;
  border-radius: ${spacer8};
  overflow: hidden;
  background: ${gray90};
  margin: 0 auto;
  height: ${IMAGE_SIZE};
  width: ${IMAGE_SIZE};
`;
const ImageTooltip = styled(TooltipTrigger) `
  position: absolute;
  left: 0;
  top: 0;
  margin-left: ${spacer8};
  margin-top: ${spacer8};
`;
const ImageTooltipButton = styled(ButtonIcon) `
  color: ${gray50};
  &:hover {
    color: ${white};
  }
  &:after {
    display: none;
  }
`;
export const ImageContainer = ({ imageUrl, imageTooltipMessage, children, }) => {
    const [showTooltip, setShowTooltip] = React.useState(false);
    return (React.createElement(ImageWrapper, null,
        imageTooltipMessage && (React.createElement(ImageTooltip, { onShow: () => setShowTooltip(true), onHide: () => setShowTooltip(false), placement: TooltipTrigger.right, withPortal: true, overlay: showTooltip && React.createElement(Tooltip, { id: "help" }, imageTooltipMessage) },
            React.createElement(ImageTooltipButton, { "data-testid": "tooltip-btn", "aria-label": "help" },
                React.createElement(IconHelpAlt, { iconSize: 16 })))),
        imageUrl ? React.createElement(Image, { src: imageUrl }) : children));
};
