// spotify-transport/types/status_code.js
'use strict';

var debugEnum = require('spotify-debug-tools/debug_enum');

/**
 * HTTP Status Codes
 *
 * @enum {number}
 */
var StatusCode = {
  // This status code is usually seen in the browser when the XMLHttpRequest
  // cannot properly complete the request.
  UNKNOWN: 0,

  // 2xx Family: Success
  OK: 200,
  CREATED: 201,
  ACCEPTED: 202,
  NO_CONTENT: 204,
  PARTIAL_CONTENT: 206,

  // 3xx Family: Redirect
  // Status-codes are not exported as they are handled automatically by
  // the browser.

  // 4xx Family: Client Error
  BAD_REQUEST: 400,
  UNAUTHORIZED: 401,
  PAYMENT_REQUIRED: 402,
  FORBIDDEN: 403,
  NOT_FOUND: 404,
  METHOD_NOT_ALLOWED: 405,
  NOT_ACCEPTABLE: 406,
  PROXY_AUTHENTICATION_REQUIRED: 407,
  TIMED_OUT: 408, // deprecated
  REQUEST_TIMEOUT: 408,
  CONFLICT: 409,
  GONE: 410,
  LENGTH_REQUIRED: 411,
  PRECONDITION_FAILED: 412,
  REQUEST_ENTITY_TOO_LARGE: 413,
  REQUEST_URI_TOO_LONG: 414,
  UNSUPPORTED_MEDIA_TYPE: 415,
  REQUESTED_RANGE_NOT_SATISFIABLE: 416,
  EXPECTATION_FAILED: 417,

  // 5xx Family: Server Error
  INTERNAL_SERVER_ERROR: 500,
  NOT_IMPLEMENTED: 501,
  BAD_GATEWAY: 502,
  SERVICE_UNAVAILABLE: 503,
  GATEWAY_TIMEOUT: 504,
  HTTP_VERSION_NOT_SUPPORTED: 505
};

module.exports = debugEnum(StatusCode);
