// @spotify-internal/encore-advertising-web/esm/components/PaymentDetails/constants.js
import IconVisa from '../../icons/Payment/IconVisa';
import IconMastercard from '../../icons/Payment/IconMastercard';
import IconAmex from '../../icons/Payment/IconAmex';
import IconDiscover from '../../icons/Payment/IconDiscover';
import IconDefault from '../../icons/Payment/IconDefault';
export var DetailsType;
(function (DetailsType) {
    DetailsType["Card"] = "CARD";
    DetailsType["Invoice"] = "INVOICE";
})(DetailsType || (DetailsType = {}));
export const CREDIT_CARD_CONFIG = {
    visa: {
        format: [4, 4, 4, 4],
        name: 'Visa',
        icon: IconVisa,
    },
    mc: {
        format: [4, 4, 4, 4],
        name: 'Mastercard',
        icon: IconMastercard,
    },
    amex: {
        format: [4, 6, 5],
        name: 'Amex',
        icon: IconAmex,
    },
    discover: {
        format: [4, 4, 4, 4],
        name: 'Discover',
        icon: IconDiscover,
    },
};
export const SUPPORTED_CARDS = [
    CREDIT_CARD_CONFIG.visa,
    CREDIT_CARD_CONFIG.mc,
    CREDIT_CARD_CONFIG.amex,
];
export const DEFAULT_CARD_CONFIG = {
    format: CREDIT_CARD_CONFIG.mc.format,
    icon: IconDefault,
};
export const DEFAULT_ADD_CARD_TEXT = '**Add your card to continue.**';
export const DEFAULT_ADD_CARD_SUBTEXT = `You'll be redirected to a new page. Your work will be saved.`;
export const DEFAULT_ADD_CARD_CTA_TEXT = 'Add card';
export const DEFAULT_ADD_CARD_DESCRIPTION = 'Payment method must be from the same country as your account.';
export const DEFAULT_CARD_CTA_TEXT = 'Update';
export const DEFAULT_CARD_DESCRIPTION = 'We’ll charge your credit card at the end of each month or when your ad spend reaches a given [billing threshold](https://adstudio.spotify.com/help/billing-cycle) across all your campaigns.';
export const DEFAULT_CARD_DESCRIPTION_SUB = '**Note:** only one credit card can be kept on file for billing.';
export const DEFAULT_INVOICE_TEXT = 'You have invoicing set up.';
export const DEFAULT_INVOICE_DESCRIPTION = `
  If you have questions about billing, visit our FAQ or please email us at [adstudio@spotify.com](mailto:adstudio@spotify.com)
`;
export const DEFAULT_REDIRECTING_CTA_TEXT = 'Redirecting';
export const DEFAULT_ERROR_HELP_TEXT = `
  Contact [adstudio@spotify.com](mailto:adstudio@spotify.com)
  if you're having trouble.
`;
export const RETRY_TEXT = 'Retry';
export const DEFAULT_CARD_SUCCESS_TEXT = 'Card saved';
export const DEFAULT_CANNOT_ADD_CARD_SUBTEXT = `Viewers aren't permitted to add a payment method. Please contact your Admin.`;
export const DEFAULT_NONE_TEXT = 'none';
export const DEFAULT_PO_NUMBER_TEXT = 'PO number';
