// @spotify-internal/encore-web/es/components/PopoverNavigation/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React from 'react';
import { ThemeProvider } from 'styled-components';
import { overlayDirections } from "../../styles/mixins/overlays";
import { List } from "../List";
import { PopoverDesktop } from "./PopoverDesktop";
import { PopoverMobile } from "./PopoverMobile";

function PopoverNavigationComponent(_ref) {
  var isMobile = _ref.isMobile,
      children = _ref.children,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["isMobile", "children", "theme"]);

  var Popover = isMobile ? PopoverMobile : PopoverDesktop;
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(Popover, props, /*#__PURE__*/React.createElement(List, null, children)));
}

var popoverNavigationHelpers = _objectSpread(_objectSpread({}, overlayDirections), {}, {
  darkTheme: {
    popover: 'dark'
  }
});
/**
* **PopoverNavigation** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/PopoverNavigation) | [Storybook](https://encore-web.spotify.net/?path=/story/components-popover-popovernavigation--default-desktop) | [Encore](https://encore.spotify.net/web/components/popover?format=figma-figma#popovernavigation) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=613:402) |
*
* A popover that renders a simple list of PopoverNavigationLinks
*
* @example
* export const DefaultDesktop = () => (
*  <PopoverNavigation arrow={PopoverNavigation.bottomRight}>
*    <PopoverNavigationItem>
*      <PopoverNavigationLink href="#">Share</PopoverNavigationLink>
*    </PopoverNavigationItem>
*    <PopoverNavigationItem>
*      <PopoverNavigationLink component="button">View details</PopoverNavigationLink>
*    </PopoverNavigationItem>
*  </PopoverNavigation>
*);
*
*/


export var PopoverNavigation = Object.assign(PopoverNavigationComponent, popoverNavigationHelpers);
export default PopoverNavigation;
