// @spotify-internal/encore-advertising-web/esm/components/ButtonGroup/Button.js
import { ButtonPrimary, gray30, gray70, spacer8, white, } from '@spotify-internal/encore-web';
import styled, { css } from 'styled-components';
import { plumHover, plum } from '../../styles/colors';
const defaultButtonBackgroundColor = white;
const defaultButtonBorderColor = gray70;
const defaultButtonBorder = `1px solid ${defaultButtonBorderColor}`;
const defaultButtonBorderRadius = spacer8;
const defaultButtonTextColor = gray30;
const hoverStyles = css `
  transform: none !important;
  background-color: ${plumHover};
  color: ${white};
  opacity: 0.9;
`;
export const Button = styled(ButtonPrimary) `
  && {
    background: ${defaultButtonBackgroundColor};
    color: ${defaultButtonTextColor};
    border-radius: 0px;
    border-top: ${defaultButtonBorder};
    border-bottom: ${defaultButtonBorder};

    /* Border styles for first button */
    &:first-of-type {
      border-left: ${defaultButtonBorder};
      border-bottom-left-radius: ${defaultButtonBorderRadius};
      border-top-left-radius: ${defaultButtonBorderRadius};
    }

    /* Border styles for last button */
    &:last-of-type {
      border-right: ${defaultButtonBorder};
      border-bottom-right-radius: ${defaultButtonBorderRadius};
      border-top-right-radius: ${defaultButtonBorderRadius};
      &:after {
        border-right: none;
      }
    }

    /* Dividers between buttons */
    &:after {
      border: none;
      border-right: ${defaultButtonBorder};
      border-color: ${defaultButtonBorderColor} !important;
      top: -1px;
      left: 0px;
      width: 100%;
      height: calc(100% + 2px);
      border-radius: 0;
    }

    ${props => !props.disabled &&
    css `
        ${props.hover && hoverStyles}
        &:hover {
          ${hoverStyles}
        }
      `};
    ${props => props.$isSelected &&
    css `
        background: ${plum} !important;
        border-color: transparent !important;
        color: ${white};
      `}
  }
`;
