// @spotify-internal/spoticharts/cjs/components/A11yInterface/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const encore_foundation_1 = require("@spotify-internal/encore-foundation");
const VisuallyHiddenRect = styled_components_1.default.rect `
  fill: transparent;
  pointer-events: none;

  &:focus {
    outline: 3px inset ${encore_foundation_1.gray10};
  }
`;
A11yInterface.propTypes = {
    ariaLabelGenerator: prop_types_1.default.func.isRequired,
    numFrames: prop_types_1.default.number,
    onKeyDown: prop_types_1.default.func,
    height: prop_types_1.default.number,
    width: prop_types_1.default.number,
};
function A11yInterface(props) {
    const { ariaLabelGenerator, numFrames, onKeyDown, height, width } = props;
    const sliceWidth = width / numFrames;
    return (react_1.default.createElement("g", { className: "spt-chart-a11y" }, [...Array(numFrames).keys()].map((d, frameIndex) => (react_1.default.createElement(VisuallyHiddenRect, { "aria-label": ariaLabelGenerator(frameIndex), key: frameIndex, height: height, width: sliceWidth, x: sliceWidth * frameIndex, y: 0, role: "button", tabIndex: 0, onKeyDown: event => {
            if (!!onKeyDown) {
                onKeyDown(event, frameIndex);
            }
        } })))));
}
exports.default = A11yInterface;
