// @spotify-internal/adstudio-shared/lib/components/Autosuggest/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.handlers = exports.AutosuggestField = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _recompose = require("recompose");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _classnames = _interopRequireDefault(require("classnames"));

var _utils = require("./utils");

var _Input = _interopRequireDefault(require("./Input"));

var _Suggestion = _interopRequireDefault(require("./Suggestion"));

var _SelectedPillsList = _interopRequireDefault(require("./SelectedPillsList"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) { try { var info = gen[key](arg); var value = info.value; } catch (error) { reject(error); return; } if (info.done) { resolve(value); } else { Promise.resolve(value).then(_next, _throw); } }

function _asyncToGenerator(fn) { return function () { var self = this, args = arguments; return new Promise(function (resolve, reject) { var gen = fn.apply(self, args); function _next(value) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value); } function _throw(err) { asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err); } _next(undefined); }); }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ENTER_KEY = 13;

var Container = _styledComponents["default"].div.withConfig({
  displayName: "Autosuggest__Container",
  componentId: "sc-1e8tqb5-0"
})(["ul{list-style-type:none;padding:0;margin:0;}p{margin:0;}"]);

var AutosuggestField = function AutosuggestField(props) {
  var className = props.className,
      exclude = props.exclude,
      handleAdd = props.handleAdd,
      handleInputChange = props.handleInputChange,
      handleFetchMore = props.handleFetchMore,
      hasLazyLoading = props.hasLazyLoading,
      inlineValue = props.inlineValue,
      keyField = props.keyField,
      _props$options = props.options,
      options = _props$options === void 0 ? [] : _props$options,
      placeholder = props.placeholder,
      renderValue = props.renderValue,
      _props$value = props.value,
      value = _props$value === void 0 ? [] : _props$value,
      valueField = props.valueField,
      disabled = props.disabled,
      limit = props.limit,
      rest = _objectWithoutProperties(props, ["className", "exclude", "handleAdd", "handleInputChange", "handleFetchMore", "hasLazyLoading", "inlineValue", "keyField", "options", "placeholder", "renderValue", "value", "valueField", "disabled", "limit"]);

  var availableItems = (0, _utils.filterOutExcludedItems)(options, value, keyField);
  var shouldDisable = limit ? disabled || value.length >= limit : disabled;
  return /*#__PURE__*/_react["default"].createElement(Container, null, value.length > 0 && renderValue(props), (!inlineValue || value.length === 0) && /*#__PURE__*/_react["default"].createElement(_Input["default"], _extends({}, rest, {
    disabled: shouldDisable,
    availableItems: availableItems,
    browseEnabled: Boolean(options),
    className: (0, _classnames["default"])(className, "ta-".concat(name, "-autosuggest")),
    exclude: exclude,
    existingValue: value,
    keyField: keyField,
    onChange: handleInputChange,
    onFetchMore: handleFetchMore,
    onSuggestionSelect: handleAdd,
    searchPlaceholder: placeholder,
    selectPlaceholder: placeholder,
    valueField: valueField,
    hasLazyLoading: hasLazyLoading
  })));
};

exports.AutosuggestField = AutosuggestField;
AutosuggestField.propTypes = {
  className: _propTypes["default"].string,
  debounced: _propTypes["default"].bool,
  disabled: _propTypes["default"].bool,
  exclude: _propTypes["default"].oneOfType([_propTypes["default"].array, _propTypes["default"].func]),
  handleAdd: _propTypes["default"].func.isRequired,
  handleInputChange: _propTypes["default"].func.isRequired,
  handleRemove: _propTypes["default"].func.isRequired,
  inlineValue: _propTypes["default"].bool,
  keyField: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onBlur: _propTypes["default"].func,
  onChange: _propTypes["default"].func,
  onFocus: _propTypes["default"].func,
  onInputChange: _propTypes["default"].func,
  onFetchMore: _propTypes["default"].func,
  onSortSuggestions: _propTypes["default"].func,
  options: _propTypes["default"].array,
  placeholder: _propTypes["default"].string,
  renderSuggestion: _propTypes["default"].func.isRequired,
  renderValue: _propTypes["default"].func.isRequired,
  handleFetchMore: function handleFetchMore(props) {
    if (props.hasLazyLoading === true) return _propTypes["default"].isRequired;
  },
  hasLazyLoading: _propTypes["default"].func,
  value: _propTypes["default"].array.isRequired,
  valueField: _propTypes["default"].string,
  limit: _propTypes["default"].number
};
/* eslint-disable react/prop-types */

var defaultRenderValue = function defaultRenderValue(_ref) {
  var disabled = _ref.disabled,
      clickthroughUrl = _ref.clickthroughUrl,
      keyField = _ref.keyField,
      valueField = _ref.valueField,
      value = _ref.value,
      handleRemove = _ref.handleRemove,
      handleKeyEnter = _ref.handleKeyEnter;
  return /*#__PURE__*/_react["default"].createElement(_SelectedPillsList["default"], {
    disabled: disabled,
    clickthroughUrl: clickthroughUrl,
    items: value,
    name: name,
    keyField: keyField,
    onClick: handleRemove,
    onKeyEnter: handleKeyEnter,
    valueField: valueField
  });
};

var defaultRenderSuggestion = function defaultRenderSuggestion(props) {
  return /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], props);
};
/* eslint-enable react/prop-types */

/* eslint-disable consistent-return */


var removeItem = function removeItem(_ref2) {
  var onBlur = _ref2.onBlur,
      value = _ref2.value,
      keyField = _ref2.keyField,
      item = _ref2.item;

  if (typeof onBlur === 'function') {
    return onBlur((0, _utils.filterOutExcludedItems)(value, [item], keyField));
  }
};

var handlers = {
  handleAdd: function handleAdd(_ref3) {
    var onBlur = _ref3.onBlur,
        value = _ref3.value;
    return function (item) {
      if (typeof onBlur === 'function') {
        return onBlur(value.concat(item));
      }
    };
  },
  handleRemove: function handleRemove(_ref4) {
    var onBlur = _ref4.onBlur,
        value = _ref4.value,
        keyField = _ref4.keyField;
    return function (item) {
      removeItem({
        onBlur: onBlur,
        value: value,
        keyField: keyField,
        item: item
      });
    };
  },
  handleInputChange: function handleInputChange(_ref5) {
    var onInputChange = _ref5.onInputChange;
    return /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee() {
      var _args = arguments;
      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              if (!(typeof onInputChange === 'function')) {
                _context.next = 2;
                break;
              }

              return _context.abrupt("return", onInputChange.apply(void 0, _args));

            case 2:
            case "end":
              return _context.stop();
          }
        }
      }, _callee);
    }));
  },
  handleFetchMore: function handleFetchMore(_ref7) {
    var onFetchMore = _ref7.onFetchMore;
    return /*#__PURE__*/_asyncToGenerator( /*#__PURE__*/regeneratorRuntime.mark(function _callee2() {
      var _args2 = arguments;
      return regeneratorRuntime.wrap(function _callee2$(_context2) {
        while (1) {
          switch (_context2.prev = _context2.next) {
            case 0:
              if (!(typeof onFetchMore === 'function')) {
                _context2.next = 2;
                break;
              }

              return _context2.abrupt("return", onFetchMore.apply(void 0, _args2));

            case 2:
            case "end":
              return _context2.stop();
          }
        }
      }, _callee2);
    }));
  },
  handleKeyEnter: function handleKeyEnter(_ref9) {
    var onBlur = _ref9.onBlur,
        value = _ref9.value,
        keyField = _ref9.keyField;
    return function (item) {
      if (event.keyCode === ENTER_KEY) {
        removeItem({
          onBlur: onBlur,
          value: value,
          keyField: keyField,
          item: item
        });
      }
    };
  }
};
exports.handlers = handlers;

var _default = (0, _recompose.compose)((0, _recompose.defaultProps)({
  keyField: 'id',
  placeholder: 'Search',
  renderValue: defaultRenderValue,
  renderSuggestion: defaultRenderSuggestion,
  valueField: 'name'
}), (0, _recompose.withProps)(function (_ref10) {
  var value = _ref10.value;
  return {
    value: value || []
  };
}), (0, _recompose.withHandlers)(handlers))(AutosuggestField);

exports["default"] = _default;
