// @spotify-internal/adstudio-tape/lib/components/Animator/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Animator;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactTransitionGroup = require("react-transition-group");

var _parseToMillis = _interopRequireDefault(require("../../utils/helpers/parseToMillis"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Animator(_ref) {
  var children = _ref.children,
      defaultState = _ref.defaultState,
      delay = _ref.delay,
      duration = _ref.duration,
      enteredState = _ref.enteredState,
      enteringState = _ref.enteringState,
      exitedState = _ref.exitedState,
      exitingState = _ref.exitingState,
      transition = _ref.transition,
      props = _objectWithoutProperties(_ref, ["children", "defaultState", "delay", "duration", "enteredState", "enteringState", "exitedState", "exitingState", "transition"]);

  var defaultStyle = _objectSpread({
    transition: Array.isArray(transition) ? transition.join(', ') : "all ".concat(duration, " ").concat(transition, " ").concat(delay)
  }, defaultState || exitedState);

  var transitionStyles = {
    entered: enteredState,
    entering: enteringState || exitedState,
    exited: exitedState,
    exiting: exitingState || exitedState
  };
  var timeout = props.timeout || (0, _parseToMillis["default"])(delay || 0) + (0, _parseToMillis["default"])(duration || 0);
  return /*#__PURE__*/_react["default"].createElement(_reactTransitionGroup.Transition, _extends({
    timeout: timeout
  }, props), function (state) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      style: _objectSpread(_objectSpread({}, defaultStyle), transitionStyles[state])
    }, children);
  });
}

Animator.propTypes = {
  addEndListener: _propTypes["default"].func,
  appear: _propTypes["default"].bool,
  children: _propTypes["default"].oneOfType([_propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].element, _propTypes["default"].node]).isRequired,
  defaultState: _propTypes["default"].object,
  delay: _propTypes["default"].string,
  duration: _propTypes["default"].string,
  enter: _propTypes["default"].bool,
  enteringState: _propTypes["default"].object,
  enteredState: _propTypes["default"].object.isRequired,
  exit: _propTypes["default"].bool,
  exitingState: _propTypes["default"].object,
  exitedState: _propTypes["default"].object.isRequired,
  "in": _propTypes["default"].bool.isRequired,
  mountOnEnter: _propTypes["default"].bool,
  onEnter: _propTypes["default"].func,
  onEntered: _propTypes["default"].func,
  onEntering: _propTypes["default"].func,
  onExit: _propTypes["default"].func,
  onExited: _propTypes["default"].func,
  onExiting: _propTypes["default"].func,
  timeout: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].shape({
    appear: _propTypes["default"].number,
    enter: _propTypes["default"].number,
    exit: _propTypes["default"].number
  })]),
  transition: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].arrayOf(_propTypes["default"].string)]).required,
  unmountOnExit: _propTypes["default"].bool
};
Animator.defaultProps = {
  appear: true,
  delay: '0s',
  duration: '1s',
  enter: true,
  exit: true,
  "in": false,
  mountOnEnter: true,
  transition: 'linear',
  unmountOnExit: true
};
