// @spotify-internal/encore-web/es/components/TableRow/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { forwardRef } from 'react';
import { ThemeProvider } from 'styled-components';
import { TableRowElement } from "./TableRow";
var TableRowComponent = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(TableRowElement, Object.assign({
    ref: ref
  }, props)));
});
var tableRowHelpers = {
  darkTheme: {
    tableRow: 'dark'
  },
  displayName: 'TableRow'
};
/**
* **TableRow** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/TableRow) | [Storybook](https://encore-web.spotify.net/?path=/story/components-table-tablerow--default) | [Encore](https://encore.spotify.net/web/components/table#row) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=525:1503) |
*
* A TableRow is a horizontal stack of related cells.
*
* @example
* export const Default = () => (
*  <Table>
*    <tbody>
*      <TableRow>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*      </TableRow>
*      <TableRow>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*        <TableCell>Cell</TableCell>
*      </TableRow>
*    </tbody>
*  </Table>
*);
*
*/

export var TableRow = Object.assign(TableRowComponent, tableRowHelpers);
export default TableRow;
