// @spotify-internal/adstudio-tape/lib/components/Table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Table;
exports.CELL_TYPES = exports.CELL_SIZES = void 0;

var _get2 = _interopRequireDefault(require("lodash/get"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _date = require("../../utils/helpers/date");

var _TableContainer = _interopRequireDefault(require("../TableContainer"));

var _TableBody = _interopRequireDefault(require("../TableBody"));

var _TableRow = _interopRequireDefault(require("../TableRow"));

var _TableData = _interopRequireDefault(require("../TableData"));

var _TableHeader = _interopRequireDefault(require("../TableHeader"));

var _TableHeaderItem = _interopRequireDefault(require("../TableHeaderItem"));

var _StatusIndicator = _interopRequireDefault(require("../StatusIndicator"));

var _styles = require("./styles");

var _constants = require("./constants");

var _CELL_STYLES;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var CELL_SIZES = {
  XL: 'xl',
  LG: 'lg',
  MD: 'md',
  SM: 'sm',
  ICON: 'icon'
};
exports.CELL_SIZES = CELL_SIZES;
var CELL_TYPES = {
  DATE: 'date',
  STATUS: 'status',
  CUSTOM: 'custom'
};
exports.CELL_TYPES = CELL_TYPES;
var CELL_SIZE_VALUES = Object.values(CELL_SIZES);
var CELL_STYLES = (_CELL_STYLES = {}, _defineProperty(_CELL_STYLES, CELL_SIZES.XL, _styles.extraLargeCol), _defineProperty(_CELL_STYLES, CELL_SIZES.LG, _styles.largeCol), _defineProperty(_CELL_STYLES, CELL_SIZES.MD, _styles.mediumCol), _defineProperty(_CELL_STYLES, CELL_SIZES.SM, _styles.smallCol), _defineProperty(_CELL_STYLES, CELL_SIZES.ICON, _styles.iconCol), _CELL_STYLES);

var generateStyledVariants = function generateStyledVariants(Component) {
  return CELL_SIZE_VALUES.reduce(function (accum, cellSize) {
    return _objectSpread(_objectSpread({}, accum), {}, _defineProperty({}, cellSize, (0, _styledComponents["default"])(Component).withConfig({
      displayName: "Table__cellSize",
      componentId: "sc-1264zi4-0"
    })(["&&{", "}"], CELL_STYLES[cellSize])));
  }, {});
};

var TH_VARIANTS = generateStyledVariants(_TableHeaderItem["default"]);
var TD_VARIANTS = generateStyledVariants(_TableData["default"]);

var StyledNoResults = _styledComponents["default"].span.withConfig({
  displayName: "Table__StyledNoResults",
  componentId: "sc-1264zi4-1"
})(["display:block;font-weight:bold;margin-bottom:10px;"]);

var propTypes = {
  condensed: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  empty: _propTypes["default"].bool,
  graybox: _propTypes["default"].bool,
  handleRowClick: _propTypes["default"].func,
  handleTitleClick: _propTypes["default"].func,
  isLoading: _propTypes["default"].bool,
  noResultsText: _propTypes["default"].oneOf([_propTypes["default"].func, _propTypes["default"].string]),
  order: _propTypes["default"].string,
  orderBy: _propTypes["default"].string,
  rows: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    adState: _propTypes["default"].string,
    currency: _propTypes["default"].string,
    user: _propTypes["default"].object,
    totalBudget: _propTypes["default"].number,
    creationDate: _propTypes["default"].string,
    dateEnd: _propTypes["default"].string,
    impressions: _propTypes["default"].number,
    pacing: _propTypes["default"].number,
    dateBegin: _propTypes["default"].string,
    internalName: _propTypes["default"].string,
    name: _propTypes["default"].string,
    imageUri: _propTypes["default"].string,
    id: _propTypes["default"].string,
    extraContent: _propTypes["default"].node
  })),
  searchSuggestionText: _propTypes["default"].string,
  statusStrings: _propTypes["default"].object,
  tableSchema: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    heading: _propTypes["default"].string,
    sort: _propTypes["default"].string,
    type: _propTypes["default"].string,
    cell: _propTypes["default"].any
  })),
  searchWord: _propTypes["default"].string,
  sticky: _propTypes["default"].bool
};

var curryRowClick = function curryRowClick(cb, id) {
  return function () {
    cb(id);
  };
};

var NoResultsFound = function NoResultsFound(_ref) {
  var _ref$searchWord = _ref.searchWord,
      searchWord = _ref$searchWord === void 0 ? '' : _ref$searchWord,
      noResultsText = _ref.noResultsText,
      searchSuggestionText = _ref.searchSuggestionText;
  return /*#__PURE__*/_react["default"].createElement("p", null, /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
    source: typeof noResultsText === 'function' ? noResultsText(searchWord) : noResultsText,
    renderers: {
      paragraph: function paragraph(_ref2) {
        var children = _ref2.children;
        return (
          /*#__PURE__*/
          // eslint-disable-line
          _react["default"].createElement(StyledNoResults, null, children)
        );
      }
    }
  }), !!searchWord && /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
    source: searchSuggestionText,
    renderers: {
      paragraph: function paragraph(_ref3) {
        var children = _ref3.children;
        return (
          /*#__PURE__*/
          // eslint-disable-line
          _react["default"].createElement("span", null, children)
        );
      }
    }
  }));
};

var getBody = function getBody(schema, row, props) {
  switch (schema.type) {
    case CELL_TYPES.DATE:
      return (0, _date.formatDateWithZeros)((0, _get2["default"])(row, schema.cell));

    case CELL_TYPES.STATUS:
      return /*#__PURE__*/_react["default"].createElement(_StatusIndicator["default"], {
        status: row.adState,
        strings: props.statusStrings
      });

    case CELL_TYPES.CUSTOM:
      return schema.cell(row, props);

    default:
      return (0, _get2["default"])(row, schema.cell);
  }
};

var GrayboxCell = _styledComponents["default"].p.withConfig({
  displayName: "Table__GrayboxCell",
  componentId: "sc-1264zi4-2"
})(["background:", ";"], _encoreFoundation.gray90);

var GrayboxOverlay = _styledComponents["default"].div.withConfig({
  displayName: "Table__GrayboxOverlay",
  componentId: "sc-1264zi4-3"
})(["background:rgba(0,0,0,0.05);position:absolute;top:0;left:0;right:0;bottom:0;"]);

var getTitleText = function getTitleText(schema, row) {
  return (0, _get2["default"])(row, schema.titleText);
};

NoResultsFound.propTypes = {
  noResultsText: _propTypes["default"].oneOf([_propTypes["default"].func, _propTypes["default"].string]),
  searchSuggestionText: _propTypes["default"].string,
  searchWord: _propTypes["default"].string
};

function Table(props) {
  var tableClass = props.className,
      _props$condensed = props.condensed,
      condensed = _props$condensed === void 0 ? false : _props$condensed,
      empty = props.empty,
      graybox = props.graybox,
      handleRowClick = props.handleRowClick,
      handleTitleClick = props.handleTitleClick,
      isLoading = props.isLoading,
      noResultsText = props.noResultsText,
      order = props.order,
      orderBy = props.orderBy,
      rows = props.rows,
      searchSuggestionText = props.searchSuggestionText,
      searchWord = props.searchWord,
      _props$tableSchema = props.tableSchema,
      tableSchema = _props$tableSchema === void 0 ? [] : _props$tableSchema,
      sticky = props.sticky;

  var th = function th(name, value, className) {
    var isNumeric = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;
    var key = arguments.length > 4 ? arguments[4] : undefined;
    var tooltipText = arguments.length > 5 ? arguments[5] : undefined;
    var tooltipPlacement = arguments.length > 6 ? arguments[6] : undefined;
    var title = arguments.length > 7 ? arguments[7] : undefined;
    var size = arguments.length > 8 ? arguments[8] : undefined;
    var headerComponent = arguments.length > 9 ? arguments[9] : undefined;
    var valign = arguments.length > 10 ? arguments[10] : undefined;
    var wrap = arguments.length > 11 ? arguments[11] : undefined;
    var StyledTableHeaderItem = TH_VARIANTS[size] || TH_VARIANTS[CELL_SIZES.MD];
    return /*#__PURE__*/_react["default"].createElement(StyledTableHeaderItem, {
      alignRight: isNumeric,
      className: className,
      condensed: condensed,
      key: "Table-Header-".concat(key),
      name: name,
      onClick: value && handleTitleClick,
      order: order,
      orderBy: orderBy,
      tooltipPlacement: tooltipPlacement,
      tooltipText: tooltipText,
      title: title,
      value: value,
      headerComponent: headerComponent,
      valign: valign,
      wrap: wrap
    });
  };

  var td = function td(content, className, formatter) {
    var isNumeric = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : false;
    var key = arguments.length > 4 ? arguments[4] : undefined;
    var isPrimary = arguments.length > 5 ? arguments[5] : undefined;
    var hasOverflow = arguments.length > 6 ? arguments[6] : undefined;
    var columnName = arguments.length > 7 ? arguments[7] : undefined;
    var title = arguments.length > 8 ? arguments[8] : undefined;
    var size = arguments.length > 9 ? arguments[9] : undefined;
    var StyledTableData = TD_VARIANTS[size] || TD_VARIANTS[CELL_SIZES.MD];
    return /*#__PURE__*/_react["default"].createElement(StyledTableData, {
      alignRight: isNumeric,
      className: className,
      condensed: condensed,
      content: content,
      formatter: formatter,
      hasOverflow: hasOverflow,
      isPrimary: isPrimary,
      title: title,
      key: "Table-Data-".concat(key),
      "data-test": "table-data-".concat(columnName.toLowerCase().replace(' ', '-'))
    });
  };

  var body;

  if (rows) {
    body = /*#__PURE__*/_react["default"].createElement(_TableBody["default"], null, rows && rows.map(function (row) {
      var id = row.id,
          active = row.active;
      var handleClick = handleRowClick && curryRowClick(handleRowClick, id);
      return /*#__PURE__*/_react["default"].createElement(_TableRow["default"], {
        active: active,
        href: row.href,
        key: "".concat(row.id, "-tr"),
        onClick: handleClick,
        extraContent: row.extraContent
      }, tableSchema.map(function (item, key) {
        return td(getBody(item, row, props), item.className, item.formatter, item.isNumeric, key, item.isPrimary, item.hasOverflow, item.heading, getTitleText(item, row), item.size);
      }));
    }));
  } else if (graybox) {
    body = /*#__PURE__*/_react["default"].createElement(_TableBody["default"], {
      fixedHeight: 648
    }, /*#__PURE__*/_react["default"].createElement(_TableRow["default"], null, tableSchema.map(function (item, key) {
      return td( /*#__PURE__*/_react["default"].createElement(GrayboxCell, null, "\xA0"), item.className, item.formatter, item.isNumeric, key, item.isPrimary, item.hasOverflow, item.heading, '', item.size);
    })), /*#__PURE__*/_react["default"].createElement(GrayboxOverlay, null));
  }

  return /*#__PURE__*/_react["default"].createElement(_TableContainer["default"], {
    className: tableClass,
    body: body,
    empty: !graybox && (empty || !rows) && /*#__PURE__*/_react["default"].createElement(NoResultsFound, {
      noResultsText: noResultsText,
      searchWord: searchWord,
      searchSuggestionText: searchSuggestionText
    }),
    header: /*#__PURE__*/_react["default"].createElement(_TableHeader["default"], {
      sticky: sticky
    }, tableSchema.map(function (item, key) {
      var heading = item.heading;
      var headingLines = heading.split('\n');

      if (headingLines.length > 1) {
        heading = headingLines.map(function (headingLine) {
          return /*#__PURE__*/_react["default"].createElement("span", null, headingLine, /*#__PURE__*/_react["default"].createElement("br", null));
        });
      }

      return th(heading, item.sort, item.className, item.isNumeric, key, item.tooltipText, item.tooltipPlacement, item.title, item.size, item.headerComponent, item.valign, item.wrap);
    })),
    isLoading: isLoading
  });
}

Table.propTypes = propTypes;
Table.defaultProps = {
  noResultsText: _constants.noResultsText,
  searchSuggestionText: _constants.searchSuggestionText
};
