// @spotify-internal/adstudio-shared/lib/components/BulkLocationsUploader/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BULK_LOCATION_STRINGS = exports.LOCATION_INPUT_FIELD_NAME = exports.ERRORS_TAB = exports.MATCHES_TAB = void 0;
// Element identifiers
var MATCHES_TAB = 'matches-tab';
exports.MATCHES_TAB = MATCHES_TAB;
var ERRORS_TAB = 'errors-tab'; // Redux-form

exports.ERRORS_TAB = ERRORS_TAB;
var LOCATION_INPUT_FIELD_NAME = 'bulk-location-input';
exports.LOCATION_INPUT_FIELD_NAME = LOCATION_INPUT_FIELD_NAME;
var BULK_LOCATION_STRINGS = {
  // Modal Text
  CANCEL_BUTTON: 'Cancel',
  DIALOG_TITLE: 'Bulk add postal codes',
  UPLOAD_BODY_TEXT: 'Type or paste postal codes below',
  UPLOAD_SUBDESCRIPTION_TEXT: 'Put each code on a new line, or use semicolons between them.',
  PLACEHOLDER_TEXT: '90210\n90211\n89109; 89108; 13579',
  TABLE_HEADER_TEXT: 'Location',
  LOCATION_COUNT_TEXT: '[num]/[limit] location(s) entered',
  LOCATION_LIMIT_ERROR_TEXT: 'A total of [limit] locations (including cities, states, etc.) is allowed.',
  // Matches Text
  MATCHES_TAB_TITLE: 'Matched Postal Codes',
  MATCHES_TAB_BODY_TEXT: 'Review matches',
  MATCHES_TAB_SUBDESCRIPTION_TEXT: 'You can remove items in the table',
  NO_MATCHES_TEXT: 'No matches found',
  // Errors Text
  ERRORS_TAB_TITLE: 'Errors',
  ERRORS_TAB_BODY_TEXT: 'Review errors',
  ERRORS_TAB_SUBDESCRIPTION_TEXT: 'You can remove items in the table',
  NO_ERRORS_TEXT: 'No errors found',
  // Button Text
  MATCH_LOCATIONS_BUTTON_TEXT: 'Match Codes',
  ADD_LOCATIONS_BUTTON_TEXT: 'Add [num] codes'
};
exports.BULK_LOCATION_STRINGS = BULK_LOCATION_STRINGS;
