// @spotify-internal/adstudio-shared/lib/components/BulkLocationsUploader/LocationsTable.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LocationsTable = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _isEmpty = _interopRequireDefault(require("lodash/isEmpty"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _constants = require("./constants");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconX = _adstudioTape.Icons.IconX;
var TITLE_HEIGHT = _encoreAdvertisingWeb.spacer40;
var LOCATION_ROW_HEIGHT = _encoreAdvertisingWeb.spacer48;
var StyledIconX = (0, _styledComponents["default"])(IconX).withConfig({
  displayName: "LocationsTable__StyledIconX",
  componentId: "kbiee5-0"
})(["margin-left:auto;visibility:hidden;"]);
var LocationItem = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "LocationsTable__LocationItem",
  componentId: "kbiee5-1"
})(["align-items:center;display:flex;color:", ";&&:hover{", "{visibility:visible;}}"], function (props) {
  return props.color || 'black';
}, StyledIconX);
var TabSubText = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "LocationsTable__TabSubText",
  componentId: "kbiee5-2"
})(["padding-bottom:", ";"], _encoreAdvertisingWeb.spacer8);
var MatchesContainer = (0, _styledComponents["default"])(_encoreAdvertisingWeb.TableContainer).withConfig({
  displayName: "LocationsTable__MatchesContainer",
  componentId: "kbiee5-3"
})(["margin-top:", ";"], _encoreAdvertisingWeb.spacer8);
var NoMatchesText = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "LocationsTable__NoMatchesText",
  componentId: "kbiee5-4"
})(["display:table;margin:auto;color:", ";margin-top:1em;"], function (props) {
  return props.color || 'black';
});
var NoMatchesErrorRow = (0, _styledComponents["default"])(_encoreAdvertisingWeb.TableRow).withConfig({
  displayName: "LocationsTable__NoMatchesErrorRow",
  componentId: "kbiee5-5"
})(["border-bottom:none;"]);

var LoadingSpinnerContainer = _styledComponents["default"].div.withConfig({
  displayName: "LocationsTable__LoadingSpinnerContainer",
  componentId: "kbiee5-6"
})(["display:flex;justify-content:center;overflow:hidden;"]);

var StyledTableRow = (0, _styledComponents["default"])(_encoreAdvertisingWeb.TableRow).withConfig({
  displayName: "LocationsTable__StyledTableRow",
  componentId: "kbiee5-7"
})(["display:grid !important;"]);

var renderTableBody = function renderTableBody(_ref) {
  var items = _ref.items,
      strings = _ref.strings,
      styleForErrors = _ref.styleForErrors,
      onDeleteLocation = _ref.onDeleteLocation;
  var NO_ERRORS_TEXT = strings.NO_ERRORS_TEXT,
      NO_MATCHES_TEXT = strings.NO_MATCHES_TEXT;

  if ((0, _isEmpty["default"])(items)) {
    return /*#__PURE__*/_react["default"].createElement(NoMatchesErrorRow, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.TableCell, null, /*#__PURE__*/_react["default"].createElement(NoMatchesText, {
      variant: _encoreAdvertisingWeb.Type.body1,
      weight: _encoreAdvertisingWeb.Type.book,
      color: _encoreAdvertisingWeb.gray75
    }, styleForErrors ? NO_ERRORS_TEXT : NO_MATCHES_TEXT)));
  }

  return items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(StyledTableRow, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.TableCell, {
      key: "locations-table-".concat(item)
    }, /*#__PURE__*/_react["default"].createElement(LocationItem, {
      id: "matched-item-".concat(item.name),
      variant: _encoreAdvertisingWeb.Type.body1,
      weight: _encoreAdvertisingWeb.Type.book,
      condensed: true,
      color: styleForErrors && _encoreAdvertisingWeb.failure
    }, styleForErrors ? "".concat(item) : "".concat(item.name, ": ").concat(item.parentName), /*#__PURE__*/_react["default"].createElement(StyledIconX, {
      color: _encoreAdvertisingWeb.gray75,
      iconSize: 16,
      onClick: function onClick() {
        onDeleteLocation(item);
      }
    }))));
  });
};

var LocationsTable = function LocationsTable(_ref2) {
  var hasFetchedLocations = _ref2.hasFetchedLocations,
      items = _ref2.items,
      onDeleteLocation = _ref2.onDeleteLocation,
      styleForErrors = _ref2.styleForErrors,
      strings = _ref2.strings,
      id = _ref2.id;
  var ERRORS_TAB_BODY_TEXT = strings.ERRORS_TAB_BODY_TEXT,
      MATCHES_TAB_BODY_TEXT = strings.MATCHES_TAB_BODY_TEXT,
      ERRORS_TAB_SUBDESCRIPTION_TEXT = strings.ERRORS_TAB_SUBDESCRIPTION_TEXT,
      MATCHES_TAB_SUBDESCRIPTION_TEXT = strings.MATCHES_TAB_SUBDESCRIPTION_TEXT,
      TABLE_HEADER_TEXT = strings.TABLE_HEADER_TEXT;
  var isLoading = (0, _isEmpty["default"])(items) && !hasFetchedLocations;
  return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
    variant: _encoreAdvertisingWeb.Type.body1,
    weight: _encoreAdvertisingWeb.Type.bold,
    condensed: true
  }, styleForErrors ? ERRORS_TAB_BODY_TEXT : MATCHES_TAB_BODY_TEXT), /*#__PURE__*/_react["default"].createElement(TabSubText, {
    variant: _encoreAdvertisingWeb.Type.body1,
    weight: _encoreAdvertisingWeb.Type.book,
    condensed: true
  }, styleForErrors ? ERRORS_TAB_SUBDESCRIPTION_TEXT : MATCHES_TAB_SUBDESCRIPTION_TEXT), /*#__PURE__*/_react["default"].createElement(MatchesContainer, {
    style: {
      height: "calc(".concat(TITLE_HEIGHT, " + ").concat(LOCATION_ROW_HEIGHT, "*6)")
    },
    stickyHeader: true,
    id: id
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Table, null, /*#__PURE__*/_react["default"].createElement("thead", null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.TableRow, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.TableHeaderCell, null, /*#__PURE__*/_react["default"].createElement(LocationItem, {
    color: _encoreAdvertisingWeb.gray60,
    variant: _encoreAdvertisingWeb.Type.body3,
    weight: _encoreAdvertisingWeb.Type.bold,
    condensed: true
  }, TABLE_HEADER_TEXT)))), /*#__PURE__*/_react["default"].createElement("tbody", null, isLoading ? /*#__PURE__*/_react["default"].createElement(LoadingSpinnerContainer, null, /*#__PURE__*/_react["default"].createElement(_adstudioTape.LoadingSpinner, {
    diameter: "15vh"
  })) : renderTableBody({
    items: items,
    strings: strings,
    styleForErrors: styleForErrors,
    onDeleteLocation: onDeleteLocation
  })))));
};

exports.LocationsTable = LocationsTable;
var _default = LocationsTable;
exports["default"] = _default;
LocationsTable.propTypes = {
  hasFetchedLocations: _propTypes["default"].bool,
  items: _propTypes["default"].array,
  onDeleteLocation: _propTypes["default"].func,
  strings: _propTypes["default"].obbject,
  styleForErrors: _propTypes["default"].bool,
  id: _propTypes["default"].string
};
LocationsTable.defaultProps = {
  strings: _constants.BULK_LOCATION_STRINGS
};
