// @spotify-internal/adstudio-shared/lib/components/BulkLocationsUploader/LocationsReview.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.LocationsReview = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _recompose = require("recompose");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _constants = require("./constants");

var _LocationsTable = _interopRequireDefault(require("./LocationsTable"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var StyledTabsListItem = (0, _styledComponents["default"])(_adstudioTape.TabsListItem).withConfig({
  displayName: "LocationsReview__StyledTabsListItem",
  componentId: "g3vxpu-0"
})(["p{padding-top:0.5em;padding-bottom:0.5em;}"]);

var LocationsReview = /*#__PURE__*/function (_Component) {
  _inherits(LocationsReview, _Component);

  var _super = _createSuper(LocationsReview);

  function LocationsReview() {
    _classCallCheck(this, LocationsReview);

    return _super.apply(this, arguments);
  }

  _createClass(LocationsReview, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          hasFetchedLocations = _this$props.hasFetchedLocations,
          matchedLocations = _this$props.matchedLocations,
          _this$props$numErrors = _this$props.numErrors,
          numErrors = _this$props$numErrors === void 0 ? 0 : _this$props$numErrors,
          _this$props$numMatche = _this$props.numMatches,
          numMatches = _this$props$numMatche === void 0 ? 0 : _this$props$numMatche,
          onDeleteError = _this$props.onDeleteError,
          onDeleteLocation = _this$props.onDeleteLocation,
          selectedTab = _this$props.selectedTab,
          setTab = _this$props.setTab,
          strings = _this$props.strings,
          unmatchedLocations = _this$props.unmatchedLocations;
      var MATCHES_TAB_TITLE = strings.MATCHES_TAB_TITLE,
          ERRORS_TAB_TITLE = strings.ERRORS_TAB_TITLE;
      return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_adstudioTape.Tabs, {
        list: /*#__PURE__*/_react["default"].createElement(_adstudioTape.TabsList, null, /*#__PURE__*/_react["default"].createElement(StyledTabsListItem, {
          isActive: selectedTab === _constants.MATCHES_TAB,
          label: "".concat(MATCHES_TAB_TITLE, " (").concat(numMatches, ")"),
          onClick: function onClick() {
            return setTab(_constants.MATCHES_TAB);
          }
        }), /*#__PURE__*/_react["default"].createElement(StyledTabsListItem, {
          isActive: selectedTab === _constants.ERRORS_TAB,
          label: "".concat(ERRORS_TAB_TITLE, " (").concat(numErrors, ")"),
          onClick: function onClick() {
            return setTab(_constants.ERRORS_TAB);
          }
        }))
      }), /*#__PURE__*/_react["default"].createElement("div", null, selectedTab === _constants.MATCHES_TAB && /*#__PURE__*/_react["default"].createElement(_LocationsTable["default"], {
        id: "bulk-locations-upload-review",
        hasFetchedLocations: hasFetchedLocations,
        items: matchedLocations,
        onDeleteLocation: onDeleteLocation,
        strings: strings
      }), selectedTab === _constants.ERRORS_TAB && /*#__PURE__*/_react["default"].createElement(_LocationsTable["default"], {
        id: "bulk-locations-upload-errors",
        hasFetchedLocations: hasFetchedLocations,
        items: unmatchedLocations,
        onDeleteLocation: onDeleteError,
        strings: strings,
        styleForErrors: true
      })));
    }
  }]);

  return LocationsReview;
}(_react.Component);

exports.LocationsReview = LocationsReview;
LocationsReview.propTypes = {
  hasFetchedLocations: _propTypes["default"].bool,
  matchedLocations: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    geoId: _propTypes["default"].string,
    name: _propTypes["default"].string,
    parentName: _propTypes["default"].string,
    type: _propTypes["default"].string,
    typeDisplayName: _propTypes["default"].string
  })),
  numErrors: _propTypes["default"].number,
  numMatches: _propTypes["default"].number,
  onDeleteError: _propTypes["default"].func.isRequired,
  onDeleteLocation: _propTypes["default"].func,
  selectedTab: _propTypes["default"].string,
  setTab: _propTypes["default"].func,
  strings: _propTypes["default"].object,
  unmatchedLocations: _propTypes["default"].array
};
LocationsReview.defaultProps = {
  strings: _constants.BULK_LOCATION_STRINGS
};

var _default = (0, _recompose.compose)((0, _recompose.withState)('selectedTab', 'setTab', _constants.MATCHES_TAB))(LocationsReview);

exports["default"] = _default;
