// @spotify-internal/encore-web/es/components/FormSelect/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { Select } from "./Select";
import { SelectArrow } from "./SelectArrow";
import { SelectContainer } from "./SelectContainer";
var FormSelectComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      props = _objectWithoutProperties(_ref, ["theme"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(SelectContainer, null, /*#__PURE__*/React.createElement(Select, Object.assign({
    ref: ref,
    isUsingKeyboard: isUsingKeyboard
  }, props)), /*#__PURE__*/React.createElement(SelectArrow, null)));
});
var formSelectHelpers = {
  darkTheme: {
    formSelect: 'dark'
  },
  displayName: 'FormSelect'
};
/**
* **FormSelect** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/FormSelect) | [Storybook](https://encore-web.spotify.net/?path=/story/components-form-formselect--default) | [Encore](https://encore.spotify.net/web/components/form?format=figma-figma#formselect) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:844) |
*
* FormSelect allows the user to select one option from a long list of potential options and is used within a Form
*
* @example
* export const Default = () => (
*  <FormSelect defaultValue="">
*    <option disabled value="">
*      Choose one
*    </option>
*    <option value="hello">Hello</option>
*    <option value="hola">Hola</option>
*    <option value="hej">Hej</option>
*    <option value="hallo">Hallo</option>
*  </FormSelect>
*);
*
*/

export var FormSelect = Object.assign(FormSelectComponent, formSelectHelpers);
export default FormSelect;
