// @spotify-internal/adstudio-shared/lib/components/SelectableCell/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.mapDispatchToProps = exports.mapStateToProps = exports.Index = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactRedux = require("react-redux");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var DEFAULT_STRINGS = {
  ALL: 'All selected',
  NONE: 'None selected',
  PARTIAL: 'Partially selected'
};
var SelectableCellContainer = (0, _styledComponents["default"])(_encoreAdvertisingWeb.TableCell).withConfig({
  displayName: "SelectableCell__SelectableCellContainer",
  componentId: "vc37jm-0"
})(["", ""], function (_ref) {
  var color = _ref.color,
      size = _ref.size;
  return "\n    color: ".concat(color, ";\n    display: flex;\n    font-size: ").concat(size, "px;\n    padding-left: 0;\n    padding-top: ").concat(size, "px;\n  ");
});
var SelectableHeaderCellContainer = (0, _styledComponents["default"])(_encoreAdvertisingWeb.TableHeaderCell).withConfig({
  displayName: "SelectableCell__SelectableHeaderCellContainer",
  componentId: "vc37jm-1"
})(["", ""], function (_ref2) {
  var color = _ref2.color,
      size = _ref2.size;
  return "\n    color: ".concat(color, ";\n    display: flex;\n    font-size: ").concat(size * 0.75, "px;\n    line-height: ").concat(size * 1.5, "px;\n    padding-left: ").concat(size, "px;\n    padding-top: ").concat(size, "px;\n  ");
});

var SelectableContentContainer = _styledComponents["default"].div.withConfig({
  displayName: "SelectableCell__SelectableContentContainer",
  componentId: "vc37jm-2"
})(["", ""], function (_ref3) {
  var size = _ref3.size;
  return "\n    padding-bottom: ".concat(size / 8, "px;\n    position: relative;\n    top: 0;\n  ");
});

var StyledCheckboxContainer = _styledComponents["default"].div.withConfig({
  displayName: "SelectableCell__StyledCheckboxContainer",
  componentId: "vc37jm-3"
})(["", ""], function (_ref4) {
  var isHeader = _ref4.isHeader,
      size = _ref4.size;
  return "\n    padding: ".concat(size / (isHeader ? 4 : 8), "px ").concat(size, "px ").concat(size / 4, "px 0;\n  ");
});

var StyledLink = _styledComponents["default"].a.withConfig({
  displayName: "SelectableCell__StyledLink",
  componentId: "vc37jm-4"
})(["text-decoration:none;"]);

var Index = /*#__PURE__*/function (_PureComponent) {
  _inherits(Index, _PureComponent);

  var _super = _createSuper(Index);

  function Index() {
    _classCallCheck(this, Index);

    return _super.apply(this, arguments);
  }

  _createClass(Index, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          checked = _this$props.checked,
          children = _this$props.children,
          indeterminate = _this$props.indeterminate,
          isHeader = _this$props.isHeader,
          linkUrl = _this$props.linkUrl,
          size = _this$props.size,
          strings = _this$props.strings,
          textColor = _this$props.textColor,
          toggler = _this$props.toggler;
      var color = textColor || isHeader ? _encoreAdvertisingWeb.gray60 : _encoreAdvertisingWeb.gray50;
      var SelectableContainer = isHeader ? SelectableHeaderCellContainer : SelectableCellContainer;
      var stateLabel = strings.NONE;

      if (indeterminate) {
        stateLabel = strings.PARTIAL;
      }

      if (checked) {
        stateLabel = strings.ALL;
      }

      return /*#__PURE__*/_react["default"].createElement(SelectableContainer, {
        color: color,
        size: size
      }, /*#__PURE__*/_react["default"].createElement(StyledCheckboxContainer, {
        isHeader: isHeader,
        size: size
      }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.TableCheckbox, {
        "aria-label": stateLabel,
        onChange: toggler,
        checked: checked,
        indeterminate: indeterminate
      })), /*#__PURE__*/_react["default"].createElement(SelectableContentContainer, {
        size: size
      }, linkUrl ? /*#__PURE__*/_react["default"].createElement(StyledLink, {
        textColor: textColor,
        href: linkUrl
      }, children) : children));
    }
  }]);

  return Index;
}(_react.PureComponent);

exports.Index = Index;
Index.defaultProps = {
  checked: false,
  indeterminate: false,
  isHeader: false,
  size: 16,
  strings: DEFAULT_STRINGS
};

var mapStateToProps = function mapStateToProps(state, ownProps) {
  return {
    checked: ownProps.selector(state, ownProps.id),
    indeterminate: ownProps.indeterminateSelector ? ownProps.indeterminateSelector(state) : false
  };
};

exports.mapStateToProps = mapStateToProps;

var mapDispatchToProps = function mapDispatchToProps(dispatch, ownProps) {
  return {
    toggler: function toggler(value) {
      return dispatch(ownProps.action({
        id: ownProps.id,
        value: value.target.checked
      }));
    }
  };
};

exports.mapDispatchToProps = mapDispatchToProps;

var _default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(Index);

exports["default"] = _default;
