// @spotify-internal/encore-web-v3/es/components/NavBarListItem/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { Link } from "./Link";
import { ListItem } from "./ListItem";
var NavBarListItemComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var label = _ref.label,
      active = _ref.active,
      disabled = _ref.disabled,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      _ref$sentenceCase = _ref.sentenceCase,
      sentenceCase = _ref$sentenceCase === void 0 ? false : _ref$sentenceCase,
      props = _objectWithoutProperties(_ref, ["label", "active", "disabled", "theme", "component", "sentenceCase"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(ListItem, null, /*#__PURE__*/React.createElement(Link, Object.assign({
    ref: ref,
    isActive: active,
    isDisabled: disabled,
    component: component,
    sentenceCase: sentenceCase,
    "aria-current": active ? 'page' : undefined
  }, props), label)));
});
var navBarListItemHelpers = {
  darkTheme: {
    navBarListItem: 'dark'
  },
  displayName: 'NavBarListItem'
};
/**
* **NavBarListItem** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavBarListItem) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navbar-navbarlistitem--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-bar) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=377:171) |
*
* An individual tab within the NavBar component
*
* @example
* export const Default = () => (
*  <NavBarList>
*    <NavBarListItem label="Overview" href="#navbar-overview" />
*  </NavBarList>
*);
*
*/

export var NavBarListItem = Object.assign(NavBarListItemComponent, navBarListItemHelpers);
export default NavBarListItem;
