// @spotify-internal/ubi-sdk-s4a-web-common/WebCommonEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebCommonEventFactory = exports.createWebCommonEventFactory = void 0;
var SPECIFICATION_ID = "web-common";
var APP = "s4a";
var SPECIFICATION_VERSION = "3.0.2";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebCommonEventFactory(options) {
    return new WebCommonEventFactory(options);
}
exports.createWebCommonEventFactory = createWebCommonEventFactory;
/**
 * common components across s4a
 */
var WebCommonEventFactory = /** @class */ (function () {
    function WebCommonEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of ArtistSwitcherEventFactory
    */
    WebCommonEventFactory.prototype.artistSwitcherFactory = function () {
        return new WebCommonEventFactory.ArtistSwitcherEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SideNavPanelEventFactory
    */
    WebCommonEventFactory.prototype.sideNavPanelFactory = function () {
        return new WebCommonEventFactory.SideNavPanelEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of TopNavBarEventFactory
    */
    WebCommonEventFactory.prototype.topNavBarFactory = function () {
        return new WebCommonEventFactory.TopNavBarEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of FooterEventFactory
    */
    WebCommonEventFactory.prototype.footerFactory = function () {
        return new WebCommonEventFactory.FooterEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of HelpButtonEventFactory
    */
    WebCommonEventFactory.prototype.helpButtonFactory = function () {
        return new WebCommonEventFactory.HelpButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebCommonEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebCommonEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebCommonEventFactory;
}());
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    /**
     * Create ArtistSwitcherEventFactory class with path
     * @param path to this node
     */
    var ArtistSwitcherEventFactory = /** @class */ (function () {
        function ArtistSwitcherEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "artist_switcher"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        ArtistSwitcherEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ArtistSwitcherEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        ArtistSwitcherEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ArtistSwitcherEventFactory;
    }());
    WebCommonEventFactory.ArtistSwitcherEventFactory = ArtistSwitcherEventFactory;
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    /**
     * Create SideNavPanelEventFactory class with path
     * @param path to this node
     */
    var SideNavPanelEventFactory = /** @class */ (function () {
        function SideNavPanelEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "side_nav_panel"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of SearchButtonEventFactory
        */
        SideNavPanelEventFactory.prototype.searchButtonFactory = function () {
            return new WebCommonEventFactory.SideNavPanelEventFactory.SearchButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of SearchBarEventFactory
        */
        SideNavPanelEventFactory.prototype.searchBarFactory = function () {
            return new WebCommonEventFactory.SideNavPanelEventFactory.SearchBarEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ArtistSelectorEventFactory
        */
        SideNavPanelEventFactory.prototype.artistSelectorFactory = function () {
            return new WebCommonEventFactory.SideNavPanelEventFactory.ArtistSelectorEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of OtherLinksEventFactory
        */
        SideNavPanelEventFactory.prototype.otherLinksFactory = function () {
            return new WebCommonEventFactory.SideNavPanelEventFactory.OtherLinksEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of UserRowEventFactory
        */
        SideNavPanelEventFactory.prototype.userRowFactory = function () {
            return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        SideNavPanelEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SideNavPanelEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for SideNavPanelEventFactory.
         */
        SideNavPanelEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SideNavPanelEventFactory;
    }());
    WebCommonEventFactory.SideNavPanelEventFactory = SideNavPanelEventFactory;
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    /**
     * Create TopNavBarEventFactory class with path
     * @param path to this node
     */
    var TopNavBarEventFactory = /** @class */ (function () {
        function TopNavBarEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "top_nav_bar"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of HomeLinkEventFactory
        */
        TopNavBarEventFactory.prototype.homeLinkFactory = function () {
            return new WebCommonEventFactory.TopNavBarEventFactory.HomeLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of MusicLinkEventFactory
        */
        TopNavBarEventFactory.prototype.musicLinkFactory = function () {
            return new WebCommonEventFactory.TopNavBarEventFactory.MusicLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of AudienceLinkEventFactory
        */
        TopNavBarEventFactory.prototype.audienceLinkFactory = function () {
            return new WebCommonEventFactory.TopNavBarEventFactory.AudienceLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ProfileLinkEventFactory
        */
        TopNavBarEventFactory.prototype.profileLinkFactory = function () {
            return new WebCommonEventFactory.TopNavBarEventFactory.ProfileLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of CampaignsLinkEventFactory
        */
        TopNavBarEventFactory.prototype.campaignsLinkFactory = function () {
            return new WebCommonEventFactory.TopNavBarEventFactory.CampaignsLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        TopNavBarEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TopNavBarEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for TopNavBarEventFactory.
         */
        TopNavBarEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return TopNavBarEventFactory;
    }());
    WebCommonEventFactory.TopNavBarEventFactory = TopNavBarEventFactory;
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    /**
     * Create FooterEventFactory class with path
     * @param path to this node
     */
    var FooterEventFactory = /** @class */ (function () {
        function FooterEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "footer"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of LegalLinkEventFactory
        */
        FooterEventFactory.prototype.legalLinkFactory = function () {
            return new WebCommonEventFactory.FooterEventFactory.LegalLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of PrivacyLinkEventFactory
        */
        FooterEventFactory.prototype.privacyLinkFactory = function () {
            return new WebCommonEventFactory.FooterEventFactory.PrivacyLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of CookiesLinkEventFactory
        */
        FooterEventFactory.prototype.cookiesLinkFactory = function () {
            return new WebCommonEventFactory.FooterEventFactory.CookiesLinkEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        FooterEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        FooterEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for FooterEventFactory.
         */
        FooterEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return FooterEventFactory;
    }());
    WebCommonEventFactory.FooterEventFactory = FooterEventFactory;
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    /**
     * Create HelpButtonEventFactory class with path
     * @param path to this node
     */
    var HelpButtonEventFactory = /** @class */ (function () {
        function HelpButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "help_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        HelpButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        HelpButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        HelpButtonEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return HelpButtonEventFactory;
    }());
    WebCommonEventFactory.HelpButtonEventFactory = HelpButtonEventFactory;
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        /**
         * Create SearchButtonEventFactory class with path
         * @param path to this node
         */
        var SearchButtonEventFactory = /** @class */ (function () {
            function SearchButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "search_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SearchButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SearchButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_element_toggle action.
             */
            SearchButtonEventFactory.prototype.hitUiElementToggle = function () {
                var data = {
                    actionName: "ui_element_toggle",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SearchButtonEventFactory;
        }());
        SideNavPanelEventFactory.SearchButtonEventFactory = SearchButtonEventFactory;
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        /**
         * Create SearchBarEventFactory class with path
         * @param path to this node
         */
        var SearchBarEventFactory = /** @class */ (function () {
            function SearchBarEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "search_bar"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of TextAreaEventFactory
            */
            SearchBarEventFactory.prototype.textAreaFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.SearchBarEventFactory.TextAreaEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of CloseButtonEventFactory
            */
            SearchBarEventFactory.prototype.closeButtonFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.SearchBarEventFactory.CloseButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            SearchBarEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SearchBarEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for SearchBarEventFactory.
             */
            SearchBarEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SearchBarEventFactory;
        }());
        SideNavPanelEventFactory.SearchBarEventFactory = SearchBarEventFactory;
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        /**
         * Create ArtistSelectorEventFactory class with path
         * @param path to this node
         */
        var ArtistSelectorEventFactory = /** @class */ (function () {
            function ArtistSelectorEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "artist_selector"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param uri of the ArtistRowEventFactory
            * @returns A new instance of ArtistRowEventFactory
            */
            ArtistSelectorEventFactory.prototype.artistRowFactory = function (data) {
                return new WebCommonEventFactory.SideNavPanelEventFactory.ArtistSelectorEventFactory.ArtistRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            ArtistSelectorEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ArtistSelectorEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return ArtistSelectorEventFactory;
        }());
        SideNavPanelEventFactory.ArtistSelectorEventFactory = ArtistSelectorEventFactory;
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        /**
         * Create OtherLinksEventFactory class with path
         * @param path to this node
         */
        var OtherLinksEventFactory = /** @class */ (function () {
            function OtherLinksEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "other_links"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of YourRosterRowEventFactory
            */
            OtherLinksEventFactory.prototype.yourRosterRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.OtherLinksEventFactory.YourRosterRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of YourTeamsRowEventFactory
            */
            OtherLinksEventFactory.prototype.yourTeamsRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.OtherLinksEventFactory.YourTeamsRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            OtherLinksEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            OtherLinksEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return OtherLinksEventFactory;
        }());
        SideNavPanelEventFactory.OtherLinksEventFactory = OtherLinksEventFactory;
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        /**
         * Create UserRowEventFactory class with path
         * @param path to this node
         */
        var UserRowEventFactory = /** @class */ (function () {
            function UserRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "user_row"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of EmployeeFeedbackRowEventFactory
            */
            UserRowEventFactory.prototype.employeeFeedbackRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory.EmployeeFeedbackRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of UserSettingsRowEventFactory
            */
            UserRowEventFactory.prototype.userSettingsRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory.UserSettingsRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of AddTeamRowEventFactory
            */
            UserRowEventFactory.prototype.addTeamRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory.AddTeamRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of HelpRowEventFactory
            */
            UserRowEventFactory.prototype.helpRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory.HelpRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of LogOutRowEventFactory
            */
            UserRowEventFactory.prototype.logOutRowFactory = function () {
                return new WebCommonEventFactory.SideNavPanelEventFactory.UserRowEventFactory.LogOutRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            UserRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            UserRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            UserRowEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return UserRowEventFactory;
        }());
        SideNavPanelEventFactory.UserRowEventFactory = UserRowEventFactory;
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var TopNavBarEventFactory;
    (function (TopNavBarEventFactory) {
        /**
         * Create HomeLinkEventFactory class with path
         * @param path to this node
         */
        var HomeLinkEventFactory = /** @class */ (function () {
            function HomeLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "home_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            HomeLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            HomeLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            HomeLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return HomeLinkEventFactory;
        }());
        TopNavBarEventFactory.HomeLinkEventFactory = HomeLinkEventFactory;
    })(TopNavBarEventFactory = WebCommonEventFactory.TopNavBarEventFactory || (WebCommonEventFactory.TopNavBarEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var TopNavBarEventFactory;
    (function (TopNavBarEventFactory) {
        /**
         * Create MusicLinkEventFactory class with path
         * @param path to this node
         */
        var MusicLinkEventFactory = /** @class */ (function () {
            function MusicLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "music_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            MusicLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            MusicLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            MusicLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return MusicLinkEventFactory;
        }());
        TopNavBarEventFactory.MusicLinkEventFactory = MusicLinkEventFactory;
    })(TopNavBarEventFactory = WebCommonEventFactory.TopNavBarEventFactory || (WebCommonEventFactory.TopNavBarEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var TopNavBarEventFactory;
    (function (TopNavBarEventFactory) {
        /**
         * Create AudienceLinkEventFactory class with path
         * @param path to this node
         */
        var AudienceLinkEventFactory = /** @class */ (function () {
            function AudienceLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "audience_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AudienceLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AudienceLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            AudienceLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AudienceLinkEventFactory;
        }());
        TopNavBarEventFactory.AudienceLinkEventFactory = AudienceLinkEventFactory;
    })(TopNavBarEventFactory = WebCommonEventFactory.TopNavBarEventFactory || (WebCommonEventFactory.TopNavBarEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var TopNavBarEventFactory;
    (function (TopNavBarEventFactory) {
        /**
         * Create ProfileLinkEventFactory class with path
         * @param path to this node
         */
        var ProfileLinkEventFactory = /** @class */ (function () {
            function ProfileLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "profile_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ProfileLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ProfileLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            ProfileLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ProfileLinkEventFactory;
        }());
        TopNavBarEventFactory.ProfileLinkEventFactory = ProfileLinkEventFactory;
    })(TopNavBarEventFactory = WebCommonEventFactory.TopNavBarEventFactory || (WebCommonEventFactory.TopNavBarEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var TopNavBarEventFactory;
    (function (TopNavBarEventFactory) {
        /**
         * Create CampaignsLinkEventFactory class with path
         * @param path to this node
         */
        var CampaignsLinkEventFactory = /** @class */ (function () {
            function CampaignsLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "campaigns_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CampaignsLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CampaignsLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            CampaignsLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create impression for CampaignsLinkEventFactory.
             */
            CampaignsLinkEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CampaignsLinkEventFactory;
        }());
        TopNavBarEventFactory.CampaignsLinkEventFactory = CampaignsLinkEventFactory;
    })(TopNavBarEventFactory = WebCommonEventFactory.TopNavBarEventFactory || (WebCommonEventFactory.TopNavBarEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var FooterEventFactory;
    (function (FooterEventFactory) {
        /**
         * Create LegalLinkEventFactory class with path
         * @param path to this node
         */
        var LegalLinkEventFactory = /** @class */ (function () {
            function LegalLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "legal_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            LegalLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            LegalLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            LegalLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return LegalLinkEventFactory;
        }());
        FooterEventFactory.LegalLinkEventFactory = LegalLinkEventFactory;
    })(FooterEventFactory = WebCommonEventFactory.FooterEventFactory || (WebCommonEventFactory.FooterEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var FooterEventFactory;
    (function (FooterEventFactory) {
        /**
         * Create PrivacyLinkEventFactory class with path
         * @param path to this node
         */
        var PrivacyLinkEventFactory = /** @class */ (function () {
            function PrivacyLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "privacy_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            PrivacyLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            PrivacyLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            PrivacyLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return PrivacyLinkEventFactory;
        }());
        FooterEventFactory.PrivacyLinkEventFactory = PrivacyLinkEventFactory;
    })(FooterEventFactory = WebCommonEventFactory.FooterEventFactory || (WebCommonEventFactory.FooterEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var FooterEventFactory;
    (function (FooterEventFactory) {
        /**
         * Create CookiesLinkEventFactory class with path
         * @param path to this node
         */
        var CookiesLinkEventFactory = /** @class */ (function () {
            function CookiesLinkEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "cookies_link"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CookiesLinkEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CookiesLinkEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            CookiesLinkEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CookiesLinkEventFactory;
        }());
        FooterEventFactory.CookiesLinkEventFactory = CookiesLinkEventFactory;
    })(FooterEventFactory = WebCommonEventFactory.FooterEventFactory || (WebCommonEventFactory.FooterEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var SearchBarEventFactory;
        (function (SearchBarEventFactory) {
            /**
             * Create TextAreaEventFactory class with path
             * @param path to this node
             */
            var TextAreaEventFactory = /** @class */ (function () {
                function TextAreaEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "text_area"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TextAreaEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TextAreaEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for key_stroke search action.
                 */
                TextAreaEventFactory.prototype.keyStrokeSearch = function () {
                    var data = {
                        actionName: "search",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "key_stroke",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TextAreaEventFactory;
            }());
            SearchBarEventFactory.TextAreaEventFactory = TextAreaEventFactory;
        })(SearchBarEventFactory = SideNavPanelEventFactory.SearchBarEventFactory || (SideNavPanelEventFactory.SearchBarEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var SearchBarEventFactory;
        (function (SearchBarEventFactory) {
            /**
             * Create CloseButtonEventFactory class with path
             * @param path to this node
             */
            var CloseButtonEventFactory = /** @class */ (function () {
                function CloseButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "close_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                CloseButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                CloseButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_element_toggle action.
                 */
                CloseButtonEventFactory.prototype.hitUiElementToggle = function () {
                    var data = {
                        actionName: "ui_element_toggle",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return CloseButtonEventFactory;
            }());
            SearchBarEventFactory.CloseButtonEventFactory = CloseButtonEventFactory;
        })(SearchBarEventFactory = SideNavPanelEventFactory.SearchBarEventFactory || (SideNavPanelEventFactory.SearchBarEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var ArtistSelectorEventFactory;
        (function (ArtistSelectorEventFactory) {
            /**
             * Create ArtistRowEventFactory class with path
             * @param path to this node
             */
            var ArtistRowEventFactory = /** @class */ (function () {
                function ArtistRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "artist_row" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ArtistRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ArtistRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                ArtistRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ArtistRowEventFactory;
            }());
            ArtistSelectorEventFactory.ArtistRowEventFactory = ArtistRowEventFactory;
        })(ArtistSelectorEventFactory = SideNavPanelEventFactory.ArtistSelectorEventFactory || (SideNavPanelEventFactory.ArtistSelectorEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var OtherLinksEventFactory;
        (function (OtherLinksEventFactory) {
            /**
             * Create YourRosterRowEventFactory class with path
             * @param path to this node
             */
            var YourRosterRowEventFactory = /** @class */ (function () {
                function YourRosterRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "your_roster_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                YourRosterRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                YourRosterRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                YourRosterRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                /**
                 * Create impression for YourRosterRowEventFactory.
                 */
                YourRosterRowEventFactory.prototype.impression = function () {
                    var data = {
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        specificationVersion: SPECIFICATION_VERSION,
                        specificationMode: SPECIFICATION_MODE,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return YourRosterRowEventFactory;
            }());
            OtherLinksEventFactory.YourRosterRowEventFactory = YourRosterRowEventFactory;
        })(OtherLinksEventFactory = SideNavPanelEventFactory.OtherLinksEventFactory || (SideNavPanelEventFactory.OtherLinksEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var OtherLinksEventFactory;
        (function (OtherLinksEventFactory) {
            /**
             * Create YourTeamsRowEventFactory class with path
             * @param path to this node
             */
            var YourTeamsRowEventFactory = /** @class */ (function () {
                function YourTeamsRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "your_teams_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                YourTeamsRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                YourTeamsRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                YourTeamsRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                /**
                 * Create impression for YourTeamsRowEventFactory.
                 */
                YourTeamsRowEventFactory.prototype.impression = function () {
                    var data = {
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        specificationVersion: SPECIFICATION_VERSION,
                        specificationMode: SPECIFICATION_MODE,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return YourTeamsRowEventFactory;
            }());
            OtherLinksEventFactory.YourTeamsRowEventFactory = YourTeamsRowEventFactory;
        })(OtherLinksEventFactory = SideNavPanelEventFactory.OtherLinksEventFactory || (SideNavPanelEventFactory.OtherLinksEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var UserRowEventFactory;
        (function (UserRowEventFactory) {
            /**
             * Create EmployeeFeedbackRowEventFactory class with path
             * @param path to this node
             */
            var EmployeeFeedbackRowEventFactory = /** @class */ (function () {
                function EmployeeFeedbackRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "employee_feedback_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                EmployeeFeedbackRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                EmployeeFeedbackRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                EmployeeFeedbackRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return EmployeeFeedbackRowEventFactory;
            }());
            UserRowEventFactory.EmployeeFeedbackRowEventFactory = EmployeeFeedbackRowEventFactory;
        })(UserRowEventFactory = SideNavPanelEventFactory.UserRowEventFactory || (SideNavPanelEventFactory.UserRowEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var UserRowEventFactory;
        (function (UserRowEventFactory) {
            /**
             * Create UserSettingsRowEventFactory class with path
             * @param path to this node
             */
            var UserSettingsRowEventFactory = /** @class */ (function () {
                function UserSettingsRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "user_settings_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                UserSettingsRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                UserSettingsRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                UserSettingsRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return UserSettingsRowEventFactory;
            }());
            UserRowEventFactory.UserSettingsRowEventFactory = UserSettingsRowEventFactory;
        })(UserRowEventFactory = SideNavPanelEventFactory.UserRowEventFactory || (SideNavPanelEventFactory.UserRowEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var UserRowEventFactory;
        (function (UserRowEventFactory) {
            /**
             * Create AddTeamRowEventFactory class with path
             * @param path to this node
             */
            var AddTeamRowEventFactory = /** @class */ (function () {
                function AddTeamRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "add_team_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AddTeamRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AddTeamRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                AddTeamRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return AddTeamRowEventFactory;
            }());
            UserRowEventFactory.AddTeamRowEventFactory = AddTeamRowEventFactory;
        })(UserRowEventFactory = SideNavPanelEventFactory.UserRowEventFactory || (SideNavPanelEventFactory.UserRowEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var UserRowEventFactory;
        (function (UserRowEventFactory) {
            /**
             * Create HelpRowEventFactory class with path
             * @param path to this node
             */
            var HelpRowEventFactory = /** @class */ (function () {
                function HelpRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "help_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                HelpRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                HelpRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                HelpRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return HelpRowEventFactory;
            }());
            UserRowEventFactory.HelpRowEventFactory = HelpRowEventFactory;
        })(UserRowEventFactory = SideNavPanelEventFactory.UserRowEventFactory || (SideNavPanelEventFactory.UserRowEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
(function (WebCommonEventFactory) {
    var SideNavPanelEventFactory;
    (function (SideNavPanelEventFactory) {
        var UserRowEventFactory;
        (function (UserRowEventFactory) {
            /**
             * Create LogOutRowEventFactory class with path
             * @param path to this node
             */
            var LogOutRowEventFactory = /** @class */ (function () {
                function LogOutRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "log_out_row"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                LogOutRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                LogOutRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit logout action.
                 */
                LogOutRowEventFactory.prototype.hitLogout = function () {
                    var data = {
                        actionName: "logout",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return LogOutRowEventFactory;
            }());
            UserRowEventFactory.LogOutRowEventFactory = LogOutRowEventFactory;
        })(UserRowEventFactory = SideNavPanelEventFactory.UserRowEventFactory || (SideNavPanelEventFactory.UserRowEventFactory = {}));
    })(SideNavPanelEventFactory = WebCommonEventFactory.SideNavPanelEventFactory || (WebCommonEventFactory.SideNavPanelEventFactory = {}));
})(WebCommonEventFactory = exports.WebCommonEventFactory || (exports.WebCommonEventFactory = {}));
exports.WebCommonEventFactory = WebCommonEventFactory;
