// @spotify-internal/adstudio-tape/lib/components/TableBody/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = TableBody;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _flex = require("../../styles/common/flex");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var propTypes = {
  /* Can accept TableRow or any generic HTML */
  children: _propTypes["default"].node,
  fitToWidth: _propTypes["default"].bool,
  fixedHeight: _propTypes["default"].number
};

var StyledTableBody = _styledComponents["default"].section.withConfig({
  displayName: "TableBody__StyledTableBody",
  componentId: "vkp683-0"
})(["background-color:transparent;display:flex;margin-bottom:", ";max-width:100%;width:100%;", ""], _encoreFoundation.spacer24, _flex.column);

var StyledTableContent = _styledComponents["default"].div.withConfig({
  displayName: "TableBody__StyledTableContent",
  componentId: "vkp683-1"
})(["", " ", ""], function (props) {
  return props.fitToWidth ? 'min-width: 100%;' : '';
}, function (props) {
  return props.fixedHeight ? "\n    height: ".concat(props.fixedHeight, "px;\n    position: relative;\n  ") : '';
});

function TableBody(_ref) {
  var children = _ref.children,
      _ref$fitToWidth = _ref.fitToWidth,
      fitToWidth = _ref$fitToWidth === void 0 ? true : _ref$fitToWidth,
      fixedHeight = _ref.fixedHeight;

  var content = _react["default"].Children.map(children, function (child) {
    return child;
  });

  return /*#__PURE__*/_react["default"].createElement(StyledTableBody, {
    "data-test": "table-body"
  }, /*#__PURE__*/_react["default"].createElement(StyledTableContent, {
    fitToWidth: fitToWidth,
    fixedHeight: fixedHeight
  }, content));
}

TableBody.propTypes = propTypes;
