// @spotify-internal/betamax-sdk-desktop/esm/src/tracking/types.js
import PlaybackEvents from './VideoPlaybackEvents';
export var PlaybackSessionKeys;
(function (PlaybackSessionKeys) {
    PlaybackSessionKeys["SESSION_ID"] = "session_id";
    PlaybackSessionKeys["MEDIA_URL"] = "media_url";
    PlaybackSessionKeys["MS_START_POSITION"] = "ms_start_position";
    PlaybackSessionKeys["MS_DURATION"] = "ms_duration";
    PlaybackSessionKeys["MS_START_TIME"] = "ms_start_time";
    PlaybackSessionKeys["MS_MANIFEST_LOAD_TIME"] = "ms_manifest_load_time";
    PlaybackSessionKeys["MS_ENCRYPTION_LOAD_TIME"] = "ms_encryption_load_time";
    PlaybackSessionKeys["MS_INITIAL_BUFFERING"] = "ms_initial_buffering";
    PlaybackSessionKeys["MS_PLAYED"] = "ms_played";
    PlaybackSessionKeys["MS_PLAYED_SUBTITLES"] = "ms_played_subtitles";
    PlaybackSessionKeys["MS_PLAYED_FULLSCREEN"] = "ms_played_fullscreen";
    PlaybackSessionKeys["ENCRYPTION_TYPE"] = "encryption_type";
    PlaybackSessionKeys["EXIT_REASON"] = "exit_reason";
    PlaybackSessionKeys["LAST_USED_SUBTITLE"] = "last_used_subtitle";
    PlaybackSessionKeys["N_STALLS"] = "n_stalls";
    PlaybackSessionKeys["MS_BUFFERING_STALL"] = "ms_buffering_stall";
    PlaybackSessionKeys["MS_BUFFERING_STALL_LONGEST"] = "ms_buffering_stall_longest";
    PlaybackSessionKeys["N_SEEKBACK"] = "n_seekback";
    PlaybackSessionKeys["N_SEEKFORWARD"] = "n_seekforward";
    PlaybackSessionKeys["MS_BUFFERING_SEEK"] = "ms_buffering_seek";
    PlaybackSessionKeys["MS_BUFFERING_SEEK_LONGEST"] = "ms_buffering_seek_longest";
    PlaybackSessionKeys["KBPS_AVERAGE_BANDWIDTH"] = "kbps_average_bandwidth";
    PlaybackSessionKeys["KBYTES_DOWNLOADED"] = "kbytes_downloaded";
    PlaybackSessionKeys["N_DROPPED_FRAMES"] = "n_dropped_frames";
    PlaybackSessionKeys["ASPECT_RATIO"] = "aspect_ratio";
    PlaybackSessionKeys["FEATURE_IDENTIFIER"] = "feature_identifier";
    PlaybackSessionKeys["START_BITRATE"] = "start_bitrate";
    PlaybackSessionKeys["MS_PLAYED_BACKGROUND"] = "ms_played_background";
    PlaybackSessionKeys["CONNECTION_TYPE_START"] = "connection_type_start";
    PlaybackSessionKeys["CONNECTION_TYPE_END"] = "connection_type_end";
    PlaybackSessionKeys["N_VIEW_TRANSITIONS"] = "n_view_transitions";
    PlaybackSessionKeys["AVERAGE_DEVICE_ORIENTATION"] = "average_device_orientation";
    PlaybackSessionKeys["KBYTES_PLAYED"] = "kbytes_played";
})(PlaybackSessionKeys || (PlaybackSessionKeys = {}));
export var EndVideoKeys;
(function (EndVideoKeys) {
    EndVideoKeys["PLAY_CONTEXT"] = "play_context";
    EndVideoKeys["REFERRER_VENDOR"] = "referrer_vendor";
    EndVideoKeys["PLAYBACK_ID"] = "playback_id";
    EndVideoKeys["MS_TOTAL_EST"] = "ms_total_est";
    EndVideoKeys["AUDIOCODEC"] = "audiocodec";
    EndVideoKeys["N_SEEKFWD"] = "n_seekfwd";
    EndVideoKeys["PREFETCHED_BYTES_LOADED"] = "prefetched_bytes_loaded";
    EndVideoKeys["TOTAL_BYTES"] = "total_bytes";
    EndVideoKeys["FEATURE_IDENTIFIER"] = "feature_identifier";
    EndVideoKeys["VIEW_URI"] = "view_uri";
    EndVideoKeys["START_BITRATE"] = "start_bitrate";
    EndVideoKeys["KEY_SYSTEM"] = "key_system";
    EndVideoKeys["CONNECTION_TYPE_END"] = "connection_type_end";
    EndVideoKeys["VIDEO_SESSION_ID"] = "video_session_id";
    EndVideoKeys["LOCAL_TIME_MS"] = "local_time_ms";
    EndVideoKeys["TIME_WEIGHTED_BITRATE"] = "time_weighted_bitrate";
    EndVideoKeys["MS_LATENCY"] = "ms_latency";
    EndVideoKeys["CONNECTION_TYPE_START"] = "connection_type_start";
    EndVideoKeys["MS_PLAYED_BACKGROUND"] = "ms_played_background";
    EndVideoKeys["FILE_ID"] = "file_id";
    EndVideoKeys["MS_SEEKFWD"] = "ms_seekfwd";
    EndVideoKeys["UI_LATENCY"] = "ui_latency";
    EndVideoKeys["REFERRER_VERSION"] = "referrer_version";
    EndVideoKeys["REASON_START"] = "reason_start";
    EndVideoKeys["N_STALLS"] = "n_stalls";
    EndVideoKeys["VIDEOCODEC"] = "videocodec";
    EndVideoKeys["FEATURE_VERSION"] = "feature_version";
    EndVideoKeys["N_SEEKBACK"] = "n_seekback";
    EndVideoKeys["MAX_MS_SEEK_REBUFFERING"] = "max_ms_seek_rebuffering";
    EndVideoKeys["INCOGNITO_MODE"] = "incognito_mode";
    EndVideoKeys["MAX_MS_STALLED"] = "max_ms_stalled";
    EndVideoKeys["MS_MANIFEST_LATENCY"] = "ms_manifest_latency";
    EndVideoKeys["PROVIDER"] = "provider";
    EndVideoKeys["SEQUENCE_ID"] = "sequence_id";
    EndVideoKeys["PLAY_TRACK"] = "play_track";
    EndVideoKeys["MS_STALLED"] = "ms_stalled";
    EndVideoKeys["LIVE"] = "live";
    EndVideoKeys["START_OFFSET_MS"] = "start_offset_ms";
    EndVideoKeys["MS_SEEK_REBUFFERING"] = "ms_seek_rebuffering";
    EndVideoKeys["GAIA_DEV_ID"] = "gaia_dev_id";
    EndVideoKeys["PREFETCHED_BYTES"] = "prefetched_bytes";
    EndVideoKeys["MS_KEY_LATENCY"] = "ms_key_latency";
    EndVideoKeys["MS_INITIAL_BUFFERING"] = "ms_initial_buffering";
    EndVideoKeys["SEQUENCE_NUMBER"] = "sequence_number";
    EndVideoKeys["REFERRER"] = "referrer";
    EndVideoKeys["MS_SEEKBACK"] = "ms_seekback";
    EndVideoKeys["UNION_PLAYED"] = "union_played";
    EndVideoKeys["STREAMING_RULE"] = "streaming_rule";
    EndVideoKeys["MS_PLAYED"] = "ms_played";
    EndVideoKeys["REASON_END"] = "reason_end";
    EndVideoKeys["MAX_CONTINOUS"] = "max_continous";
    EndVideoKeys["SHUFFLE"] = "shuffle";
    EndVideoKeys["FULL_SCREEN"] = "full_screen";
})(EndVideoKeys || (EndVideoKeys = {}));
