// @spotify-internal/betamax-sdk-desktop/esm/src/tracking/VideoPlaybackEvents.js
export var BufferType;
(function (BufferType) {
    BufferType["INITIAL"] = "initial";
    BufferType["SEEK"] = "seek";
    BufferType["STALL"] = "stall";
})(BufferType || (BufferType = {}));
export var EndReason;
(function (EndReason) {
    EndReason["PLAYED_TO_END"] = "playedToEnd";
    EndReason["FATAL_ERROR"] = "fatalError";
    EndReason["REQUESTED"] = "requested";
})(EndReason || (EndReason = {}));
export var InternalEvents = {
    CHANGE: 'change',
};
var PlaybackEvents;
(function (PlaybackEvents) {
    PlaybackEvents["CREATED"] = "created";
    PlaybackEvents["MANIFEST_LOADING"] = "manifestLoading";
    PlaybackEvents["MANIFEST_LOADED"] = "manifestLoaded";
    PlaybackEvents["ENCRYPTION_KEY_LOADING"] = "encryptionKeyLoading";
    PlaybackEvents["ENCRYPTION_KEY_LOADED"] = "encryptionKeyLoaded";
    PlaybackEvents["BUFFERING"] = "buffering";
    PlaybackEvents["DROPPED_FRAMES"] = "droppedFrames";
    PlaybackEvents["READY"] = "ready";
    PlaybackEvents["STARTED"] = "started";
    PlaybackEvents["PAUSED"] = "paused";
    PlaybackEvents["RESUMED"] = "resumed";
    PlaybackEvents["SEEKING"] = "seeking";
    PlaybackEvents["ENDED"] = "ended";
    PlaybackEvents["RECOVERABLE_ERROR"] = "recoverableError";
    PlaybackEvents["FATAL_ERROR"] = "fatalError";
    PlaybackEvents["BYTES_TRANSFERRED"] = "bytesTransferred";
    PlaybackEvents["SUBTITLE_CHANGED"] = "subtitleChanged";
    PlaybackEvents["VIDEO_FORMAT_CHANGED"] = "videoFormatChanged";
    PlaybackEvents["AUDIO_FORMAT_CHANGED"] = "audioFormatChanged";
    PlaybackEvents["FULLSCREEN_CHANGED"] = "fullscreenChanged";
    PlaybackEvents["VOLUME_CHANGED"] = "volumeChanged";
    PlaybackEvents["PROGRESS"] = "progress";
    PlaybackEvents["POSITION_UPDATE"] = "positionUpdate";
    PlaybackEvents["DURATION_CHANGED"] = "durationChanged";
    PlaybackEvents["NEW_SESSION_LOADED"] = "newSessionLoaded";
})(PlaybackEvents || (PlaybackEvents = {}));
export default PlaybackEvents;
