// spotify-playback/types/errors.js
'use strict';

var debugEnum = require('spotify-debug-tools/debug_enum');

var Errors = {
  // Capping
  CAPPING_USER_IS_CAPPED: 'CAPPING_USER_IS_CAPPED',

  // EME Testing
  EME_API_NOT_SUPPORTED: 'EME_API_NOT_SUPPORTED',
  EME_MEDIA_KEYS_NOT_SUPPORTED: 'EME_MEDIA_KEYS_NOT_SUPPORTED',
  EME_MEDIA_KEY_SESSION_NOT_SUPPORTED: 'EME_MEDIA_KEY_SESSION_NOT_SUPPORTED',
  EME_NO_SUPPORTED_KEYSYSTEM: 'EME_NO_SUPPORTED_KEYSYSTEM',
  EME_PLAYER_MEDIA_KEYS_SETTING_FAILED: 'EME_PLAYER_MEDIA_KEYS_SETTING_FAILED',

  // EME Lifecycle
  EME_ERROR_UNKNOWN: 'EME_ERROR_UNKNOWN',
  EME_LICENSE_REQUEST_EMPTY_RESPONSE: 'EME_LICENSE_REQUEST_EMPTY_RESPONSE',
  EME_LICENSE_REQUEST_FAILED_WITH_STATUS:
    'EME_LICENSE_REQUEST_FAILED_WITH_STATUS',
  EME_LICENSE_REQUEST_WIDEVINE_ERROR: 'EME_LICENSE_REQUEST_WIDEVINE_ERROR',
  EME_LICENSE_UPDATE_FAILED: 'EME_LICENSE_UPDATE_FAILED',

  // PlayReady Specific Errors
  EME_HEADER_KEY_VALUE_MISMATCH: 'EME_HEADER_KEY_VALUE_MISMATCH',
  EME_HEADER_MISSING_CHALLENGE: 'EME_HEADER_MISSING_CHALLENGE',

  // EME Polyfill errors
  EME_CANNOT_SET_CERTIFICATE_FOR_PLATFORM:
    'EME_CANNOT_SET_CERTIFICATE_FOR_PLATFORM',
  EME_MEDIA_KEY_SESSION_V0_1B_ERROR: 'EME_MEDIA_KEY_SESSION_V0_1B_ERROR',
  EME_NO_SUPPORTED_CONFIGURATION: 'EME_NO_SUPPORTED_CONFIGURATION',

  EME_NOT_SUPPORTED_ERROR: 'EME_NOT_SUPPORTED_ERROR',
  EME_INVALID_STATE_ERROR: 'EME_INVALID_STATE_ERROR',
  EME_UNKNOWN_ERROR: 'EME_UNKNOWN_ERROR',

  DISALLOW_PROTECTED_TRACK_ERROR: 'DISALLOW_PROTECTED_TRACK_ERROR',

  // File Errors
  FILE_FORMAT_NOT_SUPPORTED: 'FILE_FORMAT_NOT_SUPPORTED',
  FILE_MALFORMED_SEEKTABLE: 'FILE_MALFORMED_SEEKTABLE',
  FILE_MALFORMED_PSSH: 'FILE_MALFORMED_PSSH',
  FILE_NOT_RESOLVED: 'FILE_NOT_RESOLVED',

  // Fragment Errors
  FRAGMENT_ONLINE_REQUEST_FAILED_WITH_ZERO:
    'FRAGMENT_ONLINE_REQUEST_FAILED_WITH_ZERO',
  FRAGMENT_OFFLINE_REQUEST_FAILED_WITH_ZERO:
    'FRAGMENT_OFFLINE_REQUEST_FAILED_WITH_ZERO',
  FRAGMENT_REQUEST_FAILED_WITH_ZERO: 'FRAGMENT_REQUEST_FAILED_WITH_ZERO',
  FRAGMENT_REQUEST_FAILED_WITH_STATUS: 'FRAGMENT_REQUEST_FAILED_WITH_STATUS',
  FRAGMENT_REQUEST_EMPTY_RESPONSE: 'FRAGMENT_REQUEST_EMPTY_RESPONSE',
  FRAGMENT_REQUEST_UNEXPECTED_LENGTH: 'FRAGMENT_REQUEST_UNEXPECTED_LENGTH',

  // Generic Playback Errors
  PLAYER_ATTEMPTED_VOLUME_OUT_OF_RANGE: 'PLAYER_ATTEMPTED_VOLUME_OUT_OF_RANGE',
  PLAYER_BUFFER_QUOTA_EXCEEDED: 'PLAYER_BUFFER_QUOTA_EXCEEDED',
  PLAYER_CANNOT_FIND_PLAYABLE_URI: 'PLAYER_CANNOT_FIND_PLAYABLE_URI',
  PLAYER_MEDIA_ERROR: 'PLAYER_MEDIA_ERROR',
  PLAYER_PLAYBACK_ERROR: 'PLAYER_PLAYBACK_ERROR',

  // Media Errors
  MEDIA_ABORTED: 'MEDIA_ABORTED',
  MEDIA_DECODING_ERROR: 'MEDIA_DECODING_ERROR',
  MEDIA_NETWORK_ERROR: 'MEDIA_NETWORK_ERROR',
  MEDIA_NOT_SUPPORTED: 'MEDIA_NOT_SUPPORTED',

  // License Errors
  LICENSE_RESOLVER_CANT_RESOLVE_URL: 'LICENSE_RESOLVER_CANT_RESOLVE_URL',
  LICENSE_RESOLVER_DEPRECATED_VERSION: 'LICENSE_RESOLVER_DEPRECATED_VERSION',

  // List Player Errors
  LIST_PLAYER_NO_TRACK_PLAYER: 'LIST_PLAYER_NO_TRACK_PLAYER',
  LIST_PLAYER_NO_LIST: 'LIST_PLAYER_NO_LIST',
  LIST_PLAYER_INVALID_ARGUMENT: 'LIST_PLAYER_INVALID_ARGUMENT',
  LIST_PLAYER_FORBIDDEN: 'LIST_PLAYER_FORBIDDEN',

  // Storage Errors
  STORAGE_ERROR: 'STORAGE_ERROR',
  STORAGE_FAILED_WITH_STATUS: 'STORAGE_FAILED_WITH_STATUS',
  STORAGE_RETURNED_NO_TRACKS: 'STORAGE_RETURNED_NO_TRACKS',
  STORAGE_TRACK_MANIFEST_FAILED: 'STORAGE_TRACK_MANIFEST_FAILED',
  STORAGE_TRACK_MANIFEST_EMPTY: 'STORAGE_TRACK_MANIFEST_EMPTY',
  STORAGE_VIDEO_MANIFEST_FAILED: 'STORAGE_VIDEO_MANIFEST_FAILED',

  // Tracking errors
  TRACK_DATA_ALREADY_FINALIZED: 'TRACK_DATA_ALREADY_FINALIZED',

  TSV_SENDING_FAILED: 'TSV_SENDING_FAILED',
  PLAYBACK_STATS_SENDING_FAILED: 'PLAYBACK_STATS_SENDING_FAILED',
  PLAYBACK_START_SENDING_FAILED: 'PLAYBACK_START_SENDING_FAILED',

  UNKNOWN: 'UNKNOWN'
};

module.exports = debugEnum(Errors);
