// spotify-playback/types/reason.js
'use strict';

/**
 * Defines the different "reasons" for playback.
 *
 * @enum {string}
 */
var Reason = {
  /** The program started and the song autostarted */
  APPLOAD: 'appload',
  /** The back button was pressed. */
  BACK_BUTTON: 'backbtn',
  /** A row in the song list was clicked/opened. */
  CLICK_ROW: 'clickrow',
  /** The song was started from a click on the sidebar. */
  CLICK_SIDE: 'clickside',
  /** The song was ended (will never cause a song to start). */
  END_PLAY: 'endplay',
  /** The forward button was pressed. */
  FORWARD_BUTTON: 'fwdbtn',
  /** The user logged out (will never cause a song to start). */
  LOGOUT: 'logout',
  /** The play button was pressed. */
  PLAY_BUTTON: 'playbtn',
  /** Play was chosen from a popup menu. */
  POPUP: 'popup',
  /** A remote control request. */
  REMOTE: 'remote',
  /** A track ended because it had been played in full. */
  TRACK_DONE: 'trackdone',
  /**  A track ended because of errors in the track. */
  TRACK_ERROR: 'trackerror',
  /** Client doesn't know. */
  UNKNOWN: 'unknown',
  /** A URL was opened. */
  URI_OPEN: 'uriopen'
};


module.exports = Reason;
