// @spotify-internal/adstudioasset/proto/com/spotify/ads/protobuf/base_pb.js
// source: com/spotify/ads/protobuf/base.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.com.spotify.ads.protobuf.CostType', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Currency', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.CurrencyMicroAmount', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int32Range', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int32Range.LowerboundoptionalCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int32Range.UpperboundoptionalCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int64Range', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int64Range.LowerboundoptionalCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Int64Range.UpperboundoptionalCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.KeyValuePair', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.MediaType', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.ModelData', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.ModelType', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.OptimizationData', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.OptimizationData.OptimizationtargetCase', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.ProbabilityCutoff', null, global);
goog.exportSymbol('proto.com.spotify.ads.protobuf.Uuid', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.Int32Range = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_);
};
goog.inherits(proto.com.spotify.ads.protobuf.Int32Range, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.Int32Range.displayName = 'proto.com.spotify.ads.protobuf.Int32Range';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.Int64Range = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_);
};
goog.inherits(proto.com.spotify.ads.protobuf.Int64Range, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.Int64Range.displayName = 'proto.com.spotify.ads.protobuf.Int64Range';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.Uuid = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.Uuid, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.Uuid.displayName = 'proto.com.spotify.ads.protobuf.Uuid';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.KeyValuePair = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.KeyValuePair, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.KeyValuePair.displayName = 'proto.com.spotify.ads.protobuf.KeyValuePair';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.MediaType = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.MediaType, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.MediaType.displayName = 'proto.com.spotify.ads.protobuf.MediaType';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.Currency = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.Currency, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.Currency.displayName = 'proto.com.spotify.ads.protobuf.Currency';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.CurrencyMicroAmount, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.CurrencyMicroAmount.displayName = 'proto.com.spotify.ads.protobuf.CurrencyMicroAmount';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.ModelData = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.ModelData, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.ModelData.displayName = 'proto.com.spotify.ads.protobuf.ModelData';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.OptimizationData = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_);
};
goog.inherits(proto.com.spotify.ads.protobuf.OptimizationData, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.OptimizationData.displayName = 'proto.com.spotify.ads.protobuf.OptimizationData';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.ads.protobuf.ProbabilityCutoff, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.ads.protobuf.ProbabilityCutoff.displayName = 'proto.com.spotify.ads.protobuf.ProbabilityCutoff';
}

/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_ = [[1,2],[3,4]];

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.LowerboundoptionalCase = {
  LOWERBOUNDOPTIONAL_NOT_SET: 0,
  LOWERBOUNDINCLUSIVE: 1,
  LOWERBOUNDEXCLUSIVE: 2
};

/**
 * @return {proto.com.spotify.ads.protobuf.Int32Range.LowerboundoptionalCase}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getLowerboundoptionalCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.Int32Range.LowerboundoptionalCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[0]));
};

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.UpperboundoptionalCase = {
  UPPERBOUNDOPTIONAL_NOT_SET: 0,
  UPPERBOUNDINCLUSIVE: 3,
  UPPERBOUNDEXCLUSIVE: 4
};

/**
 * @return {proto.com.spotify.ads.protobuf.Int32Range.UpperboundoptionalCase}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getUpperboundoptionalCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.Int32Range.UpperboundoptionalCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[1]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.Int32Range.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.Int32Range} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Int32Range.toObject = function(includeInstance, msg) {
  var f, obj = {
    lowerboundinclusive: jspb.Message.getFieldWithDefault(msg, 1, 0),
    lowerboundexclusive: jspb.Message.getFieldWithDefault(msg, 2, 0),
    upperboundinclusive: jspb.Message.getFieldWithDefault(msg, 3, 0),
    upperboundexclusive: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range}
 */
proto.com.spotify.ads.protobuf.Int32Range.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.Int32Range;
  return proto.com.spotify.ads.protobuf.Int32Range.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.Int32Range} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range}
 */
proto.com.spotify.ads.protobuf.Int32Range.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readSint32());
      msg.setLowerboundinclusive(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readSint32());
      msg.setLowerboundexclusive(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readSint32());
      msg.setUpperboundinclusive(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readSint32());
      msg.setUpperboundexclusive(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.Int32Range.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.Int32Range} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Int32Range.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {number} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeSint32(
      1,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeSint32(
      2,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeSint32(
      3,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeSint32(
      4,
      f
    );
  }
};


/**
 * optional sint32 lowerBoundInclusive = 1;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getLowerboundinclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.setLowerboundinclusive = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.clearLowerboundinclusive = function() {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.hasLowerboundinclusive = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional sint32 lowerBoundExclusive = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getLowerboundexclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.setLowerboundexclusive = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.clearLowerboundexclusive = function() {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.hasLowerboundexclusive = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional sint32 upperBoundInclusive = 3;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getUpperboundinclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.setUpperboundinclusive = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[1], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.clearUpperboundinclusive = function() {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[1], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.hasUpperboundinclusive = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional sint32 upperBoundExclusive = 4;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.getUpperboundexclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.setUpperboundexclusive = function(value) {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[1], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int32Range} returns this
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.clearUpperboundexclusive = function() {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.Int32Range.oneofGroups_[1], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int32Range.prototype.hasUpperboundexclusive = function() {
  return jspb.Message.getField(this, 4) != null;
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_ = [[1,2],[3,4]];

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.LowerboundoptionalCase = {
  LOWERBOUNDOPTIONAL_NOT_SET: 0,
  LOWERBOUNDINCLUSIVE: 1,
  LOWERBOUNDEXCLUSIVE: 2
};

/**
 * @return {proto.com.spotify.ads.protobuf.Int64Range.LowerboundoptionalCase}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getLowerboundoptionalCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.Int64Range.LowerboundoptionalCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[0]));
};

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.UpperboundoptionalCase = {
  UPPERBOUNDOPTIONAL_NOT_SET: 0,
  UPPERBOUNDINCLUSIVE: 3,
  UPPERBOUNDEXCLUSIVE: 4
};

/**
 * @return {proto.com.spotify.ads.protobuf.Int64Range.UpperboundoptionalCase}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getUpperboundoptionalCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.Int64Range.UpperboundoptionalCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[1]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.Int64Range.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.Int64Range} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Int64Range.toObject = function(includeInstance, msg) {
  var f, obj = {
    lowerboundinclusive: jspb.Message.getFieldWithDefault(msg, 1, 0),
    lowerboundexclusive: jspb.Message.getFieldWithDefault(msg, 2, 0),
    upperboundinclusive: jspb.Message.getFieldWithDefault(msg, 3, 0),
    upperboundexclusive: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range}
 */
proto.com.spotify.ads.protobuf.Int64Range.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.Int64Range;
  return proto.com.spotify.ads.protobuf.Int64Range.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.Int64Range} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range}
 */
proto.com.spotify.ads.protobuf.Int64Range.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readSint64());
      msg.setLowerboundinclusive(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readSint64());
      msg.setLowerboundexclusive(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readSint64());
      msg.setUpperboundinclusive(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readSint64());
      msg.setUpperboundexclusive(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.Int64Range.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.Int64Range} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Int64Range.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {number} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeSint64(
      1,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeSint64(
      2,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeSint64(
      3,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeSint64(
      4,
      f
    );
  }
};


/**
 * optional sint64 lowerBoundInclusive = 1;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getLowerboundinclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.setLowerboundinclusive = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.clearLowerboundinclusive = function() {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.hasLowerboundinclusive = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional sint64 lowerBoundExclusive = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getLowerboundexclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.setLowerboundexclusive = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.clearLowerboundexclusive = function() {
  return jspb.Message.setOneofField(this, 2, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.hasLowerboundexclusive = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional sint64 upperBoundInclusive = 3;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getUpperboundinclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.setUpperboundinclusive = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[1], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.clearUpperboundinclusive = function() {
  return jspb.Message.setOneofField(this, 3, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[1], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.hasUpperboundinclusive = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional sint64 upperBoundExclusive = 4;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.getUpperboundexclusive = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.setUpperboundexclusive = function(value) {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[1], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.Int64Range} returns this
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.clearUpperboundexclusive = function() {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.ads.protobuf.Int64Range.oneofGroups_[1], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.Int64Range.prototype.hasUpperboundexclusive = function() {
  return jspb.Message.getField(this, 4) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.Uuid.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.Uuid} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Uuid.toObject = function(includeInstance, msg) {
  var f, obj = {
    mostsigbits: jspb.Message.getFieldWithDefault(msg, 1, 0),
    leastsigbits: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.Uuid}
 */
proto.com.spotify.ads.protobuf.Uuid.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.Uuid;
  return proto.com.spotify.ads.protobuf.Uuid.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.Uuid} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.Uuid}
 */
proto.com.spotify.ads.protobuf.Uuid.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setMostsigbits(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readFixed64());
      msg.setLeastsigbits(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.Uuid.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.Uuid} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Uuid.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMostsigbits();
  if (f !== 0) {
    writer.writeFixed64(
      1,
      f
    );
  }
  f = message.getLeastsigbits();
  if (f !== 0) {
    writer.writeFixed64(
      2,
      f
    );
  }
};


/**
 * optional fixed64 mostSigBits = 1;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.getMostsigbits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Uuid} returns this
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.setMostsigbits = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional fixed64 leastSigBits = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.getLeastsigbits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Uuid} returns this
 */
proto.com.spotify.ads.protobuf.Uuid.prototype.setLeastsigbits = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.KeyValuePair.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.KeyValuePair} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.KeyValuePair.toObject = function(includeInstance, msg) {
  var f, obj = {
    key: jspb.Message.getFieldWithDefault(msg, 1, ""),
    value: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.KeyValuePair}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.KeyValuePair;
  return proto.com.spotify.ads.protobuf.KeyValuePair.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.KeyValuePair} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.KeyValuePair}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setKey(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setValue(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.KeyValuePair.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.KeyValuePair} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.KeyValuePair.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getKey();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValue();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string key = 1;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.getKey = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.KeyValuePair} returns this
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.setKey = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string value = 2;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.getValue = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.KeyValuePair} returns this
 */
proto.com.spotify.ads.protobuf.KeyValuePair.prototype.setValue = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.MediaType.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.MediaType} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MediaType.toObject = function(includeInstance, msg) {
  var f, obj = {
    type: jspb.Message.getFieldWithDefault(msg, 1, ""),
    subtype: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.MediaType}
 */
proto.com.spotify.ads.protobuf.MediaType.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.MediaType;
  return proto.com.spotify.ads.protobuf.MediaType.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.MediaType} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.MediaType}
 */
proto.com.spotify.ads.protobuf.MediaType.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setType(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setSubtype(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.MediaType.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.MediaType} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.MediaType.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getType();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getSubtype();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string type = 1;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.getType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.MediaType} returns this
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.setType = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string subtype = 2;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.getSubtype = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.MediaType} returns this
 */
proto.com.spotify.ads.protobuf.MediaType.prototype.setSubtype = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.Currency.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.Currency.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.Currency} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Currency.toObject = function(includeInstance, msg) {
  var f, obj = {
    currencycode: jspb.Message.getFieldWithDefault(msg, 1, ""),
    numericcode: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.Currency}
 */
proto.com.spotify.ads.protobuf.Currency.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.Currency;
  return proto.com.spotify.ads.protobuf.Currency.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.Currency} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.Currency}
 */
proto.com.spotify.ads.protobuf.Currency.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCurrencycode(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setNumericcode(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.Currency.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.Currency.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.Currency} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.Currency.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCurrencycode();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getNumericcode();
  if (f !== 0) {
    writer.writeInt32(
      2,
      f
    );
  }
};


/**
 * optional string currencyCode = 1;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.Currency.prototype.getCurrencycode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.Currency} returns this
 */
proto.com.spotify.ads.protobuf.Currency.prototype.setCurrencycode = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional int32 numericCode = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.Currency.prototype.getNumericcode = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.Currency} returns this
 */
proto.com.spotify.ads.protobuf.Currency.prototype.setNumericcode = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.CurrencyMicroAmount.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.toObject = function(includeInstance, msg) {
  var f, obj = {
    currencycode: jspb.Message.getFieldWithDefault(msg, 1, ""),
    microamount: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.CurrencyMicroAmount;
  return proto.com.spotify.ads.protobuf.CurrencyMicroAmount.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCurrencycode(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setMicroamount(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.CurrencyMicroAmount.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCurrencycode();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getMicroamount();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
};


/**
 * optional string currencyCode = 1;
 * @return {string}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.getCurrencycode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount} returns this
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.setCurrencycode = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional int64 microAmount = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.getMicroamount = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.CurrencyMicroAmount} returns this
 */
proto.com.spotify.ads.protobuf.CurrencyMicroAmount.prototype.setMicroamount = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.ModelData.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.ModelData} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ModelData.toObject = function(includeInstance, msg) {
  var f, obj = {
    modeltype: jspb.Message.getFieldWithDefault(msg, 1, 0),
    modelprediction: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.ModelData}
 */
proto.com.spotify.ads.protobuf.ModelData.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.ModelData;
  return proto.com.spotify.ads.protobuf.ModelData.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.ModelData} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.ModelData}
 */
proto.com.spotify.ads.protobuf.ModelData.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.com.spotify.ads.protobuf.ModelType} */ (reader.readEnum());
      msg.setModeltype(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readDouble());
      msg.setModelprediction(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.ModelData.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.ModelData} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ModelData.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getModeltype();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getModelprediction();
  if (f !== 0.0) {
    writer.writeDouble(
      2,
      f
    );
  }
};


/**
 * optional ModelType modelType = 1;
 * @return {!proto.com.spotify.ads.protobuf.ModelType}
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.getModeltype = function() {
  return /** @type {!proto.com.spotify.ads.protobuf.ModelType} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.com.spotify.ads.protobuf.ModelType} value
 * @return {!proto.com.spotify.ads.protobuf.ModelData} returns this
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.setModeltype = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional double modelPrediction = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.getModelprediction = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.ModelData} returns this
 */
proto.com.spotify.ads.protobuf.ModelData.prototype.setModelprediction = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_ = [[1,2]];

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.OptimizationData.OptimizationtargetCase = {
  OPTIMIZATIONTARGET_NOT_SET: 0,
  CONTROL: 1,
  PROBABILITYCUTOFF: 2
};

/**
 * @return {proto.com.spotify.ads.protobuf.OptimizationData.OptimizationtargetCase}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.getOptimizationtargetCase = function() {
  return /** @type {proto.com.spotify.ads.protobuf.OptimizationData.OptimizationtargetCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.OptimizationData.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.OptimizationData} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.OptimizationData.toObject = function(includeInstance, msg) {
  var f, obj = {
    control: jspb.Message.getBooleanFieldWithDefault(msg, 1, false),
    probabilitycutoff: (f = msg.getProbabilitycutoff()) && proto.com.spotify.ads.protobuf.ProbabilityCutoff.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData}
 */
proto.com.spotify.ads.protobuf.OptimizationData.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.OptimizationData;
  return proto.com.spotify.ads.protobuf.OptimizationData.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.OptimizationData} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData}
 */
proto.com.spotify.ads.protobuf.OptimizationData.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setControl(value);
      break;
    case 2:
      var value = new proto.com.spotify.ads.protobuf.ProbabilityCutoff;
      reader.readMessage(value,proto.com.spotify.ads.protobuf.ProbabilityCutoff.deserializeBinaryFromReader);
      msg.setProbabilitycutoff(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.OptimizationData.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.OptimizationData} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.OptimizationData.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {boolean} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeBool(
      1,
      f
    );
  }
  f = message.getProbabilitycutoff();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.com.spotify.ads.protobuf.ProbabilityCutoff.serializeBinaryToWriter
    );
  }
};


/**
 * optional bool control = 1;
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.getControl = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData} returns this
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.setControl = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData} returns this
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.clearControl = function() {
  return jspb.Message.setOneofField(this, 1, proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.hasControl = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional ProbabilityCutoff probabilityCutoff = 2;
 * @return {?proto.com.spotify.ads.protobuf.ProbabilityCutoff}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.getProbabilitycutoff = function() {
  return /** @type{?proto.com.spotify.ads.protobuf.ProbabilityCutoff} */ (
    jspb.Message.getWrapperField(this, proto.com.spotify.ads.protobuf.ProbabilityCutoff, 2));
};


/**
 * @param {?proto.com.spotify.ads.protobuf.ProbabilityCutoff|undefined} value
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData} returns this
*/
proto.com.spotify.ads.protobuf.OptimizationData.prototype.setProbabilitycutoff = function(value) {
  return jspb.Message.setOneofWrapperField(this, 2, proto.com.spotify.ads.protobuf.OptimizationData.oneofGroups_[0], value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.com.spotify.ads.protobuf.OptimizationData} returns this
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.clearProbabilitycutoff = function() {
  return this.setProbabilitycutoff(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.ads.protobuf.OptimizationData.prototype.hasProbabilitycutoff = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.ads.protobuf.ProbabilityCutoff.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.toObject = function(includeInstance, msg) {
  var f, obj = {
    modeltype: jspb.Message.getFieldWithDefault(msg, 1, 0),
    modelpredictioncutoff: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0),
    percentilecutoff: jspb.Message.getFloatingPointFieldWithDefault(msg, 3, 0.0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.ads.protobuf.ProbabilityCutoff}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.ads.protobuf.ProbabilityCutoff;
  return proto.com.spotify.ads.protobuf.ProbabilityCutoff.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.ads.protobuf.ProbabilityCutoff}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.com.spotify.ads.protobuf.ModelType} */ (reader.readEnum());
      msg.setModeltype(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readDouble());
      msg.setModelpredictioncutoff(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readDouble());
      msg.setPercentilecutoff(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.ads.protobuf.ProbabilityCutoff.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getModeltype();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getModelpredictioncutoff();
  if (f !== 0.0) {
    writer.writeDouble(
      2,
      f
    );
  }
  f = message.getPercentilecutoff();
  if (f !== 0.0) {
    writer.writeDouble(
      3,
      f
    );
  }
};


/**
 * optional ModelType modelType = 1;
 * @return {!proto.com.spotify.ads.protobuf.ModelType}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.getModeltype = function() {
  return /** @type {!proto.com.spotify.ads.protobuf.ModelType} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.com.spotify.ads.protobuf.ModelType} value
 * @return {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} returns this
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.setModeltype = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional double modelPredictionCutoff = 2;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.getModelpredictioncutoff = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} returns this
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.setModelpredictioncutoff = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};


/**
 * optional double percentileCutoff = 3;
 * @return {number}
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.getPercentilecutoff = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 3, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.ads.protobuf.ProbabilityCutoff} returns this
 */
proto.com.spotify.ads.protobuf.ProbabilityCutoff.prototype.setPercentilecutoff = function(value) {
  return jspb.Message.setProto3FloatField(this, 3, value);
};


/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.CostType = {
  UNSET: 0,
  CPM: 1,
  CPCL: 2,
  CPC: 3
};

/**
 * @enum {number}
 */
proto.com.spotify.ads.protobuf.ModelType = {
  CLICK: 0,
  COMPLETE: 1,
  RENDER: 2,
  STREAM: 3
};

goog.object.extend(exports, proto.com.spotify.ads.protobuf);
