// @spotify-internal/ubi-sdk-s4a-web-access-user-details-form/WebAccessUserDetailsFormEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAccessUserDetailsFormEventFactory = exports.createWebAccessUserDetailsFormEventFactory = void 0;
var SPECIFICATION_ID = "web-access-user-details-form";
var APP = "s4a";
var SPECIFICATION_VERSION = "0.1.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.2.0";
function createWebAccessUserDetailsFormEventFactory(options) {
    return new WebAccessUserDetailsFormEventFactory(options);
}
exports.createWebAccessUserDetailsFormEventFactory = createWebAccessUserDetailsFormEventFactory;
/**
 *
 */
var WebAccessUserDetailsFormEventFactory = /** @class */ (function () {
    function WebAccessUserDetailsFormEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of RestartFlowLinkEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.restartFlowLinkFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.RestartFlowLinkEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @param id of the LabelOrDistributorFormRadioEventFactory
    * @returns A new instance of LabelOrDistributorFormRadioEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.labelOrDistributorFormRadioFactory = function (data) {
        return new WebAccessUserDetailsFormEventFactory.LabelOrDistributorFormRadioEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation,
            data: data
        });
    };
    /**
    * @returns A new instance of LogOutButtonEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.logOutButtonFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.LogOutButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of FirstNameTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.firstNameTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.FirstNameTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of LastNameTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.lastNameTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.LastNameTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of BusinessEmailTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.businessEmailTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.BusinessEmailTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of RoleDropdownEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.roleDropdownFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.RoleDropdownEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CompanyTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.companyTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.CompanyTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CompanyWebsiteTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.companyWebsiteTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.CompanyWebsiteTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SocialLinkTextboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.socialLinkTextboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.SocialLinkTextboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of TermsAndConditionsCheckboxEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.termsAndConditionsCheckboxFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.TermsAndConditionsCheckboxEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ActionButtonsEventFactory
    */
    WebAccessUserDetailsFormEventFactory.prototype.actionButtonsFactory = function () {
        return new WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebAccessUserDetailsFormEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebAccessUserDetailsFormEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebAccessUserDetailsFormEventFactory;
}());
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create RestartFlowLinkEventFactory class with path
     * @param path to this node
     */
    var RestartFlowLinkEventFactory = /** @class */ (function () {
        function RestartFlowLinkEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "restart_flow_link"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        RestartFlowLinkEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        RestartFlowLinkEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_navigate action.
       * @param destination the destination field
         */
        RestartFlowLinkEventFactory.prototype.hitUiNavigate = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "ui_navigate",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return RestartFlowLinkEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.RestartFlowLinkEventFactory = RestartFlowLinkEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create LabelOrDistributorFormRadioEventFactory class with path
     * @param path to this node
     */
    var LabelOrDistributorFormRadioEventFactory = /** @class */ (function () {
        function LabelOrDistributorFormRadioEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat(__assign(__assign({}, data), { name: "label_or_distributor_form_radio" }));
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LabelOrDistributorFormRadioEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LabelOrDistributorFormRadioEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_select action.
         */
        LabelOrDistributorFormRadioEventFactory.prototype.hitUiSelect = function () {
            var data = {
                actionName: "ui_select",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LabelOrDistributorFormRadioEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.LabelOrDistributorFormRadioEventFactory = LabelOrDistributorFormRadioEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create LogOutButtonEventFactory class with path
     * @param path to this node
     */
    var LogOutButtonEventFactory = /** @class */ (function () {
        function LogOutButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "log_out_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LogOutButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LogOutButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit logout action.
         */
        LogOutButtonEventFactory.prototype.hitLogout = function () {
            var data = {
                actionName: "logout",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LogOutButtonEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.LogOutButtonEventFactory = LogOutButtonEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create FirstNameTextboxEventFactory class with path
     * @param path to this node
     */
    var FirstNameTextboxEventFactory = /** @class */ (function () {
        function FirstNameTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "first_name_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        FirstNameTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        FirstNameTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        FirstNameTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return FirstNameTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.FirstNameTextboxEventFactory = FirstNameTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create LastNameTextboxEventFactory class with path
     * @param path to this node
     */
    var LastNameTextboxEventFactory = /** @class */ (function () {
        function LastNameTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "last_name_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LastNameTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LastNameTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        LastNameTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LastNameTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.LastNameTextboxEventFactory = LastNameTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create BusinessEmailTextboxEventFactory class with path
     * @param path to this node
     */
    var BusinessEmailTextboxEventFactory = /** @class */ (function () {
        function BusinessEmailTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "business_email_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        BusinessEmailTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        BusinessEmailTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        BusinessEmailTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return BusinessEmailTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.BusinessEmailTextboxEventFactory = BusinessEmailTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create RoleDropdownEventFactory class with path
     * @param path to this node
     */
    var RoleDropdownEventFactory = /** @class */ (function () {
        function RoleDropdownEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "role_dropdown"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the RoleOptionsEventFactory
        * @returns A new instance of RoleOptionsEventFactory
        */
        RoleDropdownEventFactory.prototype.roleOptionsFactory = function (data) {
            return new WebAccessUserDetailsFormEventFactory.RoleDropdownEventFactory.RoleOptionsEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        RoleDropdownEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        RoleDropdownEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        RoleDropdownEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return RoleDropdownEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.RoleDropdownEventFactory = RoleDropdownEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create CompanyTextboxEventFactory class with path
     * @param path to this node
     */
    var CompanyTextboxEventFactory = /** @class */ (function () {
        function CompanyTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "company_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        CompanyTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CompanyTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        CompanyTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CompanyTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.CompanyTextboxEventFactory = CompanyTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create CompanyWebsiteTextboxEventFactory class with path
     * @param path to this node
     */
    var CompanyWebsiteTextboxEventFactory = /** @class */ (function () {
        function CompanyWebsiteTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "company_website_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        CompanyWebsiteTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CompanyWebsiteTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        CompanyWebsiteTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CompanyWebsiteTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.CompanyWebsiteTextboxEventFactory = CompanyWebsiteTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create SocialLinkTextboxEventFactory class with path
     * @param path to this node
     */
    var SocialLinkTextboxEventFactory = /** @class */ (function () {
        function SocialLinkTextboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "social_link_textbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        SocialLinkTextboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SocialLinkTextboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit focus_input_field action.
         */
        SocialLinkTextboxEventFactory.prototype.hitFocusInputField = function () {
            var data = {
                actionName: "focus_input_field",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SocialLinkTextboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.SocialLinkTextboxEventFactory = SocialLinkTextboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create TermsAndConditionsCheckboxEventFactory class with path
     * @param path to this node
     */
    var TermsAndConditionsCheckboxEventFactory = /** @class */ (function () {
        function TermsAndConditionsCheckboxEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "terms_and_conditions_checkbox"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        TermsAndConditionsCheckboxEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TermsAndConditionsCheckboxEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit accept_terms action.
         */
        TermsAndConditionsCheckboxEventFactory.prototype.hitAcceptTerms = function () {
            var data = {
                actionName: "accept_terms",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return TermsAndConditionsCheckboxEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.TermsAndConditionsCheckboxEventFactory = TermsAndConditionsCheckboxEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    /**
     * Create ActionButtonsEventFactory class with path
     * @param path to this node
     */
    var ActionButtonsEventFactory = /** @class */ (function () {
        function ActionButtonsEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "action_buttons"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of GoBackButtonEventFactory
        */
        ActionButtonsEventFactory.prototype.goBackButtonFactory = function () {
            return new WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory.GoBackButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of NextButtonEventFactory
        */
        ActionButtonsEventFactory.prototype.nextButtonFactory = function () {
            return new WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory.NextButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of SubmitButtonEventFactory
        */
        ActionButtonsEventFactory.prototype.submitButtonFactory = function () {
            return new WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory.SubmitButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        ActionButtonsEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ActionButtonsEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return ActionButtonsEventFactory;
    }());
    WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory = ActionButtonsEventFactory;
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    var RoleDropdownEventFactory;
    (function (RoleDropdownEventFactory) {
        /**
         * Create RoleOptionsEventFactory class with path
         * @param path to this node
         */
        var RoleOptionsEventFactory = /** @class */ (function () {
            function RoleOptionsEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "role_options" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            RoleOptionsEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            RoleOptionsEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_select action.
             */
            RoleOptionsEventFactory.prototype.hitUiSelect = function () {
                var data = {
                    actionName: "ui_select",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return RoleOptionsEventFactory;
        }());
        RoleDropdownEventFactory.RoleOptionsEventFactory = RoleOptionsEventFactory;
    })(RoleDropdownEventFactory = WebAccessUserDetailsFormEventFactory.RoleDropdownEventFactory || (WebAccessUserDetailsFormEventFactory.RoleDropdownEventFactory = {}));
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    var ActionButtonsEventFactory;
    (function (ActionButtonsEventFactory) {
        /**
         * Create GoBackButtonEventFactory class with path
         * @param path to this node
         */
        var GoBackButtonEventFactory = /** @class */ (function () {
            function GoBackButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "go_back_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            GoBackButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            GoBackButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            GoBackButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return GoBackButtonEventFactory;
        }());
        ActionButtonsEventFactory.GoBackButtonEventFactory = GoBackButtonEventFactory;
    })(ActionButtonsEventFactory = WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory || (WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory = {}));
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    var ActionButtonsEventFactory;
    (function (ActionButtonsEventFactory) {
        /**
         * Create NextButtonEventFactory class with path
         * @param path to this node
         */
        var NextButtonEventFactory = /** @class */ (function () {
            function NextButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "next_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            NextButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            NextButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            NextButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return NextButtonEventFactory;
        }());
        ActionButtonsEventFactory.NextButtonEventFactory = NextButtonEventFactory;
    })(ActionButtonsEventFactory = WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory || (WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory = {}));
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
(function (WebAccessUserDetailsFormEventFactory) {
    var ActionButtonsEventFactory;
    (function (ActionButtonsEventFactory) {
        /**
         * Create SubmitButtonEventFactory class with path
         * @param path to this node
         */
        var SubmitButtonEventFactory = /** @class */ (function () {
            function SubmitButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "submit_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SubmitButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SubmitButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit request_team_access action.
             */
            SubmitButtonEventFactory.prototype.hitRequestTeamAccess = function () {
                var data = {
                    actionName: "request_team_access",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SubmitButtonEventFactory;
        }());
        ActionButtonsEventFactory.SubmitButtonEventFactory = SubmitButtonEventFactory;
    })(ActionButtonsEventFactory = WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory || (WebAccessUserDetailsFormEventFactory.ActionButtonsEventFactory = {}));
})(WebAccessUserDetailsFormEventFactory = exports.WebAccessUserDetailsFormEventFactory || (exports.WebAccessUserDetailsFormEventFactory = {}));
exports.WebAccessUserDetailsFormEventFactory = WebAccessUserDetailsFormEventFactory;
