// @spotify-internal/adstudio-shared/lib/components/LocationTargetingByCountry/CountrySearch/Suggestion/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var CountrySuggestionContainer = _styledComponents["default"].div.withConfig({
  displayName: "Suggestion__CountrySuggestionContainer",
  componentId: "sc-17o3ual-0"
})(["display:-webkit-box;display:-ms-flexbox;display:flex;margin:0;padding:14px 16px;span:first-child{text-transform:capitalize;}"]);

var Highlight = _styledComponents["default"].span.withConfig({
  displayName: "Suggestion__Highlight",
  componentId: "sc-17o3ual-1"
})(["", ""], function (_ref) {
  var highlight = _ref.highlight;
  return highlight && "\n      color: ".concat(_encoreAdvertisingWeb.gray75, ";\n      position: relative;\n    ");
});

var CountrySuggestion = function CountrySuggestion(_ref2) {
  var nameParts = _ref2.parts,
      _ref2$small = _ref2.small,
      small = _ref2$small === void 0 ? false : _ref2$small,
      suggestion = _ref2.suggestion;
  var name = suggestion.name,
      id = suggestion.id;
  return /*#__PURE__*/_react["default"].createElement(CountrySuggestionContainer, {
    small: small,
    "data-test": "country-autosuggest-option"
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.p, {
    condensed: true,
    variant: _encoreAdvertisingWeb.Type.body1
  }, nameParts && nameParts.map(function (_ref3, i) {
    var highlight = _ref3.highlight,
        text = _ref3.text;
    return /*#__PURE__*/_react["default"].createElement(Highlight, {
      highlight: highlight,
      key: i
    }, text);
  }), !nameParts && /*#__PURE__*/_react["default"].createElement("span", {
    id: id
  }, name)));
};

CountrySuggestion.propTypes = {
  parts: _propTypes["default"].array,
  small: _propTypes["default"].bool,
  suggestion: _propTypes["default"].shape({
    name: _propTypes["default"].string,
    id: _propTypes["default"].string
  })
};
var _default = CountrySuggestion;
exports["default"] = _default;
