// @spotify-internal/ubi-sdk-s4a-web-profile-merch/WebProfileMerchEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebProfileMerchEventFactory = exports.createWebProfileMerchEventFactory = void 0;
var SPECIFICATION_ID = "web-profile-merch";
var APP = "s4a";
var SPECIFICATION_VERSION = "1.0.2";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebProfileMerchEventFactory(options) {
    return new WebProfileMerchEventFactory(options);
}
exports.createWebProfileMerchEventFactory = createWebProfileMerchEventFactory;
/**
 * merch tab under profile in S4A
 */
var WebProfileMerchEventFactory = /** @class */ (function () {
    function WebProfileMerchEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @param id of the LinkShopButtonEventFactory
    * @param uri of the LinkShopButtonEventFactory
    * @returns A new instance of LinkShopButtonEventFactory
    */
    WebProfileMerchEventFactory.prototype.linkShopButtonFactory = function (data) {
        return new WebProfileMerchEventFactory.LinkShopButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation,
            data: data
        });
    };
    /**
    * @returns A new instance of PopupModalEventFactory
    */
    WebProfileMerchEventFactory.prototype.popupModalFactory = function () {
        return new WebProfileMerchEventFactory.PopupModalEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SpotifyShopEventFactory
    */
    WebProfileMerchEventFactory.prototype.spotifyShopFactory = function () {
        return new WebProfileMerchEventFactory.SpotifyShopEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ShopifyProductsEventFactory
    */
    WebProfileMerchEventFactory.prototype.shopifyProductsFactory = function () {
        return new WebProfileMerchEventFactory.ShopifyProductsEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ManageButtonEventFactory
    */
    WebProfileMerchEventFactory.prototype.manageButtonFactory = function () {
        return new WebProfileMerchEventFactory.ManageButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebProfileMerchEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebProfileMerchEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    /**
     * Create impression for WebProfileMerchEventFactory.
     */
    WebProfileMerchEventFactory.prototype.impression = function () {
        var data = {
            app: APP,
            generatorVersion: GENERATOR_VERSION,
            specificationVersion: SPECIFICATION_VERSION,
            specificationMode: SPECIFICATION_MODE,
            annotatorConfigurationVersion: '',
            annotatorVersion: '',
            location: this._getCurrentLocation(),
            parentAbsoluteLocation: this._parentAbsoluteLocation
        };
        return data;
    };
    return WebProfileMerchEventFactory;
}());
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    /**
     * Create LinkShopButtonEventFactory class with path
     * @param path to this node
     */
    var LinkShopButtonEventFactory = /** @class */ (function () {
        function LinkShopButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat(__assign(__assign({}, data), { name: "link_shop_button" }));
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        LinkShopButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        LinkShopButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        LinkShopButtonEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        /**
         * Create impression for LinkShopButtonEventFactory.
         */
        LinkShopButtonEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return LinkShopButtonEventFactory;
    }());
    WebProfileMerchEventFactory.LinkShopButtonEventFactory = LinkShopButtonEventFactory;
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    /**
     * Create PopupModalEventFactory class with path
     * @param path to this node
     */
    var PopupModalEventFactory = /** @class */ (function () {
        function PopupModalEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "popup_modal"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of CancelButtonEventFactory
        */
        PopupModalEventFactory.prototype.cancelButtonFactory = function () {
            return new WebProfileMerchEventFactory.PopupModalEventFactory.CancelButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the ContinueButtonEventFactory
        * @param uri of the ContinueButtonEventFactory
        * @returns A new instance of ContinueButtonEventFactory
        */
        PopupModalEventFactory.prototype.continueButtonFactory = function (data) {
            return new WebProfileMerchEventFactory.PopupModalEventFactory.ContinueButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        PopupModalEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        PopupModalEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for PopupModalEventFactory.
         */
        PopupModalEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return PopupModalEventFactory;
    }());
    WebProfileMerchEventFactory.PopupModalEventFactory = PopupModalEventFactory;
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    /**
     * Create SpotifyShopEventFactory class with path
     * @param path to this node
     */
    var SpotifyShopEventFactory = /** @class */ (function () {
        function SpotifyShopEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "spotify_shop"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of OfferingEventFactory
        */
        SpotifyShopEventFactory.prototype.offeringFactory = function () {
            return new WebProfileMerchEventFactory.SpotifyShopEventFactory.OfferingEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        SpotifyShopEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SpotifyShopEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for SpotifyShopEventFactory.
         */
        SpotifyShopEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SpotifyShopEventFactory;
    }());
    WebProfileMerchEventFactory.SpotifyShopEventFactory = SpotifyShopEventFactory;
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    /**
     * Create ShopifyProductsEventFactory class with path
     * @param path to this node
     */
    var ShopifyProductsEventFactory = /** @class */ (function () {
        function ShopifyProductsEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "shopify_products"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the OfferingEventFactory
        * @param uri of the OfferingEventFactory
        * @returns A new instance of OfferingEventFactory
        */
        ShopifyProductsEventFactory.prototype.offeringFactory = function (data) {
            return new WebProfileMerchEventFactory.ShopifyProductsEventFactory.OfferingEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        ShopifyProductsEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ShopifyProductsEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for ShopifyProductsEventFactory.
         */
        ShopifyProductsEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ShopifyProductsEventFactory;
    }());
    WebProfileMerchEventFactory.ShopifyProductsEventFactory = ShopifyProductsEventFactory;
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    /**
     * Create ManageButtonEventFactory class with path
     * @param path to this node
     */
    var ManageButtonEventFactory = /** @class */ (function () {
        function ManageButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "manage_button"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        ManageButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ManageButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit navigate_to_external_uri action.
       * @param destination the destination field
         */
        ManageButtonEventFactory.prototype.hitNavigateToExternalUri = function (_a) {
            var destination = _a.destination;
            var data = {
                actionParameterNames: ['destination'],
                actionParameterValues: [destination.toString()],
                actionName: "navigate_to_external_uri",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ManageButtonEventFactory;
    }());
    WebProfileMerchEventFactory.ManageButtonEventFactory = ManageButtonEventFactory;
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var PopupModalEventFactory;
    (function (PopupModalEventFactory) {
        /**
         * Create CancelButtonEventFactory class with path
         * @param path to this node
         */
        var CancelButtonEventFactory = /** @class */ (function () {
            function CancelButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "cancel_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CancelButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CancelButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_hide action.
             */
            CancelButtonEventFactory.prototype.hitUiHide = function () {
                var data = {
                    actionName: "ui_hide",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CancelButtonEventFactory;
        }());
        PopupModalEventFactory.CancelButtonEventFactory = CancelButtonEventFactory;
    })(PopupModalEventFactory = WebProfileMerchEventFactory.PopupModalEventFactory || (WebProfileMerchEventFactory.PopupModalEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var PopupModalEventFactory;
    (function (PopupModalEventFactory) {
        /**
         * Create ContinueButtonEventFactory class with path
         * @param path to this node
         */
        var ContinueButtonEventFactory = /** @class */ (function () {
            function ContinueButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "continue_button" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ContinueButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ContinueButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit navigate_to_external_uri action.
           * @param destination the destination field
             */
            ContinueButtonEventFactory.prototype.hitNavigateToExternalUri = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "navigate_to_external_uri",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ContinueButtonEventFactory;
        }());
        PopupModalEventFactory.ContinueButtonEventFactory = ContinueButtonEventFactory;
    })(PopupModalEventFactory = WebProfileMerchEventFactory.PopupModalEventFactory || (WebProfileMerchEventFactory.PopupModalEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var SpotifyShopEventFactory;
    (function (SpotifyShopEventFactory) {
        /**
         * Create OfferingEventFactory class with path
         * @param path to this node
         */
        var OfferingEventFactory = /** @class */ (function () {
            function OfferingEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "offering"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the RemoveButtonEventFactory
            * @param uri of the RemoveButtonEventFactory
            * @returns A new instance of RemoveButtonEventFactory
            */
            OfferingEventFactory.prototype.removeButtonFactory = function (data) {
                return new WebProfileMerchEventFactory.SpotifyShopEventFactory.OfferingEventFactory.RemoveButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            OfferingEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            OfferingEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return OfferingEventFactory;
        }());
        SpotifyShopEventFactory.OfferingEventFactory = OfferingEventFactory;
    })(SpotifyShopEventFactory = WebProfileMerchEventFactory.SpotifyShopEventFactory || (WebProfileMerchEventFactory.SpotifyShopEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var ShopifyProductsEventFactory;
    (function (ShopifyProductsEventFactory) {
        /**
         * Create OfferingEventFactory class with path
         * @param path to this node
         */
        var OfferingEventFactory = /** @class */ (function () {
            function OfferingEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "offering" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the AddButtonEventFactory
            * @param uri of the AddButtonEventFactory
            * @returns A new instance of AddButtonEventFactory
            */
            OfferingEventFactory.prototype.addButtonFactory = function (data) {
                return new WebProfileMerchEventFactory.ShopifyProductsEventFactory.OfferingEventFactory.AddButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            OfferingEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            OfferingEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return OfferingEventFactory;
        }());
        ShopifyProductsEventFactory.OfferingEventFactory = OfferingEventFactory;
    })(ShopifyProductsEventFactory = WebProfileMerchEventFactory.ShopifyProductsEventFactory || (WebProfileMerchEventFactory.ShopifyProductsEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var SpotifyShopEventFactory;
    (function (SpotifyShopEventFactory) {
        var OfferingEventFactory;
        (function (OfferingEventFactory) {
            /**
             * Create RemoveButtonEventFactory class with path
             * @param path to this node
             */
            var RemoveButtonEventFactory = /** @class */ (function () {
                function RemoveButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "remove_button" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                RemoveButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                RemoveButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit remove_product_from_shop action.
               * @param product_id Unique identifier for the product in the shop
                 */
                RemoveButtonEventFactory.prototype.hitRemoveProductFromShop = function (_a) {
                    var productId = _a.productId;
                    var data = {
                        actionParameterNames: ['product_id'],
                        actionParameterValues: [productId.toString()],
                        actionName: "remove_product_from_shop",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return RemoveButtonEventFactory;
            }());
            OfferingEventFactory.RemoveButtonEventFactory = RemoveButtonEventFactory;
        })(OfferingEventFactory = SpotifyShopEventFactory.OfferingEventFactory || (SpotifyShopEventFactory.OfferingEventFactory = {}));
    })(SpotifyShopEventFactory = WebProfileMerchEventFactory.SpotifyShopEventFactory || (WebProfileMerchEventFactory.SpotifyShopEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
(function (WebProfileMerchEventFactory) {
    var ShopifyProductsEventFactory;
    (function (ShopifyProductsEventFactory) {
        var OfferingEventFactory;
        (function (OfferingEventFactory) {
            /**
             * Create AddButtonEventFactory class with path
             * @param path to this node
             */
            var AddButtonEventFactory = /** @class */ (function () {
                function AddButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "add_button" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                AddButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                AddButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit add_product_to_shop action.
               * @param product_id Unique identifier for the product in the shop
                 */
                AddButtonEventFactory.prototype.hitAddProductToShop = function (_a) {
                    var productId = _a.productId;
                    var data = {
                        actionParameterNames: ['product_id'],
                        actionParameterValues: [productId.toString()],
                        actionName: "add_product_to_shop",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return AddButtonEventFactory;
            }());
            OfferingEventFactory.AddButtonEventFactory = AddButtonEventFactory;
        })(OfferingEventFactory = ShopifyProductsEventFactory.OfferingEventFactory || (ShopifyProductsEventFactory.OfferingEventFactory = {}));
    })(ShopifyProductsEventFactory = WebProfileMerchEventFactory.ShopifyProductsEventFactory || (WebProfileMerchEventFactory.ShopifyProductsEventFactory = {}));
})(WebProfileMerchEventFactory = exports.WebProfileMerchEventFactory || (exports.WebProfileMerchEventFactory = {}));
exports.WebProfileMerchEventFactory = WebProfileMerchEventFactory;
