// @spotify-internal/encore-web/es/components/NavigationAction/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { Link } from "./Link";
export function NavigationAction(_ref) {
  var active = _ref.active,
      disabled = _ref.disabled,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      restProps = _objectWithoutProperties(_ref, ["active", "disabled", "component"]);

  return /*#__PURE__*/React.createElement(Link, Object.assign({}, restProps, {
    active: active,
    component: component,
    disabled: disabled,
    "aria-disabled": disabled,
    rel: restProps.href ? 'noopener noreferrer' : undefined,
    "aria-current": active ? 'page' : undefined
  }));
}
/**
* **NavigationAction** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavigationAction) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navigation-navigationaction--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=5384:16) |
*
* NavigationAction is a primary action button that fills it's container width and is used at the bottom of the Navigation component.
*
* @example
* export const Default = () => (
*  <NavigationAction target="_blank" title="Start your first release whenever you're ready">
*    <IconPlus />
*  </NavigationAction>
*);
*
*/

export default NavigationAction;
