// @spotify-internal/adstudio-tape/lib/components/NavigationList/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = NavigationList;
exports.NavigationListDl = exports.NavigationListNav = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var propTypes = {
  /** Can accept `NavigationListItem` or any HTML */
  children: _propTypes["default"].node
};

var NavigationListNav = _styledComponents["default"].nav.withConfig({
  displayName: "NavigationList__NavigationListNav",
  componentId: "piaa56-0"
})(["display:block;"]);

exports.NavigationListNav = NavigationListNav;

var NavigationListDl = _styledComponents["default"].dl.withConfig({
  displayName: "NavigationList__NavigationListDl",
  componentId: "piaa56-1"
})(["margin-bottom:0;padding-bottom:0;"]);

exports.NavigationListDl = NavigationListDl;

function NavigationList(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/_react["default"].createElement(NavigationListNav, props, /*#__PURE__*/_react["default"].createElement(NavigationListDl, null, children));
}

NavigationList.propTypes = propTypes;
