// @spotify-internal/adstudio-shared/lib/components/FrequencyCaps/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.validation = exports.defaultProps = exports.Units = void 0;

var _react = _interopRequireWildcard(require("react"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

var StyledPanel = _styledComponents["default"].div.withConfig({
  displayName: "FrequencyCaps__StyledPanel",
  componentId: "sc-1pnw7mx-0"
})(["border-color:", ";border-radius:4px;box-shadow:0px 1px 3px rgba(0,0,0,0.2);padding:1rem;display:flex;"], _encoreAdvertisingWeb.gray90);

var NoHeaderContainer = _styledComponents["default"].div.withConfig({
  displayName: "FrequencyCaps__NoHeaderContainer",
  componentId: "sc-1pnw7mx-1"
})(["padding-top:2.3rem;margin-right:3.5rem;"]);

var StyledFormSelect = (0, _styledComponents["default"])(_encoreAdvertisingWeb.FormSelect).withConfig({
  displayName: "FrequencyCaps__StyledFormSelect",
  componentId: "sc-1pnw7mx-2"
})(["width:5rem;"]);
var StyledFormInput = (0, _styledComponents["default"])(_encoreAdvertisingWeb.FormInput).withConfig({
  displayName: "FrequencyCaps__StyledFormInput",
  componentId: "sc-1pnw7mx-3"
})(["margin-top:0.5rem;"]);
var InfoLabel = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "FrequencyCaps__InfoLabel",
  componentId: "sc-1pnw7mx-4"
})(["margin-top:", ";color:", ";font-size:14px;"], _encoreAdvertisingWeb.spacer8, _encoreAdvertisingWeb.gray60);
var ErrorLabel = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type).withConfig({
  displayName: "FrequencyCaps__ErrorLabel",
  componentId: "sc-1pnw7mx-5"
})(["margin-top:", ";color:", ";font-size:14px;"], _encoreAdvertisingWeb.spacer8, _encoreAdvertisingWeb.red);
var Units;
exports.Units = Units;

(function (Units) {
  Units[Units["days"] = 1] = "days";
  Units[Units["weeks"] = 2] = "weeks";
})(Units || (exports.Units = Units = {}));

var strings = {
  daysOption: 'Days',
  weeksOption: 'Weeks',
  description: 'This determines the maximum number of ads a user will be served in a chosen time period.',
  capsLabel: 'Set cap by:',
  maxAdsLabel: 'Maximum ads served',
  maxAdsPlaceholder: 'Enter a number',
  error: 'Number must be equal to or below'
};
var defaultProps = {
  limits: {
    days: 4,
    weeks: 31
  },
  init: {
    time: 1,
    timeUnit: Units.days,
    maxImpressions: 5,
    valid: false
  },
  onChange: function onChange() {
    return null;
  },
  error: false,
  logUserAction: function logUserAction() {
    return null;
  },
  strings: strings
};
exports.defaultProps = defaultProps;

var validation = function validation(value) {
  var valid = value.valid,
      maxImpressions = value.maxImpressions;

  if (!valid || isNaN(maxImpressions)) {
    return true;
  }

  return;
};

exports.validation = validation;

var FrequencyCapsElement = function FrequencyCapsElement(configs) {
  var limits = configs.limits,
      onChange = configs.onChange,
      init = configs.init;

  var days = _toConsumableArray(new Array(31));

  var weeks = _toConsumableArray(new Array(4));

  var maxAllowed = (0, _react.useCallback)(function (unit, value) {
    var timeUnit = unit === Units.weeks ? limits.weeks : limits.days;
    return timeUnit * value - 1;
  }, [limits]);

  var unitsArray = function unitsArray(unit) {
    return unit === Units.weeks ? weeks : days;
  };

  var _useState = (0, _react.useState)(init.timeUnit),
      _useState2 = _slicedToArray(_useState, 2),
      unit = _useState2[0],
      setUnit = _useState2[1];

  var _useState3 = (0, _react.useState)(init.time),
      _useState4 = _slicedToArray(_useState3, 2),
      value = _useState4[0],
      setValue = _useState4[1];

  var _useState5 = (0, _react.useState)(init.maxImpressions),
      _useState6 = _slicedToArray(_useState5, 2),
      inputMax = _useState6[0],
      setInputMax = _useState6[1];

  var hasError = inputMax < 1 || inputMax > maxAllowed(unit, value);
  (0, _react.useEffect)(function () {
    onChange({
      time: value,
      timeUnit: unit,
      maxImpressions: inputMax,
      valid: !hasError
    });
  }, [unit, value, hasError, inputMax, onChange]);

  var handleMaxInput = function handleMaxInput(ev) {
    var maxInputValue = ev.target.value;

    if (!isNaN(maxInputValue)) {
      setInputMax(Number(maxInputValue));
    }
  };

  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(StyledPanel, null, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.FormGroup, {
    role: "radiogroup",
    label: strings.capsLabel
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.FormRadio, {
    name: "units",
    defaultChecked: unit === Units.days,
    onChange: function onChange() {
      setUnit(Units.days);
      setValue(1);
    }
  }, strings.daysOption), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.FormRadio, {
    name: "units",
    defaultChecked: unit === Units.weeks,
    onChange: function onChange() {
      setUnit(Units.weeks);
      setValue(1);
    }
  }, strings.weeksOption)), /*#__PURE__*/_react["default"].createElement(NoHeaderContainer, null, /*#__PURE__*/_react["default"].createElement(StyledFormSelect, {
    value: value,
    onChange: function onChange(ev) {
      setValue(ev.target.value);
    },
    defaultValue: value
  }, unitsArray(unit).map(function (_, i) {
    return /*#__PURE__*/_react["default"].createElement("option", {
      value: i + 1
    }, i + 1);
  }))), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.FormGroup, {
    label: strings.maxAdsLabel
  }, /*#__PURE__*/_react["default"].createElement(StyledFormInput, {
    error: hasError,
    onInput: handleMaxInput,
    value: inputMax,
    placeholder: strings.maxAdsPlaceholder
  }))), hasError && /*#__PURE__*/_react["default"].createElement(ErrorLabel, {
    as: "div"
  }, strings.error), /*#__PURE__*/_react["default"].createElement(InfoLabel, {
    as: "div"
  }, strings.description));
};

FrequencyCapsElement.defaultProps = defaultProps;
var _default = FrequencyCapsElement;
exports["default"] = _default;
