// @spotify-internal/encore-web/es/components/TableCell/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { TableCellElement } from "./TableCell";

function TableCellComponent(_ref) {
  var _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$align = _ref.align,
      align = _ref$align === void 0 ? 'left' : _ref$align,
      props = _objectWithoutProperties(_ref, ["theme", "align"]);

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(TableCellElement, Object.assign({
    align: align
  }, props)));
}

var tableCellHelpers = {
  darkTheme: {
    tableCell: 'dark'
  }
};
/**
* **TableCell** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/TableCell) | [Storybook](https://encore-web.spotify.net/?path=/story/components-table-tablecell--highlight) | [Encore](https://encore.spotify.net/web/components/table?format=figma-figma#cell) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=525:1495) |
*
* Cells are the most basic unit of a table. They are contained within rows and influenced by other cells in the same column.
*
* @example
* export const Highlight = () => (
*  <Table>
*    <tbody>
*      <TableRow>
*        <TableCell as="th" scope="row" highlight>
*          Highlight
*        </TableCell>
*        <TableCell>Default</TableCell>
*        <TableCell>Default</TableCell>
*      </TableRow>
*    </tbody>
*  </Table>
*);
*
*/

export var TableCell = Object.assign(TableCellComponent, tableCellHelpers);
export default TableCell;
