// @spotify-internal/encore-web/es/components/Image/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _slicedToArray from "@babel/runtime/helpers/esm/slicedToArray";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useRef, useState } from 'react';
import StyledImage, { PlaceholderImage, PlaceholderImageWrapper, StyledLazyImage } from "./Image";
import { useInView } from "./useInView";

function LazyImage(_ref) {
  var _ref$alt = _ref.alt,
      alt = _ref$alt === void 0 ? '' : _ref$alt,
      props = _objectWithoutProperties(_ref, ["alt"]);

  var imageRef = useRef(null);

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      inView = _useState2[0],
      setInView = _useState2[1];

  var _useState3 = useState(false),
      _useState4 = _slicedToArray(_useState3, 2),
      isLoaded = _useState4[0],
      setIsLoaded = _useState4[1];

  var lazyImage = /*#__PURE__*/React.createElement(StyledLazyImage, Object.assign({}, props, {
    alt: alt,
    ref: imageRef,
    onLoad: function onLoad() {
      return setIsLoaded(true);
    },
    isLoaded: isLoaded
  }));
  var placeholder = /*#__PURE__*/React.createElement(PlaceholderImageWrapper, props, /*#__PURE__*/React.createElement(PlaceholderImage, {
    alt: alt,
    ref: imageRef,
    src: props.placeholderSrc
  }));
  useInView(imageRef, function () {
    setInView(true);
  });

  if (inView) {
    return /*#__PURE__*/React.createElement(React.Fragment, null, !isLoaded && placeholder, lazyImage);
  }

  return placeholder;
}

export function Image(_ref2) {
  var props = _extends({}, _ref2);

  if ('placeholderSrc' in props && props.placeholderSrc) {
    return /*#__PURE__*/React.createElement(LazyImage, props);
  }

  return /*#__PURE__*/React.createElement(StyledImage, props);
}
/**
* **Image** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Image) | [Storybook](https://encore-web.spotify.net/?path=/story/components-image--default) | [Encore](https://encore.spotify.net/web/components/image) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=265:13) |
*
* Use Image to render a simple image by using a link to its location.
*
* @example
* export const Default = () => (
*  <Image src="https://i.scdn.co/image/54470e5e056b93d7022076960d3a6b2267ece226" alt="Vacationer" />
*);
*
*/

export default Image;
