// @spotify-internal/adstudio-shared/lib/components/BudgetPicker/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HELP_SETUP_LARGER_CAMPAIGNS = exports.HAS_MAXIMUM_BUDGET = exports.HAS_MINIMUM_BUDGET = exports.NUMERIC_ERROR = exports.PREMIUM_ERROR = exports.BUDGET_PICKER_DESCRIPTION = exports.CUSTOM_BUDGET_PLACEHOLDER = void 0;
var CUSTOM_BUDGET_PLACEHOLDER = 'Custom amount';
exports.CUSTOM_BUDGET_PLACEHOLDER = CUSTOM_BUDGET_PLACEHOLDER;
var BUDGET_PICKER_DESCRIPTION = 'Your budget will be evenly distributed between start and end dates.';
exports.BUDGET_PICKER_DESCRIPTION = BUDGET_PICKER_DESCRIPTION;
var PREMIUM_ERROR = 'Please contact adstudio@spotify.com for premium ad options.';
exports.PREMIUM_ERROR = PREMIUM_ERROR;
var NUMERIC_ERROR = 'Please enter a valid amount.';
exports.NUMERIC_ERROR = NUMERIC_ERROR;
var HAS_MINIMUM_BUDGET = "Ad Studio has a minimum budget of";
exports.HAS_MINIMUM_BUDGET = HAS_MINIMUM_BUDGET;
var HAS_MAXIMUM_BUDGET = "Ad Studio's maximum budget is";
exports.HAS_MAXIMUM_BUDGET = HAS_MAXIMUM_BUDGET;
var HELP_SETUP_LARGER_CAMPAIGNS = "Contact adstudio@spotify.com for help setting up larger campaigns.";
exports.HELP_SETUP_LARGER_CAMPAIGNS = HELP_SETUP_LARGER_CAMPAIGNS;
