// @react-aria/combobox/dist/module.js
import { useTextField } from "@react-aria/textfield";
import { useMessageFormatter } from "@react-aria/i18n";
import { useMenuTrigger } from "@react-aria/menu";
import { ListKeyboardDelegate, useSelectableCollection } from "@react-aria/selection";
import { getItemCount } from "@react-stately/collections";
import { useEffect, useMemo, useRef } from "react";
import { chain, isAppleDevice, mergeProps, useLabels } from "@react-aria/utils";
import { getItemId, listData } from "@react-aria/listbox";
import { ariaHideOutside } from "@react-aria/overlays";
import { announce } from "@react-aria/live-announcer";
import _babelRuntimeHelpersEsmInteropRequireDefault from "@babel/runtime/helpers/esm/interopRequireDefault";
import _babelRuntimeHelpersEsmExtends from "@babel/runtime/helpers/esm/extends";
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/ar-AE.json
var $d78683a0e7fef7294b62031fc43054c6$exports = {};
$d78683a0e7fef7294b62031fc43054c6$exports = JSON.parse("{\"buttonLabel\":\"عرض المقترحات\",\"countAnnouncement\":\"{optionCount, plural, one {# خيار} other {# خيارات}} متاحة.\",\"focusAnnouncement\":\"{isGroupChange, select, true {المجموعة المدخلة {groupTitle}, مع {groupCount, plural, one {# خيار} other {# خيارات}}. } other {}}{optionText}{isSelected, select, true {, محدد} other {}}\",\"listboxLabel\":\"مقترحات\",\"selectedAnnouncement\":\"{optionText}، محدد\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/bg-BG.json
var $f98ae6ce2700d4a20c4e60ff9d68eed9$exports = {};
$f98ae6ce2700d4a20c4e60ff9d68eed9$exports = JSON.parse("{\"buttonLabel\":\"Покажи предложения\",\"countAnnouncement\":\"{optionCount, plural, one {# опция} other {# опции}} на разположение.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Въведена група {groupTitle}, с {groupCount, plural, one {# опция} other {# опции}}. } other {}}{optionText}{isSelected, select, true {, избрани} other {}}\",\"listboxLabel\":\"Предложения\",\"selectedAnnouncement\":\"{optionText}, избрани\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/cs-CZ.json
var $e18f88738dd7cf3410585df50c4a$exports = {};
$e18f88738dd7cf3410585df50c4a$exports = JSON.parse("{\"buttonLabel\":\"Zobrazit doporučení\",\"countAnnouncement\":\"K dispozici {optionCount, plural, one {je # možnost} other {jsou/je # možnosti/-í}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Zadaná skupina „{groupTitle}“ {groupCount, plural, one {s # možností} other {se # možnostmi}}. } other {}}{optionText}{isSelected, select, true { (vybráno)} other {}}\",\"listboxLabel\":\"Návrhy\",\"selectedAnnouncement\":\"{optionText}, vybráno\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/da-DK.json
var $bc87a01dd17f8e08d9b5f2d709d95ab$exports = {};
$bc87a01dd17f8e08d9b5f2d709d95ab$exports = JSON.parse("{\"buttonLabel\":\"Vis forslag\",\"countAnnouncement\":\"{optionCount, plural, one {# mulighed tilgængelig} other {# muligheder tilgængelige}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Angivet gruppe {groupTitle}, med {groupCount, plural, one {# mulighed} other {# muligheder}}. } other {}}{optionText}{isSelected, select, true {, valgt} other {}}\",\"listboxLabel\":\"Forslag\",\"selectedAnnouncement\":\"{optionText}, valgt\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/de-DE.json
var $a2876bc991816077d10a6a3c448303e$exports = {};
$a2876bc991816077d10a6a3c448303e$exports = JSON.parse("{\"buttonLabel\":\"Empfehlungen anzeigen\",\"countAnnouncement\":\"{optionCount, plural, one {# Option} other {# Optionen}} verfügbar.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Eingetretene Gruppe {groupTitle}, mit {groupCount, plural, one {# Option} other {# Optionen}}. } other {}}{optionText}{isSelected, select, true {, ausgewählt} other {}}\",\"listboxLabel\":\"Empfehlungen\",\"selectedAnnouncement\":\"{optionText}, ausgewählt\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/el-GR.json
var $b2fff3cd75c7191d2e6133d07588de86$exports = {};
$b2fff3cd75c7191d2e6133d07588de86$exports = JSON.parse("{\"buttonLabel\":\"Προβολή προτάσεων\",\"countAnnouncement\":\"{optionCount, plural, one {# επιλογή} other {# επιλογές }} διαθέσιμες.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Εισαγμένη ομάδα {groupTitle}, με {groupCount, plural, one {# επιλογή} other {# επιλογές}}. } other {}}{optionText}{isSelected, select, true {, επιλεγμένο} other {}}\",\"listboxLabel\":\"Προτάσεις\",\"selectedAnnouncement\":\"{optionText}, επιλέχθηκε\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/en-US.json
var $f1639f2754c31f58763b8cb015a6ecd6$exports = {};
$f1639f2754c31f58763b8cb015a6ecd6$exports = JSON.parse("{\"focusAnnouncement\":\"{isGroupChange, select, true {Entered group {groupTitle}, with {groupCount, plural, one {# option} other {# options}}. } other {}}{optionText}{isSelected, select, true {, selected} other {}}\",\"countAnnouncement\":\"{optionCount, plural, one {# option} other {# options}} available.\",\"selectedAnnouncement\":\"{optionText}, selected\",\"buttonLabel\":\"Show suggestions\",\"listboxLabel\":\"Suggestions\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/es-ES.json
var $f0fff8208defd2a975d21faa464abfd$exports = {};
$f0fff8208defd2a975d21faa464abfd$exports = JSON.parse("{\"buttonLabel\":\"Mostrar sugerencias\",\"countAnnouncement\":\"{optionCount, plural, one {# opción} other {# opciones}} disponible(s).\",\"focusAnnouncement\":\"{isGroupChange, select, true {Se ha unido al grupo {groupTitle}, con {groupCount, plural, one {# opción} other {# opciones}}. } other {}}{optionText}{isSelected, select, true {, seleccionado} other {}}\",\"listboxLabel\":\"Sugerencias\",\"selectedAnnouncement\":\"{optionText}, seleccionado\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/et-EE.json
var $c22ee6a7566d1cf5578bf8d4f5c2a6$exports = {};
$c22ee6a7566d1cf5578bf8d4f5c2a6$exports = JSON.parse("{\"buttonLabel\":\"Kuva soovitused\",\"countAnnouncement\":\"{optionCount, plural, one {# valik} other {# valikud}} saadaval.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Sisestatud rühm {groupTitle}, valikuga {groupCount, plural, one {# valik} other {# valikud}}. } other {}}{optionText}{isSelected, select, true {, valitud} other {}}\",\"listboxLabel\":\"Soovitused\",\"selectedAnnouncement\":\"{optionText}, valitud\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/fi-FI.json
var $c0df15e7925dcb04d7c1787b03a5e$exports = {};
$c0df15e7925dcb04d7c1787b03a5e$exports = JSON.parse("{\"buttonLabel\":\"Näytä ehdotukset\",\"countAnnouncement\":\"{optionCount, plural, one {# vaihtoehto} other {# vaihtoehdot}} saatavilla.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Mentiin ryhmään {groupTitle}, {groupCount, plural, one {# vaihtoehdon} other {# vaihtoehdon}} kanssa.} other {}}{optionText}{isSelected, select, true {, valittu} other {}}\",\"listboxLabel\":\"Ehdotukset\",\"selectedAnnouncement\":\"{optionText}, valittu\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/fr-FR.json
var $b36e7dd5de5dec60de8490c4382ec$exports = {};
$b36e7dd5de5dec60de8490c4382ec$exports = JSON.parse("{\"buttonLabel\":\"Afficher les suggestions\",\"countAnnouncement\":\"{optionCount, plural, one {# option} other {# options}} disponible(s).\",\"focusAnnouncement\":\"{isGroupChange, select, true {Groupe {groupTitle} saisi, avec {groupCount, plural, one {# option} other {# options}}. } other {}}{optionText}{isSelected, select, true {, sélectionné(s)} other {}}\",\"listboxLabel\":\"Suggestions\",\"selectedAnnouncement\":\"{optionText}, sélectionné\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/he-IL.json
var $c0bb0e2e7b387fc7e7e478556f7f7d7$exports = {};
$c0bb0e2e7b387fc7e7e478556f7f7d7$exports = JSON.parse("{\"buttonLabel\":\"הצג הצעות\",\"countAnnouncement\":\"{optionCount, plural, one {אפשרות #} other {# אפשרויות}} במצב זמין.\",\"focusAnnouncement\":\"{isGroupChange, select, true {נכנס לקבוצה {groupTitle}, עם {groupCount, plural, one {אפשרות #} other {# אפשרויות}}. } other {}}{optionText}{isSelected, select, true {, נבחר} other {}}\",\"listboxLabel\":\"הצעות\",\"selectedAnnouncement\":\"{optionText}, נבחר\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/hr-HR.json
var $c2ad6267acd85a3c0b84c89b176b1b$exports = {};
$c2ad6267acd85a3c0b84c89b176b1b$exports = JSON.parse("{\"buttonLabel\":\"Prikaži prijedloge\",\"countAnnouncement\":\"Dostupno još: {optionCount, plural, one {# opcija} other {# opcije/a}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Unesena skupina {groupTitle}, s {groupCount, plural, one {# opcijom} other {# opcije/a}}. } other {}}{optionText}{isSelected, select, true {, odabranih} other {}}\",\"listboxLabel\":\"Prijedlozi\",\"selectedAnnouncement\":\"{optionText}, odabrano\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/hu-HU.json
var $b72435e6c734787569d9dde4943b4667$exports = {};
$b72435e6c734787569d9dde4943b4667$exports = JSON.parse("{\"buttonLabel\":\"Javaslatok megjelenítése\",\"countAnnouncement\":\"{optionCount, plural, one {# lehetőség} other {# lehetőség}} áll rendelkezésre.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Belépett a(z) {groupTitle} csoportba, amely {groupCount, plural, one {# lehetőséget} other {# lehetőséget}} tartalmaz. } other {}}{optionText}{isSelected, select, true {, kijelölve} other {}}\",\"listboxLabel\":\"Javaslatok\",\"selectedAnnouncement\":\"{optionText}, kijelölve\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/it-IT.json
var $ec385ad6d76803c0e13ef3155a37731f$exports = {};
$ec385ad6d76803c0e13ef3155a37731f$exports = JSON.parse("{\"buttonLabel\":\"Mostra suggerimenti\",\"countAnnouncement\":\"{optionCount, plural, one {# opzione disponibile} other {# opzioni disponibili}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Ingresso nel gruppo {groupTitle}, con {groupCount, plural, one {# opzione} other {# opzioni}}. } other {}}{optionText}{isSelected, select, true {, selezionato} other {}}\",\"listboxLabel\":\"Suggerimenti\",\"selectedAnnouncement\":\"{optionText}, selezionato\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/ja-JP.json
var $f6712684539b6a07d53079e26b70d9df$exports = {};
$f6712684539b6a07d53079e26b70d9df$exports = JSON.parse("{\"buttonLabel\":\"候補を表示\",\"countAnnouncement\":\"{optionCount, plural, one {# 個のオプション} other {# 個のオプション}}を利用できます。\",\"focusAnnouncement\":\"{isGroupChange, select, true {入力されたグループ {groupTitle}、{groupCount, plural, one {# 個のオプション} other {# 個のオプション}}を含む。} other {}}{optionText}{isSelected, select, true {、選択済み} other {}}\",\"listboxLabel\":\"候補\",\"selectedAnnouncement\":\"{optionText}、選択済み\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/ko-KR.json
var $c670ae89d3913146f04a762d163$exports = {};
$c670ae89d3913146f04a762d163$exports = JSON.parse("{\"buttonLabel\":\"제안 사항 표시\",\"countAnnouncement\":\"{optionCount, plural, one {#개 옵션} other {#개 옵션}}을 사용할 수 있습니다.\",\"focusAnnouncement\":\"{isGroupChange, select, true {입력한 그룹 {groupTitle}, {groupCount, plural, one {#개 옵션} other {#개 옵션}}. } other {}}{optionText}{isSelected, select, true {, 선택됨} other {}}\",\"listboxLabel\":\"제안\",\"selectedAnnouncement\":\"{optionText}, 선택됨\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/lt-LT.json
var $a36750eea27db776e11cc103d172017$exports = {};
$a36750eea27db776e11cc103d172017$exports = JSON.parse("{\"buttonLabel\":\"Rodyti pasiūlymus\",\"countAnnouncement\":\"Yra {optionCount, plural, one {# parinktis} other {# parinktys (-ių)}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Įvesta grupė {groupTitle}, su {groupCount, plural, one {# parinktimi} other {# parinktimis (-ių)}}. } other {}}{optionText}{isSelected, select, true {, pasirinkta} other {}}\",\"listboxLabel\":\"Pasiūlymai\",\"selectedAnnouncement\":\"{optionText}, pasirinkta\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/lv-LV.json
var $e840ed8916bd1f210105d1fb3f88697$exports = {};
$e840ed8916bd1f210105d1fb3f88697$exports = JSON.parse("{\"buttonLabel\":\"Rādīt ieteikumus\",\"countAnnouncement\":\"Pieejamo opciju skaits: {optionCount, plural, one {# opcija} other {# opcijas}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Ievadīta grupa {groupTitle}, ar {groupCount, plural, one {# opciju} other {# opcijām}}. } other {}}{optionText}{isSelected, select, true {, atlasīta} other {}}\",\"listboxLabel\":\"Ieteikumi\",\"selectedAnnouncement\":\"{optionText}, atlasīta\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/nb-NO.json
var $a9d36ce32fd5aea219762aea75a1360$exports = {};
$a9d36ce32fd5aea219762aea75a1360$exports = JSON.parse("{\"buttonLabel\":\"Vis forslag\",\"countAnnouncement\":\"{optionCount, plural, one {# alternativ} other {# alternativer}} finnes.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Angitt gruppe {groupTitle}, med {groupCount, plural, one {# alternativ} other {# alternativer}}. } other {}}{optionText}{isSelected, select, true {, valgt} other {}}\",\"listboxLabel\":\"Forslag\",\"selectedAnnouncement\":\"{optionText}, valgt\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/nl-NL.json
var $c4a143186784391623f115223360f80$exports = {};
$c4a143186784391623f115223360f80$exports = JSON.parse("{\"buttonLabel\":\"Suggesties weergeven\",\"countAnnouncement\":\"{optionCount, plural, one {# optie} other {# opties}} beschikbaar.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Groep {groupTitle} ingevoerd met {groupCount, plural, one {# optie} other {# opties}}. } other {}}{optionText}{isSelected, select, true {, geselecteerd} other {}}\",\"listboxLabel\":\"Suggesties\",\"selectedAnnouncement\":\"{optionText}, geselecteerd\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/pl-PL.json
var $a32834a3efc5c6ca32d91bd08$exports = {};
$a32834a3efc5c6ca32d91bd08$exports = JSON.parse("{\"buttonLabel\":\"Wyświetlaj sugestie\",\"countAnnouncement\":\"dostępna/dostępne(-nych) {optionCount, plural, one {# opcja} other {# opcje(-i)}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Dołączono do grupy {groupTitle}, z {groupCount, plural, one {# opcją} other {# opcjami}}. } other {}}{optionText}{isSelected, select, true {, wybrano} other {}}\",\"listboxLabel\":\"Sugestie\",\"selectedAnnouncement\":\"{optionText}, wybrano\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/pt-BR.json
var $e1b551c29bce2e3141bec9bc8b1ddba1$exports = {};
$e1b551c29bce2e3141bec9bc8b1ddba1$exports = JSON.parse("{\"buttonLabel\":\"Mostrar sugestões\",\"countAnnouncement\":\"{optionCount, plural, one {# opção} other {# opções}} disponível.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Grupo inserido {groupTitle}, com {groupCount, plural, one {# opção} other {# opções}}. } other {}}{optionText}{isSelected, select, true {, selecionado} other {}}\",\"listboxLabel\":\"Sugestões\",\"selectedAnnouncement\":\"{optionText}, selecionado\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/pt-PT.json
var $e10cdad02310abdd1550ffa8b7db9ba$exports = {};
$e10cdad02310abdd1550ffa8b7db9ba$exports = JSON.parse("{\"buttonLabel\":\"Apresentar sugestões\",\"countAnnouncement\":\"{optionCount, plural, one {# opção} other {# opções}} disponível.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Grupo introduzido {groupTitle}, com {groupCount, plural, one {# opção} other {# opções}}. } other {}}{optionText}{isSelected, select, true {, selecionado} other {}}\",\"listboxLabel\":\"Sugestões\",\"selectedAnnouncement\":\"{optionText}, selecionado\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/ro-RO.json
var $ca354d1ee6bae344a049de3e819052c$exports = {};
$ca354d1ee6bae344a049de3e819052c$exports = JSON.parse("{\"buttonLabel\":\"Afișare sugestii\",\"countAnnouncement\":\"{optionCount, plural, one {# opțiune} other {# opțiuni}} disponibile.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Grup {groupTitle} introdus, cu {groupCount, plural, one {# opțiune} other {# opțiuni}}. } other {}}{optionText}{isSelected, select, true {, selectat} other {}}\",\"listboxLabel\":\"Sugestii\",\"selectedAnnouncement\":\"{optionText}, selectat\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/ru-RU.json
var $f7006032145148654e07a693955fb9c2$exports = {};
$f7006032145148654e07a693955fb9c2$exports = JSON.parse("{\"buttonLabel\":\"Показать предложения\",\"countAnnouncement\":\"{optionCount, plural, one {# параметр} other {# параметров}} доступно.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Введенная группа {groupTitle}, с {groupCount, plural, one {# параметром} other {# параметрами}}. } other {}}{optionText}{isSelected, select, true {, выбранными} other {}}\",\"listboxLabel\":\"Предложения\",\"selectedAnnouncement\":\"{optionText}, выбрано\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/sk-SK.json
var $c60520bb4ea0d2c95ea4b69dc1046132$exports = {};
$c60520bb4ea0d2c95ea4b69dc1046132$exports = JSON.parse("{\"buttonLabel\":\"Zobraziť návrhy\",\"countAnnouncement\":\"{optionCount, plural, one {# možnosť} other {# možnosti/-í}} k dispozícii.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Zadaná skupina {groupTitle}, s {groupCount, plural, one {# možnosťou} other {# možnosťami}}. } other {}}{optionText}{isSelected, select, true {, vybraté} other {}}\",\"listboxLabel\":\"Návrhy\",\"selectedAnnouncement\":\"{optionText}, vybraté\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/sl-SI.json
var $fd97d8e462e8521b6da71e6762a5c$exports = {};
$fd97d8e462e8521b6da71e6762a5c$exports = JSON.parse("{\"buttonLabel\":\"Prikaži predloge\",\"countAnnouncement\":\"Na voljo je {optionCount, plural, one {# opcija} other {# opcije}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Vnesena skupina {groupTitle}, z {groupCount, plural, one {# opcija} other {# opcije}}. } other {}}{optionText}{isSelected, select, true {, izbrano} other {}}\",\"listboxLabel\":\"Predlogi\",\"selectedAnnouncement\":\"{optionText}, izbrano\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/sr-SP.json
var $b615e8cf79bac7ecb7173b279f2a7$exports = {};
$b615e8cf79bac7ecb7173b279f2a7$exports = JSON.parse("{\"buttonLabel\":\"Prikaži predloge\",\"countAnnouncement\":\"Dostupno još: {optionCount, plural, one {# opcija} other {# opcije/a}}.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Unesena grupa {groupTitle}, s {groupCount, plural, one {# opcijom} other {# optione/a}}. } other {}}{optionText}{isSelected, select, true {, izabranih} other {}}\",\"listboxLabel\":\"Predlozi\",\"selectedAnnouncement\":\"{optionText}, izabrano\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/sv-SE.json
var $f7c7084482b80ff47daa2804ba64c$exports = {};
$f7c7084482b80ff47daa2804ba64c$exports = JSON.parse("{\"buttonLabel\":\"Visa förslag\",\"countAnnouncement\":\"{optionCount, plural, one {# alternativ} other {# alternativ}} tillgängliga.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Ingick i gruppen {groupTitle} med {groupCount, plural, one {# alternativ} other {# alternativ}}. } other {}}{optionText}{isSelected, select, true {, valda} other {}}\",\"listboxLabel\":\"Förslag\",\"selectedAnnouncement\":\"{optionText}, valda\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/tr-TR.json
var $cee59db5f22d3d988bb91129e61c1b$exports = {};
$cee59db5f22d3d988bb91129e61c1b$exports = JSON.parse("{\"buttonLabel\":\"Önerileri göster\",\"countAnnouncement\":\"{optionCount, plural, one {# seçenek} other {# seçenekler}} kullanılabilir.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Girilen grup {groupTitle}, ile {groupCount, plural, one {# seçenek} other {# seçenekler}}. } other {}}{optionText}{isSelected, select, true {, seçildi} other {}}\",\"listboxLabel\":\"Öneriler\",\"selectedAnnouncement\":\"{optionText}, seçildi\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/uk-UA.json
var $dc5dd33b9d2d8eb21d5be848da84b518$exports = {};
$dc5dd33b9d2d8eb21d5be848da84b518$exports = JSON.parse("{\"buttonLabel\":\"Показати пропозиції\",\"countAnnouncement\":\"{optionCount, plural, one {# параметр} other {# параметри(-ів)}} доступно.\",\"focusAnnouncement\":\"{isGroupChange, select, true {Введена група {groupTitle}, з {groupCount, plural, one {# параметр} other {# параметри(-ів)}}. } other {}}{optionText}{isSelected, select, true {, вибрано} other {}}\",\"listboxLabel\":\"Пропозиції\",\"selectedAnnouncement\":\"{optionText}, вибрано\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/zh-CN.json
var $b630874da1ff9a0aebc4f95508e93369$exports = {};
$b630874da1ff9a0aebc4f95508e93369$exports = JSON.parse("{\"buttonLabel\":\"显示建议\",\"countAnnouncement\":\"有 {optionCount, plural, one {# 个选项} other {# 个选项}}可用。\",\"focusAnnouncement\":\"{isGroupChange, select, true {进入了 {groupTitle} 组，其中有 {groupCount, plural, one {# 个选项} other {# 个选项}}. } other {}}{optionText}{isSelected, select, true {, 已选择} other {}}\",\"listboxLabel\":\"建议\",\"selectedAnnouncement\":\"{optionText}, 已选择\"}");
// ASSET: /Users/govett/dev/react-spectrum-v3/packages/@react-aria/combobox/intl/zh-TW.json
var $da4492b53f7bfcb42b50dfd904cc8dc9$exports = {};
$da4492b53f7bfcb42b50dfd904cc8dc9$exports = JSON.parse("{\"buttonLabel\":\"顯示建議\",\"countAnnouncement\":\"{optionCount, plural, one {# 選項} other {# 選項}} 可用。\",\"focusAnnouncement\":\"{isGroupChange, select, true {輸入的群組 {groupTitle}, 有 {groupCount, plural, one {# 選項} other {# 選項}}. } other {}}{optionText}{isSelected, select, true {, 已選取} other {}}\",\"listboxLabel\":\"建議\",\"selectedAnnouncement\":\"{optionText}, 已選取\"}");
// @ts-ignore
const $be26f9e765ed670e72d5b46ae9eff69a$var$intlMessages = {
  "ar-AE": _babelRuntimeHelpersEsmInteropRequireDefault($d78683a0e7fef7294b62031fc43054c6$exports).default,
  "bg-BG": _babelRuntimeHelpersEsmInteropRequireDefault($f98ae6ce2700d4a20c4e60ff9d68eed9$exports).default,
  "cs-CZ": _babelRuntimeHelpersEsmInteropRequireDefault($e18f88738dd7cf3410585df50c4a$exports).default,
  "da-DK": _babelRuntimeHelpersEsmInteropRequireDefault($bc87a01dd17f8e08d9b5f2d709d95ab$exports).default,
  "de-DE": _babelRuntimeHelpersEsmInteropRequireDefault($a2876bc991816077d10a6a3c448303e$exports).default,
  "el-GR": _babelRuntimeHelpersEsmInteropRequireDefault($b2fff3cd75c7191d2e6133d07588de86$exports).default,
  "en-US": _babelRuntimeHelpersEsmInteropRequireDefault($f1639f2754c31f58763b8cb015a6ecd6$exports).default,
  "es-ES": _babelRuntimeHelpersEsmInteropRequireDefault($f0fff8208defd2a975d21faa464abfd$exports).default,
  "et-EE": _babelRuntimeHelpersEsmInteropRequireDefault($c22ee6a7566d1cf5578bf8d4f5c2a6$exports).default,
  "fi-FI": _babelRuntimeHelpersEsmInteropRequireDefault($c0df15e7925dcb04d7c1787b03a5e$exports).default,
  "fr-FR": _babelRuntimeHelpersEsmInteropRequireDefault($b36e7dd5de5dec60de8490c4382ec$exports).default,
  "he-IL": _babelRuntimeHelpersEsmInteropRequireDefault($c0bb0e2e7b387fc7e7e478556f7f7d7$exports).default,
  "hr-HR": _babelRuntimeHelpersEsmInteropRequireDefault($c2ad6267acd85a3c0b84c89b176b1b$exports).default,
  "hu-HU": _babelRuntimeHelpersEsmInteropRequireDefault($b72435e6c734787569d9dde4943b4667$exports).default,
  "it-IT": _babelRuntimeHelpersEsmInteropRequireDefault($ec385ad6d76803c0e13ef3155a37731f$exports).default,
  "ja-JP": _babelRuntimeHelpersEsmInteropRequireDefault($f6712684539b6a07d53079e26b70d9df$exports).default,
  "ko-KR": _babelRuntimeHelpersEsmInteropRequireDefault($c670ae89d3913146f04a762d163$exports).default,
  "lt-LT": _babelRuntimeHelpersEsmInteropRequireDefault($a36750eea27db776e11cc103d172017$exports).default,
  "lv-LV": _babelRuntimeHelpersEsmInteropRequireDefault($e840ed8916bd1f210105d1fb3f88697$exports).default,
  "nb-NO": _babelRuntimeHelpersEsmInteropRequireDefault($a9d36ce32fd5aea219762aea75a1360$exports).default,
  "nl-NL": _babelRuntimeHelpersEsmInteropRequireDefault($c4a143186784391623f115223360f80$exports).default,
  "pl-PL": _babelRuntimeHelpersEsmInteropRequireDefault($a32834a3efc5c6ca32d91bd08$exports).default,
  "pt-BR": _babelRuntimeHelpersEsmInteropRequireDefault($e1b551c29bce2e3141bec9bc8b1ddba1$exports).default,
  "pt-PT": _babelRuntimeHelpersEsmInteropRequireDefault($e10cdad02310abdd1550ffa8b7db9ba$exports).default,
  "ro-RO": _babelRuntimeHelpersEsmInteropRequireDefault($ca354d1ee6bae344a049de3e819052c$exports).default,
  "ru-RU": _babelRuntimeHelpersEsmInteropRequireDefault($f7006032145148654e07a693955fb9c2$exports).default,
  "sk-SK": _babelRuntimeHelpersEsmInteropRequireDefault($c60520bb4ea0d2c95ea4b69dc1046132$exports).default,
  "sl-SI": _babelRuntimeHelpersEsmInteropRequireDefault($fd97d8e462e8521b6da71e6762a5c$exports).default,
  "sr-SP": _babelRuntimeHelpersEsmInteropRequireDefault($b615e8cf79bac7ecb7173b279f2a7$exports).default,
  "sv-SE": _babelRuntimeHelpersEsmInteropRequireDefault($f7c7084482b80ff47daa2804ba64c$exports).default,
  "tr-TR": _babelRuntimeHelpersEsmInteropRequireDefault($cee59db5f22d3d988bb91129e61c1b$exports).default,
  "uk-UA": _babelRuntimeHelpersEsmInteropRequireDefault($dc5dd33b9d2d8eb21d5be848da84b518$exports).default,
  "zh-CN": _babelRuntimeHelpersEsmInteropRequireDefault($b630874da1ff9a0aebc4f95508e93369$exports).default,
  "zh-TW": _babelRuntimeHelpersEsmInteropRequireDefault($da4492b53f7bfcb42b50dfd904cc8dc9$exports).default
};

/**
 * Provides the behavior and accessibility implementation for a combo box component.
 * A combo box combines a text input with a listbox, allowing users to filter a list of options to items matching a query.
 * @param props - Props for the combo box.
 * @param state - State for the select, as returned by `useComboBoxState`.
 */
export function useComboBox(props, state) {
  var _focusedItem$parentKe, _state$selectionManag;

  let {
    buttonRef,
    popoverRef,
    inputRef,
    listBoxRef,
    keyboardDelegate,
    // completionMode = 'suggest',
    shouldFocusWrap,
    isReadOnly,
    isDisabled
  } = props;
  let formatMessage = useMessageFormatter($be26f9e765ed670e72d5b46ae9eff69a$var$intlMessages);
  let {
    menuTriggerProps,
    menuProps
  } = useMenuTrigger({
    type: 'listbox',
    isDisabled: isDisabled || isReadOnly
  }, state, buttonRef); // Set listbox id so it can be used when calling getItemId later

  listData.set(state, {
    id: menuProps.id
  }); // By default, a KeyboardDelegate is provided which uses the DOM to query layout information (e.g. for page up/page down).
  // When virtualized, the layout object will be passed in as a prop and override this.

  let delegate = useMemo(() => keyboardDelegate || new ListKeyboardDelegate(state.collection, state.disabledKeys, listBoxRef), [keyboardDelegate, state.collection, state.disabledKeys, listBoxRef]); // Use useSelectableCollection to get the keyboard handlers to apply to the textfield

  let {
    collectionProps
  } = useSelectableCollection({
    selectionManager: state.selectionManager,
    keyboardDelegate: delegate,
    disallowTypeAhead: true,
    disallowEmptySelection: true,
    shouldFocusWrap,
    ref: inputRef,
    // Prevent item scroll behavior from being applied here, should be handled in the user's Popover + ListBox component
    isVirtualized: true
  }); // For textfield specific keydown operations

  let onKeyDown = e => {
    switch (e.key) {
      case 'Enter':
      case 'Tab':
        // Prevent form submission if menu is open since we may be selecting a option
        if (state.isOpen && e.key === 'Enter') {
          e.preventDefault();
        }

        state.commit();
        break;

      case 'Escape':
        state.revert();
        break;

      case 'ArrowDown':
        state.open('first', 'manual');
        break;

      case 'ArrowUp':
        state.open('last', 'manual');
        break;

      case 'ArrowLeft':
      case 'ArrowRight':
        state.selectionManager.setFocusedKey(null);
        break;
    }
  };

  let onBlur = e => {
    var _popoverRef$current;

    // Ignore blur if focused moved to the button or into the popover.
    if (e.relatedTarget === (buttonRef == null ? void 0 : buttonRef.current) || (_popoverRef$current = popoverRef.current) != null && _popoverRef$current.contains(e.relatedTarget)) {
      return;
    }

    if (props.onBlur) {
      props.onBlur(e);
    }

    state.setFocused(false);
  };

  let onFocus = e => {
    if (state.isFocused) {
      return;
    }

    if (props.onFocus) {
      props.onFocus(e);
    }

    state.setFocused(true);
  };

  let {
    labelProps,
    inputProps,
    descriptionProps,
    errorMessageProps
  } = useTextField(_babelRuntimeHelpersEsmExtends({}, props, {
    onChange: state.setInputValue,
    onKeyDown: !isReadOnly && chain(state.isOpen && collectionProps.onKeyDown, onKeyDown),
    onBlur,
    value: state.inputValue,
    onFocus,
    autoComplete: 'off'
  }), inputRef); // Press handlers for the ComboBox button

  let onPress = e => {
    if (e.pointerType === 'touch') {
      // Focus the input field in case it isn't focused yet
      inputRef.current.focus();
      state.toggle(null, 'manual');
    }
  };

  let onPressStart = e => {
    if (e.pointerType !== 'touch') {
      inputRef.current.focus();
      state.toggle(e.pointerType === 'keyboard' || e.pointerType === 'virtual' ? 'first' : null, 'manual');
    }
  };

  let triggerLabelProps = useLabels({
    id: menuTriggerProps.id,
    'aria-label': formatMessage('buttonLabel'),
    'aria-labelledby': props['aria-labelledby'] || labelProps.id
  });
  let listBoxProps = useLabels({
    id: menuProps.id,
    'aria-label': formatMessage('listboxLabel'),
    'aria-labelledby': props['aria-labelledby'] || labelProps.id
  }); // If a touch happens on direct center of ComboBox input, might be virtual click from iPad so open ComboBox menu

  let lastEventTime = useRef(0);

  let onTouchEnd = e => {
    if (isDisabled || isReadOnly) {
      return;
    } // Sometimes VoiceOver on iOS fires two touchend events in quick succession. Ignore the second one.


    if (e.timeStamp - lastEventTime.current < 500) {
      e.preventDefault();
      inputRef.current.focus();
      return;
    }

    let rect = e.target.getBoundingClientRect();
    let touch = e.changedTouches[0];
    let centerX = Math.ceil(rect.left + .5 * rect.width);
    let centerY = Math.ceil(rect.top + .5 * rect.height);

    if (touch.clientX === centerX && touch.clientY === centerY) {
      e.preventDefault();
      inputRef.current.focus();
      state.toggle(null, 'manual');
      lastEventTime.current = e.timeStamp;
    }
  }; // VoiceOver has issues with announcing aria-activedescendant properly on change
  // (especially on iOS). We use a live region announcer to announce focus changes
  // manually. In addition, section titles are announced when navigating into a new section.


  let focusedItem = state.selectionManager.focusedKey != null && state.isOpen ? state.collection.getItem(state.selectionManager.focusedKey) : undefined;
  let sectionKey = (_focusedItem$parentKe = focusedItem == null ? void 0 : focusedItem.parentKey) != null ? _focusedItem$parentKe : null;
  let itemKey = (_state$selectionManag = state.selectionManager.focusedKey) != null ? _state$selectionManag : null;
  let lastSection = useRef(sectionKey);
  let lastItem = useRef(itemKey);
  useEffect(() => {
    if (isAppleDevice() && focusedItem != null && itemKey !== lastItem.current) {
      let isSelected = state.selectionManager.isSelected(itemKey);
      let section = sectionKey != null ? state.collection.getItem(sectionKey) : null;
      let sectionTitle = (section == null ? void 0 : section['aria-label']) || (typeof (section == null ? void 0 : section.rendered) === 'string' ? section.rendered : '') || '';
      let announcement = formatMessage('focusAnnouncement', {
        isGroupChange: section && sectionKey !== lastSection.current,
        groupTitle: sectionTitle,
        groupCount: section ? [...section.childNodes].length : 0,
        optionText: focusedItem['aria-label'] || focusedItem.textValue || '',
        isSelected
      });
      announce(announcement);
    }

    lastSection.current = sectionKey;
    lastItem.current = itemKey;
  }); // Announce the number of available suggestions when it changes

  let optionCount = getItemCount(state.collection);
  let lastSize = useRef(optionCount);
  let lastOpen = useRef(state.isOpen);
  useEffect(() => {
    // Only announce the number of options available when the menu opens if there is no
    // focused item, otherwise screen readers will typically read e.g. "1 of 6".
    // The exception is VoiceOver since this isn't included in the message above.
    let didOpenWithoutFocusedItem = state.isOpen !== lastOpen.current && (state.selectionManager.focusedKey == null || isAppleDevice());

    if (state.isOpen && (didOpenWithoutFocusedItem || optionCount !== lastSize.current)) {
      let announcement = formatMessage('countAnnouncement', {
        optionCount
      });
      announce(announcement);
    }

    lastSize.current = optionCount;
    lastOpen.current = state.isOpen;
  }); // Announce when a selection occurs for VoiceOver. Other screen readers typically do this automatically.

  let lastSelectedKey = useRef(state.selectedKey);
  useEffect(() => {
    if (isAppleDevice() && state.isFocused && state.selectedItem && state.selectedKey !== lastSelectedKey.current) {
      let optionText = state.selectedItem['aria-label'] || state.selectedItem.textValue || '';
      let announcement = formatMessage('selectedAnnouncement', {
        optionText
      });
      announce(announcement);
    }

    lastSelectedKey.current = state.selectedKey;
  });
  useEffect(() => {
    if (state.isOpen) {
      return ariaHideOutside([inputRef.current, popoverRef.current]);
    }
  }, [state.isOpen, inputRef, popoverRef]);
  return {
    labelProps,
    buttonProps: _babelRuntimeHelpersEsmExtends({}, menuTriggerProps, triggerLabelProps, {
      excludeFromTabOrder: true,
      onPress,
      onPressStart,
      isDisabled: isDisabled || isReadOnly
    }),
    inputProps: mergeProps(inputProps, {
      role: 'combobox',
      'aria-expanded': menuTriggerProps['aria-expanded'],
      'aria-controls': state.isOpen ? menuProps.id : undefined,
      // TODO: readd proper logic for completionMode = complete (aria-autocomplete: both)
      'aria-autocomplete': 'list',
      'aria-activedescendant': focusedItem ? getItemId(state, focusedItem.key) : undefined,
      onTouchEnd,
      // This disable's iOS's autocorrect suggestions, since the combo box provides its own suggestions.
      autoCorrect: 'off',
      // This disable's the macOS Safari spell check auto corrections.
      spellCheck: 'false'
    }),
    listBoxProps: mergeProps(menuProps, listBoxProps, {
      autoFocus: state.focusStrategy,
      shouldUseVirtualFocus: true,
      shouldSelectOnPressUp: true,
      shouldFocusOnHover: true
    }),
    descriptionProps,
    errorMessageProps
  };
}
