// @spotify-internal/adstudio-shared/lib/components/IntlFormatCurrency/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IntlFormatCurrency = function IntlFormatCurrency(_ref) {
  var value = _ref.value,
      country = _ref.country,
      currency = _ref.currency,
      _ref$currencyDisplay = _ref.currencyDisplay,
      currencyDisplay = _ref$currencyDisplay === void 0 ? 'symbol' : _ref$currencyDisplay,
      _ref$decimalPlaces = _ref.decimalPlaces,
      decimalPlaces = _ref$decimalPlaces === void 0 ? 2 : _ref$decimalPlaces,
      _ref$useGrouping = _ref.useGrouping,
      useGrouping = _ref$useGrouping === void 0 ? true : _ref$useGrouping;
  var formatter = new Intl.NumberFormat(country, {
    style: 'currency',
    currency: currency,
    currencyDisplay: currencyDisplay,
    minimumFractionDigits: decimalPlaces,
    useGrouping: useGrouping
  }).format;
  return /*#__PURE__*/_react["default"].createElement("span", null, formatter(value));
};

IntlFormatCurrency.propTypes = {
  country: _propTypes["default"].string.isRequired,
  currency: _propTypes["default"].string.isRequired,
  currencyDisplay: _propTypes["default"].oneOf(['symbol', 'code', 'name']),
  decimalPlaces: _propTypes["default"].number,
  useGrouping: _propTypes["default"].bool,
  value: _propTypes["default"].oneOf([_propTypes["default"].string, _propTypes["default"].number])
};
var _default = IntlFormatCurrency;
exports["default"] = _default;
