// @spotify-internal/adstudio-tape/lib/components/LayoutCommon/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = LayoutCommon;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _tokens = require("../../styles/tokens");

var _Navigation = _interopRequireDefault(require("../Navigation"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var propTypes = {
  /** Should accept the <Alert /> component */
  alert: _propTypes["default"].node,

  /** Can accept any JSX */
  content: _propTypes["default"].node,

  /** Should accept the <Navigation /> component */
  navigation: _propTypes["default"].node,

  /** Should accept the <Footer /> component */
  footer: _propTypes["default"].node,

  /** Used when an <Alert /> is present without any page content */
  noContent: _propTypes["default"].bool,

  /** Reset padding on layout to zero */
  condensed: _propTypes["default"].bool
};
var defaultProps = {
  navigation: /*#__PURE__*/_react["default"].createElement(_Navigation["default"], null),
  noContent: false,
  condensed: false
};

var Aside = _styledComponents["default"].aside.withConfig({
  displayName: "LayoutCommon__Aside",
  componentId: "sc-19gw67q-0"
})(["height:100%;left:0;position:fixed;top:0;width:", ";z-index:", ";"], _tokens.sidebarBaseWidth, _tokens.zIndexFixed);

var Main = _styledComponents["default"].main.withConfig({
  displayName: "LayoutCommon__Main",
  componentId: "sc-19gw67q-1"
})(["background:", ";display:block;margin-left:", ";position:relative;", ""], _encoreFoundation.white, _tokens.sidebarBaseWidth, function (props) {
  return props.hasAlert && (0, _styledComponents.css)(["padding-top:", ";"], _tokens.alertHeight);
});

var Alert = _styledComponents["default"].div.withConfig({
  displayName: "LayoutCommon__Alert",
  componentId: "sc-19gw67q-2"
})(["height:", ";left:", ";position:fixed;right:0;top:0;z-index:", ";"], _tokens.alertHeight, _tokens.sidebarBaseWidth, _tokens.zIndexFixed);

var Body = _styledComponents["default"].div.withConfig({
  displayName: "LayoutCommon__Body",
  componentId: "sc-19gw67q-3"
})(["display:flex;flex-direction:column;max-width:calc(", " - ", ");min-height:100vh;", ";"], _encoreFoundation.screenXlMin, _tokens.sidebarBaseWidth, function (props) {
  return !props.condensed && (0, _styledComponents.css)(["@media (max-width:", "){padding-left:", ";padding-right:", ";}@media (min-width:", "){padding-left:", ";padding-right:", ";}"], _encoreFoundation.screenXsMax, _encoreFoundation.spacer24, _encoreFoundation.spacer24, _encoreFoundation.screenSmMin, _encoreFoundation.spacer32, _encoreFoundation.spacer32);
});

function LayoutCommon(props) {
  var alert = props.alert,
      navigation = props.navigation,
      content = props.content,
      footer = props.footer,
      noContent = props.noContent,
      condensed = props.condensed;
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(Aside, null, " ", navigation), /*#__PURE__*/_react["default"].createElement(Main, {
    hasAlert: alert && !noContent
  }, alert && /*#__PURE__*/_react["default"].createElement(Alert, null, alert), /*#__PURE__*/_react["default"].createElement(Body, {
    condensed: condensed
  }, content, footer)));
}

LayoutCommon.propTypes = propTypes;
LayoutCommon.defaultProps = defaultProps;
