// @spotify-internal/betamax-sdk-desktop/esm/src/tracking/error/ErrorLogger.js
var ErrorLogger = (function () {
    function ErrorLogger(logger) {
        this.logger = logger;
    }
    ErrorLogger.prototype.onPlaybackCreated = function (playbackIdentity) {
        if (playbackIdentity.metadata) {
            var metadata = playbackIdentity.metadata;
            if (metadata['media.manifest_id']) {
                this.fileId = metadata['media.manifest_id'];
            }
            if (metadata.endvideo_play_track) {
                this.uri = metadata.endvideo_play_track;
            }
            if (metadata.endvideo_playback_id) {
                this.playbackId = metadata.endvideo_playback_id;
            }
        }
    };
    ErrorLogger.prototype.onRecoverableError = function (error, _position, timestamp) {
        this.logErrorMessage(timestamp, error);
    };
    ErrorLogger.prototype.onFatalError = function (error, _position, timestamp) {
        this.logErrorMessage(timestamp, error);
    };
    ErrorLogger.prototype.logErrorMessage = function (timestamp, error) {
        var errorMessage = {
            file_id: this.fileId,
            uri: this.uri,
            playback_id: this.playbackId,
            local_time: timestamp,
            name: error.kind,
            message: JSON.stringify({
                kind: error.kind,
                type: error ? error.data.type : null,
                detail: error ? error.data.detail : null,
            }),
        };
        this.logger.log(errorMessage);
    };
    ErrorLogger.prototype.onPlaybackEnded = function () { };
    ErrorLogger.prototype.onAudioFormatChanged = function () { };
    ErrorLogger.prototype.onBuffering = function () { };
    ErrorLogger.prototype.onBytesTransferred = function () { };
    ErrorLogger.prototype.onDroppedVideoFrames = function () { };
    ErrorLogger.prototype.onDurationChanged = function () { };
    ErrorLogger.prototype.onEncryptionKeyLoaded = function () { };
    ErrorLogger.prototype.onEncryptionKeyLoading = function () { };
    ErrorLogger.prototype.onEndingPlayback = function () { };
    ErrorLogger.prototype.onFullscreenChanged = function () { };
    ErrorLogger.prototype.onManifestLoaded = function () { };
    ErrorLogger.prototype.onManifestLoading = function () { };
    ErrorLogger.prototype.onPaused = function () { };
    ErrorLogger.prototype.onReady = function () { };
    ErrorLogger.prototype.onResumed = function () { };
    ErrorLogger.prototype.onSeeking = function () { };
    ErrorLogger.prototype.onSubtitleChanged = function () { };
    ErrorLogger.prototype.onVideoFormatChanged = function () { };
    ErrorLogger.prototype.onVolumeChanged = function () { };
    return ErrorLogger;
}());
export default ErrorLogger;
