// @spotify-internal/encore-web/es/components/Button/Button.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import styled from 'styled-components';
import { buttonStyle } from "../../styles/mixins/buttons";
// eslint-disable-next-line prefer-arrow-callback
export default styled( /*#__PURE__*/React.forwardRef(function Button(_ref, ref) {
  var _ref$component = _ref.component,
      Component = _ref$component === void 0 ? 'button' : _ref$component,
      variantRules = _ref.variantRules,
      buttonSize = _ref.buttonSize,
      condensed = _ref.condensed,
      hover = _ref.hover,
      active = _ref.active,
      focus = _ref.focus,
      tertiary = _ref.tertiary,
      isUsingKeyboard = _ref.isUsingKeyboard,
      props = _objectWithoutProperties(_ref, ["component", "variantRules", "buttonSize", "condensed", "hover", "active", "focus", "tertiary", "isUsingKeyboard"]);

  return /*#__PURE__*/React.createElement(Component, Object.assign({}, props, {
    ref: ref
  }));
})).withConfig({
  displayName: "Button",
  componentId: "oyfj48-0"
})(["", ""], function (props) {
  return buttonStyle({
    buttonSize: props.buttonSize,
    condensed: props.condensed,
    variantRules: props.variantRules,
    theme: props.theme,
    focus: props.focus,
    tertiary: props.tertiary,
    isUsingKeyboard: props.isUsingKeyboard
  });
});
