// @spotify-internal/encore-web/es/components/Button/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { buttonSizes } from "../../styles/mixins/buttons";
import ButtonElement from "./Button";
var ButtonComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var component = _ref.component,
      _ref$buttonSize = _ref.buttonSize,
      buttonSize = _ref$buttonSize === void 0 ? 'md' : _ref$buttonSize,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      elementProps = _objectWithoutProperties(_ref, ["component", "buttonSize", "theme"]);

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  var propOverrides = {};

  if (!component && elementProps.href) {
    propOverrides.component = 'a';
  }

  if (elementProps.href && elementProps.target === '_blank') {
    propOverrides.rel = 'noopener noreferrer';
  }

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(ButtonElement, Object.assign({
    buttonSize: buttonSize,
    component: component
  }, elementProps, propOverrides, {
    isUsingKeyboard: isUsingKeyboard,
    ref: ref
  })));
});

var buttonHelpers = _objectSpread(_objectSpread({}, buttonSizes), {}, {
  darkTheme: {
    button: 'dark'
  },
  displayName: 'Button'
});
/**
* **Button** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Button) |   
*
* null
*
* @example
* No example for this component
*
*/


export var Button = Object.assign(ButtonComponent, buttonHelpers);
export default Button;
