// @spotify-internal/ubi-sdk-s4a-web-music/WebMusicEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebMusicEventFactory = exports.createWebMusicEventFactory = void 0;
var SPECIFICATION_ID = "web-music";
var APP = "s4a";
var SPECIFICATION_VERSION = "4.1.1";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebMusicEventFactory(options) {
    return new WebMusicEventFactory(options);
}
exports.createWebMusicEventFactory = createWebMusicEventFactory;
/**
 * The spec for the Music tab in S4A
 */
var WebMusicEventFactory = /** @class */ (function () {
    function WebMusicEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of HeaderEventFactory
    */
    WebMusicEventFactory.prototype.headerFactory = function () {
        return new WebMusicEventFactory.HeaderEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @param uri of the SearchBarEventFactory
    * @returns A new instance of SearchBarEventFactory
    */
    WebMusicEventFactory.prototype.searchBarFactory = function (data) {
        return new WebMusicEventFactory.SearchBarEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation,
            data: data
        });
    };
    /**
    * @returns A new instance of DateFilterEventFactory
    */
    WebMusicEventFactory.prototype.dateFilterFactory = function () {
        return new WebMusicEventFactory.DateFilterEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @param uri of the DownloadCsvButtonEventFactory
    * @returns A new instance of DownloadCsvButtonEventFactory
    */
    WebMusicEventFactory.prototype.downloadCsvButtonFactory = function (data) {
        return new WebMusicEventFactory.DownloadCsvButtonEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation,
            data: data
        });
    };
    /**
    * @returns A new instance of SongsTabEventFactory
    */
    WebMusicEventFactory.prototype.songsTabFactory = function () {
        return new WebMusicEventFactory.SongsTabEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ReleasesTabEventFactory
    */
    WebMusicEventFactory.prototype.releasesTabFactory = function () {
        return new WebMusicEventFactory.ReleasesTabEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of PlaylistsTabEventFactory
    */
    WebMusicEventFactory.prototype.playlistsTabFactory = function () {
        return new WebMusicEventFactory.PlaylistsTabEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of UpcomingTabEventFactory
    */
    WebMusicEventFactory.prototype.upcomingTabFactory = function () {
        return new WebMusicEventFactory.UpcomingTabEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebMusicEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebMusicEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebMusicEventFactory;
}());
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create HeaderEventFactory class with path
     * @param path to this node
     */
    var HeaderEventFactory = /** @class */ (function () {
        function HeaderEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "header"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of SongsButtonEventFactory
        */
        HeaderEventFactory.prototype.songsButtonFactory = function () {
            return new WebMusicEventFactory.HeaderEventFactory.SongsButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ReleasesButtonEventFactory
        */
        HeaderEventFactory.prototype.releasesButtonFactory = function () {
            return new WebMusicEventFactory.HeaderEventFactory.ReleasesButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of PlaylistsButtonEventFactory
        */
        HeaderEventFactory.prototype.playlistsButtonFactory = function () {
            return new WebMusicEventFactory.HeaderEventFactory.PlaylistsButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of UpcomingButtonEventFactory
        */
        HeaderEventFactory.prototype.upcomingButtonFactory = function () {
            return new WebMusicEventFactory.HeaderEventFactory.UpcomingButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the AnnouncementModalEventFactory
        * @returns A new instance of AnnouncementModalEventFactory
        */
        HeaderEventFactory.prototype.announcementModalFactory = function (data) {
            return new WebMusicEventFactory.HeaderEventFactory.AnnouncementModalEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        HeaderEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        HeaderEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return HeaderEventFactory;
    }());
    WebMusicEventFactory.HeaderEventFactory = HeaderEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create SearchBarEventFactory class with path
     * @param path to this node
     */
    var SearchBarEventFactory = /** @class */ (function () {
        function SearchBarEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat(__assign(__assign({}, data), { name: "search_bar" }));
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        SearchBarEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SearchBarEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for key_stroke search action.
         */
        SearchBarEventFactory.prototype.keyStrokeSearch = function () {
            var data = {
                actionName: "search",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "key_stroke",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SearchBarEventFactory;
    }());
    WebMusicEventFactory.SearchBarEventFactory = SearchBarEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create DateFilterEventFactory class with path
     * @param path to this node
     */
    var DateFilterEventFactory = /** @class */ (function () {
        function DateFilterEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "date_filter"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the OptionEventFactory
        * @param uri of the OptionEventFactory
        * @returns A new instance of OptionEventFactory
        */
        DateFilterEventFactory.prototype.optionFactory = function (data) {
            return new WebMusicEventFactory.DateFilterEventFactory.OptionEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        DateFilterEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        DateFilterEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        DateFilterEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return DateFilterEventFactory;
    }());
    WebMusicEventFactory.DateFilterEventFactory = DateFilterEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create DownloadCsvButtonEventFactory class with path
     * @param path to this node
     */
    var DownloadCsvButtonEventFactory = /** @class */ (function () {
        function DownloadCsvButtonEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat(__assign(__assign({}, data), { name: "download_csv_button" }));
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        DownloadCsvButtonEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        DownloadCsvButtonEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit download_file action.
         */
        DownloadCsvButtonEventFactory.prototype.hitDownloadFile = function () {
            var data = {
                actionName: "download_file",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        /**
         * Create impression for DownloadCsvButtonEventFactory.
         */
        DownloadCsvButtonEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return DownloadCsvButtonEventFactory;
    }());
    WebMusicEventFactory.DownloadCsvButtonEventFactory = DownloadCsvButtonEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create SongsTabEventFactory class with path
     * @param path to this node
     */
    var SongsTabEventFactory = /** @class */ (function () {
        function SongsTabEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "songs_tab"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the TableHeaderEventFactory
        * @param reason of the TableHeaderEventFactory
        * @returns A new instance of TableHeaderEventFactory
        */
        SongsTabEventFactory.prototype.tableHeaderFactory = function (data) {
            return new WebMusicEventFactory.SongsTabEventFactory.TableHeaderEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @param position of the SongRowEventFactory
        * @param uri of the SongRowEventFactory
        * @returns A new instance of SongRowEventFactory
        */
        SongsTabEventFactory.prototype.songRowFactory = function (data) {
            return new WebMusicEventFactory.SongsTabEventFactory.SongRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        SongsTabEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SongsTabEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for SongsTabEventFactory.
         */
        SongsTabEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return SongsTabEventFactory;
    }());
    WebMusicEventFactory.SongsTabEventFactory = SongsTabEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create ReleasesTabEventFactory class with path
     * @param path to this node
     */
    var ReleasesTabEventFactory = /** @class */ (function () {
        function ReleasesTabEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "releases_tab"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of TableHeaderEventFactory
        */
        ReleasesTabEventFactory.prototype.tableHeaderFactory = function () {
            return new WebMusicEventFactory.ReleasesTabEventFactory.TableHeaderEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param position of the ReleaseRowEventFactory
        * @param uri of the ReleaseRowEventFactory
        * @returns A new instance of ReleaseRowEventFactory
        */
        ReleasesTabEventFactory.prototype.releaseRowFactory = function (data) {
            return new WebMusicEventFactory.ReleasesTabEventFactory.ReleaseRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @returns A new instance of ErrorStateEventFactory
        */
        ReleasesTabEventFactory.prototype.errorStateFactory = function () {
            return new WebMusicEventFactory.ReleasesTabEventFactory.ErrorStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        ReleasesTabEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ReleasesTabEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for ReleasesTabEventFactory.
         */
        ReleasesTabEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ReleasesTabEventFactory;
    }());
    WebMusicEventFactory.ReleasesTabEventFactory = ReleasesTabEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create PlaylistsTabEventFactory class with path
     * @param path to this node
     */
    var PlaylistsTabEventFactory = /** @class */ (function () {
        function PlaylistsTabEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "playlists_tab"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the PlaylistCategoryEventFactory
        * @returns A new instance of PlaylistCategoryEventFactory
        */
        PlaylistsTabEventFactory.prototype.playlistCategoryFactory = function (data) {
            return new WebMusicEventFactory.PlaylistsTabEventFactory.PlaylistCategoryEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        PlaylistsTabEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        PlaylistsTabEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for PlaylistsTabEventFactory.
         */
        PlaylistsTabEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return PlaylistsTabEventFactory;
    }());
    WebMusicEventFactory.PlaylistsTabEventFactory = PlaylistsTabEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    /**
     * Create UpcomingTabEventFactory class with path
     * @param path to this node
     */
    var UpcomingTabEventFactory = /** @class */ (function () {
        function UpcomingTabEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "upcoming_tab"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of TableHeaderEventFactory
        */
        UpcomingTabEventFactory.prototype.tableHeaderFactory = function () {
            return new WebMusicEventFactory.UpcomingTabEventFactory.TableHeaderEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the UpcomingReleaseRowEventFactory
        * @param uri of the UpcomingReleaseRowEventFactory
        * @returns A new instance of UpcomingReleaseRowEventFactory
        */
        UpcomingTabEventFactory.prototype.upcomingReleaseRowFactory = function (data) {
            return new WebMusicEventFactory.UpcomingTabEventFactory.UpcomingReleaseRowEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        UpcomingTabEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        UpcomingTabEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for UpcomingTabEventFactory.
         */
        UpcomingTabEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return UpcomingTabEventFactory;
    }());
    WebMusicEventFactory.UpcomingTabEventFactory = UpcomingTabEventFactory;
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var HeaderEventFactory;
    (function (HeaderEventFactory) {
        /**
         * Create SongsButtonEventFactory class with path
         * @param path to this node
         */
        var SongsButtonEventFactory = /** @class */ (function () {
            function SongsButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "songs_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SongsButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SongsButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            SongsButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SongsButtonEventFactory;
        }());
        HeaderEventFactory.SongsButtonEventFactory = SongsButtonEventFactory;
    })(HeaderEventFactory = WebMusicEventFactory.HeaderEventFactory || (WebMusicEventFactory.HeaderEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var HeaderEventFactory;
    (function (HeaderEventFactory) {
        /**
         * Create ReleasesButtonEventFactory class with path
         * @param path to this node
         */
        var ReleasesButtonEventFactory = /** @class */ (function () {
            function ReleasesButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "releases_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ReleasesButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ReleasesButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            ReleasesButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ReleasesButtonEventFactory;
        }());
        HeaderEventFactory.ReleasesButtonEventFactory = ReleasesButtonEventFactory;
    })(HeaderEventFactory = WebMusicEventFactory.HeaderEventFactory || (WebMusicEventFactory.HeaderEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var HeaderEventFactory;
    (function (HeaderEventFactory) {
        /**
         * Create PlaylistsButtonEventFactory class with path
         * @param path to this node
         */
        var PlaylistsButtonEventFactory = /** @class */ (function () {
            function PlaylistsButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "playlists_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            PlaylistsButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            PlaylistsButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            PlaylistsButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return PlaylistsButtonEventFactory;
        }());
        HeaderEventFactory.PlaylistsButtonEventFactory = PlaylistsButtonEventFactory;
    })(HeaderEventFactory = WebMusicEventFactory.HeaderEventFactory || (WebMusicEventFactory.HeaderEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var HeaderEventFactory;
    (function (HeaderEventFactory) {
        /**
         * Create UpcomingButtonEventFactory class with path
         * @param path to this node
         */
        var UpcomingButtonEventFactory = /** @class */ (function () {
            function UpcomingButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "upcoming_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            UpcomingButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            UpcomingButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            UpcomingButtonEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return UpcomingButtonEventFactory;
        }());
        HeaderEventFactory.UpcomingButtonEventFactory = UpcomingButtonEventFactory;
    })(HeaderEventFactory = WebMusicEventFactory.HeaderEventFactory || (WebMusicEventFactory.HeaderEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var HeaderEventFactory;
    (function (HeaderEventFactory) {
        /**
         * Create AnnouncementModalEventFactory class with path
         * @param path to this node
         */
        var AnnouncementModalEventFactory = /** @class */ (function () {
            function AnnouncementModalEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "announcement_modal" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AnnouncementModalEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AnnouncementModalEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for AnnouncementModalEventFactory.
             */
            AnnouncementModalEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AnnouncementModalEventFactory;
        }());
        HeaderEventFactory.AnnouncementModalEventFactory = AnnouncementModalEventFactory;
    })(HeaderEventFactory = WebMusicEventFactory.HeaderEventFactory || (WebMusicEventFactory.HeaderEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var DateFilterEventFactory;
    (function (DateFilterEventFactory) {
        /**
         * Create OptionEventFactory class with path
         * @param path to this node
         */
        var OptionEventFactory = /** @class */ (function () {
            function OptionEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "option" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            OptionEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            OptionEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit filter action.
             */
            OptionEventFactory.prototype.hitFilter = function () {
                var data = {
                    actionName: "filter",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return OptionEventFactory;
        }());
        DateFilterEventFactory.OptionEventFactory = OptionEventFactory;
    })(DateFilterEventFactory = WebMusicEventFactory.DateFilterEventFactory || (WebMusicEventFactory.DateFilterEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var SongsTabEventFactory;
    (function (SongsTabEventFactory) {
        /**
         * Create TableHeaderEventFactory class with path
         * @param path to this node
         */
        var TableHeaderEventFactory = /** @class */ (function () {
            function TableHeaderEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "table_header" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TableHeaderEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit sort action.
             */
            TableHeaderEventFactory.prototype.hitSort = function () {
                var data = {
                    actionName: "sort",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TableHeaderEventFactory;
        }());
        SongsTabEventFactory.TableHeaderEventFactory = TableHeaderEventFactory;
    })(SongsTabEventFactory = WebMusicEventFactory.SongsTabEventFactory || (WebMusicEventFactory.SongsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var SongsTabEventFactory;
    (function (SongsTabEventFactory) {
        /**
         * Create SongRowEventFactory class with path
         * @param path to this node
         */
        var SongRowEventFactory = /** @class */ (function () {
            function SongRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "song_row" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SongRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SongRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            SongRowEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SongRowEventFactory;
        }());
        SongsTabEventFactory.SongRowEventFactory = SongRowEventFactory;
    })(SongsTabEventFactory = WebMusicEventFactory.SongsTabEventFactory || (WebMusicEventFactory.SongsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var ReleasesTabEventFactory;
    (function (ReleasesTabEventFactory) {
        /**
         * Create TableHeaderEventFactory class with path
         * @param path to this node
         */
        var TableHeaderEventFactory = /** @class */ (function () {
            function TableHeaderEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "table_header"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TableHeaderEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return TableHeaderEventFactory;
        }());
        ReleasesTabEventFactory.TableHeaderEventFactory = TableHeaderEventFactory;
    })(ReleasesTabEventFactory = WebMusicEventFactory.ReleasesTabEventFactory || (WebMusicEventFactory.ReleasesTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var ReleasesTabEventFactory;
    (function (ReleasesTabEventFactory) {
        /**
         * Create ReleaseRowEventFactory class with path
         * @param path to this node
         */
        var ReleaseRowEventFactory = /** @class */ (function () {
            function ReleaseRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "release_row" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param position of the SongRowEventFactory
            * @param uri of the SongRowEventFactory
            * @returns A new instance of SongRowEventFactory
            */
            ReleaseRowEventFactory.prototype.songRowFactory = function (data) {
                return new WebMusicEventFactory.ReleasesTabEventFactory.ReleaseRowEventFactory.SongRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            ReleaseRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ReleaseRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_navigate action.
           * @param destination the destination field
             */
            ReleaseRowEventFactory.prototype.hitUiNavigate = function (_a) {
                var destination = _a.destination;
                var data = {
                    actionParameterNames: ['destination'],
                    actionParameterValues: [destination.toString()],
                    actionName: "ui_navigate",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create impression for ReleaseRowEventFactory.
             */
            ReleaseRowEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ReleaseRowEventFactory;
        }());
        ReleasesTabEventFactory.ReleaseRowEventFactory = ReleaseRowEventFactory;
    })(ReleasesTabEventFactory = WebMusicEventFactory.ReleasesTabEventFactory || (WebMusicEventFactory.ReleasesTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var ReleasesTabEventFactory;
    (function (ReleasesTabEventFactory) {
        /**
         * Create ErrorStateEventFactory class with path
         * @param path to this node
         */
        var ErrorStateEventFactory = /** @class */ (function () {
            function ErrorStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "error_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ErrorStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ErrorStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ErrorStateEventFactory.
             */
            ErrorStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ErrorStateEventFactory;
        }());
        ReleasesTabEventFactory.ErrorStateEventFactory = ErrorStateEventFactory;
    })(ReleasesTabEventFactory = WebMusicEventFactory.ReleasesTabEventFactory || (WebMusicEventFactory.ReleasesTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var PlaylistsTabEventFactory;
    (function (PlaylistsTabEventFactory) {
        /**
         * Create PlaylistCategoryEventFactory class with path
         * @param path to this node
         */
        var PlaylistCategoryEventFactory = /** @class */ (function () {
            function PlaylistCategoryEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "playlist_category" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the TableHeaderEventFactory
            * @param reason of the TableHeaderEventFactory
            * @returns A new instance of TableHeaderEventFactory
            */
            PlaylistCategoryEventFactory.prototype.tableHeaderFactory = function (data) {
                return new WebMusicEventFactory.PlaylistsTabEventFactory.PlaylistCategoryEventFactory.TableHeaderEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            /**
            * @param id of the PlaylistRowEventFactory
            * @param uri of the PlaylistRowEventFactory
            * @returns A new instance of PlaylistRowEventFactory
            */
            PlaylistCategoryEventFactory.prototype.playlistRowFactory = function (data) {
                return new WebMusicEventFactory.PlaylistsTabEventFactory.PlaylistCategoryEventFactory.PlaylistRowEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            /**
            * @returns A new instance of ShowMoreButtonEventFactory
            */
            PlaylistCategoryEventFactory.prototype.showMoreButtonFactory = function () {
                return new WebMusicEventFactory.PlaylistsTabEventFactory.PlaylistCategoryEventFactory.ShowMoreButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of ShowLessButtonEventFactory
            */
            PlaylistCategoryEventFactory.prototype.showLessButtonFactory = function () {
                return new WebMusicEventFactory.PlaylistsTabEventFactory.PlaylistCategoryEventFactory.ShowLessButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            PlaylistCategoryEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            PlaylistCategoryEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for PlaylistCategoryEventFactory.
             */
            PlaylistCategoryEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return PlaylistCategoryEventFactory;
        }());
        PlaylistsTabEventFactory.PlaylistCategoryEventFactory = PlaylistCategoryEventFactory;
    })(PlaylistsTabEventFactory = WebMusicEventFactory.PlaylistsTabEventFactory || (WebMusicEventFactory.PlaylistsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var UpcomingTabEventFactory;
    (function (UpcomingTabEventFactory) {
        /**
         * Create TableHeaderEventFactory class with path
         * @param path to this node
         */
        var TableHeaderEventFactory = /** @class */ (function () {
            function TableHeaderEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "table_header"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TableHeaderEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TableHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return TableHeaderEventFactory;
        }());
        UpcomingTabEventFactory.TableHeaderEventFactory = TableHeaderEventFactory;
    })(UpcomingTabEventFactory = WebMusicEventFactory.UpcomingTabEventFactory || (WebMusicEventFactory.UpcomingTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var UpcomingTabEventFactory;
    (function (UpcomingTabEventFactory) {
        /**
         * Create UpcomingReleaseRowEventFactory class with path
         * @param path to this node
         */
        var UpcomingReleaseRowEventFactory = /** @class */ (function () {
            function UpcomingReleaseRowEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "upcoming_release_row" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the ManageButtonEventFactory
            * @param uri of the ManageButtonEventFactory
            * @returns A new instance of ManageButtonEventFactory
            */
            UpcomingReleaseRowEventFactory.prototype.manageButtonFactory = function (data) {
                return new WebMusicEventFactory.UpcomingTabEventFactory.UpcomingReleaseRowEventFactory.ManageButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            UpcomingReleaseRowEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            UpcomingReleaseRowEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit no_action action.
             */
            UpcomingReleaseRowEventFactory.prototype.hitNoAction = function () {
                var data = {
                    actionName: "no_action",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            /**
             * Create impression for UpcomingReleaseRowEventFactory.
             */
            UpcomingReleaseRowEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return UpcomingReleaseRowEventFactory;
        }());
        UpcomingTabEventFactory.UpcomingReleaseRowEventFactory = UpcomingReleaseRowEventFactory;
    })(UpcomingTabEventFactory = WebMusicEventFactory.UpcomingTabEventFactory || (WebMusicEventFactory.UpcomingTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var ReleasesTabEventFactory;
    (function (ReleasesTabEventFactory) {
        var ReleaseRowEventFactory;
        (function (ReleaseRowEventFactory) {
            /**
             * Create SongRowEventFactory class with path
             * @param path to this node
             */
            var SongRowEventFactory = /** @class */ (function () {
                function SongRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "song_row" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                SongRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                SongRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_navigate action.
               * @param destination the destination field
                 */
                SongRowEventFactory.prototype.hitUiNavigate = function (_a) {
                    var destination = _a.destination;
                    var data = {
                        actionParameterNames: ['destination'],
                        actionParameterValues: [destination.toString()],
                        actionName: "ui_navigate",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return SongRowEventFactory;
            }());
            ReleaseRowEventFactory.SongRowEventFactory = SongRowEventFactory;
        })(ReleaseRowEventFactory = ReleasesTabEventFactory.ReleaseRowEventFactory || (ReleasesTabEventFactory.ReleaseRowEventFactory = {}));
    })(ReleasesTabEventFactory = WebMusicEventFactory.ReleasesTabEventFactory || (WebMusicEventFactory.ReleasesTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var PlaylistsTabEventFactory;
    (function (PlaylistsTabEventFactory) {
        var PlaylistCategoryEventFactory;
        (function (PlaylistCategoryEventFactory) {
            /**
             * Create TableHeaderEventFactory class with path
             * @param path to this node
             */
            var TableHeaderEventFactory = /** @class */ (function () {
                function TableHeaderEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "table_header" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TableHeaderEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TableHeaderEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit sort action.
                 */
                TableHeaderEventFactory.prototype.hitSort = function () {
                    var data = {
                        actionName: "sort",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TableHeaderEventFactory;
            }());
            PlaylistCategoryEventFactory.TableHeaderEventFactory = TableHeaderEventFactory;
        })(PlaylistCategoryEventFactory = PlaylistsTabEventFactory.PlaylistCategoryEventFactory || (PlaylistsTabEventFactory.PlaylistCategoryEventFactory = {}));
    })(PlaylistsTabEventFactory = WebMusicEventFactory.PlaylistsTabEventFactory || (WebMusicEventFactory.PlaylistsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var PlaylistsTabEventFactory;
    (function (PlaylistsTabEventFactory) {
        var PlaylistCategoryEventFactory;
        (function (PlaylistCategoryEventFactory) {
            /**
             * Create PlaylistRowEventFactory class with path
             * @param path to this node
             */
            var PlaylistRowEventFactory = /** @class */ (function () {
                function PlaylistRowEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "playlist_row" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                PlaylistRowEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                PlaylistRowEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create impression for PlaylistRowEventFactory.
                 */
                PlaylistRowEventFactory.prototype.impression = function () {
                    var data = {
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        specificationVersion: SPECIFICATION_VERSION,
                        specificationMode: SPECIFICATION_MODE,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return PlaylistRowEventFactory;
            }());
            PlaylistCategoryEventFactory.PlaylistRowEventFactory = PlaylistRowEventFactory;
        })(PlaylistCategoryEventFactory = PlaylistsTabEventFactory.PlaylistCategoryEventFactory || (PlaylistsTabEventFactory.PlaylistCategoryEventFactory = {}));
    })(PlaylistsTabEventFactory = WebMusicEventFactory.PlaylistsTabEventFactory || (WebMusicEventFactory.PlaylistsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var PlaylistsTabEventFactory;
    (function (PlaylistsTabEventFactory) {
        var PlaylistCategoryEventFactory;
        (function (PlaylistCategoryEventFactory) {
            /**
             * Create ShowMoreButtonEventFactory class with path
             * @param path to this node
             */
            var ShowMoreButtonEventFactory = /** @class */ (function () {
                function ShowMoreButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "show_more_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ShowMoreButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ShowMoreButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                ShowMoreButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ShowMoreButtonEventFactory;
            }());
            PlaylistCategoryEventFactory.ShowMoreButtonEventFactory = ShowMoreButtonEventFactory;
        })(PlaylistCategoryEventFactory = PlaylistsTabEventFactory.PlaylistCategoryEventFactory || (PlaylistsTabEventFactory.PlaylistCategoryEventFactory = {}));
    })(PlaylistsTabEventFactory = WebMusicEventFactory.PlaylistsTabEventFactory || (WebMusicEventFactory.PlaylistsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var PlaylistsTabEventFactory;
    (function (PlaylistsTabEventFactory) {
        var PlaylistCategoryEventFactory;
        (function (PlaylistCategoryEventFactory) {
            /**
             * Create ShowLessButtonEventFactory class with path
             * @param path to this node
             */
            var ShowLessButtonEventFactory = /** @class */ (function () {
                function ShowLessButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "show_less_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ShowLessButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ShowLessButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_hide action.
                 */
                ShowLessButtonEventFactory.prototype.hitUiHide = function () {
                    var data = {
                        actionName: "ui_hide",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ShowLessButtonEventFactory;
            }());
            PlaylistCategoryEventFactory.ShowLessButtonEventFactory = ShowLessButtonEventFactory;
        })(PlaylistCategoryEventFactory = PlaylistsTabEventFactory.PlaylistCategoryEventFactory || (PlaylistsTabEventFactory.PlaylistCategoryEventFactory = {}));
    })(PlaylistsTabEventFactory = WebMusicEventFactory.PlaylistsTabEventFactory || (WebMusicEventFactory.PlaylistsTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
(function (WebMusicEventFactory) {
    var UpcomingTabEventFactory;
    (function (UpcomingTabEventFactory) {
        var UpcomingReleaseRowEventFactory;
        (function (UpcomingReleaseRowEventFactory) {
            /**
             * Create ManageButtonEventFactory class with path
             * @param path to this node
             */
            var ManageButtonEventFactory = /** @class */ (function () {
                function ManageButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "manage_button" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ManageButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ManageButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                ManageButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ManageButtonEventFactory;
            }());
            UpcomingReleaseRowEventFactory.ManageButtonEventFactory = ManageButtonEventFactory;
        })(UpcomingReleaseRowEventFactory = UpcomingTabEventFactory.UpcomingReleaseRowEventFactory || (UpcomingTabEventFactory.UpcomingReleaseRowEventFactory = {}));
    })(UpcomingTabEventFactory = WebMusicEventFactory.UpcomingTabEventFactory || (WebMusicEventFactory.UpcomingTabEventFactory = {}));
})(WebMusicEventFactory = exports.WebMusicEventFactory || (exports.WebMusicEventFactory = {}));
exports.WebMusicEventFactory = WebMusicEventFactory;
