// @spotify-internal/adstudio-tape/lib/components/PlaybackScrubber/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.thumbDiameter = exports.thumbRadius = exports.thumbColor = exports.thumbBorderWidth = exports.hoverPlayedFill = exports.playedFill = exports.unplayedFill = exports.trackRadius = exports.trackHeight = exports.trackBorderWidth = void 0;

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var trackBorderWidth = 1;
exports.trackBorderWidth = trackBorderWidth;
var trackHeight = 4;
exports.trackHeight = trackHeight;
var trackRadius = 4;
exports.trackRadius = trackRadius;
var unplayedFill = _encoreFoundation.gray90;
exports.unplayedFill = unplayedFill;
var playedFill = _encoreFoundation.gray40;
exports.playedFill = playedFill;
var hoverPlayedFill = _encoreFoundation.spotifyGreen;
exports.hoverPlayedFill = hoverPlayedFill;
var thumbBorderWidth = 1;
exports.thumbBorderWidth = thumbBorderWidth;
var thumbColor = _encoreFoundation.gray90;
exports.thumbColor = thumbColor;
var thumbRadius = 8 - trackBorderWidth * 2;
exports.thumbRadius = thumbRadius;
var thumbDiameter = thumbRadius * 2;
exports.thumbDiameter = thumbDiameter;
