// @spotify-internal/adstudio-shared/lib/components/Slider/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Slider = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _constants = require("./constants");

var _DEFAULTS$LABEL, _DEFAULTS$LABEL2, _DEFAULTS$LABEL3, _DEFAULTS$SLIDER, _DEFAULTS$SLIDER2, _DEFAULTS$SLIDER3, _DEFAULTS$SLIDER4, _DEFAULTS$SLIDER5, _DEFAULTS$SLIDER6, _DEFAULTS$VALUES4, _DEFAULTS$VALUES5, _DEFAULTS$VALUES6, _DEFAULTS$VALUES7, _DEFAULTS$VALUES8;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var calcLabelPos = function calcLabelPos(_ref) {
  var handle = _ref.handle,
      index = _ref.index,
      lastIndex = _ref.lastIndex,
      listWidth = _ref.listWidth;
  var itemWidth = (0, _styledComponents.css)(["calc((", " - ", ") / ", ")"], listWidth, handle, lastIndex);

  switch (index) {
    case 0:
      return (0, _styledComponents.css)(["text-align:left;width:calc(", " / 2 + ", " / 2);"], itemWidth, handle);

    case lastIndex:
      return (0, _styledComponents.css)(["text-align:right;width:calc(", " / 2 + ", " / 2);"], itemWidth, handle);

    default:
      return (0, _styledComponents.css)(["text-align:center;width:", "};"], itemWidth);
  }
};

var sliderDefaults = (0, _styledComponents.css)(["width:", ";height:", ";border-radius:calc( ", " * 0.6 );cursor:", ";background:", ";border:1px solid ", ";"], function (props) {
  var _props$styles, _props$styles$slider;

  return ((_props$styles = props.styles) === null || _props$styles === void 0 ? void 0 : (_props$styles$slider = _props$styles.slider) === null || _props$styles$slider === void 0 ? void 0 : _props$styles$slider.handle) || _constants.DEFAULTS.SLIDER.HANDLE;
}, function (props) {
  var _props$styles2, _props$styles2$slider;

  return ((_props$styles2 = props.styles) === null || _props$styles2 === void 0 ? void 0 : (_props$styles2$slider = _props$styles2.slider) === null || _props$styles2$slider === void 0 ? void 0 : _props$styles2$slider.handle) || _constants.DEFAULTS.SLIDER.HANDLE;
}, function (props) {
  var _props$styles3, _props$styles3$slider;

  return ((_props$styles3 = props.styles) === null || _props$styles3 === void 0 ? void 0 : (_props$styles3$slider = _props$styles3.slider) === null || _props$styles3$slider === void 0 ? void 0 : _props$styles3$slider.handle) || _constants.DEFAULTS.SLIDER.HANDLE;
}, function (props) {
  return props.disabled ? 'not-allowed' : 'pointer';
}, function (props) {
  var _props$styles5, _props$styles5$slider;

  if (props.disabled) {
    var _props$styles4, _props$styles4$slider;

    return ((_props$styles4 = props.styles) === null || _props$styles4 === void 0 ? void 0 : (_props$styles4$slider = _props$styles4.slider) === null || _props$styles4$slider === void 0 ? void 0 : _props$styles4$slider.inactive) || _constants.DEFAULTS.SLIDER.INACTIVE;
  } // @ts-ignore


  return (_props$styles5 = props.styles) !== null && _props$styles5 !== void 0 && (_props$styles5$slider = _props$styles5.slider) !== null && _props$styles5$slider !== void 0 && _props$styles5$slider.filled ? props.theme.colors.primaryColor : props.theme.colors.primaryColorText;
}, function (props) {
  var _props$styles6, _props$styles6$slider;

  return props.disabled ? ((_props$styles6 = props.styles) === null || _props$styles6 === void 0 ? void 0 : (_props$styles6$slider = _props$styles6.slider) === null || _props$styles6$slider === void 0 ? void 0 : _props$styles6$slider.inactive) || _constants.DEFAULTS.SLIDER.INACTIVE : props.theme.colors.primaryColor;
});

var Container = _styledComponents["default"].div.withConfig({
  displayName: "Slider__Container",
  componentId: "odpwnt-0"
})(["display:flex;flex-direction:column;justify-content:center;"]);

var StyledInput = _styledComponents["default"].input.withConfig({
  displayName: "Slider__StyledInput",
  componentId: "odpwnt-1"
})(["appearance:none;-webkit-appearance:none;width:", ";height:", ";border-radius:calc( ", " * 0.6 );cursor:", ";background:linear-gradient(90deg,", " 0% ", "%,", " ", "% 100%);outline:none;&::-webkit-slider-thumb{appearance:none;-webkit-appearance:none;", "}&::-moz-range-thumb{", "}&::-ms-thumb{", "}}"], function (props) {
  return props.width || _constants.DEFAULTS.SLIDER.WIDTH;
}, function (props) {
  var _props$styles7, _props$styles7$slider;

  return ((_props$styles7 = props.styles) === null || _props$styles7 === void 0 ? void 0 : (_props$styles7$slider = _props$styles7.slider) === null || _props$styles7$slider === void 0 ? void 0 : _props$styles7$slider.size) || _constants.DEFAULTS.SLIDER.SIZE;
}, function (props) {
  var _props$styles8, _props$styles8$slider;

  return ((_props$styles8 = props.styles) === null || _props$styles8 === void 0 ? void 0 : (_props$styles8$slider = _props$styles8.slider) === null || _props$styles8$slider === void 0 ? void 0 : _props$styles8$slider.size) || _constants.DEFAULTS.SLIDER.SIZE;
}, function (props) {
  return props.disabled ? 'not-allowed' : 'pointer';
}, function (props) {
  var _props$styles9, _props$styles9$slider;

  return props.disabled ? ((_props$styles9 = props.styles) === null || _props$styles9 === void 0 ? void 0 : (_props$styles9$slider = _props$styles9.slider) === null || _props$styles9$slider === void 0 ? void 0 : _props$styles9$slider.inactive) || _constants.DEFAULTS.SLIDER.INACTIVE : props.theme.colors.primaryColor;
}, function (props) {
  return props.progress;
}, function (props) {
  var _props$styles10, _props$styles10$slide;

  return ((_props$styles10 = props.styles) === null || _props$styles10 === void 0 ? void 0 : (_props$styles10$slide = _props$styles10.slider) === null || _props$styles10$slide === void 0 ? void 0 : _props$styles10$slide.inactive) || _constants.DEFAULTS.SLIDER.INACTIVE;
}, function (props) {
  return props.progress;
}, sliderDefaults, sliderDefaults, sliderDefaults);

var List = _styledComponents["default"].ul.withConfig({
  displayName: "Slider__List",
  componentId: "odpwnt-2"
})(["list-style:none;margin-left:2px;margin-right:2px;margin-top:calc(", " * 0.25);width:", ";"], function (props) {
  return props.handle || _constants.DEFAULTS.SLIDER.HANDLE;
}, function (props) {
  return props.width || _constants.DEFAULTS.SLIDER.WIDTH;
});

var ListItem = _styledComponents["default"].li.withConfig({
  displayName: "Slider__ListItem",
  componentId: "odpwnt-3"
})(["display:inline;float:left;overflow:clip;position:relative;color:", ";font-size:", ";line-height:", ";", ""], function (props) {
  var _props$styles11, _props$styles11$slide;

  return props.disabled ? ((_props$styles11 = props.styles) === null || _props$styles11 === void 0 ? void 0 : (_props$styles11$slide = _props$styles11.slider) === null || _props$styles11$slide === void 0 ? void 0 : _props$styles11$slide.inactive) || _constants.DEFAULTS.SLIDER.INACTIVE : props.color || _constants.DEFAULTS.LABEL.COLOR;
}, function (props) {
  return props.size || _constants.DEFAULTS.LABEL.SIZE;
}, function (props) {
  return props.height || _constants.DEFAULTS.LABEL.HEIGHT;
}, function (props) {
  return calcLabelPos(props);
});

var Slider = /*#__PURE__*/function (_PureComponent) {
  _inherits(Slider, _PureComponent);

  var _super = _createSuper(Slider);

  function Slider() {
    var _DEFAULTS$VALUES, _DEFAULTS$VALUES2;

    var _this;

    _classCallCheck(this, Slider);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.state = {
      value: _this.props.values.initial || (_constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES === void 0 ? void 0 : _DEFAULTS$VALUES.INI),
      progress: (_this.props.values.initial || (_constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES2 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES2 === void 0 ? void 0 : _DEFAULTS$VALUES2.INI)) * 100
    };

    _this.onChange = function (e) {
      // @ts-ignore
      var value = +e.target.value;

      _this.setValue(value); // change the bg color of the slider track to highlight only selected value (left of thumb)


      _this.setProgress(_this.calculateProgress(value));

      _this.props.events.onChange(value);
    };

    _this.calculateProgress = function (value) {
      var _this$props;

      var _this$props$values = (_this$props = _this.props) === null || _this$props === void 0 ? void 0 : _this$props.values,
          max = _this$props$values.max,
          min = _this$props$values.min;

      return (value - min) / (max - min) * 100;
    };

    _this.setProgress = function (progress) {
      _this.setState({
        progress: progress
      });
    };

    _this.setValue = function (value) {
      _this.setState({
        value: value
      });
    };

    return _this;
  }

  _createClass(Slider, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.setProgress(this.calculateProgress(this.state.value));
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _DEFAULTS$VALUES3;

      var initValue = prevProps.values.initial || (_constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES3 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES3 === void 0 ? void 0 : _DEFAULTS$VALUES3.INI);
      var value = this.state.value || initValue;

      if (value !== prevProps.values.initial) {
        this.setValue(value);
        this.setProgress(this.calculateProgress(value));
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$state, _this$state2;

      var _this$props2 = this.props,
          dataTestId = _this$props2.dataTestId,
          disabled = _this$props2.disabled,
          events = _this$props2.events,
          label = _this$props2.label,
          _this$props2$list = _this$props2.list,
          list = _this$props2$list === void 0 ? [] : _this$props2$list,
          slider = _this$props2.slider,
          value = _this$props2.value,
          values = _this$props2.values;
      var min = values.min,
          max = values.max,
          step = values.step;
      var width = slider.width;
      var onBlur = events.onBlur;
      var val = value || typeof ((_this$state = this.state) === null || _this$state === void 0 ? void 0 : _this$state.value) === 'undefined' ? values.initial : (_this$state2 = this.state) === null || _this$state2 === void 0 ? void 0 : _this$state2.value;
      var progress = this.calculateProgress(val);
      var listitems = list.map(function (item, index) {
        return /*#__PURE__*/_react["default"].createElement(ListItem, _extends({
          disabled: disabled,
          handle: slider.handle,
          index: index
        }, label, {
          listWidth: width || _constants.DEFAULTS.SLIDER.WIDTH,
          lastIndex: list.length - 1,
          value: item
        }), item);
      });
      return /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(StyledInput, {
        disabled: disabled,
        width: width,
        type: "range",
        min: min,
        max: max,
        step: step,
        value: val,
        progress: progress,
        onChange: this.onChange,
        onBlur: onBlur,
        styles: {
          label: label,
          slider: slider
        },
        "data-test": dataTestId
      }), /*#__PURE__*/_react["default"].createElement(List, _extends({}, label, {
        width: width,
        handle: slider.handle
      }), listitems));
    }
  }]);

  return Slider;
}(_react.PureComponent);

exports.Slider = Slider;
Slider.defaultProps = {
  disabled: false,
  label: {
    color: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$LABEL = _constants.DEFAULTS.LABEL) === null || _DEFAULTS$LABEL === void 0 ? void 0 : _DEFAULTS$LABEL.COLOR,
    height: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$LABEL2 = _constants.DEFAULTS.LABEL) === null || _DEFAULTS$LABEL2 === void 0 ? void 0 : _DEFAULTS$LABEL2.HEIGHT,
    size: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$LABEL3 = _constants.DEFAULTS.LABEL) === null || _DEFAULTS$LABEL3 === void 0 ? void 0 : _DEFAULTS$LABEL3.SIZE
  },
  events: {
    onBlur: function onBlur() {},
    onChange: function onChange() {}
  },
  list: [],
  slider: {
    filled: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER === void 0 ? void 0 : _DEFAULTS$SLIDER.FILLED,
    handle: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER2 = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER2 === void 0 ? void 0 : _DEFAULTS$SLIDER2.HANDLE,
    inactive: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER3 = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER3 === void 0 ? void 0 : _DEFAULTS$SLIDER3.INACTIVE,
    labels: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER4 = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER4 === void 0 ? void 0 : _DEFAULTS$SLIDER4.LABELS,
    size: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER5 = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER5 === void 0 ? void 0 : _DEFAULTS$SLIDER5.SIZE,
    width: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$SLIDER6 = _constants.DEFAULTS.SLIDER) === null || _DEFAULTS$SLIDER6 === void 0 ? void 0 : _DEFAULTS$SLIDER6.WIDTH
  },
  value: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES4 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES4 === void 0 ? void 0 : _DEFAULTS$VALUES4.INI,
  values: {
    initial: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES5 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES5 === void 0 ? void 0 : _DEFAULTS$VALUES5.INI,
    max: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES6 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES6 === void 0 ? void 0 : _DEFAULTS$VALUES6.MAX,
    min: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES7 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES7 === void 0 ? void 0 : _DEFAULTS$VALUES7.MIN,
    step: _constants.DEFAULTS === null || _constants.DEFAULTS === void 0 ? void 0 : (_DEFAULTS$VALUES8 = _constants.DEFAULTS.VALUES) === null || _DEFAULTS$VALUES8 === void 0 ? void 0 : _DEFAULTS$VALUES8.STEP
  }
};
var _default = Slider;
exports["default"] = _default;
