// @spotify-internal/encore-web/es/components/FlyOut/Wrapper.js
import { screenSmMin, screenXsMax, spacer16 } from '@spotify-internal/encore-foundation';
import styled, { css } from 'styled-components';
import { defaultTheme, rootStyle, triangle } from "../../styles";
import { flyoutControlsBg, flyoutDefaultColorSet, flyoutMainBg, flyoutWidth } from "./variables";
export var Wrapper = styled.nav.withConfig({
  displayName: "Wrapper",
  componentId: "sc-1haoc8j-0"
})(["", ";color:var(--text-base,", ");", ";", ";@media (min-width:", "){width:", ";}"], rootStyle(), defaultTheme[flyoutDefaultColorSet].text.base, function (props) {
  return props.isLightArrow ? css(["@media (min-width:", "){", ";}"], screenSmMin, triangle({
    color: flyoutControlsBg,
    position: 'top',
    offset: spacer16
  })) : css(["@media (min-width:", "){", ";}"], screenSmMin, triangle({
    color: flyoutMainBg,
    position: 'top',
    offset: spacer16
  }));
}, function (props) {
  return props.isMobileFlex && css(["@media (max-width:", "){display:flex;flex-direction:column;height:100vh;}"], screenXsMax);
}, screenSmMin, flyoutWidth);
