// @spotify-internal/adstudio-shared/lib/components/RadialMeter/archGradient.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ArchGradient = function ArchGradient(_ref) {
  var children = _ref.children,
      _ref$gradientFill = _ref.gradientFill,
      gradientFill = _ref$gradientFill === void 0 ? '#979797' : _ref$gradientFill;
  return /*#__PURE__*/_react["default"].createElement("svg", {
    viewBox: "0 0 139 75",
    xmlns: "http://www.w3.org/2000/svg",
    xmlnsXlink: "http://www.w3.org/1999/xlink"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, children), /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M139 69a6 6 0 0 1-12 0c0-31.47-25.732-57-57.5-57S12 37.53 12 69a6 6 0 0 1-12 0C0 30.882 31.127 0 69.5 0S139 30.882 139 69z",
    id: "audience-sizer-radial-meter"
  })), /*#__PURE__*/_react["default"].createElement("clipPath", {
    id: "b"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    overflow: "visible",
    xlinkHref: "#audience-sizer-radial-meter"
  })), /*#__PURE__*/_react["default"].createElement("g", {
    clipPath: "url(#b)"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M-487-205h882v615h-882z",
    id: "c"
  })), /*#__PURE__*/_react["default"].createElement("clipPath", {
    id: "d"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    overflow: "visible",
    xlinkHref: "#c"
  })), /*#__PURE__*/_react["default"].createElement("g", {
    clipPath: "url(#d)"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M-5-5h149v144H-5z",
    id: "e"
  })), /*#__PURE__*/_react["default"].createElement("clipPath", {
    id: "f"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    overflow: "visible",
    xlinkHref: "#e"
  })), /*#__PURE__*/_react["default"].createElement("g", {
    clipPath: "url(#f)"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M0 0h139v75H0z",
    id: "g"
  })), /*#__PURE__*/_react["default"].createElement("clipPath", {
    id: "h"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    overflow: "visible",
    xlinkHref: "#g"
  })), /*#__PURE__*/_react["default"].createElement("path", {
    clipPath: "url(#h)",
    d: "M-5-5h149v85H-5z",
    fill: gradientFill
  })), /*#__PURE__*/_react["default"].createElement("g", {
    clipPath: "url(#f)"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("circle", {
    cx: "69.5",
    cy: "68.5",
    id: "j",
    r: "70.5"
  })), /*#__PURE__*/_react["default"].createElement("clipPath", {
    id: "k"
  }, /*#__PURE__*/_react["default"].createElement("use", {
    overflow: "visible",
    xlinkHref: "#j"
  })), /*#__PURE__*/_react["default"].createElement("g", {
    clipPath: "url(#k)"
  }, /*#__PURE__*/_react["default"].createElement("defs", null, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M0 0h139v75H0z",
    id: "l"
  })))))));
};

ArchGradient.propTypes = {
  children: _propTypes["default"].node,
  gradientFill: _propTypes["default"].string
};
var _default = ArchGradient;
exports["default"] = _default;
