// @spotify-internal/adstudio-tape/lib/components/PaymentDetails/AddCard.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = AddCard;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _reactMarkdown = _interopRequireDefault(require("react-markdown"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _Type = _interopRequireDefault(require("../Type"));

var _constants = require("./constants");

var _Container = _interopRequireDefault(require("./Container"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconsDescription = (0, _styledComponents["default"])(_Type["default"]).withConfig({
  displayName: "AddCard__IconsDescription",
  componentId: "sc-1pivk4f-0"
})(["svg{height:", ";position:relative;top:", ";width:auto;&:last-of-type{margin-right:", ";}}"], _encoreFoundation.spacer16, _encoreFoundation.spacer4, _encoreFoundation.spacer8);

function AddCard(_ref) {
  var addCardCTAtext = _ref.addCardCTAtext,
      addCardDescription = _ref.addCardDescription,
      addCardSubtext = _ref.addCardSubtext,
      addCardText = _ref.addCardText,
      hideShadow = _ref.hideShadow,
      onClickCheckout = _ref.onClickCheckout,
      redirecting = _ref.redirecting,
      redirectingCTAtext = _ref.redirectingCTAtext,
      viewOnly = _ref.viewOnly,
      viewOnlySubText = _ref.viewOnlySubText;
  return /*#__PURE__*/_react["default"].createElement(_Container["default"], {
    cta: !viewOnly && {
      label: redirecting ? redirectingCTAtext : addCardCTAtext,
      disabled: redirecting,
      onClick: onClickCheckout
    },
    hideShadow: hideShadow
  }, /*#__PURE__*/_react["default"].createElement(_Type["default"], {
    element: "p",
    variant: "body-1-book",
    flush: true
  }, /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
    source: addCardText,
    linkTarget: "_blank",
    renderers: {
      /* eslint-disable react/prop-types */
      paragraph: function paragraph(_ref2) {
        var children = _ref2.children;
        return /*#__PURE__*/_react["default"].createElement("span", null, children);
      }
      /* eslint-enable react/prop-types */

    }
  })), /*#__PURE__*/_react["default"].createElement(_Type["default"], {
    element: "p",
    variant: "body-2-book"
  }, /*#__PURE__*/_react["default"].createElement(_reactMarkdown["default"], {
    linkTarget: "_blank",
    source: viewOnly ? viewOnlySubText : addCardSubtext,
    renderers: {
      /* eslint-disable react/prop-types */
      paragraph: function paragraph(_ref3) {
        var children = _ref3.children;
        return /*#__PURE__*/_react["default"].createElement("span", null, children);
      }
      /* eslint-enable react/prop-types */

    }
  })), !viewOnly && /*#__PURE__*/_react["default"].createElement(IconsDescription, {
    element: "p",
    variant: "body-3-book",
    flush: true
  }, _constants.SUPPORTED_CARDS.map(function (_ref4) {
    var Icon = _ref4.icon,
        name = _ref4.name;
    return /*#__PURE__*/_react["default"].createElement(Icon, {
      key: name
    });
  }), addCardDescription));
}

AddCard.propTypes = {
  addCardCTAtext: _propTypes["default"].string,
  addCardDescription: _propTypes["default"].string,
  addCardSubtext: _propTypes["default"].string,
  addCardText: _propTypes["default"].string,
  hideShadow: _propTypes["default"].bool,
  onClickCheckout: _propTypes["default"].func,
  redirecting: _propTypes["default"].bool,
  redirectingCTAtext: _propTypes["default"].string,
  viewOnly: _propTypes["default"].bool,
  viewOnlySubText: _propTypes["default"].string
};
AddCard.defaultProps = {
  addCardCTAtext: _constants.DEFAULT_ADD_CARD_CTA_TEXT,
  addCardDescription: _constants.DEFAULT_ADD_CARD_DESCRIPTION,
  addCardSubtext: _constants.DEFAULT_ADD_CARD_SUBTEXT,
  addCardText: _constants.DEFAULT_ADD_CARD_TEXT,
  redirectingCTAtext: _constants.DEFAULT_REDIRECTING_CTA_TEXT,
  viewOnlySubText: _constants.DEFAULT_CANNOT_ADD_CARD_SUBTEXT
};
