// @spotify-internal/spoticharts/cjs/components/DonutChart/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const reactochart_1 = require("reactochart");
const global_1 = __importDefault(require("../../styles/global"));
function DonutChart(_a) {
    var { data, slice, sliceStyle, sliceLabel, sliceLabelStyle, sliceLabelDistance, holeRadius, width, height, marginLeft, marginRight, marginTop, marginBottom, onMouseEnterSlice, onMouseLeaveSlice } = _a, rest = __rest(_a, ["data", "slice", "sliceStyle", "sliceLabel", "sliceLabelStyle", "sliceLabelDistance", "holeRadius", "width", "height", "marginLeft", "marginRight", "marginTop", "marginBottom", "onMouseEnterSlice", "onMouseLeaveSlice"]);
    return (react_1.default.createElement(react_1.Fragment, null,
        react_1.default.createElement(reactochart_1.PieChart, Object.assign({ data: data, slice: slice, pieSliceStyle: sliceStyle, getPieSliceLabel: sliceLabel, holeRadius: holeRadius, pieSliceLabelStyle: sliceLabelStyle, pieSliceLabelDistance: sliceLabelDistance, width: width, height: height, marginLeft: marginLeft, marginRight: marginRight, marginTop: marginTop, marginBottom: marginBottom, onMouseEnterSlice: onMouseEnterSlice, onMouseLeaveSlice: onMouseLeaveSlice }, rest)),
        react_1.default.createElement(global_1.default, null)));
}
exports.default = DonutChart;
DonutChart.propTypes = {
    data: prop_types_1.default.array.isRequired,
    slice: prop_types_1.default.func.isRequired,
    sliceLabel: prop_types_1.default.oneOfType([prop_types_1.default.func, prop_types_1.default.string]),
    sliceStyle: prop_types_1.default.oneOfType([prop_types_1.default.func, prop_types_1.default.object]),
    holeRadius: prop_types_1.default.number,
    sliceLabelStyle: prop_types_1.default.oneOfType([prop_types_1.default.func, prop_types_1.default.object]),
    sliceLabelDistance: prop_types_1.default.number,
    width: prop_types_1.default.number,
    height: prop_types_1.default.number,
    marginLeft: prop_types_1.default.number,
    marginRight: prop_types_1.default.number,
    marginTop: prop_types_1.default.number,
    marginBottom: prop_types_1.default.number,
    onMouseEnterSlice: prop_types_1.default.func,
    onMouseLeaveSlice: prop_types_1.default.func,
};
DonutChart.defaultProps = {
    holeRadius: 50,
};
