// @spotify-internal/encore-web/es/components/ButtonSecondary/index.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["component", "buttonSize", "iconLeading", "iconTrailing", "iconOnly", "children", "aria-label", "aria-labelledby", "buttonLegacy", "theme"],
    _excluded2 = ["buttonLegacy"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import React, { useContext } from 'react';
import { BrowserDefaultFocusStyleContext } from "../../contexts/BrowserDefaultFocusStyleContext";
import { EncoreContext, encoreContextKeyword, getStatus } from "../../contexts/EncoreContext";
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { deprecatedPropWarning, renamedPropWarning } from "../../deprecationWarnings";
import { buttonSizes } from "../../styles";
import { Button } from "./Button";
import { ButtonChildren } from "./ButtonChildren";
import { ButtonLegacyComponent } from "./ButtonLegacy";
var ButtonComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var component = _ref.component,
      _ref$buttonSize = _ref.buttonSize,
      buttonSize = _ref$buttonSize === void 0 ? 'md' : _ref$buttonSize,
      iconLeading = _ref.iconLeading,
      iconTrailing = _ref.iconTrailing,
      iconOnly = _ref.iconOnly,
      children = _ref.children,
      ariaLabel = _ref['aria-label'],
      ariaLabelledby = _ref['aria-labelledby'],
      buttonLegacy = _ref.buttonLegacy,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    renamedPropWarning('theme', 'colorSet');
  }

  if (iconOnly && !ariaLabel && !ariaLabelledby) {
    // eslint-disable-next-line no-console
    console.warn('ButtonSecondary must have an aria-label or aria-labelledby attribute defined when using the iconOnly prop in order to be accessible.');
  }

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  var _useContext2 = useContext(BrowserDefaultFocusStyleContext),
      useBrowserDefaultFocusStyle = _useContext2.useBrowserDefaultFocusStyle;

  var config = useContext(EncoreContext);
  var status = getStatus(encoreContextKeyword.button, config);
  var rel = props.href && props.target === '_blank' ? 'noopener noreferrer' : null;
  return /*#__PURE__*/React.createElement(Button, Object.assign({}, props, {
    rel: rel,
    ref: ref,
    component: !component && props.href ? 'a' : component,
    "aria-label": ariaLabel,
    "aria-labelledby": ariaLabelledby,
    isUsingKeyboard: isUsingKeyboard,
    useBrowserDefaultFocusStyle: useBrowserDefaultFocusStyle,
    buttonSize: buttonSize,
    iconOnly: iconOnly,
    iconLeading: iconLeading,
    iconTrailing: iconTrailing,
    status: status
  }), /*#__PURE__*/React.createElement(ButtonChildren, {
    iconOnly: iconOnly,
    iconLeading: iconLeading,
    iconTrailing: iconTrailing,
    buttonSize: buttonSize,
    children: children
  }));
});
/** Component that figures out if we are rendering the legacy or current button */

var ButtonToRender = /*#__PURE__*/React.forwardRef(function (_ref2, ref) {
  var buttonLegacy = _ref2.buttonLegacy,
      props = _objectWithoutProperties(_ref2, _excluded2);

  if (buttonLegacy) {
    deprecatedPropWarning('buttonLegacy', "This prop will be removed soon. Please migrate to new button styles by removing this prop.\nRead more: https://encore.spotify.net/web/whats-new/green-buttons/?format=web-js");
    return /*#__PURE__*/React.createElement(ButtonLegacyComponent, Object.assign({
      ref: ref
    }, props));
  }

  return /*#__PURE__*/React.createElement(ButtonComponent, Object.assign({
    ref: ref
  }, props));
});
/**
 * TODO: Remove when we deprecate `buttonSize` helpers like `ButtonSecondary.sm`.
 */

var buttonHelpers = _objectSpread(_objectSpread({}, buttonSizes), {}, {
  displayName: 'ButtonSecondary'
});
/**
* **ButtonSecondary** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/ButtonSecondary) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-button-buttonsecondary--default) | [Encore](https://encore.spotify.net/web/components/button?format=figma-figma#buttonsecondary) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=7643:5628) |
*
* Secondary buttons have a stroke and no fill and indicate an alternate or less important action.
*
* @example
* () => <ButtonSecondary>Default</ButtonSecondary>;
*
*/


export var ButtonSecondary = Object.assign(ButtonToRender, buttonHelpers);
export default ButtonSecondary;
