// @spotify-internal/adstudio-shared/lib/components/ArtistTargetingSearch/Value/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SelectedArtistValue = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var SelectedArtistValue = function SelectedArtistValue(props) {
  var _props$disabled = props.disabled,
      disabled = _props$disabled === void 0 ? false : _props$disabled,
      item = props.item,
      onClick = props.onClick;

  var SelectedArtist = _styledComponents["default"].div.withConfig({
    displayName: "Value__SelectedArtist",
    componentId: "sc-1tqrec9-0"
  })(["", " ", " border:1px solid ", ";border-radius:4px;overflow:hidden;"], function (props) {
    return props.disabled && "background-color: ".concat(_encoreAdvertisingWeb.gray95, ";");
  }, function (props) {
    return props.disabled && "color: ".concat(_encoreAdvertisingWeb.gray50, ";");
  }, _encoreAdvertisingWeb.gray60);

  return /*#__PURE__*/_react["default"].createElement(SelectedArtist, {
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement(_Item["default"], {
    artist: item,
    disabled: disabled,
    onClick: onClick
  }));
};

exports.SelectedArtistValue = SelectedArtistValue;
SelectedArtistValue.propTypes = {
  disabled: _propTypes["default"].bool,
  onClick: _propTypes["default"].func,
  item: _propTypes["default"].object.isRequired
};
var _default = SelectedArtistValue;
exports["default"] = _default;
