// @spotify-internal/adstudio-shared/lib/components/ArtistTargetingSearch/Value/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _Suggestion = _interopRequireDefault(require("../Suggestion"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconX = _adstudioTape.Icons.IconX;

var SelectedArtistValueItem = function SelectedArtistValueItem(props) {
  var disabled = props.disabled,
      artist = props.artist,
      onClick = props.onClick;
  var deselectBtnHeight = _encoreAdvertisingWeb.spacer16;
  var deselectBtnWidth = _encoreAdvertisingWeb.spacer24;

  var ArtistName = _styledComponents["default"].div.withConfig({
    displayName: "Item__ArtistName",
    componentId: "sc-1hkhjep-0"
  })(["display:inline;float:left;width:calc(100% - ", ");"], deselectBtnWidth);

  var DeselectBtnWrapper = _styledComponents["default"].div.withConfig({
    displayName: "Item__DeselectBtnWrapper",
    componentId: "sc-1hkhjep-1"
  })(["display:inline;float:left;width:", ";"], deselectBtnWidth);

  var DeselectBtn = _styledComponents["default"].div.withConfig({
    displayName: "Item__DeselectBtn",
    componentId: "sc-1hkhjep-2"
  })(["color:", ";cursor:pointer;margin:", " 0;height:", ";width:", ";&:hover{color:", ";}"], _encoreAdvertisingWeb.gray75, _encoreAdvertisingWeb.spacer12, deselectBtnHeight, deselectBtnHeight, _encoreAdvertisingWeb.gray60);

  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(ArtistName, null, /*#__PURE__*/_react["default"].createElement(_Suggestion["default"], {
    suggestion: artist,
    small: true
  })), /*#__PURE__*/_react["default"].createElement(DeselectBtnWrapper, {
    key: artist.id
  }, !disabled && /*#__PURE__*/_react["default"].createElement(DeselectBtn, {
    onClick: onClick,
    role: "button",
    tabIndex: "0"
  }, /*#__PURE__*/_react["default"].createElement(IconX, {
    iconSize: 16
  }))));
};

SelectedArtistValueItem.propTypes = {
  disabled: _propTypes["default"].bool,
  artist: _propTypes["default"].object.isRequired,
  onClick: _propTypes["default"].func
};
var _default = SelectedArtistValueItem;
exports["default"] = _default;
