// @spotify-internal/ubi-sdk-s4a-web-audience/WebAudienceEventFactory.js
"use strict";
// DO NOT EDIT THIS FILE!
// This file was generated by UBI-generators.
// For support contact us on the #ubi slack channel.
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAudienceEventFactory = exports.createWebAudienceEventFactory = void 0;
var SPECIFICATION_ID = "web-audience";
var APP = "s4a";
var SPECIFICATION_VERSION = "7.1.0";
var SPECIFICATION_MODE = "default";
var GENERATOR_VERSION = "9.1.11";
function createWebAudienceEventFactory(options) {
    return new WebAudienceEventFactory(options);
}
exports.createWebAudienceEventFactory = createWebAudienceEventFactory;
/**
 *
 */
var WebAudienceEventFactory = /** @class */ (function () {
    function WebAudienceEventFactory(options) {
        this._path = [__assign({ name: SPECIFICATION_ID }, options.data)];
        if (options === null || options === void 0 ? void 0 : options.parentAbsoluteLocation) {
            this._parentAbsoluteLocation = options.parentAbsoluteLocation;
        }
    }
    /**
    * @returns A new instance of DataDelayBannerEventFactory
    */
    WebAudienceEventFactory.prototype.dataDelayBannerFactory = function () {
        return new WebAudienceEventFactory.DataDelayBannerEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of CountryFilterDropdownEventFactory
    */
    WebAudienceEventFactory.prototype.countryFilterDropdownFactory = function () {
        return new WebAudienceEventFactory.CountryFilterDropdownEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of AudienceStatsVersionEventFactory
    */
    WebAudienceEventFactory.prototype.audienceStatsVersionFactory = function () {
        return new WebAudienceEventFactory.AudienceStatsVersionEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ToplineNumberEventFactory
    */
    WebAudienceEventFactory.prototype.toplineNumberFactory = function () {
        return new WebAudienceEventFactory.ToplineNumberEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of AudienceTimelineEventFactory
    */
    WebAudienceEventFactory.prototype.audienceTimelineFactory = function () {
        return new WebAudienceEventFactory.AudienceTimelineEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of SourceOfStreamsEventFactory
    */
    WebAudienceEventFactory.prototype.sourceOfStreamsFactory = function () {
        return new WebAudienceEventFactory.SourceOfStreamsEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ListenersAgeAndGenderEventFactory
    */
    WebAudienceEventFactory.prototype.listenersAgeAndGenderFactory = function () {
        return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of ListenersAlsoLikeEventFactory
    */
    WebAudienceEventFactory.prototype.listenersAlsoLikeFactory = function () {
        return new WebAudienceEventFactory.ListenersAlsoLikeEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of TopCountriesEventFactory
    */
    WebAudienceEventFactory.prototype.topCountriesFactory = function () {
        return new WebAudienceEventFactory.TopCountriesEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of TopCitiesEventFactory
    */
    WebAudienceEventFactory.prototype.topCitiesFactory = function () {
        return new WebAudienceEventFactory.TopCitiesEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    /**
    * @returns A new instance of EngagementEventFactory
    */
    WebAudienceEventFactory.prototype.engagementFactory = function () {
        return new WebAudienceEventFactory.EngagementEventFactory({
            path: this._path,
            parentAbsoluteLocation: this._parentAbsoluteLocation
        });
    };
    WebAudienceEventFactory.prototype._getCurrentLocation = function () {
        return {
            pathNodes: this._path,
            specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
            specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
        };
    };
    WebAudienceEventFactory.prototype.getAbsoluteLocation = function () {
        var _a;
        var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
        return {
            locations: __spreadArrays((parentLocations || []), [
                this._getCurrentLocation()
            ]),
        };
    };
    return WebAudienceEventFactory;
}());
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create DataDelayBannerEventFactory class with path
     * @param path to this node
     */
    var DataDelayBannerEventFactory = /** @class */ (function () {
        function DataDelayBannerEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "data_delay_banner"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        DataDelayBannerEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        DataDelayBannerEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for DataDelayBannerEventFactory.
         */
        DataDelayBannerEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return DataDelayBannerEventFactory;
    }());
    WebAudienceEventFactory.DataDelayBannerEventFactory = DataDelayBannerEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create CountryFilterDropdownEventFactory class with path
     * @param path to this node
     */
    var CountryFilterDropdownEventFactory = /** @class */ (function () {
        function CountryFilterDropdownEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "country_filter_dropdown"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param id of the CountryValueEventFactory
        * @returns A new instance of CountryValueEventFactory
        */
        CountryFilterDropdownEventFactory.prototype.countryValueFactory = function (data) {
            return new WebAudienceEventFactory.CountryFilterDropdownEventFactory.CountryValueEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        CountryFilterDropdownEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        CountryFilterDropdownEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create interaction for hit ui_reveal action.
         */
        CountryFilterDropdownEventFactory.prototype.hitUiReveal = function () {
            var data = {
                actionName: "ui_reveal",
                actionVersion: 1,
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                interactionType: "hit",
                specificationMode: SPECIFICATION_MODE,
                specificationVersion: SPECIFICATION_VERSION,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return CountryFilterDropdownEventFactory;
    }());
    WebAudienceEventFactory.CountryFilterDropdownEventFactory = CountryFilterDropdownEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create AudienceStatsVersionEventFactory class with path
     * @param path to this node
     */
    var AudienceStatsVersionEventFactory = /** @class */ (function () {
        function AudienceStatsVersionEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "audience_stats_version"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of OverviewTabEventFactory
        */
        AudienceStatsVersionEventFactory.prototype.overviewTabFactory = function () {
            return new WebAudienceEventFactory.AudienceStatsVersionEventFactory.OverviewTabEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of EngagementTabEventFactory
        */
        AudienceStatsVersionEventFactory.prototype.engagementTabFactory = function () {
            return new WebAudienceEventFactory.AudienceStatsVersionEventFactory.EngagementTabEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        AudienceStatsVersionEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AudienceStatsVersionEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for AudienceStatsVersionEventFactory.
         */
        AudienceStatsVersionEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AudienceStatsVersionEventFactory;
    }());
    WebAudienceEventFactory.AudienceStatsVersionEventFactory = AudienceStatsVersionEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create ToplineNumberEventFactory class with path
     * @param path to this node
     */
    var ToplineNumberEventFactory = /** @class */ (function () {
        function ToplineNumberEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "topline_number"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        ToplineNumberEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ToplineNumberEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for ToplineNumberEventFactory.
         */
        ToplineNumberEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return ToplineNumberEventFactory;
    }());
    WebAudienceEventFactory.ToplineNumberEventFactory = ToplineNumberEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create AudienceTimelineEventFactory class with path
     * @param path to this node
     */
    var AudienceTimelineEventFactory = /** @class */ (function () {
        function AudienceTimelineEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "audience_timeline"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of MetricSelectionEventFactory
        */
        AudienceTimelineEventFactory.prototype.metricSelectionFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.MetricSelectionEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of DownloadCsvButtonEventFactory
        */
        AudienceTimelineEventFactory.prototype.downloadCsvButtonFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.DownloadCsvButtonEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of DownloadCsvButtonHoverTextEventFactory
        */
        AudienceTimelineEventFactory.prototype.downloadCsvButtonHoverTextFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.DownloadCsvButtonHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of DatePickerEventFactory
        */
        AudienceTimelineEventFactory.prototype.datePickerFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.DatePickerEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of TimelineChartEventFactory
        */
        AudienceTimelineEventFactory.prototype.timelineChartFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.TimelineChartEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the TimelineChartHoverTextEventFactory
        * @returns A new instance of TimelineChartHoverTextEventFactory
        */
        AudienceTimelineEventFactory.prototype.timelineChartHoverTextFactory = function (data) {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.TimelineChartHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @returns A new instance of EmptyStateEventFactory
        */
        AudienceTimelineEventFactory.prototype.emptyStateFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.EmptyStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of ErrorStateEventFactory
        */
        AudienceTimelineEventFactory.prototype.errorStateFactory = function () {
            return new WebAudienceEventFactory.AudienceTimelineEventFactory.ErrorStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        AudienceTimelineEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        AudienceTimelineEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for AudienceTimelineEventFactory.
         */
        AudienceTimelineEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return AudienceTimelineEventFactory;
    }());
    WebAudienceEventFactory.AudienceTimelineEventFactory = AudienceTimelineEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create SourceOfStreamsEventFactory class with path
     * @param path to this node
     */
    var SourceOfStreamsEventFactory = /** @class */ (function () {
        function SourceOfStreamsEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "source_of_streams"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of ChartEventFactory
        */
        SourceOfStreamsEventFactory.prototype.chartFactory = function () {
            return new WebAudienceEventFactory.SourceOfStreamsEventFactory.ChartEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the ChartHoverTextEventFactory
        * @returns A new instance of ChartHoverTextEventFactory
        */
        SourceOfStreamsEventFactory.prototype.chartHoverTextFactory = function (data) {
            return new WebAudienceEventFactory.SourceOfStreamsEventFactory.ChartHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @returns A new instance of ErrorStateEventFactory
        */
        SourceOfStreamsEventFactory.prototype.errorStateFactory = function () {
            return new WebAudienceEventFactory.SourceOfStreamsEventFactory.ErrorStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of EmptyStateEventFactory
        */
        SourceOfStreamsEventFactory.prototype.emptyStateFactory = function () {
            return new WebAudienceEventFactory.SourceOfStreamsEventFactory.EmptyStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        SourceOfStreamsEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        SourceOfStreamsEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return SourceOfStreamsEventFactory;
    }());
    WebAudienceEventFactory.SourceOfStreamsEventFactory = SourceOfStreamsEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create ListenersAgeAndGenderEventFactory class with path
     * @param path to this node
     */
    var ListenersAgeAndGenderEventFactory = /** @class */ (function () {
        function ListenersAgeAndGenderEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "listeners_age_and_gender"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of GenderChartEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.genderChartFactory = function () {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.GenderChartEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the GenderChartHoverTextEventFactory
        * @returns A new instance of GenderChartHoverTextEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.genderChartHoverTextFactory = function (data) {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.GenderChartHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @returns A new instance of AgeChartEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.ageChartFactory = function () {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.AgeChartEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the AgeChartHoverTextEventFactory
        * @returns A new instance of AgeChartHoverTextEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.ageChartHoverTextFactory = function (data) {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.AgeChartHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        /**
        * @returns A new instance of ErrorStateEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.errorStateFactory = function () {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.ErrorStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of EmptyStateEventFactory
        */
        ListenersAgeAndGenderEventFactory.prototype.emptyStateFactory = function () {
            return new WebAudienceEventFactory.ListenersAgeAndGenderEventFactory.EmptyStateEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        ListenersAgeAndGenderEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ListenersAgeAndGenderEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return ListenersAgeAndGenderEventFactory;
    }());
    WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = ListenersAgeAndGenderEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create ListenersAlsoLikeEventFactory class with path
     * @param path to this node
     */
    var ListenersAlsoLikeEventFactory = /** @class */ (function () {
        function ListenersAlsoLikeEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "listeners_also_like"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @param uri of the SimilarArtistEventFactory
        * @returns A new instance of SimilarArtistEventFactory
        */
        ListenersAlsoLikeEventFactory.prototype.similarArtistFactory = function (data) {
            return new WebAudienceEventFactory.ListenersAlsoLikeEventFactory.SimilarArtistEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        ListenersAlsoLikeEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        ListenersAlsoLikeEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return ListenersAlsoLikeEventFactory;
    }());
    WebAudienceEventFactory.ListenersAlsoLikeEventFactory = ListenersAlsoLikeEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create TopCountriesEventFactory class with path
     * @param path to this node
     */
    var TopCountriesEventFactory = /** @class */ (function () {
        function TopCountriesEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "top_countries"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of TopCountriesListEventFactory
        */
        TopCountriesEventFactory.prototype.topCountriesListFactory = function () {
            return new WebAudienceEventFactory.TopCountriesEventFactory.TopCountriesListEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @returns A new instance of CountryMapEventFactory
        */
        TopCountriesEventFactory.prototype.countryMapFactory = function () {
            return new WebAudienceEventFactory.TopCountriesEventFactory.CountryMapEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        /**
        * @param id of the CountryMapHoverTextEventFactory
        * @returns A new instance of CountryMapHoverTextEventFactory
        */
        TopCountriesEventFactory.prototype.countryMapHoverTextFactory = function (data) {
            return new WebAudienceEventFactory.TopCountriesEventFactory.CountryMapHoverTextEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation,
                data: data
            });
        };
        TopCountriesEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TopCountriesEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        return TopCountriesEventFactory;
    }());
    WebAudienceEventFactory.TopCountriesEventFactory = TopCountriesEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create TopCitiesEventFactory class with path
     * @param path to this node
     */
    var TopCitiesEventFactory = /** @class */ (function () {
        function TopCitiesEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "top_cities"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        TopCitiesEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        TopCitiesEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for TopCitiesEventFactory.
         */
        TopCitiesEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return TopCitiesEventFactory;
    }());
    WebAudienceEventFactory.TopCitiesEventFactory = TopCitiesEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    /**
     * Create EngagementEventFactory class with path
     * @param path to this node
     */
    var EngagementEventFactory = /** @class */ (function () {
        function EngagementEventFactory(_a) {
            var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
            this._path = path.concat({
                name: "engagement"
            });
            if (parentAbsoluteLocation) {
                this._parentAbsoluteLocation = parentAbsoluteLocation;
            }
        }
        /**
        * @returns A new instance of StatsThisPeriodEventFactory
        */
        EngagementEventFactory.prototype.statsThisPeriodFactory = function () {
            return new WebAudienceEventFactory.EngagementEventFactory.StatsThisPeriodEventFactory({
                path: this._path,
                parentAbsoluteLocation: this._parentAbsoluteLocation
            });
        };
        EngagementEventFactory.prototype._getCurrentLocation = function () {
            return {
                pathNodes: this._path,
                specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
            };
        };
        EngagementEventFactory.prototype.getAbsoluteLocation = function () {
            var _a;
            var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
            return {
                locations: __spreadArrays((parentLocations || []), [
                    this._getCurrentLocation()
                ]),
            };
        };
        /**
         * Create impression for EngagementEventFactory.
         */
        EngagementEventFactory.prototype.impression = function () {
            var data = {
                app: APP,
                generatorVersion: GENERATOR_VERSION,
                specificationVersion: SPECIFICATION_VERSION,
                specificationMode: SPECIFICATION_MODE,
                annotatorConfigurationVersion: '',
                annotatorVersion: '',
                location: this._getCurrentLocation(),
                parentAbsoluteLocation: this._parentAbsoluteLocation
            };
            return data;
        };
        return EngagementEventFactory;
    }());
    WebAudienceEventFactory.EngagementEventFactory = EngagementEventFactory;
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var CountryFilterDropdownEventFactory;
    (function (CountryFilterDropdownEventFactory) {
        /**
         * Create CountryValueEventFactory class with path
         * @param path to this node
         */
        var CountryValueEventFactory = /** @class */ (function () {
            function CountryValueEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "country_value" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CountryValueEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CountryValueEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit filter action.
             */
            CountryValueEventFactory.prototype.hitFilter = function () {
                var data = {
                    actionName: "filter",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CountryValueEventFactory;
        }());
        CountryFilterDropdownEventFactory.CountryValueEventFactory = CountryValueEventFactory;
    })(CountryFilterDropdownEventFactory = WebAudienceEventFactory.CountryFilterDropdownEventFactory || (WebAudienceEventFactory.CountryFilterDropdownEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceStatsVersionEventFactory;
    (function (AudienceStatsVersionEventFactory) {
        /**
         * Create OverviewTabEventFactory class with path
         * @param path to this node
         */
        var OverviewTabEventFactory = /** @class */ (function () {
            function OverviewTabEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "overview_tab"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            OverviewTabEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            OverviewTabEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            OverviewTabEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return OverviewTabEventFactory;
        }());
        AudienceStatsVersionEventFactory.OverviewTabEventFactory = OverviewTabEventFactory;
    })(AudienceStatsVersionEventFactory = WebAudienceEventFactory.AudienceStatsVersionEventFactory || (WebAudienceEventFactory.AudienceStatsVersionEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceStatsVersionEventFactory;
    (function (AudienceStatsVersionEventFactory) {
        /**
         * Create EngagementTabEventFactory class with path
         * @param path to this node
         */
        var EngagementTabEventFactory = /** @class */ (function () {
            function EngagementTabEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "engagement_tab"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            EngagementTabEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            EngagementTabEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            EngagementTabEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return EngagementTabEventFactory;
        }());
        AudienceStatsVersionEventFactory.EngagementTabEventFactory = EngagementTabEventFactory;
    })(AudienceStatsVersionEventFactory = WebAudienceEventFactory.AudienceStatsVersionEventFactory || (WebAudienceEventFactory.AudienceStatsVersionEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create MetricSelectionEventFactory class with path
         * @param path to this node
         */
        var MetricSelectionEventFactory = /** @class */ (function () {
            function MetricSelectionEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "metric_selection"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of ListenersButtonEventFactory
            */
            MetricSelectionEventFactory.prototype.listenersButtonFactory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.MetricSelectionEventFactory.ListenersButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of StreamsButtonEventFactory
            */
            MetricSelectionEventFactory.prototype.streamsButtonFactory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.MetricSelectionEventFactory.StreamsButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of FollowersButtonEventFactory
            */
            MetricSelectionEventFactory.prototype.followersButtonFactory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.MetricSelectionEventFactory.FollowersButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            MetricSelectionEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            MetricSelectionEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return MetricSelectionEventFactory;
        }());
        AudienceTimelineEventFactory.MetricSelectionEventFactory = MetricSelectionEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create DownloadCsvButtonEventFactory class with path
         * @param path to this node
         */
        var DownloadCsvButtonEventFactory = /** @class */ (function () {
            function DownloadCsvButtonEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "download_csv_button"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            DownloadCsvButtonEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            DownloadCsvButtonEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit download_file action.
             */
            DownloadCsvButtonEventFactory.prototype.hitDownloadFile = function () {
                var data = {
                    actionName: "download_file",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return DownloadCsvButtonEventFactory;
        }());
        AudienceTimelineEventFactory.DownloadCsvButtonEventFactory = DownloadCsvButtonEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create DownloadCsvButtonHoverTextEventFactory class with path
         * @param path to this node
         */
        var DownloadCsvButtonHoverTextEventFactory = /** @class */ (function () {
            function DownloadCsvButtonHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "download_csv_button_hover_text"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            DownloadCsvButtonHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            DownloadCsvButtonHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for DownloadCsvButtonHoverTextEventFactory.
             */
            DownloadCsvButtonHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return DownloadCsvButtonHoverTextEventFactory;
        }());
        AudienceTimelineEventFactory.DownloadCsvButtonHoverTextEventFactory = DownloadCsvButtonHoverTextEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create DatePickerEventFactory class with path
         * @param path to this node
         */
        var DatePickerEventFactory = /** @class */ (function () {
            function DatePickerEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "date_picker"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @returns A new instance of Last7DaysPresetEventFactory
            */
            DatePickerEventFactory.prototype.last7DaysPresetFactory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.DatePickerEventFactory.Last7DaysPresetEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of Last28DaysPresetEventFactory
            */
            DatePickerEventFactory.prototype.last28DaysPresetFactory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.DatePickerEventFactory.Last28DaysPresetEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @returns A new instance of Since2015EventFactory
            */
            DatePickerEventFactory.prototype.since2015Factory = function () {
                return new WebAudienceEventFactory.AudienceTimelineEventFactory.DatePickerEventFactory.Since2015EventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            DatePickerEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            DatePickerEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create interaction for hit ui_reveal action.
             */
            DatePickerEventFactory.prototype.hitUiReveal = function () {
                var data = {
                    actionName: "ui_reveal",
                    actionVersion: 1,
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    interactionType: "hit",
                    specificationMode: SPECIFICATION_MODE,
                    specificationVersion: SPECIFICATION_VERSION,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return DatePickerEventFactory;
        }());
        AudienceTimelineEventFactory.DatePickerEventFactory = DatePickerEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create TimelineChartEventFactory class with path
         * @param path to this node
         */
        var TimelineChartEventFactory = /** @class */ (function () {
            function TimelineChartEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "timeline_chart"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TimelineChartEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TimelineChartEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for TimelineChartEventFactory.
             */
            TimelineChartEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TimelineChartEventFactory;
        }());
        AudienceTimelineEventFactory.TimelineChartEventFactory = TimelineChartEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create TimelineChartHoverTextEventFactory class with path
         * @param path to this node
         */
        var TimelineChartHoverTextEventFactory = /** @class */ (function () {
            function TimelineChartHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "timeline_chart_hover_text" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TimelineChartHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TimelineChartHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for TimelineChartHoverTextEventFactory.
             */
            TimelineChartHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TimelineChartHoverTextEventFactory;
        }());
        AudienceTimelineEventFactory.TimelineChartHoverTextEventFactory = TimelineChartHoverTextEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create EmptyStateEventFactory class with path
         * @param path to this node
         */
        var EmptyStateEventFactory = /** @class */ (function () {
            function EmptyStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "empty_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            EmptyStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            EmptyStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for EmptyStateEventFactory.
             */
            EmptyStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return EmptyStateEventFactory;
        }());
        AudienceTimelineEventFactory.EmptyStateEventFactory = EmptyStateEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        /**
         * Create ErrorStateEventFactory class with path
         * @param path to this node
         */
        var ErrorStateEventFactory = /** @class */ (function () {
            function ErrorStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "error_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ErrorStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ErrorStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ErrorStateEventFactory.
             */
            ErrorStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ErrorStateEventFactory;
        }());
        AudienceTimelineEventFactory.ErrorStateEventFactory = ErrorStateEventFactory;
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var SourceOfStreamsEventFactory;
    (function (SourceOfStreamsEventFactory) {
        /**
         * Create ChartEventFactory class with path
         * @param path to this node
         */
        var ChartEventFactory = /** @class */ (function () {
            function ChartEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "chart"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ChartEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ChartEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ChartEventFactory.
             */
            ChartEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ChartEventFactory;
        }());
        SourceOfStreamsEventFactory.ChartEventFactory = ChartEventFactory;
    })(SourceOfStreamsEventFactory = WebAudienceEventFactory.SourceOfStreamsEventFactory || (WebAudienceEventFactory.SourceOfStreamsEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var SourceOfStreamsEventFactory;
    (function (SourceOfStreamsEventFactory) {
        /**
         * Create ChartHoverTextEventFactory class with path
         * @param path to this node
         */
        var ChartHoverTextEventFactory = /** @class */ (function () {
            function ChartHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "chart_hover_text" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ChartHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ChartHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ChartHoverTextEventFactory.
             */
            ChartHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ChartHoverTextEventFactory;
        }());
        SourceOfStreamsEventFactory.ChartHoverTextEventFactory = ChartHoverTextEventFactory;
    })(SourceOfStreamsEventFactory = WebAudienceEventFactory.SourceOfStreamsEventFactory || (WebAudienceEventFactory.SourceOfStreamsEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var SourceOfStreamsEventFactory;
    (function (SourceOfStreamsEventFactory) {
        /**
         * Create ErrorStateEventFactory class with path
         * @param path to this node
         */
        var ErrorStateEventFactory = /** @class */ (function () {
            function ErrorStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "error_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ErrorStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ErrorStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ErrorStateEventFactory.
             */
            ErrorStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ErrorStateEventFactory;
        }());
        SourceOfStreamsEventFactory.ErrorStateEventFactory = ErrorStateEventFactory;
    })(SourceOfStreamsEventFactory = WebAudienceEventFactory.SourceOfStreamsEventFactory || (WebAudienceEventFactory.SourceOfStreamsEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var SourceOfStreamsEventFactory;
    (function (SourceOfStreamsEventFactory) {
        /**
         * Create EmptyStateEventFactory class with path
         * @param path to this node
         */
        var EmptyStateEventFactory = /** @class */ (function () {
            function EmptyStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "empty_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            EmptyStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            EmptyStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for EmptyStateEventFactory.
             */
            EmptyStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return EmptyStateEventFactory;
        }());
        SourceOfStreamsEventFactory.EmptyStateEventFactory = EmptyStateEventFactory;
    })(SourceOfStreamsEventFactory = WebAudienceEventFactory.SourceOfStreamsEventFactory || (WebAudienceEventFactory.SourceOfStreamsEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create GenderChartEventFactory class with path
         * @param path to this node
         */
        var GenderChartEventFactory = /** @class */ (function () {
            function GenderChartEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "gender_chart"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            GenderChartEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            GenderChartEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            return GenderChartEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.GenderChartEventFactory = GenderChartEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create GenderChartHoverTextEventFactory class with path
         * @param path to this node
         */
        var GenderChartHoverTextEventFactory = /** @class */ (function () {
            function GenderChartHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "gender_chart_hover_text" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            GenderChartHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            GenderChartHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for GenderChartHoverTextEventFactory.
             */
            GenderChartHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return GenderChartHoverTextEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.GenderChartHoverTextEventFactory = GenderChartHoverTextEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create AgeChartEventFactory class with path
         * @param path to this node
         */
        var AgeChartEventFactory = /** @class */ (function () {
            function AgeChartEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "age_chart"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AgeChartEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AgeChartEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for AgeChartEventFactory.
             */
            AgeChartEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AgeChartEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.AgeChartEventFactory = AgeChartEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create AgeChartHoverTextEventFactory class with path
         * @param path to this node
         */
        var AgeChartHoverTextEventFactory = /** @class */ (function () {
            function AgeChartHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "age_chart_hover_text" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            AgeChartHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            AgeChartHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for AgeChartHoverTextEventFactory.
             */
            AgeChartHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return AgeChartHoverTextEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.AgeChartHoverTextEventFactory = AgeChartHoverTextEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create ErrorStateEventFactory class with path
         * @param path to this node
         */
        var ErrorStateEventFactory = /** @class */ (function () {
            function ErrorStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "error_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            ErrorStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            ErrorStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for ErrorStateEventFactory.
             */
            ErrorStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return ErrorStateEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.ErrorStateEventFactory = ErrorStateEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAgeAndGenderEventFactory;
    (function (ListenersAgeAndGenderEventFactory) {
        /**
         * Create EmptyStateEventFactory class with path
         * @param path to this node
         */
        var EmptyStateEventFactory = /** @class */ (function () {
            function EmptyStateEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "empty_state"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            EmptyStateEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            EmptyStateEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for EmptyStateEventFactory.
             */
            EmptyStateEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return EmptyStateEventFactory;
        }());
        ListenersAgeAndGenderEventFactory.EmptyStateEventFactory = EmptyStateEventFactory;
    })(ListenersAgeAndGenderEventFactory = WebAudienceEventFactory.ListenersAgeAndGenderEventFactory || (WebAudienceEventFactory.ListenersAgeAndGenderEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var ListenersAlsoLikeEventFactory;
    (function (ListenersAlsoLikeEventFactory) {
        /**
         * Create SimilarArtistEventFactory class with path
         * @param path to this node
         */
        var SimilarArtistEventFactory = /** @class */ (function () {
            function SimilarArtistEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "similar_artist" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            SimilarArtistEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            SimilarArtistEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for SimilarArtistEventFactory.
             */
            SimilarArtistEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return SimilarArtistEventFactory;
        }());
        ListenersAlsoLikeEventFactory.SimilarArtistEventFactory = SimilarArtistEventFactory;
    })(ListenersAlsoLikeEventFactory = WebAudienceEventFactory.ListenersAlsoLikeEventFactory || (WebAudienceEventFactory.ListenersAlsoLikeEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var TopCountriesEventFactory;
    (function (TopCountriesEventFactory) {
        /**
         * Create TopCountriesListEventFactory class with path
         * @param path to this node
         */
        var TopCountriesListEventFactory = /** @class */ (function () {
            function TopCountriesListEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "top_countries_list"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            TopCountriesListEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            TopCountriesListEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for TopCountriesListEventFactory.
             */
            TopCountriesListEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return TopCountriesListEventFactory;
        }());
        TopCountriesEventFactory.TopCountriesListEventFactory = TopCountriesListEventFactory;
    })(TopCountriesEventFactory = WebAudienceEventFactory.TopCountriesEventFactory || (WebAudienceEventFactory.TopCountriesEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var TopCountriesEventFactory;
    (function (TopCountriesEventFactory) {
        /**
         * Create CountryMapEventFactory class with path
         * @param path to this node
         */
        var CountryMapEventFactory = /** @class */ (function () {
            function CountryMapEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "country_map"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CountryMapEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CountryMapEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for CountryMapEventFactory.
             */
            CountryMapEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CountryMapEventFactory;
        }());
        TopCountriesEventFactory.CountryMapEventFactory = CountryMapEventFactory;
    })(TopCountriesEventFactory = WebAudienceEventFactory.TopCountriesEventFactory || (WebAudienceEventFactory.TopCountriesEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var TopCountriesEventFactory;
    (function (TopCountriesEventFactory) {
        /**
         * Create CountryMapHoverTextEventFactory class with path
         * @param path to this node
         */
        var CountryMapHoverTextEventFactory = /** @class */ (function () {
            function CountryMapHoverTextEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat(__assign(__assign({}, data), { name: "country_map_hover_text" }));
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            CountryMapHoverTextEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            CountryMapHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for CountryMapHoverTextEventFactory.
             */
            CountryMapHoverTextEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return CountryMapHoverTextEventFactory;
        }());
        TopCountriesEventFactory.CountryMapHoverTextEventFactory = CountryMapHoverTextEventFactory;
    })(TopCountriesEventFactory = WebAudienceEventFactory.TopCountriesEventFactory || (WebAudienceEventFactory.TopCountriesEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var EngagementEventFactory;
    (function (EngagementEventFactory) {
        /**
         * Create StatsThisPeriodEventFactory class with path
         * @param path to this node
         */
        var StatsThisPeriodEventFactory = /** @class */ (function () {
            function StatsThisPeriodEventFactory(_a) {
                var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                this._path = path.concat({
                    name: "stats_this_period"
                });
                if (parentAbsoluteLocation) {
                    this._parentAbsoluteLocation = parentAbsoluteLocation;
                }
            }
            /**
            * @param id of the MetricSelectionEventFactory
            * @returns A new instance of MetricSelectionEventFactory
            */
            StatsThisPeriodEventFactory.prototype.metricSelectionFactory = function (data) {
                return new WebAudienceEventFactory.EngagementEventFactory.StatsThisPeriodEventFactory.MetricSelectionEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            /**
            * @returns A new instance of MetricTooltipButtonEventFactory
            */
            StatsThisPeriodEventFactory.prototype.metricTooltipButtonFactory = function () {
                return new WebAudienceEventFactory.EngagementEventFactory.StatsThisPeriodEventFactory.MetricTooltipButtonEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                });
            };
            /**
            * @param id of the TimelineChartEventFactory
            * @returns A new instance of TimelineChartEventFactory
            */
            StatsThisPeriodEventFactory.prototype.timelineChartFactory = function (data) {
                return new WebAudienceEventFactory.EngagementEventFactory.StatsThisPeriodEventFactory.TimelineChartEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            /**
            * @param id of the TimelineChartHoverTextEventFactory
            * @returns A new instance of TimelineChartHoverTextEventFactory
            */
            StatsThisPeriodEventFactory.prototype.timelineChartHoverTextFactory = function (data) {
                return new WebAudienceEventFactory.EngagementEventFactory.StatsThisPeriodEventFactory.TimelineChartHoverTextEventFactory({
                    path: this._path,
                    parentAbsoluteLocation: this._parentAbsoluteLocation,
                    data: data
                });
            };
            StatsThisPeriodEventFactory.prototype._getCurrentLocation = function () {
                return {
                    pathNodes: this._path,
                    specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                    specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                };
            };
            StatsThisPeriodEventFactory.prototype.getAbsoluteLocation = function () {
                var _a;
                var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                return {
                    locations: __spreadArrays((parentLocations || []), [
                        this._getCurrentLocation()
                    ]),
                };
            };
            /**
             * Create impression for StatsThisPeriodEventFactory.
             */
            StatsThisPeriodEventFactory.prototype.impression = function () {
                var data = {
                    app: APP,
                    generatorVersion: GENERATOR_VERSION,
                    specificationVersion: SPECIFICATION_VERSION,
                    specificationMode: SPECIFICATION_MODE,
                    annotatorConfigurationVersion: '',
                    annotatorVersion: '',
                    location: this._getCurrentLocation(),
                    parentAbsoluteLocation: this._parentAbsoluteLocation
                };
                return data;
            };
            return StatsThisPeriodEventFactory;
        }());
        EngagementEventFactory.StatsThisPeriodEventFactory = StatsThisPeriodEventFactory;
    })(EngagementEventFactory = WebAudienceEventFactory.EngagementEventFactory || (WebAudienceEventFactory.EngagementEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var MetricSelectionEventFactory;
        (function (MetricSelectionEventFactory) {
            /**
             * Create ListenersButtonEventFactory class with path
             * @param path to this node
             */
            var ListenersButtonEventFactory = /** @class */ (function () {
                function ListenersButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "listeners_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                ListenersButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                ListenersButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                ListenersButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return ListenersButtonEventFactory;
            }());
            MetricSelectionEventFactory.ListenersButtonEventFactory = ListenersButtonEventFactory;
        })(MetricSelectionEventFactory = AudienceTimelineEventFactory.MetricSelectionEventFactory || (AudienceTimelineEventFactory.MetricSelectionEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var MetricSelectionEventFactory;
        (function (MetricSelectionEventFactory) {
            /**
             * Create StreamsButtonEventFactory class with path
             * @param path to this node
             */
            var StreamsButtonEventFactory = /** @class */ (function () {
                function StreamsButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "streams_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                StreamsButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                StreamsButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                StreamsButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return StreamsButtonEventFactory;
            }());
            MetricSelectionEventFactory.StreamsButtonEventFactory = StreamsButtonEventFactory;
        })(MetricSelectionEventFactory = AudienceTimelineEventFactory.MetricSelectionEventFactory || (AudienceTimelineEventFactory.MetricSelectionEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var MetricSelectionEventFactory;
        (function (MetricSelectionEventFactory) {
            /**
             * Create FollowersButtonEventFactory class with path
             * @param path to this node
             */
            var FollowersButtonEventFactory = /** @class */ (function () {
                function FollowersButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "followers_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                FollowersButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                FollowersButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                FollowersButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return FollowersButtonEventFactory;
            }());
            MetricSelectionEventFactory.FollowersButtonEventFactory = FollowersButtonEventFactory;
        })(MetricSelectionEventFactory = AudienceTimelineEventFactory.MetricSelectionEventFactory || (AudienceTimelineEventFactory.MetricSelectionEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var DatePickerEventFactory;
        (function (DatePickerEventFactory) {
            /**
             * Create Last7DaysPresetEventFactory class with path
             * @param path to this node
             */
            var Last7DaysPresetEventFactory = /** @class */ (function () {
                function Last7DaysPresetEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "last_7_days_preset"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                Last7DaysPresetEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                Last7DaysPresetEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit filter action.
                 */
                Last7DaysPresetEventFactory.prototype.hitFilter = function () {
                    var data = {
                        actionName: "filter",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return Last7DaysPresetEventFactory;
            }());
            DatePickerEventFactory.Last7DaysPresetEventFactory = Last7DaysPresetEventFactory;
        })(DatePickerEventFactory = AudienceTimelineEventFactory.DatePickerEventFactory || (AudienceTimelineEventFactory.DatePickerEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var DatePickerEventFactory;
        (function (DatePickerEventFactory) {
            /**
             * Create Last28DaysPresetEventFactory class with path
             * @param path to this node
             */
            var Last28DaysPresetEventFactory = /** @class */ (function () {
                function Last28DaysPresetEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "last_28_days_preset"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                Last28DaysPresetEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                Last28DaysPresetEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit filter action.
                 */
                Last28DaysPresetEventFactory.prototype.hitFilter = function () {
                    var data = {
                        actionName: "filter",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return Last28DaysPresetEventFactory;
            }());
            DatePickerEventFactory.Last28DaysPresetEventFactory = Last28DaysPresetEventFactory;
        })(DatePickerEventFactory = AudienceTimelineEventFactory.DatePickerEventFactory || (AudienceTimelineEventFactory.DatePickerEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var AudienceTimelineEventFactory;
    (function (AudienceTimelineEventFactory) {
        var DatePickerEventFactory;
        (function (DatePickerEventFactory) {
            /**
             * Create Since2015EventFactory class with path
             * @param path to this node
             */
            var Since2015EventFactory = /** @class */ (function () {
                function Since2015EventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "since_2015"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                Since2015EventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                Since2015EventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit filter action.
                 */
                Since2015EventFactory.prototype.hitFilter = function () {
                    var data = {
                        actionName: "filter",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return Since2015EventFactory;
            }());
            DatePickerEventFactory.Since2015EventFactory = Since2015EventFactory;
        })(DatePickerEventFactory = AudienceTimelineEventFactory.DatePickerEventFactory || (AudienceTimelineEventFactory.DatePickerEventFactory = {}));
    })(AudienceTimelineEventFactory = WebAudienceEventFactory.AudienceTimelineEventFactory || (WebAudienceEventFactory.AudienceTimelineEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var EngagementEventFactory;
    (function (EngagementEventFactory) {
        var StatsThisPeriodEventFactory;
        (function (StatsThisPeriodEventFactory) {
            /**
             * Create MetricSelectionEventFactory class with path
             * @param path to this node
             */
            var MetricSelectionEventFactory = /** @class */ (function () {
                function MetricSelectionEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "metric_selection" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                MetricSelectionEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                MetricSelectionEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                MetricSelectionEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return MetricSelectionEventFactory;
            }());
            StatsThisPeriodEventFactory.MetricSelectionEventFactory = MetricSelectionEventFactory;
        })(StatsThisPeriodEventFactory = EngagementEventFactory.StatsThisPeriodEventFactory || (EngagementEventFactory.StatsThisPeriodEventFactory = {}));
    })(EngagementEventFactory = WebAudienceEventFactory.EngagementEventFactory || (WebAudienceEventFactory.EngagementEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var EngagementEventFactory;
    (function (EngagementEventFactory) {
        var StatsThisPeriodEventFactory;
        (function (StatsThisPeriodEventFactory) {
            /**
             * Create MetricTooltipButtonEventFactory class with path
             * @param path to this node
             */
            var MetricTooltipButtonEventFactory = /** @class */ (function () {
                function MetricTooltipButtonEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat({
                        name: "metric_tooltip_button"
                    });
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                MetricTooltipButtonEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                MetricTooltipButtonEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create interaction for hit ui_reveal action.
                 */
                MetricTooltipButtonEventFactory.prototype.hitUiReveal = function () {
                    var data = {
                        actionName: "ui_reveal",
                        actionVersion: 1,
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        interactionType: "hit",
                        specificationMode: SPECIFICATION_MODE,
                        specificationVersion: SPECIFICATION_VERSION,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return MetricTooltipButtonEventFactory;
            }());
            StatsThisPeriodEventFactory.MetricTooltipButtonEventFactory = MetricTooltipButtonEventFactory;
        })(StatsThisPeriodEventFactory = EngagementEventFactory.StatsThisPeriodEventFactory || (EngagementEventFactory.StatsThisPeriodEventFactory = {}));
    })(EngagementEventFactory = WebAudienceEventFactory.EngagementEventFactory || (WebAudienceEventFactory.EngagementEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var EngagementEventFactory;
    (function (EngagementEventFactory) {
        var StatsThisPeriodEventFactory;
        (function (StatsThisPeriodEventFactory) {
            /**
             * Create TimelineChartEventFactory class with path
             * @param path to this node
             */
            var TimelineChartEventFactory = /** @class */ (function () {
                function TimelineChartEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "timeline_chart" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TimelineChartEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TimelineChartEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create impression for TimelineChartEventFactory.
                 */
                TimelineChartEventFactory.prototype.impression = function () {
                    var data = {
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        specificationVersion: SPECIFICATION_VERSION,
                        specificationMode: SPECIFICATION_MODE,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TimelineChartEventFactory;
            }());
            StatsThisPeriodEventFactory.TimelineChartEventFactory = TimelineChartEventFactory;
        })(StatsThisPeriodEventFactory = EngagementEventFactory.StatsThisPeriodEventFactory || (EngagementEventFactory.StatsThisPeriodEventFactory = {}));
    })(EngagementEventFactory = WebAudienceEventFactory.EngagementEventFactory || (WebAudienceEventFactory.EngagementEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
(function (WebAudienceEventFactory) {
    var EngagementEventFactory;
    (function (EngagementEventFactory) {
        var StatsThisPeriodEventFactory;
        (function (StatsThisPeriodEventFactory) {
            /**
             * Create TimelineChartHoverTextEventFactory class with path
             * @param path to this node
             */
            var TimelineChartHoverTextEventFactory = /** @class */ (function () {
                function TimelineChartHoverTextEventFactory(_a) {
                    var path = _a.path, parentAbsoluteLocation = _a.parentAbsoluteLocation, data = _a.data;
                    this._path = path.concat(__assign(__assign({}, data), { name: "timeline_chart_hover_text" }));
                    if (parentAbsoluteLocation) {
                        this._parentAbsoluteLocation = parentAbsoluteLocation;
                    }
                }
                TimelineChartHoverTextEventFactory.prototype._getCurrentLocation = function () {
                    return {
                        pathNodes: this._path,
                        specMode: __spreadArrays([SPECIFICATION_MODE], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; })),
                        specVersion: __spreadArrays([SPECIFICATION_VERSION], __spreadArrays(Array(this._path.length - 1)).map(function () { return ''; }))
                    };
                };
                TimelineChartHoverTextEventFactory.prototype.getAbsoluteLocation = function () {
                    var _a;
                    var parentLocations = (_a = this._parentAbsoluteLocation) === null || _a === void 0 ? void 0 : _a.locations;
                    return {
                        locations: __spreadArrays((parentLocations || []), [
                            this._getCurrentLocation()
                        ]),
                    };
                };
                /**
                 * Create impression for TimelineChartHoverTextEventFactory.
                 */
                TimelineChartHoverTextEventFactory.prototype.impression = function () {
                    var data = {
                        app: APP,
                        generatorVersion: GENERATOR_VERSION,
                        specificationVersion: SPECIFICATION_VERSION,
                        specificationMode: SPECIFICATION_MODE,
                        annotatorConfigurationVersion: '',
                        annotatorVersion: '',
                        location: this._getCurrentLocation(),
                        parentAbsoluteLocation: this._parentAbsoluteLocation
                    };
                    return data;
                };
                return TimelineChartHoverTextEventFactory;
            }());
            StatsThisPeriodEventFactory.TimelineChartHoverTextEventFactory = TimelineChartHoverTextEventFactory;
        })(StatsThisPeriodEventFactory = EngagementEventFactory.StatsThisPeriodEventFactory || (EngagementEventFactory.StatsThisPeriodEventFactory = {}));
    })(EngagementEventFactory = WebAudienceEventFactory.EngagementEventFactory || (WebAudienceEventFactory.EngagementEventFactory = {}));
})(WebAudienceEventFactory = exports.WebAudienceEventFactory || (exports.WebAudienceEventFactory = {}));
exports.WebAudienceEventFactory = WebAudienceEventFactory;
