// @spotify-internal/adstudio-shared/lib/components/IntlFormatNumber/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.IntlFormatNumber = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _constants = require("./constants");

var _i18n = require("../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IntlFormatNumber = function IntlFormatNumber(_ref) {
  var number = _ref.number,
      _ref$countryCode = _ref.countryCode,
      countryCode = _ref$countryCode === void 0 ? _i18n.DEFAULT_COUNTRY : _ref$countryCode,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? _constants.DECIMAL : _ref$type,
      _ref$useGrouping = _ref.useGrouping,
      useGrouping = _ref$useGrouping === void 0 ? true : _ref$useGrouping,
      minimumFractionDigits = _ref.minimumFractionDigits,
      maximumFractionDigits = _ref.maximumFractionDigits;
  var formatter = new Intl.NumberFormat(countryCode, {
    style: type,
    useGrouping: useGrouping,
    minimumFractionDigits: type === _constants.PERCENT ? 2 : minimumFractionDigits,
    maximumFractionDigits: type === _constants.PERCENT ? 2 : maximumFractionDigits
  }).format;
  var value = type === _constants.PERCENT ? number / 100 : number;
  return /*#__PURE__*/_react["default"].createElement("span", null, formatter(value));
};

exports.IntlFormatNumber = IntlFormatNumber;
IntlFormatNumber.DECIMAL = _constants.DECIMAL;
IntlFormatNumber.PERCENT = _constants.PERCENT;
IntlFormatNumber.propTypes = {
  countryCode: _propTypes["default"].string,
  maximumFractionDigits: _propTypes["default"].number,
  minimumFractionDigits: _propTypes["default"].number,
  number: _propTypes["default"].oneOf([_propTypes["default"].string, _propTypes["default"].number]).isRequired,
  type: _propTypes["default"].oneOf([_constants.PERCENT, _constants.DECIMAL]),
  useGrouping: _propTypes["default"].bool
};
var _default = IntlFormatNumber;
exports["default"] = _default;
