// @spotify-internal/betamax-sdk-desktop/esm/src/controller-x/VideoContextPlayerCoordinatorXObserver.js
import { EndReason } from '..';
var VideoContextPlayerCoordinatorXObserver = (function () {
    function VideoContextPlayerCoordinatorXObserver(coordinator) {
        this.cosmosSender = coordinator;
    }
    VideoContextPlayerCoordinatorXObserver.prototype.onPlaybackCreated = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onPlaybackEnded = function (reason) {
        this.cosmosSender.sendPlaybackState();
        if (reason === EndReason.PLAYED_TO_END) {
            this.cosmosSender.sendAdvance('finished');
        }
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onAudioFormatChanged = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onBuffering = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onBytesTransferred = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onDroppedVideoFrames = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onDurationChanged = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onEncryptionKeyLoaded = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onEncryptionKeyLoading = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onEndingPlayback = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onFatalError = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onFullscreenChanged = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onManifestLoaded = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onManifestLoading = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onPaused = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onReady = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onRecoverableError = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onResumed = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onSeeking = function () {
        this.cosmosSender.sendPlaybackState();
    };
    VideoContextPlayerCoordinatorXObserver.prototype.onSubtitleChanged = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onVideoFormatChanged = function () { };
    VideoContextPlayerCoordinatorXObserver.prototype.onVolumeChanged = function () { };
    return VideoContextPlayerCoordinatorXObserver;
}());
export default VideoContextPlayerCoordinatorXObserver;
