// @spotify-internal/adstudio-shared/lib/components/AdTargeting/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SHOW_ALL_TEXT = exports.TOOLTIP_DESCRIPTION = exports.LOCATIONS_DESCRIPTION = exports.PLATFORM_DESCRIPTION = exports.TARGET_BASED_TOOLTIP = exports.TARGET_BASED_DESCRIPTION = exports.SEARCH_FOR_LOCATION = exports.SELECT_ONE_LOCATION = exports.SELECT_ONE_PLATFORM = exports.DESKTOP_TOOLTIP_MESSAGE = exports.PLATFORMS = exports.PLACEHOLDERS = exports.LABELS = exports.HEADERS = exports.SELECT_GENDER = exports.SUPPORTED_GENDERS = exports.NO_GENDERS = exports.MALE_GENDER = exports.FEMALE_GENDER = exports.ALL_GENDERS = exports.GENDER = exports.LISTENING_BEHAVIOR = exports.AUDIENCE = exports.KEY_FIELD = void 0;

var _HEADERS, _GENDER, _LABELS;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var KEY_FIELD = {
  ID: 'id',
  GEO_ID: 'geoId'
};
exports.KEY_FIELD = KEY_FIELD;
var AUDIENCE = 'audience';
exports.AUDIENCE = AUDIENCE;
var LISTENING_BEHAVIOR = 'listening-behavior';
exports.LISTENING_BEHAVIOR = LISTENING_BEHAVIOR;
var GENDER = 'Gender';
exports.GENDER = GENDER;
var ALL_GENDERS = 'All';
exports.ALL_GENDERS = ALL_GENDERS;
var FEMALE_GENDER = 'Female';
exports.FEMALE_GENDER = FEMALE_GENDER;
var MALE_GENDER = 'Male'; // This was added to hack Redux forms.

exports.MALE_GENDER = MALE_GENDER;
var NO_GENDERS = 'None';
exports.NO_GENDERS = NO_GENDERS;
var SUPPORTED_GENDERS = [FEMALE_GENDER, MALE_GENDER];
exports.SUPPORTED_GENDERS = SUPPORTED_GENDERS;
var SELECT_GENDER = 'Please select at least one gender.';
exports.SELECT_GENDER = SELECT_GENDER;
var HEADERS = (_HEADERS = {}, _defineProperty(_HEADERS, AUDIENCE, 'Audience'), _defineProperty(_HEADERS, LISTENING_BEHAVIOR, 'Listening behavior'), _HEADERS);
exports.HEADERS = HEADERS;
var LABELS = (_LABELS = {}, _defineProperty(_LABELS, AUDIENCE, {
  Locations: 'Location',
  Age: 'Age',
  Gender: 'Gender'
}), _defineProperty(_LABELS, GENDER, (_GENDER = {}, _defineProperty(_GENDER, MALE_GENDER, 'Male'), _defineProperty(_GENDER, FEMALE_GENDER, 'Female'), _GENDER)), _defineProperty(_LABELS, LISTENING_BEHAVIOR, {
  Genres: 'Genres',
  Playlist: 'Playlist',
  Fans: 'Fans',
  Platform: 'Platform',
  Interests: 'Interests'
}), _LABELS);
exports.LABELS = LABELS;

var PLACEHOLDERS = _defineProperty({}, LISTENING_BEHAVIOR, {
  Genres: 'E.g., Hip hop, Indie',
  Playlist: 'E.g., Party, Focus',
  Fans: 'Choose artist(s)'
});

exports.PLACEHOLDERS = PLACEHOLDERS;
var PLATFORMS = {
  ios: 'iOS (iPhone/iPad)',
  android: 'Android',
  desktop: 'Desktop'
};
exports.PLATFORMS = PLATFORMS;
var DESKTOP_TOOLTIP_MESSAGE = 'To target listeners on desktop, select horizontal video or audio.';
exports.DESKTOP_TOOLTIP_MESSAGE = DESKTOP_TOOLTIP_MESSAGE;
var SELECT_ONE_PLATFORM = 'Please select at least one platform.';
exports.SELECT_ONE_PLATFORM = SELECT_ONE_PLATFORM;
var SELECT_ONE_LOCATION = 'Please select at least one location and remove any invalid locations.';
exports.SELECT_ONE_LOCATION = SELECT_ONE_LOCATION;
var SEARCH_FOR_LOCATION = 'Search for location';
exports.SEARCH_FOR_LOCATION = SEARCH_FOR_LOCATION;
var TARGET_BASED_DESCRIPTION = "You can further target your ad based on your audience's music taste. This will impact reach.";
exports.TARGET_BASED_DESCRIPTION = TARGET_BASED_DESCRIPTION;
var TARGET_BASED_TOOLTIP = 'Targeting based on genre, playlist, or fan will further narrow the reach of your ad. If you don\'t know your audience\'s listening behavior, we suggest leaving "All music" selected.';
exports.TARGET_BASED_TOOLTIP = TARGET_BASED_TOOLTIP;
var PLATFORM_DESCRIPTION = 'We recommend choosing all platforms to maximize your potential audience reach, unless your ad is platform specific.';
exports.PLATFORM_DESCRIPTION = PLATFORM_DESCRIPTION;
var LOCATIONS_DESCRIPTION = 'Reach listeners in a specific country, state, city, or DMA.';
exports.LOCATIONS_DESCRIPTION = LOCATIONS_DESCRIPTION;
var TOOLTIP_DESCRIPTION = 'A DMA (Designated Market Area) is a group of cities that form a broader metropolitan area. For example, the Los Angeles DMA includes the city of Los Angeles as well as the neighboring towns and counties.';
exports.TOOLTIP_DESCRIPTION = TOOLTIP_DESCRIPTION;
var SHOW_ALL_TEXT = 'All';
exports.SHOW_ALL_TEXT = SHOW_ALL_TEXT;
