// @spotify-internal/adstudio-tape/lib/components/MobilePreview/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _encoreWeb = require("@spotify-internal/encore-web");

var _PlaybackControls = _interopRequireDefault(require("./styled/PlaybackControls"));

var _ProgressBar = _interopRequireDefault(require("./styled/ProgressBar"));

var _Header = _interopRequireDefault(require("./styled/Header"));

var _CTAButton = _interopRequireDefault(require("./styled/CTAButton"));

var _ImageContainer = _interopRequireDefault(require("./styled/ImageContainer"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var PreviewBody = _styledComponents["default"].div.withConfig({
  displayName: "MobilePreview__PreviewBody",
  componentId: "sc-19zh7r2-0"
})(["background:", ";border-radius:4px;color:white;height:482px;margin:", ";text-align:center;width:271px;"], _encoreFoundation.black, function (props) {
  return props.showPanel ? '0 auto' : '0';
});

var Panel = _styledComponents["default"].div.withConfig({
  displayName: "MobilePreview__Panel",
  componentId: "sc-19zh7r2-1"
})(["border-radius:4px;box-shadow:0 3px 9px 0 ", ";padding:20px 0;width:300px;"], _encoreFoundation.gray90);

var PodcastPanel = _styledComponents["default"].div.withConfig({
  displayName: "MobilePreview__PodcastPanel",
  componentId: "sc-19zh7r2-2"
})(["text-align:left;margin-top:", ";margin-left:", ";"], _encoreFoundation.spacer16, _encoreFoundation.spacer16);

var MobilePreview = function MobilePreview(_ref) {
  var title = _ref.title,
      headline = _ref.headline,
      tagLine = _ref.tagLine,
      imageUrl = _ref.imageUrl,
      podcastSponsorMessage = _ref.podcastSponsorMessage,
      podcastName = _ref.podcastName,
      ctaText = _ref.ctaText,
      showSkipControls = _ref.showSkipControls,
      showPanel = _ref.showPanel,
      showCtaText = _ref.showCtaText,
      showPodcastControls = _ref.showPodcastControls,
      imageTooltipMessage = _ref.imageTooltipMessage,
      children = _ref.children;

  var body = /*#__PURE__*/_react["default"].createElement(PreviewBody, {
    showPanel: showPanel
  }, /*#__PURE__*/_react["default"].createElement(_Header["default"], {
    title: title,
    headline: headline,
    tagLine: tagLine
  }), /*#__PURE__*/_react["default"].createElement(_ImageContainer["default"], {
    imageTooltipMessage: imageTooltipMessage,
    imageUrl: imageUrl
  }, children), showPodcastControls && /*#__PURE__*/_react["default"].createElement(PodcastPanel, null, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    variant: _encoreWeb.Type.body2,
    weight: _encoreWeb.Type.book,
    condensed: !!podcastSponsorMessage
  }, podcastSponsorMessage), /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    color: _encoreFoundation.gray75,
    variant: _encoreWeb.Type.body4,
    weight: _encoreWeb.Type.book,
    condensed: !!podcastName
  }, podcastName)), showCtaText && /*#__PURE__*/_react["default"].createElement(_CTAButton["default"], {
    ctaText: ctaText
  }), /*#__PURE__*/_react["default"].createElement(_ProgressBar["default"], null), /*#__PURE__*/_react["default"].createElement(_PlaybackControls["default"], {
    showSkipControls: showSkipControls,
    showPodcastControls: showPodcastControls
  }));

  if (showPanel) return /*#__PURE__*/_react["default"].createElement(Panel, null, body);
  return body;
};

MobilePreview.propTypes = {
  title: _propTypes["default"].string,
  headline: _propTypes["default"].string,
  tagLine: _propTypes["default"].string,
  imageUrl: _propTypes["default"].string,
  ctaText: _propTypes["default"].string,
  podcastSponsorMessage: _propTypes["default"].string,
  podcastName: _propTypes["default"].string,
  showSkipControls: _propTypes["default"].bool,
  showPanel: _propTypes["default"].bool,
  showCtaText: _propTypes["default"].bool,
  showPodcastControls: _propTypes["default"].bool,
  imageTooltipMessage: _propTypes["default"].string,
  children: _propTypes["default"].node
};
MobilePreview.defaultProps = {
  title: 'Advertisement',
  ctaText: 'Learn More',
  showPanel: true,
  showSkipControls: false,
  showCtaText: true
};
var _default = MobilePreview;
exports["default"] = _default;
