// @spotify-internal/encore-web/es/components/NavigationListItem/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React from 'react';
import { Link } from "./Link";
import { ListItem } from "./ListItem";
export function NavigationListItem(_ref) {
  var active = _ref.active,
      disabled = _ref.disabled,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      restProps = _objectWithoutProperties(_ref, ["active", "disabled", "component"]);

  return /*#__PURE__*/React.createElement(ListItem, null, /*#__PURE__*/React.createElement(Link, Object.assign({}, restProps, {
    active: active,
    disabled: disabled,
    "aria-disabled": disabled,
    rel: restProps.href ? 'noopener noreferrer' : undefined,
    "aria-current": active ? 'page' : undefined,
    component: component
  })));
}
/**
* **NavigationListItem** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavigationListItem) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navigation-navigationlistitem--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=889:71) |
*
* NavigationListItem is an individual link within the NavigationList component
*
* @example
* export const Default = () => (
*  <Fragment>
*    <NavigationListItem>NavigationListItem</NavigationListItem>
*    <NavigationListItem active>NavigationListItem</NavigationListItem>
*    <NavigationListItem disabled>NavigationListItem</NavigationListItem>
*    <NavigationListItem>NavigationListItem</NavigationListItem>
*  </Fragment>
*);
*
*/

export default NavigationListItem;
