// @spotify-internal/adstudio-shared/lib/components/GeoTargetingSearch/Suggestion/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _selectors = require("../selectors");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Suggestion = _styledComponents["default"].div.withConfig({
  displayName: "Suggestion",
  componentId: "sc-1xher07-0"
})(["margin:0;display:grid;grid-template-columns:180px 1fr;align-items:center;.highlight{color:", ";font-weight:300;position:relative;}span:first-child{text-transform:capitalize;}"], _encoreAdvertisingWeb.gray75);

var PaddingLeftEmpty = _styledComponents["default"].div.withConfig({
  displayName: "Suggestion__PaddingLeftEmpty",
  componentId: "sc-1xher07-1"
})(["background-color:", ";display:grid;padding-left:0;width:100%;height:100%;text-align:right;align-items:center;&& p{text-align:right;margin-bottom:0;margin-top:0;padding:14px 16px;}"], _encoreAdvertisingWeb.gray95);

var LocationType = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "Suggestion__LocationType",
  componentId: "sc-1xher07-2"
})(["background-clip:content-box;text-transform:capitalize;.dma-geo-type{text-transform:uppercase;}"]);

var Highlight = _styledComponents["default"].span.withConfig({
  displayName: "Suggestion__Highlight",
  componentId: "sc-1xher07-3"
})(["", ""], function (_ref) {
  var highlight = _ref.highlight;
  return highlight && "\n        color: ".concat(_encoreAdvertisingWeb.gray75, ";\n      ");
});

var SuggestionName = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "Suggestion__SuggestionName",
  componentId: "sc-1xher07-4"
})(["padding:14px 16px;"]);

var LocationSuggestion = function LocationSuggestion(_ref2) {
  var suggestion = _ref2.suggestion,
      parts = _ref2.parts,
      strings = _ref2.strings;
  return /*#__PURE__*/_react["default"].createElement(Suggestion, {
    "data-test": "location-autosuggest-option"
  }, /*#__PURE__*/_react["default"].createElement(PaddingLeftEmpty, null, /*#__PURE__*/_react["default"].createElement(LocationType, {
    condensed: true,
    variant: _encoreAdvertisingWeb.Type.body2,
    color: _encoreAdvertisingWeb.gray40
  }, (0, _selectors.getMappedGeoType)(suggestion.type, strings))), /*#__PURE__*/_react["default"].createElement(SuggestionName, {
    variant: _encoreAdvertisingWeb.Type.body1,
    condensed: true
  }, parts && parts.map(function (_ref3, i) {
    var highlight = _ref3.highlight,
        text = _ref3.text;
    return /*#__PURE__*/_react["default"].createElement(Highlight, {
      highlight: highlight,
      key: i
    }, text);
  }), !parts && /*#__PURE__*/_react["default"].createElement("span", null, suggestion.name)));
};

LocationSuggestion.propTypes = {
  parts: _propTypes["default"].array,
  strings: _propTypes["default"].object,
  suggestion: _propTypes["default"].shape({
    type: _propTypes["default"].string.isRequired,
    name: _propTypes["default"].string,
    typeDisplayName: _propTypes["default"].string
  })
};
var _default = LocationSuggestion;
exports["default"] = _default;
