// @spotify-internal/encore-web/es/styles/theme.js
import { useContext } from 'react';
import { black, gray20, gray85, green, white } from '@spotify-internal/encore-foundation';
import getColor from 'color';
import { ThemeContext } from 'styled-components';
export var colors = {
  primaryColor: 'primaryColor',
  primaryColorText: 'primaryColorText',
  primaryColorActive: 'primaryColorActive',
  primaryColorActiveText: 'primaryColorActiveText',
  primaryColorHover: 'primaryColorHover',
  primaryColorFocus: 'primaryColorFocus',
  primaryColorFocusText: 'primaryColorFocusText',
  primaryColorSelected: 'primaryColorSelected',
  primaryColorSelectedText: 'primaryColorSelectedText',
  primaryColorOutline: 'primaryColorOutline',
  primaryColorDisabledLight: 'primaryColorDisabledLight',
  primaryColorDisabledDark: 'primaryColorDisabledDark'
};

var getTextColor = function getTextColor(value) {
  var convertedHex = getColor(value);
  return convertedHex.isLight() ? black : white;
};

var getFocusTextColor = function getFocusTextColor(value) {
  var convertedHex = getColor(value);
  return convertedHex.isLight() ? getColor('black').alpha(0.9).rgb().string() : getColor('white').alpha(0.8).rgb().string();
};

var getActiveTextColor = function getActiveTextColor(value) {
  var convertedHex = getColor(value);
  return convertedHex.isLight() ? getColor('black').alpha(0.5).rgb().string() : getColor('white').alpha(0.5).rgb().string();
};

export var colorThemeGenerator = function colorThemeGenerator(_ref) {
  var primaryColor = _ref.primaryColor,
      primaryColorText = _ref.primaryColorText,
      primaryColorHover = _ref.primaryColorHover,
      primaryColorActive = _ref.primaryColorActive,
      primaryColorActiveText = _ref.primaryColorActiveText,
      primaryColorFocus = _ref.primaryColorFocus,
      primaryColorFocusText = _ref.primaryColorFocusText,
      primaryColorOutline = _ref.primaryColorOutline,
      primaryColorSelected = _ref.primaryColorSelected,
      primaryColorSelectedText = _ref.primaryColorSelectedText,
      primaryColorDisabledLight = _ref.primaryColorDisabledLight,
      primaryColorDisabledDark = _ref.primaryColorDisabledDark;

  if (!primaryColor) {
    return {};
  }

  return {
    colors: {
      primaryColor: primaryColor,
      primaryColorText: primaryColorText || getTextColor(primaryColor),
      primaryColorHover: primaryColorHover || getColor(primaryColor).mix(getColor('white'), 0.25).hex(),
      primaryColorActive: primaryColorActive || getColor(primaryColor).mix(getColor('black'), 0.3).hex(),
      primaryColorActiveText: primaryColorActiveText || getActiveTextColor(primaryColor),
      primaryColorFocus: primaryColorFocus || getColor(primaryColor).mix(getColor('black'), 0.15).hex(),
      primaryColorFocusText: primaryColorFocusText || getFocusTextColor(primaryColor),
      primaryColorOutline: primaryColorOutline || getColor(primaryColor).alpha(0.2).rgb().string(),
      primaryColorSelected: primaryColorSelected || getColor(primaryColor).mix(getColor('black'), 0.52).hex(),
      primaryColorSelectedText: primaryColorSelectedText || white,
      primaryColorDisabledLight: primaryColorDisabledLight || getColor(primaryColor).mix(getColor(gray85), 0.8).hex(),
      primaryColorDisabledDark: primaryColorDisabledDark || getColor(primaryColor).mix(getColor(gray20), 0.8).hex()
    }
  };
};
export var getColorOrDefault = function getColorOrDefault(theme, color, defaultColor) {
  return theme && theme.colors && theme.colors[color] || defaultColor;
};
/* Export hook for theme usage outside of styled-components */

export var usePrimaryColor = function usePrimaryColor() {
  var defaultColor = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : green;
  var themeContext = useContext(ThemeContext);
  return themeContext && themeContext.colors && themeContext.colors.primaryColor || defaultColor;
};
