// @spotify-internal/checkout-sdk/esm/components/FormLayout/Input/mask.js
import { applyBirthdateMask, handleBirthdateBackspace, validateBirthdateMask, } from './birthdateMask';
import { applyTwoDigitPinMask, validateTwoDigitPinMask, } from './twoDigitPinMask';
export var Mask;
(function (Mask) {
    Mask["Birthdate"] = "birthdate";
    Mask["TwoDigitPin"] = "two_digit_pin";
})(Mask || (Mask = {}));
export var handleBackspace = function (inputMask, event) {
    var text = event.target.value;
    switch (inputMask) {
        case Mask.Birthdate:
            text = handleBirthdateBackspace(text);
            break;
        default:
    }
    return text;
};
export var getValueByMask = function (inputMask, event) {
    var maskedText = event.target.value;
    switch (inputMask) {
        case Mask.Birthdate:
            maskedText = applyBirthdateMask(maskedText);
            break;
        case Mask.TwoDigitPin:
            maskedText = applyTwoDigitPinMask(maskedText);
            break;
        default:
    }
    return maskedText;
};
export var validateMask = function (rule, inputText) {
    switch (rule.value) {
        case Mask.Birthdate:
            return validateBirthdateMask(inputText, rule.errorMessage);
        case Mask.TwoDigitPin:
            return validateTwoDigitPinMask(inputText, rule.errorMessage);
        default:
            return '';
    }
};
