// @spotify-internal/encore-web/es/components/UnsupportedBrowser/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["className", "colorSet", "children"];
import React from 'react';
import { spacer24, spacer64 } from '@spotify-internal/encore-foundation';
import styled from 'styled-components';
import { addColorSet, defaultTheme, rootStyle, zIndexFixed } from "../../styles";
export var Wrapper = styled.div.withConfig({
  displayName: "UnsupportedBrowser__Wrapper",
  componentId: "sc-9pnsws-0"
})(["", ";color:var(--text-base,", ");background-color:var(--background-base,", ");position:fixed;top:0;left:0;width:100%;height:100%;z-index:", ";display:flex;justify-content:center;align-items:center;flex-direction:column;text-align:center;padding:", ";"], rootStyle(), defaultTheme.invertedDark.text.base, defaultTheme.invertedDark.background.base, zIndexFixed, spacer24);
var BodySvg = styled.svg.withConfig({
  displayName: "UnsupportedBrowser__BodySvg",
  componentId: "sc-9pnsws-1"
})(["stroke:var(--background-base,", ");margin-bottom:", ";"], defaultTheme.invertedDark.background.base, spacer64);
export var UnsupportedBrowser = function UnsupportedBrowser(_ref) {
  var className = _ref.className,
      _ref$colorSet = _ref.colorSet,
      colorSet = _ref$colorSet === void 0 ? 'invertedDark' : _ref$colorSet,
      children = _ref.children,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(Wrapper, Object.assign({}, props, {
    className: addColorSet(colorSet, className)
  }), /*#__PURE__*/React.createElement(BodySvg, {
    role: "img",
    width: "214",
    height: "155"
  }, /*#__PURE__*/React.createElement("rect", {
    stroke: "#FFF",
    strokeWidth: "3",
    x: "1.5",
    y: "1.5",
    width: "211",
    height: "152",
    rx: "9"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "16",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "32",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("circle", {
    fill: "#FFF",
    cx: "48",
    cy: "18",
    r: "4"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M55 83c3.28 2.667 7.28 4 12 4s8.72-1.333 12-4M135 83c3.28 2.667 7.28 4 12 4s8.72-1.333 12-4",
    stroke: "#FFF",
    strokeWidth: "3",
    strokeLinecap: "square"
  }), /*#__PURE__*/React.createElement("path", {
    d: "M68 122.374c13-4.25 26-6.374 39-6.374s26 2.125 39 6.374M2.5 34.5h210",
    stroke: "#FFF",
    strokeWidth: "3",
    strokeLinecap: "square"
  })), children);
};
/**
* **UnsupportedBrowser** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/UnsupportedBrowser) | [Storybook](https://encore-web.spotify.net/?path=/docs/components-unsupportedbrowser--default) | [Encore](https://encore.spotify.net/web/components/unsupported-browser) | 
*
* UnsupportedBrowser shows a simple message and image when a browser is unsupported.
*
* @example
* () => (
*  <UnsupportedBrowser>
*    <Type as="p" variant="altoBrio">
*      Your browser doesn’t work with this application.
*    </Type>
*    <Type paddingBottom="0.67em" as="p" variant="altoBrio">
*      <TextLink href="https://browser-update.org/" standalone>
*        Switch or update browsers.
*      </TextLink>
*    </Type>
*  </UnsupportedBrowser>
*);
*
*/

export default UnsupportedBrowser;
