// @spotify-internal/adstudio-shared/lib/components/CpmMetrics/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CpmMetrics = CpmMetrics;
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _DetailViewerList = _interopRequireDefault(require("../DetailViewerList"));

var _DetailViewerListItem = _interopRequireDefault(require("../DetailViewerListItem"));

var _IntlFormatNumber = _interopRequireDefault(require("../IntlFormatNumber"));

var _LegendDot = _interopRequireDefault(require("../LegendDot"));

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _QuartileFormatNumber = _interopRequireDefault(require("../QuartileFormatNumber"));

var _i18n = require("../../config/i18n");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function CpmMetrics(_ref) {
  var metrics = _ref.metrics,
      hasMetrics = _ref.hasMetrics,
      adsServedLegend = _ref.adsServedLegend,
      reachLegend = _ref.reachLegend,
      strings = _ref.strings,
      _ref$showClicks = _ref.showClicks,
      showClicks = _ref$showClicks === void 0 ? true : _ref$showClicks,
      _ref$countryCode = _ref.countryCode,
      countryCode = _ref$countryCode === void 0 ? _i18n.DEFAULT_COUNTRY : _ref$countryCode;
  var _metrics$clicks = metrics.clicks,
      clicks = _metrics$clicks === void 0 ? 0 : _metrics$clicks,
      _metrics$ctr = metrics.ctr,
      ctr = _metrics$ctr === void 0 ? 0.0 : _metrics$ctr,
      _metrics$frequency = metrics.frequency,
      frequency = _metrics$frequency === void 0 ? 0.0 : _metrics$frequency,
      _metrics$reach = metrics.reach,
      reach = _metrics$reach === void 0 ? 0 : _metrics$reach,
      _metrics$adsServed = metrics.adsServed,
      adsServed = _metrics$adsServed === void 0 ? 0 : _metrics$adsServed,
      _metrics$completionRa = metrics.completionRate,
      completionRate = _metrics$completionRa === void 0 ? 0.0 : _metrics$completionRa,
      _metrics$quartiles = metrics.quartiles,
      quartiles = _metrics$quartiles === void 0 ? {
    firstQuartile: 0.0,
    midpoint: 0.0,
    thirdQuartile: 0.0,
    complete: 0.0
  } : _metrics$quartiles;
  var ADS_SERVER = strings.ADS_SERVER,
      REACH_OF_ADS = strings.REACH_OF_ADS,
      FREQ_OF_ADS = strings.FREQ_OF_ADS,
      CLICKS = strings.CLICKS,
      COMPLETION_RATE = strings.COMPLETION_RATE,
      NA = strings.NA,
      QUARTILES = strings.QUARTILES;
  var hasQuartileData = !!(quartiles.firstQuartile || quartiles.midpoint || quartiles.thirdQuartile || quartiles.complete);
  return /*#__PURE__*/_react["default"].createElement(_DetailViewerList["default"], {
    inline: true
  }, /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, adsServedLegend && /*#__PURE__*/_react["default"].createElement(_LegendDot["default"], {
      color: adsServedLegend
    }), ADS_SERVER),
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "ads-served-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    number: adsServed
  }))), /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, reachLegend && /*#__PURE__*/_react["default"].createElement(_LegendDot["default"], {
      color: reachLegend
    }), REACH_OF_ADS),
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "reach-of-ads-served-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    number: reach
  }))), /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: FREQ_OF_ADS,
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "freq-of-ads-served-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    maximumFractionDigits: 2,
    number: frequency
  }))), /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: COMPLETION_RATE,
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "completion-rate-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    type: _IntlFormatNumber["default"].PERCENT,
    number: completionRate
  }))), showClicks && /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: CLICKS,
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "clicks-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    number: clicks
  }))), showClicks && /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: "CTR",
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "ctr-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, ctr === 0 ? NA : /*#__PURE__*/_react["default"].createElement(_IntlFormatNumber["default"], {
    countryCode: countryCode,
    number: ctr,
    type: _IntlFormatNumber["default"].PERCENT
  }))), /*#__PURE__*/_react["default"].createElement(_DetailViewerListItem["default"], {
    empty: !hasMetrics,
    label: QUARTILES,
    performanceItem: true
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h2, {
    "data-test": "quartiles-value",
    variant: _encoreAdvertisingWeb.Type.heading2
  }, !hasQuartileData ? NA : /*#__PURE__*/_react["default"].createElement(_QuartileFormatNumber["default"], {
    quartiles: quartiles,
    adsServed: adsServed,
    countryCode: countryCode
  }))));
}

CpmMetrics.propTypes = {
  adsServedLegend: _propTypes["default"].string,
  countryCode: _propTypes["default"].string,
  hasMetrics: _propTypes["default"].bool,
  metrics: _propTypes["default"].shape({
    clicks: _propTypes["default"].number,
    ctr: _propTypes["default"].number,
    frequency: _propTypes["default"].number,
    reach: _propTypes["default"].number,
    adsServed: _propTypes["default"].number,
    completionRate: _propTypes["default"].number,
    quartiles: _propTypes["default"].shape({
      firstQuartile: _propTypes["default"].number,
      midpoint: _propTypes["default"].number,
      thirdQuartile: _propTypes["default"].number,
      complete: _propTypes["default"].number
    })
  }),
  reachLegend: _propTypes["default"].string,
  showClicks: _propTypes["default"].bool,
  strings: {
    ADS_SERVER: _propTypes["default"].string,
    REACH_OF_ADS: _propTypes["default"].string,
    FREQ_OF_ADS: _propTypes["default"].string,
    CLICKS: _propTypes["default"].string,
    COMPLETION_RATE: _propTypes["default"].string,
    NA: _propTypes["default"].string,
    QUARTILES: _propTypes["default"].string
  }
};
CpmMetrics.defaultProps = {
  strings: {
    ADS_SERVER: 'Ads served',
    REACH_OF_ADS: 'Reach of ads served',
    FREQ_OF_ADS: 'Freq of Ads Served',
    CLICKS: 'Clicks',
    COMPLETION_RATE: 'Completion rate',
    NA: 'N/A',
    QUARTILES: 'Ads played to'
  }
};
var _default = CpmMetrics;
exports["default"] = _default;
