// @spotify-internal/adstudio-shared/lib/components/ExpandToModalContainer/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ExpandToModalContainer = void 0;

var _get2 = _interopRequireDefault(require("lodash/get"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _recompose = require("recompose");

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

// This is the usual size of the unexpanded view, but can be increased by height of optional
// details component property.
var UNEXPANDED_HEIGHT = 150;
var ViewMoreType = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "ExpandToModalContainer__ViewMoreType",
  componentId: "sc-1huq1y6-0"
})(["color:", ";cursor:pointer;padding-bottom:0;"], _encoreAdvertisingWeb.azure);

var DisplayContainer = _styledComponents["default"].div.withConfig({
  displayName: "ExpandToModalContainer__DisplayContainer",
  componentId: "sc-1huq1y6-1"
})(["", " overflow:hidden;text-overflow:ellipsis;white-space:pre-wrap;display:-webkit-box;-webkit-line-clamp:4;-webkit-box-orient:vertical;"], function (props) {
  return "\n    max-height: ".concat(props.height, ";\n    width: ").concat(props.width, ";\n  ");
});

var DisplayType = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "ExpandToModalContainer__DisplayType",
  componentId: "sc-1huq1y6-2"
})(["padding-bottom:0;"]);

var ModalDiv = _styledComponents["default"].div.withConfig({
  displayName: "ExpandToModalContainer__ModalDiv",
  componentId: "sc-1huq1y6-3"
})(["white-space:pre-wrap;"]);

var ExpandToModalContainer = /*#__PURE__*/function (_React$Component) {
  _inherits(ExpandToModalContainer, _React$Component);

  var _super = _createSuper(ExpandToModalContainer);

  function ExpandToModalContainer(props) {
    var _this;

    _classCallCheck(this, ExpandToModalContainer);

    _this = _super.call(this, props);
    _this.detailsRef = /*#__PURE__*/_react["default"].createRef();
    _this.Details = /*#__PURE__*/_react["default"].createElement("div", {
      ref: _this.detailsRef
    }, (0, _get2["default"])(_assertThisInitialized(_this), 'props.details'));
    return _this;
  }

  _createClass(ExpandToModalContainer, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.setState({
        detailsHeight: (0, _get2["default"])(this, 'detailsRef.current.offsetHeight', 0)
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          details = _this$props.details,
          detailsExpanded = _this$props.detailsExpanded,
          isExpandable = _this$props.isExpandable,
          openExpandedModal = _this$props.openExpandedModal,
          modalOpened = _this$props.modalOpened,
          formattedModalBody = _this$props.formattedModalBody,
          width = _this$props.width,
          typeVariant = _this$props.typeVariant,
          typeWeight = _this$props.typeWeight,
          dialogTitle = _this$props.dialogTitle,
          secondaryButton = _this$props.secondaryButton,
          strings = _this$props.strings;
      var height = "".concat(UNEXPANDED_HEIGHT + parseInt((0, _get2["default"])(this, 'state.detailsHeight', 0), 10), "px");
      return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(DisplayContainer, {
        width: width,
        height: height
      }, details && this.Details, /*#__PURE__*/_react["default"].createElement(DisplayType, {
        variant: typeVariant,
        weight: typeWeight
      }, children)), isExpandable ? /*#__PURE__*/_react["default"].createElement(ViewMoreType, {
        variant: _encoreAdvertisingWeb.Type.cta3,
        onClick: function onClick() {
          return openExpandedModal(!modalOpened);
        },
        id: "view-more"
      }, strings.viewMore) : null, modalOpened && /*#__PURE__*/_react["default"].createElement(_adstudioTape.DialogConfirmation, {
        id: "expanded-locations-modal",
        details: detailsExpanded,
        dialogTitle: dialogTitle,
        body: /*#__PURE__*/_react["default"].createElement(ModalDiv, {
          id: "locations-body"
        }, formattedModalBody),
        footer: /*#__PURE__*/_react["default"].createElement("div", null, secondaryButton && /*#__PURE__*/_react["default"].createElement("span", null, secondaryButton), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.ButtonPrimary, {
          id: "locations-close-btn",
          onClick: function onClick() {
            return openExpandedModal(!modalOpened);
          }
        }, strings.close))
      }));
    }
  }]);

  return ExpandToModalContainer;
}(_react["default"].Component);

exports.ExpandToModalContainer = ExpandToModalContainer;
ExpandToModalContainer.propTypes = {
  children: _propTypes["default"].string,
  details: _propTypes["default"].node,
  detailsExpanded: _propTypes["default"].node,
  secondaryButton: _propTypes["default"].node,
  formattedModalBody: _propTypes["default"].string,
  width: _propTypes["default"].string.isRequired,
  isExpandable: _propTypes["default"].bool.isRequired,
  openExpandedModal: _propTypes["default"].func.isRequired,
  modalOpened: _propTypes["default"].bool.isRequired,
  strings: _propTypes["default"].shape({
    close: _propTypes["default"].string,
    viewMore: _propTypes["default"].string
  }),
  typeVariant: _propTypes["default"].string,
  typeWeight: _propTypes["default"].string,
  dialogTitle: _propTypes["default"].string
};
ExpandToModalContainer.defaultProps = {
  isExpandable: false,
  strings: {
    close: 'Close',
    viewMore: 'View More'
  },
  typeVariant: _encoreAdvertisingWeb.Type.body1,
  typeWeight: _encoreAdvertisingWeb.Type.book
};

var _default = (0, _recompose.compose)((0, _recompose.withState)('modalOpened', 'openExpandedModal', false))(ExpandToModalContainer);

exports["default"] = _default;
