// @spotify-internal/creator-tape/es/styles/mixins/type.js
import { css } from 'styled-components';
import { fontSizeBody1, fontSizeBody2, fontSizeBody3, fontSizeBody4, fontWeightBook, fontWeightBold, fontWeightBlack, screenSmMin, screenMdMin } from '../variables';
export var typeStyleDisplay1 = function typeStyleDisplay1() {
  return css(["font-size:56px;font-weight:", ";letter-spacing:-1.75px;line-height:64px;padding-bottom:0.67em;@media (min-width:", "){font-size:80px;line-height:88px;letter-spacing:-2px;}@media (min-width:", "){font-size:96px;line-height:104px;}"], fontWeightBlack, screenSmMin, screenMdMin);
};
export var typeStyleDisplay2 = function typeStyleDisplay2() {
  return css(["font-size:48px;font-weight:", ";letter-spacing:-1.75px;line-height:56px;padding-bottom:0.67em;@media (min-width:", "){font-size:64px;line-height:72px;letter-spacing:-2px;}@media (min-width:", "){font-size:80px;line-height:88px;}"], fontWeightBlack, screenSmMin, screenMdMin);
};
export var typeStyleH1 = function typeStyleH1() {
  return css(["font-size:32px;font-weight:", ";letter-spacing:-1px;line-height:40px;padding-bottom:0.67em;@media (min-width:", "){font-size:40px;line-height:48px;}@media (min-width:", "){font-size:48px;line-height:56px;}"], fontWeightBlack, screenSmMin, screenMdMin);
};
export var typeStyleH2 = function typeStyleH2() {
  return css(["font-size:28px;font-weight:", ";letter-spacing:-0.25px;line-height:36px;padding-bottom:0.75em;@media (min-width:", "){font-size:32px;line-height:40px;letter-spacing:-0.5px;}"], fontWeightBlack, screenSmMin);
};
export var typeStyleH3 = function typeStyleH3() {
  return css(["font-size:20px;font-weight:", ";line-height:28px;letter-spacing:-0.25px;@media (min-width:", "){font-size:24px;line-height:32px;}"], fontWeightBlack, screenSmMin);
};
export var typeStyleH4 = function typeStyleH4() {
  return css(["font-size:18px;font-weight:", ";line-height:26px;letter-spacing:0;"], fontWeightBook);
};
export var typeStyleBody1 = function typeStyleBody1() {
  return css(["font-size:", ";font-weight:", ";line-height:24px;letter-spacing:0;"], fontSizeBody1, fontWeightBook);
};
export var typeStyleBody2 = function typeStyleBody2() {
  return css(["font-size:", ";font-weight:", ";line-height:20px;letter-spacing:0.25px;"], fontSizeBody2, fontWeightBook);
};
export var typeStyleBody3 = function typeStyleBody3() {
  return css(["font-size:", ";font-weight:", ";line-height:16px;letter-spacing:0.25px;"], fontSizeBody3, fontWeightBook);
};
export var typeStyleBody4 = function typeStyleBody4() {
  return css(["font-size:", ";font-weight:", ";line-height:16px;letter-spacing:0.25px;"], fontSizeBody4, fontWeightBook);
};
export var typeStyleCta1 = function typeStyleCta1() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:2px;line-height:16px;text-transform:uppercase;"], fontSizeBody1, fontWeightBold);
};
export var typeStyleCta2 = function typeStyleCta2() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:1px;line-height:16px;text-transform:uppercase;"], fontSizeBody2, fontWeightBold);
};
export var typeStyleCta3 = function typeStyleCta3() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:1px;line-height:16px;text-transform:uppercase;"], fontSizeBody3, fontWeightBold);
};
export var typeStyleCta4 = function typeStyleCta4() {
  return css(["font-size:", ";font-weight:", ";letter-spacing:1px;line-height:16px;text-transform:uppercase;"], fontSizeBody4, fontWeightBold);
};
