// @spotify-internal/encore-web/es/components/NavBarListItem/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
var _excluded = ["label", "active", "disabled", "isHover", "isFocus", "component", "sentenceCase", "theme"];
import React from 'react';
import { deprecatedPropWarning } from "../../deprecationWarnings";
import { Link } from "./Link";
import { ListItem } from "./ListItem";
export var NavBarListItem = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var label = _ref.label,
      active = _ref.active,
      disabled = _ref.disabled,
      isHover = _ref.isHover,
      isFocus = _ref.isFocus,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      _ref$sentenceCase = _ref.sentenceCase,
      sentenceCase = _ref$sentenceCase === void 0 ? false : _ref$sentenceCase,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, _excluded);

  if (theme) {
    deprecatedPropWarning('theme', 'This component is intended to inherit its parent’s color set or color.');
  }

  return /*#__PURE__*/React.createElement(ListItem, null, /*#__PURE__*/React.createElement(Link, Object.assign({
    ref: ref,
    isActive: active,
    isDisabled: disabled,
    isHover: isHover,
    isFocus: isFocus,
    component: component,
    sentenceCase: sentenceCase,
    "aria-current": active ? 'page' : undefined
  }, props), label));
});
/**
* **NavBarListItem** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/core/components/NavBarListItem) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navbar-navbarlistitem--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-bar) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=377:171) |
*
* An individual tab within the NavBar component
*
* @example
* () => (
*  <NavBarList>
*    <NavBarListItem label="Overview" href="#navbar-overview" />
*  </NavBarList>
*);
*
*/

export default NavBarListItem;
