// reactochart/utils/Margin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.innerWidth = innerWidth;
exports.innerHeight = innerHeight;
exports.innerSize = innerSize;
exports.innerRangeX = innerRangeX;
exports.innerRangeY = innerRangeY;
exports.prefixKeys = prefixKeys;
exports.maxMargins = maxMargins;
exports.sumMargins = sumMargins;
exports.zeroMargin = void 0;

var _mapKeys = _interopRequireDefault(require("lodash/mapKeys"));

var _upperFirst = _interopRequireDefault(require("lodash/upperFirst"));

var _mapValues = _interopRequireDefault(require("lodash/mapValues"));

var _clone = _interopRequireDefault(require("lodash/clone"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const zeroMargin = {
  top: 0,
  bottom: 0,
  left: 0,
  right: 0
}; // find a fuzzy match for key in object and return the value
// eg getFuzzy({marginLeft: 10}, 'left') returns 10

exports.zeroMargin = zeroMargin;

function getFuzzy(obj = {}, fuzzyKey) {
  const keyMatch = Object.keys(obj).find(key => {
    if (!!key.match(new RegExp(fuzzyKey, 'i'))) {
      return true;
    }

    return false;
  });

  if (keyMatch === undefined) {
    return;
  }

  return obj[keyMatch];
}

function innerWidth(width, margin = {}) {
  return Math.max(width - ((getFuzzy(margin, 'left') || 0) + (getFuzzy(margin, 'right') || 0)), 0);
}

function innerHeight(height, margin = {}) {
  return Math.max(height - ((getFuzzy(margin, 'top') || 0) + (getFuzzy(margin, 'bottom') || 0)), 0);
}

function innerSize({
  width,
  height
} = {}, margin = {}) {
  return {
    width: innerWidth(width, margin),
    height: innerHeight(height, margin)
  };
}

function innerRangeX(outerWidth, margin = {}) {
  const left = getFuzzy(margin, 'left') || 0;
  return [Math.min(left, outerWidth), Math.min(left + innerWidth(outerWidth, margin), outerWidth)];
}

function innerRangeY(outerHeight, margin = {}) {
  const top = getFuzzy(margin, 'top') || 0;
  return [Math.min(top + innerHeight(outerHeight, margin), outerHeight), Math.min(top, outerHeight)];
}

function prefixKeys(obj, prefix) {
  if (!prefix) return obj;
  return (0, _mapKeys.default)(obj, (value, key) => prefix + (0, _upperFirst.default)(key));
} // TODO this isn't used anywhere, deprecate?


function maxMargins(margins = [], keyPrefix) {
  return margins.reduce((result, margin) => {
    return (0, _mapValues.default)(result, (value, key) => {
      return Math.max(margin[key] || 0, result[key] || 0);
    });
  }, (0, _clone.default)(prefixKeys(zeroMargin, keyPrefix)));
}

function sumMargins(margins = [], keyPrefix) {
  return margins.reduce((result, margin) => {
    return (0, _mapValues.default)(result, (value, key) => {
      return (result[key] || 0) + (margin[key] || 0);
    });
  }, (0, _clone.default)(prefixKeys(zeroMargin, keyPrefix)));
}
