// @spotify-internal/adstudio-tape/lib/components/RadioList/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.RadioList = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _FormGroup = _interopRequireDefault(require("../FormGroup"));

var _StyledType = require("../FormGroup/StyledType");

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _Item = _interopRequireDefault(require("./Item"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var RadioFormGroup = (0, _styledComponents["default"])(_FormGroup["default"]).withConfig({
  displayName: "RadioList__RadioFormGroup",
  componentId: "sc-15rfxcc-0"
})(["margin:0;border-color:", ";border-width:", ";", ""], function (props) {
  if (props.error) return _encoreFoundation.failure;
  if (props.warning) return _encoreFoundation.warning;
  return _encoreFoundation.gray70;
}, function (props) {
  return props.error || props.warning && _encoreFoundation.spacer4;
}, function (props) {
  return props.orientation === 'horizontal' && (0, _styledComponents.css)(["& ", "{display:grid;grid-template-columns:repeat(", ",1fr);grid-column-gap:12px;padding-bottom:0;}"], _StyledType.Span, props.numColumns);
});
var RadioItem = (0, _styledComponents["default"])(_Item["default"]).withConfig({
  displayName: "RadioList__RadioItem",
  componentId: "sc-15rfxcc-1"
})(["margin-bottom:", ";", " ", ""], _encoreFoundation.spacer8, function (props) {
  var bWidth = props.shadow ? '2px' : '1px';
  return props.group ? (0, _styledComponents.css)(["border-radius:0;border-top-width:", ";border-bottom-width:0;margin-bottom:0;&:first-of-type{border-top-left-radius:", ";border-top-right-radius:", ";border-top-width:", ";}&:last-of-type{border-bottom-left-radius:", ";border-bottom-right-radius:", ";border-bottom-width:", ";margin-bottom:", ";}"], props.heading ? bWidth : 0, _encoreFoundation.spacer4, _encoreFoundation.spacer4, bWidth, _encoreFoundation.spacer4, _encoreFoundation.spacer4, bWidth, _encoreFoundation.spacer8) : '';
}, function (props) {
  return props.orientation === 'horizontal' ? (0, _styledComponents.css)(["& div:first-child{margin-bottom:0;}"]) : (0, _styledComponents.css)(["&:last-of-type{margin-bottom:0;}"]);
});

var RadioList = function RadioList(_ref) {
  var options = _ref.options,
      id = _ref.id,
      description = _ref.description,
      error = _ref.error,
      animation = _ref.animation,
      group = _ref.group,
      label = _ref.label,
      subDescription = _ref.subDescription,
      warning = _ref.warning,
      shadow = _ref.shadow,
      orientation = _ref.orientation,
      invertOnSelect = _ref.invertOnSelect,
      props = _objectWithoutProperties(_ref, ["options", "id", "description", "error", "animation", "group", "label", "subDescription", "warning", "shadow", "orientation", "invertOnSelect"]);

  return /*#__PURE__*/_react["default"].createElement(RadioFormGroup, {
    subDescription: subDescription,
    description: description,
    flush: true,
    forLabel: id,
    id: id,
    label: label,
    error: error,
    warning: warning,
    orientation: orientation,
    numColumns: options.length
  }, options.map(function (_ref2) {
    var key = _ref2.key,
        value = _ref2.value,
        children = _ref2.children,
        optionProps = _objectWithoutProperties(_ref2, ["key", "value", "children"]);

    return /*#__PURE__*/_react["default"].createElement(RadioItem, _extends({
      key: key,
      label: value,
      radioValue: key,
      animation: animation,
      group: group,
      shadow: shadow,
      orientation: orientation,
      invertOnSelect: invertOnSelect
    }, props, optionProps), children);
  }));
};

exports.RadioList = RadioList;
RadioList.propTypes = {
  id: _propTypes["default"].string,

  /* eslint-disable */
  animation: _propTypes["default"].object,

  /* eslint-enable */
  border: _propTypes["default"].bool,
  description: _propTypes["default"].string,
  error: _propTypes["default"].string,
  group: _propTypes["default"].bool,
  label: _propTypes["default"].string.isRequired,
  subDescription: _propTypes["default"].string,
  value: _propTypes["default"].string,
  warning: _propTypes["default"].string,
  options: _propTypes["default"].arrayOf(_propTypes["default"].shape(_objectSpread(_objectSpread({}, _FormGroup["default"].propTypes), {}, {
    key: _propTypes["default"].string.isRequired,
    value: _propTypes["default"].node.isRequired,
    children: _propTypes["default"].node
  }))),
  shadow: _propTypes["default"].bool,
  invertOnSelect: _propTypes["default"].bool,
  orientation: _propTypes["default"].oneOf(['vertical', 'horizontal'])
};
RadioList.defaultProps = {
  group: false,
  shadow: true,
  orientation: 'vertical'
};
var _default = RadioList;
exports["default"] = _default;
