// @spotify-internal/adstudio-gainz-client/models/com/spotify/ads/adstudiogainz/BulkSheetRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var BulkSheetRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.adstudiogainz = (function() {
                /**
                 * Namespace adstudiogainz.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var adstudiogainz = {};

                adstudiogainz.BulkSheetRequest = (function() {
                    /**
                     * Properties of a BulkSheetRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @interface IBulkSheetRequest
                     * @property {string|null} [bulkSheetId] BulkSheetRequest bulkSheetId
                     * @property {string|null} [adAccountId] BulkSheetRequest adAccountId
                     * @property {string|null} [setBulkSheetId] BulkSheetRequest setBulkSheetId
                     * @property {string|null} [setAdAccountId] BulkSheetRequest setAdAccountId
                     */
                    
                    /**
                     * Constructs a new BulkSheetRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @classdesc Represents a BulkSheetRequest.
                     * @implements IBulkSheetRequest
                     * @constructor
                     * @param {com.spotify.ads.adstudiogainz.IBulkSheetRequest=} [properties] Properties to set
                     */
                    function BulkSheetRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * BulkSheetRequest bulk_sheet_id.
                     * @member {string} bulk_sheet_id
                     * @memberof com.spotify.ads.adstudiogainz.BulkSheetRequest
                     * @instance
                     */
                    BulkSheetRequest.prototype.bulkSheetId = "";

                    /**
                     * BulkSheetRequest ad_account_id.
                     * @member {string} ad_account_id
                     * @memberof com.spotify.ads.adstudiogainz.BulkSheetRequest
                     * @instance
                     */
                    BulkSheetRequest.prototype.adAccountId = "";


                    /**
                     * Sets bulk_sheet_id
                     * @function setBulkSheetId
                     * @memberof com.spotify.ads.adstudiogainz.BulkSheetRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.BulkSheetRequest} BulkSheetRequest
                     */
                    BulkSheetRequest.prototype.setBulkSheetId = function(value) {
                        this['bulkSheetId'] = value;
                        return this;
                    };

                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.ads.adstudiogainz.BulkSheetRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.BulkSheetRequest} BulkSheetRequest
                     */
                    BulkSheetRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Converts the BulkSheetRequest to an object
                     * @function toObject
                     * @memberof com.spotify.ads.adstudiogainz.BulkSheetRequest
                     * @instance
                     * @returns {object} object
                     */
                    BulkSheetRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return BulkSheetRequest;
                })();
                
                BulkSheetRequest = adstudiogainz.BulkSheetRequest;
                return adstudiogainz;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["BulkSheetRequest"] = BulkSheetRequest;
exports["default"] = $root;
