// @spotify-internal/adstudio-gainz-client/models/com/spotify/ads/adstudiogainz/DownloadRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var DownloadRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.adstudiogainz = (function() {
                /**
                 * Namespace adstudiogainz.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var adstudiogainz = {};

                adstudiogainz.DownloadRequest = (function() {
                    /**
                     * Properties of a DownloadRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @interface IDownloadRequest
                     * @property {string|null} [adAccountId] DownloadRequest adAccountId
                     * @property {Array.<string>|null} [campaignIds] DownloadRequest campaignIds
                     * @property {Array.<string>|null} [flightIds] DownloadRequest flightIds
                     * @property {Array.<string>|null} [creativeIds] DownloadRequest creativeIds
                     * @property {string|null} [setAdAccountId] DownloadRequest setAdAccountId
                     * @property {Array.<string>|null} [setCampaignIds] DownloadRequest setCampaignIds
                     * @property {Array.<string>|null} [setFlightIds] DownloadRequest setFlightIds
                     * @property {Array.<string>|null} [setCreativeIds] DownloadRequest setCreativeIds
                     */
                    
                    /**
                     * Constructs a new DownloadRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @classdesc Represents a DownloadRequest.
                     * @implements IDownloadRequest
                     * @constructor
                     * @param {com.spotify.ads.adstudiogainz.IDownloadRequest=} [properties] Properties to set
                     */
                    function DownloadRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * DownloadRequest ad_account_id.
                     * @member {string} ad_account_id
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @instance
                     */
                    DownloadRequest.prototype.adAccountId = "";

                    /**
                     * DownloadRequest campaign_ids.
                     * @member {Array.<string>} campaign_ids
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @instance
                     */
                    DownloadRequest.prototype.campaignIds = [];

                    /**
                     * DownloadRequest flight_ids.
                     * @member {Array.<string>} flight_ids
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @instance
                     */
                    DownloadRequest.prototype.flightIds = [];

                    /**
                     * DownloadRequest creative_ids.
                     * @member {Array.<string>} creative_ids
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @instance
                     */
                    DownloadRequest.prototype.creativeIds = [];


                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.DownloadRequest} DownloadRequest
                     */
                    DownloadRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets campaign_ids
                     * @function setCampaignIds
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.DownloadRequest} DownloadRequest
                     */
                    DownloadRequest.prototype.setCampaignIds = function(value) {
                        this['campaignIds'] = value;
                        return this;
                    };

                    /**
                     * Sets flight_ids
                     * @function setFlightIds
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.DownloadRequest} DownloadRequest
                     */
                    DownloadRequest.prototype.setFlightIds = function(value) {
                        this['flightIds'] = value;
                        return this;
                    };

                    /**
                     * Sets creative_ids
                     * @function setCreativeIds
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @param {Array.<string>} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.DownloadRequest} DownloadRequest
                     */
                    DownloadRequest.prototype.setCreativeIds = function(value) {
                        this['creativeIds'] = value;
                        return this;
                    };

                    /**
                     * Converts the DownloadRequest to an object
                     * @function toObject
                     * @memberof com.spotify.ads.adstudiogainz.DownloadRequest
                     * @instance
                     * @returns {object} object
                     */
                    DownloadRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return DownloadRequest;
                })();
                
                DownloadRequest = adstudiogainz.DownloadRequest;
                return adstudiogainz;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["DownloadRequest"] = DownloadRequest;
exports["default"] = $root;
