// @spotify-internal/adstudio-gainz-client/models/com/spotify/ads/adstudiogainz/PdfUploadRequest.js
// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";

Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var PdfUploadRequest = {};

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.adstudiogainz = (function() {
                /**
                 * Namespace adstudiogainz.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var adstudiogainz = {};

                adstudiogainz.PdfUploadRequest = (function() {
                    /**
                     * Properties of a PdfUploadRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @interface IPdfUploadRequest
                     * @property {string|null} [adAccountId] PdfUploadRequest adAccountId
                     * @property {string|null} [bulkSheetId] PdfUploadRequest bulkSheetId
                     * @property {string|null} [setAdAccountId] PdfUploadRequest setAdAccountId
                     * @property {string|null} [setBulkSheetId] PdfUploadRequest setBulkSheetId
                     */
                    
                    /**
                     * Constructs a new PdfUploadRequest.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @classdesc Represents a PdfUploadRequest.
                     * @implements IPdfUploadRequest
                     * @constructor
                     * @param {com.spotify.ads.adstudiogainz.IPdfUploadRequest=} [properties] Properties to set
                     */
                    function PdfUploadRequest(properties) {
                        if (properties)
                            for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                                if (properties[keys[i]] != null)
                                    this[keys[i]] = properties[keys[i]];
                    }


                    /**
                     * PdfUploadRequest ad_account_id.
                     * @member {string} ad_account_id
                     * @memberof com.spotify.ads.adstudiogainz.PdfUploadRequest
                     * @instance
                     */
                    PdfUploadRequest.prototype.adAccountId = "";

                    /**
                     * PdfUploadRequest bulk_sheet_id.
                     * @member {string} bulk_sheet_id
                     * @memberof com.spotify.ads.adstudiogainz.PdfUploadRequest
                     * @instance
                     */
                    PdfUploadRequest.prototype.bulkSheetId = "";


                    /**
                     * Sets ad_account_id
                     * @function setAdAccountId
                     * @memberof com.spotify.ads.adstudiogainz.PdfUploadRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.PdfUploadRequest} PdfUploadRequest
                     */
                    PdfUploadRequest.prototype.setAdAccountId = function(value) {
                        this['adAccountId'] = value;
                        return this;
                    };

                    /**
                     * Sets bulk_sheet_id
                     * @function setBulkSheetId
                     * @memberof com.spotify.ads.adstudiogainz.PdfUploadRequest
                     * @param {string} value
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.PdfUploadRequest} PdfUploadRequest
                     */
                    PdfUploadRequest.prototype.setBulkSheetId = function(value) {
                        this['bulkSheetId'] = value;
                        return this;
                    };

                    /**
                     * Converts the PdfUploadRequest to an object
                     * @function toObject
                     * @memberof com.spotify.ads.adstudiogainz.PdfUploadRequest
                     * @instance
                     * @returns {object} object
                     */
                    PdfUploadRequest.prototype.toObject = function() {
                        return JSON.parse(JSON.stringify(this));
                    };


                    return PdfUploadRequest;
                })();
                
                PdfUploadRequest = adstudiogainz.PdfUploadRequest;
                return adstudiogainz;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();

exports["$root"] = $root;
exports["PdfUploadRequest"] = PdfUploadRequest;
exports["default"] = $root;
