// @spotify-internal/adstudio-gainz-client/clients/AdstudioGainzService.js

// THIS FILE IS AUTO-GENERATED BY PROTO-REQUEST
/* eslint-disable */
"use strict";


// Exported root namespace
Object.defineProperty(exports, '__esModule', { value: true });
var $root = {};
var AdstudioGainzService = {};
    

$root.com = (function() {
    /**
     * Namespace com.
     * @exports com
     * @namespace
     */
    var com = {};

    com.spotify = (function() {
        /**
         * Namespace spotify.
         * @memberof com
         * @namespace
         */
        var spotify = {};

        spotify.ads = (function() {
            /**
             * Namespace ads.
             * @memberof com.spotify
             * @namespace
             */
            var ads = {};

            ads.adstudiogainz = (function() {
                /**
                 * Namespace adstudiogainz.
                 * @memberof com.spotify.ads
                 * @namespace
                 */
                var adstudiogainz = {};

                adstudiogainz.AdstudioGainzService = (function() {
                        
                    /**
                     * Properties of a AdstudioGainzService.
                     * @memberof com.spotify.ads.adstudiogainz
                     */
                     
                     /**
                      * Fetch function type, set when instantiating this service class.
                      * @callback FetchFunction
                      * @param {string} uri
                      * @param {Object} options
                      * @returns {Promise<any>}
                      */
                     
                    /**
                     * An HttpServiceClient class to send requests to the AdstudioGainzService.
                     * @memberof com.spotify.ads.adstudiogainz
                     * @classdesc AdstudioGainzService to send requests to the AdstudioGainzService.
                     * @param {FetchFunction} fetchFn
                     * @param {protorequest.HttpServiceConfigOptions} options
                     * @constructor
                     */
                    AdstudioGainzService = function(fetchFn, options) {
                        this.fetchFn = fetchFn;
                        this.host = options.host;
                        this.baseUrl = options.baseUrl;
                        this.version = options.version || "";
                        this.headers = options.headers || new Headers();
                    };
                
                    /**
                     * Sends the fetch request decorated with the configured host.  Only for internal use.
                     */
                    AdstudioGainzService.prototype.protoFetch = function(url, opts) {
                      return this.fetchFn(this.host + url, opts);
                    };
                
                    /**
                     * Sets the headers to be sent with all requests to this service.
                     * @function setHeaders
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {Headers} headers
                     * @instance
                     * @returns {com.spotify.ads.adstudiogainz.AdstudioGainzService} AdstudioGainzService
                     */
                    AdstudioGainzService.prototype.setHeaders = function(headers) {
                      this.headers = headers;
                      return this;
                    };
                
                    /**
                     * Get the headers being sent with all requests to this service.
                     * @function getHeaders
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @instance
                     * @returns {Headers}
                     */
                    AdstudioGainzService.prototype.getHeaders = function() {
                      return this.headers;
                    };
                
                    /**
                     * Calls AdstudioGainzService.generateBulkSheetSignUrl
                     * @function generateBulkSheetSignUrl
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.BulkUploadRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.BulkSheetResponse>} generateBulkSheetSignUrl
                     */
                    AdstudioGainzService.prototype.generateBulkSheetSignUrl = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'generateBulkSheetSignUrl';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.validateCsv
                     * @function validateCsv
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.BulkSheetRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.BulkValidationResponse>} validateCsv
                     */
                    AdstudioGainzService.prototype.validateCsv = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'validateCsv';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.submitBulkSheet
                     * @function submitBulkSheet
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.BulkSheetRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.ResultSheetResponse>} submitBulkSheet
                     */
                    AdstudioGainzService.prototype.submitBulkSheet = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'submitBulkSheet';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.getBulkSheet
                     * @function getBulkSheet
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.BulkSheetRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.BulkSheetResponse>} getBulkSheet
                     */
                    AdstudioGainzService.prototype.getBulkSheet = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'getBulkSheet';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.getErrorSheet
                     * @function getErrorSheet
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.ErrorSheetRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.ErrorSheetResponses>} getErrorSheet
                     */
                    AdstudioGainzService.prototype.getErrorSheet = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'getErrorSheet';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.downloadRequest
                     * @function downloadRequest
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.DownloadRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.DownloadUrlResponse>} downloadRequest
                     */
                    AdstudioGainzService.prototype.downloadRequest = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'downloadRequest';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.downloadBulkSheet
                     * @function downloadBulkSheet
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.DownloadRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.DownloadUrlResponse>} downloadBulkSheet
                     */
                    AdstudioGainzService.prototype.downloadBulkSheet = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'downloadBulkSheet';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    /**
                     * Calls AdstudioGainzService.generatePdfSignedUrl
                     * @function generatePdfSignedUrl
                     * @memberof com.spotify.ads.adstudiogainz.AdstudioGainzService
                     * @param {com.spotify.ads.adstudiogainz.PdfUploadRequest} request
                     * @instance
                     * @returns {Promise<com.spotify.ads.adstudiogainz.PdfEntityResponse>} generatePdfSignedUrl
                     */
                    AdstudioGainzService.prototype.generatePdfSignedUrl = function(request) {
                        const url = '/' + this.baseUrl + '/AdstudioGainzService/' + this.version +  (this.version ? '/' : '') + 'generatePdfSignedUrl';
                        const opts = {
                            method: 'POST',
                            body: JSON.stringify(request),
                            headers: this.headers
                        };
                        return this.protoFetch(url, opts)
                            .then(function(r) { return r.ok ? r.json() : Promise.reject(r); });
                    };

                    return AdstudioGainzService;
                })();

                return adstudiogainz;
            })();

            return ads;
        })();

        return spotify;
    })();

    return com;
})();


exports["default"] = $root;
exports["$root"] = $root;
exports["AdstudioGainzService"] = AdstudioGainzService;
