// @spotify-internal/encore-web/cjs/styles/semantic-theme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addColorSet = addColorSet;
exports.createBasicColorSet = exports.calcAccessibleContrast = void 0;
exports.cssColorSet = cssColorSet;
exports.cssColorValue = cssColorValue;
exports.cssColorVar = cssColorVar;
exports.getFlatCamelCaseMap = exports.defaultTheme = exports.defaultColorSetName = exports.defaultColorSet = void 0;
exports.makeCampaignCategoryColorSetClass = makeCampaignCategoryColorSetClass;
exports.makeColorSetClass = makeColorSetClass;
exports.semanticTextColors = exports.semanticEssentialColors = exports.semanticDecorativeColors = exports.semanticColors = exports.semanticBackgroundColors = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _color = _interopRequireDefault(require("color"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

/**
 * Theme default constants for Encore Web
 * */
var defaultTheme = _encoreFoundation.lightTheme;
exports.defaultTheme = defaultTheme;
var defaultColorSetName = 'base';
exports.defaultColorSetName = defaultColorSetName;
var defaultColorSet = defaultTheme[defaultColorSetName];
exports.defaultColorSet = defaultColorSet;

var capitalize = function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
};
/**
 * Takes a deeply nested object and returns a flattend map of all the values
 * with the keys camelCased.
 * @param obj - a nested object of string values
 * @param string - the parent key of an object used for recursion.
 * @returns an object with camcelCased keys and string values
 */


var getFlatCamelCaseMap = function getFlatCamelCaseMap(obj) {
  var previousKey = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : '';
  return Object.entries(obj).reduce(function (prevObj, _ref) {
    var _ref2 = (0, _slicedToArray2["default"])(_ref, 2),
        key = _ref2[0],
        value = _ref2[1];

    var camelKey = key;

    if (previousKey !== '') {
      camelKey = "".concat(previousKey).concat(capitalize(key));
    }

    if (typeof value === 'object') {
      return _objectSpread(_objectSpread({}, prevObj), getFlatCamelCaseMap(value, camelKey));
    } else if (typeof value === 'string') {
      prevObj[camelKey] = value;
      return prevObj;
    }

    return prevObj;
  }, {});
};

exports.getFlatCamelCaseMap = getFlatCamelCaseMap;

function camelToKebab(inputString) {
  return inputString.replace(/[A-Z0-9]/g, function (m) {
    return "-".concat(m.toLowerCase());
  });
}

function makeColorSetClass(colorSet) {
  return colorSet ? "encore-".concat(camelToKebab(colorSet), "-set") : '';
}

function makeCampaignCategoryColorSetClass(colorSet) {
  if (!colorSet) return '';
  var removedSetKeyword = colorSet.replace('Set', '');
  return "encore-".concat(camelToKebab(removedSetKeyword), "-set");
}

/**
 * Appends colorSet classname to the passed className string
 * @param colorSet - A colorSet (eg. 'base', 'negative', 'brightAccent')
 * @param className - Other classnames passed from the elements
 * @returns string of passed classnames plus the colorset class
 */
function addColorSet(colorSet, className) {
  if (colorSet in _encoreFoundation.lightTheme) {
    return [makeColorSetClass(colorSet), className].join(' ').trim();
  }

  return [makeCampaignCategoryColorSetClass(colorSet), className].join(' ').trim();
}
/**
 * Returns the css variable format of a semanticColor
 * @param semanticColor - a semanticColor (eg 'textBase,' 'decorativeSubdued')
 * @returns a css variable in kebabcase format with a '--' prefix
 */


function cssColorVar(semanticColor) {
  return "--".concat(camelToKebab(semanticColor));
}

function cssColorSet(colorSetObject) {
  return Object.entries(getFlatCamelCaseMap(colorSetObject)).map(function (_ref3) {
    var _ref4 = (0, _slicedToArray2["default"])(_ref3, 2),
        token = _ref4[0],
        hex = _ref4[1];

    return "\n      ".concat(cssColorVar(token), ": ").concat(hex, " !important;\n    ");
  });
}
/**
 * Get a CSS value that can be used when authoring CSS color rules
 * @param semanticColor - a semanticColor (eg 'textBase,' 'decorativeSubdued')
 * @param [fallbackColor] - a valid CSS <color> value as fallback
 * @returns a string in the form of `var(--semantic-color-variable, <fallback>)`
 */


function cssColorValue(semanticColor, fallbackColor) {
  var flattenedDefaultColorSet = getFlatCamelCaseMap(defaultColorSet);
  var fallback = flattenedDefaultColorSet[semanticColor];

  if (fallbackColor) {
    fallback = fallbackColor;
  }

  return "var(".concat(cssColorVar(semanticColor), ", ").concat(fallback, ")");
}

var semanticBackgroundColors = {
  backgroundBase: 'backgroundBase',
  backgroundHighlight: 'backgroundHighlight',
  backgroundPress: 'backgroundPress',
  backgroundElevatedBase: 'backgroundElevatedBase',
  backgroundElevatedHighlight: 'backgroundElevatedHighlight',
  backgroundElevatedPress: 'backgroundElevatedPress',
  backgroundTintedBase: 'backgroundTintedBase',
  backgroundTintedHighlight: 'backgroundTintedHighlight',
  backgroundTintedPress: 'backgroundTintedPress',
  backgroundUnsafeForSmallTextBase: 'backgroundUnsafeForSmallTextBase',
  backgroundUnsafeForSmallTextHighlight: 'backgroundUnsafeForSmallTextHighlight',
  backgroundUnsafeForSmallTextPress: 'backgroundUnsafeForSmallTextPress'
};
exports.semanticBackgroundColors = semanticBackgroundColors;
var semanticTextColors = {
  textBase: 'textBase',
  textSubdued: 'textSubdued',
  textBrightAccent: 'textBrightAccent',
  textNegative: 'textNegative',
  textWarning: 'textWarning',
  textPositive: 'textPositive',
  textAnnouncement: 'textAnnouncement'
};
exports.semanticTextColors = semanticTextColors;
var semanticEssentialColors = {
  essentialBase: 'essentialBase',
  essentialSubdued: 'essentialSubdued',
  essentialBrightAccent: 'essentialBrightAccent',
  essentialNegative: 'essentialNegative',
  essentialWarning: 'essentialWarning',
  essentialPositive: 'essentialPositive',
  essentialAnnouncement: 'essentialAnnouncement'
};
exports.semanticEssentialColors = semanticEssentialColors;
var semanticDecorativeColors = {
  decorativeBase: 'decorativeBase',
  decorativeSubdued: 'decorativeSubdued'
};
exports.semanticDecorativeColors = semanticDecorativeColors;

var semanticColors = _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, semanticBackgroundColors), semanticTextColors), semanticEssentialColors), semanticDecorativeColors); // TODO export this from foundation?
// https://ghe.spotify.net/encore/foundation/blob/master/src/extensions/theme/color-tune/predicates.ts


exports.semanticColors = semanticColors;

var calcAccessibleContrast = function calcAccessibleContrast(color1, color2) {
  return color1.contrast(color2);
}; // TODO export this from foundation?
// https://ghe.spotify.net/encore/foundation/blob/master/src/extensions/theme/utils.js#L8


exports.calcAccessibleContrast = calcAccessibleContrast;

var getHighContrastColor = function getHighContrastColor(color) {
  var white = (0, _color["default"])('#FFFFFF');
  var black = (0, _color["default"])('#000000');
  var contrastWhite = calcAccessibleContrast(color, white);
  var contrastBlack = calcAccessibleContrast(color, black);
  return contrastWhite >= contrastBlack ? white : black;
};
/**
 * Create a simple color set from a custom color, that meets contrast requirements
 * @param backgroundValue
 * @param [suggestedForegroundValue]
 */


var createBasicColorSet = function createBasicColorSet(backgroundValue, suggestedForegroundValue) {
  var adjustment = 5;
  var accessibleContrast = 4.45;
  var background = (0, _color["default"])(backgroundValue);
  var suggestedForeground = undefined;

  if (suggestedForegroundValue) {
    suggestedForeground = (0, _color["default"])(suggestedForegroundValue);
  }

  var foreground = suggestedForeground && calcAccessibleContrast(background, suggestedForeground) >= accessibleContrast ? suggestedForeground : getHighContrastColor(background);
  var foregroundIsLighter = foreground.luminosity() >= background.luminosity();
  var backgroundHighlight = foregroundIsLighter ? (0, _color["default"])(background.lightness(background.hsl().object().l - adjustment)) : (0, _color["default"])(background.lightness(background.hsl().object().l + adjustment));
  var backgroundPress = foregroundIsLighter ? (0, _color["default"])(backgroundHighlight.lightness(backgroundHighlight.hsl().object().l - adjustment)) : (0, _color["default"])(backgroundHighlight.lightness(backgroundHighlight.hsl().object().l + adjustment));
  return {
    background: {
      base: background.hex(),
      highlight: backgroundHighlight.hex(),
      press: backgroundPress.hex(),
      elevated: {
        base: background.hex(),
        highlight: backgroundHighlight.hex(),
        press: backgroundPress.hex()
      },
      tinted: {
        base: background.hex(),
        highlight: backgroundHighlight.hex(),
        press: backgroundPress.hex()
      },
      unsafeForSmallText: {
        base: background.hex(),
        highlight: backgroundHighlight.hex(),
        press: backgroundPress.hex()
      }
    },
    text: {
      base: foreground.hex(),
      subdued: foreground.hex(),
      brightAccent: foreground.hex(),
      negative: foreground.hex(),
      warning: foreground.hex(),
      positive: foreground.hex(),
      announcement: foreground.hex()
    },
    essential: {
      base: foreground.hex(),
      subdued: foreground.hex(),
      brightAccent: foreground.hex(),
      negative: foreground.hex(),
      warning: foreground.hex(),
      positive: foreground.hex(),
      announcement: foreground.hex()
    },
    decorative: {
      base: foreground.hex(),
      subdued: foreground.hex()
    }
  };
};

exports.createBasicColorSet = createBasicColorSet;
