// @spotify-internal/adstudio-tape/lib/components/TableContainer/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = TableContainer;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _spReactSticky = require("sp-react-sticky");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _LoadingSpinner = _interopRequireDefault(require("../LoadingSpinner"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _flex = require("../../styles/common/flex");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var propTypes = {
  /** Can accept TableBody or any HTML */
  body: _propTypes["default"].node.isRequired,

  /** Can accept TableHeader or any HTML */
  className: _propTypes["default"].string,
  header: _propTypes["default"].node.isRequired,
  empty: _propTypes["default"].node,
  isLoading: _propTypes["default"].bool
};

var StyledTempContentOuter = _styledComponents["default"].div.withConfig({
  displayName: "TableContainer__StyledTempContentOuter",
  componentId: "sc-1e280tc-0"
})(["", " ", ""], _flex.row, _flex.rowAlignCenter);

var StyledTempContentInner = _styledComponents["default"].div.withConfig({
  displayName: "TableContainer__StyledTempContentInner",
  componentId: "sc-1e280tc-1"
})(["color:", ";margin-top:85px;text-align:center;", ""], _encoreFoundation.gray75, _flex.column);

var StyledTableContainer = (0, _styledComponents["default"])(_spReactSticky.StickyContainer).withConfig({
  displayName: "TableContainer__StyledTableContainer",
  componentId: "sc-1e280tc-2"
})(["overflow-x:auto;position:relative;"]);

var TemporaryContent = function TemporaryContent(props) {
  return /*#__PURE__*/_react["default"].createElement(StyledTempContentOuter, null, /*#__PURE__*/_react["default"].createElement(StyledTempContentInner, null, props.children));
};

function TableContainer(_ref) {
  var className = _ref.className,
      header = _ref.header,
      body = _ref.body,
      empty = _ref.empty,
      isLoading = _ref.isLoading;
  var content;

  if (isLoading) {
    content = /*#__PURE__*/_react["default"].createElement(TemporaryContent, null, /*#__PURE__*/_react["default"].createElement(_LoadingSpinner["default"], null));
  } else if (empty) {
    content = /*#__PURE__*/_react["default"].createElement(TemporaryContent, null, empty);
  } else {
    content = body;
  }

  return /*#__PURE__*/_react["default"].createElement(StyledTableContainer, {
    className: className,
    "data-test": "table"
  }, header, content);
}

TemporaryContent.propTypes = {
  children: _propTypes["default"].node
};
TableContainer.propTypes = propTypes;
