// @spotify-internal/encore-web/es/components/Backdrop/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useEffect } from 'react';
import { OverlayBackdrop } from "./OverlayBackdrop";
export function Backdrop(_ref) {
  var _ref$onClose = _ref.onClose,
      onClose = _ref$onClose === void 0 ? function () {} : _ref$onClose,
      props = _objectWithoutProperties(_ref, ["onClose"]);

  var backdropRef = /*#__PURE__*/React.createRef();
  /* If backdrop is clicked, execute onClose */

  var onClick = function onClick(event) {
    if (event.target === backdropRef.current) {
      onClose(event);
    }
  };
  /* if escape key is pressed, execute onClose */


  var onKeydown = function onKeydown(event) {
    if (event.keyCode === 27) {
      onClose(event);
    }
  };

  useEffect(function () {
    /* add event listener for escape key press */
    window.addEventListener('keydown', onKeydown, true);
    /* clean up */

    return function () {
      window.removeEventListener('keydown', onKeydown, true);
    };
  });
  return /*#__PURE__*/React.createElement(OverlayBackdrop, Object.assign({
    ref: backdropRef,
    onClick: onClick
  }, props));
}
/**
* **Backdrop** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/Backdrop) | [Storybook](https://encore-web.spotify.net/?path=/story/components-backdrop--default) | [Encore](https://encore.spotify.net/web/components/backdrop) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=0:1206) |
*
* Backdrop is a tinted overlay on page content, under a popover or model. It disables interactive elements on the page, and uses the "Over media" color set.
*
* @example
* export const Default = () => <Backdrop />;
*
*/

export default Backdrop;
