// @spotify-internal/adstudio-shared/lib/components/TSTableBrowser/Pagination/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Pagination = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var DEFAULT_PAGE_SIZE_OPTIONS = [{
  key: 5,
  value: 5
}, {
  key: 10,
  value: 10
}, {
  key: 20,
  value: 20
}, {
  key: 50,
  value: 50
}, {
  key: 100,
  value: 100
}];
var GrayType = (0, _styledComponents["default"])(_adstudioTape.Type).withConfig({
  displayName: "Pagination__GrayType",
  componentId: "sc-2uzcz6-0"
})(["color:", ";vertical-align:top;"], _encoreAdvertisingWeb.gray50);

var PageSizeSection = _styledComponents["default"].div.withConfig({
  displayName: "Pagination__PageSizeSection",
  componentId: "sc-2uzcz6-1"
})(["text-align:left;"]);

var StyledPaginationControls = (0, _styledComponents["default"])(_encoreAdvertisingWeb.PaginationControls).withConfig({
  displayName: "Pagination__StyledPaginationControls",
  componentId: "sc-2uzcz6-2"
})(["justify-content:flex-end;"]);

var GridContainer = _styledComponents["default"].div.withConfig({
  displayName: "Pagination__GridContainer",
  componentId: "sc-2uzcz6-3"
})(["display:grid;grid-template-columns:repeat(2,1fr);height:48px;align-items:center;"]); // Fallback if generator function for translated string is not provided


var defaultStringGenerator = function defaultStringGenerator(rangeStart, rangeEnd, total) {
  return "".concat(rangeStart, " \u2013 ").concat(rangeEnd, " of ").concat(total || 'many');
};

var Pagination = /*#__PURE__*/function (_PureComponent) {
  _inherits(Pagination, _PureComponent);

  var _super = _createSuper(Pagination);

  function Pagination() {
    var _this;

    _classCallCheck(this, Pagination);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));
    _this.state = {
      showPageSizeDropdown: false
    };

    _this.handlePrev = function () {
      var _this$props = _this.props,
          handleSelect = _this$props.handleSelect,
          offset = _this$props.offset,
          limit = _this$props.limit;
      handleSelect(Math.max(offset - limit, 0).toString());
    };

    _this.handleNext = function () {
      var _this$props2 = _this.props,
          handleSelect = _this$props2.handleSelect,
          offset = _this$props2.offset,
          limit = _this$props2.limit,
          total = _this$props2.total;
      var newOffset = total !== undefined ? Math.min(offset + limit, total - limit) : offset + limit;
      handleSelect(newOffset.toString());
    };

    _this.handleItemClick = function (limit) {
      _this.setState({
        showPageSizeDropdown: false
      });

      _this.props.handlePageSizeChange(limit);
    };

    return _this;
  }

  _createClass(Pagination, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props3 = this.props,
          offset = _this$props3.offset,
          limit = _this$props3.limit,
          total = _this$props3.total,
          _this$props3$pageSize = _this$props3.pageSizeOptions,
          pageSizeOptions = _this$props3$pageSize === void 0 ? DEFAULT_PAGE_SIZE_OPTIONS : _this$props3$pageSize,
          _this$props3$stringGe = _this$props3.stringGenerator,
          stringGenerator = _this$props3$stringGe === void 0 ? defaultStringGenerator : _this$props3$stringGe;
      var showPageSizeDropdown = this.state.showPageSizeDropdown;
      var firstElementNumber = Math.max(offset + 1, 1);
      var lastElementNumber = total !== undefined ? Math.min(offset + limit, total) : offset + limit;
      var selectedPageSizeOption = pageSizeOptions.find(function (opt) {
        return opt.key === limit;
      }) || pageSizeOptions[0];
      var canGoToPrevPage = offset !== 0; // If total is undefined, we assume there are more results

      var canGoToNextPage = total === undefined || offset + limit < total;
      return /*#__PURE__*/_react["default"].createElement(GridContainer, null, /*#__PURE__*/_react["default"].createElement(_adstudioTape.GridColumns, {
        defaultColumnSpanCount: 1
      }, /*#__PURE__*/_react["default"].createElement(PageSizeSection, null, /*#__PURE__*/_react["default"].createElement(GrayType, {
        variant: "body-2-bold"
      }, "Rows per page: "), /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.PaginationDropdownTrigger, {
        onShow: function onShow() {
          return _this2.setState({
            showPageSizeDropdown: true
          });
        },
        onHide: function onHide() {
          return _this2.setState({
            showPageSizeDropdown: false
          });
        },
        overlay: showPageSizeDropdown && /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.PaginationDropdownList, null, pageSizeOptions.map(function (opt) {
          return /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.PaginationDropdownItem, {
            key: opt.value
          }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.PaginationDropdownLink, {
            href: "#",
            onClick: function onClick() {
              return _this2.handleItemClick(opt.value.toString());
            },
            selected: selectedPageSizeOption.value === opt.value
          }, opt.value));
        }))
      }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.PaginationDropdown, null, selectedPageSizeOption.value)))), /*#__PURE__*/_react["default"].createElement(_adstudioTape.GridColumns, {
        defaultColumnSpanCount: 1
      }, /*#__PURE__*/_react["default"].createElement(StyledPaginationControls, {
        "data-test": 'table-browser-pagination-controls',
        onDecrement: canGoToPrevPage ? this.handlePrev : undefined,
        onIncrement: canGoToNextPage ? this.handleNext : undefined
      }, /*#__PURE__*/_react["default"].createElement(GrayType, {
        variant: "body-2-bold",
        "data-test": 'table-browser-pagination-text'
      }, stringGenerator(firstElementNumber, lastElementNumber, total)))));
    }
  }]);

  return Pagination;
}(_react.PureComponent);

exports.Pagination = Pagination;
var _default = Pagination;
exports["default"] = _default;
