// @spotify-internal/encore-web/advertising/components/Panel/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Panel = Panel;
exports.PanelContent = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _ButtonTertiary = require("../../../cjs/components/ButtonTertiary");

var _semanticTheme = require("../../../cjs/styles/semantic-theme");

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _excluded = ["label", "className"];

var Container = _styledComponents["default"].div.withConfig({
  displayName: "Panel__Container",
  componentId: "sc-1kl3bck-0"
})(["align-items:center;border-radius:", ";border:", ";border-color:", ";box-shadow:", ";display:flex;justify-content:space-between;padding:", ";"], _encoreFoundation.spacer4, function (props) {
  if (props.shadow) return 'none';
  if (props.border) return "1px solid ".concat((0, _semanticTheme.cssColorValue)(_semanticTheme.semanticColors.essentialSubdued));
  return 'none';
}, function (props) {
  if (props.error) return (0, _semanticTheme.cssColorValue)(_semanticTheme.semanticColors.essentialNegative);
  if (props.warning) return (0, _semanticTheme.cssColorValue)(_semanticTheme.semanticColors.essentialWarning);
  return null;
}, function (props) {
  return props.shadow && "0 1px 3px 0 ".concat((0, _semanticTheme.cssColorValue)(_semanticTheme.semanticColors.decorativeSubdued));
}, function (props) {
  return props.padding ? _encoreFoundation.spacer16 : 0;
});

var PanelContent = _styledComponents["default"].div.withConfig({
  displayName: "Panel__PanelContent",
  componentId: "sc-1kl3bck-1"
})(["align-items:center;display:flex;flex-grow:1;justify-content:flex-start;min-width:0;"]);

exports.PanelContent = PanelContent;

var CTA = _styledComponents["default"].div.withConfig({
  displayName: "Panel__CTA",
  componentId: "sc-1kl3bck-2"
})(["flex-shrink:0;"]);

var CTAButton = (0, _styledComponents["default"])(_ButtonTertiary.ButtonTertiary).withConfig({
  displayName: "Panel__CTAButton",
  componentId: "sc-1kl3bck-3"
})(["padding-right:0;display:", ";flex-direction:", ";flex-shrink:0;"], function (props) {
  return props.$ctaListStacked && 'flex';
}, function (props) {
  return props.$ctaListStacked && 'column';
});

function Panel(_ref) {
  var _ref$border = _ref.border,
      border = _ref$border === void 0 ? true : _ref$border,
      children = _ref.children,
      className = _ref.className,
      ctaList = _ref.ctaList,
      ctaListStacked = _ref.ctaListStacked,
      error = _ref.error,
      _ref$padding = _ref.padding,
      padding = _ref$padding === void 0 ? true : _ref$padding,
      shadow = _ref.shadow,
      warning = _ref.warning;

  var result = /*#__PURE__*/_react["default"].createElement(Container, {
    className: className,
    error: error,
    warning: warning,
    border: border,
    shadow: shadow,
    padding: padding
  }, /*#__PURE__*/_react["default"].createElement(PanelContent, {
    className: className
  }, children), ctaList && ctaList.length > 0 && /*#__PURE__*/_react["default"].createElement(CTA, null, ctaList.map(function (_ref2) {
    var label = _ref2.label,
        ctaClassName = _ref2.className,
        ctaProps = (0, _objectWithoutProperties2["default"])(_ref2, _excluded);
    return /*#__PURE__*/_react["default"].createElement(CTAButton, Object.assign({
      semanticColor: _semanticTheme.semanticColors.textBrightAccent,
      key: "panel-cta-".concat(label),
      buttonSize: "sm",
      className: ctaClassName,
      $ctaListStacked: ctaListStacked
    }, ctaProps), label);
  })));

  return result;
}
