// @spotify-internal/adstudioasset/proto/com/spotify/adstudioasset/model_pb.js
// source: com/spotify/adstudioasset/model.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.com.spotify.adstudioasset.AssetState', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.AssetStateMix', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.AssetStateVo', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.AudioMatchReviewState', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.AudioMatchState', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.AudioMediaType', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.BackgroundType', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.ImageType', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.MixMode', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.VideoFormatType', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.VoiceoverRejectionCode', null, global);
goog.exportSymbol('proto.com.spotify.adstudioasset.VoiceoverSystem', null, global);
/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AssetState = {
  STATE_UNKNOWN: 0,
  WAITING_UPLOAD: 1,
  WAITING_TRANSCODING: 2,
  TRANSCODING_FINISHED: 3,
  UPLOAD_TIMEOUT: 4,
  TRANSCODING_FAILED: 5,
  INTERNAL_ERROR: 6,
  INVALID_FAILED: 7,
  AUDIO_MATCH_PROCESSING: 8
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AssetStateVo = {
  VO_STATE_UNKNOWN: 0,
  WAITING_SCRIPT_APPROVAL: 1,
  VO_SCRIPT_REJECTED: 2,
  WAITING_VOICEOVER: 3,
  VO_READY: 4,
  VO_REVISION_CANCELLED: 5,
  VO_FAILED: 6,
  VO_APPROVED: 7,
  VO_REJECTED: 8,
  WAITING_VO_ADMIN_APPROVAL: 9,
  WAITING_VO_APPROVAL: 10,
  WAITING_VO_UPLOAD: 11,
  UPLOAD_VO_TIMEOUT: 12
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AssetStateMix = {
  MIX_STATE_UNKNOWN: 0,
  WAITING_MIX_DEPENDENCIES: 1,
  WAITING_MIX_TRANSCODE: 2,
  MIX_STATE_COMPLETE: 3,
  MIX_STATE_APPROVED: 4,
  MIX_STATE_REJECTED: 5,
  MIX_STATE_FAILED: 6
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.BackgroundType = {
  STOCK: 0,
  USER_UPLOADED: 1
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AudioMediaType = {
  AUDIO_UNSET: 0,
  AUDIO_OGG: 1,
  AUDIO_MP3: 2,
  AUDIO_WAV: 3
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.VideoFormatType = {
  VIDEO_UNSET: 0,
  VIDEO_MP4: 1,
  VIDEO_QUICKTIME: 2
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AudioMatchState = {
  UNKNOWN: 0,
  SONALYTIC_PROCESSING: 1,
  SONALYTIC_COMPLETE: 2,
  UNSENT: 3
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.AudioMatchReviewState = {
  UNSET: 0,
  CONFIRMED_MATCH: 1,
  INVALID_MATCH: 2
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.MixMode = {
  MIX_MODE_UNSET: 0,
  VOCAL_RIDE: 1,
  AUTODUCK: 2
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.VoiceoverSystem = {
  VOICEOVER_SYSTEM_UNSET: 0,
  VOICE_BUNNY: 1,
  USER_UPLOAD: 2,
  TEXT_TO_SPEECH_UPLOAD: 3
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.ImageType = {
  STOCK_IMAGE: 0,
  USER_UPLOADED_IMAGE: 1
};

/**
 * @enum {number}
 */
proto.com.spotify.adstudioasset.VoiceoverRejectionCode = {
  REJECTION_CODE_UNSET: 0,
  PROJECT_CANCELLED: 1,
  PROJECT_SUBMITTED_INCORRECTLY: 2,
  WANT_TO_CHANGE_ACTOR: 3,
  PROJECT_TOO_LATE: 4,
  UNEXPECTED_AGE_OR_GENDER: 5,
  UNEXPECTED_ACCENT: 6,
  TOO_ROBOTIC: 7,
  UNLIKE_SAMPLE: 8,
  WRONG_FIT_TONE: 9,
  ACTOR_DIDNT_FOLLOW_INSTRUCTIONS: 10,
  UNEXPECTED_PRONUNCIATION: 11,
  DISTORTION_OR_TOO_LOUD: 12,
  BACKGROUND_NOISE_OR_ECHO: 13,
  POORLY_EDITED: 14
};

goog.object.extend(exports, proto.com.spotify.adstudioasset);
