// @spotify-internal/encore-web/es/components/NavPillListItem/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { KeyboardDetectionContext } from "../../contexts/KeyboardDetectionContext";
import { Link } from "./Link";
import { ListItem } from "./ListItem";
var NavPillListItemComponent = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var label = _ref.label,
      active = _ref.active,
      _ref$theme = _ref.theme,
      theme = _ref$theme === void 0 ? {} : _ref$theme,
      _ref$component = _ref.component,
      component = _ref$component === void 0 ? 'a' : _ref$component,
      props = _objectWithoutProperties(_ref, ["label", "active", "theme", "component"]);

  var tabIndexAttribute = active ? null : {
    tabIndex: '-1'
  };

  var _useContext = useContext(KeyboardDetectionContext),
      isUsingKeyboard = _useContext.isUsingKeyboard;

  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: theme
  }, /*#__PURE__*/React.createElement(ListItem, {
    role: "presentation"
  }, /*#__PURE__*/React.createElement(Link, Object.assign({
    ref: ref,
    role: "tab"
  }, tabIndexAttribute, {
    isActive: active,
    component: component,
    "aria-selected": active,
    isUsingKeyboard: isUsingKeyboard
  }, props), label)));
});
var navPillListItemHelpers = {
  darkTheme: {
    navPillListItem: 'dark'
  }
};
/**
* **NavPillListItem** ![Status: Production](https://img.shields.io/badge/PRODUCTION-%2357B560|height=14)
*
* [GitHub](https://ghe.spotify.net/encore/web/tree/master/src/components/NavPillListItem) | [Storybook](https://encore-web.spotify.net/?path=/story/components-navpill-navpilllistitem--default) | [Encore](https://encore.spotify.net/web/components/navigation?format=figma-figma#nav-pill) | [Figma](https://www.figma.com/file/bnQXqGkLeiTlbk8JXGDcsX/Encore-Web---Light-Theme-Toolkit?node-id=423:1550) |
*
* An individual tab within the NavPill component
*
* @example
* export const Default = () => (
*  <NavPillList>
*    <NavPillListItem label="Overview" href="#NavPill-overview" />
*  </NavPillList>
*);
*
*/

export var NavPillListItem = Object.assign(NavPillListItemComponent, navPillListItemHelpers);
export default NavPillListItem;
