// @spotify-internal/adstudio-shared/lib/components/AutosuggestInput/SelectedInput.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var IconX = _adstudioTape.Icons.IconX;
var ICON_SIZE = '24px';
var SEARCH_CONTAINER_HEIGHT = '48px';
var SEARCH_CONTAINER_MOBILE_HEIGHT = '32px';

var SelectedInputContainer = _styledComponents["default"].div.withConfig({
  displayName: "SelectedInput__SelectedInputContainer",
  componentId: "sc-1sabfke-0"
})(["display:block;position:relative;width:100%;"]);

var SelectedInputValue = _styledComponents["default"].div.withConfig({
  displayName: "SelectedInput__SelectedInputValue",
  componentId: "sc-1sabfke-1"
})(["@media (max-width:", "){border-bottom:1px solid ", ";box-sizing:content-box;height:32px;padding:0 8px;}@media (min-width:", "){border:1px solid ", ";border-radius:4px;height:48px;padding:12px ", ";&::before{background-color:", ";content:'';height:100%;left:0;opacity:0.1;position:absolute;top:0;width:100%;z-index:-1;}", "}"], _encoreAdvertisingWeb.screenXsMax, _encoreAdvertisingWeb.gray75, _encoreAdvertisingWeb.screenSmMin, _encoreAdvertisingWeb.gray75, _encoreAdvertisingWeb.spacer16, _encoreAdvertisingWeb.azure, function (props) {
  return props.disabled && (0, _styledComponents.css)(["color:", ";cursor:not-allowed;&::before{background-color:", ";opacity:1;}"], _encoreAdvertisingWeb.gray45, _encoreAdvertisingWeb.gray95);
});

var CloseIcon = (0, _styledComponents["default"])(IconX).withConfig({
  displayName: "SelectedInput__CloseIcon",
  componentId: "sc-1sabfke-2"
})(["@media (max-width:", "){cursor:pointer;position:absolute;right:calc((", " - ", ") / 2);top:calc((", " - ", ") / 2);}@media (min-width:", "){cursor:pointer;position:absolute;right:calc((", " - ", ") / 2);top:calc((", " - ", ") / 2);}"], _encoreAdvertisingWeb.screenXsMax, SEARCH_CONTAINER_MOBILE_HEIGHT, ICON_SIZE, SEARCH_CONTAINER_MOBILE_HEIGHT, ICON_SIZE, _encoreAdvertisingWeb.screenSmMin, SEARCH_CONTAINER_HEIGHT, ICON_SIZE, SEARCH_CONTAINER_HEIGHT, ICON_SIZE);

var SelectedInput = function SelectedInput(_ref) {
  var disabled = _ref.disabled,
      value = _ref.value,
      onClickRemove = _ref.onClickRemove,
      valueField = _ref.valueField;
  return /*#__PURE__*/_react["default"].createElement(SelectedInputContainer, null, /*#__PURE__*/_react["default"].createElement(SelectedInputValue, {
    disabled: disabled,
    value: value[valueField]
  }, value[valueField]), !disabled && /*#__PURE__*/_react["default"].createElement(CloseIcon, {
    onClick: function onClick() {
      return onClickRemove(value);
    }
  }));
};

SelectedInput.propTypes = {
  disabled: _propTypes["default"].bool,
  onClickRemove: _propTypes["default"].func,
  value: _propTypes["default"].object,
  valueField: _propTypes["default"].string.isRequired
};
var _default = SelectedInput;
exports["default"] = _default;
