// @spotify-internal/adstudio-tape/lib/components/ThumbName/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = NameNav;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _AudioStampPlayer = _interopRequireDefault(require("../AudioStampPlayer"));

var _Thumb = _interopRequireDefault(require("../Thumb"));

var _encoreWeb = require("@spotify-internal/encore-web");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var FlexContainer = _styledComponents["default"].div.withConfig({
  displayName: "ThumbName__FlexContainer",
  componentId: "jqehqp-0"
})(["align-items:center;display:flex;flex-direction:row;"]);

var NameContainer = _styledComponents["default"].div.withConfig({
  displayName: "ThumbName__NameContainer",
  componentId: "jqehqp-1"
})(["display:block;overflow:hidden;padding-left:15px;text-overflow:ellipsis;white-space:nowrap;"]);

var LinkableName = _styledComponents["default"].a.withConfig({
  displayName: "ThumbName__LinkableName",
  componentId: "jqehqp-2"
})(["color:", ";text-decoration:none;& p:hover{text-decoration:underline;}"], _encoreFoundation.gray10);

var propTypes = {
  audioSrc: _propTypes["default"].string,
  image: _propTypes["default"].shape({
    alt: _propTypes["default"].string,
    src: _propTypes["default"].string,
    size: _propTypes["default"].number
  }),
  name: _propTypes["default"].string.isRequired,
  subContent: _propTypes["default"].node,
  url: _propTypes["default"].string
};

function NameNav(_ref) {
  var image = _ref.image,
      name = _ref.name,
      subContent = _ref.subContent,
      audioSrc = _ref.audioSrc,
      url = _ref.url;
  return /*#__PURE__*/_react["default"].createElement(FlexContainer, null, audioSrc && /*#__PURE__*/_react["default"].createElement(_AudioStampPlayer["default"], {
    audioSrc: audioSrc,
    imageSrc: image.src,
    size: 64
  }) || /*#__PURE__*/_react["default"].createElement(_Thumb["default"], image), /*#__PURE__*/_react["default"].createElement(NameContainer, null, url ? /*#__PURE__*/_react["default"].createElement(LinkableName, {
    href: url
  }, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    variant: _encoreWeb.Type.body2,
    weight: _encoreWeb.Type.bold,
    condensed: true
  }, name)) : /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
    variant: _encoreWeb.Type.body2,
    weight: _encoreWeb.Type.bold,
    condensed: true
  }, name), !!subContent && subContent));
}

NameNav.propTypes = propTypes;
