// @spotify-internal/adstudio-shared/lib/components/GeoTargetingSearch/Value/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _adstudioTape = require("@spotify-internal/adstudio-tape");

var _encoreAdvertisingWeb = require("@spotify-internal/encore-advertising-web");

var _selectors = require("../selectors");

var _deprecatedGeoHelpers = require("../../../utils/deprecatedGeoHelpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var GridContainer = _styledComponents["default"].div.withConfig({
  displayName: "Item__GridContainer",
  componentId: "azngxo-0"
})(["display:grid;grid-template-columns:180px 1fr;align-items:center;p{margin-bottom:0;margin-top:0;padding:14px;padding-right:", ";}"], _encoreAdvertisingWeb.spacer48);

var LocationType = _styledComponents["default"].div.withConfig({
  displayName: "Item__LocationType",
  componentId: "azngxo-1"
})(["height:100%;background-clip:content-box;background-color:", ";color:", ";font-weight:400;text-transform:capitalize;display:grid;h6{text-align:right;padding:", ";line-height:20px;}"], _encoreAdvertisingWeb.gray95, function (props) {
  return props.disabled ? _encoreAdvertisingWeb.gray60 : _encoreAdvertisingWeb.gray40;
}, _encoreAdvertisingWeb.spacer16);

var LocationName = _styledComponents["default"].div.withConfig({
  displayName: "Item__LocationName",
  componentId: "azngxo-2"
})(["color:", ";", ""], _encoreAdvertisingWeb.gray60, function (props) {
  return props.disabled && "background-color: ".concat(_encoreAdvertisingWeb.gray95, ";");
});

var DeselectButton = (0, _styledComponents["default"])(_encoreAdvertisingWeb.ButtonIcon).withConfig({
  displayName: "Item__DeselectButton",
  componentId: "azngxo-3"
})(["position:absolute;display:grid;top:50%;transform:translateY(-50%);right:12px;opacity:0;"]);

var CloseableListItem = _styledComponents["default"].div.withConfig({
  displayName: "Item__CloseableListItem",
  componentId: "azngxo-4"
})(["position:relative;&:hover ", "{opacity:1;}"], DeselectButton);

var DeprecationMarker = (0, _styledComponents["default"])(_encoreAdvertisingWeb.Type.p).withConfig({
  displayName: "Item__DeprecationMarker",
  componentId: "azngxo-5"
})(["", ""], function (props) {
  if (props.deprecated) {
    return "color: ".concat(props.disabled ? _encoreAdvertisingWeb.pink : _encoreAdvertisingWeb.redLight, ";");
  }
});
var IconX = _adstudioTape.Icons.IconX;

var SelectedLocationsListItem = function SelectedLocationsListItem(_ref) {
  var disabled = _ref.disabled,
      handleDeprecatedGeos = _ref.handleDeprecatedGeos,
      locations = _ref.locations,
      _onClick = _ref.onClick,
      strings = _ref.strings,
      type = _ref.type;
  return /*#__PURE__*/_react["default"].createElement(GridContainer, {
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement(LocationType, {
    disabled: disabled
  }, /*#__PURE__*/_react["default"].createElement(_encoreAdvertisingWeb.Type.h6, {
    condensed: true,
    variant: _encoreAdvertisingWeb.Type.cta3,
    className: "".concat((0, _selectors.getMappedGeoType)(type).toLowerCase(), "-geo-type")
  }, (0, _selectors.getMappedGeoType)(type, strings))), /*#__PURE__*/_react["default"].createElement(LocationName, {
    disabled: disabled
  }, locations.map(function (loc) {
    var deprecated = !!(handleDeprecatedGeos && (0, _deprecatedGeoHelpers.isGeoDeprecated)(loc));
    var ariaLabel = deprecated ? strings.ARIA_REMOVE_THIS_UNAVAILABLE_LOCATION : strings.ARIA_REMOVE_THIS_LOCATION;
    return /*#__PURE__*/_react["default"].createElement(CloseableListItem, {
      key: loc.id
    }, /*#__PURE__*/_react["default"].createElement(DeprecationMarker, {
      condensed: true,
      deprecated: deprecated,
      disabled: disabled,
      variant: _encoreAdvertisingWeb.Type.body1,
      color: _encoreAdvertisingWeb.gray60
    }, (0, _selectors.getGeoLabelFromSuggestion)(loc)), !disabled && /*#__PURE__*/_react["default"].createElement(DeselectButton, {
      "aria-label": ariaLabel,
      onClick: function onClick() {
        return _onClick(loc);
      },
      role: "button",
      tabIndex: "0"
    }, /*#__PURE__*/_react["default"].createElement(IconX, null)));
  })));
};

SelectedLocationsListItem.propTypes = {
  disabled: _propTypes["default"].bool,
  handleDeprecatedGeos: _propTypes["default"].bool,
  locations: _propTypes["default"].array.isRequired,
  onClick: _propTypes["default"].func.isRequired,
  strings: _propTypes["default"].object,
  type: _propTypes["default"].string.isRequired
};
var _default = SelectedLocationsListItem;
exports["default"] = _default;
