// @spotify-internal/adstudio-tape/lib/components/TableHeaderItem/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _TooltipInfo = _interopRequireDefault(require("../TooltipInfo"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _Icons = _interopRequireDefault(require("../Icons"));

var _orders = _interopRequireDefault(require("../../utils/const/orders"));

var _positions = _interopRequireDefault(require("../../utils/const/positions"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _flex = require("../../styles/common/flex");

var _overflow = require("../../styles/common/overflow");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var clickableStyle = (0, _styledComponents.css)(["cursor:pointer;&:focus{color:$black;outline:none;}"]);

var applyValign = function applyValign(props) {
  var valign = props.valign;

  if (!valign && props.wrap) {
    valign = 'middle';
  }

  switch (valign) {
    case 'middle':
      return (0, _styledComponents.css)(["display:flex;align-items:center;", ";"], props.alignRight && "justify-content: flex-end");

    case "bottom":
      return (0, _styledComponents.css)(["align-self:flex-end;"]);

    default:
      return '';
  }
};

var StyledComponentHeader = _styledComponents["default"].div.withConfig({
  displayName: "TableHeaderItem__StyledComponentHeader",
  componentId: "sc-5pz2gk-0"
})(["border-bottom:1px solid ", ";", ""], _encoreFoundation.gray75, _flex.column);

var StyledTableHeaderItem = _styledComponents["default"].div.withConfig({
  displayName: "TableHeaderItem__StyledTableHeaderItem",
  componentId: "sc-5pz2gk-1"
})(["border-bottom:1px solid ", ";white-space:", ";padding:", " ", " ", ";", ""], _encoreFoundation.gray75, function (props) {
  return props.wrap ? 'normal' : 'nowrap';
}, function (props) {
  return props.ensmallen ? "".concat(_encoreFoundation.spacer12, " ").concat(_encoreFoundation.spacer12, " ").concat(_encoreFoundation.spacer12, " ").concat(_encoreFoundation.spacer20, ";") : "".concat(_encoreFoundation.spacer20, " ").concat(_encoreFoundation.spacer16, ";");
}, function (props) {
  return props.alignRight ? 'text-align: right;' : '';
}, _flex.column, function (props) {
  return applyValign(props);
});

var StyledSortIcon = (0, _styledComponents["default"])(_Icons["default"].IconChartUp).withConfig({
  displayName: "TableHeaderItem__StyledSortIcon",
  componentId: "sc-5pz2gk-2"
})(["transform:translateY(-10%);vertical-align:middle;", ""], function (props) {
  return props.order === _orders["default"].DESC ? 'transform : translateY(-10%) rotate(180deg);' : '';
});

var StyledContent = _styledComponents["default"].div.withConfig({
  displayName: "TableHeaderItem__StyledContent",
  componentId: "sc-5pz2gk-3"
})(["", " ", " ", ""], function (props) {
  return props.onClick ? clickableStyle : '';
}, function (props) {
  return props.active ? "color: ".concat(_encoreFoundation.black) : '';
}, function (props) {
  return props.ellipseOverflow ? _overflow.ellipseOverflow : '';
});

var StyledType = (0, _styledComponents["default"])(_encoreWeb.Type).withConfig({
  displayName: "TableHeaderItem__StyledType",
  componentId: "sc-5pz2gk-4"
})(["&&{color:", ";display:block;line-height:12px;position:relative;", "}"], _encoreFoundation.gray60, function (props) {
  return props.ensmallen ? 'font-size: 10px; line-height: 10px;' : '';
});
var propTypes = {
  alignRight: _propTypes["default"].bool,
  condensed: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onClick: _propTypes["default"].func,
  order: _propTypes["default"].string,
  orderBy: _propTypes["default"].string,
  tooltipPlacement: _propTypes["default"].oneOf(_positions["default"]),
  tooltipText: _propTypes["default"].string,
  title: _propTypes["default"].string,
  value: _propTypes["default"].string,
  ellipseOverflow: _propTypes["default"].bool,
  headerComponent: _propTypes["default"].node,
  valign: _propTypes["default"].string,
  wrap: _propTypes["default"].bool
};

var TableHeaderItem = /*#__PURE__*/function (_Component) {
  _inherits(TableHeaderItem, _Component);

  var _super = _createSuper(TableHeaderItem);

  function TableHeaderItem() {
    _classCallCheck(this, TableHeaderItem);

    return _super.apply(this, arguments);
  }

  _createClass(TableHeaderItem, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          alignRight = _this$props.alignRight,
          className = _this$props.className,
          ensmallen = _this$props.condensed,
          ellipseOverflow = _this$props.ellipseOverflow,
          name = _this$props.name,
          onClick = _this$props.onClick,
          order = _this$props.order,
          orderBy = _this$props.orderBy,
          tooltipPlacement = _this$props.tooltipPlacement,
          tooltipText = _this$props.tooltipText,
          title = _this$props.title,
          value = _this$props.value,
          headerComponent = _this$props.headerComponent,
          valign = _this$props.valign,
          wrap = _this$props.wrap;
      var HeaderComponent = headerComponent;

      var applyFilter = function applyFilter() {
        var params = {
          orderBy: value,
          order: order === _orders["default"].ASC ? _orders["default"].DESC : _orders["default"].ASC
        };
        onClick(params);
      };

      var isActive = orderBy && value && orderBy === value;

      var Container = function Container(_ref) {
        var children = _ref.children;
        return tooltipText ? /*#__PURE__*/_react["default"].createElement(_TooltipInfo["default"], {
          icon: false,
          tooltipText: tooltipText,
          placement: tooltipPlacement,
          wrapperAlignRight: alignRight
        }, children) : /*#__PURE__*/_react["default"].createElement("div", {
          title: title
        }, children);
      };

      var SortIcon = function SortIcon() {
        return /*#__PURE__*/_react["default"].createElement(StyledSortIcon, {
          order: order,
          iconSize: ensmallen ? 8 : 16,
          color: "black"
        });
      };

      var clickProps = onClick && {
        onClick: applyFilter,
        tabIndex: '0',
        role: 'button'
      };

      if (HeaderComponent) {
        return /*#__PURE__*/_react["default"].createElement(StyledComponentHeader, {
          alignRight: alignRight,
          className: className
        }, /*#__PURE__*/_react["default"].createElement(HeaderComponent, null));
      }

      return /*#__PURE__*/_react["default"].createElement(StyledTableHeaderItem, {
        ensmallen: ensmallen,
        alignRight: alignRight,
        className: className,
        valign: valign,
        wrap: wrap,
        "data-test": "table-header-item"
      }, /*#__PURE__*/_react["default"].createElement(StyledType, {
        ensmallen: ensmallen,
        condensed: true,
        variant: _encoreWeb.Type.cta3
      }, /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(StyledContent, _extends({
        active: isActive,
        ellipseOverflow: ellipseOverflow
      }, clickProps), isActive && alignRight && /*#__PURE__*/_react["default"].createElement(SortIcon, null), name, isActive && !alignRight && /*#__PURE__*/_react["default"].createElement(SortIcon, null)))));
    }
  }]);

  return TableHeaderItem;
}(_react.Component);

exports["default"] = TableHeaderItem;
TableHeaderItem.propTypes = propTypes;
