// @spotify-internal/adstudio-tape/lib/components/UserProfile/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _encoreWeb = require("@spotify-internal/encore-web");

var _Icons = _interopRequireDefault(require("../Icons"));

var _PopoverNavigation = _interopRequireDefault(require("../PopoverNavigation"));

var _PopoverNavigationLink = _interopRequireDefault(require("../PopoverNavigationLink"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var IconChevronDownAlt = _Icons["default"].IconChevronDownAlt,
    IconChevronUpAlt = _Icons["default"].IconChevronUpAlt,
    ButtonIcon = _Icons["default"].ButtonIcon;

var UserProfileContainer = _styledComponents["default"].div.withConfig({
  displayName: "UserProfile__UserProfileContainer",
  componentId: "btgp7s-0"
})(["max-width:250px;padding-right:", ";position:relative;"], _encoreFoundation.spacer32);

var ProfileImage = _styledComponents["default"].img.withConfig({
  displayName: "UserProfile__ProfileImage",
  componentId: "btgp7s-1"
})(["border-radius:50%;box-shadow:0 0 5px 2px rgba(0,0,0,0.1);display:inline-block;height:34px;margin-right:", ";vertical-align:middle;width:34px;"], _encoreFoundation.spacer8);

var Username = _styledComponents["default"].div.withConfig({
  displayName: "UserProfile__Username",
  componentId: "btgp7s-2"
})(["display:inline-block;max-width:60%;p{max-width:100%;display:inline-block;overflow:hidden;text-overflow:ellipsis;vertical-align:middle;white-space:nowrap;}span{&::before{content:'\\A';white-space:pre;}}"]);

var Menu = _styledComponents["default"].div.withConfig({
  displayName: "UserProfile__Menu",
  componentId: "btgp7s-3"
})(["display:inline-block;font-size:0;position:absolute;right:0;top:50%;transform:translateY(-50%);vertical-align:middle;"]);

var Chevron = (0, _styledComponents["default"])(IconChevronDownAlt).withConfig({
  displayName: "UserProfile__Chevron",
  componentId: "btgp7s-4"
})(["", ""], function (_ref) {
  var theme = _ref.theme;
  return theme === 'dark' && (0, _styledComponents.css)(["color:", ";&:hover{color:", ";}"], _encoreFoundation.gray60, _encoreFoundation.white);
});
var OpenChevron = (0, _styledComponents["default"])(IconChevronUpAlt).withConfig({
  displayName: "UserProfile__OpenChevron",
  componentId: "btgp7s-5"
})(["", ""], function (_ref2) {
  var theme = _ref2.theme;
  return theme === 'dark' && (0, _styledComponents.css)(["color:", ";&:hover{color:", ";}"], _encoreFoundation.gray60, _encoreFoundation.white);
});

var UserProfile = /*#__PURE__*/function (_Component) {
  _inherits(UserProfile, _Component);

  var _super = _createSuper(UserProfile);

  function UserProfile() {
    var _this;

    _classCallCheck(this, UserProfile);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      show: false
    });

    return _this;
  }

  _createClass(UserProfile, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          links = _this$props.links,
          username = _this$props.username,
          description = _this$props.description,
          img = _this$props.img,
          theme = _this$props.theme;
      return /*#__PURE__*/_react["default"].createElement(UserProfileContainer, null, img && /*#__PURE__*/_react["default"].createElement(ProfileImage, {
        src: img,
        alt: "profile"
      }), /*#__PURE__*/_react["default"].createElement(Username, null, /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.p, {
        condensed: true,
        title: username,
        variant: _encoreWeb.Type.body3,
        weight: _encoreWeb.Type.book
      }, username, ' ', description && /*#__PURE__*/_react["default"].createElement(_encoreWeb.Type.span, {
        condensed: true,
        title: username,
        variant: _encoreWeb.Type.body4,
        weight: _encoreWeb.Type.book
      }, description))), /*#__PURE__*/_react["default"].createElement(Menu, null, /*#__PURE__*/_react["default"].createElement(_encoreWeb.PopoverTrigger, {
        onShow: function onShow() {
          return _this2.setState({
            show: true
          });
        },
        onHide: function onHide() {
          return _this2.setState({
            show: false
          });
        },
        overlay: this.state.show && /*#__PURE__*/_react["default"].createElement(_PopoverNavigation["default"], {
          arrow: "bottom",
          theme: "dark"
        }, links.map(function (navItem) {
          return /*#__PURE__*/_react["default"].createElement(_PopoverNavigationLink["default"], {
            href: navItem.href,
            key: "pop-nav-".concat(navItem.title),
            onClick: navItem.onClick,
            theme: "dark",
            target: navItem.target
          }, navItem.title);
        })),
        placement: "top"
      }, /*#__PURE__*/_react["default"].createElement(ButtonIcon, null, this.state.show ? /*#__PURE__*/_react["default"].createElement(OpenChevron, {
        iconSize: 24,
        theme: theme
      }) : /*#__PURE__*/_react["default"].createElement(Chevron, {
        iconSize: 24,
        theme: theme
      })))));
    }
  }]);

  return UserProfile;
}(_react.Component);

exports["default"] = UserProfile;

_defineProperty(UserProfile, "defaultProps", {
  links: [],
  theme: 'light'
});

UserProfile.propTypes = {
  img: _propTypes["default"].string,
  links: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    title: _propTypes["default"].string,
    href: _propTypes["default"].string,
    target: _propTypes["default"].string
  })),
  theme: _propTypes["default"].string,
  username: _propTypes["default"].string,
  description: _propTypes["default"].string
};
