// @spotify-internal/adstudio-tape/lib/components/ApproveReject/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ApproveReject = exports.DEFAULT_REJECTED_VALUE = exports.DEFAULT_APPROVED_VALUE = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Icons = _interopRequireDefault(require("../Icons"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var DEFAULT_APPROVED_VALUE = 'approved';
exports.DEFAULT_APPROVED_VALUE = DEFAULT_APPROVED_VALUE;
var DEFAULT_REJECTED_VALUE = 'rejected';
exports.DEFAULT_REJECTED_VALUE = DEFAULT_REJECTED_VALUE;

var ApproveContainer = _styledComponents["default"].div.withConfig({
  displayName: "ApproveReject__ApproveContainer",
  componentId: "ed94st-0"
})(["display:inline-block;cursor:pointer;", ";"], function (props) {
  return props.disabled && (0, _styledComponents.css)(["cursor:not-allowed;opacity:0.4;"]);
});

var RejectContainer = _styledComponents["default"].div.withConfig({
  displayName: "ApproveReject__RejectContainer",
  componentId: "ed94st-1"
})(["display:inline-block;cursor:pointer;margin-left:", ";", ";"], _encoreFoundation.spacer16, function (props) {
  return props.disabled && (0, _styledComponents.css)(["cursor:not-allowed;opacity:0.4;"]);
});

var ApproveReject = /*#__PURE__*/function (_PureComponent) {
  _inherits(ApproveReject, _PureComponent);

  var _super = _createSuper(ApproveReject);

  function ApproveReject() {
    var _this;

    _classCallCheck(this, ApproveReject);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      approvalHovered: false,
      rejectionHovered: false
    });

    return _this;
  }

  _createClass(ApproveReject, [{
    key: "setApprovalHoverState",
    value: function setApprovalHoverState(hovered) {
      this.setState({
        approvalHovered: hovered
      });
    }
  }, {
    key: "setRejectedHoverState",
    value: function setRejectedHoverState(hovered) {
      this.setState({
        rejectedHovered: hovered
      });
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          value = _this$props.value,
          disabled = _this$props.disabled,
          _this$props$approvedV = _this$props.approvedValue,
          approvedValue = _this$props$approvedV === void 0 ? DEFAULT_APPROVED_VALUE : _this$props$approvedV,
          _this$props$rejectedV = _this$props.rejectedValue,
          rejectedValue = _this$props$rejectedV === void 0 ? DEFAULT_REJECTED_VALUE : _this$props$rejectedV,
          onClickApprove = _this$props.onClickApprove,
          onClickReject = _this$props.onClickReject;
      var approvalHovered = this.state.approvalHovered;
      var rejectedHovered = this.state.rejectedHovered;
      var ApprovalIcon = value === approvedValue ? _Icons["default"].IconApproveAlt : _Icons["default"].IconApprove;
      var RejectionIcon = value === rejectedValue ? _Icons["default"].IconRejectAlt : _Icons["default"].IconReject;
      var approvalColor = value === approvedValue || approvalHovered ? _encoreFoundation.success : _encoreFoundation.gray30;
      var rejectionColor = value === rejectedValue || rejectedHovered ? _encoreFoundation.failure : _encoreFoundation.gray30;

      if (disabled) {
        return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(ApproveContainer, {
          disabled: true,
          "data-test": "approve-button"
        }, /*#__PURE__*/_react["default"].createElement(ApprovalIcon, {
          color: approvalColor
        })), /*#__PURE__*/_react["default"].createElement(RejectContainer, {
          disabled: true,
          "data-test": "reject-button"
        }, /*#__PURE__*/_react["default"].createElement(RejectionIcon, {
          color: rejectionColor
        })));
      }

      return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(ApproveContainer, {
        onMouseEnter: function onMouseEnter() {
          return _this2.setApprovalHoverState(true);
        },
        onMouseLeave: function onMouseLeave() {
          return _this2.setApprovalHoverState(false);
        },
        onClick: onClickApprove,
        "data-test": "approve-button"
      }, /*#__PURE__*/_react["default"].createElement(ApprovalIcon, {
        color: approvalColor
      })), /*#__PURE__*/_react["default"].createElement(RejectContainer, {
        onMouseEnter: function onMouseEnter() {
          return _this2.setRejectedHoverState(true);
        },
        onMouseLeave: function onMouseLeave() {
          return _this2.setRejectedHoverState(false);
        },
        onClick: onClickReject,
        "data-test": "reject-button"
      }, /*#__PURE__*/_react["default"].createElement(RejectionIcon, {
        color: rejectionColor
      })));
    }
  }]);

  return ApproveReject;
}(_react.PureComponent);

exports.ApproveReject = ApproveReject;

_defineProperty(ApproveReject, "propTypes", {
  value: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  approvedValue: _propTypes["default"].string.isRequired,
  rejectedValue: _propTypes["default"].string.isRequired,
  onClickApprove: _propTypes["default"].func,
  onClickReject: _propTypes["default"].func
});

_defineProperty(ApproveReject, "defaultProps", {
  approvedValue: DEFAULT_APPROVED_VALUE,
  rejectedValue: DEFAULT_REJECTED_VALUE
});

var _default = ApproveReject;
exports["default"] = _default;
