// @spotify-internal/targetingservice/proto/com/spotify/targetingservice/proto/service_pb.js
// source: com/spotify/targetingservice/proto/service.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var google_protobuf_empty_pb = require('google-protobuf/google/protobuf/empty_pb.js');
goog.object.extend(proto, google_protobuf_empty_pb);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.Artist', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.ArtistImage', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.ArtistTargetingResponse', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocation', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocation.IdentifierOfCase', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocation.Type', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.PagingAndSorting', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.PagingParameters', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.QueryType', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.SortDirection', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.SortDirection.Type', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.Sorting', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.Sorting.SortField', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.TargetingRequest', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.TargetingResultsResponse', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.TargetingUnitResponse', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest', null, global);
goog.exportSymbol('proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.TargetingResultsResponse.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.TargetingResultsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.TargetingResultsResponse.displayName = 'proto.com.spotify.targetingservice.proto.TargetingResultsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.TargetingUnitResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.TargetingUnitResponse.displayName = 'proto.com.spotify.targetingservice.proto.TargetingUnitResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.TargetingRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.TargetingRequest.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.TargetingRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.TargetingRequest.displayName = 'proto.com.spotify.targetingservice.proto.TargetingRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.displayName = 'proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.PagingParameters = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.PagingParameters, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.PagingParameters.displayName = 'proto.com.spotify.targetingservice.proto.PagingParameters';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.SortDirection = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.SortDirection, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.SortDirection.displayName = 'proto.com.spotify.targetingservice.proto.SortDirection';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.Sorting = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.Sorting, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.Sorting.displayName = 'proto.com.spotify.targetingservice.proto.Sorting';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.PagingAndSorting.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.PagingAndSorting, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.PagingAndSorting.displayName = 'proto.com.spotify.targetingservice.proto.PagingAndSorting';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.displayName = 'proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.displayName = 'proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.GeoLocation = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_);
};
goog.inherits(proto.com.spotify.targetingservice.proto.GeoLocation, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.GeoLocation.displayName = 'proto.com.spotify.targetingservice.proto.GeoLocation';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.displayName = 'proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.displayName = 'proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.ArtistImage = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.ArtistImage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.ArtistImage.displayName = 'proto.com.spotify.targetingservice.proto.ArtistImage';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.Artist = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.Artist.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.Artist, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.Artist.displayName = 'proto.com.spotify.targetingservice.proto.Artist';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.repeatedFields_, null);
};
goog.inherits(proto.com.spotify.targetingservice.proto.ArtistTargetingResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.displayName = 'proto.com.spotify.targetingservice.proto.ArtistTargetingResponse';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.TargetingResultsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    resultsList: jspb.Message.toObjectList(msg.getResultsList(),
    proto.com.spotify.targetingservice.proto.TargetingUnitResponse.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.TargetingResultsResponse;
  return proto.com.spotify.targetingservice.proto.TargetingResultsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.com.spotify.targetingservice.proto.TargetingUnitResponse;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.TargetingUnitResponse.deserializeBinaryFromReader);
      msg.addResults(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.TargetingResultsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getResultsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.com.spotify.targetingservice.proto.TargetingUnitResponse.serializeBinaryToWriter
    );
  }
};


/**
 * repeated TargetingUnitResponse results = 1;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.TargetingUnitResponse>}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.getResultsList = function() {
  return /** @type{!Array<!proto.com.spotify.targetingservice.proto.TargetingUnitResponse>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.spotify.targetingservice.proto.TargetingUnitResponse, 1));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.TargetingUnitResponse>} value
 * @return {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse} returns this
*/
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.setResultsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse}
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.addResults = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.com.spotify.targetingservice.proto.TargetingUnitResponse, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingResultsResponse} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingResultsResponse.prototype.clearResultsList = function() {
  return this.setResultsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.TargetingUnitResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    name: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.TargetingUnitResponse;
  return proto.com.spotify.targetingservice.proto.TargetingUnitResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.TargetingUnitResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string name = 2;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.TargetingUnitResponse} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingUnitResponse.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.TargetingRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.TargetingRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    keywordList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    queryType: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.TargetingRequest;
  return proto.com.spotify.targetingservice.proto.TargetingRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addKeyword(value);
      break;
    case 2:
      var value = /** @type {!proto.com.spotify.targetingservice.proto.QueryType} */ (reader.readEnum());
      msg.setQueryType(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.TargetingRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.TargetingRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getKeywordList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
  f = message.getQueryType();
  if (f !== 0.0) {
    writer.writeEnum(
      2,
      f
    );
  }
};


/**
 * repeated string keyword = 1;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.getKeywordList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.setKeywordList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.addKeyword = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.clearKeywordList = function() {
  return this.setKeywordList([]);
};


/**
 * optional QueryType query_type = 2;
 * @return {!proto.com.spotify.targetingservice.proto.QueryType}
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.getQueryType = function() {
  return /** @type {!proto.com.spotify.targetingservice.proto.QueryType} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.QueryType} value
 * @return {!proto.com.spotify.targetingservice.proto.TargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.TargetingRequest.prototype.setQueryType = function(value) {
  return jspb.Message.setProto3EnumField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.repeatedFields_ = [1,2,3,4,5,6];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.toObject = function(includeInstance, msg) {
  var f, obj = {
    typesList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    geonameIdsList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f,
    isoCodesList: (f = jspb.Message.getRepeatedField(msg, 3)) == null ? undefined : f,
    keywordsList: (f = jspb.Message.getRepeatedField(msg, 4)) == null ? undefined : f,
    geoIdsList: (f = jspb.Message.getRepeatedField(msg, 5)) == null ? undefined : f,
    parentCountryIdsList: (f = jspb.Message.getRepeatedField(msg, 6)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters;
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation.Type>} */ (reader.readPackedEnum());
      msg.setTypesList(value);
      break;
    case 2:
      var value = /** @type {!Array<number>} */ (reader.readPackedInt32());
      msg.setGeonameIdsList(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.addIsoCodes(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.addKeywords(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.addGeoIds(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.addParentCountryIds(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTypesList();
  if (f.length > 0) {
    writer.writePackedEnum(
      1,
      f
    );
  }
  f = message.getGeonameIdsList();
  if (f.length > 0) {
    writer.writePackedInt32(
      2,
      f
    );
  }
  f = message.getIsoCodesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      3,
      f
    );
  }
  f = message.getKeywordsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      4,
      f
    );
  }
  f = message.getGeoIdsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      5,
      f
    );
  }
  f = message.getParentCountryIdsList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      6,
      f
    );
  }
};


/**
 * repeated GeoLocation.Type types = 1;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation.Type>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getTypesList = function() {
  return /** @type {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation.Type>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation.Type>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setTypesList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation.Type} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addTypes = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearTypesList = function() {
  return this.setTypesList([]);
};


/**
 * repeated int32 geoname_ids = 2;
 * @return {!Array<number>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getGeonameIdsList = function() {
  return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<number>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setGeonameIdsList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addGeonameIds = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearGeonameIdsList = function() {
  return this.setGeonameIdsList([]);
};


/**
 * repeated string iso_codes = 3;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getIsoCodesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 3));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setIsoCodesList = function(value) {
  return jspb.Message.setField(this, 3, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addIsoCodes = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 3, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearIsoCodesList = function() {
  return this.setIsoCodesList([]);
};


/**
 * repeated string keywords = 4;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getKeywordsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 4));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setKeywordsList = function(value) {
  return jspb.Message.setField(this, 4, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addKeywords = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 4, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearKeywordsList = function() {
  return this.setKeywordsList([]);
};


/**
 * repeated string geo_ids = 5;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getGeoIdsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 5));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setGeoIdsList = function(value) {
  return jspb.Message.setField(this, 5, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addGeoIds = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 5, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearGeoIdsList = function() {
  return this.setGeoIdsList([]);
};


/**
 * repeated string parent_country_ids = 6;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.getParentCountryIdsList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 6));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.setParentCountryIdsList = function(value) {
  return jspb.Message.setField(this, 6, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.addParentCountryIds = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 6, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.prototype.clearParentCountryIdsList = function() {
  return this.setParentCountryIdsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.PagingParameters.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.PagingParameters} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.PagingParameters.toObject = function(includeInstance, msg) {
  var f, obj = {
    limit: jspb.Message.getFieldWithDefault(msg, 1, 0),
    offset: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.PagingParameters}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.PagingParameters;
  return proto.com.spotify.targetingservice.proto.PagingParameters.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.PagingParameters} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.PagingParameters}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setLimit(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setOffset(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.PagingParameters.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.PagingParameters} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.PagingParameters.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getLimit();
  if (f !== 0) {
    writer.writeInt32(
      1,
      f
    );
  }
  f = message.getOffset();
  if (f !== 0) {
    writer.writeInt32(
      2,
      f
    );
  }
};


/**
 * optional int32 limit = 1;
 * @return {number}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.getLimit = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.targetingservice.proto.PagingParameters} returns this
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.setLimit = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int32 offset = 2;
 * @return {number}
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.getOffset = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.targetingservice.proto.PagingParameters} returns this
 */
proto.com.spotify.targetingservice.proto.PagingParameters.prototype.setOffset = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.SortDirection.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.SortDirection.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.SortDirection} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.SortDirection.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.SortDirection}
 */
proto.com.spotify.targetingservice.proto.SortDirection.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.SortDirection;
  return proto.com.spotify.targetingservice.proto.SortDirection.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.SortDirection} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.SortDirection}
 */
proto.com.spotify.targetingservice.proto.SortDirection.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.SortDirection.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.SortDirection.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.SortDirection} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.SortDirection.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};


/**
 * @enum {number}
 */
proto.com.spotify.targetingservice.proto.SortDirection.Type = {
  UNSET: 0,
  ASC: 1,
  DESC: 2
};




if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.Sorting.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.Sorting} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.Sorting.toObject = function(includeInstance, msg) {
  var f, obj = {
    direction: jspb.Message.getFieldWithDefault(msg, 1, 0),
    field: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.Sorting}
 */
proto.com.spotify.targetingservice.proto.Sorting.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.Sorting;
  return proto.com.spotify.targetingservice.proto.Sorting.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.Sorting} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.Sorting}
 */
proto.com.spotify.targetingservice.proto.Sorting.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.com.spotify.targetingservice.proto.SortDirection.Type} */ (reader.readEnum());
      msg.setDirection(value);
      break;
    case 2:
      var value = /** @type {!proto.com.spotify.targetingservice.proto.Sorting.SortField} */ (reader.readEnum());
      msg.setField(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.Sorting.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.Sorting} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.Sorting.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDirection();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getField();
  if (f !== 0.0) {
    writer.writeEnum(
      2,
      f
    );
  }
};


/**
 * @enum {number}
 */
proto.com.spotify.targetingservice.proto.Sorting.SortField = {
  UNSET: 0,
  NAME: 1,
  GEO_TYPE_SORT_ORDER: 2
};

/**
 * optional SortDirection.Type direction = 1;
 * @return {!proto.com.spotify.targetingservice.proto.SortDirection.Type}
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.getDirection = function() {
  return /** @type {!proto.com.spotify.targetingservice.proto.SortDirection.Type} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.SortDirection.Type} value
 * @return {!proto.com.spotify.targetingservice.proto.Sorting} returns this
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.setDirection = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional SortField field = 2;
 * @return {!proto.com.spotify.targetingservice.proto.Sorting.SortField}
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.getField = function() {
  return /** @type {!proto.com.spotify.targetingservice.proto.Sorting.SortField} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.Sorting.SortField} value
 * @return {!proto.com.spotify.targetingservice.proto.Sorting} returns this
 */
proto.com.spotify.targetingservice.proto.Sorting.prototype.setField = function(value) {
  return jspb.Message.setProto3EnumField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.PagingAndSorting.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.PagingAndSorting} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.toObject = function(includeInstance, msg) {
  var f, obj = {
    parameters: (f = msg.getParameters()) && proto.com.spotify.targetingservice.proto.PagingParameters.toObject(includeInstance, f),
    sortingList: jspb.Message.toObjectList(msg.getSortingList(),
    proto.com.spotify.targetingservice.proto.Sorting.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.PagingAndSorting;
  return proto.com.spotify.targetingservice.proto.PagingAndSorting.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.PagingAndSorting} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.com.spotify.targetingservice.proto.PagingParameters;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.PagingParameters.deserializeBinaryFromReader);
      msg.setParameters(value);
      break;
    case 2:
      var value = new proto.com.spotify.targetingservice.proto.Sorting;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.Sorting.deserializeBinaryFromReader);
      msg.addSorting(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.PagingAndSorting.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.PagingAndSorting} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getParameters();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.com.spotify.targetingservice.proto.PagingParameters.serializeBinaryToWriter
    );
  }
  f = message.getSortingList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.com.spotify.targetingservice.proto.Sorting.serializeBinaryToWriter
    );
  }
};


/**
 * optional PagingParameters parameters = 1;
 * @return {?proto.com.spotify.targetingservice.proto.PagingParameters}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.getParameters = function() {
  return /** @type{?proto.com.spotify.targetingservice.proto.PagingParameters} */ (
    jspb.Message.getWrapperField(this, proto.com.spotify.targetingservice.proto.PagingParameters, 1));
};


/**
 * @param {?proto.com.spotify.targetingservice.proto.PagingParameters|undefined} value
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting} returns this
*/
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.setParameters = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting} returns this
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.clearParameters = function() {
  return this.setParameters(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.hasParameters = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated Sorting sorting = 2;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.Sorting>}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.getSortingList = function() {
  return /** @type{!Array<!proto.com.spotify.targetingservice.proto.Sorting>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.spotify.targetingservice.proto.Sorting, 2));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.Sorting>} value
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting} returns this
*/
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.setSortingList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.Sorting=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.Sorting}
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.addSorting = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.com.spotify.targetingservice.proto.Sorting, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.PagingAndSorting} returns this
 */
proto.com.spotify.targetingservice.proto.PagingAndSorting.prototype.clearSortingList = function() {
  return this.setSortingList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    filters: (f = msg.getFilters()) && proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.toObject(includeInstance, f),
    pagingAndSorting: (f = msg.getPagingAndSorting()) && proto.com.spotify.targetingservice.proto.PagingAndSorting.toObject(includeInstance, f),
    language: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest;
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.deserializeBinaryFromReader);
      msg.setFilters(value);
      break;
    case 2:
      var value = new proto.com.spotify.targetingservice.proto.PagingAndSorting;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.PagingAndSorting.deserializeBinaryFromReader);
      msg.setPagingAndSorting(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setLanguage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getFilters();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters.serializeBinaryToWriter
    );
  }
  f = message.getPagingAndSorting();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.com.spotify.targetingservice.proto.PagingAndSorting.serializeBinaryToWriter
    );
  }
  f = message.getLanguage();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional GeoLocationTargetingFilters filters = 1;
 * @return {?proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.getFilters = function() {
  return /** @type{?proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters} */ (
    jspb.Message.getWrapperField(this, proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters, 1));
};


/**
 * @param {?proto.com.spotify.targetingservice.proto.GeoLocationTargetingFilters|undefined} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} returns this
*/
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.setFilters = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.clearFilters = function() {
  return this.setFilters(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.hasFilters = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional PagingAndSorting paging_and_sorting = 2;
 * @return {?proto.com.spotify.targetingservice.proto.PagingAndSorting}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.getPagingAndSorting = function() {
  return /** @type{?proto.com.spotify.targetingservice.proto.PagingAndSorting} */ (
    jspb.Message.getWrapperField(this, proto.com.spotify.targetingservice.proto.PagingAndSorting, 2));
};


/**
 * @param {?proto.com.spotify.targetingservice.proto.PagingAndSorting|undefined} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} returns this
*/
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.setPagingAndSorting = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.clearPagingAndSorting = function() {
  return this.setPagingAndSorting(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.hasPagingAndSorting = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string language = 3;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.getLanguage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingRequest.prototype.setLanguage = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    resultsList: jspb.Message.toObjectList(msg.getResultsList(),
    proto.com.spotify.targetingservice.proto.GeoLocation.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse;
  return proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.com.spotify.targetingservice.proto.GeoLocation;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.GeoLocation.deserializeBinaryFromReader);
      msg.addResults(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getResultsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.com.spotify.targetingservice.proto.GeoLocation.serializeBinaryToWriter
    );
  }
};


/**
 * repeated GeoLocation results = 1;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation>}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.getResultsList = function() {
  return /** @type{!Array<!proto.com.spotify.targetingservice.proto.GeoLocation>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.spotify.targetingservice.proto.GeoLocation, 1));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.GeoLocation>} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse} returns this
*/
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.setResultsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation}
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.addResults = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.com.spotify.targetingservice.proto.GeoLocation, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocationTargetingResponse.prototype.clearResultsList = function() {
  return this.setResultsList([]);
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_ = [[4,5]];

/**
 * @enum {number}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.IdentifierOfCase = {
  IDENTIFIER_OF_NOT_SET: 0,
  ID: 4,
  ISO_CODE: 5
};

/**
 * @return {proto.com.spotify.targetingservice.proto.GeoLocation.IdentifierOfCase}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getIdentifierOfCase = function() {
  return /** @type {proto.com.spotify.targetingservice.proto.GeoLocation.IdentifierOfCase} */(jspb.Message.computeOneofCase(this, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.GeoLocation.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocation.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    type: jspb.Message.getFieldWithDefault(msg, 2, 0),
    countryCode: jspb.Message.getFieldWithDefault(msg, 3, ""),
    id: jspb.Message.getFieldWithDefault(msg, 4, 0),
    isoCode: jspb.Message.getFieldWithDefault(msg, 5, ""),
    geoId: jspb.Message.getFieldWithDefault(msg, 6, ""),
    typeDisplayName: jspb.Message.getFieldWithDefault(msg, 7, ""),
    parentName: jspb.Message.getFieldWithDefault(msg, 8, ""),
    searchable: jspb.Message.getBooleanFieldWithDefault(msg, 9, false),
    targetableByAdserving: jspb.Message.getBooleanFieldWithDefault(msg, 10, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.GeoLocation;
  return proto.com.spotify.targetingservice.proto.GeoLocation.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {!proto.com.spotify.targetingservice.proto.GeoLocation.Type} */ (reader.readEnum());
      msg.setType(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCountryCode(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setId(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setIsoCode(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setGeoId(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setTypeDisplayName(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setParentName(value);
      break;
    case 9:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setSearchable(value);
      break;
    case 10:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setTargetableByAdserving(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.GeoLocation.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.GeoLocation.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getType();
  if (f !== 0.0) {
    writer.writeEnum(
      2,
      f
    );
  }
  f = message.getCountryCode();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeInt32(
      4,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 5));
  if (f != null) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getGeoId();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getTypeDisplayName();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getParentName();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
  f = message.getSearchable();
  if (f) {
    writer.writeBool(
      9,
      f
    );
  }
  f = message.getTargetableByAdserving();
  if (f) {
    writer.writeBool(
      10,
      f
    );
  }
};


/**
 * @enum {number}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.Type = {
  UNSET: 0,
  CITY: 1,
  DMA_REGION: 2,
  COUNTRY: 3,
  REGION: 4,
  POSTAL_CODE: 5,
  COUNTY: 6
};

/**
 * optional string name = 1;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional Type type = 2;
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation.Type}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getType = function() {
  return /** @type {!proto.com.spotify.targetingservice.proto.GeoLocation.Type} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.GeoLocation.Type} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setType = function(value) {
  return jspb.Message.setProto3EnumField(this, 2, value);
};


/**
 * optional string country_code = 3;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getCountryCode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setCountryCode = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional int32 id = 4;
 * @return {number}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getId = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setId = function(value) {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.clearId = function() {
  return jspb.Message.setOneofField(this, 4, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.hasId = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional string iso_code = 5;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getIsoCode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setIsoCode = function(value) {
  return jspb.Message.setOneofField(this, 5, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.clearIsoCode = function() {
  return jspb.Message.setOneofField(this, 5, proto.com.spotify.targetingservice.proto.GeoLocation.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.hasIsoCode = function() {
  return jspb.Message.getField(this, 5) != null;
};


/**
 * optional string geo_id = 6;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getGeoId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setGeoId = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string type_display_name = 7;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getTypeDisplayName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setTypeDisplayName = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional string parent_name = 8;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getParentName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setParentName = function(value) {
  return jspb.Message.setProto3StringField(this, 8, value);
};


/**
 * optional bool searchable = 9;
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getSearchable = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 9, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setSearchable = function(value) {
  return jspb.Message.setProto3BooleanField(this, 9, value);
};


/**
 * optional bool targetable_by_adserving = 10;
 * @return {boolean}
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.getTargetableByAdserving = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 10, false));
};


/**
 * @param {boolean} value
 * @return {!proto.com.spotify.targetingservice.proto.GeoLocation} returns this
 */
proto.com.spotify.targetingservice.proto.GeoLocation.prototype.setTargetableByAdserving = function(value) {
  return jspb.Message.setProto3BooleanField(this, 10, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    countryCode: jspb.Message.getFieldWithDefault(msg, 1, ""),
    postalCodesList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest;
  return proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCountryCode(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addPostalCodes(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCountryCode();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPostalCodesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional string country_code = 1;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.getCountryCode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.setCountryCode = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated string postal_codes = 2;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.getPostalCodesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.setPostalCodesList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.addPostalCodes = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeRequest.prototype.clearPostalCodesList = function() {
  return this.setPostalCodesList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.repeatedFields_ = [1,2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    validPostalCodeListList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    invalidPostalCodeListList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse;
  return proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addValidPostalCodeList(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addInvalidPostalCodeList(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getValidPostalCodeListList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
  f = message.getInvalidPostalCodeListList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * repeated string valid_postal_code_list = 1;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.getValidPostalCodeListList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.setValidPostalCodeListList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.addValidPostalCodeList = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.clearValidPostalCodeListList = function() {
  return this.setValidPostalCodeListList([]);
};


/**
 * repeated string invalid_postal_code_list = 2;
 * @return {!Array<string>}
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.getInvalidPostalCodeListList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.setInvalidPostalCodeListList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.addInvalidPostalCodeList = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ValidatePostalCodeResponse.prototype.clearInvalidPostalCodeListList = function() {
  return this.setInvalidPostalCodeListList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.ArtistImage.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.ArtistImage} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ArtistImage.toObject = function(includeInstance, msg) {
  var f, obj = {
    width: jspb.Message.getFieldWithDefault(msg, 1, 0),
    height: jspb.Message.getFieldWithDefault(msg, 2, 0),
    url: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.ArtistImage;
  return proto.com.spotify.targetingservice.proto.ArtistImage.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.ArtistImage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setWidth(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setHeight(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setUrl(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.ArtistImage.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.ArtistImage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ArtistImage.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getWidth();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getHeight();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
  f = message.getUrl();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional int64 width = 1;
 * @return {number}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.getWidth = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage} returns this
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.setWidth = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int64 height = 2;
 * @return {number}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.getHeight = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage} returns this
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.setHeight = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional string url = 3;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.getUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage} returns this
 */
proto.com.spotify.targetingservice.proto.ArtistImage.prototype.setUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.Artist.repeatedFields_ = [3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.Artist.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.Artist} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.Artist.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    name: jspb.Message.getFieldWithDefault(msg, 2, ""),
    imagesList: jspb.Message.toObjectList(msg.getImagesList(),
    proto.com.spotify.targetingservice.proto.ArtistImage.toObject, includeInstance),
    spotifyUrl: jspb.Message.getFieldWithDefault(msg, 4, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.Artist}
 */
proto.com.spotify.targetingservice.proto.Artist.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.Artist;
  return proto.com.spotify.targetingservice.proto.Artist.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.Artist} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.Artist}
 */
proto.com.spotify.targetingservice.proto.Artist.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 3:
      var value = new proto.com.spotify.targetingservice.proto.ArtistImage;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.ArtistImage.deserializeBinaryFromReader);
      msg.addImages(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setSpotifyUrl(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.Artist.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.Artist} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.Artist.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getImagesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      proto.com.spotify.targetingservice.proto.ArtistImage.serializeBinaryToWriter
    );
  }
  f = message.getSpotifyUrl();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.Artist} returns this
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string name = 2;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.Artist} returns this
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * repeated ArtistImage images = 3;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.ArtistImage>}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.getImagesList = function() {
  return /** @type{!Array<!proto.com.spotify.targetingservice.proto.ArtistImage>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.spotify.targetingservice.proto.ArtistImage, 3));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.ArtistImage>} value
 * @return {!proto.com.spotify.targetingservice.proto.Artist} returns this
*/
proto.com.spotify.targetingservice.proto.Artist.prototype.setImagesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.ArtistImage=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.ArtistImage}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.addImages = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.com.spotify.targetingservice.proto.ArtistImage, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.Artist} returns this
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.clearImagesList = function() {
  return this.setImagesList([]);
};


/**
 * optional string spotify_url = 4;
 * @return {string}
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.getSpotifyUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.com.spotify.targetingservice.proto.Artist} returns this
 */
proto.com.spotify.targetingservice.proto.Artist.prototype.setSpotifyUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    resultsList: jspb.Message.toObjectList(msg.getResultsList(),
    proto.com.spotify.targetingservice.proto.Artist.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.com.spotify.targetingservice.proto.ArtistTargetingResponse;
  return proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 2:
      var value = new proto.com.spotify.targetingservice.proto.Artist;
      reader.readMessage(value,proto.com.spotify.targetingservice.proto.Artist.deserializeBinaryFromReader);
      msg.addResults(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getResultsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.com.spotify.targetingservice.proto.Artist.serializeBinaryToWriter
    );
  }
};


/**
 * repeated Artist results = 2;
 * @return {!Array<!proto.com.spotify.targetingservice.proto.Artist>}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.getResultsList = function() {
  return /** @type{!Array<!proto.com.spotify.targetingservice.proto.Artist>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.com.spotify.targetingservice.proto.Artist, 2));
};


/**
 * @param {!Array<!proto.com.spotify.targetingservice.proto.Artist>} value
 * @return {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse} returns this
*/
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.setResultsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.com.spotify.targetingservice.proto.Artist=} opt_value
 * @param {number=} opt_index
 * @return {!proto.com.spotify.targetingservice.proto.Artist}
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.addResults = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.com.spotify.targetingservice.proto.Artist, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.com.spotify.targetingservice.proto.ArtistTargetingResponse} returns this
 */
proto.com.spotify.targetingservice.proto.ArtistTargetingResponse.prototype.clearResultsList = function() {
  return this.setResultsList([]);
};


/**
 * @enum {number}
 */
proto.com.spotify.targetingservice.proto.QueryType = {
  ID: 0,
  NAME: 1
};

goog.object.extend(exports, proto.com.spotify.targetingservice.proto);
