// @spotify-internal/adstudio-tape/lib/components/Navigation/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Navigation;
exports.NavigationFooter = exports.NavigationHeader = exports.NavigationBody = exports.NavigationNav = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _encoreFoundation = require("@spotify-internal/encore-foundation");

var _SpotifyLogo = _interopRequireDefault(require("../SpotifyLogo"));

var _tokens = require("../../styles/tokens");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var NavigationNav = _styledComponents["default"].nav.withConfig({
  displayName: "Navigation__NavigationNav",
  componentId: "sc-1p54o6n-0"
})(["background-color:", ";color:", ";display:flex;flex-direction:column;height:100vh;padding-top:", ";position:relative;width:", ";"], _encoreFoundation.gray10, _encoreFoundation.gray70, _tokens.sidebarBasePadding, _tokens.sidebarBaseWidth);

exports.NavigationNav = NavigationNav;

var NavigationBody = _styledComponents["default"].section.withConfig({
  displayName: "Navigation__NavigationBody",
  componentId: "sc-1p54o6n-1"
})(["flex:1 0 auto;padding:", " 0;"], _tokens.sidebarBasePadding);

exports.NavigationBody = NavigationBody;

var NavigationHeader = _styledComponents["default"].header.withConfig({
  displayName: "Navigation__NavigationHeader",
  componentId: "sc-1p54o6n-2"
})(["padding:0 ", ";"], _tokens.sidebarBasePadding);

exports.NavigationHeader = NavigationHeader;

var NavigationFooter = _styledComponents["default"].footer.withConfig({
  displayName: "Navigation__NavigationFooter",
  componentId: "sc-1p54o6n-3"
})(["border-top:1px solid ", ";margin:", ";padding-top:", ";"], _encoreFoundation.gray30, _tokens.sidebarBasePadding, _tokens.sidebarBasePadding);

exports.NavigationFooter = NavigationFooter;

function Navigation(_ref) {
  var body = _ref.body,
      footer = _ref.footer,
      header = _ref.header,
      props = _objectWithoutProperties(_ref, ["body", "footer", "header"]);

  return /*#__PURE__*/_react["default"].createElement(NavigationNav, props, /*#__PURE__*/_react["default"].createElement(NavigationHeader, null, header), /*#__PURE__*/_react["default"].createElement(NavigationBody, null, body), footer && /*#__PURE__*/_react["default"].createElement(NavigationFooter, null, footer));
}

Navigation.propTypes = {
  header: _propTypes["default"].node,
  body: _propTypes["default"].node,
  footer: _propTypes["default"].node
};
Navigation.defaultProps = {
  header: /*#__PURE__*/_react["default"].createElement(_SpotifyLogo["default"], null)
}; // these are used to override styles if necessary.

Navigation.NavigationNav = NavigationNav;
Navigation.NavigationBody = NavigationBody;
Navigation.NavigationHeader = NavigationHeader;
Navigation.NavigationFooter = NavigationFooter;
Navigation.sidebarBasePadding = _tokens.sidebarBasePadding;
Navigation.sidebarBaseWidth = _tokens.sidebarBaseWidth;
